--- ./src/lib/WPXContentListener.cpp.orig	2018-12-25 10:19:19.000000000 -0600
+++ ./src/lib/WPXContentListener.cpp	2021-08-06 00:05:53.939363381 -0500
@@ -23,6 +23,7 @@
  * Corel Corporation or Corel Corporation Limited."
  */
 
+#include <cstddef>
 #include "WPXContentListener.h"
 #include "WPXPageSpan.h"
 #include "libwpd_internal.h"
--- ./src/lib/WPXTable.cpp.orig	2018-12-25 10:19:19.000000000 -0600
+++ ./src/lib/WPXTable.cpp	2021-08-06 00:05:19.424365290 -0500
@@ -25,6 +25,8 @@
  * Corel Corporation or Corel Corporation Limited."
  */
 
+#include <cstddef>
+
 #include "WPXTable.h"
 
 #include "libwpd_internal.h"
