diff -rupN openjpeg-2.1.1/CMakeLists.txt openjpeg-2.1.1-new/CMakeLists.txt
--- openjpeg-2.1.1/CMakeLists.txt	2016-07-05 16:54:17.000000000 +0200
+++ openjpeg-2.1.1-new/CMakeLists.txt	2016-07-06 09:38:26.083029127 +0200
@@ -270,7 +270,6 @@ if(BUILD_CODEC OR BUILD_MJ2)
   # OFF: It will only build 3rd party libs if they are not found on the system
   # ON: 3rd party libs will ALWAYS be build, and used
   option(BUILD_THIRDPARTY "Build the thirdparty executables if it is needed" OFF)
-  add_subdirectory(thirdparty)
   add_subdirectory(src/bin)
 endif ()
 add_subdirectory(wrapping)
