diff -Nur polkit-gnome-0.95.orig//src/polkit-gnome-authentication-agent-1.desktop.in.in polkit-gnome-0.95/src/polkit-gnome-authentication-agent-1.desktop.in.in
--- polkit-gnome-0.95.orig//src/polkit-gnome-authentication-agent-1.desktop.in.in	2009-07-20 18:28:31.000000000 -0500
+++ polkit-gnome-0.95/src/polkit-gnome-authentication-agent-1.desktop.in.in	2010-05-03 01:03:47.021291164 -0500
@@ -5,4 +5,4 @@
 Terminal=false
 Type=Application
 Categories=
-OnlyShowIn=GNOME;
+NotShowIn=KDE;
