From af2fe0d1dcbcd982d84ddf2bbd174afe90976ed9 Mon Sep 17 00:00:00 2001
From: Harald Sitter <sitter@kde.org>
Date: Tue, 30 Jan 2024 11:24:41 +0100
Subject: xdg-icon-resource: unbreak syntax by removing stray grave accent

---
 scripts/xdg-icon-resource.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/xdg-icon-resource.in b/scripts/xdg-icon-resource.in
index 0bfa26b..47ba6d6 100644
--- a/scripts/xdg-icon-resource.in
+++ b/scripts/xdg-icon-resource.in
@@ -306,7 +306,7 @@ fi
 need_kde_icon_path()
 {
   local path
-  path="$(xdg_realpath "$1")" 2> /dev/null` # Normalize path
+  path="$(xdg_realpath "$1")" 2> /dev/null # Normalize path
   DEBUG 2 "need_kde_icon_path $path"
   if [ -z "$path" ] ; then
      DEBUG 2 "need_kde_icon_path RETURN 1 (not needed, no xdg icon dir)"
-- 
cgit v1.2.3
