--- ./scripts/xdg-open.in.orig	2024-02-03 12:40:07.000000000 -0600
+++ ./scripts/xdg-open.in	2024-02-03 12:45:57.900659000 -0600
@@ -533,11 +533,6 @@
         local file
         file="$(file_url_to_path "$1")"
 
-        # handle relative paths
-        if ! echo "$file" | grep -q ^/; then
-            file="$(pwd)/$file"
-        fi
-
         pcmanfm "$file"
     else
         open_generic "$1"
