--- ./scripts/xdg-open.in.orig	2018-05-10 10:02:31.000000000 -0500
+++ ./scripts/xdg-open.in	2019-01-25 13:01:17.492733618 -0600
@@ -453,12 +453,6 @@
     # pcmanfm only knows how to handle file:// urls and filepaths, it seems.
     if pcmanfm --help >/dev/null 2>&1 -a is_file_url_or_path "$1"; then
         local file="$(file_url_to_path "$1")"
-
-        # handle relative paths
-        if ! echo "$file" | grep -q ^/; then
-            file="$(pwd)/$file"
-        fi
-
         pcmanfm "$file"
     else
         open_generic "$1"
