--- ./localedata/Makefile.orig	2003-11-20 15:31:38.000000000 -0800
+++ ./localedata/Makefile	2004-08-03 17:20:54.000000000 -0700
@@ -222,6 +222,7 @@
 	echo -n '...'; \
 	input=`echo $$locale | sed 's/\([^.]*\)[^@]*\(.*\)/\1\2/'`; \
 	$(LOCALEDEF) --alias-file=../intl/locale.alias \
+		     --no-archive \
 		     -i locales/$$input -c -f charmaps/$$charset \
 		     $(addprefix --prefix=,$(install_root)) $$locale; \
 	echo ' done'; \
