--- ./setuptools/_distutils/sysconfig.py.orig	2022-08-21 02:05:13.000000000 -0500
+++ ./setuptools/_distutils/sysconfig.py	2022-08-21 13:33:52.738969315 -0500
@@ -234,7 +234,7 @@
             libdir = getattr(sys, "platlibdir", "lib")
         else:
             # Pure Python
-            libdir = "lib"
+            libdir = "lib64"
         implementation = 'pypy' if IS_PYPY else 'python'
         libpython = os.path.join(prefix, libdir, implementation + get_python_version())
         return _posix_lib(standard_lib, libpython, early_prefix, prefix)
--- ./setuptools/_distutils/command/install.py.orig	2022-08-21 13:33:52.738969315 -0500
+++ ./setuptools/_distutils/command/install.py	2022-08-21 13:35:11.919964935 -0500
@@ -34,7 +34,7 @@
 
 INSTALL_SCHEMES = {
     'posix_prefix': {
-        'purelib': '{base}/lib/{implementation_lower}{py_version_short}/site-packages',
+        'purelib': '{base}/lib64/{implementation_lower}{py_version_short}/site-packages',
         'platlib': '{platbase}/{platlibdir}/{implementation_lower}'
         '{py_version_short}/site-packages',
         'headers': '{base}/include/{implementation_lower}'
@@ -43,7 +43,7 @@
         'data': '{base}',
     },
     'posix_home': {
-        'purelib': '{base}/lib/{implementation_lower}',
+        'purelib': '{base}/lib64/{implementation_lower}',
         'platlib': '{base}/{platlibdir}/{implementation_lower}',
         'headers': '{base}/include/{implementation_lower}/{dist_name}',
         'scripts': '{base}/bin',
