--- ./CL/cl_version.h.orig	2019-01-17 12:40:56.000000000 -0600
+++ ./CL/cl_version.h	2019-02-06 15:22:44.022548935 -0600
@@ -31,7 +31,7 @@
 
 /* Detect which version to target */
 #if !defined(CL_TARGET_OPENCL_VERSION)
-#pragma message("cl_version.h: CL_TARGET_OPENCL_VERSION is not defined. Defaulting to 220 (OpenCL 2.2)")
+/* #pragma message("cl_version.h: CL_TARGET_OPENCL_VERSION is not defined. Defaulting to 220 (OpenCL 2.2)") */
 #define CL_TARGET_OPENCL_VERSION 220
 #endif
 #if CL_TARGET_OPENCL_VERSION != 100 && \
