--- ./src/shared/sleep-config.c.orig	2024-04-16 02:21:44.000000000 -0500
+++ ./src/shared/sleep-config.c	2024-04-22 13:31:58.033078814 -0500
@@ -45,7 +45,7 @@
 #if 0 /// elogind supports suspend modes (deep s2idle) so we need defaults, too
         /* Not used by SLEEP_SUSPEND */
 #else // 0
-        [SLEEP_SUSPEND]      = STRV_MAKE("s2idle", "deep"),
+        [SLEEP_SUSPEND]      = STRV_MAKE("deep", "s2idle"),
 #endif // 0
         [SLEEP_HIBERNATE]    = STRV_MAKE("platform", "shutdown"),
         [SLEEP_HYBRID_SLEEP] = STRV_MAKE("suspend"),
