From e8adc2fbf697bd683eb538585d0645e4541d05f1 Mon Sep 17 00:00:00 2001
From: Shi Pujin <shipujin.t@gmail.com>
Date: Mon, 11 Jul 2022 02:02:27 +0000
Subject: [PATCH] elogind add loongarch64 support


diff --git a/src/basic/architecture.h b/src/basic/architecture.h
index 620b522..ef4d2dc 100644
--- a/src/basic/architecture.h
+++ b/src/basic/architecture.h
@@ -42,6 +42,7 @@ enum {
         ARCHITECTURE_NIOS2,
         ARCHITECTURE_RISCV32,
         ARCHITECTURE_RISCV64,
+        ARCHITECTURE_LOONGARCH64,
         ARCHITECTURE_ARC,
         ARCHITECTURE_ARC_BE,
         _ARCHITECTURE_MAX,
@@ -223,6 +224,9 @@ int uname_architecture(void);
 #  else
 #    error "Unrecognized riscv architecture variant"
 #  endif
+#elif defined(__loongarch64)
+#    define native_architecture() ARCHITECTURE_LOONGARCH64
+#    define LIB_ARCH_TUPLE "loongarch64-linux-gnu"
 #elif defined(__arc__)
 #  if __BYTE_ORDER == __BIG_ENDIAN
 #    define native_architecture() ARCHITECTURE_ARC_BE
-- 
2.37.0

