--- ./src/login/10-elogind.conf.orig	2025-01-13 09:07:01.000000000 -0600
+++ ./src/login/10-elogind.conf	2025-01-17 11:44:28.470397665 -0600
@@ -9,7 +9,7 @@
 # Local configuration should be created here with a higher leading number so
 # they are parsed later overriding the defaults.
 #
-# See sleep.conf(5) for details.
+# See logind.conf(5) for details.
 
 [Login]
 # elogind is not a system manager, so it does not remove IPC-items by default
