--- ./setup.py.orig	2018-12-10 00:13:32.000000000 -0600
+++ ./setup.py	2019-04-28 13:11:33.836514225 -0500
@@ -44,7 +44,7 @@
 requires = [
     'chardet>=3.0.2,<3.1.0',
     'idna>=2.5,<2.9',
-    'urllib3>=1.21.1,<1.25',
+    'urllib3>=1.21.1,<1.26',
     'certifi>=2017.4.17'
 
 ]
--- ./requests/__init__.py.orig	2018-12-10 00:13:32.000000000 -0600
+++ ./requests/__init__.py	2019-04-28 13:13:44.711502000 -0500
@@ -57,10 +57,10 @@
     # Check urllib3 for compatibility.
     major, minor, patch = urllib3_version  # noqa: F811
     major, minor, patch = int(major), int(minor), int(patch)
-    # urllib3 >= 1.21.1, <= 1.24
+    # urllib3 >= 1.21.1, < 1.26
     assert major == 1
     assert minor >= 21
-    assert minor <= 24
+    assert minor < 26
 
     # Check chardet for compatibility.
     major, minor, patch = chardet_version.split('.')[:3]
