--- ./fortune/datfiles/fortunes2.orig	2001-08-29 10:11:30.000000000 -0500
+++ ./fortune/datfiles/fortunes2	2012-03-02 10:00:47.753999471 -0600
@@ -35689,7 +35689,7 @@
 order to get power we would have to become very much like them.  (Lenin's
 fatal mistake, both in theory and in practice.)
 %
-Perhaps the world's second words crime is boredom.  The first is
+Perhaps the world's second worst crime is boredom.  The first is
 being a bore.
 		-- Cecil Beaton
 %
