--- ./misc/terminfo.src.orig	2018-10-27 18:10:56.000000000 -0500
+++ ./misc/terminfo.src	2018-11-06 18:51:34.708372527 -0600
@@ -941,7 +941,7 @@
 # themselves; this entry assumes that capability.
 #
 linux-basic|linux console,
-	am, bce, eo, mir, msgr, xenl, xon,
+	am, eo, mir, msgr, xenl, xon,
 	it#8, ncv#18, U8#1,
 	acsc=+\020\,\021-\030.^Y0\333`\004a\261f\370g\361h\260i
 	     \316j\331k\277l\332m\300n\305o~p\304q\304r\304s_t\303u
