--- ./include/MKhashsize.sh.orig	2006-10-21 19:45:39.000000000 -0500
+++ ./include/MKhashsize.sh	2006-12-27 17:21:06.000000000 -0600
@@ -35,7 +35,7 @@
 echo " */"
 
 CAPS="${1-Caps}"
-TABSIZE=`grep -v '^[ #]' $CAPS | grep -v "^$" | grep -v "^capalias"| grep -v "^infoalias" | wc -l`
+TABSIZE=$(grep -v '^[ #]' $CAPS | grep -v "^$" | grep -v "^capalias"| grep -v "^infoalias" | wc -l)
 
 echo ""
 echo "#define CAPTABSIZE	${TABSIZE}"
