From 3acecbbab8ecaf3e3b324a2286e51cf9d7950ad5 Mon Sep 17 00:00:00 2001
From: Jean Delvare <jdelvare@suse.de>
Date: Tue, 20 Oct 2015 08:47:15 +0200
Subject: [PATCH 4/9] dmidecode: Handle OEM-specific types in group
 associations

---
 CHANGELOG   | 5 +++++
 dmidecode.c | 2 ++
 2 files changed, 7 insertions(+)

diff --git a/CHANGELOG b/CHANGELOG
index c940c9f..2aa1082 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -1,3 +1,8 @@
+2015-10-20  Jean Delvare  <jdelvare@suse.de>
+
+	* dmidecode.c: Handle OEM-specific types in group associations
+	  (DMI type 14).
+
 2015-10-14  Jean Delvare  <jdelvare@suse.de>
 
 	* util.c: Avoid SIGBUS on mmap failure.
diff --git a/dmidecode.c b/dmidecode.c
index f41c85b..ce0511b 100644
--- a/dmidecode.c
+++ b/dmidecode.c
@@ -172,6 +172,8 @@ static const char *dmi_smbios_structure_type(u8 code)
 		"Management Controller Host Interface", /* 42 */
 	};
 
+	if (code >= 128)
+		return "OEM-specific";
 	if (code <= 42)
 		return type[code];
 	return out_of_spec;
-- 
2.6.4

