diff -Nur libtool-2.4.3.orig/build-aux/ltmain.in libtool-2.4.3/build-aux/ltmain.in
--- libtool-2.4.3.orig/build-aux/ltmain.in	2014-10-27 12:20:51.000000000 -0500
+++ libtool-2.4.3/build-aux/ltmain.in	2014-10-28 11:01:00.731257807 -0500
@@ -5978,7 +5978,7 @@
 	# Find the relevant object directory and library name.
 	if test yes = "$installed"; then
 	  if test ! -f "$lt_sysroot$libdir/$linklib" && test -f "$abs_ladir/$linklib"; then
-	    func_warning "library '$lib' was moved."
+	    #func_warning "library '$lib' was moved."
 	    dir=$ladir
 	    absdir=$abs_ladir
 	    libdir=$abs_ladir
@@ -6546,7 +6546,7 @@
 		  test -z "$libdir" && \
 		    func_fatal_error "'$deplib' is not a valid libtool archive"
 		  test "$absdir" != "$libdir" && \
-		    func_warning "'$deplib' seems to be moved"
+		    #func_warning "'$deplib' seems to be moved"
 
 		  path=-L$absdir
 		fi
diff -Nur libtool-2.4.3.orig/build-aux/ltmain.sh libtool-2.4.3/build-aux/ltmain.sh
--- libtool-2.4.3.orig/build-aux/ltmain.sh	2014-10-27 13:04:31.000000000 -0500
+++ libtool-2.4.3/build-aux/ltmain.sh	2014-10-28 11:01:14.196484821 -0500
@@ -7890,7 +7890,7 @@
 	# Find the relevant object directory and library name.
 	if test yes = "$installed"; then
 	  if test ! -f "$lt_sysroot$libdir/$linklib" && test -f "$abs_ladir/$linklib"; then
-	    func_warning "library '$lib' was moved."
+	    #func_warning "library '$lib' was moved."
 	    dir=$ladir
 	    absdir=$abs_ladir
 	    libdir=$abs_ladir
@@ -8458,7 +8458,7 @@
 		  test -z "$libdir" && \
 		    func_fatal_error "'$deplib' is not a valid libtool archive"
 		  test "$absdir" != "$libdir" && \
-		    func_warning "'$deplib' seems to be moved"
+		    #func_warning "'$deplib' seems to be moved"
 
 		  path=-L$absdir
 		fi
