diff -Nur libtool-1.5.26.orig/libltdl/ltmain.sh libtool-1.5.26/libltdl/ltmain.sh
--- libtool-1.5.26.orig/libltdl/ltmain.sh	2008-02-01 10:39:51.000000000 -0600
+++ libtool-1.5.26/libltdl/ltmain.sh	2008-10-01 13:59:24.654020673 -0500
@@ -2952,9 +2952,9 @@
 		    $echo "$modename: \`$deplib' is not a valid libtool archive" 1>&2
 		    exit $EXIT_FAILURE
 		  fi
-		  if test "$absdir" != "$libdir"; then
-		    $echo "$modename: warning: \`$deplib' seems to be moved" 1>&2
-		  fi
+		  #if test "$absdir" != "$libdir"; then
+		  #  $echo "$modename: warning: \`$deplib' seems to be moved" 1>&2
+		  #fi
 		  path="$absdir"
 		fi
 		depdepl=
diff -Nur libtool-1.5.26.orig/ltmain.sh libtool-1.5.26/ltmain.sh
--- libtool-1.5.26.orig/ltmain.sh	2008-02-01 10:39:51.000000000 -0600
+++ libtool-1.5.26/ltmain.sh	2008-10-01 13:59:24.690024584 -0500
@@ -2952,9 +2952,9 @@
 		    $echo "$modename: \`$deplib' is not a valid libtool archive" 1>&2
 		    exit $EXIT_FAILURE
 		  fi
-		  if test "$absdir" != "$libdir"; then
-		    $echo "$modename: warning: \`$deplib' seems to be moved" 1>&2
-		  fi
+		  #if test "$absdir" != "$libdir"; then
+		  #  $echo "$modename: warning: \`$deplib' seems to be moved" 1>&2
+		  #fi
 		  path="$absdir"
 		fi
 		depdepl=
