--- ./php.ini-development.orig	2023-02-13 19:06:39.000000000 -0600
+++ ./php.ini-development	2023-03-11 13:29:08.667004041 -0600
@@ -905,7 +905,49 @@
 ; 'extension='php_<ext>.dll') is supported for legacy reasons and may be
 ; deprecated in a future PHP major version. So, when it is possible, please
 ; move to the new ('extension=<ext>) syntax.
+
+; These are the extensions that are available for use on this PHP build.
+; If you do NOT need any of these extensions, you should comment them out.
+; That will probably increase both your security and performance.
 ;
+extension=bcmath
+extension=bz2
+extension=calendar
+extension=ctype
+extension=curl
+extension=dba
+extension=dom
+extension=enchant
+extension=exif
+extension=ftp
+extension=gd
+extension=gettext
+extension=gmp
+extension=iconv
+extension=intl
+extension=ldap
+extension=mbstring
+extension=mysqlnd
+extension=mysqli
+extension=odbc
+extension=openssl
+zend_extension=opcache
+extension=pdo
+extension=pdo_mysql
+extension=pdo_sqlite
+extension=pdo_odbc
+extension=pspell
+extension=shmop
+extension=snmp
+extension=soap
+extension=sqlite3
+extension=tidy
+extension=tokenizer
+extension=xmlreader
+extension=xsl
+extension=zip
+extension=zlib
+
 ; Notes for Windows environments :
 ;
 ; - Many DLL files are located in the ext/
@@ -1363,7 +1405,7 @@
 ; where MODE is the octal representation of the mode. Note that this
 ; does not overwrite the process's umask.
 ; https://php.net/session.save-path
-;session.save_path = "/tmp"
+session.save_path = "/var/lib/php"
 
 ; Whether to use strict session mode.
 ; Strict session mode does not accept an uninitialized session ID, and
