--- ./ext/imap/php_imap.c.orig	2020-03-17 05:40:25.000000000 -0500
+++ ./ext/imap/php_imap.c	2020-03-17 16:20:15.600685744 -0500
@@ -5081,7 +5081,7 @@
 	   php debugger? */
 }
 
-PHP_IMAP_EXPORT void mm_login(NETMBX *mb, char *user, char *pwd, long trial)
+PHP_IMAP_EXPORT void mm_login(NETMBX *mb, char *user, char **pwd, long trial)
 {
 
 	if (*mb->user) {
@@ -5089,7 +5089,13 @@
 	} else {
 		strlcpy (user, IMAPG(imap_user), MAILTMPLEN);
 	}
-	strlcpy (pwd, IMAPG(imap_password), MAILTMPLEN);
+	*pwd = cpystr(IMAPG(imap_password));	/* alpine-2.22 c-client.a */
+}
+
+/* alpine-2.22 c-client.a workaround */
+PHP_IMAP_EXPORT void mm_login_method(NETMBX *mb, char *user, void *pwd, long trial, char *method)
+{
+   pwd = NULL;
 }
 
 PHP_IMAP_EXPORT void mm_critical(MAILSTREAM *stream)
--- ./ext/imap/config.m4.orig	2020-03-17 05:40:25.000000000 -0500
+++ ./ext/imap/config.m4	2020-03-17 16:20:15.588685745 -0500
@@ -34,6 +34,7 @@
     PHP_IMAP_EXPORT void mm_nocritical(void){}
     PHP_IMAP_EXPORT void mm_notify(void){}
     PHP_IMAP_EXPORT void mm_login(void){}
+    PHP_IMAP_EXPORT void mm_login_method(void){}
     PHP_IMAP_EXPORT void mm_diskerror(void){}
     PHP_IMAP_EXPORT void mm_status(void){}
     PHP_IMAP_EXPORT void mm_lsub(void){}
@@ -262,6 +263,7 @@
       PHP_IMAP_EXPORT void mm_nocritical(void){}
       PHP_IMAP_EXPORT void mm_notify(void){}
       PHP_IMAP_EXPORT void mm_login(void){}
+      PHP_IMAP_EXPORT void mm_login_method(void){}
       PHP_IMAP_EXPORT void mm_diskerror(void){}
       PHP_IMAP_EXPORT void mm_status(void){}
       PHP_IMAP_EXPORT void mm_lsub(void){}
--- ./configure.orig	2020-03-17 05:40:20.000000000 -0500
+++ ./configure	2020-03-17 16:22:03.768679101 -0500
@@ -39870,6 +39870,7 @@
     PHP_IMAP_EXPORT void mm_nocritical(void){}
     PHP_IMAP_EXPORT void mm_notify(void){}
     PHP_IMAP_EXPORT void mm_login(void){}
+    PHP_IMAP_EXPORT void mm_login_method(void){}
     PHP_IMAP_EXPORT void mm_diskerror(void){}
     PHP_IMAP_EXPORT void mm_status(void){}
     PHP_IMAP_EXPORT void mm_lsub(void){}
@@ -39937,6 +39938,8 @@
     PHP_IMAP_EXPORT void mm_nocritical(void){}
     PHP_IMAP_EXPORT void mm_notify(void){}
     PHP_IMAP_EXPORT void mm_login(void){}
+    PHP_IMAP_EXPORT void mm_login_method(void){}
+    PHP_IMAP_EXPORT void mm_login_method(void){}
     PHP_IMAP_EXPORT void mm_diskerror(void){}
     PHP_IMAP_EXPORT void mm_status(void){}
     PHP_IMAP_EXPORT void mm_lsub(void){}
@@ -40078,6 +40081,7 @@
     PHP_IMAP_EXPORT void mm_nocritical(void){}
     PHP_IMAP_EXPORT void mm_notify(void){}
     PHP_IMAP_EXPORT void mm_login(void){}
+    PHP_IMAP_EXPORT void mm_login_method(void){}
     PHP_IMAP_EXPORT void mm_diskerror(void){}
     PHP_IMAP_EXPORT void mm_status(void){}
     PHP_IMAP_EXPORT void mm_lsub(void){}
