--- ./ext/intl/timezone/timezone_class.cpp.orig	2020-10-27 10:01:59.000000000 -0500
+++ ./ext/intl/timezone/timezone_class.cpp	2020-10-30 04:10:46.429866229 -0500
@@ -37,6 +37,14 @@
 #include <ext/date/php_date.h>
 }
 
+#ifndef FALSE
+#define FALSE (0)
+#endif
+
+#ifndef TRUE
+#define TRUE (!FALSE)
+#endif
+
 using icu::Calendar;
 
 /* {{{ Global variables */
--- ./ext/intl/timezone/timezone_methods.cpp.orig	2020-10-27 10:01:59.000000000 -0500
+++ ./ext/intl/timezone/timezone_methods.cpp	2020-10-30 04:11:06.722865106 -0500
@@ -38,6 +38,14 @@
 }
 #include "common/common_enum.h"
 
+#ifndef FALSE
+#define FALSE (0)
+#endif
+
+#ifndef TRUE
+#define TRUE (!FALSE)
+#endif
+
 using icu::Locale;
 using icu::Calendar;
 
--- ./ext/intl/normalizer/normalizer_normalize.c.orig	2020-10-27 10:01:59.000000000 -0500
+++ ./ext/intl/normalizer/normalizer_normalize.c	2020-10-30 04:09:56.149869010 -0500
@@ -30,6 +30,13 @@
 #include "intl_convert.h"
 #include <unicode/utf8.h>
 
+#ifndef FALSE
+#define FALSE (0)
+#endif
+
+#ifndef TRUE
+#define TRUE (!FALSE)
+#endif
 
 #if U_ICU_VERSION_MAJOR_NUM >= 56
 static const UNormalizer2 *intl_get_normalizer(zend_long form, UErrorCode *err)
--- ./ext/intl/collator/collator_sort.c.orig	2020-10-27 10:01:59.000000000 -0500
+++ ./ext/intl/collator/collator_sort.c	2020-10-30 04:09:12.468871426 -0500
@@ -26,6 +26,14 @@
 #include "collator_convert.h"
 #include "intl_convert.h"
 
+#ifndef FALSE
+#define FALSE (0)
+#endif
+
+#ifndef TRUE
+#define TRUE (!FALSE)
+#endif
+
 #if !defined(HAVE_PTRDIFF_T) && !defined(_PTRDIFF_T_DEFINED)
 typedef zend_long ptrdiff_t;
 #endif
--- ./ext/intl/dateformat/dateformat_attr.c.orig	2020-10-27 10:01:59.000000000 -0500
+++ ./ext/intl/dateformat/dateformat_attr.c	2020-10-30 04:10:23.069867521 -0500
@@ -26,6 +26,14 @@
 #include <unicode/ustring.h>
 #include <unicode/udat.h>
 
+#ifndef FALSE
+#define FALSE (0)
+#endif
+
+#ifndef TRUE
+#define TRUE (!FALSE)
+#endif
+
 /* {{{ proto unicode IntlDateFormatter::getDateType( )
  * Get formatter datetype. }}} */
 /* {{{ proto string datefmt_get_datetype( IntlDateFormatter $mf )
--- ./ext/intl/breakiterator/codepointiterator_internal.cpp.orig	2020-10-27 10:01:59.000000000 -0500
+++ ./ext/intl/breakiterator/codepointiterator_internal.cpp	2020-10-30 04:11:32.563863677 -0500
@@ -27,6 +27,14 @@
     void   *t3;
 } UAlignedMemory;
 
+#ifndef FALSE
+#define FALSE (0)
+#endif
+
+#ifndef TRUE
+#define TRUE (!FALSE)
+#endif
+
 #define U_POINTER_MASK_LSB(ptr, mask) (((ptrdiff_t)(char *)(ptr)) & (mask))
 #define U_ALIGNMENT_OFFSET(ptr) U_POINTER_MASK_LSB(ptr, sizeof(UAlignedMemory) - 1)
 #define U_ALIGNMENT_OFFSET_UP(ptr) (sizeof(UAlignedMemory) - U_ALIGNMENT_OFFSET(ptr))
