diff -ru php-5.2.13.orig/ext/gd/libgd/gd_png.c php-5.2.13/ext/gd/libgd/gd_png.c
--- php-5.2.13.orig/ext/gd/libgd/gd_png.c	2009-03-14 12:48:42.000000000 -0500
+++ php-5.2.13/ext/gd/libgd/gd_png.c	2010-03-21 02:17:03.000000000 -0500
@@ -139,7 +139,7 @@
 		return NULL;
 	}
 
-	if (!png_check_sig (sig, 8)) { /* bad signature */
+	if (png_sig_cmp (sig, 0, 8)) { /* bad signature */
 		return NULL;
 	}
 
