--- ./Makefile.orig	2024-05-11 13:17:31.232871231 -0500
+++ ./Makefile	2024-05-11 13:18:04.399871973 -0500
@@ -3,7 +3,7 @@
 
 WARNINGS=-Wall -W -Wpointer-arith -Wbad-function-cast -Wcast-qual \
          -Wstrict-prototypes -Wmissing-prototypes \
-         -Wmissing-declarations -Wnested-externs -Winline -Wcast-align
+         -Wmissing-declarations -Wnested-externs -Winline -Wcast-align -Wno-error=implicit-int
 
 CFLAGS = -g $(WARNINGS)
 
