From 6f452937366bb1599fb0ae986958786d52946f15 Mon Sep 17 00:00:00 2001
From: Andre Miranda <andreldm@xfce.org>
Date: Sat, 6 Jul 2019 13:42:20 -0300
Subject: [PATCH] Fix use after free crash (Bug #15679)

---
 garcon/garcon-menu-item.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/garcon/garcon-menu-item.c b/garcon/garcon-menu-item.c
index 35f6bed..5f23467 100644
--- a/garcon/garcon-menu-item.c
+++ b/garcon/garcon-menu-item.c
@@ -1154,6 +1154,8 @@ garcon_menu_item_reload_from_file (GarconMenuItem  *item,
 
   /* Update application actions */
   _garcon_g_list_free_full (item->priv->actions, garcon_menu_item_action_unref);
+  item->priv->actions = NULL;
+
   str_list = xfce_rc_read_list_entry (rc, G_KEY_FILE_DESKTOP_KEY_ACTIONS, ";");
   if (G_LIKELY (str_list != NULL))
     {
-- 
2.22.0

