--- ./plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp.orig	2017-05-06 16:40:46.273796232 -0500
+++ ./plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp	2017-05-06 16:47:19.594787855 -0500
@@ -261,7 +261,7 @@
     AVDictionaryEntry *ade = av_dict_get( d->formatContext->metadata, "TITLE", NULL, 0 );
     if( ade == NULL )
         return QString();
-    if( ade->value != '\0' )
+    if( ade->value )
         return QString::fromLocal8Bit( ade->value );
     else
         return QString();
@@ -274,7 +274,7 @@
     AVDictionaryEntry *ade = av_dict_get( d->formatContext->metadata, "ARTIST", NULL, 0 );
     if( ade == NULL )
         return QString();
-    if( ade->value != '\0' )
+    if( ade->value )
         return QString::fromLocal8Bit( ade->value );
     else
         return QString();
@@ -287,7 +287,7 @@
     AVDictionaryEntry *ade = av_dict_get( d->formatContext->metadata, "COMMENT", NULL, 0 );
     if( ade == NULL )
         return QString();
-    if( ade->value != '\0' )
+    if( ade->value )
         return QString::fromLocal8Bit( ade->value );
     else
         return QString();
