--- ./makefile.orig	2024-05-11 01:34:14.429927290 -0500
+++ ./makefile	2024-05-11 01:35:16.723928683 -0500
@@ -102,7 +102,7 @@
 
 # Linux
 linux:
-	$(MAKE) CC="gcc" CFLAGS="-c $(OPTIM) $(LINTFLAGS) -DLINUX -DANSI_HDRS" LDFLAGS="$(LDFLAGS)" $(TARGETS)
+	$(MAKE) CC="gcc" CFLAGS="-c $(OPTIM) $(LINTFLAGS) -DLINUX -DANSI_HDRS -Wno-error=implicit-int" LDFLAGS="$(LDFLAGS)" $(TARGETS)
 
 # Linux64
 linux64:
