--- ./gemini/qml/welcomepages/WelcomePageCloud.qml	2020-05-14 03:51:31.000000000 -0300
+++ ./gemini/qml/welcomepages/WelcomePageCloud.qml	2021-01-26 09:18:37.220265667 -0300
@@ -201,7 +201,7 @@
     }
     Component { id: addDropBox; AddDropbox { } }
     Component { id: addWebDav; AddWebdav { } }
-    Component { id: addGit; AddGit { } }
+    // Component { id: addGit; AddGit { } }
 
     function editComponentFromName(name) {
         var elements = {
@@ -213,7 +213,7 @@
     }
     Component { id: editDropBox; AddDropbox { } }
     Component { id: editWebDav; EditDetailsBase { } }
-    Component { id: editGit; EditGit { } }
+    // Component { id: editGit; EditGit { } }
 
     Component { id: removeAccountDlg; RemoveAccountDlg { } }
 
