diff --git a/psx/psx.c b/psx/psx.c
index d66a7bf..ba515bb 100644
--- a/psx/psx.c
+++ b/psx/psx.c
@@ -486,7 +486,7 @@ long int __psx_syscall(long int syscall_nr, ...) {
 
 	for (;;) {
 	    char buf[BUF_SIZE];
-	    size_t nread = syscall(SYS_getdents, fd, buf, BUF_SIZE);
+	    size_t nread = syscall(SYS_getdents64, fd, buf, BUF_SIZE);
 	    if (nread == 0) {
 		break;
 	    } else if (nread < 0) {
