--- ./libcap/Makefile.orig	2019-12-22 18:12:31.000000000 -0600
+++ ./libcap/Makefile	2019-12-27 13:07:04.168648079 -0600
@@ -88,9 +88,9 @@
 	install -m 0644 libcap.pc $(FAKEROOT)$(PKGCONFIGDIR)/libcap.pc
 ifeq ($(GOLANG),yes)
 	mkdir -p -m 0755 $(FAKEROOT)$(GOPKGDIR)/libcap/cap
-	install -m 0644 src/libcap/cap/* $(FAKEROOT)$(GOPKGDIR)/libcap/cap/
+	install -m 0644 ../go/src/libcap/cap/* $(FAKEROOT)$(GOPKGDIR)/libcap/cap/
 	mkdir -p -m 0755 $(FAKEROOT)$(GOPKGDIR)/libcap/psx
-	install -m 0644 src/libcap/psx/* $(FAKEROOT)$(GOPKGDIR)/libcap/psx/
+	install -m 0644 ../go/src/libcap/psx/* $(FAKEROOT)$(GOPKGDIR)/libcap/psx/
 endif
 
 clean:
