--- ./meson.build.orig	2025-03-10 06:54:03.000000000 -0500
+++ ./meson.build	2025-03-12 12:33:57.230022598 -0500
@@ -494,7 +494,12 @@
     )
   endif
 else
-  git_version_h = files('git-version.h')
+  git_version_h = vcs_tag(
+    input : 'git-version.h.in',
+    output: 'git-version.h',
+    replace_string: '@BABL_GIT_VERSION@',
+    fallback: '@0@'.format(version),
+  )
 endif
 
 ################################################################################
