diff -Nur librsvg-2.22.3.orig/configure librsvg-2.22.3/configure
--- librsvg-2.22.3.orig/configure	2008-09-22 19:54:49.000000000 -0500
+++ librsvg-2.22.3/configure	2008-10-05 17:10:04.772100035 -0500
@@ -24636,7 +24636,7 @@
 
 if test "x$test_mozilla_plugin" != "xno"; then
    if test -z "$MOZILLA_CONFIG"; then
-      for ac_prog in mozilla-config firefox-config xulrunner-config
+      for ac_prog in mozilla-config firefox-config xulrunner-config seamonkey-config
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
 set dummy $ac_prog; ac_word=$2
