diff --git a/Source/WTF/wtf/Platform.h b/Source/WTF/wtf/Platform.h
index 67ac0df..bf09ef0 100644
--- a/Source/WTF/wtf/Platform.h
+++ b/Source/WTF/wtf/Platform.h
@@ -166,11 +166,11 @@
 #if defined(__SSE2__) || (defined(_M_IX86_FP) && _M_IX86_FP >= 2)
 #define WTF_CPU_X86_SSE2 1
 #endif
+#endif
 
 /* CPU(LOONGARCH64) - LOONGARCH64 */
 #if defined(__loongarch64)
 #define WTF_CPU_LOONGARCH64 1
-#endif
 
 #endif
 
