--- ./dist/Makefile.in.orig	2010-04-12 15:25:22.000000000 -0500
+++ ./dist/Makefile.in	2018-04-09 11:58:49.634283283 -0500
@@ -1001,11 +1001,6 @@
 
 install_docs:
 	@echo "Installing documentation: $(DESTDIR)$(docdir) ..."
-	@test -d $(DESTDIR)$(docdir) || \
-	    ($(MKDIR) -p $(DESTDIR)$(docdir) && \
-	    $(CHMOD) $(dmode) $(DESTDIR)$(docdir))
-	@cd $(DESTDIR)$(docdir) && $(RM) -r $(DOCLIST)
-	@cd $(srcdir)/docs && $(CP) -pr $(DOCLIST) $(DESTDIR)$(docdir)/
 
 uninstall_docs:
 	@cd $(DESTDIR)$(docdir) && $(RM) -r $(DOCLIST)
