diff --git a/dist/configure b/dist/configure
index 295fec1..6e8b0ed 100755
--- a/dist/configure
+++ b/dist/configure
@@ -17364,6 +17364,7 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu
 /* end confdefs.h.  */
 #include <stdlib.h>
          #include <pthread.h>
+#include <stdlib.h>
 
          static pthread_once_t once_control_ = PTHREAD_ONCE_INIT;
          static pthread_key_t key;
@@ -18756,6 +18757,7 @@ if test "$cross_compiling" = yes; then :
 /* end confdefs.h.  */
 
 #include <pthread.h>
+#include <stdlib.h>
 int
 main ()
 {
@@ -18792,7 +18794,8 @@ else
 /* end confdefs.h.  */
 
 #include <pthread.h>
-main() {
+#include <stdlib.h>
+int main() {
 	pthread_cond_t cond;
 	pthread_mutex_t mutex;
 	pthread_condattr_t condattr;
@@ -18828,6 +18831,7 @@ if test "$cross_compiling" = yes; then :
 /* end confdefs.h.  */
 
 #include <pthread.h>
+#include <stdlib.h>
 int
 main ()
 {
@@ -18864,7 +18868,8 @@ else
 /* end confdefs.h.  */
 
 #include <pthread.h>
-main() {
+#include <stdlib.h>
+int main() {
 	pthread_cond_t cond;
 	pthread_mutex_t mutex;
 	pthread_condattr_t condattr;
@@ -18899,6 +18904,7 @@ if test "$cross_compiling" = yes; then :
 /* end confdefs.h.  */
 
 #include <pthread.h>
+#include <stdlib.h>
 int
 main ()
 {
@@ -18933,6 +18939,7 @@ else
 /* end confdefs.h.  */
 
 #include <pthread.h>
+#include <stdlib.h>
 main() {
 	pthread_cond_t cond;
 	pthread_mutex_t mutex;
@@ -18967,6 +18974,7 @@ if test "$cross_compiling" = yes; then :
 /* end confdefs.h.  */
 
 #include <pthread.h>
+#include <stdlib.h>
 int
 main ()
 {
@@ -19001,6 +19009,7 @@ else
 /* end confdefs.h.  */
 
 #include <pthread.h>
+#include <stdlib.h>
 main() {
 	pthread_cond_t cond;
 	pthread_mutex_t mutex;
