--- ./CMakeLists.txt.orig	2020-10-28 06:42:28.000000000 -0500
+++ ./CMakeLists.txt	2020-12-14 14:12:57.970780187 -0600
@@ -568,6 +568,10 @@
 add_definitions(-DBOOST_ALL_NO_LIB)
 find_package(Boost 1.55 REQUIRED COMPONENTS system)
 include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
+if(${Boost_VERSION} VERSION_GREATER_EQUAL "1.75")
+    message(WARNING "Boost 1.75 or higher detected, upgrading to C++14.")
+    set(CMAKE_CXX_STANDARD 14)
+endif()
 ##
 ## Test for GNU Scientific Library
 ##
