--- ./tools/c_rehash.in.orig	2018-03-27 08:50:41.000000000 -0500
+++ ./tools/c_rehash.in	2018-05-14 13:47:20.553710437 -0500
@@ -11,8 +11,8 @@
 # Perl c_rehash script, scan all files in a directory
 # and add symbolic links to their hash values.
 
-my $dir = {- quotify1($config{openssldir}) -};
-my $prefix = {- quotify1($config{prefix}) -};
+my $dir = "{- quotify1($config{openssldir}) -}";
+my $prefix = "{- quotify1($config{prefix}) -}";
 
 my $errorcount = 0;
 my $openssl = $ENV{OPENSSL} || "openssl";
