--- ./tools/clang/lib/Driver/ToolChains/Gnu.cpp.orig	2017-08-10 20:47:32.000000000 -0500
+++ ./tools/clang/lib/Driver/ToolChains/Gnu.cpp	2017-09-07 14:41:16.170110145 -0500
@@ -1834,8 +1834,8 @@
       "i686-linux-gnu",       "i686-pc-linux-gnu",     "i486-linux-gnu",
       "i386-linux-gnu",       "i386-redhat-linux6E",   "i686-redhat-linux",
       "i586-redhat-linux",    "i386-redhat-linux",     "i586-suse-linux",
-      "i486-slackware-linux", "i686-montavista-linux", "i686-linux-android",
-      "i586-linux-gnu"};
+      "i486-slackware-linux", "i586-slackware-linux",  "i686-montavista-linux",
+      "i686-linux-android",   "i586-linux-gnu"};
 
   static const char *const MIPSLibDirs[] = {"/lib"};
   static const char *const MIPSTriples[] = {"mips-linux-gnu", "mips-mti-linux",
