--- ./tools/clang/tools/extra/CMakeLists.txt.orig	2018-03-22 18:34:46.000000000 -0500
+++ ./tools/clang/tools/extra/CMakeLists.txt	2018-09-20 18:03:37.034041551 -0500
@@ -10,17 +10,10 @@
 add_subdirectory(clang-doc)
 add_subdirectory(clang-query)
 add_subdirectory(clang-move)
-add_subdirectory(clangd)
 add_subdirectory(include-fixer)
 add_subdirectory(pp-trace)
 add_subdirectory(tool-template)
 
-# Add the common testsuite after all the tools.
-if(CLANG_INCLUDE_TESTS)
-add_subdirectory(test)
-add_subdirectory(unittests)
-endif()
-
 option(CLANG_TOOLS_EXTRA_INCLUDE_DOCS "Generate build targets for the Clang Extra Tools docs."
   ${LLVM_INCLUDE_DOCS})
 if( CLANG_TOOLS_EXTRA_INCLUDE_DOCS )
