diff --git a/tools/clang/lib/Driver/ToolChains/Gnu.cpp b/tools/clang/lib/Driver/ToolChains/Gnu.cpp
index 8d152e4cc..2a726ace8 100644
--- a/tools/clang/lib/Driver/ToolChains/Gnu.cpp
+++ b/tools/clang/lib/Driver/ToolChains/Gnu.cpp
@@ -2262,7 +2262,8 @@ void Generic_GCC::GCCInstallationDetector::AddDefaultGCCPrefixes(
 
   static const char *const LoongArch64LibDirs[] = {"/lib64", "/lib"};
   static const char *const LoongArch64Triples[] = {
-      "loongarch64-linux-gnu", "loongarch64-unknown-linux-gnu"};
+      "loongarch64-linux-gnu", "loongarch64-unknown-linux-gnu",
+      "loongarch64-slackware-linux"};
 
   static const char *const M68kLibDirs[] = {"/lib"};
   static const char *const M68kTriples[] = {
