--- ./tools/clang/lib/CodeGen/CodeGenPGO.cpp.orig	2018-03-08 19:50:01.115345121 -0600
+++ ./tools/clang/lib/CodeGen/CodeGenPGO.cpp	2018-03-08 19:52:24.016358481 -0600
@@ -22,10 +22,7 @@
 #include "llvm/Support/FileSystem.h"
 #include "llvm/Support/MD5.h"
 
-static llvm::cl::opt<bool>
-    EnableValueProfiling("enable-value-profiling", llvm::cl::ZeroOrMore,
-                         llvm::cl::desc("Enable value profiling"),
-                         llvm::cl::Hidden, llvm::cl::init(false));
+static bool EnableValueProfiling = false;
 
 using namespace clang;
 using namespace CodeGen;
