--- ./tools/lldb/source/Utility/CMakeLists.txt.orig	2022-06-22 11:46:24.000000000 -0500
+++ ./tools/lldb/source/Utility/CMakeLists.txt	2022-06-27 22:33:56.715179825 -0500
@@ -19,9 +19,9 @@
   list(APPEND LLDB_SYSTEM_LIBS ws2_32 rpcrt4)
 endif ()
 
-if (NOT HAVE_CXX_ATOMICS64_WITHOUT_LIB )
+#if (NOT HAVE_CXX_ATOMICS64_WITHOUT_LIB )
     list(APPEND LLDB_SYSTEM_LIBS atomic)
-endif()
+#endif()
 
 add_lldb_library(lldbUtility
   ArchSpec.cpp
