diff -ru curl-7.27.0.orig/curl-config.in curl-7.27.0/curl-config.in
--- curl-7.27.0.orig/curl-config.in	2012-05-25 16:56:17.000000000 -0500
+++ curl-7.27.0/curl-config.in	2012-08-17 12:01:44.000000000 -0500
@@ -135,9 +135,9 @@
           CPPFLAG_CURL_STATICLIB=""
         fi
        	if test "X@includedir@" = "X/usr/include"; then
-          echo "$(CPPFLAG_CURL_STATICLIB)"
+          echo "${CPPFLAG_CURL_STATICLIB}"
         else
-          echo "$(CPPFLAG_CURL_STATICLIB)-I@includedir@"
+          echo "${CPPFLAG_CURL_STATICLIB}-I@includedir@"
         fi
        	;;
 
