From cb0ebfa54639619c11ea8277ab7573a5db7484f9 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Tue, 29 Jul 2014 09:37:44 -0400
Subject: [PATCH 4/5] Stop using deprecated xf86PciInfo.h

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
 src/xgi_driver.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/xgi_driver.c b/src/xgi_driver.c
index 637e360..b440e85 100644
--- a/src/xgi_driver.c
+++ b/src/xgi_driver.c
@@ -43,9 +43,6 @@
 #include <xf86RandR12.h>
 #include "vbe.h"
 
-/* PCI vendor/device definitions */
-#include "xf86PciInfo.h"
-
 /* fbdevhw & vgahw */
 #include "fbdevhw.h"
 #include "vgaHW.h"
-- 
2.3.5

