From e36331f3f2641900efe8363fe3dc15877a72c027 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Wed, 8 Oct 2014 10:34:45 +0200
Subject: [PATCH 5/5] Update for new damage API

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
 src/xgi_shadow.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/xgi_shadow.c b/src/xgi_shadow.c
index 80f3645..c146ce4 100644
--- a/src/xgi_shadow.c
+++ b/src/xgi_shadow.c
@@ -86,7 +86,7 @@ void XGIRefreshArea(ScrnInfoPtr pScrn, int num, BoxPtr pbox)
 
 void XGIShadowUpdate (ScreenPtr pScreen, shadowBufPtr pBuf)
 {
-    RegionPtr       pRegion = &pBuf->damage;
+    RegionPtr       pRegion = DamageRegion(pBuf->pDamage);
     ScrnInfoPtr     pScrn;
     pScrn = xf86ScreenToScrn(pScreen);
 
-- 
2.3.5

