--- ./contrib/pzstd/Makefile.orig	2017-12-21 03:16:40.000000000 -0600
+++ ./contrib/pzstd/Makefile	2018-02-12 16:25:19.626150873 -0600
@@ -171,8 +171,8 @@
 	$(TESTPROG) ./test/RoundTripTest$(EXT) $(TESTFLAGS)
 
 # Build the main binary
-pzstd$(EXT): main.o Options.o Pzstd.o SkippableFrame.o $(ZSTDDIR)/libzstd.a
-	$(LD_COMMAND)
+pzstd$(EXT): main.o Options.o Pzstd.o SkippableFrame.o
+	$(LD_COMMAND) -L ../../lib -lzstd
 
 # Target that depends on all the tests
 .PHONY: tests
