--- ./docs/kdepim/korganizer/index.docbook.orig	2011-01-19 10:56:02.410999994 -0600
+++ ./docs/kdepim/korganizer/index.docbook	2011-01-19 12:40:25.712999555 -0600
@@ -108,6 +108,15 @@
 <releaseinfo
 >2.20.00</releaseinfo>
 
+<abstract
+><para
+>&korganizer; est un agenda personnel facile à utiliser (un <acronym
+>PIM</acronym
+>, acronyme anglais de <foreignphrase
+>personal information manager</foreignphrase
+>). Vous pouvez planifier des rendez-vous, des événements et des tâches à effectuer. &korganizer; vous remémorera les tâches en instances et vous aidera à respecter votre emploi du temps. </para
+></abstract>
+
 <keywordset>
 <keyword
 >KDE</keyword>
@@ -127,15 +136,6 @@
 >rendez-vous</keyword>
 </keywordset>
 
-<abstract
-><para
->&korganizer; est un agenda personnel facile à utiliser (un <acronym
->PIM</acronym
->, acronyme anglais de <foreignphrase
->personal information manager</foreignphrase
->). Vous pouvez planifier des rendez-vous, des événements et des tâches à effectuer. &korganizer; vous remémorera les tâches en instances et vous aidera à respecter votre emploi du temps. </para
-></abstract>
-
 </bookinfo>
 
 <chapter id="introduction">
