diff --git a/libfrog/crc32.c b/libfrog/crc32.c
index 526ce95..ea21610 100644
--- a/libfrog/crc32.c
+++ b/libfrog/crc32.c
@@ -32,6 +32,7 @@
 #include <inttypes.h>
 #include <asm/types.h>
 #include <sys/time.h>
+#include <stdio.h>
 #include "platform_defs.h"
 /* For endian conversion routines */
 #include "xfs_arch.h"
diff --git a/libfrog/linux.c b/libfrog/linux.c
index a45d99a..d643c52 100644
--- a/libfrog/linux.c
+++ b/libfrog/linux.c
@@ -4,6 +4,7 @@
  * All Rights Reserved.
  */
 
+#include <stdlib.h>
 #include <mntent.h>
 #include <sys/stat.h>
 #include <sys/ioctl.h>
diff --git a/libfrog/ptvar.c b/libfrog/ptvar.c
index 7ac8c54..316108c 100644
--- a/libfrog/ptvar.c
+++ b/libfrog/ptvar.c
@@ -10,6 +10,7 @@
 #include <assert.h>
 #include <pthread.h>
 #include <unistd.h>
+#include <errno.h>
 #include "platform_defs.h"
 #include "ptvar.h"
 
diff --git a/libfrog/topology.c b/libfrog/topology.c
index b1b470c..e14c1bc 100644
--- a/libfrog/topology.c
+++ b/libfrog/topology.c
@@ -3,7 +3,7 @@
  * Copyright (c) 2000-2001,2005 Silicon Graphics, Inc.
  * All Rights Reserved.
  */
-
+#include <stdlib.h>
 #include "libxfs.h"
 #include "libxcmd.h"
 #ifdef ENABLE_BLKID
