Description: Close file on error
Origin: other, https://bugzilla.xfce.org/attachment.cgi?id=4447
Bug: https://bugzilla.xfce.org/show_bug.cgi?id=8303
Bug-Ubuntu: https://launchpad.net/bugs/995918

--- tumbler-0.1.25.orig/plugins/gst-thumbnailer/gst-thumbnailer.c
+++ tumbler-0.1.25/plugins/gst-thumbnailer/gst-thumbnailer.c
@@ -233,6 +233,7 @@ make_pipeline (TumblerFileInfo *info,
   if (state == GST_STATE_CHANGE_FAILURE || state == GST_STATE_CHANGE_ASYNC)
     {
       LOG ("failed to or still changing state, aborting (state change %d)", state);
+      gst_element_set_state (playbin, GST_STATE_NULL);
       g_object_unref (playbin);
       return NULL;
     }
@@ -241,6 +242,7 @@ make_pipeline (TumblerFileInfo *info,
   if (n_video == 0)
     {
       LOG ("no video stream, aborting");
+      gst_element_set_state (playbin, GST_STATE_NULL);
       g_object_unref (playbin);
       return NULL;
     }
