--- ./texk/web2c/luatexdir/luazlib/lzlib.c.orig	2024-04-09 12:32:20.232075369 -0500
+++ ./texk/web2c/luatexdir/luazlib/lzlib.c	2024-04-09 12:33:53.746077461 -0500
@@ -544,13 +544,16 @@
 
     /* ====================================================================== */
 
+    /* There's no justification for breaking this every time zlib does a minor
+       upgrade without changing the ABI. Commenting out the below block: */
+
     /* make sure header and library version are consistent */
-    const char* version = zlibVersion();
+    /* const char* version = zlibVersion();
     if (strncmp(version, ZLIB_VERSION, 2))
     {
         lua_pushfstring(L, "zlib library version does not match - header: %s, library: %s", ZLIB_VERSION, version);
         lua_error(L);
-    }
+    } */
 
     /* create new metatable for zlib compression structures */
     luaL_newmetatable(L, ZSTREAMMETA);
