diff --git a/M2Crypto/RSA.py b/M2Crypto/RSA.py
index 1292e1d..1517387 100644
--- a/M2Crypto/RSA.py
+++ b/M2Crypto/RSA.py
@@ -18,7 +18,7 @@ m2.rsa_init(RSAError)
 
 no_padding = m2.no_padding
 pkcs1_padding = m2.pkcs1_padding
-sslv23_padding = m2.sslv23_padding
+#sslv23_padding = m2.sslv23_padding
 pkcs1_oaep_padding = m2.pkcs1_oaep_padding
 
 
diff --git a/SWIG/_rsa.i b/SWIG/_rsa.i
index 4ffae01..b66ddfc 100644
--- a/SWIG/_rsa.i
+++ b/SWIG/_rsa.i
@@ -23,7 +23,9 @@ extern int RSA_check_key(const RSA *);
 
 %constant int no_padding        = RSA_NO_PADDING;
 %constant int pkcs1_padding     = RSA_PKCS1_PADDING;
+#if OPENSSL_VERSION_MAJOR < 3
 %constant int sslv23_padding    = RSA_SSLV23_PADDING;
+#endif
 %constant int pkcs1_oaep_padding = RSA_PKCS1_OAEP_PADDING;
 
 %constant int NID_sha1 = NID_sha1;
