--- ./genpowerfail.orig	2008-12-02 17:49:36.000000000 -0600
+++ ./genpowerfail	2008-12-02 18:22:37.000000000 -0600
@@ -36,23 +36,23 @@
 	                stats=`head -1 $statpath`
 			case "$stats" in
 				FAIL)  # Power is down
-                                	shutdown -r +2 "THE POWER IS DOWN! SHUTTING DOWN SYSTEM! PLEASE LOG OFF NOW!" < /dev/console &
+                                	shutdown -h +2 "THE POWER IS DOWN! SHUTTING DOWN SYSTEM! PLEASE LOG OFF NOW!" < /dev/console &
 					;;
 				SCRAM) # Battery is low
-					shutdown -r now "THE POWER IS DOWN! BATTERY POWER IS LOW!  EMERGENCY SHUTDOWN!" < /dev/console &
+					shutdown -h now "THE POWER IS DOWN! BATTERY POWER IS LOW!  EMERGENCY SHUTDOWN!" < /dev/console &
 					;;
 				CABLE) # Possible bad cable
- 	                                shutdown -r +1 "POSSIBLE BAD CABLE! SHUTTING DOWN SYSTEM! PLEASE LOG OFF NOW!" < /dev/console &
+ 	                                shutdown -h +1 "POSSIBLE BAD CABLE! SHUTTING DOWN SYSTEM! PLEASE LOG OFF NOW!" < /dev/console &
 					;;
                                 *)     # Unknown message, assume power is down
-                                        shutdown -r +2 "THE POWER IS DOWN! SHUTTING DOWN SYSTEM! PLEASE LOG OFF NOW!" < /dev/console &
+                                        shutdown -h +2 "THE POWER IS DOWN! SHUTTING DOWN SYSTEM! PLEASE LOG OFF NOW!" < /dev/console &
                                         ;;
 
 			esac
 		else
-			# genowerfail called, and upsstatus dosen't exist.
+			# genpowerfail called, and upsstatus dosen't exist.
 			# Assume user is using powerd, and shutdown.
-			shutdown -r +2 "THE POWER IS DOWN! SHUTTING DOWN SYSTEM! PLEASE LOG OFF NOW!" < /dev/console &
+			shutdown -h +2 "THE POWER IS DOWN! SHUTTING DOWN SYSTEM! PLEASE LOG OFF NOW!" < /dev/console &
  
 		fi
 		;;
