--- ./genpowerfail.orig	Wed Apr 10 18:25:07 2002
+++ ./genpowerfail	Wed Apr 10 19:14:27 2002
@@ -6,19 +6,19 @@
 #		This script is also being run when the power comes
 #		up again (if it does in time!)
 #
-# Version:	/etc/genpowerfail 1.0.1
+# Version:	/sbin/genpowerfail 1.0.1
 #
 # Author:	Tom Webster <webster@kaiwan.com>
 #
 
 # Set the path.
-PATH=/sbin:/etc:/bin:/usr/bin
+PATH=/sbin:/bin:/usr/bin:/usr/sbin
 
 # Set location of upsstatus file
 statpath="/etc/upsstatus"
 
 # Set location of file containing PID of running shutdowns
-spidpath="/etc/shutdownpid"
+spidpath="/var/run/shutdownpid"
 
 # See what happened.
 case "$1" in
@@ -63,7 +63,7 @@
 		shutdown -c "THE POWER IS BACK"
 		;;
   	*)
-		echo "Usage: /etc/genpowerfail {start|stop}"
+		echo "Usage: /sbin/genpowerfail {start|stop}"
 		exit 1
 		;;
   esac
