--- ./SDL_ttf.c.orig	2012-01-14 22:44:08.000000000 -0600
+++ ./SDL_ttf.c	2016-06-08 11:59:50.163770281 -0500
@@ -1747,7 +1747,7 @@
 	/* Copy the character from the pixmap */
 	src = glyph->pixmap.buffer;
 	dst = (Uint8*) textbuf->pixels;
-	for ( row = 0; row < glyph->bitmap.rows; ++row ) {
+	for ( row = 0; row < glyph->pixmap.rows; ++row ) {
 		memcpy( dst, src, glyph->pixmap.width );
 		src += glyph->pixmap.pitch;
 		dst += textbuf->pitch;
