--- ./applets/applets.c.orig	2007-09-16 13:48:10.000000000 -0500
+++ ./applets/applets.c	2007-10-04 22:07:47.000000000 -0500
@@ -16,7 +16,7 @@
 #include "busybox.h"
 
 /* Apparently uclibc defines __GLIBC__ (compat trick?). Oh well. */
-#if ENABLE_STATIC && defined(__GLIBC__) && !defined(__UCLIBC__)
+/* #if ENABLE_STATIC && defined(__GLIBC__) && !defined(__UCLIBC__) 
 #warning Static linking against glibc produces buggy executables
 #warning (glibc does not cope well with ld --gc-sections).
 #warning See sources.redhat.com/bugzilla/show_bug.cgi?id=3400
@@ -24,7 +24,7 @@
 #warning If you still want to do it, remove -Wl,--gc-sections
 #warning from top-level Makefile and remove this warning.
 #error Aborting compilation.
-#endif
+#endif */
 
 
 /* Declare <applet>_main() */
