--- ./Makefile.orig	2007-09-29 18:54:25.000000000 -0500
+++ ./Makefile	2007-11-27 15:35:42.000000000 -0600
@@ -570,7 +570,7 @@
 quiet_cmd_busybox__ ?= LINK    $@
       cmd_busybox__ ?= $(srctree)/scripts/trylink $(CC) $(LDFLAGS) \
       -o $@ -Wl,-Map -Wl,$@.map \
-      -Wl,--warn-common -Wl,--sort-common -Wl,--gc-sections \
+      -Wl,--warn-common -Wl,--sort-common \
       -Wl,--start-group $(busybox-all) -Wl,--end-group \
       $(LDLIBS)
 
