--- ./config/makefiles/rust.mk.orig	2019-08-26 16:32:05.000000000 -0500
+++ ./config/makefiles/rust.mk	2019-08-28 15:33:32.393489657 -0500
@@ -243,13 +243,13 @@
 # When we are building in --enable-release mode; we add an additional check to confirm
 # that we are not importing any networking-related functions in rust code. This reduces
 # the chance of proxy bypasses originating from rust code.
-ifndef DEVELOPER_OPTIONS
-ifndef MOZ_DEBUG_RUST
-ifeq ($(OS_ARCH), Linux)
-	$(call py_action,check_binary,--target --networking $@)
-endif
-endif
-endif
+#ifndef DEVELOPER_OPTIONS
+#ifndef MOZ_DEBUG_RUST
+#ifeq ($(OS_ARCH), Linux)
+#	$(call py_action,check_binary,--target --networking $@)
+#endif
+#endif
+#endif
 
 force-cargo-library-check:
 	$(call CARGO_CHECK) --lib $(cargo_target_flag) $(rust_features_flag)
