--- ./defaults/pref/all-thunderbird.js.orig	2005-10-25 22:03:39.000000000 -0500
+++ ./defaults/pref/all-thunderbird.js	2005-12-09 16:22:00.000000000 -0600
@@ -440,3 +440,8 @@
 
 // prevent status-bar spoofing even if people are foolish enough to turn on JS
 pref("dom.disable_window_status_change",          true);
+
+// Use Firefox as the default for http:// and https:// links
+pref("network.protocol-handler.app.http", "/usr/bin/firefox");
+pref("network.protocol-handler.app.https", "/usr/bin/firefox");
+
