--- ./src/device.c.orig	2014-03-10 03:57:48.000000000 -0500
+++ ./src/device.c	2018-09-18 19:06:22.494073271 -0500
@@ -33,6 +33,7 @@
 #include <sys/stat.h>
 #include <sys/time.h>
 #include <sys/resource.h>
+#include <sys/sysmacros.h>
 #include <fcntl.h>
 #include <pwd.h>
 #include <grp.h>
--- ./src/daemon.c.orig	2014-03-10 03:57:48.000000000 -0500
+++ ./src/daemon.c	2018-09-18 19:06:22.490073271 -0500
@@ -45,6 +45,7 @@
 #include <sys/socket.h>
 #include <sys/un.h>
 #include <sys/utsname.h>
+#include <sys/sysmacros.h>
 #include <net/if_arp.h>
 #include <fcntl.h>
 #include <signal.h>
--- ./src/mount-monitor.c.orig	2014-03-10 03:58:42.000000000 -0500
+++ ./src/mount-monitor.c	2018-09-18 19:06:22.495073271 -0500
@@ -28,7 +28,9 @@
 #include <errno.h>
 #include <string.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/stat.h>
+
 #include <mntent.h>
 
 #include <glib.h>
--- ./tools/umount-udisks.c.orig	2014-03-10 03:57:48.000000000 -0500
+++ ./tools/umount-udisks.c	2018-09-18 19:06:22.497073271 -0500
@@ -30,6 +30,7 @@
 #include <string.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <fcntl.h>
 #include <pwd.h>
 #include <grp.h>
--- ./tools/udisks.c.orig	2014-03-10 03:57:48.000000000 -0500
+++ ./tools/udisks.c	2018-09-18 19:06:22.496073271 -0500
@@ -31,6 +31,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/wait.h>
+#include <sys/sysmacros.h>
 #include <fcntl.h>
 #include <pwd.h>
 #include <grp.h>
