From 5edf86dcebd2ab55a4a7221db77e10676723bbb1 Mon Sep 17 00:00:00 2001
From: piterpunk <piterpunk@slackware.com>
Date: Mon, 8 Feb 2021 18:22:33 -0300
Subject: [PATCH] Allow new-config after slackpkg upgrade itself

Signed-off-by: Robby Workman <rworkman@slackware.com>
---
 files/core-functions.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/files/core-functions.sh b/files/core-functions.sh
index dcbceae..9816458 100644
--- a/files/core-functions.sh
+++ b/files/core-functions.sh
@@ -345,6 +345,7 @@ use slackpkg.\n"
 		CHECKMD5=off
 	elif ! [ -f ${ROOT}/${WORKDIR}/CHECKSUMS.md5 ] && \
 		[ "$CMD" != "update" ] && \
+		[ "$CMD" != "new-config" ] && \
 		[ "$CHECKMD5" = "on" ]; then
 		echo -e "\n\
 No CHECKSUMS.md5 found!  Please disable md5sums checking\n\
-- 
2.30.0

