From ff0b9ff70a6031e712035e98b1dd2845774cb137 Mon Sep 17 00:00:00 2001
From: Robby Workman <rworkman@slackware.com>
Date: Wed, 5 Sep 2018 22:49:24 -0500
Subject: [PATCH 6/6] Mention possible stale mirror if CHECKSUMS.md5 gpg verify
 fails

Signed-off-by: Robby Workman <rworkman@slackware.com>
---
 files/core-functions.sh | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/files/core-functions.sh b/files/core-functions.sh
index 76cab6f..908edb0 100644
--- a/files/core-functions.sh
+++ b/files/core-functions.sh
@@ -1067,7 +1067,10 @@ function updatefilelists()
 				echo -e "\
 \n\t\tERROR: Verification of the  gpg signature on CHECKSUMS.md5\n\
 \t\t       failed! This could mean that the file is out of date\n\
-\t\t       or has been tampered with.\n"
+\t\t       or has been tampered with. If you use mirrors.slackware.com\n\
+\t\t       as your mirror, this could also mean that the mirror to\n\
+\t\t       which you got redirected is not yet updated with the most\n\
+\t\t       recent changes in the Slackware tree.\n"
 				cleanup
 			fi
 		elif [ "$SLACKKEY" != "" ]; then
-- 
2.18.0

