--- ./hostname.c.orig	2001-04-08 12:04:23.000000000 -0500
+++ ./hostname.c	2010-08-12 12:19:29.181510641 -0500
@@ -78,6 +78,7 @@
             fprintf(stderr, _("%s: name too long\n"), program_name);
             break;
         default:
+            break;
         }
 	exit(1);
     }
@@ -98,6 +99,7 @@
 	    fprintf(stderr, _("%s: name too long\n"), program_name);
 	    break;
 	default:
+            break;
 	}
 	exit(1);
     };
@@ -117,6 +119,7 @@
 	    fprintf(stderr, _("%s: name too long\n"), program_name);
 	    break;
 	default:
+	    break;
 	}
 	exit(1);
     };
@@ -174,6 +177,7 @@
 	printf("%s\n", hp->h_name);
 	break;
     default:
+        break;
     }
 }
 
@@ -330,7 +334,6 @@
 	case 'h':
 	default:
 	    usage();
-
 	};
 
 
