--- ./src/widgets/kpropertiesdialog.cpp.orig	2021-09-05 14:40:56.000000000 -0500
+++ ./src/widgets/kpropertiesdialog.cpp	2021-10-07 13:51:57.301053426 -0500
@@ -1227,6 +1227,7 @@
     Q_ASSERT(ro); // false isn't supported
     if (ro && d->m_ui->fileNameLabel->isHidden()) {
         Q_ASSERT(!d->m_bFromTemplate);
+        disconnect(d->m_ui->fileNameLineEdit, &QLineEdit::textChanged, this, &KFilePropsPlugin::nameFileChanged);
         d->m_ui->fileNameLineEdit->setText(QString());
         d->m_ui->fileNameLineEdit->hide();
 
