--- ./src/widgets/kpropertiesdialog.cpp.orig	2021-10-02 09:28:47.000000000 -0500
+++ ./src/widgets/kpropertiesdialog.cpp	2021-10-10 13:23:29.757051042 -0500
@@ -1239,6 +1239,8 @@
     if (readOnly) {
         Q_ASSERT(!d->m_bFromTemplate);
 
+        disconnect(d->m_ui->fileNameLineEdit, &QLineEdit::textChanged, this, &KFilePropsPlugin::nameFileChanged);
+
         d->m_ui->fileNameLineEdit->setText(QString());
         d->m_ui->fileNameLineEdit->hide();
 
