--- ./lib/libv4lconvert/control/libv4lcontrol.c.orig	2017-12-27 08:33:07.000000000 -0600
+++ ./lib/libv4lconvert/control/libv4lcontrol.c	2018-09-18 19:07:39.598074096 -0500
@@ -20,9 +20,7 @@
  */
 
 #include <sys/types.h>
-#if defined(MAJOR_IN_SYSMACROS)
 #include <sys/sysmacros.h>
-#endif
 #include <sys/mman.h>
 #include <fcntl.h>
 #include <sys/stat.h>
--- ./lib/libv4l2/libv4l2.c.orig	2017-12-27 07:50:55.000000000 -0600
+++ ./lib/libv4l2/libv4l2.c	2018-09-18 19:07:39.596074096 -0500
@@ -70,6 +70,7 @@
 #include <sys/types.h>
 #include <sys/mman.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include "libv4l2.h"
 #include "libv4l2-priv.h"
 #include "libv4l-plugin.h"
--- ./utils/v4l2-ctl/v4l2-ctl.cpp.orig	2017-12-27 07:50:55.000000000 -0600
+++ ./utils/v4l2-ctl/v4l2-ctl.cpp	2018-09-18 19:07:39.599074096 -0500
@@ -33,6 +33,7 @@
 #include <errno.h>
 #include <sys/ioctl.h>
 #include <sys/time.h>
+#include <sys/sysmacros.h>
 #include <dirent.h>
 #include <math.h>
 
