--- ./contrib/gconv/meson.build.orig	2023-11-25 07:27:49.000000000 -0600
+++ ./contrib/gconv/meson.build	2023-12-03 12:52:06.319957999 -0600
@@ -16,6 +16,7 @@
                              arib_std_b24_sources,
                              name_prefix : '',
                              dependencies : arib_std_b24_deps,
+                             link_args : [ '-Wl,--enable-new-dtags', '-Wl,-rpath=$ORIGIN' ],
                              install : true,
                              install_dir : gconv_install_dir,
                              include_directories : v4l2_utils_incdir)
