--- ./misc/mke2fs.conf.in.orig	2023-02-08 13:32:07.116015773 -0600
+++ ./misc/mke2fs.conf.in	2023-02-08 13:33:43.568017931 -0600
@@ -11,7 +11,8 @@
 		features = has_journal
 	}
 	ext4 = {
-		features = has_journal,extent,huge_file,flex_bg,metadata_csum,metadata_csum_seed,64bit,dir_nlink,extra_isize,orphan_file
+		features = has_journal,extent,huge_file,flex_bg,metadata_csum,metadata_csum_seed,dir_nlink,extra_isize,orphan_file
+		auto_64-bit_support = 1
 	}
 	small = {
 		blocksize = 1024
