--- ./misc/mke2fs.conf.in.orig	2016-05-17 00:32:58.000000000 -0500
+++ ./misc/mke2fs.conf.in	2016-05-24 19:04:24.956000477 -0500
@@ -11,7 +11,8 @@
 		features = has_journal
 	}
 	ext4 = {
-		features = has_journal,extent,huge_file,flex_bg,64bit,dir_nlink,extra_isize
+		features = has_journal,extent,huge_file,flex_bg,dir_nlink,extra_isize
+		auto_64-bit_support = 1
 		inode_size = 256
 	}
 	ext4dev = {
