--- xxgdb-1.12/source.c~	Fri Apr 21 12:16:02 1995
+++ xxgdb-1.12/source.c	Mon Dec  2 17:42:39 1996
@@ -79,7 +79,7 @@
 #endif
 #include <stdlib.h>
 
-#include <X11/Xos.h>
+/* #include <X11/Xos.h> */
 #include <sys/stat.h>
 #include <pwd.h>
 #include "global.h"
--- xxgdb-1.12/global.h~	Thu Apr  6 12:23:42 1995
+++ xxgdb-1.12/global.h	Mon Dec  2 17:38:21 1996
@@ -62,8 +62,9 @@
  *
  *    Contain extern declarations of variables and functions.
  */
-
+#undef SYSV
 #include "defs.h"
+#define SYSV
 
 
 #if defined(__GNUC__) && !defined(alloca)
