diff -Naur xxgdb-1.12/Imakefile xxgdb-1.12.fixed/Imakefile
--- xxgdb-1.12/Imakefile	1995-06-19 19:20:55.000000000 -0300
+++ xxgdb-1.12.fixed/Imakefile	2007-02-21 10:40:57.000000000 -0200
@@ -59,7 +59,7 @@
 #endif
 
 #ifdef LinuxArchitecture
-DEFINES = -DSYSV $(DEFGDB)
+DEFINES = $(DEFGDB)
 #endif
 
 #if defined(NeXTArchitecture) || defined(NextArchitecture)
diff -Naur xxgdb-1.12/filemenu.c xxgdb-1.12.fixed/filemenu.c
--- xxgdb-1.12/filemenu.c	1994-08-29 16:38:28.000000000 -0300
+++ xxgdb-1.12.fixed/filemenu.c	2007-02-21 10:47:47.000000000 -0200
@@ -207,12 +207,12 @@
 	return False;
     if (statbuf.st_mode & S_IFDIR) {		/* is directory */
 	strcat(entry->d_name, "/");
-	++(entry->d_namlen);
+	++(entry->d_reclen);
 	return True;
     }
     if (statbuf.st_mode & S_IEXEC) {		/* is executable */
 	strcat(entry->d_name, "*");
-	++(entry->d_namlen);
+	++(entry->d_reclen);
 	return True;
     }
     return True;
diff -Naur xxgdb-1.12/gdb_handler.c xxgdb-1.12.fixed/gdb_handler.c
--- xxgdb-1.12/gdb_handler.c	1995-04-19 13:46:22.000000000 -0300
+++ xxgdb-1.12.fixed/gdb_handler.c	2007-02-21 10:51:30.000000000 -0200
@@ -85,10 +85,7 @@
  *    query_gdb():			Send command to gdb.
  */
 
-#ifdef SYSV 
-#   include <signal.h>
-#endif
-
+#include <signal.h>
 
 /*  
  *  Display an outlined arrow to locate the calling routine in a stack
