--- ./configure.ac.orig	2015-04-27 09:07:43.000000000 -0500
+++ ./configure.ac	2015-12-14 13:34:54.592931654 -0600
@@ -52,6 +52,11 @@
   fi
 AM_CONDITIONAL(SYSTEMD, [test -n "$with_systemdsystemunitdir" -a "x$with_systemdsystemunitdir" != xno ])
 
+AC_SEARCH_LIBS(yp_get_default_domain, nsl)
+AC_MSG_CHECKING([for hosts_access in -lwrap])
+saved_LIBS="$LIBS"
+LIBS="$saved_LIBS -lwrap"
+
 AS_IF([test x$enable_libwrap = xyes], [
 	AC_CHECK_LIB([wrap], [hosts_access], ,
 		AC_MSG_ERROR([libwrap support requested but unable to find libwrap]))
