diff -Nur pango-1.28.3.orig//pango/Makefile.am pango-1.28.3/pango/Makefile.am
--- pango-1.28.3.orig//pango/Makefile.am	2010-09-29 05:42:31.000000000 -0500
+++ pango-1.28.3/pango/Makefile.am	2010-10-13 15:52:27.163733791 -0500
@@ -27,6 +27,7 @@
 	-DPANGO_ENABLE_BACKEND				\
 	-DPANGO_ENABLE_ENGINE				\
 	-DSYSCONFDIR=\"$(sysconfdir)\"			\
+	-HOST=\"$(host_triplet)\"                       \
 	-DLIBDIR=\"$(libdir)\"				\
 	-I$(top_srcdir)					\
 	$(PANGO_DEBUG_FLAGS)				\
diff -Nur pango-1.28.3.orig//pango/Makefile.in pango-1.28.3/pango/Makefile.in
--- pango-1.28.3.orig//pango/Makefile.in	2010-09-29 06:02:35.000000000 -0500
+++ pango-1.28.3/pango/Makefile.in	2010-10-13 15:52:53.182319091 -0500
@@ -535,6 +535,7 @@
 	-DPANGO_ENABLE_BACKEND				\
 	-DPANGO_ENABLE_ENGINE				\
 	-DSYSCONFDIR=\"$(sysconfdir)\"			\
+	-DHOST=\"$(host_triplet)\"                      \
 	-DLIBDIR=\"$(libdir)\"				\
 	-I$(top_srcdir)					\
 	$(PANGO_DEBUG_FLAGS)				\
diff -Nur pango-1.28.3.orig//pango/modules.c pango-1.28.3/pango/modules.c
--- pango-1.28.3.orig//pango/modules.c	2010-02-09 06:06:28.000000000 -0600
+++ pango-1.28.3/pango/modules.c	2010-10-13 15:53:31.016716077 -0500
@@ -529,6 +529,7 @@
 
   if (!file_str)
     file_str = g_build_filename (pango_get_sysconf_subdirectory (),
+				 HOST,
 				 "pango.modules",
 				 NULL);
 
