diff -uarN pango-1.22.4.orig/pango/Makefile.am pango-1.22.4/pango/Makefile.am
--- pango-1.22.4.orig/pango/Makefile.am	2008-12-16 07:14:00.000000000 +0100
+++ pango-1.22.4/pango/Makefile.am	2009-03-04 23:58:26.000000000 +0100
@@ -19,6 +19,7 @@
 	-DPANGO_ENABLE_BACKEND				\
 	-DPANGO_ENABLE_ENGINE				\
 	-DSYSCONFDIR=\"$(sysconfdir)\"			\
+	-DHOST=\"$(host_triplet)\"                      \
 	-DLIBDIR=\"$(libdir)\"				\
 	-I$(top_srcdir)					\
 	$(PANGO_DEBUG_FLAGS)				\
diff -uarN pango-1.22.4.orig/pango/Makefile.in pango-1.22.4/pango/Makefile.in
--- pango-1.22.4.orig/pango/Makefile.in	2008-12-16 07:14:36.000000000 +0100
+++ pango-1.22.4/pango/Makefile.in	2009-03-04 23:58:26.000000000 +0100
@@ -427,6 +427,7 @@
 	-DPANGO_ENABLE_BACKEND				\
 	-DPANGO_ENABLE_ENGINE				\
 	-DSYSCONFDIR=\"$(sysconfdir)\"			\
+	-DHOST=\"$(host_triplet)\"                      \
 	-DLIBDIR=\"$(libdir)\"				\
 	-I$(top_srcdir)					\
 	$(PANGO_DEBUG_FLAGS)				\
diff -uarN pango-1.22.4.orig/pango/modules.c pango-1.22.4/pango/modules.c
--- pango-1.22.4.orig/pango/modules.c	2008-12-16 07:14:00.000000000 +0100
+++ pango-1.22.4/pango/modules.c	2009-03-04 23:58:26.000000000 +0100
@@ -508,6 +508,7 @@
 
   if (!file_str)
     file_str = g_build_filename (pango_get_sysconf_subdirectory (),
+				 HOST,
 				 "pango.modules",
 				 NULL);
 
