--- ./plugins/sftp/mounter.cpp.orig	2015-02-03 20:53:33.000000000 -0600
+++ ./plugins/sftp/mounter.cpp	2015-11-19 17:22:51.795962646 -0600
@@ -136,7 +136,8 @@
         << "-f"
         << "-o" << "IdentityFile=" + m_sftp->device()->privateKeyPath()
         << "-o" << "StrictHostKeyChecking=no" //Do not ask for confirmation because it is not a known host
-        << "-o" << "UserKnownHostsFile=/dev/null"; //Prevent storing as a known host
+        << "-o" << "UserKnownHostsFile=/dev/null" //Prevent storing as a known host
+        << "-o" << "HostKeyAlgorithms=+ssh-dss"; //https://bugs.kde.org/show_bug.cgi?id=351725
     
     m_proc->setProgram(program, arguments);
 
@@ -146,7 +147,7 @@
 
     cleanMountPoint();
     
-    kDebug(debugArea()) << "Staring process: " << m_proc->program().join(" ");
+    kDebug(debugArea()) << "Starting process: " << m_proc->program().join(" ");
     m_proc->start();
 }
 
