From 118b94bdffca7e33a5e0d6b596eb7e51e7109e68 Mon Sep 17 00:00:00 2001
From: Roy Marples <roy@marples.name>
Date: Tue, 6 Aug 2013 12:34:08 +0100
Subject: [PATCH 1/2] Actually validate the search list. Fixes #280

---
 dhcpcd-hooks/20-resolv.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dhcpcd-hooks/20-resolv.conf b/dhcpcd-hooks/20-resolv.conf
index d0dce41..05316c9 100644
--- a/dhcpcd-hooks/20-resolv.conf
+++ b/dhcpcd-hooks/20-resolv.conf
@@ -115,7 +115,7 @@ add_resolv_conf()
 	if [ -n "$new_domain_search" -a \
 	    "$new_domain_search" != "$new_domain_name" ]
 	then
-		if valid_domainname_list; then
+		if valid_domainname_list $new_domain_search; then
 			conf="${conf}search $new_domain_search$NL"
 		else
 			syslog err "Invalid domain name in list: $new_domain_search"
-- 
1.8.3.2

