diff -Nur dhcpcd-3.2.3.orig/config.h dhcpcd-3.2.3/config.h
--- dhcpcd-3.2.3.orig/config.h	2008-02-25 02:28:19.000000000 -0600
+++ dhcpcd-3.2.3/config.h	2009-03-06 14:55:01.998436374 -0600
@@ -71,10 +71,10 @@
 
 #define OPENNTPFILE         ETCDIR "/ntpd.conf"
 
-#define DEFAULT_SCRIPT      ETCDIR "/" PACKAGE ".sh"
+#define DEFAULT_SCRIPT      INFODIR "/" PACKAGE ".sh"
 
-#define STATEDIR            "/var"
-#define PIDFILE             STATEDIR "/run/" PACKAGE "-%s.pid"
+#define STATEDIR            INFODIR
+#define PIDFILE             INFODIR "/" PACKAGE "-%s.pid"
 
 #ifndef INFODIR
 # define INFODIR            "/var/lib/dhcpcd"
