diff -Nur dhcpcd-3.2.3.orig/dhcpcd.8.in dhcpcd-3.2.3/dhcpcd.8.in
--- dhcpcd-3.2.3.orig/dhcpcd.8.in	2008-02-25 02:28:19.000000000 -0600
+++ dhcpcd-3.2.3/dhcpcd.8.in	2009-03-06 19:36:02.207109931 -0600
@@ -117,7 +117,7 @@
 up, down or new depending on the state of
 .Nm .
 .Nm
-ignores the exist code of the script.
+ignores the exit code of the script.
 .Ss Fine tuning
 You can fine tune the behaviour of
 .Nm
