--- a/dhcpcd.sh	2009-03-06 19:22:41.247171672 -0600
+++ b/dhcpcd.sh	2009-03-06 19:24:19.186014526 -0600
@@ -1,10 +1,10 @@
 #!/bin/sh
 #
-#  This is a sample /etc/dhcpcd.sh script.
-#  /etc/dhcpcd.sh script is executed by dhcpcd daemon
+#  This is a sample /etc/dhcpc/dhcpcd.sh script.
+#  /etc/dhcpc/dhcpcd.sh script is executed by dhcpcd daemon
 #  any time it configures or shuts down interface.
-#  The following parameters are passed to dhcpcd.exe script:
-#  $1 = HostInfoFilePath, e.g  "/var/lib/dhcpcd/dhcpcd-eth0.info"
+#  The following parameters are passed to the dhcpcd.sh script:
+#  $1 = HostInfoFilePath, e.g  "/etc/dhcpc/dhcpcd-eth0.info"
 #  $2 = "up" if interface has been configured with the same
 #       IP address as before reboot;
 #  $2 = "down" if interface has been shut down;
