--- ./shared/nm-service-defines.h.orig	2020-03-06 06:38:55.000000000 -0600
+++ ./shared/nm-service-defines.h	2020-05-03 20:12:26.997028745 -0500
@@ -126,8 +126,8 @@
 #define NM_OPENVPN_VERIFY_X509_NAME_TYPE_SUBJECT     "subject"
 
 /* User name and group to run nm-openvpn-service under */
-#define NM_OPENVPN_USER   "nm-openvpn"
-#define NM_OPENVPN_GROUP  "nm-openvpn"
+#define NM_OPENVPN_USER   "nobody"
+#define NM_OPENVPN_GROUP  "nogroup"
 #define NM_OPENVPN_CHROOT LOCALSTATEDIR "/lib/openvpn/chroot"
 
 #endif /* __NM_SERVICE_DEFINES_H__ */
