diff -Nur tar-1.23.orig//src/buffer.c tar-1.23/src/buffer.c
--- tar-1.23.orig//src/buffer.c	2010-03-09 06:52:41.000000000 -0600
+++ tar-1.23/src/buffer.c	2010-03-15 23:47:30.122668444 -0500
@@ -734,7 +734,7 @@
   more = record_start->buffer + status;
   left = record_size - status;
 
-  if (left && left % BLOCKSIZE == 0
+  /* if (left && left % BLOCKSIZE == 0
       && verbose_option
       && record_start_block == 0 && status != 0
       && archive_is_dev ())
@@ -745,7 +745,7 @@
                        "Record size = %lu blocks",
                        rsize),
              rsize));
-    }
+    } */
 
   while (left % BLOCKSIZE != 0
          || (left && status && read_full_records))
