--- ./src/buffer.c.orig	2009-03-05 01:04:13.000000000 -0600
+++ ./src/buffer.c	2009-03-09 20:17:50.000000000 -0500
@@ -688,7 +688,7 @@
   more = record_start->buffer + status;
   left = record_size - status;
 
-  if (left && left % BLOCKSIZE == 0
+  /* if (left && left % BLOCKSIZE == 0
       && verbose_option
       && record_start_block == 0 && status != 0)
     {
@@ -698,7 +698,7 @@
                        "Record size = %lu blocks",
                        rsize),
              rsize));
-    }
+    } */
 
   while (left % BLOCKSIZE != 0
          || (left && status && read_full_records))
