--- ./src/list.c.orig	2004-05-10 04:49:09.000000000 -0700
+++ ./src/list.c	2004-06-06 13:13:23.000000000 -0700
@@ -138,6 +138,14 @@
 
 	  if (!ignore_zeros_option)
 	    {
+
+	      /* 
+	       * According to POSIX tar specs, this is wrong, but on the web
+	       * there are some tar specs that can trigger this, and some tar
+	       * implementations create tars according to that spec.  For now,
+	       * let's not be pedantic about issuing the warning.
+	       */
+#if 0	       
 	      char buf[UINTMAX_STRSIZE_BOUND];
 
 	      status = read_header (false);
@@ -145,6 +153,7 @@
 		break;
 	      WARN ((0, 0, _("A lone zero block at %s"),
 		     STRINGIFY_BIGINT (current_block_ordinal (), buf)));
+#endif
 	      break;
 	    }
 	  status = prev_status;
