diff -Nur tar-1.23.orig//src/list.c tar-1.23//src/list.c
--- tar-1.23.orig//src/list.c	2010-01-26 05:28:09.000000000 -0600
+++ tar-1.23//src/list.c	2010-03-15 23:46:19.841670026 -0500
@@ -138,6 +138,14 @@
 
 	  if (!ignore_zeros_option)
 	    {
+
+             /* 
+              * According to POSIX tar specs, this is wrong, but on the web
+              * there are some tar specs that can trigger this, and some tar
+              * implementations create tars according to that spec.  For now,
+              * let's not be pedantic about issuing the warning.
+              */
+#if 0         
 	      char buf[UINTMAX_STRSIZE_BOUND];
 
 	      status = read_header (&current_header, &current_stat_info, 
@@ -147,6 +155,7 @@
 	      WARNOPT (WARN_ALONE_ZERO_BLOCK,
 		       (0, 0, _("A lone zero block at %s"),
 			STRINGIFY_BIGINT (current_block_ordinal (), buf)));
+#endif
 	      break;
 	    }
 	  status = prev_status;
