diff -Nur tar-1.27.1.orig/src/list.c tar-1.27.1/src/list.c
--- tar-1.27.1.orig/src/list.c	2013-03-14 15:18:10.000000000 -0500
+++ tar-1.27.1/src/list.c	2014-07-26 20:42:16.613004098 -0500
@@ -211,6 +211,14 @@
 
 	  if (!ignore_zeros_option)
 	    {
+
+	      /* 
+	       * According to POSIX tar specs, this is wrong, but on the web
+	       * there are some tar specs that can trigger this, and some tar
+	       * implementations create tars according to that spec.  For now,
+	       * let's not be pedantic about issuing the warning.
+	       */
+#if 0	       
 	      char buf[UINTMAX_STRSIZE_BOUND];
 
 	      status = read_header (&current_header, &current_stat_info,
@@ -220,6 +228,7 @@
 	      WARNOPT (WARN_ALONE_ZERO_BLOCK,
 		       (0, 0, _("A lone zero block at %s"),
 			STRINGIFY_BIGINT (current_block_ordinal (), buf)));
+#endif
 	      break;
 	    }
 	  status = prev_status;
