--- ./mcelog.init.orig	2010-01-20 20:36:52.000000000 -0600
+++ ./mcelog.init	2010-09-27 17:23:24.964394022 -0500
@@ -2,11 +2,6 @@
 # 
 # Startup script for mcelog
 # 
-# This should be customized for distribution standards
-# (using rc_status etc.)
-# The paths are hardcoded and are not automatically adjusted
-# for different prefix
-#
 ### BEGIN INIT INFO
 # Provides:		mcelog 
 # Default-Start:	3 5
@@ -16,8 +11,6 @@
 #                       This logs and handles CPU hardware errors on x86 systems.
 ### END INIT INFO
 
-# put this is sysconfig
-
 # mcelog mode
 # valid values: daemon, trigger, cron
 # Recommended value daemon
@@ -52,8 +45,8 @@
 case "$1" in
 start)
 	if [ "$MCELOG_MODE" = "daemon" ] ; then
-		echo "Starting mcelog daemon"
-		startproc $MCELOG --daemon $MCELOG_OPTIONS
+		echo "Starting mcelog daemon:  $MCELOG --daemon $MCELOG_OPTIONS"
+		$MCELOG --daemon $MCELOG_OPTIONS
 	elif [ -f "$TRIGGER" ] ; then
 		echo $MCELOG > "$TRIGGER"
 	else
@@ -62,8 +55,8 @@
 	;;
 stop)
 	if [ "$MCELOG_MODE" = "daemon" ] ; then
-		echo "Stopping mcelog"
-		killproc -TERM $MCELOG
+		echo "Stopping mcelog daemon:  killall -TERM $MCELOG"
+		killall -TERM $MCELOG
 	elif [ "$MCELOG_MODE" = "trigger" -a -f "$TRIGGER" ]; then 
 		echo "" > "$TRIGGER"
 	else
@@ -85,8 +78,8 @@
 	;;
 status)
 	if [ "$MCELOG_MODE" = "daemon" ] ; then
-		echo "Checking for mcelog"
-		checkproc $MCELOG
+		echo "Checking for mcelog:"
+		ps ax | grep -v grep | grep $MCELOG | cut -f 1 -d ' '
 	fi
 	;;
 *)
