--- ./CMakeLists.txt.orig	2024-07-15 15:50:34.000000000 -0500
+++ ./CMakeLists.txt	2025-01-02 13:57:21.072502116 -0600
@@ -18,15 +18,10 @@
 if (QT_VERSION_MAJOR GREATER_EQUAL 6)
     set(KF_MIN_VERSION "5.240.0")
     find_package(ECM ${KF_MIN_VERSION} NO_MODULE)
-    if (ECM_FOUND)
-        set(KF_MAJOR_VERSION "6")
-        set(QT_MIN_VERSION "6.2.2")		# openSUSE 15.4 version
-    else ()
-        MESSAGE (STATUS "KF6 not found! Trying Qt5")
-	find_package(QT NAMES Qt5 REQUIRED COMPONENTS Core)
-	MESSAGE (STATUS "Found Qt version ${QT_VERSION_MAJOR}")
-        set(KF_MIN_VERSION "5.32.0")
-    endif ()
+    MESSAGE (STATUS "KF6 not found! Trying Qt5")
+    find_package(QT NAMES Qt5 REQUIRED COMPONENTS Core)
+    MESSAGE (STATUS "Found Qt version ${QT_VERSION_MAJOR}")
+    set(KF_MIN_VERSION "5.32.0")
 endif ()
 set(QT_MAJOR_VERSION ${QT_VERSION_MAJOR})
 
