--- slack-desc.orig	2019-01-01 03:24:18.000000000 -0600
+++ slack-desc	2018-11-28 16:37:00.644976546 -0600
@@ -1,14 +1,14 @@
 # HOW TO EDIT THIS FILE:
-# The "handy ruler" below makes it easier to edit a package description.  Line
+# The "handy ruler" below makes it easier to edit a package description. Line
 # up the first '|' above the ':' following the base package name, and the '|'
-# on the right side marks the last column you can put a character in.  You must
-# make exactly 11 lines for the formatting to be correct.  It's also
+# on the right side marks the last column you can put a character in. You must
+# make exactly 11 lines for the formatting to be correct. It's also
 # customary to leave one space after the ':'.
 
     |-----handy-ruler------------------------------------------------------|
-soma: Soma - a dialog/mplayer internet radio player and recorder.
+soma: soma (a dialog/MPlayer internet radio player and recorder)
 soma:
-soma: Soma handles playing any internet audio stream supported by mplayer.
+soma: Soma handles playing any internet audio stream supported by MPlayer.
 soma: It also supports playlists such as pls,m3u and ram.
 soma:
 soma: Homepage: http://tty1.uk/slackbuilds/noversion/soma
