--- ./soma.desktop.orig	2021-02-07 12:21:29.000000000 -0600
+++ ./soma.desktop	2024-02-16 13:58:32.684887399 -0600
@@ -1,11 +1,11 @@
-[Desktop Entry]                        
-Version=1.0                            
-Type=Application                       
+[Desktop Entry]
+Version=1.0
+Type=Application
 Name=Internet Radio
-Comment=Soma - Live Internet Streams          
-Exec=soma                              
-Icon=Radio                    
-Terminal=true                          
-StartupNotify=false                    
+Comment=Soma - Live Internet Streams
+Exec=soma
+Icon=Radio
+Terminal=true
+StartupNotify=false
 Categories=Application;Network;Audio;Player;
 GenericName=Live Internet Radio Streams
