diff -Naur autofs-3.1.7/modules/lookup_file.c autofs-3.1.7-gcc/modules/lookup_file.c
--- autofs-3.1.7/modules/lookup_file.c	2000-11-04 07:05:17.000000000 +0000
+++ autofs-3.1.7-gcc/modules/lookup_file.c	2006-08-16 18:21:21.000000000 +0100
@@ -197,7 +197,7 @@
       }
       break;
     }
-  next_char:			/* End of loop, since we can't continue;
+  next_char:;			/* End of loop, since we can't continue;
 				   inside a switch */
   }
   
