From 2ed4cd66401bc1d9683c94238e9024218d7be3ba Mon Sep 17 00:00:00 2001
From: Albert Astals Cid <aacid@kde.org>
Date: Wed, 7 Apr 2021 00:14:59 +0200
Subject: [PATCH] TextOutputDev: Fix crash in malformed file

oss-fuzz/32952
---
 poppler/TextOutputDev.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/poppler/TextOutputDev.cc b/poppler/TextOutputDev.cc
index ea12b4689..956c13289 100644
--- a/poppler/TextOutputDev.cc
+++ b/poppler/TextOutputDev.cc
@@ -931,7 +931,7 @@ TextPool::~TextPool()
 int TextPool::getBaseIdx(double base) const
 {
     const double baseIdxDouble = base / textPoolStep;
-    if (baseIdxDouble < minBaseIdx) {
+    if (std::isnan(baseIdxDouble) || baseIdxDouble < minBaseIdx) {
         return minBaseIdx;
     }
     if (baseIdxDouble > maxBaseIdx) {
-- 
GitLab

