--- ./splash/SplashPath.h.orig	2018-03-18 13:23:49.000000000 -0500
+++ ./splash/SplashPath.h	2017-12-03 13:25:06.000000000 -0600
@@ -4,20 +4,6 @@
 //
 //========================================================================
 
-//========================================================================
-//
-// Modified under the Poppler project - http://poppler.freedesktop.org
-//
-// All changes made under the Poppler project to this file are licensed
-// under GPL version 2 or later
-//
-// Copyright (C) 2018 Albert Astals Cid <aacid@kde.org>
-//
-// To see a description of the changes please see the Changelog file that
-// came with your tarball or type make ChangeLog if you are building from git
-//
-//========================================================================
-
 #ifndef SPLASHPATH_H
 #define SPLASHPATH_H
 
@@ -76,9 +62,6 @@
 
   ~SplashPath();
 
-  SplashPath(const SplashPath&) = delete;
-  SplashPath& operator=(const SplashPath&) = delete;
-
   // Append <path> to <this>.
   void append(SplashPath *path);
 
@@ -133,6 +116,8 @@
 
   friend class SplashXPath;
   friend class Splash;
+  // this is a temporary hack, until we read FreeType paths directly
+  friend class ArthurOutputDev;
 };
 
 #endif
