--- docbook2x-0.8.8.orig/debian/changelog
+++ docbook2x-0.8.8/debian/changelog
@@ -0,0 +1,219 @@
+docbook2x (0.8.8-8) unstable; urgency=low
+
+  * debian/rules (binary-post-install): Fix manpages to talk about the renamed
+    commands.
+
+ -- Daniel Leidert (dale) <daniel.leidert@wgdd.de>  Sat, 06 Sep 2008 10:59:50 +0200
+
+docbook2x (0.8.8-7) unstable; urgency=low
+
+  * debian/docbook2x.doc-base (Section): Fixed accordingly to version 0.8.10
+    of the doc-base manual.
+  * debian/watch: Fixed.
+
+ -- Daniel Leidert (dale) <daniel.leidert@wgdd.de>  Tue, 08 Jul 2008 14:53:08 +0200
+
+docbook2x (0.8.8-6) unstable; urgency=low
+
+  * debian/compat: Raised to v5.
+  * debian/control: Added DM-Upload-Allowed for DM status.
+    (Build-Depends): Raised debhelper to v5.
+    (Standards-Version): Raised to 3.7.3.
+  * debian/copyright: Fixed typo. Thanks to lintian.
+  * debian/docbook2x.doc-base (Document): Fixed uppercase letter. Thanks to
+    lintian.
+  * debian/patches/05_fix_439214_error_on_missing_refentry.dpatch: Added.
+    - xslt/man/docbook.xsl: Print a warning if no refentry element can be
+      found (closes: #439214).
+  * debian/patches/00list: Adjusted.
+
+ -- Daniel Leidert (dale) <daniel.leidert@wgdd.de>  Wed, 13 Feb 2008 18:08:08 +0100
+
+docbook2x (0.8.8-5) unstable; urgency=low
+
+  * debian/control: Homepage field transition.
+  * debian/rules (DEB_CONFIGURE_EXTRA_FLAGS): Added --with-html-xsl configure
+    option to prevent docbook2X FTBFS building twice in a row. Thanks to
+    Steven Cheng, Patrick Winnertz and Lucas Nussbaum (closes: #441700).
+  * debian/watch: Added.
+  * debian/patches/01_fix_static_datadir_evaluation.dpatch: Adjusted.
+    - Removed change in configure.ac to avoid rerunning all the autotools.
+  * debian/patches/04_fix_442782_preprocessor_declaration_syntax.dpatch: Added.
+    - Fixes preprocessor declaration syntax to be '\" (closes: #442782).
+      Thanks to Colin Watson for the report and the patch.
+  * debian/patches/00list: Adjusted.
+
+ -- Daniel Leidert (dale) <daniel.leidert@wgdd.de>  Sun, 25 Nov 2007 03:40:18 +0000
+
+docbook2x (0.8.8-4) unstable; urgency=high
+
+  [ Daniel Leidert ]
+  * debian/control (Depends): Added xsltproc. Thanks to Peter Eisentraut and
+    Adeodato Simó for the hint (closes: #420345). Further added the necessary
+    sp | opensp and texinfo dependencies.
+  * debian/patches/03_fix_420153_filename_whitespace_handling.dpatch: Added.
+    - First normalize the refentrytitle content and then replace spaces with
+      underlines to create the manpage filename (closes: #420153). Thanks to
+      Peter Eisentraut for reporting this issue.
+  * debian/patches/00list: Adjusted.
+    - Added 03_fix_420153_filename_whitespace_handling.dpatch.
+
+ -- Daniel Leidert (dale) <daniel.leidert@wgdd.de>  Sat, 21 Apr 2007 21:05:46 +0200
+
+docbook2x (0.8.8-3) unstable; urgency=high
+
+  [ Daniel Leidert ]
+  * debian/control (Depends): Added ${misc:Depends} variable to fix missing
+    dependency on xml-core (closes: #419419). Thanks to Kurt Roeckx for the
+    report.
+
+ -- Daniel Leidert (dale) <daniel.leidert@wgdd.de>  Sun, 15 Apr 2007 19:10:50 +0200
+
+docbook2x (0.8.8-2) unstable; urgency=high
+
+  [ Daniel Leidert ]
+  * debian/control (Build-Depends): Added xml-core and re-added dpatch..
+  * debian/docbook2x.xmlcatalogs: Added to register XML catalogs.
+  * debian/rules: Re-added dpatch ruleset.
+    (binary-install/docbook2x): Register the XML catalogs shipped with the
+    package source (closes: #418703).
+  * debian/patches/01_fix_static_datadir_evaluation.dpatch: Added. Fixes the
+    broken ${datadir} evaluation in configure(.ac) that leads to an unusable
+    package.
+  * debian/patches/02_fix_418703_dont_use_abbreviated_sfnet_address.dpatch:
+    Added. Don't use the short sf.net URLs. This with the above issue probably
+    caused bug #418703. However, having the catalogs registered was already in
+    TODO.
+  * debian/patches/00list: Adjusted.
+
+ -- Daniel Leidert (dale) <daniel.leidert@wgdd.de>  Thu, 12 Apr 2007 16:27:10 +0200
+
+docbook2x (0.8.8-1) unstable; urgency=low
+
+  * New upstream version 0.8.8.
+  * Upload to unstable (closes: #409524).
+
+  [ Daniel Leidert ]
+  * debian/control: Added XS-Vcs-Browser and XS-Vcs-Svn fields.
+    (Build-Depends): Removed dpatch dependency.
+  * debian/rules: Removed dpatch ruleset inclusion.
+    (binary-install/docbook2x): Added the upstream changelog as well.
+  * debian/patches/00list:
+  * debian/patches/01_add_missing_documentation.dpatch: Removed. Fixed
+    upstream.
+
+ -- Daniel Leidert (dale) <daniel.leidert@wgdd.de>  Tue, 10 Apr 2007 22:54:42 +0200
+
+docbook2x (0.8.7-1) experimental; urgency=low
+
+  * New upstream release 0.8.7 (closes: #292046).
+    - This release also doesn't fail to build any longer with the latest
+      libxml-sax-perl (closes: #409524).
+
+  [ Daniel Leidert ]
+  * debian/control (Uploaders): Added myself.
+  (Build-Depends): Added dpatch, opensp, sp and xsltproc and removed
+  libxslt1-dev.
+  (Build-Conflicts): Added tidy to avoid further tests.
+  * debian/copyright: Small update.
+  * debian/docbook2x.doc-base: Added for registering the docbook2X HTML docs
+    with doc-base.
+  * debian/rules: Removed simple-patchsys.mk inclusion and made the makefile
+    smaller. Added the dpatch ruleset to apply changes.
+    (common-install-prehook-arch, binary-install): Fixed. The info files come
+    with a INFO-DIR-SECTION entry. Further there is also info documentation
+    for the xslt stylesheets.
+    (binary-post-install): Removed a lot of obsolete stuff. The Perl stuff is
+    not longer installed. No CVS content is installed.
+    (binary-post-install, DEB_CONFIGURE_EXTRA_FLAGS): Replaced the manual
+    program/man-page renaming with adding --program-transform-name to
+    configure flags.
+    (clean): Removed unused stuff and improved the rest a bit.
+  * debian/patches/01_add_missing_documentation.dpatch: Added.
+    - This patch adds the missing docbook2{man,texi}.html files to the
+      documentation.
+  * debian/patches/00list: Added. Apply patch(es).
+
+ -- Daniel Leidert (dale) <daniel.leidert@wgdd.de>  Sun, 25 Feb 2007 22:28:34 +0100
+
+docbook2x (0.8.3-1) unstable; urgency=low
+
+  * New upstream release.  Removed all the patches that used to be in
+    debian/patches, since they have been applied upstream.
+  * debian/control:
+    - In Build-Depends, dropped libxml-perl and added libxml-sax-perl.
+    - Added Build-Conflicts on libxml2-utils, because the presence of
+      /usr/bin/xmllint triggers unecessary buildings in the doc
+      directory.
+  * debian/rules: Added an extra clean target for files left after building
+    the upstream package that are not removed by make clean.  Also, make
+    sure that debian/examples is clean.
+
+ -- Rafael Laboissiere <rafael@debian.org>  Sun, 18 Jul 2004 22:24:57 +0200
+
+docbook2x (0.8.2-6) unstable; urgency=low
+
+  * debian/control: Added dependencies on ${perl:Depends}, libxml-perl,
+    and libtext-wrapi18n-perl.  The variable ${perl:Depends} is set by
+    dh_perl.
+
+ -- Rafael Laboissiere <rafael@debian.org>  Thu, 29 Apr 2004 14:32:06 +0200
+
+docbook2x (0.8.2-5) unstable; urgency=low
+
+  * debian/patches/50-malocc-return.patch: Added this patch to fix a
+    serious upstream bug which was preventing the package of being built
+    by the ia64, hppa, sparc, and powerpc build daemons.
+
+ -- Rafael Laboissiere <rafael@debian.org>  Thu, 29 Apr 2004 12:12:12 +0200
+
+docbook2x (0.8.2-4) unstable; urgency=low
+
+  * debian/control: Added build-dependency on texinfo, since the makeinfo
+    program is needed for building the package.
+
+ -- Rafael Laboissiere <rafael@debian.org>  Sun, 25 Apr 2004 21:29:15 +0200
+
+docbook2x (0.8.2-3) unstable; urgency=low
+
+  * debian/control: Added build-dependency on libxml-perl.
+
+ -- Rafael Laboissiere <rafael@debian.org>  Sun, 25 Apr 2004 17:09:09 +0200
+
+docbook2x (0.8.2-2) unstable; urgency=low
+
+  * debian/patches/50-gentext-dingbat-template.patch: Patch for
+    xslt/common/gentext.xsl, defining the lacking gentext-dingbat template
+    (closes: #245340).
+  * debian/patches/50-i-element.patch: Patch for xslt/texi/biblio.xsl,
+    fixing wrong <I> elements (closes: #245343).
+  * debian/patches/50-xml-fixes.patch: Patch for fixing XML problems in
+    the upstream files doc/docbook2X.xml and doc/faq.xml.
+  * debian/rules:
+    - Include cdbs' simple-patchsys.mk for coping with the patches above.
+    - Removed hard-coded Perl version from some installation directories
+      names.  There is now a PERL_INSTALLPRIVLIB variable, properly
+      initialized through the Perl Config module (closes: #245291).
+    - Install Perl module in standard place, complying with the Debian
+      Perl Policy (closes: #245305).
+    - Run make clean in perl/XML/Handler directory to avoid inclusion of
+      extraneous patches in the Debian diff.gz file.  Remove also
+      Makefile.old.
+  * debian/control:
+    - Added versioned build-dependency on libxslt1-dev.  Also, changed
+      Build-Depends-Indep to Build-Depends, since the package is
+      Architecture: any (closes: #245285).
+    - Cite upstream URL homepage in debian/control in accordance with the
+      Debian Best Packaging Practices manual (closes: #245447)
+    - Added Uploaders field with Ardo, Wolfgang, and Rafael.
+
+ -- Rafael Laboissiere <rafael@debian.org>  Sun, 25 Apr 2004 14:45:14 +0200
+
+docbook2x (0.8.2-1) unstable; urgency=low
+
+  * Initial package.  Closes: #185295.
+  * The file install.xml is missing from the upstream archive, but is
+    available at the web site.
+  * Beware: <qandaset> can only be handled with defaultlabel attribute.
+
+ -- W. Borgert <debacle@debian.org>  Sat, 13 Mar 2004 09:41:07 +0000
--- docbook2x-0.8.8.orig/debian/control
+++ docbook2x-0.8.8/debian/control
@@ -0,0 +1,23 @@
+Source: docbook2x
+Section: text
+Priority: optional
+Maintainer: Debian XML/SGML Group <debian-xml-sgml-pkgs@lists.alioth.debian.org>
+Uploaders: Ardo van Rangelrooij <ardo@debian.org>, W. Borgert <debacle@debian.org>, Rafael Laboissiere <rafael@debian.org>, Daniel Leidert (dale) <daniel.leidert@wgdd.de>
+Build-Depends: debhelper (>= 5), cdbs (>= 0.4.21), dpatch, libxml-sax-perl, opensp, sp, texinfo, xml-core, xsltproc
+Build-Conflicts: libxml2-utils, tidy
+Standards-Version: 3.7.3
+Homepage: http://docbook2x.sourceforge.net
+Vcs-Browser: http://svn.debian.org/wsvn/debian-xml-sgml/packages/docbook2x/trunk/
+Vcs-Svn: svn://svn.debian.org/svn/debian-xml-sgml/packages/docbook2x/
+DM-Upload-Allowed: yes
+
+Package: docbook2x
+Architecture: any
+Depends: ${shlibs:Depends}, ${misc:Depends}, ${perl:Depends}, libxml-sax-expat-perl, libtext-wrapi18n-perl, opensp | sp, texinfo, xsltproc
+Recommends: docbook-xml, docbook-xsl
+Description: Converts DocBook/XML documents into man pages and TeXinfo
+ The main tools are:
+  * docbook2x-man: converts a DocBook/XML refentry into a manual page,
+    with table support
+  * docbook2x-texi: converts a DocBook/XML refentry into an info page,
+    most usable for Emacs users
--- docbook2x-0.8.8.orig/debian/compat
+++ docbook2x-0.8.8/debian/compat
@@ -0,0 +1 @@
+5
--- docbook2x-0.8.8.orig/debian/TODO
+++ docbook2x-0.8.8/debian/TODO
@@ -0,0 +1,22 @@
+Things that should (IMHO definitely) be done in the near future:
+
+  * make the package XML policy compliant.
+
+    - move the stylesheets and DTDs into /usr/share/xml
+
+  * solve the conflict with the docbook-utils package binaries
+    (including: man-pages and docs currently use the names chosen by upstream,
+    not "our" tranformed binary names)
+
+  * the man-pages contain references to files in /usr/share
+    [.. citing Steve Cheng ..] So you have two choices:
+
+    1. You can either use a sed script afterwards to fix the paths in the
+    installed man pages (and HTML, Texinfo documentation).
+
+    2. Or, force the rebuild of the documentation at your end.
+    After running the configure --with-html-xsl, touch doc/docbook2X.xml and
+    then make.  (Note: don't use VPATH builds in this situation; it breaks.)
+    Of course you will need the xsltproc and the DocBook XSL stylesheets for
+    this procedure. [..]
+
--- docbook2x-0.8.8.orig/debian/watch
+++ docbook2x-0.8.8/debian/watch
@@ -0,0 +1,4 @@
+# See uscan(1) for format
+
+version=3
+http://sf.net/docbook2x/docbook2X-([\d\.]+)\.tar\.gz
--- docbook2x-0.8.8.orig/debian/docbook2x.doc-base
+++ docbook2x-0.8.8/debian/docbook2x.doc-base
@@ -0,0 +1,13 @@
+Document: docbook2x
+Title: docbook2X user documentation
+Author: Steve Cheng <stevecheng@users.sourceforge.net>
+Abstract: docbook2X converts DocBook documents into man pages and
+ Texinfo documents. This documentation contains the man-pages, a
+ quick start guide, the FAQ and some information about the
+ project (history).
+Section: Text
+
+Format: HTML
+Index: /usr/share/doc/docbook2x/docbook2X/docbook2X.html
+Files: /usr/share/doc/docbook2x/docbook2X/*.html
+
--- docbook2x-0.8.8.orig/debian/docbook2x.xmlcatalogs
+++ docbook2x-0.8.8/debian/docbook2x.xmlcatalogs
@@ -0,0 +1,19 @@
+root;public;-//Steve Cheng//DTD
+root;public;-//Steve Cheng//DOCUMENT DocBook
+
+root;system;http://docbook2x.sf.net/latest/
+root;system;http://docbook2x.sourceforge.net/latest/
+
+local;dtd/catalog.xml;/usr/share/docbook2X/dtd/catalog.xml
+local;xslt/catalog.xml;/usr/share/docbook2X/xslt/catalog.xml
+
+package;public;-//Steve Cheng//DTD Man-XML V0.8.6//EN;/usr/share/docbook2X/dtd/catalog.xml
+package;public;-//Steve Cheng//DTD Texi-XML V0.8.6//EN;/usr/share/docbook2X/dtd/catalog.xml
+package;system;http://docbook2x.sourceforge.net/latest/dtd/;/usr/share/docbook2X/dtd/catalog.xml
+package;system;http://docbook2x.sf.net/latest/dtd/;/usr/share/docbook2X/dtd/catalog.xml
+
+package;public;-//Steve Cheng//DOCUMENT DocBook Texinfo Stylesheet//EN;/usr/share/docbook2X/xslt/catalog.xml
+package;public;-//Steve Cheng//DOCUMENT DocBook Man-page Stylesheet//EN;/usr/share/docbook2X/xslt/catalog.xml
+package;system;http://docbook2x.sourceforge.net/latest/xslt/;/usr/share/docbook2X/xslt/catalog.xml
+package;system;http://docbook2x.sf.net/latest/xslt/;/usr/share/docbook2X/xslt/catalog.xml
+
--- docbook2x-0.8.8.orig/debian/docbook2x.examples
+++ docbook2x-0.8.8/debian/docbook2x.examples
@@ -0,0 +1 @@
+debian/examples/*
--- docbook2x-0.8.8.orig/debian/copyright
+++ docbook2x-0.8.8/debian/copyright
@@ -0,0 +1,67 @@
+This package was debianized by W. Borgert <debacle@debian.org> on
+Sat, 2004-03-13 17:04:55 GMT 2004.
+
+It was downloaded from Sourceforge:
+
+  <URL:http://docbook2x.sourceforge.net/>
+
+Upstream Author:
+
+  Steve Cheng <stevecheng@users.sourceforge.net>
+
+Copyright:
+
+  Copyright (C) 2000-2006 Steve Cheng <stevecheng@users.sourceforge.net>.
+
+  Parts have been derived from the docbook-xsl stylesheet distribution, that
+  is:
+
+  Copyright (C) 1999, 2000, 2001, 2002 Norman Walsh.
+
+License and Warranty for docbook2X:
+
+  Permission is hereby granted, free of charge, to any person obtaining
+  a copy of this software and associated documentation files (the
+  "Software"), to deal in the Software without restriction, including
+  without limitation the rights to use, copy, modify, merge, publish,
+  distribute, sublicense, and/or sell copies of the Software, and to
+  permit persons to whom the Software is furnished to do so, subject to
+  the following conditions:
+  
+  The above copyright notice and this permission notice shall be included
+  in all copies or substantial portions of the Software.
+  
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+  MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+  IN NO EVENT SHALL THE AUTHOR(S) BE LIABLE FOR ANY CLAIM, DAMAGES OR
+  OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+  ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+  OTHER DEALNGS IN THE SOFTWARE.
+
+License and Warranty for derived parts:
+
+  Permission is hereby granted, free of charge, to any person
+  obtaining a copy of this software and associated documentation
+  files (the ``Software''), to deal in the Software without
+  restriction, including without limitation the rights to use,
+  copy, modify, merge, publish, distribute, sublicense, and/or
+  sell copies of the Software, and to permit persons to whom the
+  Software is furnished to do so, subject to the following
+  conditions:
+  
+  The above copyright notice and this permission notice shall be
+  included in al copies or substantial portions of the Software.
+  
+  Except as contained in this notice, the names of individuals
+  credited with contribution to this software shall not be used in
+  advertising or otherwise to promote the sale, use or other
+  dealings in this Software without prior written authorization
+  from the individuals in question.
+  
+  Any stylesheet derived from this Software that is publicly
+  distributed will be identified with a different name and the
+  version strings in any derived Software will be changed s that
+  no possibility of confusion between the derived package and this
+  Software will exist.
+
--- docbook2x-0.8.8.orig/debian/rules
+++ docbook2x-0.8.8/debian/rules
@@ -0,0 +1,38 @@
+#!/usr/bin/make -f
+
+include /usr/share/cdbs/1/rules/debhelper.mk
+include /usr/share/cdbs/1/rules/dpatch.mk
+include /usr/share/cdbs/1/class/autotools.mk
+
+
+PACKAGE := docbook2x
+
+DEB_CONFIGURE_EXTRA_FLAGS := --program-transform-name="s/^docbook2/docbook2x-/" --with-html-xsl
+
+
+common-install-prehook-arch::
+	sed "s/^INFO-DIR-SECTION Document Preparation/INFO-DIR-SECTION Text/" \
+	    doc/docbook2X.info > debian/docbook2X.info
+	sed "s/^INFO-DIR-SECTION Document Preparation/INFO-DIR-SECTION Text/" \
+	    xslt/documentation/docbook2man-xslt.info > debian/docbook2man-xslt.info
+	sed "s/^INFO-DIR-SECTION Document Preparation/INFO-DIR-SECTION Text/" \
+	    xslt/documentation/docbook2texi-xslt.info > debian/docbook2texi-xslt.info
+
+binary-install/docbook2x::
+	dh_installinfo -pdocbook2x debian/docbook2X.info \
+	                           debian/docbook2man-xslt.info \
+	                           debian/docbook2texi-xslt.info
+	dh_installchangelogs doc/changes.html
+	dh_installxmlcatalogs
+
+binary-post-install/docbook2x::
+	mv $(CURDIR)/debian/$(PACKAGE)/usr/share/doc/docbook2X \
+	   $(CURDIR)/debian/$(PACKAGE)/usr/share/doc/docbook2x
+	sed -i -e 's/docbook2man/docbook2x\\-man/g' \
+	   $(CURDIR)/debian/$(PACKAGE)/usr/share/man/man1/docbook2x-man.1
+	sed -i -e 's/docbook2texi/docbook2x\\-texi/g' \
+	   $(CURDIR)/debian/$(PACKAGE)/usr/share/man/man1/docbook2x-texi.1
+
+clean::
+	make -C debian/examples clean
+	rm -f debian/docbook2X.info
--- docbook2x-0.8.8.orig/debian/patches/00list
+++ docbook2x-0.8.8/debian/patches/00list
@@ -0,0 +1,5 @@
+01_fix_static_datadir_evaluation
+02_fix_418703_dont_use_abbreviated_sfnet_address
+03_fix_420153_filename_whitespace_handling
+04_fix_442782_preprocessor_declaration_syntax
+05_fix_439214_error_on_missing_refentry
--- docbook2x-0.8.8.orig/debian/patches/03_fix_420153_filename_whitespace_handling
+++ docbook2x-0.8.8/debian/patches/03_fix_420153_filename_whitespace_handling
@@ -0,0 +1,43 @@
+#! /bin/sh /usr/share/dpatch/dpatch-run
+## 03_fix_420153_filename_whitespace_handling.dpatch by
+##     Daniel Leidert (dale) <daniel.leidert@wgdd.de>
+##
+## All lines beginning with `## DP:' are a description of the patch.
+## DP: Peter Eisentraut reported a regression in the whitespace handling of
+## DP: refentrytitle content during filename creation:
+## DP: http://bugs.debian.org/420153. The problem is, that upstream first
+## DP: replaces all spaces (but not linebreaks btw) with underlines and then
+## DP: it tries to normalize the result. This means, that a linebreak with
+## DP: additional whitespaces results in manpage names like 'foo_ ____bar.9'.
+## DP: So what we basically do in this patch is, that we first normalize the
+## DP: refentrytitle and then replace any spaces left with underlines.
+
+@DPATCH@
+diff -urNad docbook2x-0.8.8~/xslt/man/manpage.xsl docbook2x-0.8.8/xslt/man/manpage.xsl
+--- docbook2x-0.8.8~/xslt/man/manpage.xsl	2006-04-20 15:45:55.000000000 +0200
++++ docbook2x-0.8.8/xslt/man/manpage.xsl	2007-04-20 16:19:28.000000000 +0200
+@@ -30,7 +30,7 @@
+ 
+ <xsl:template name="manpage-filename">
+   <xsl:param name="filename" />
+-  <xsl:value-of select="normalize-space(translate($filename, &quot; /&quot;, &quot;__&quot;))" />
++  <xsl:value-of select="translate(normalize-space($filename), ' /', '__')" />
+ </xsl:template>
+ 
+ 
+diff -urNad docbook2x-0.8.8~/xslt/man/refentry.xsl docbook2x-0.8.8/xslt/man/refentry.xsl
+--- docbook2x-0.8.8~/xslt/man/refentry.xsl	2006-04-21 04:39:55.000000000 +0200
++++ docbook2x-0.8.8/xslt/man/refentry.xsl	2007-04-20 16:21:53.000000000 +0200
+@@ -38,7 +38,11 @@
+ <xsl:template name="refentry-filename">
+   <xsl:param name="title" />
+ 
+-  <xsl:variable name="title2" select="translate($title, &quot; /&quot;, &quot;__&quot;)" />
++  <xsl:variable name="title2">
++    <xsl:call-template name="manpage-filename">
++      <xsl:with-param name="filename" select="$title" />
++    </xsl:call-template>
++  </xsl:variable>
+ 
+   <!-- not using gentext here since man page names tend not to have
+        accented chars / non-Latin chars ...
--- docbook2x-0.8.8.orig/debian/patches/02_fix_418703_dont_use_abbreviated_sfnet_address.dpatch
+++ docbook2x-0.8.8/debian/patches/02_fix_418703_dont_use_abbreviated_sfnet_address.dpatch
@@ -0,0 +1,27 @@
+#! /bin/sh /usr/share/dpatch/dpatch-run
+## 02_fix_418703_dont_use_abbreviated_sfnet_address.dpatch by Daniel Leidert (dale) <daniel.leidert@wgdd.de>
+##
+## All lines beginning with `## DP:' are a description of the patch.
+## DP: Ondrej Certik reported a resolver issue: http://bugs.debian.org/418703.
+## DP: The error seems to be caused by using the abbreviated sf.net URLs. But
+## DP: it is possible, that this issue only occurs together with the issue
+## DP: described in 01_fix_static_datadir_evaluation.dpatch, because the path
+## DP: to the catalog also suffers from this issue.
+
+@DPATCH@
+diff -urNad docbook2x-0.8.8~/perl/db2x_xsltproc.pl docbook2x-0.8.8/perl/db2x_xsltproc.pl
+--- docbook2x-0.8.8~/perl/db2x_xsltproc.pl	2004-08-18 16:21:52.000000000 +0200
++++ docbook2x-0.8.8/perl/db2x_xsltproc.pl	2007-04-12 16:07:20.000000000 +0200
+@@ -110,10 +110,10 @@
+     
+     if($options->{'stylesheet'} eq 'texi') {
+         $options->{'stylesheet'} = 
+-            "http://docbook2x.sf.net/latest/xslt/texi/docbook.xsl";
++            "http://docbook2x.sourceforge.net/latest/xslt/texi/docbook.xsl";
+     } elsif($options->{'stylesheet'} eq 'man') {
+         $options->{'stylesheet'} = 
+-            "http://docbook2x.sf.net/latest/xslt/man/docbook.xsl";
++            "http://docbook2x.sourceforge.net/latest/xslt/man/docbook.xsl";
+     }
+ 
+     if(scalar(@argv) != 1) {
--- docbook2x-0.8.8.orig/debian/patches/04_fix_442782_preprocessor_declaration_syntax.dpatch
+++ docbook2x-0.8.8/debian/patches/04_fix_442782_preprocessor_declaration_syntax.dpatch
@@ -0,0 +1,90 @@
+#! /bin/sh /usr/share/dpatch/dpatch-run
+## 04_fix_442782_preprocessor_declaration_syntax.dpatch by Colin Watson <cjwatson@debian.org>.
+##
+## All lines beginning with `## DP:' are a description of the patch.
+## DP: Conventionally, preprocessor declarations should start with
+## DP:       '\"
+## DP:   rather than
+## DP:       .\"
+## DP:   Current man-db only supports the first (and recommended) syntax. So
+## DP:   Colin Watson provided this patch to fix docbook2x.
+## DP:
+## DP: <URL:http://bugs.debian.org/442782>
+## DP: <URL:http://lists.gnu.org/archive/html/groff/2007-11/msg00023.html>
+
+@DPATCH@
+diff -urNad trunk~/perl/db2x_manxml.pl trunk/perl/db2x_manxml.pl
+--- trunk~/perl/db2x_manxml.pl	2006-04-22 17:21:32.000000000 +0200
++++ trunk/perl/db2x_manxml.pl	2007-11-24 01:27:37.000000000 +0100
+@@ -342,6 +342,25 @@
+     $self->{line_start} = 1;
+ }
+ 
++#
++# Print a comment in the output, without causing a break.
++# Params: comment - the comment text.  
++#                   May use any characters; they need not be escaped.
++#
++sub comment_nobreak
++{
++    my ($self, $comment) = @_;
++    $self->write("\n") unless $self->{line_start};
++
++    foreach my $line (split(/\n/, $comment)) {
++        $self->write('\'\" ');
++        $self->write($line);
++        $self->write("\n");
++    }
++
++    $self->{line_start} = 1;
++}
++
+ 
+ #
+ # Use a roff "escape" i.e. commands embedded in text starting with \
+@@ -510,16 +529,20 @@
+ 
+     $self->{'adjust-stack'} = [ 'b' ];
+ 
+-    $self->{rw}->comment($elem->attr('preprocessors'))
+-        if($elem->attr('preprocessors') ne '');
+-
++    my $preprocessors = $elem->attr('preprocessors');
+     # I've dug through the Internet to see if there was any
+     # standard way to specify encoding with man pages.
+     # The following seems to be a reasonable proposal:
+     # <URL:http://mail.nl.linux.org/linux-utf8/2001-04/msg00168.html>
+     my $encoding = $self->{options}->{'encoding'};
+     $encoding =~ s#//TRANSLIT$##i;
+-    $self->{rw}->comment("-*- coding: $encoding -*-");
++    $encoding = "-*- coding: $encoding -*-";
++    if ($preprocessors eq '') {
++        $preprocessors = $encoding;
++    } else {
++        $preprocessors = "$preprocessors $encoding";
++    }
++    $self->{rw}->comment_nobreak($preprocessors);
+     
+     # Define escapes for switching to and from monospace fonts (groff only)
+     $self->{rw}->request(qw{ .if \n(.g .ds T< \\\\FC});
+diff -urNad trunk~/xslt/backend/db2x_manxml.xsl trunk/xslt/backend/db2x_manxml.xsl
+--- trunk~/xslt/backend/db2x_manxml.xsl	2006-04-23 16:44:52.000000000 +0200
++++ trunk/xslt/backend/db2x_manxml.xsl	2007-11-24 01:27:37.000000000 +0100
+@@ -528,7 +528,7 @@
+       <exslt:document method="text" 
+                       encoding="{$encoding}" 
+                       href="{$path}">
+-        <xsl:text>.\" -*- coding: </xsl:text>
++        <xsl:text>'\" -*- coding: </xsl:text>
+         <xsl:value-of select="$encoding" />
+         <xsl:text> -*-&#10;</xsl:text>
+         <xsl:copy-of select="$content" />
+@@ -538,7 +538,7 @@
+       <saxon:output method="text" 
+                     encoding="{$encoding}" 
+                     href="{$path}">
+-        <xsl:text>.\" -*- coding: </xsl:text>
++        <xsl:text>'\" -*- coding: </xsl:text>
+         <xsl:value-of select="$encoding" />
+         <xsl:text> -*-&#10;</xsl:text>
+         <xsl:copy-of select="$content" />
--- docbook2x-0.8.8.orig/debian/patches/01_fix_static_datadir_evaluation.dpatch
+++ docbook2x-0.8.8/debian/patches/01_fix_static_datadir_evaluation.dpatch
@@ -0,0 +1,20 @@
+#! /bin/sh /usr/share/dpatch/dpatch-run
+## 01_fix_static_datadir_evaluation.dpatch by Daniel Leidert (dale) <daniel.leidert@wgdd.de>
+##
+## All lines beginning with `## DP:' are a description of the patch.
+## DP: The evaluation of datadir results in "${prefix}/share" without
+## DP: evaluation of the ${prefix} variable with autoconf 2.60.
+
+@DPATCH@
+diff -urNad docbook2x-0.8.8~/configure docbook2x-0.8.8/configure
+--- docbook2x-0.8.8~/configure	2007-04-12 15:16:48.000000000 +0200
++++ docbook2x-0.8.8/configure	2007-04-12 16:04:16.000000000 +0200
+@@ -5265,7 +5265,7 @@
+ test "x$exec_prefix" = xNONE && exec_prefix='${prefix}'
+ 
+ 
+-eval static_datadir="$datadir"
++eval eval static_datadir="$datadir"
+ 
+ eval static_bindir="$bindir"
+ old_val=""
--- docbook2x-0.8.8.orig/debian/patches/05_fix_439214_error_on_missing_refentry.dpatch
+++ docbook2x-0.8.8/debian/patches/05_fix_439214_error_on_missing_refentry.dpatch
@@ -0,0 +1,33 @@
+#! /bin/sh /usr/share/dpatch/dpatch-run
+## 05_fix_439214_error_on_missing_refentry.dpatch by Daniel Leidert <daniel.leidert@wgdd.de>
+##
+## All lines beginning with `## DP:' are a description of the patch.
+## DP: If there is no refentry element, the manpage stylesheets silently
+## DP:   "ignores" this. It has been requested to print at least a
+## DP:   warning.
+## DP:
+## DP: <URL:http://bugs.debian.org/439214>
+## DP: <URL:http://lists.gnu.org/archive/html/groff/2007-11/msg00023.html>
+
+@DPATCH@
+diff -urNad trunk~/xslt/man/docbook.xsl trunk/xslt/man/docbook.xsl
+--- trunk~/xslt/man/docbook.xsl	2006-04-11 21:00:19.000000000 +0200
++++ trunk/xslt/man/docbook.xsl	2008-02-05 03:31:48.000000000 +0100
+@@ -111,11 +111,15 @@
+     <xsl:when test="child::refentry">
+       <xsl:apply-templates />
+     </xsl:when>
+-
+-    <xsl:otherwise>
++    <xsl:when test="descendant-or-self::refentry">
+       <manpageset>
+         <xsl:apply-templates select="descendant-or-self::refentry" />
+       </manpageset>
++    </xsl:when>
++    <xsl:otherwise>
++      <xsl:message terminate="no">
++        <xsl:text>WARNING: Sorry, but I cannot find a refentry element in your source!</xsl:text>
++      </xsl:message>
+     </xsl:otherwise>
+   </xsl:choose>
+ </xsl:template>
--- docbook2x-0.8.8.orig/debian/examples/ttcn-el.dbk
+++ docbook2x-0.8.8/debian/examples/ttcn-el.dbk
@@ -0,0 +1,178 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"
+"http://www.oasis-open.org/docbook/xml/4.2/docbookx.dtd" [
+  <!ENTITY debian "<productname>Debian GNU/Linux</productname>">
+  <!ENTITY emacs  "<abbrev>GNU Emacs</abbrev>">
+  <!ENTITY ttcn3  "<acronym>TTCN-3</acronym>">
+  <!ENTITY ttcnmp "<acronym>TTCN.MP</acronym>">
+]>
+
+<!-- Manual page for ttcn-el, DocBook source file (C) 2000 W. Borgert
+     debacle@debian.org
+
+     $Id: ttcn-el.dbk,v 1.1 2007/02/24 06:33:26 dleidert-guest Exp $ -->
+
+<refentry>
+  <refentryinfo>
+    <title>&emacs; modes for protocol testing languages</title>
+    <titleabbrev>TTCN-3 mode</titleabbrev>
+    <address>
+      <email>debacle@debian.org</email>
+    </address>
+    <author>
+      <firstname>W.</firstname>
+      <surname>Borgert</surname>
+    </author>
+    <copyright>
+      <year>2000</year>
+      <holder>W. Borgert</holder>
+    </copyright>
+    <date>2000-07-28</date>
+  </refentryinfo>
+  <refmeta>
+    <refentrytitle>ttcn-el</refentrytitle>
+
+    <manvolnum>7</manvolnum>
+    <refmiscinfo role="infodirsection">Development</refmiscinfo>
+  </refmeta>
+  <refnamediv>
+    <refname>ttcn-el</refname>
+
+    <refpurpose>&emacs; modes for protocol testing languages.</refpurpose>
+  </refnamediv>
+  <refsynopsisdiv>
+    <cmdsynopsis>
+      <command>ttcn-3-mode</command>
+    </cmdsynopsis>
+    <cmdsynopsis>
+      <command>ttcn-mode</command>
+    </cmdsynopsis>
+    <cmdsynopsis>
+      <command>tm-functions</command>
+    </cmdsynopsis>
+  </refsynopsisdiv>
+  <refsect1>
+    <title>Description</title>
+
+    <para>The programming modes <command>ttcn-3-mode</command>,
+      <command>ttcn-mode</command>, and
+      <command>tm-functions</command> for &emacs; are for programming
+      in the &ttcn3;, core language (Testing and Test Control Notation
+      - 3), &ttcnmp; format (Tree and Tabulart Combined Notation), or
+      the Test Manager scripts, an enhancement to the Forth
+      programming language, used in the Tektronix
+      <productname>K1297</productname> protocol tester.</para>
+
+    <para>The &ttcn3; programming mode
+      (<command>ttcn-3-mode</command>, file
+      <filename>ttcn3.el</filename>) does indentation, code
+      colourisation (so-called font-locking) and gives you an index of
+      test cases and other important definitions via the
+      <guimenu>Module-Index</guimenu>.  With the command
+      <command>ttcn3-parse</command> you can run an external
+      TTCN3Parser or ttthreeparser, if available.  The keywords are
+      according to the TTCN-3 BNF of 2001-10.  To switch off the logo
+      splash screen, set the variable
+      <varname>ttcn3-show-logo</varname> to <constant>nil</constant>
+      in your <filename>~/.emacs</filename> file.  The logo is by
+      Patrick Andrieu, the ASCII art version by W. Borgert.  This mode
+      is based on <productname>CC Mode</productname>, please look in
+      the <productname>CC Mode</productname> documentation for further
+      information.</para>
+
+    <para>The &ttcnmp; programming mode (<command>ttcn-mode</command>,
+      file <filename>ttcn.el</filename>) is not for writing test
+      suites from scratch, but for fixing small errors or doing little
+      changes or just looking at the MP format.  No indentation is
+      done, but some code colourisation, an index, and a menu with
+      some navigation functions are given.  Note: For large test
+      suites startup might become very slow.</para>
+
+    <para>If one want that &emacs; treats files with the
+      <filename>.mp</filename> extension automatically as
+       &ttcnmp; files instead of MetaPost files, add the following
+      lines to your <filename>~/.emacs</filename> file:
+	<programlisting>(setq auto-mode-alist
+	  (cons '("\\.mp\\'" . ttcn-mode) auto-mode-alist))</programlisting>
+    </para>
+
+    <para>The Test Manager programming functions
+      (<command>tm-functions</command>, file <filename>tm.el</filename>)is
+      an addition to the Forth programming mode
+      (<command>forth-mode</command>, file
+      <filename>forth.el</filename>).  It includes only minor
+      additions to the regular Forth mode, i.e. font-locking, index
+      menu, and support of the Test Manager specific words.</para>
+
+    <para>To enable the <command>tm-functions</command> by default,
+      add the following line to your
+      <filename>~/.emacs</filename>:
+      <programlisting>(add-hook 'forth-mode-hook 'tm-functions)
+      </programlisting></para>
+
+  </refsect1>
+  <refsect1>
+    <title>Files</title>
+
+    <para>All lisp source files are in
+      <filename>/usr/share/emacs/site-lisp/ttcn-el/</filename>.</para>
+  </refsect1>
+  <refsect1>
+    <title>Author</title>
+
+      <para>These programmes are made by <author><firstname>W.</firstname>
+	<surname>Borgert</surname></author>
+	&lt;<email>debacle@debian.org</email>&gt;, as well as this
+	documentation.</para>
+
+  </refsect1>
+  <refsect1>
+    <title>Thanks</title>
+
+    <para>All important ideas and some code are taken from other
+      &emacs; modes, and adapted to work with TTCN-3, &ttcnmp; and
+      Test Manager files.  Thanks to Csaba Koppany and Theofanis
+      Vassiliou-Gioles for valuable hints on current TTCN-3 keywords
+      and regular expressions.</para>
+
+  </refsect1>
+  <refsect1>
+    <title>Copyright</title>
+
+    <para>Copyright 2000 W. Borgert and released under the
+      <acronym>GNU</acronym> General Public License
+      (<abbrev>GPL</abbrev>).  Permission is granted to copy,
+      distribute and/or modify this document under the terms of the
+      <acronym>GNU</acronym> Free Documentation License, Version 1.1;
+      with no Invariant Sections, no Front-Cover Texts and no
+      Back-Cover Texts.</para>
+  </refsect1>
+  <refsect1>
+    <title>Bugs</title>
+
+    <para>The indentation is not correct in many cases.</para>
+  </refsect1>
+  <refsect1>
+    <title>History</title>
+
+    <para>Version 0.6 released on 2001-10-10.  Updated
+      keywords.</para>
+
+    <para>Version 0.5 released on 2001-05-05.  Updated
+      keywords.</para>
+
+    <para>Version 0.4 released on 2001-02-02.  Calls external
+      parser.</para>
+
+    <para>Version 0.3 released on 2001-01-31.
+      <filename>ttcn3.el</filename> should now work with GNU Emacs
+      21.0 and XEmacs 21.1, too.</para>
+
+    <para>Version 0.2 released on 2000-07-28.  Includes updates for
+      &ttcn3; keywords in <filename>ttcn3.el</filename> and makes
+      <filename>tm.el</filename> a function library instead of a major
+      mode.</para>
+
+    <para>Version 0.1 released on 2000-03-31.  Initial release.</para>
+  </refsect1>
+</refentry>
--- docbook2x-0.8.8.orig/debian/examples/Makefile
+++ docbook2x-0.8.8/debian/examples/Makefile
@@ -0,0 +1,29 @@
+# -*- makefile -*-
+
+# Notes:
+# 1. you must have package libxml2-python2.3 installed
+# 2. your document must have a titleabbrev in the refentryinfo
+# 3. your document must have a refmiscinfo with role=infodirsection
+
+XTR=python -c 'import sys, libxml2; \
+print libxml2.parseFile(sys.argv[1]).xpathEval(sys.argv[2])[0].content'
+
+all: $(patsubst %.dbk,%.info,$(wildcard *.dbk))
+
+%.info: %.dbk
+	docbook2x-texi --info --to-stdout $< | \
+	  awk -v base=`basename $< .dbk` \
+	    -v sect="`${XTR} $< '//refmiscinfo[@role="infodirsection"]'`" \
+	    -v title="`${XTR} $< '/refentry/refentryinfo/titleabbrev'`" \
+	    'BEGIN { p=1 } \
+	    /^START-INFO-DIR-ENTRY/ { print \
+	        "INFO-DIR-SECTION " sect "\n" $$0;  p=0 } \
+	    /^END-INFO-DIR-ENTRY/ { p=1 } \
+	    /^\* / { if (p==0) { gsub(/.*\./, ""); \
+	        print "* " title ": (" base ")." $$0 } } \
+	    /^File: stdout,/ { if (p==1) { \
+	        gsub(/File: stdout/, "File: " base ".info"); p=2 } } \
+	    { if (p) { print } }' > $@
+
+clean::
+	rm -f *.info *~
