--- docbook-4.5.orig/debian/patches/00list
+++ docbook-4.5/debian/patches/00list
@@ -0,0 +1,4 @@
+01_remove_entity_sets_and_dcl_from_catalogs
+10_fix_101779_mention_artheader_to_articleinfo_change
+11_fix_dbpoolx_mod_header_comment_pubid
+12_fix_256119_and_turn_on_sgml_mode
--- docbook-4.5.orig/debian/patches/01_remove_entity_sets_and_dcl_from_catalogs.dpatch
+++ docbook-4.5/debian/patches/01_remove_entity_sets_and_dcl_from_catalogs.dpatch
@@ -0,0 +1,567 @@
+#! /bin/sh /usr/share/dpatch/dpatch-run
+## 01_remove_entity_sets_and_dcl_from_catalogs.dpatch by
+##     Mark Eichin <eichin@kitten.gen.ma.us> and Adam Di Carlo <aph@debian.org>.
+##
+## All lines beginning with `## DP:' are a description of the patch.
+## DP: This patch:
+## DP: * removes all SGML declarations from the catalogs, for more info, see
+## DP:   debian/README.docbook.dcl
+## DP: * removes the ISO entity pointers since they are supplied by sgml-data
+## DP: * removes Public-IDs for DocBook XML
+## DP: * forces Public-ID rather than the System-ID in DTD declarations
+## DP:   (closes: <URL:http://bugs.debian.org/97330>)
+## DP:
+## DP: <URL:http://packages.debian.org/changelogs/pool/main/d/docbook/current/changelog#versionversion4.4-1>
+## DP: <URL:http://packages.debian.org/changelogs/pool/main/d/docbook/current/changelog#versionversion4.1-5>
+## DP: <URL:http://packages.debian.org/changelogs/pool/main/d/docbook/current/changelog#versionversion4.1-2>
+## DP: <URL:http://packages.debian.org/changelogs/pool/main/d/docbook/current/changelog#versionversion3.0-4>
+
+@DPATCH@
+diff -urNad docbook-4.5~/docbook-2.4.1/docbook.cat docbook-4.5/docbook-2.4.1/docbook.cat
+--- docbook-4.5~/docbook-2.4.1/docbook.cat	1996-04-16 08:53:16.000000000 +0200
++++ docbook-4.5/docbook-2.4.1/docbook.cat	2007-03-09 16:39:46.000000000 +0100
+@@ -27,8 +27,13 @@
+   -- ...................................................................... --
+   -- SGML declaration associated with DocBook ............................. --
+ 
+-SGMLDECL "docbook.dcl"
+-DTDDECL "-//Davenport//DTD DocBook V2.4.1//EN" "docbook.dcl"
++-- disabled, see README.docbook.dcl --
++-- SGMLDECL "docbook.dcl" --
++-- DTDDECL "-//Davenport//DTD DocBook V2.4.1//EN" "docbook.dcl" --
++
++  -- ...................................................................... --
++  -- Prefer PUBID to SYSID ................................................ --
++OVERRIDE YES
+ 
+   -- ...................................................................... --
+   -- DocBook driver file .................................................. --
+@@ -43,28 +48,5 @@
+ PUBLIC "-//Davenport//ELEMENTS DocBook Document Hierarchy V2.4.1//EN" "dbhier.mod"
+ PUBLIC "-//Davenport//ELEMENTS DocBook Additional General Entities//EN" "dbgenent.mod"
+ 
+-  -- ...................................................................... --
+-  -- ISO entity sets ...................................................... --
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Diacritical Marks//EN" "iso-dia.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Numeric and Special Graphic//EN" "iso-num.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Publishing//EN" "iso-pub.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES General Technical//EN" "iso-tech.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Latin 1//EN" "iso-lat1.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Latin 2//EN" "iso-lat2.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Greek Letters//EN" "iso-grk1.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Monotoniko Greek//EN" "iso-grk2.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Greek Symbols//EN" "iso-grk3.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Alternative Greek Symbols//EN" "iso-grk4.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Arrow Relations//EN" "iso-amsa.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Binary Operators//EN" "iso-amsb.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Delimiters//EN" "iso-amsc.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Negated Relations//EN" "iso-amsn.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Ordinary//EN" "iso-amso.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Relations//EN" "iso-amsr.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Box and Line Drawing//EN" "iso-box.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Russian Cyrillic//EN" "iso-cyr1.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Non-Russian Cyrillic//EN" "iso-cyr2.gml"
+-
+   -- End of DocBook catalog data V2.4.1 ................................... --
+   -- ...................................................................... --
+diff -urNad docbook-4.5~/docbook-3.0/docbook.cat docbook-4.5/docbook-3.0/docbook.cat
+--- docbook-4.5~/docbook-3.0/docbook.cat	1996-12-10 16:35:22.000000000 +0100
++++ docbook-4.5/docbook-3.0/docbook.cat	2007-03-09 16:44:17.000000000 +0100
+@@ -27,7 +27,12 @@
+   -- ...................................................................... --
+   -- SGML declaration associated with DocBook ............................. --
+ 
+-DTDDECL "-//Davenport//DTD DocBook V3.0//EN" "docbook.dcl"
++-- disabled, see README.docbook.dcl --
++-- DTDDECL "-//Davenport//DTD DocBook V3.0//EN" "docbook.dcl" --
++
++  -- ...................................................................... --
++  -- Prefer PUBID to SYSID ................................................ --
++OVERRIDE YES
+ 
+   -- ...................................................................... --
+   -- DocBook driver file .................................................. --
+@@ -42,28 +47,5 @@
+ PUBLIC "-//Davenport//ELEMENTS DocBook Document Hierarchy V3.0//EN" "dbhier.mod"
+ PUBLIC "-//Davenport//ENTITIES DocBook Additional General Entities V3.0//EN" "dbgenent.mod"
+ 
+-  -- ...................................................................... --
+-  -- ISO entity sets ...................................................... --
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Diacritical Marks//EN" "iso-dia.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Numeric and Special Graphic//EN" "iso-num.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Publishing//EN" "iso-pub.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES General Technical//EN" "iso-tech.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Latin 1//EN" "iso-lat1.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Latin 2//EN" "iso-lat2.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Greek Letters//EN" "iso-grk1.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Monotoniko Greek//EN" "iso-grk2.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Greek Symbols//EN" "iso-grk3.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Alternative Greek Symbols//EN" "iso-grk4.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Arrow Relations//EN" "iso-amsa.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Binary Operators//EN" "iso-amsb.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Delimiters//EN" "iso-amsc.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Negated Relations//EN" "iso-amsn.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Ordinary//EN" "iso-amso.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Relations//EN" "iso-amsr.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Box and Line Drawing//EN" "iso-box.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Russian Cyrillic//EN" "iso-cyr1.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Non-Russian Cyrillic//EN" "iso-cyr2.gml"
+-
+   -- End of catalog data for DocBook V3.0 ................................. --
+   -- ...................................................................... --
+diff -urNad docbook-4.5~/docbook-3.1/docbook.cat docbook-4.5/docbook-3.1/docbook.cat
+--- docbook-4.5~/docbook-3.1/docbook.cat	1999-02-02 06:53:32.000000000 +0100
++++ docbook-4.5/docbook-3.1/docbook.cat	2007-03-09 16:44:12.000000000 +0100
+@@ -19,7 +19,12 @@
+   -- ...................................................................... --
+   -- SGML declaration associated with DocBook ............................. --
+ 
+-DTDDECL "-//OASIS//DTD DocBook V3.1//EN" "docbook.dcl"
++-- disabled, see README.docbook.dcl --
++-- DTDDECL "-//OASIS//DTD DocBook V3.1//EN" "docbook.dcl" --
++
++  -- ...................................................................... --
++  -- Prefer PUBID to SYSID ................................................ --
++OVERRIDE YES
+ 
+   -- ...................................................................... --
+   -- DocBook driver file .................................................. --
+@@ -36,28 +41,5 @@
+ PUBLIC "-//OASIS//ENTITIES DocBook Notations V3.1//EN" "dbnotn.mod"
+ PUBLIC "-//OASIS//ENTITIES DocBook Character Entities V3.1//EN" "dbcent.mod"
+ 
+-  -- ...................................................................... --
+-  -- ISO entity sets ...................................................... --
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Diacritical Marks//EN" "iso-dia.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Numeric and Special Graphic//EN" "iso-num.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Publishing//EN" "iso-pub.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES General Technical//EN" "iso-tech.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Latin 1//EN" "iso-lat1.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Latin 2//EN" "iso-lat2.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Greek Letters//EN" "iso-grk1.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Monotoniko Greek//EN" "iso-grk2.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Greek Symbols//EN" "iso-grk3.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Alternative Greek Symbols//EN" "iso-grk4.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Arrow Relations//EN" "iso-amsa.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Binary Operators//EN" "iso-amsb.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Delimiters//EN" "iso-amsc.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Negated Relations//EN" "iso-amsn.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Ordinary//EN" "iso-amso.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Relations//EN" "iso-amsr.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Box and Line Drawing//EN" "iso-box.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Russian Cyrillic//EN" "iso-cyr1.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Non-Russian Cyrillic//EN" "iso-cyr2.gml"
+-
+   -- End of catalog data for DocBook V3.1 ................................. --
+   -- ...................................................................... --
+diff -urNad docbook-4.5~/docbook-4.0/docbook.cat docbook-4.5/docbook-4.0/docbook.cat
+--- docbook-4.5~/docbook-4.0/docbook.cat	2000-05-18 21:27:28.000000000 +0200
++++ docbook-4.5/docbook-4.0/docbook.cat	2007-03-09 16:45:08.000000000 +0100
+@@ -19,7 +19,12 @@
+   -- ...................................................................... --
+   -- SGML declaration associated with DocBook ............................. --
+ 
+-DTDDECL "-//OASIS//DTD DocBook V4.0//EN" "docbook.dcl"
++-- disabled, see README.docbook.dcl --
++-- DTDDECL "-//OASIS//DTD DocBook V4.0//EN" "docbook.dcl" --
++
++  -- ...................................................................... --
++  -- Prefer PUBID to SYSID ................................................ --
++OVERRIDE YES
+ 
+   -- ...................................................................... --
+   -- DocBook driver file .................................................. --
+@@ -36,28 +41,5 @@
+ PUBLIC "-//OASIS//ENTITIES DocBook Notations V4.0//EN" "dbnotn.mod"
+ PUBLIC "-//OASIS//ENTITIES DocBook Character Entities V4.0//EN" "dbcent.mod"
+ 
+-  -- ...................................................................... --
+-  -- ISO entity sets ...................................................... --
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Diacritical Marks//EN" "iso-dia.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Numeric and Special Graphic//EN" "iso-num.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Publishing//EN" "iso-pub.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES General Technical//EN" "iso-tech.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Latin 1//EN" "iso-lat1.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Latin 2//EN" "iso-lat2.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Greek Letters//EN" "iso-grk1.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Monotoniko Greek//EN" "iso-grk2.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Greek Symbols//EN" "iso-grk3.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Alternative Greek Symbols//EN" "iso-grk4.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Arrow Relations//EN" "iso-amsa.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Binary Operators//EN" "iso-amsb.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Delimiters//EN" "iso-amsc.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Negated Relations//EN" "iso-amsn.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Ordinary//EN" "iso-amso.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Relations//EN" "iso-amsr.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Box and Line Drawing//EN" "iso-box.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Russian Cyrillic//EN" "iso-cyr1.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Non-Russian Cyrillic//EN" "iso-cyr2.gml"
+-
+   -- End of catalog data for DocBook V4.0 ................................. --
+   -- ...................................................................... --
+diff -urNad docbook-4.5~/docbook-4.1/docbook.cat docbook-4.5/docbook-4.1/docbook.cat
+--- docbook-4.5~/docbook-4.1/docbook.cat	2000-06-19 16:50:28.000000000 +0200
++++ docbook-4.5/docbook-4.1/docbook.cat	2007-03-09 16:53:00.000000000 +0100
+@@ -19,7 +19,12 @@
+   -- ...................................................................... --
+   -- SGML declaration associated with DocBook ............................. --
+ 
+-DTDDECL "-//OASIS//DTD DocBook V4.1//EN" "docbook.dcl"
++-- disabled, see README.docbook.dcl --
++-- DTDDECL "-//OASIS//DTD DocBook V4.1//EN" "docbook.dcl" --
++
++  -- ...................................................................... --
++  -- Prefer PUBID to SYSID ................................................ --
++OVERRIDE YES
+ 
+   -- ...................................................................... --
+   -- DocBook driver file .................................................. --
+@@ -36,28 +41,5 @@
+ PUBLIC "-//OASIS//ENTITIES DocBook Notations V4.1//EN" "dbnotn.mod"
+ PUBLIC "-//OASIS//ENTITIES DocBook Character Entities V4.1//EN" "dbcent.mod"
+ 
+-  -- ...................................................................... --
+-  -- ISO entity sets ...................................................... --
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Diacritical Marks//EN" "iso-dia.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Numeric and Special Graphic//EN" "iso-num.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Publishing//EN" "iso-pub.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES General Technical//EN" "iso-tech.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Latin 1//EN" "iso-lat1.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Latin 2//EN" "iso-lat2.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Greek Letters//EN" "iso-grk1.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Monotoniko Greek//EN" "iso-grk2.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Greek Symbols//EN" "iso-grk3.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Alternative Greek Symbols//EN" "iso-grk4.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Arrow Relations//EN" "iso-amsa.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Binary Operators//EN" "iso-amsb.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Delimiters//EN" "iso-amsc.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Negated Relations//EN" "iso-amsn.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Ordinary//EN" "iso-amso.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Relations//EN" "iso-amsr.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Box and Line Drawing//EN" "iso-box.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Russian Cyrillic//EN" "iso-cyr1.gml"
+-PUBLIC "ISO 8879:1986//ENTITIES Non-Russian Cyrillic//EN" "iso-cyr2.gml"
+-
+   -- End of catalog data for DocBook V4.1 ................................. --
+   -- ...................................................................... --
+diff -urNad docbook-4.5~/docbook-4.2/docbook.cat docbook-4.5/docbook-4.2/docbook.cat
+--- docbook-4.5~/docbook-4.2/docbook.cat	2002-07-17 18:14:28.000000000 +0200
++++ docbook-4.5/docbook-4.2/docbook.cat	2007-03-09 16:42:30.000000000 +0100
+@@ -22,9 +22,6 @@
+ PUBLIC "-//OASIS//DTD DocBook V4.2//EN"
+        "docbook.dtd"
+ 
+-PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"
+-       "docbookx.dtd"
+-
+   -- ...................................................................... --
+   -- DocBook modules ...................................................... --
+ 
+@@ -49,65 +46,5 @@
+ PUBLIC "-//OASIS//ENTITIES DocBook Character Entities V4.2//EN"
+        "dbcentx.mod"
+ 
+-  -- ...................................................................... --
+-  -- ISO entity sets ...................................................... --
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Diacritical Marks//EN"
+-       "iso-dia.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Numeric and Special Graphic//EN"
+-       "iso-num.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Publishing//EN"
+-       "iso-pub.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES General Technical//EN"
+-       "iso-tech.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Latin 1//EN"
+-       "iso-lat1.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Latin 2//EN"
+-       "iso-lat2.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Greek Letters//EN"
+-       "iso-grk1.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Monotoniko Greek//EN"
+-       "iso-grk2.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Greek Symbols//EN"
+-       "iso-grk3.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Alternative Greek Symbols//EN"
+-       "iso-grk4.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Arrow Relations//EN"
+-       "iso-amsa.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Binary Operators//EN"
+-       "iso-amsb.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Delimiters//EN"
+-       "iso-amsc.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Negated Relations//EN"
+-       "iso-amsn.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Ordinary//EN"
+-       "iso-amso.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Relations//EN"
+-       "iso-amsr.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Box and Line Drawing//EN"
+-       "iso-box.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Russian Cyrillic//EN"
+-       "iso-cyr1.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Non-Russian Cyrillic//EN"
+-       "iso-cyr2.gml"
+-
+   -- End of catalog data for DocBook V4.2 ................................. --
+   -- ...................................................................... --
+diff -urNad docbook-4.5~/docbook-4.3/docbook.cat docbook-4.5/docbook-4.3/docbook.cat
+--- docbook-4.5~/docbook-4.3/docbook.cat	2004-03-31 13:44:16.000000000 +0200
++++ docbook-4.5/docbook-4.3/docbook.cat	2007-03-09 16:53:47.000000000 +0100
+@@ -22,9 +22,6 @@
+ PUBLIC "-//OASIS//DTD DocBook V4.3//EN"
+        "docbook.dtd"
+ 
+-PUBLIC "-//OASIS//DTD DocBook XML V4.3//EN"
+-       "docbookx.dtd"
+-
+   -- ...................................................................... --
+   -- DocBook modules ...................................................... --
+ 
+@@ -49,65 +46,5 @@
+ PUBLIC "-//OASIS//ENTITIES DocBook Character Entities V4.3//EN"
+        "dbcentx.mod"
+ 
+-  -- ...................................................................... --
+-  -- ISO entity sets ...................................................... --
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Diacritical Marks//EN"
+-       "iso-dia.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Numeric and Special Graphic//EN"
+-       "iso-num.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Publishing//EN"
+-       "iso-pub.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES General Technical//EN"
+-       "iso-tech.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Latin 1//EN"
+-       "iso-lat1.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Latin 2//EN"
+-       "iso-lat2.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Greek Letters//EN"
+-       "iso-grk1.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Monotoniko Greek//EN"
+-       "iso-grk2.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Greek Symbols//EN"
+-       "iso-grk3.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Alternative Greek Symbols//EN"
+-       "iso-grk4.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Arrow Relations//EN"
+-       "iso-amsa.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Binary Operators//EN"
+-       "iso-amsb.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Delimiters//EN"
+-       "iso-amsc.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Negated Relations//EN"
+-       "iso-amsn.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Ordinary//EN"
+-       "iso-amso.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Relations//EN"
+-       "iso-amsr.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Box and Line Drawing//EN"
+-       "iso-box.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Russian Cyrillic//EN"
+-       "iso-cyr1.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Non-Russian Cyrillic//EN"
+-       "iso-cyr2.gml"
+-
+   -- End of catalog data for DocBook V4.3 ................................. --
+   -- ...................................................................... --
+diff -urNad docbook-4.5~/docbook-4.4/docbook.cat docbook-4.5/docbook-4.4/docbook.cat
+--- docbook-4.5~/docbook-4.4/docbook.cat	2005-01-27 14:59:49.000000000 +0100
++++ docbook-4.5/docbook-4.4/docbook.cat	2007-03-09 16:54:12.000000000 +0100
+@@ -22,9 +22,6 @@
+ PUBLIC "-//OASIS//DTD DocBook V4.4//EN"
+        "docbook.dtd"
+ 
+-PUBLIC "-//OASIS//DTD DocBook XML V4.4//EN"
+-       "docbookx.dtd"
+-
+   -- ...................................................................... --
+   -- DocBook modules ...................................................... --
+ 
+@@ -52,65 +49,5 @@
+ PUBLIC "-//OASIS//ENTITIES DocBook Character Entities V4.4//EN"
+        "dbcentx.mod"
+ 
+-  -- ...................................................................... --
+-  -- ISO entity sets ...................................................... --
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Diacritical Marks//EN"
+-       "iso-dia.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Numeric and Special Graphic//EN"
+-       "iso-num.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Publishing//EN"
+-       "iso-pub.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES General Technical//EN"
+-       "iso-tech.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Latin 1//EN"
+-       "iso-lat1.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Latin 2//EN"
+-       "iso-lat2.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Greek Letters//EN"
+-       "iso-grk1.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Monotoniko Greek//EN"
+-       "iso-grk2.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Greek Symbols//EN"
+-       "iso-grk3.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Alternative Greek Symbols//EN"
+-       "iso-grk4.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Arrow Relations//EN"
+-       "iso-amsa.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Binary Operators//EN"
+-       "iso-amsb.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Delimiters//EN"
+-       "iso-amsc.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Negated Relations//EN"
+-       "iso-amsn.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Ordinary//EN"
+-       "iso-amso.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Relations//EN"
+-       "iso-amsr.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Box and Line Drawing//EN"
+-       "iso-box.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Russian Cyrillic//EN"
+-       "iso-cyr1.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Non-Russian Cyrillic//EN"
+-       "iso-cyr2.gml"
+-
+   -- End of catalog data for DocBook V4.4 ................................. --
+   -- ...................................................................... --
+diff -urNad docbook-4.5~/docbook-4.5/docbook.cat docbook-4.5/docbook-4.5/docbook.cat
+--- docbook-4.5~/docbook-4.5/docbook.cat	2006-10-03 15:27:50.000000000 +0200
++++ docbook-4.5/docbook-4.5/docbook.cat	2007-03-09 16:54:41.000000000 +0100
+@@ -22,9 +22,6 @@
+ PUBLIC "-//OASIS//DTD DocBook V4.5//EN"
+        "docbook.dtd"
+ 
+-PUBLIC "-//OASIS//DTD DocBook XML V4.5//EN"
+-       "docbookx.dtd"
+-
+   -- ...................................................................... --
+   -- DocBook modules ...................................................... --
+ 
+@@ -52,65 +49,5 @@
+ PUBLIC "-//OASIS//ENTITIES DocBook Character Entities V4.5//EN"
+        "dbcentx.mod"
+ 
+-  -- ...................................................................... --
+-  -- ISO entity sets ...................................................... --
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Diacritical Marks//EN"
+-       "iso-dia.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Numeric and Special Graphic//EN"
+-       "iso-num.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Publishing//EN"
+-       "iso-pub.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES General Technical//EN"
+-       "iso-tech.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Latin 1//EN"
+-       "iso-lat1.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Latin 2//EN"
+-       "iso-lat2.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Greek Letters//EN"
+-       "iso-grk1.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Monotoniko Greek//EN"
+-       "iso-grk2.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Greek Symbols//EN"
+-       "iso-grk3.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Alternative Greek Symbols//EN"
+-       "iso-grk4.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Arrow Relations//EN"
+-       "iso-amsa.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Binary Operators//EN"
+-       "iso-amsb.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Delimiters//EN"
+-       "iso-amsc.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Negated Relations//EN"
+-       "iso-amsn.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Ordinary//EN"
+-       "iso-amso.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Added Math Symbols: Relations//EN"
+-       "iso-amsr.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Box and Line Drawing//EN"
+-       "iso-box.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Russian Cyrillic//EN"
+-       "iso-cyr1.gml"
+-
+-PUBLIC "ISO 8879:1986//ENTITIES Non-Russian Cyrillic//EN"
+-       "iso-cyr2.gml"
+-
+   -- End of catalog data for DocBook V4.5 ................................. --
+   -- ...................................................................... --
--- docbook-4.5.orig/debian/patches/11_fix_dbpoolx_mod_header_comment_pubid.dpatch
+++ docbook-4.5/debian/patches/11_fix_dbpoolx_mod_header_comment_pubid.dpatch
@@ -0,0 +1,60 @@
+#! /bin/sh /usr/share/dpatch/dpatch-run
+## 11_fix_dbpoolx_mod_header_comment_pubid.dpatch by Adam Di Carlo <aph@debian.org>
+##     and Daniel Leidert (dale) <daniel.leidert@wgdd.de>.
+##
+## All lines beginning with `## DP:' are a description of the patch.
+## DP: The header in dbpoolx.mod references a wrong Public-ID. This patch
+## DP: simply fixes the header comment.
+## DP:
+## DP: Reported upstream:
+## DP: <URL:http://sourceforge.net/tracker/index.php?func=detail&aid=1679712&group_id=21935&atid=384107>
+
+@DPATCH@
+diff -urNad docbook-4.5~/docbook-4.2/dbpoolx.mod docbook-4.5/docbook-4.2/dbpoolx.mod
+--- docbook-4.5~/docbook-4.2/dbpoolx.mod	2002-07-17 18:14:25.000000000 +0200
++++ docbook-4.5/docbook-4.2/dbpoolx.mod	2007-03-10 08:31:23.000000000 +0100
+@@ -50,7 +50,7 @@
+      declaration that uses the public identifier shown below:
+ 
+      <!ENTITY % dbpool PUBLIC
+-     "-//OASIS//ELEMENTS DocBook XML Information Pool V4.2//EN"
++     "-//OASIS//ELEMENTS DocBook Information Pool V4.2//EN"
+      "dbpoolx.mod">
+      %dbpool;
+ 
+diff -urNad docbook-4.5~/docbook-4.3/dbpoolx.mod docbook-4.5/docbook-4.3/dbpoolx.mod
+--- docbook-4.5~/docbook-4.3/dbpoolx.mod	2004-03-31 13:43:24.000000000 +0200
++++ docbook-4.5/docbook-4.3/dbpoolx.mod	2007-03-10 08:31:49.000000000 +0100
+@@ -50,7 +50,7 @@
+      declaration that uses the public identifier shown below:
+ 
+      <!ENTITY % dbpool PUBLIC
+-     "-//OASIS//ELEMENTS DocBook XML Information Pool V4.3//EN"
++     "-//OASIS//ELEMENTS DocBook Information Pool V4.3//EN"
+      "dbpoolx.mod">
+      %dbpool;
+ 
+diff -urNad docbook-4.5~/docbook-4.4/dbpoolx.mod docbook-4.5/docbook-4.4/dbpoolx.mod
+--- docbook-4.5~/docbook-4.4/dbpoolx.mod	2005-01-27 14:59:45.000000000 +0100
++++ docbook-4.5/docbook-4.4/dbpoolx.mod	2007-03-10 08:32:00.000000000 +0100
+@@ -50,7 +50,7 @@
+      declaration that uses the public identifier shown below:
+ 
+      <!ENTITY % dbpool PUBLIC
+-     "-//OASIS//ELEMENTS DocBook XML Information Pool V4.4//EN"
++     "-//OASIS//ELEMENTS DocBook Information Pool V4.4//EN"
+      "dbpoolx.mod">
+      %dbpool;
+ 
+diff -urNad docbook-4.5~/docbook-4.5/dbpoolx.mod docbook-4.5/docbook-4.5/dbpoolx.mod
+--- docbook-4.5~/docbook-4.5/dbpoolx.mod	2006-10-03 15:27:50.000000000 +0200
++++ docbook-4.5/docbook-4.5/dbpoolx.mod	2007-03-10 08:32:10.000000000 +0100
+@@ -50,7 +50,7 @@
+      declaration that uses the public identifier shown below:
+ 
+      <!ENTITY % dbpool PUBLIC
+-     "-//OASIS//ELEMENTS DocBook XML Information Pool V4.5//EN"
++     "-//OASIS//ELEMENTS DocBook Information Pool V4.5//EN"
+      "dbpoolx.mod">
+      %dbpool;
+ 
--- docbook-4.5.orig/debian/patches/12_fix_256119_and_turn_on_sgml_mode.dpatch
+++ docbook-4.5/debian/patches/12_fix_256119_and_turn_on_sgml_mode.dpatch
@@ -0,0 +1,117 @@
+#! /bin/sh /usr/share/dpatch/dpatch-run
+## 12_fix_256119_and_turn_on_sgml_mode.dpatch by Daniel Leidert (dale) <daniel.leidert@wgdd.de>
+##
+## All lines beginning with `## DP:' are a description of the patch.
+## DP: Fixes broken element declarations by adding the tag omission characters
+## DP: for SGML and enable SGML features
+## DP: (closes: <URL:http://bugs.debian.org/256119>).
+## DP:
+## DP: <URL:http://packages.debian.org/changelogs/pool/main/d/docbook/current/changelog#versionversion4.4-1>
+
+@DPATCH@
+diff -urNad docbook-4.5~/docbook-4.3/dbpoolx.mod docbook-4.5/docbook-4.3/dbpoolx.mod
+--- docbook-4.5~/docbook-4.3/dbpoolx.mod	2007-03-10 09:01:12.000000000 +0100
++++ docbook-4.5/docbook-4.3/dbpoolx.mod	2007-03-10 09:01:13.000000000 +0100
+@@ -633,6 +633,9 @@
+ 
+ <!-- xml:base: base URI -->
+ 
++<![%sgml.features;[
++<!ENTITY % xml-base.attrib "">
++]]>
+ <!ENTITY % xml-base.attrib
+ 	"xml:base	CDATA		#IMPLIED">
+ 
+diff -urNad docbook-4.5~/docbook-4.3/docbookx.dtd docbook-4.5/docbook-4.3/docbookx.dtd
+--- docbook-4.5~/docbook-4.3/docbookx.dtd	2004-03-31 13:43:20.000000000 +0200
++++ docbook-4.5/docbook-4.3/docbookx.dtd	2007-03-10 09:01:13.000000000 +0100
+@@ -59,7 +59,7 @@
+ <!-- ...................................................................... -->
+ <!-- Enable SGML features ................................................. -->
+ 
+-<!ENTITY % sgml.features "IGNORE">
++<!ENTITY % sgml.features "INCLUDE">
+ <![%sgml.features;[
+ <!ENTITY % xml.features "IGNORE">
+ ]]>
+diff -urNad docbook-4.5~/docbook-4.3/htmltblx.mod docbook-4.5/docbook-4.3/htmltblx.mod
+--- docbook-4.5~/docbook-4.3/htmltblx.mod	2004-03-31 13:43:24.000000000 +0200
++++ docbook-4.5/docbook-4.3/htmltblx.mod	2007-03-10 09:01:13.000000000 +0100
+@@ -63,6 +63,9 @@
+   >
+ 
+ <!-- Does not contain lang or dir because they are in %common.attribs -->
++<![%sgml.features;[
++<!ENTITY % i18n "">
++]]>
+ <!ENTITY % i18n
+  "xml:lang    NMTOKEN        #IMPLIED"
+   >
+@@ -92,11 +95,11 @@
+   "valign     (top|middle|bottom|baseline) #IMPLIED"
+   >
+ 
+-<!ELEMENT colgroup (col)*>
+-<!ELEMENT col      EMPTY>
+-<!ELEMENT tr       (th|td)+>
+-<!ELEMENT th       (%para.char.mix; | %tabentry.mix;)*>
+-<!ELEMENT td       (%para.char.mix; | %tabentry.mix;)*>
++<!ELEMENT colgroup %ho; (col)*>
++<!ELEMENT col      %ho; EMPTY>
++<!ELEMENT tr       %ho; (th|td)+>
++<!ELEMENT th       %ho; (%para.char.mix; | %tabentry.mix;)*>
++<!ELEMENT td       %ho; (%para.char.mix; | %tabentry.mix;)*>
+ 
+ <!ATTLIST colgroup
+   %attrs;
+@@ -206,7 +209,7 @@
+     summary     CDATA          #IMPLIED
+     width       CDATA        #IMPLIED
+     border      CDATA        #IMPLIED
+-    rules       (none | groups | rows | cols | all)      #IMPLIED
++    rules       CDATA        #IMPLIED
+     cellspacing CDATA        #IMPLIED
+     cellpadding CDATA        #IMPLIED
+     align       (left|center|right)   #IMPLIED
+diff -urNad docbook-4.5~/docbook-4.4/docbookx.dtd docbook-4.5/docbook-4.4/docbookx.dtd
+--- docbook-4.5~/docbook-4.4/docbookx.dtd	2005-01-27 14:59:45.000000000 +0100
++++ docbook-4.5/docbook-4.4/docbookx.dtd	2007-03-10 09:01:13.000000000 +0100
+@@ -61,7 +61,7 @@
+ <!-- ...................................................................... -->
+ <!-- Enable SGML features ................................................. -->
+ 
+-<!ENTITY % sgml.features "IGNORE">
++<!ENTITY % sgml.features "INCLUDE">
+ <![%sgml.features;[
+ <!ENTITY % xml.features "IGNORE">
+ ]]>
+diff -urNad docbook-4.5~/docbook-4.4/htmltblx.mod docbook-4.5/docbook-4.4/htmltblx.mod
+--- docbook-4.5~/docbook-4.4/htmltblx.mod	2005-01-27 14:59:45.000000000 +0100
++++ docbook-4.5/docbook-4.4/htmltblx.mod	2007-03-10 09:01:13.000000000 +0100
+@@ -63,6 +63,9 @@
+   >
+ 
+ <!-- Does not contain lang or dir because they are in %common.attribs -->
++<![%sgml.features;[
++<!ENTITY % i18n "">
++]]>
+ <!ENTITY % i18n
+  "xml:lang    NMTOKEN        #IMPLIED"
+   >
+@@ -92,11 +95,11 @@
+   "valign     (top|middle|bottom|baseline) #IMPLIED"
+   >
+ 
+-<!ELEMENT colgroup (col)*>
+-<!ELEMENT col      EMPTY>
+-<!ELEMENT tr       (th|td)+>
+-<!ELEMENT th       (%para.char.mix; | %tabentry.mix; | table | informaltable)*>
+-<!ELEMENT td       (%para.char.mix; | %tabentry.mix; | table | informaltable)*>
++<!ELEMENT colgroup %ho; (col)*>
++<!ELEMENT col      %ho; EMPTY>
++<!ELEMENT tr       %ho; (th|td)+>
++<!ELEMENT th       %ho; (%para.char.mix; | %tabentry.mix; | table | informaltable)*>
++<!ELEMENT td       %ho; (%para.char.mix; | %tabentry.mix; | table | informaltable)*>
+ 
+ <!ATTLIST colgroup
+   %attrs;
--- docbook-4.5.orig/debian/patches/10_fix_101779_mention_artheader_to_articleinfo_change.dpatch
+++ docbook-4.5/debian/patches/10_fix_101779_mention_artheader_to_articleinfo_change.dpatch
@@ -0,0 +1,33 @@
+#! /bin/sh /usr/share/dpatch/dpatch-run
+## 10_fix_101779_mention_artheader_to_articleinfo_change.dpatch by
+##     Adam Di Carlo <aph@debian.org>.
+##
+## All lines beginning with `## DP:' are a description of the patch.
+## DP: The removal of ArtHeader in 4.0 is not documented in ChangeLog. This
+## DP: patch fixes it (closes: <URL:http://bugs.debian.org/bug=101779>).
+## DP:
+## DP: <URL:http://packages.debian.org/changelogs/pool/main/d/docbook/current/changelog#versionversion4.1-5>
+
+@DPATCH@
+diff -urNad docbook-4.5~/ChangeLog.upstream docbook-4.5/ChangeLog.upstream
+--- docbook-4.5~/ChangeLog.upstream	2007-03-09 14:36:17.000000000 +0100
++++ docbook-4.5/ChangeLog.upstream	2007-03-14 19:35:01.000000000 +0100
+@@ -475,6 +475,7 @@
+ - Added synopsis markup for modern programming languages (e.g, object
+   oriented languages like Java, C++, and IDL)
+ - Renamed DocInfo to PrefaceInfo, ChapterInfo, AppendixInfo, etc.
++  (and ArtHeader to ArticleInfo)
+ 
+ Other:
+ 
+diff -urNad docbook-4.5~/docbook-4.1/40chg.txt docbook-4.5/docbook-4.1/40chg.txt
+--- docbook-4.5~/docbook-4.1/40chg.txt	2000-06-19 16:50:28.000000000 +0200
++++ docbook-4.5/docbook-4.1/40chg.txt	2007-03-14 19:34:21.000000000 +0100
+@@ -29,6 +29,7 @@
+ - Added synopsis markup for modern programming languages (e.g, object
+   oriented languages like Java, C++, and IDL)
+ - Renamed DocInfo to PrefaceInfo, ChapterInfo, AppendixInfo, etc.
++  (and ArtHeader to ArticleInfo)
+ - Comment was renamed Remark
+ - InterfaceDefinition was removed
+ 
--- docbook-4.5.orig/debian/examples/docbk.sgml
+++ docbook-4.5/debian/examples/docbk.sgml
@@ -0,0 +1,2 @@
+<!DOCTYPE book PUBLIC "-//OASIS//DTD DocBook//EN">
+<book></book>
--- docbook-4.5.orig/debian/examples/docbk42.sgml
+++ docbook-4.5/debian/examples/docbk42.sgml
@@ -0,0 +1,2 @@
+<!DOCTYPE book PUBLIC "-//OASIS//DTD DocBook V4.2//EN">
+<book></book>
--- docbook-4.5.orig/debian/examples/docbk43.sgml
+++ docbook-4.5/debian/examples/docbk43.sgml
@@ -0,0 +1,2 @@
+<!DOCTYPE book PUBLIC "-//OASIS//DTD DocBook V4.3//EN">
+<book></book>
--- docbook-4.5.orig/debian/examples/docbk44.sgml
+++ docbook-4.5/debian/examples/docbk44.sgml
@@ -0,0 +1,2 @@
+<!DOCTYPE book PUBLIC "-//OASIS//DTD DocBook V4.4//EN">
+<book></book>
--- docbook-4.5.orig/debian/examples/docbk45.sgml
+++ docbook-4.5/debian/examples/docbk45.sgml
@@ -0,0 +1,2 @@
+<!DOCTYPE book PUBLIC "-//OASIS//DTD DocBook V4.5//EN">
+<book></book>
--- docbook-4.5.orig/debian/examples/Makefile
+++ docbook-4.5/debian/examples/Makefile
@@ -0,0 +1,22 @@
+SRCS	:= $(wildcard *.sgml)
+NSGMLS	:= $(shell if command -v onsgmls 2>/dev/null; then	 \
+			:					;\
+		   elif command -v nsgmls 2>/dev/null; then	 \
+			:					;\
+		   else						 \
+			echo "false"				;\
+			echo "warning: no nsgmls found" >&2	;\
+		   fi)
+
+NSGMLSARGS=
+
+.PHONY: validate
+validate:	$(addsuffix .nsgmls,$(SRCS))
+
+%.nsgmls   : %
+	$(NSGMLS) -E1 -wall $(NSGMLSARGS) -gues $<
+
+.PHONY: clean
+clean:
+	rm -f *.html *~
+
--- docbook-4.5.orig/debian/control
+++ docbook-4.5/debian/control
@@ -0,0 +1,33 @@
+Source: docbook
+Section: text
+Priority: optional
+Maintainer: Debian XML/SGML Group <debian-xml-sgml-pkgs@lists.alioth.debian.org>
+Uploaders: Adam Di Carlo <aph@debian.org>, Daniel Leidert (dale) <daniel.leidert@wgdd.de>
+Build-Depends: debhelper (>= 4.1.31), dpatch
+Standards-Version: 3.7.2
+Homepage: http://www.oasis-open.org/docbook/sgml/
+XS-Vcs-Browser: http://svn.debian.org/wsvn/debian-xml-sgml/packages/docbook/trunk/
+XS-Vcs-Svn: svn://svn.debian.org/svn/debian-xml-sgml/packages/docbook/
+
+Package: docbook
+Architecture: all
+Depends: sgml-data (>= 1.5.0), ${misc:Depends}
+Conflicts: docbook-xml (<= 4.4-4)
+Suggests: psgml, docbook-dsssl, docbook-xml, docbook-defguide
+Description: standard SGML representation system for technical documents
+ DocBook is an SGML document type definition (DTD) that is well-suited
+ to books, articles, or reference documentation about technical
+ matters, systems, or software (although it is by no means limited to
+ these applications).  It has emerged as an open and standard DTD in the
+ software industry, and is used as the documentation system of choice
+ for many free software projects.
+ .
+ This package contains the SGML DTD for DocBook, which describes
+ the formal structure for documents following this format.  If you
+ wish to author DocBook documents using XML rather than SGML, use the
+ 'docbook-xml' package.  You will need other packages in order to edit
+ (psgml), validate (opensp) or format (docbook-xsl, docbook-dsssl)
+ DocBook documents.
+ .
+ This package includes the 2.4.1, 3.0, 3.1, 4.0, 4.1, 4.2, 4.3, 4.4 and
+ 4.5 versions of the DocBook SGML DTD.
--- docbook-4.5.orig/debian/compat
+++ docbook-4.5/debian/compat
@@ -0,0 +1 @@
+4
--- docbook-4.5.orig/debian/watch
+++ docbook-4.5/debian/watch
@@ -0,0 +1,4 @@
+# See uscan(1) for format
+
+version=3
+http://www.oasis-open.org/docbook/sgml/4\.([\d]+)/docbook-([\d\.]+).zip
--- docbook-4.5.orig/debian/docbook.examples
+++ docbook-4.5/debian/docbook.examples
@@ -0,0 +1 @@
+debian/examples/*
--- docbook-4.5.orig/debian/changelog
+++ docbook-4.5/debian/changelog
@@ -0,0 +1,362 @@
+docbook (4.5-4) unstable; urgency=low
+
+  * debian/control: Homepage field transition.
+    (Suggests): Added docbook-defguide.
+  * debian/docbook.postinst: Removed. Contains only pre-etch legacy stuff.
+  * debian/README.Debian: Removed or replaced references to non-existing
+    packages.
+  * debian/watch: Added watch file.
+
+ -- Daniel Leidert (dale) <daniel.leidert@wgdd.de>  Wed, 26 Sep 2007 21:21:26 +0200
+
+docbook (4.5-3) unstable; urgency=medium
+
+  * debian/control (Maintainer, Uploaders): Set the Debian XML/SGML team as
+    maintainer.
+  * debian/docbook.examples: Fixed examples location: will re-add them back
+    to the package.
+  * debian/rules: Removed bashisms (closes: #422114). Thanks to Sebastian
+    Dröge for the report.
+
+ -- Daniel Leidert (dale) <daniel.leidert@wgdd.de>  Thu, 03 May 2007 19:46:21 +0200
+
+docbook (4.5-2) unstable; urgency=low
+
+  * Upload to unstable.
+
+  [ Daniel Leidert ]
+  * debian/rules (get-orig-source): Make the target compatible with the "Best
+    Packaging Practices" reference.
+
+ -- Daniel Leidert (dale) <daniel.leidert@wgdd.de>  Sat, 14 Apr 2007 19:49:00 +0200
+
+docbook (4.5-1) experimental; urgency=low
+
+  * New upstream release DocBook SGML 4.5 added (closes: #413166).
+
+  [ Daniel Leidert ]
+  * catalog: Made latest DocBook SGML 4.5 the default (closes: #413304).
+  * debian/control: Added XS-Vcs-Browser and XS-Vcs-Svn fields.
+    (Build-Depends-Indep): Fixed debhelper build dependency to be listed in
+    Build-Depends.
+    (Build-Depends): Added dpatch.
+    (Uploaders): Added the Debian XML/SGML group and myself (closes: #358522).
+    (Standards-Version): Bump to latest standards version 3.7.2.
+    (Description): Added newly added DocBook version.
+  * debian/copyright: Updated the packaging and copyright information.
+  * debian/rules: Added get-orig-source and dpatch targets. Cleaned the file a
+    lot. The installation of files was moved from the external
+    install_docbook.sh script to dh_install calls. The .links, .install and
+    .sgmlcatalogs debhelper files are created by debian/rules itself (atm I
+    don't remove them in the clean target).
+    (binary-indep): Fixed the dh_installdocs call to include Adams
+    README.docbook.dcl, that was not included in the package.
+    (debian/docbook.install, debian/docbook.links): Made the dbgenent.mod
+    files config-files.
+  * debian/patches/00list: Added. Moved all existing changes to dpatch patches
+    and enabled them here.
+  * debian/patches/01_remove_entity_sets_and_dcl_from_catalogs.dpatch: Added.
+  * debian/patches/10_fix_101779_mention_artheader_to_articleinfo_change.dpatch:
+    Added.
+  * debian/patches/11_fix_dbpoolx_mod_header_comment_pubid.dpatch: Added.
+  * debian/patches/12_fix_256119_and_turn_on_sgml_mode.dpatch: Added.
+  * debian/examples/docbk*.sgml: Added two new examples for DocBook 4.5 and
+    added the missing newlines to the existing examples.
+  * debian/docbook.examples: Added.
+  * debian/postinst: Renamed to docbook.postinst.
+  * debian/README.debian: Renamed to README.Debian.
+  * debian/install_docbook.sh: Removed. See debian/rules changes.
+  * debian/TODO: Removed (implemented the only item).
+  * debian/watch: Removed (empty).
+
+ -- Daniel Leidert (dale) <daniel.leidert@wgdd.de>  Sat, 10 Mar 2007 09:32:33 +0100
+
+docbook (4.4-1) unstable; urgency=low
+
+  * acknowledge NMU; closes: #301786
+  * new upstream version; closes: #339774
+  * docbook4{3,4}/docbookx.dtd: turn on SGML mode
+  * docbook4{3,4}: sync with dbpoolx.mod and htmltblx.mod from
+    docbook-xml; closes: #256119
+  * conflict with docbook-xml prior to 4.4-5, otherwise the 4.3 and 4.4
+    DTDs don't work
+  * docbook4.3 catalog: don't provide "-//OASIS//DTD DocBook XML V4.3//EN"
+  * add really basic docbook 4.2, 4.3, and 4.4 books to test with
+
+ -- Adam Di Carlo <aph@debian.org>  Thu, 25 May 2006 19:09:05 -0400
+
+docbook (4.3-1.1) unstable; urgency=high
+
+  * Non-maintainer upload.
+  * High-urgency upload for sarge-targetted RC bugfix.
+  * Drop use of symlinks for finding dangling links, as we can do the
+    same using a find command; thanks to Joey Hess for the patch.
+    Closes: #301786.
+
+ -- Steve Langasek <vorlon@debian.org>  Wed,  6 Apr 2005 11:24:24 -0700
+
+docbook (4.3-1) unstable; urgency=low
+
+  * new upstream version; closes: #243666
+  * make the usual array of minor corrections in upstream catalogs
+  * minor package description changes
+  * postinst legacy remove is more conditional and more robust; hopefully,
+    closes: #242833
+  * really don't register "-//OASIS//DTD DocBook XML V4.2//EN" in
+    catalog.xml
+
+ -- Adam Di Carlo <aph@debian.org>  Tue, 15 Jun 2004 22:15:02 -0400
+
+docbook (4.2-3) unstable; urgency=low
+
+  * stop providing "-//OASIS//DTD DocBook XML V4.2//EN", that is in the
+    docbook-xml package
+  * policy compliance updated to 3.5.8, no changes needed
+  * debhelper 4 compat mode
+  * use new dh_installcatalog, it simplifies our maintainer scripts
+  * remove build-depends on M4, we don't need to process maintainer
+    scripts anymore
+  * remove suggests on sp or jade or docbook-xml, add suggests for
+    docbook-dsssl and docbook-doc
+  * improve the package synopsis and description, add the Homepage
+  * stop using sgml-catalog-check.pl from sgml-data during build, the
+    /usr/share/sgml symlinking is no longer needed (SGML Policy update
+    pending)
+
+ -- Adam Di Carlo <aph@debian.org>  Tue, 25 Feb 2003 02:18:52 -0500
+
+docbook (4.2-2) unstable; urgency=low
+
+  * SGML catalogs should be named 'catalog' according to policy; retain
+    the 'docbook.cat' as a symlink in case anyone is using that directly
+
+ -- Adam Di Carlo <aph@debian.org>  Mon, 20 Jan 2003 04:44:52 -0500
+
+docbook (4.2-1) unstable; urgency=low
+
+  * new upstream version, closes: #155105
+  * latest DocBook version bumped to 4.2
+
+ -- Adam Di Carlo <aph@debian.org>  Wed,  7 Aug 2002 01:12:49 -0400
+
+docbook (4.1.99really4.2.cr1-2) unstable; urgency=low
+
+  * oops, remove ISO entity sets from docbook 4.2cr1 catalog
+  * minor typo in generalized.cat fixed
+
+ -- Adam Di Carlo <aph@debian.org>  Sun, 31 Mar 2002 01:40:26 -0500
+
+docbook (4.1.99really4.2.cr1-1) unstable; urgency=low
+
+  * new upstream version, pre-release of docbook 4.2;
+    note that docbook 4.1 is still considered the latest version and
+    latest 4.x version until 4.2 is actually released
+  * policy checked up to 3.5.6, no changes required
+  * disable DTDDECL in generalized.cat, I think this just slipped through
+  * debian/rules: some housekeeping
+
+ -- Adam Di Carlo <aph@debian.org>  Sun, 31 Mar 2002 01:28:04 -0500
+
+docbook (4.1-6) unstable; urgency=low
+
+  * another cut at the package description, hopefully it satisfies the
+    user; closes: #112021
+  * debian/rules: installing *everything* as documentation was not my
+    intention in the last release, fix that
+  * require sgml-data 1.5.3 or better, which calls an element an element
+    when making the creepy symlinks; closes: #51043
+
+ -- Adam Di Carlo <aph@debian.org>  Tue,  2 Oct 2001 03:06:21 -0400
+
+docbook (4.1-5) unstable; urgency=low
+
+  * all docbook catalog files now have "OVERRIDE YES", which forces the
+    PUBID rather than the SYSID in DTD declarations; closes: #97330
+  * correct upstream 40chg.txt to notate the change from ArtHeader to
+    ArticleInfo; closes: #101779
+  * README.Debian rewritten; closes: #106379
+  * it's apparently not nice to call dh_installdocs more than once per
+    package, so stop doing that; closes#98440
+
+ -- Adam Di Carlo <aph@debian.org>  Tue,  2 Oct 2001 01:49:30 -0400
+
+docbook (4.1-4) unstable; urgency=low
+
+  * debian/rules: re-work fix for files in debian not preserving execute
+    bit
+    closes: #90627
+
+ -- Adam Di Carlo <aph@debian.org>  Sun,  8 Apr 2001 14:37:42 -0400
+
+docbook (4.1-3) unstable; urgency=low
+
+  * debian/control: build-depends-indep includes m4
+  * debian/rules: cope with my local CVS not preserving executable bit,
+    making the source unbuildable
+    closes: #90627
+
+ -- Adam Di Carlo <aph@debian.org>  Thu, 22 Mar 2001 02:23:56 -0500
+
+docbook (4.1-2) unstable; urgency=low
+
+  * disable docbook.dcl as DTDDECL, otherwise lots of otherwise valid SGML
+    files will break (when using DTDDECL comprehending processors like
+    OpenJade, anyhow).  For more info, see README.docbook.dcl.
+  * include docbook 4.0
+    closes: #68666
+
+ -- Adam Di Carlo <aph@debian.org>  Wed, 21 Mar 2001 03:37:38 -0500
+
+docbook (4.1-1) unstable; urgency=low
+
+  * new maintainer, with permission from Mark Eichin
+    closes: #19682,  #19683,  #36956,  #47375,  #50243,  #51032,  #68033
+    closes: #87325,  #88679
+  * debian/control: require new sgml-base and sgml-data; suggest
+    docbook-xml; fix a grammar error in the description
+  * transition to new SGML standard system, /usr/share/sgml, and delegated
+    catalogs
+    closes: #90034
+  * rework the maintainer scripts accordingly; use m4 to dynamically set
+    some variables; move the removal work from postrm to prerm; don't
+    bother scrubbing legacy junk in prerm, since postinst configure
+    already ran, what's the point?
+  * provide new catalog entries for just referring to DocBook, generalized
+    to mean the last recommended DocBook, or the latest DocBook v4, or the
+    latest DocBook v3.
+  * delegated catalogs have proper DTDDECLs
+    closes: #63046 (again)
+  * debian/rules: stop using debstd
+    closes: #90122
+  * already fixed in earlier versions
+    closes: #90125
+  * debian/control: add Build-Depends-Indep; update standards to 3.5.2
+  * debian/copyright: update maintainership info and add coverage of the
+    debian integration scripts
+
+ -- Adam Di Carlo <aph@debian.org>  Mon, 19 Mar 2001 01:56:38 -0500
+
+docbook (4.1-0.2) unstable; urgency=low
+
+  * non-maintainer upload
+  * sgml-data/sgml-catalog-check.pl location moved
+    closes: #88679
+  * debian/postrm: fix some of the legacy FPI removal logic
+    closes: #87325
+  * debian/postinst: analogous fix, and remove some scrubbing of the SGML
+    catalog which really seemed overreaching (such as fudging with
+    SGMLDECL, and stuff which really ought to be handled by
+    install-sgmlcatalog)
+
+ -- Adam Di Carlo <aph@debian.org>  Sat, 17 Mar 2001 16:54:59 -0500
+
+docbook (4.1-0.1) unstable; urgency=low
+
+  * NMU because Mark lacks time.
+  * New upstream release (Closes: #68033).
+  * Do not include 4.0 DTD because 4.1 has no user-level changes from
+    4.0.
+  * Install doc files into /usr/share/doc/ as per policy (Closes:
+    #50243, #51032).
+  * Uncommented DTDDECL entries from the catalog (Closes: #63046).
+  * Updated download location and copyright notice.
+  * Commented out obsolete line in debian/watch.
+
+ -- Yann Dirson <dirson@debian.org>  Tue,  1 Aug 2000 01:01:59 +0200
+
+docbook (3.1-2) unstable; urgency=low
+
+  * debian/control: fix Maintainer realname for consistency.
+
+ -- Mark W. Eichin <eichin@thok.org>  Wed,  3 Nov 1999 16:14:46 -0500
+
+docbook (3.1-1) unstable; urgency=low
+
+  * Thanks for all the work; closing the bugs for real and updating my
+    email addresses. closes: #19682, #19683, #36956, #47375.
+
+ -- Mark W. Eichin <eichin@thok.org>  Wed,  3 Nov 1999 00:43:12 -0500
+
+docbook (3.1-0.4) unstable; urgency=low
+
+  * NMU so it can be removed when install-sgmlcatalog isn't present.
+    (Closes: #47375). 
+
+ -- J.H.M. Dassen (Ray) <jdassen@debian.org>  Sat, 30 Oct 1999 14:31:46 +0200
+
+docbook (3.1-0.3) unstable; urgency=low
+
+  * non-maintainer release, blessed by Mark Eichin
+  * at the suggestion of Norm Walsh, retain DTD and entity file name
+    (closes Bug#36956) -- he also suggests that we don't split entities
+    into a separate dir, which is a reasonable thought, but I leave it to
+    others to file a bug if they agree with Norm.
+
+ -- Adam Di Carlo <aph@debian.org>  Sat, 22 May 1999 16:43:06 -0400
+
+docbook (3.1-0.2) unstable; urgency=low
+
+  * non-maintainer release
+  * docbook.cat: fix problems with some system identifiers, bugs I
+    introduced
+  * use script from sgml-data to create nice SGML links (completes closing
+    of Bug#19682)
+
+ -- Adam Di Carlo <aph@debian.org>  Fri, 26 Mar 1999 04:25:09 -0500
+
+docbook (3.1-0.1) unstable; urgency=low
+
+  * non-maintainer release
+  * new upstream version (blessed by Mark Eichin)
+  * deliver docbook 2.4.1, 3.0, and 3.1 DTDs (closes Bug#19683)
+  * put DTDs, entities, and declarations in standard locations under
+    /usr/lib/sgml; make symlinks under there too in compliance with the
+    sgml-base subpolicy (closes Bug#19682)
+  * debian/control: bump standards version to 2.5.0, reconcile priority to
+    what it is in the archive, update description
+  * debian/rules: some modularization
+  * debian/copyright: updated
+  * README.debian: adding some pointer on other packages and the home page
+
+ -- Adam Di Carlo <aph@debian.org>  Fri, 26 Mar 1999 01:36:31 -0500
+
+docbook (3.0-5) unstable; urgency=low
+
+  * debian/rules: use install -m 0644 so text files don't get installed
+    gratuitously executable (lintian).
+
+ -- Mark Eichin <eichin@kitten.gen.ma.us>  Wed, 29 Apr 1998 05:39:56 -0400
+
+docbook (3.0-4) unstable; urgency=low
+
+  * docbook.cat: comment out DTDDECL to avoid warnings from jade.  Removed
+    ISO entity pointers, since they are now supplied by sgml-data.
+
+ -- Mark Eichin <eichin@kitten.gen.ma.us>  Tue,  7 Apr 1998 02:39:48 -0400
+
+docbook (3.0-3) unstable; urgency=low
+
+  * debian/postinst, debian/postrm: use sgml-base like jade does.
+  * debian/control: depend on sgml-base now.
+
+ -- Mark W. Eichin <eichin@kitten.gen.ma.us>  Tue,  4 Nov 1997 18:18:18 -0500
+
+docbook (3.0-2) unstable; urgency=low
+
+  * debian/{postinst, postrm}: change md5sum markers to make them easier
+    to find later (and consistent with jade); add cleanup of older form.
+
+ -- Mark W. Eichin <eichin@kitten.gen.ma.us>  Tue,  1 Jul 1997 01:08:31 -0400
+
+docbook (3.0-1) unstable; urgency=low
+
+  * Initial Release.
+  * docbook.cat: point to davenport subdir.
+  * rules: install everything, under binary-indep.
+  * debian/{postinst, postrm}: use md5sum markers to add/remove catalog
+    entries. Also nuke SGMLDECL line (by wrapping it in an sgml -- comment
+    --.)
+
+ -- Mark Eichin <eichin@kitten.gen.ma.us>  Mon, 30 Jun 1997 14:11:38 -0400
+
+
--- docbook-4.5.orig/debian/copyright
+++ docbook-4.5/debian/copyright
@@ -0,0 +1,39 @@
+This package was debianized by Mark Eichin <eichin@thok.org> on Mon,
+30 Jun 1997 14:11:38 -0400.  It is currently maintained by Adam Di
+Carlo <aph@debian.org> and the Debian XML/SGML group
+<debian-xml-sgml-devel@lists.alioth.debian.org>.  No changes were made
+to the upstream version of these materials, although the provided SGML
+Open catalogs were trimmed or enhanced to suit Debian's SGML
+infrastructure.  All Debian integration scripts and packages are under
+the GPL v2 or better, a copy of which can be found on Debian systems
+in '/usr/share/common-licenses/GPL'.
+
+Download:
+
+  The original upstream materials may be downloaded from
+  <URL:http://www.oasis-open.org/docbook/sgml/>. This process is
+  transparent via the `get-orig-source' target in debian/rules.
+
+Copyright:
+
+  Copyright 1992-2006 HaL Computer Systems, Inc.,
+  O'Reilly & Associates, Inc., ArborText, Inc., Fujitsu Software
+  Corporation, orman Walsh, un Microsystems, Inc.,  and the
+  Organization for the Advancement of Structured Information
+  Standards (OASIS).
+
+License and Warranty:
+
+  Permission to use, copy, modify and distribute the DocBook DTD and
+  its accompanying documentation for any purpose and without fee is
+  hereby granted in perpetuity, provided that the above copyright
+  notice and this paragraph appear in all copies.  The copyright
+  holders make no representation about the suitability of the DTD for
+  any purpose.  It is provided "as is" without expressed or implied
+  warranty.
+
+  If you modify the DocBook DTD in any way, except for declaring and
+  referencing additional sets of general entities and declaring
+  additional notations, label your DTD as a variant of DocBook.  See
+  the maintenance documentation for more information.
+
--- docbook-4.5.orig/debian/rules
+++ docbook-4.5/debian/rules
@@ -0,0 +1,121 @@
+#!/usr/bin/make -f
+# Made with the aid of debmake, by Christoph Lameter,
+# based on the sample debian/rules file for GNU hello by Ian Jackson.
+
+PACKAGE	:= docbook
+
+include /usr/share/dpatch/dpatch.make
+
+
+debian/docbook.install:
+	set -ex ; \
+	touch $@ ; \
+	for db in docbook-* ; do \
+		dbver=`echo $${db} | sed -e 's/docbook-//g'` ; \
+		for file in $${db}/*.dcl $${db}/*.dtd $${db}/*.mod ; do \
+			if [ $${file} != $${db}/dbgenent.mod ] ; then \
+				echo "$${file} usr/share/sgml/docbook/dtd/$${dbver}" >> $@ ; \
+			else \
+				echo "$${file} etc/sgml/docbook/$${dbver}" >> $@ ; \
+			fi ; \
+		done ; \
+	done
+
+debian/docbook.links:
+	set -ex ; \
+	touch $@ ; \
+	for db in docbook-* ; do \
+		dbver=`echo $${db} | sed -e 's/docbook-//g'` ; \
+		[ -f $${db}/dbgenent.mod ] && \
+		    echo -n "etc/sgml/docbook/$${dbver}/dbgenent.mod " >> $@ && \
+		        echo "usr/share/sgml/docbook/dtd/$${dbver}/dbgenent.mod" >> $@ ; \
+		[ -f $${db}/docbook.cat ] && \
+		    echo "usr/share/sgml/docbook/dtd/$${dbver}/catalog usr/share/sgml/docbook/dtd/$${dbver}/docbook.cat" >> $@ ; \
+	done
+
+debian/docbook.sgmlcatalogs:
+	set -ex ; \
+	touch $@ ; \
+	echo "catalog /usr/share/sgml/docbook/dtd/catalog" > $@ ; \
+	for db in docbook-* ; do \
+		dbver=`echo $${db} | sed -e 's/docbook-//g'` ; \
+		[ -f $${db}/docbook.cat ] && \
+		    echo "$${db}/docbook.cat /usr/share/sgml/docbook/dtd/$${dbver}/catalog" >> $@ ; \
+	done
+
+build: patch-stamp build-stamp
+
+build-stamp: debian/docbook.install debian/docbook.links debian/docbook.sgmlcatalogs
+	dh_testdir
+	touch build-stamp
+
+clean: unpatch
+	dh_testdir
+	dh_testroot
+	rm -f build-stamp patch-stamp
+	rm -f debian/docbook.install \
+	      debian/docbook.links \
+	      debian/docbook.sgmlcatalogs
+	dh_clean
+
+install: build
+	dh_testdir
+	dh_testroot
+	dh_clean -k
+	dh_installdirs
+
+binary-indep: build
+	dh_testdir
+	dh_testroot
+	dh_installchangelogs ChangeLog.upstream
+	dh_install
+	dh_installdocs debian/README.docbook.dcl
+	dh_installexamples
+	dh_installcatalogs
+	dh_link
+	find $(CURDIR)/debian/$(PACKAGE) -type l -follow >/dev/null # exits 1 on broken links
+	dh_compress
+	dh_fixperms
+	dh_installdeb
+	dh_gencontrol
+	dh_md5sums
+	dh_builddeb
+
+binary-arch: build
+
+binary: binary-indep binary-arch
+
+.PHONY: build clean binary binary-arch binary-indep
+
+# This following code is used to create the .orig.tar.gz tarball.
+# Use it to prepare it for a new release.
+
+version=$(shell dpkg-parsechangelog | sed -n -e 's/^Version: \(.*\)-[^-]*/\1/p')
+
+.PHONY: get-orig-source
+get-orig-source:
+	set -ex ; \
+	TMPDIR=`mktemp -d docbook-$(version).orig` ; \
+	touch "$$TMPDIR"/ChangeLog.upstream ; \
+	for db in 4.5 4.4 4.3 4.2 ; do \
+		mkdir -p "$$TMPDIR"/docbook-$${db} ; \
+		dbzip=docbook-$${db}.zip ; \
+		wget -P "$$TMPDIR" http://www.oasis-open.org/docbook/sgml/$${db}/"$$dbzip" ; \
+		unzip -d "$$TMPDIR"/docbook-$${db} "$$TMPDIR"/"$$dbzip" ; \
+		rm -f "$$TMPDIR"/"$$dbzip" ; \
+		[ -f "$$TMPDIR"/docbook-$${db}/ChangeLog ] && \
+		    cat "$$TMPDIR"/docbook-$${db}/ChangeLog >> "$$TMPDIR"/ChangeLog.upstream ; \
+	done; \
+	for db in 4.1 4.0 3.1 3.0 2.4.1 ; do \
+		mkdir -p "$$TMPDIR"/docbook-$${db} ; \
+		dbver=`echo $${db} | sed -e 's/\.//g'` ; \
+		dbzip=docbk"$$dbver".zip ; \
+		wget -P "$$TMPDIR" http://www.oasis-open.org/docbook/sgml/$${db}/"$$dbzip" ; \
+		unzip -d "$$TMPDIR"/docbook-$${db} "$$TMPDIR"/"$$dbzip" ; \
+		rm -f "$$TMPDIR"/"$$dbzip" ; \
+		cat "$$TMPDIR"/docbook-$${db}/"$$dbver"chg.txt >> "$$TMPDIR"/ChangeLog.upstream ; \
+	done ; \
+	find "$$TMPDIR" -type f ! -perm 644 -exec chmod 644 "{}" ";" ; \
+	GZIP=-9 tar -czf $(CURDIR)/docbook_${version}.orig.tar.gz "$$TMPDIR" ; \
+	rm -rf "$$TMPDIR"
+
--- docbook-4.5.orig/debian/README.docbook.dcl
+++ docbook-4.5/debian/README.docbook.dcl
@@ -0,0 +1,98 @@
+From: Adam Di Carlo <adam@onshore.com>
+Subject: DOCBOOK: objection to docbook.dcl
+To: docbook@lists.oasis-open.org
+Cc: docbook-tc@oasis-open.org
+Date: Wed, 21 Mar 2001 02:35:31 -0500
+
+
+Shipped with the DocBook DTDs from 2.4.1 and up is 'docbook.dcl', an
+SGML declaration for use with DocBook documents.  However, this
+declartion is unnecessarily restrictive, to the level where it is
+rather cumbersome to implement.
+
+My argument is that the DocBook declaration should diverge from the SP
+(and OpenSP) implied declarations only where the divergance expresses
+a real necessity to diverge.  This is based on the principle that
+software (including SGML parsers) should be tolerant of what they
+accept.  The unnecessarily broad divergance of the shipped Docbook
+declaration puts a burden on document engineers using DocBook.
+
+I am considering here only the DocBook SGML DTD, since I presume the
+Declaration is rather irrelevant for XML files, since all XML files
+have the same XML declaration applied to them.
+
+I consider here 'docbook.dcl' as shipped with DocBook 4.1.
+
+Major problems:
+
+ OMITTAG is turned off (why?)
+
+ NAMELEN is too short
+
+ Document Character set is too restrictive
+
+ SUBDOC is turned off (why?)
+
+
+Description:
+
+* OMITTAG is turned off
+
+'OMITTAG' is turned off in 'docbook.dcl', disallowing markup
+minimization of any sort.  This is on in the implied declaration of
+both Jade and OpenJade. This creates problems because documents using
+the default declaration for their parser will have a valid document,
+but if the user decides to be more fasidious and user the docbook SGML
+declaration, sudden their document will not be valid.
+
+The major problem is that trying to turn this on will make a large
+number of existing SGML DocBook instances invalid.
+
+
+* NAMELEN is too short
+
+The NAMELEN quantity set in docbook.dcl is set to 45, rather than the
+default SP NAMELEN of 99999999.
+
+A number of users have complained of problems due to this limitation
+(do a google search on 'docbook namelen' to see what I mean) in any
+cases (such as the SUSE Linux distribution) where the declaration is
+enforced.
+
+Quoting <URL:http://xml.coverpages.org/wlw14.html>:
+
+   Care should be used when changing these since creating a variant
+   syntax may make it difficult for some SGML systems to process
+   documents created with that syntax.  The best means of guaranteeing
+   portability between different SGML systems and applications is to
+   use the reference concrete syntax as much as possible.
+
+One wonders why we need to diverge from the reference concrete syntax
+here.
+
+
+* Document Character set it too restrictive
+
+As an example, to workaround limitations in the support of KOI-R SDATA
+entities in Jade and OpenJade, KOI-R users have to use unicode
+entities.  With the docbook.dcl file, these entities are disallowed,
+although they are perfectly valid with the implied SP declaration.
+Example of being disallowed:
+
+  jade:/usr/share/sgml/entities/sgml-iso-entities-8879.1986/ISOcyr1.ent:1:16:E: \
+  "1072" is not a character number in the document character set
+
+
+* SUBDOC is turned off
+
+Why is it necessary to disallow SUBDOC in DocBook SGML documents?
+Seems like some authors may wish to use this, even if its not fully
+supported by existing stylesheets.
+
+
+
+I hope I got my facts correct, and that this commentary is useful.
+
+-- 
+.....Adam Di Carlo....adam@onshore.com.....<URL:http://www.onshored.com/>
+
--- docbook-4.5.orig/debian/README.Debian
+++ docbook-4.5/debian/README.Debian
@@ -0,0 +1,35 @@
+                  DocBook DTDs for Debian GNU/Linux
+                  ---------------------------------
+
+DocBook is an SGML DTD for software documentation.  It is very
+thorough, and is used by many different organization to produce books,
+articles, references, and even manual pages.
+
+The home page for DocBook is <URL:http://www.oasis-open.org/docbook/>.
+
+A wonderful introduction and reference is available in the
+docbook-defguide package, which can also be read online at
+<URL:http://www.docbook.org/tdg/>.
+
+To format DocBook documents, you'll need a style-sheet package, either
+docbook-dsssl or docbook-xsl, depending on what styling language you
+would like to use.
+
+There are far too many other packages to describe here, both for
+authors, editors, stylers, or information architects.  A list of such
+packages may be perused, for instance, using 'apt-cache search
+docbook'.
+
+If you have a question about how to work with DocBook, there are many
+mailing lists available; see <URL:http://www.docbook.org/>.
+
+If you have suggestions or questions regarding debian-specific issues,
+please consult the debian-sgml@lists.debian.org mailing list,
+<URL:http://lists.debian.org/>.
+
+--
+.....Adam Di Carlo....adam@onshore.com.....<URL:http://www.onshored.com/>
+(previously, Mark Eichin <eichin@thok.org>, Mon, 30 Jun 1997 14:11:38 -0400)
+
+
+
--- docbook-4.5.orig/catalog
+++ docbook-4.5/catalog
@@ -0,0 +1,19 @@
+  -- ...................................................................... --
+  -- Prefer PUBID to SYSID ................................................ --
+OVERRIDE YES
+
+ -- latest DocBook --
+PUBLIC  "-//OASIS//DTD DocBook//EN" "4.5/docbook.dtd"
+ -- disabled, see README.docbook.dcl --
+-- DTDDECL "-//OASIS//DTD DocBook//EN" "4.5/docbook.dcl" --
+
+ -- latest DocBook v4--
+PUBLIC  "-//OASIS//DTD DocBook V4//EN" "4.5/docbook.dtd"
+ -- disabled, see README.docbook.dcl --
+-- DTDDECL "-//OASIS//DTD DocBook V4//EN" "4.5/docbook.dcl" --
+
+ -- latest DocBook v3--
+PUBLIC "-//OASIS//DTD DocBook V3//EN" "3.1/docbook.dtd"
+ -- disabled, see README.docbook.dcl --
+-- DTDDECL "-//OASIS//DTD DocBook V3//EN" "3.1/docbook.dcl" --
+
