diff -up taglib-extras-0.1/taglib-extras-config.cmake.multilib taglib-extras-0.1/taglib-extras-config.cmake
--- taglib-extras-0.1/taglib-extras-config.cmake.multilib	2009-03-12 11:12:04.000000000 -0500
+++ taglib-extras-0.1/taglib-extras-config.cmake	2009-03-23 07:57:25.000000000 -0500
@@ -16,7 +16,6 @@ EOH
 
 prefix=${CMAKE_INSTALL_PREFIX}
 exec_prefix=${CMAKE_INSTALL_PREFIX}
-libdir=${LIB_INSTALL_DIR}
 includedir=${INCLUDE_INSTALL_DIR}
 
 flags=""
@@ -29,7 +28,7 @@ while test $# -gt 0
 do
   case $1 in
     --libs)
-	  flags="$flags -L$libdir -ltag-extras"
+	  flags="$flags -ltag-extras"
 	  ;;
     --cflags)
 	  flags="$flags -I$includedir/taglib-extras"
