--- ./modes/glx/juggler3d.c.orig	2019-09-11 00:06:52.000000000 -0500
+++ ./modes/glx/juggler3d.c	2019-12-06 15:03:43.465002755 -0600
@@ -2729,7 +2729,7 @@
     double spin_accel   = 0.05;
     sp->rot = make_rotator (0, spin_speed, 0, 
                             spin_accel, wander_speed, False);
-    sp->trackball = gltrackball_init ();
+    sp->trackball = gltrackball_init ( 0 );
   }
 
   if (only && *only && strcmp(only, " ")) {
