--- ./xlock/XLock.ad.orig	2002-05-13 09:51:34.000000000 -0700
+++ ./xlock/XLock.ad	2002-05-29 21:25:39.000000000 -0700
@@ -75,6 +75,8 @@
 !XLock.*.bitmap: bitmaps/m-xlock.xbm
 !XLock.*.bitmap: pixmaps/m-xlock.xpm
 !XLock.*.bitmap: /usr/lib/X11/xlock/bitmaps
+! It's best to have no default here, or it kills the bat mode:
+!XLock.*.bitmap: /usr/X11R6/include/X11/bitmaps/s-bob.xbm
 
 ! For mailbox check application
 !XLock.mailCmd: /usr/ucb/mail -e
@@ -165,6 +167,8 @@
 !XLock.bounce.cycles: 1
 XLock.bounce.size: 0
 XLock.bounce.saturation: 1.0
+! "Bob" is taking over the bounce mode by default:
+XLock.bounce.bitmap: /usr/X11R6/include/X11/bitmaps/s-bob.xbm
 XLock.braid.delay: 1000
 XLock.braid.count: 15
 XLock.braid.cycles: 100
