--- ./CMakeLists.txt.orig	2024-02-09 15:59:05.000000000 -0600
+++ ./CMakeLists.txt	2025-03-15 16:53:09.381576999 -0500
@@ -64,11 +64,6 @@
     set_package_properties(PolyClipping PROPERTIES TYPE OPTIONAL PURPOSE "Needed only for processing coverage area data from the Transport API Repository (you don't need this for distribution)")
 endif()
 
-if (NOT WIN32)
-    find_package(Protobuf)
-    set_package_properties(Protobuf PROPERTIES TYPE OPTIONAL PURPOSE "Parsing of OSM PBF files.")
-endif()
-
 if (UNIX AND NOT ANDROID)
     find_package(KF${KF_MAJOR_VERSION}NetworkManagerQt)
     set_package_properties(KF${KF_MAJOR_VERSION}NetworkManagerQt PROPERTIES TYPE OPTIONAL PURPOSE "Onboard status detection.")
