diff -up kbd-1.15/man/man1/loadkeys.1.in_old kbd-1.15/man/man1/loadkeys.1.in
--- kbd-1.15/man/man1/loadkeys.1.in_old	2008-10-23 21:03:59.000000000 +0200
+++ kbd-1.15/man/man1/loadkeys.1.in	2009-03-05 13:11:08.000000000 +0100
@@ -19,6 +19,8 @@ loadkeys \- load keyboard translation ta
 ] [
 .I -m --mktable
 ] [
+.I -q --quiet
+] [
 .I -s --clearstrings
 ] [
 .I -v --verbose
@@ -148,6 +150,10 @@ command (and does not modify the current
 .B loadkeys
 prints its version number and a short usage message to the programs
 standard error output and exits.
+.TP
+.B \-q \-\-quiet
+.B loadkeys
+suppresses all normal output.
 .SH WARNING
 Note that anyone having read access to
 .B /dev/console
diff -up kbd-1.15/src/loadkeys.c_old kbd-1.15/src/loadkeys.c
--- kbd-1.15/src/loadkeys.c_old	2008-10-23 21:03:59.000000000 +0200
+++ kbd-1.15/src/loadkeys.c	2009-03-05 13:09:00.000000000 +0100
@@ -1966,6 +1966,7 @@ usage(void) {
 "  -d --default       load \"%s\"\n"
 "  -h --help          display this help text\n"
 "  -m --mktable       output a \"defkeymap.c\" to stdout\n"
+"  -q --quiet         suppress all normal output\n"
 "  -s --clearstrings  clear kernel string table\n"
 "  -u --unicode       implicit conversion to Unicode\n"
 "  -v --verbose       report the changes\n"), PACKAGE_VERSION, DEFMAP);
