diff -up kbd-1.15/po/Makefile.in_old kbd-1.15/po/Makefile.in
--- kbd-1.15/po/Makefile.in_old	2008-10-23 21:03:59.000000000 +0200
+++ kbd-1.15/po/Makefile.in	2008-12-09 15:35:58.000000000 +0100
@@ -4,10 +4,10 @@ srcdir = @srcdir@
 
 prefix        = @prefix@
 datarootdir   = @datarootdir@
-datadir       = @datadir@
-localedir     = $(DESTDIR)$(datadir)/locale
-gnulocaledir  = $(DESTDIR)$(datadir)/locale
-gettextsrcdir = $(DESTDIR)$(datadir)/gettext/po
+datadir       = $(DESTDIR)@datadir@
+localedir     = $(DESTDIR)@localedir@
+gnulocaledir  = $(DESTDIR)@localedir@
+gettextsrcdir = $(datadir)/gettext/po
 
 FOREIGN       = @FOREIGN@
 HAVE_XGETTEXT = @HAVE_XGETTEXT@
@@ -36,10 +36,10 @@ INCLUDES = -I.. -I$(INTL)
 COMPILE = $(CC) -c $(DEFS) $(INCLUDES) $(CPPFLAGS) $(CFLAGS) $(XCFLAGS)
 
 # Enter here all .po files
-POFILES =  cs.po da.po de.po el.po es.po fr.po gr.po nl.po pl.po \
+POFILES =  cs.po da.po de.po el.po es.po fr.po nl.po pl.po \
  ro.po ru.po sv.po tr.po
 # the same but with .gmo
-GMOFILES = cs.gmo da.gmo de.gmo el.gmo es.gmo fr.gmo gr.gmo nl.gmo \
+GMOFILES = cs.gmo da.gmo de.gmo el.gmo es.gmo fr.gmo nl.gmo \
  pl.gmo ro.gmo ru.gmo sv.gmo tr.gmo
 
 CATALOGS = $(GMOFILES)
