--- Makefile,orig	2009-03-04 04:27:22.000000000 -0300
+++ Makefile		2009-03-04 04:34:01.000000000 -0300
@@ -13,6 +13,7 @@
 DEFS= curs.h netwatch.h
 OBJECTS= curs.o dispdata.o services.o netwatch.o processinetrc.o gh.o warning.o semaphore.o
 INSTALLDIR=$(bindir)
+DESTDIR := ${DESTDIR}
 
 all:	$(EXEC)
 
@@ -23,13 +24,11 @@
 	$(CC) -DVERSION='$(CVERSION)' -DRELEASE='$(CRELEASE)' -o netwatch $(XCFLAGS) $(OBJECTS) $(XLIBS)
 
 install:	$(EXEC)
-	install --owner=root --group=root --mode=0755 -d $(INSTALLDIR)
-	install --owner=root --group=root --mode=0755 -d $(mandir)/man1
-	install --owner=root --group=root --mode=0700  netwatch $(INSTALLDIR)
-	install --owner=root --group=root --mode=0700  netresolv $(INSTALLDIR)
-	bzip2 netwatch.1
-	install -D  netwatch.1.bz2 $(mandir)/man1
-	bunzip2 netwatch.1.bz2
+	install --owner=root --group=root --mode=0755 -d $(DESTDIR)$(INSTALLDIR)
+	install --owner=root --group=root --mode=0755 -d $(DESTDIR)$(mandir)
+	install --owner=root --group=root --mode=0755  netwatch $(DESTDIR)$(INSTALLDIR)
+	install --owner=root --group=root --mode=0755  netresolv $(DESTDIR)$(INSTALLDIR)
+	install --owner=root --group=root --mode=644   netwatch.1 $(DESTDIR)$(mandir)
 	
 tar:
 	tar -C .. -czf ../netwatch-$(VERSION)-$(RELEASE).tgz netwatch-$(VERSION)
