--- ./Makefile.orig	2006-03-05 04:36:54.000000000 -0600
+++ ./Makefile	2024-05-15 19:42:54.924020444 -0500
@@ -5,26 +5,26 @@
 INSTALL = install -c
 CFLAGS = -Wall -O2 -fomit-frame-pointer
 # You may use OpenSSL's MD5 routines instead of the ones supplied here
-#CFLAGS += -DHAVE_OPENSSL
+CFLAGS += -DHAVE_OPENSSL
 LDFLAGS = -s
 LIBS =
 # Linux with glibc, FreeBSD, NetBSD
-#LIBS += -lcrypt
+LIBS += -lcrypt
 # HP-UX trusted system
 #LIBS += -lsec
 # Solaris (POP_STANDALONE, POP_VIRTUAL)
 #LIBS += -lsocket -lnsl
 # PAM
-#LIBS += -lpam
+LIBS += -lpam
 # TCP wrappers
 #LIBS += -lwrap
 # libwrap may also want this
 #LIBS += -lnsl
 # OpenSSL (-DHAVE_OPENSSL)
-#LIBS += -lcrypto
+LIBS += -lcrypto
 
 DESTDIR =
-PREFIX = /usr/local
+PREFIX = /usr
 SBINDIR = $(PREFIX)/sbin
 MANDIR = $(PREFIX)/man
 
--- ./params.h.orig	2006-03-05 07:18:32.000000000 -0600
+++ ./params.h	2024-05-15 19:43:21.877021047 -0500
@@ -13,7 +13,7 @@
 /*
  * Are we going to be a standalone server or start via an inetd clone?
  */
-#define POP_STANDALONE			0
+#define POP_STANDALONE			1
 
 #if POP_STANDALONE
 
@@ -103,7 +103,7 @@
  * A pseudo-user to run as before authentication.  The user and its UID
  * must not be used for any other purpose.
  */
-#define POP_USER			POP_SERVER
+#define POP_USER			"pop"
 
 /*
  * An empty directory to chroot to before authentication.  The directory
@@ -155,8 +155,8 @@
  * Note that there's no built-in password aging support.
  */
 #define AUTH_PASSWD			0
-#define AUTH_SHADOW			1
-#define AUTH_PAM			0
+#define AUTH_SHADOW			0
+#define AUTH_PAM			1
 #define AUTH_PAM_USERPASS		0
 #define USE_LIBPAM_USERPASS		0
 
@@ -191,7 +191,7 @@
  *
  * #undef this for qmail-style $HOME/Mailbox mailboxes.
  */
-#define MAIL_SPOOL_PATH			"/var/mail"
+#define MAIL_SPOOL_PATH			"/var/spool/mail"
 
 #ifndef MAIL_SPOOL_PATH
 /*
