diff -up ../popa3d-1.0.3.orig/Makefile ./Makefile
--- ../popa3d-1.0.3.orig/Makefile	2006-03-05 05:36:54.000000000 -0500
+++ ./Makefile	2018-09-18 12:56:05.412312376 -0400
@@ -5,26 +5,26 @@ MKDIR = mkdir -p
 INSTALL = install -c
 CFLAGS = -Wall -O2 -fomit-frame-pointer
 # You may use OpenSSL's MD5 routines instead of the ones supplied here
-#CFLAGS += -DHAVE_OPENSSL
+CFLAGS += -DHAVE_OPENSSL
 LDFLAGS = -s
 LIBS =
 # Linux with glibc, FreeBSD, NetBSD
-#LIBS += -lcrypt
+LIBS += -lcrypt
 # HP-UX trusted system
 #LIBS += -lsec
 # Solaris (POP_STANDALONE, POP_VIRTUAL)
 #LIBS += -lsocket -lnsl
 # PAM
-#LIBS += -lpam
+LIBS += -lpam
 # TCP wrappers
 #LIBS += -lwrap
 # libwrap may also want this
 #LIBS += -lnsl
 # OpenSSL (-DHAVE_OPENSSL)
-#LIBS += -lcrypto
+LIBS += -lcrypto
 
 DESTDIR =
-PREFIX = /usr/local
+PREFIX = /usr
 SBINDIR = $(PREFIX)/sbin
 MANDIR = $(PREFIX)/man
 
Common subdirectories: ../popa3d-1.0.3.orig/md5 and ./md5
diff -up ../popa3d-1.0.3.orig/params.h ./params.h
--- ../popa3d-1.0.3.orig/params.h	2006-03-05 08:18:32.000000000 -0500
+++ ./params.h	2018-09-18 12:55:55.953210742 -0400
@@ -13,7 +13,7 @@
 /*
  * Are we going to be a standalone server or start via an inetd clone?
  */
-#define POP_STANDALONE			0
+#define POP_STANDALONE			1
 
 #if POP_STANDALONE
 
@@ -103,7 +103,7 @@
  * A pseudo-user to run as before authentication.  The user and its UID
  * must not be used for any other purpose.
  */
-#define POP_USER			POP_SERVER
+#define POP_USER			"pop"
 
 /*
  * An empty directory to chroot to before authentication.  The directory
@@ -191,7 +191,7 @@
  *
  * #undef this for qmail-style $HOME/Mailbox mailboxes.
  */
-#define MAIL_SPOOL_PATH			"/var/mail"
+#define MAIL_SPOOL_PATH			"/var/spool/mail"
 
 #ifndef MAIL_SPOOL_PATH
 /*
