--- ./Makefile.orig	2024-05-11 14:15:26.383948970 -0500
+++ ./Makefile	2024-05-11 14:16:03.483949800 -0500
@@ -3,7 +3,7 @@
 RM = rm -f
 MKDIR = mkdir -p
 INSTALL = install -c
-CFLAGS = -Wall -O2 -fomit-frame-pointer
+CFLAGS = -Wall -O2 -fomit-frame-pointer -Wno-error=implicit-function-declaration -Wno-error=int-conversion
 # You may use OpenSSL's MD5 routines instead of the ones supplied here
 CFLAGS += -DHAVE_OPENSSL
 LDFLAGS = -s
