--- ./sys/v4l2/gstv4l2object.c.orig	2011-03-08 07:56:07.000000000 -0600
+++ ./sys/v4l2/gstv4l2object.c	2011-04-15 22:19:11.802999999 -0500
@@ -2019,7 +2019,7 @@
     /* ideally we would differentiate between types of interlaced video
      * but there is not sufficient information in the caps..
      */
-    field = V4L2_FIELD_SEQ_TB;
+    field = V4L2_FIELD_INTERLACED;
   } else {
     GST_DEBUG_OBJECT (v4l2object->element, "progressive video");
     field = V4L2_FIELD_NONE;
