--- ./kwrite/main.cpp.orig	2021-08-12 21:30:09.752256555 -0500
+++ ./kwrite/main.cpp	2021-08-12 21:32:05.046259134 -0500
@@ -41,17 +41,15 @@
     // Prohibit using sudo or kdesu (but allow using the root user directly)
     if (getuid() == 0) {
         if (!qEnvironmentVariableIsEmpty("SUDO_USER")) {
-            std::cout << "Executing KWrite with sudo is not possible due to unfixable security vulnerabilities. "
+            std::cout << "Executing KWrite with sudo is UNSAFE due to unfixable security vulnerabilities. "
                          "It is also not necessary; simply use KWrite normally, and you will be prompted for "
                          "elevated privileges when saving documents if needed."
                       << std::endl;
-            return EXIT_FAILURE;
         } else if (!qEnvironmentVariableIsEmpty("KDESU_USER")) {
-            std::cout << "Executing KWrite with kdesu is not possible due to unfixable security vulnerabilities. "
+            std::cout << "Executing KWrite with kdesu is UNSAFE due to unfixable security vulnerabilities. "
                          "It is also not necessary; simply use KWrite normally, and you will be prompted for "
                          "elevated privileges when saving documents if needed."
                       << std::endl;
-            return EXIT_FAILURE;
         }
     }
 #endif
--- ./kate/main.cpp.orig	2021-08-04 21:13:14.000000000 -0500
+++ ./kate/main.cpp	2021-08-12 21:30:09.753256555 -0500
@@ -49,17 +49,15 @@
     // Prohibit using sudo or kdesu (but allow using the root user directly)
     if (getuid() == 0) {
         if (!qEnvironmentVariableIsEmpty("SUDO_USER")) {
-            std::cout << "Executing Kate with sudo is not possible due to unfixable security vulnerabilities. "
+            std::cout << "Executing Kate with sudo is UNSAFE due to unfixable security vulnerabilities. "
                          "It is also not necessary; simply use Kate normally, and you will be prompted for "
                          "elevated privileges when saving documents if needed."
                       << std::endl;
-            return EXIT_FAILURE;
         } else if (!qEnvironmentVariableIsEmpty("KDESU_USER")) {
-            std::cout << "Executing Kate with kdesu is not possible due to unfixable security vulnerabilities. "
+            std::cout << "Executing Kate with kdesu is UNSAFE due to unfixable security vulnerabilities. "
                          "It is also not necessary; simply use Kate normally, and you will be prompted for "
                          "elevated privileges when saving documents if needed."
                       << std::endl;
-            return EXIT_FAILURE;
         }
     }
 #endif
