--- ./kwrite/main.cpp.orig	2021-04-22 13:26:42.398123759 -0500
+++ ./kwrite/main.cpp	2021-04-22 13:29:08.356127024 -0500
@@ -38,20 +38,18 @@
 extern "C" Q_DECL_EXPORT int main(int argc, char **argv)
 {
 #ifndef Q_OS_WIN
-    // Prohibit using sudo or kdesu (but allow using the root user directly)
+    // Warn using sudo or kdesu (but allow using the root user directly)
     if (getuid() == 0) {
         if (!qEnvironmentVariableIsEmpty("SUDO_USER")) {
-            std::cout << "Executing KWrite with sudo is not possible due to unfixable security vulnerabilities. "
+            std::cout << "Executing KWrite with sudo is UNSAFE due to unfixable security vulnerabilities. "
                          "It is also not necessary; simply use KWrite normally, and you will be prompted for "
                          "elevated privileges when saving documents if needed."
                       << std::endl;
-            return EXIT_FAILURE;
         } else if (!qEnvironmentVariableIsEmpty("KDESU_USER")) {
-            std::cout << "Executing KWrite with kdesu is not possible due to unfixable security vulnerabilities. "
+            std::cout << "Executing KWrite with kdesu is UNSAFE due to unfixable security vulnerabilities. "
                          "It is also not necessary; simply use KWrite normally, and you will be prompted for "
                          "elevated privileges when saving documents if needed."
                       << std::endl;
-            return EXIT_FAILURE;
         }
     }
 #endif
--- ./kate/main.cpp.orig	2021-04-22 13:26:42.397123759 -0500
+++ ./kate/main.cpp	2021-04-22 13:30:18.674128597 -0500
@@ -49,17 +49,15 @@
     // Prohibit using sudo or kdesu (but allow using the root user directly)
     if (getuid() == 0) {
         if (!qEnvironmentVariableIsEmpty("SUDO_USER")) {
-            std::cout << "Executing Kate with sudo is not possible due to unfixable security vulnerabilities. "
+            std::cout << "Executing Kate with sudo is UNSAFE due to unfixable security vulnerabilities. "
                          "It is also not necessary; simply use Kate normally, and you will be prompted for "
                          "elevated privileges when saving documents if needed."
                       << std::endl;
-            return EXIT_FAILURE;
         } else if (!qEnvironmentVariableIsEmpty("KDESU_USER")) {
-            std::cout << "Executing Kate with kdesu is not possible due to unfixable security vulnerabilities. "
+            std::cout << "Executing Kate with kdesu is UNSAFE due to unfixable security vulnerabilities. "
                          "It is also not necessary; simply use Kate normally, and you will be prompted for "
                          "elevated privileges when saving documents if needed."
                       << std::endl;
-            return EXIT_FAILURE;
         }
     }
 #endif
