--- ./kwrite/main.cpp.orig	2022-04-22 01:59:34.794311631 -0500
+++ ./kwrite/main.cpp	2022-04-22 02:01:06.318313678 -0500
@@ -46,13 +46,11 @@
                          "Instead use KWrite normally and you will be prompted for elevated privileges when "
                          "saving documents if needed."
                       << std::endl;
-            return EXIT_FAILURE;
         } else if (!qEnvironmentVariableIsEmpty("KDESU_USER")) {
             std::cout << "Running KWrite with kdesu can cause bugs and expose you to security vulnerabilities. "
                          "Instead use KWrite normally and you will be prompted for elevated privileges when "
                          "saving documents if needed."
                       << std::endl;
-            return EXIT_FAILURE;
         }
     }
 #endif
--- ./kate/main.cpp.orig	2022-04-22 01:59:34.794311631 -0500
+++ ./kate/main.cpp	2022-04-22 02:01:46.125314569 -0500
@@ -54,13 +54,11 @@
                          "Instead use Kate normally and you will be prompted for elevated privileges when "
                          "saving documents if needed."
                       << std::endl;
-            return EXIT_FAILURE;
         } else if (!qEnvironmentVariableIsEmpty("KDESU_USER")) {
             std::cout << "Running Kate with kdesu can cause bugs and expose you to security vulnerabilities. "
                          "Instead use Kate normally and you will be prompted for elevated privileges when "
                          "saving documents if needed."
                       << std::endl;
-            return EXIT_FAILURE;
         }
     }
 #endif
