--- ./Makefile.cfg.orig	2022-01-19 16:04:36.833030933 -0600
+++ ./Makefile.cfg	2022-01-19 16:05:13.758031759 -0600
@@ -79,7 +79,7 @@
 
 # uncomment this if your compiler fails on compiling the assembler in
 # src/vgaconvplanar.c, gl/inlstring.h, gl/line.c or gl/scale.c
-# NO_ASM = y
+NO_ASM = y
 
 # uncomment if you want to set attribute controller and dac without delay
 # This breaks original VGA, but seems to work on new cards.
