--- ./Makefile.cfg.orig	2007-02-04 15:20:18.000000000 -0600
+++ ./Makefile.cfg	2007-02-04 15:21:18.000000000 -0600
@@ -44,7 +44,7 @@
 # Common prefix for installation directories.
 # NOTE: This directory must exist when you start the install.
 TOPDIR=
-prefix = $(TOPDIR)/usr/local
+prefix = $(TOPDIR)/usr
 exec_prefix = $(prefix)
 
 # Directory where the shared stubs and static library will be installed.
