From 37058848c82512c5267961ed09c279ef78341ad5 Mon Sep 17 00:00:00 2001
From: Shi Pujin <shipujin.t@gmail.com>
Date: Mon, 23 Dec 2024 03:04:03 +0000
Subject: [PATCH] qtquick3dphysics: add LoongArch support


diff --git a/qtquick3dphysics/src/3rdparty/PhysX/pxshared/include/foundation/PxPreprocessor.h b/qtquick3dphysics/src/3rdparty/PhysX/pxshared/include/foundation/PxPreprocessor.h
index 630c7c04f9..f483912f1f 100644
--- a/qtquick3dphysics/src/3rdparty/PhysX/pxshared/include/foundation/PxPreprocessor.h
+++ b/qtquick3dphysics/src/3rdparty/PhysX/pxshared/include/foundation/PxPreprocessor.h
@@ -127,6 +127,8 @@ Architecture defines, see http://sourceforge.net/p/predef/wiki/Architectures/
 #define PX_PPC 1
 #elif defined(__mips__)
 #define PX_X64 1
+#elif defined(__loongarch_lp64)
+#define PX_LA64 1
 #elif defined(__EMSCRIPTEN__)
 #define PX_WASM
     #if defined(__LP64__)
@@ -154,7 +156,7 @@ SIMD defines
 #endif
 
 /** Disable SIMD for webassembly, mips and arm64 */
-#if defined(__EMSCRIPTEN__) || defined(__mips__) || defined(_M_ARM64) || defined(_M_ARM)
+#if defined(__EMSCRIPTEN__) || defined(__mips__) || defined(_M_ARM64) || defined(_M_ARM) || defined(__loongarch__)
 #define PX_SIMD_DISABLED 1
 #endif
 
@@ -278,7 +280,7 @@ family shortcuts
 // architecture
 #define PX_INTEL_FAMILY (PX_X64 || PX_X86)
 #define PX_ARM_FAMILY (PX_ARM || PX_A64)
-#define PX_P64_FAMILY (PX_X64 || PX_A64 || PX_WASM_64) // shortcut for 64-bit architectures
+#define PX_P64_FAMILY (PX_X64 || PX_A64 || PX_WASM_64 || PX_LA64) // shortcut for 64-bit architectures
 
 /**
 C++ standard library defines
diff --git a/qtquick3dphysics/src/3rdparty/PhysX/source/physxextensions/src/serialization/SnSerialUtils.cpp b/qtquick3dphysics/src/3rdparty/PhysX/source/physxextensions/src/serialization/SnSerialUtils.cpp
index aa53797403..ed12a93fd3 100644
--- a/qtquick3dphysics/src/3rdparty/PhysX/source/physxextensions/src/serialization/SnSerialUtils.cpp
+++ b/qtquick3dphysics/src/3rdparty/PhysX/source/physxextensions/src/serialization/SnSerialUtils.cpp
@@ -39,7 +39,7 @@ using namespace physx;
 namespace
 {
 
-#define SN_NUM_BINARY_PLATFORMS 19
+#define SN_NUM_BINARY_PLATFORMS 20
 const PxU32 sBinaryPlatformTags[SN_NUM_BINARY_PLATFORMS] =
 {
 	PX_MAKE_FOURCC('W','_','3','2'),
@@ -59,6 +59,7 @@ const PxU32 sBinaryPlatformTags[SN_NUM_BINARY_PLATFORMS] =
 	PX_MAKE_FOURCC('W','A','3','2'),
 	PX_MAKE_FOURCC('W','A','6','4'),
 	PX_MAKE_FOURCC('L','A','3','2'),
+	PX_MAKE_FOURCC('L','A','6','4'),
 	PX_MAKE_FOURCC('U','N','K','_'),
 };
 
@@ -82,6 +83,7 @@ const char* sBinaryPlatformNames[SN_NUM_BINARY_PLATFORMS] =
 	"uwparm64",
 	"macarm",
 	"linuxarm",
+	"loong64",
 	"unknown",
 };
 
@@ -127,8 +129,10 @@ PxU32 getBinaryPlatformTag()
     return sBinaryPlatformTags[16];
 #elif PX_LINUX && PX_ARM
 	return sBinaryPlatformTags[17];
-#else
+#elif PX_LINUX && PX_LA64
 	return sBinaryPlatformTags[18];
+#else
+	return sBinaryPlatformTags[19];
 #endif
 }
 
-- 
2.41.0

