--- ./qtwebengine/configure.cmake.orig	2024-02-27 13:57:02.000000000 -0600
+++ ./qtwebengine/configure.cmake	2024-02-27 17:49:26.845071893 -0600
@@ -492,13 +492,6 @@
    MESSAGE "node.js version 14 or later is required."
 )
 add_check_for_support(
-    MODULES QtWebEngine
-    CONDITION NOT (Nodejs_ARCH STREQUAL "ia32") AND
-              NOT (Nodejs_ARCH STREQUAL "x86") AND
-              NOT (Nodejs_ARCH STREQUAL "arm")
-    MESSAGE "32bit version of Nodejs is not supported."
-)
-add_check_for_support(
    MODULES QtWebEngine QtPdf
    CONDITION Python3_EXECUTABLE
    MESSAGE "Python version 3.6 or later is required."
