--- ./gtkspell/gtkspell.c.orig	2009-10-09 14:01:47.000000000 -0500
+++ ./gtkspell/gtkspell.c	2021-06-29 14:00:00.570508124 -0500
@@ -277,7 +277,7 @@
 	get_word_extents_from_mark(spell->buffer, &start, &end, spell->mark_click);
 	word = gtk_text_buffer_get_text(spell->buffer, &start, &end, FALSE);
 	
-	enchant_dict_add_to_pwl( spell->speller, word, strlen(word));
+	enchant_dict_add( spell->speller, word, strlen(word));
 
 	gtkspell_recheck_all(spell);
 
--- ./configure.ac.orig	2009-10-22 21:52:31.000000000 -0500
+++ ./configure.ac	2021-06-29 13:58:26.549513324 -0500
@@ -12,12 +12,12 @@
 AM_INIT_AUTOMAKE(AC_PACKAGE_NAME, AC_PACKAGE_VERSION)
 AC_CONFIG_HEADERS([config.h])
 
-SPELLER_LIB=-lenchant
+SPELLER_LIB=-lenchant-2
       
 AC_SUBST(SPELLER_LIB)
 GTKSPELL_PACKAGES=gtk+-2.0
 AC_SUBST(GTKSPELL_PACKAGES)
-PKG_CHECK_MODULES(GTKSPELL, $GTKSPELL_PACKAGES enchant >= 0.4.0 )
+PKG_CHECK_MODULES(GTKSPELL, $GTKSPELL_PACKAGES enchant-2 >= 2.2.0 )
 AC_SUBST(GTKSPELL_CFLAGS)
 AC_SUBST(GTKSPELL_LIBS)
 
