From 2de7bbe7e79ccf648ff4ab4c59c7cf969bdb3185 Mon Sep 17 00:00:00 2001
From: Xiaolin Zhao <zhaoxiaolin@loongson.cn>
Date: Wed, 27 Jul 2022 18:02:20 +0800
Subject: [PATCH 12/21] LoongArch64: Add CPU feature detection when init JIT
 compiler

Signed-off-by: Shi Pujin <shipujin@loongson.cn>
---
 src/lib_jit.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/lib_jit.c b/src/lib_jit.c
index c0294927..ec954f21 100644
--- a/src/lib_jit.c
+++ b/src/lib_jit.c
@@ -720,6 +720,10 @@ static uint32_t jit_cpudetect(void)
   }
 #endif
 
+#elif LJ_TARGET_LOONGARCH64
+
+  /* No optional CPU features to detect (for now). */
+
 #else
 #error "Missing CPU detection for this architecture"
 #endif
-- 
2.45.2

