From 2270a717c3c527d68f3f985d9d5a390a94db0905 Mon Sep 17 00:00:00 2001
From: Xiaolin Zhao <zhaoxiaolin@loongson.cn>
Date: Thu, 28 Jul 2022 10:24:50 +0800
Subject: [PATCH 18/21] LoongArch64: Add loongarch64 support when save/list
 bytecode

Signed-off-by: Shi Pujin <shipujin@loongson.cn>
---
 src/jit/bcsave.lua | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/jit/bcsave.lua b/src/jit/bcsave.lua
index 74699f3d..d423933b 100644
--- a/src/jit/bcsave.lua
+++ b/src/jit/bcsave.lua
@@ -97,6 +97,7 @@ local map_arch = {
   mips64el =	{ e = "le", b = 64, m = 8, f = 0x80000007, },
   mips64r6 =	{ e = "be", b = 64, m = 8, f = 0xa0000407, },
   mips64r6el =	{ e = "le", b = 64, m = 8, f = 0xa0000407, },
+  loongarch64 =	{ e = "le", b = 64, m = 258, f = 0x3, },
 }
 
 local map_os = {
-- 
2.45.2

