From d0af35f2eb7f1da7e2b6af6507b8746ed223686a Mon Sep 17 00:00:00 2001
From: Xiaolin Zhao <zhaoxiaolin@loongson.cn>
Date: Thu, 28 Jul 2022 11:02:28 +0800
Subject: [PATCH 20/21] LoongArch64: Add support in Makefile

Signed-off-by: Shi Pujin <shipujin@loongson.cn>
---
 Makefile     |  1 +
 src/Makefile | 10 +++++++++-
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index ae194428..6fb58319 100644
--- a/Makefile
+++ b/Makefile
@@ -100,6 +100,7 @@ FILES_JITLIB= bc.lua bcsave.lua dump.lua p.lua v.lua zone.lua \
 	      dis_arm64be.lua dis_ppc.lua dis_mips.lua dis_mipsel.lua \
 	      dis_mips64.lua dis_mips64el.lua \
 	      dis_mips64r6.lua dis_mips64r6el.lua \
+	      dis_loongarch64.lua \
 	      vmdef.lua
 
 ifeq (,$(findstring Windows,$(OS)))
diff --git a/src/Makefile b/src/Makefile
index 3a6a4329..622a08dc 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -52,6 +52,7 @@ CCOPT_arm=
 CCOPT_arm64=
 CCOPT_ppc=
 CCOPT_mips=
+CCOPT_loongarch64= -fwrapv
 #
 CCDEBUG=
 # Uncomment the next line to generate debug information:
@@ -243,6 +244,10 @@ else
 ifneq (,$(findstring LJ_TARGET_ARM ,$(TARGET_TESTARCH)))
   TARGET_LJARCH= arm
 else
+ifneq (,$(findstring LJ_TARGET_LOONGARCH64 ,$(TARGET_TESTARCH)))
+  TARGET_ARCH= -DLJ_ARCH_ENDIAN=LUAJIT_LE
+  TARGET_LJARCH= loongarch64
+else
 ifneq (,$(findstring LJ_TARGET_ARM64 ,$(TARGET_TESTARCH)))
   ifneq (,$(findstring __AARCH64EB__ ,$(TARGET_TESTARCH)))
     TARGET_ARCH= -D__AARCH64EB__=1
@@ -274,6 +279,7 @@ endif
 endif
 endif
 endif
+endif
 
 ifneq (,$(findstring LJ_TARGET_PS3 1,$(TARGET_TESTARCH)))
   TARGET_SYS= PS3
@@ -337,7 +343,9 @@ else
     # Find out whether the target toolchain always generates unwind tables.
     TARGET_TESTUNWIND=$(shell exec 2>/dev/null; echo 'extern void b(void);int a(void){b();return 0;}' | $(TARGET_CC) -c -x c - -o tmpunwind.o && { grep -qa -e eh_frame -e __unwind_info tmpunwind.o || grep -qU -e eh_frame -e __unwind_info tmpunwind.o; } && echo E; rm -f tmpunwind.o)
     ifneq (,$(findstring E,$(TARGET_TESTUNWIND)))
-      TARGET_XCFLAGS+= -DLUAJIT_UNWIND_EXTERNAL
+      ifeq (,$(findstring LJ_TARGET_LOONGARCH64 ,$(TARGET_TESTARCH)))
+        TARGET_XCFLAGS+= -DLUAJIT_UNWIND_EXTERNAL
+      endif
     endif
   endif
   ifneq (SunOS,$(TARGET_SYS))
-- 
2.45.2

