--- ./man/pulse-client.conf.5.xml.in.orig	2016-01-06 15:33:09.797022947 -0600
+++ ./man/pulse-client.conf.5.xml.in	2016-01-06 15:35:34.902019857 -0600
@@ -73,10 +73,7 @@
     <option>
       <p><opt>allow-autospawn-for-root=</opt> Allow autospawning also for root.
       Takes a boolean value, defaults to <opt>no</opt>. If the <opt>autospawn
-      </opt> option is disabled, this option has no effect. Autospawning for
-      root is disabled by default, because running PulseAudio as root will
-      interfere with regular users' PulseAudio instances. This option should be
-      enabled only in environments where there are no regular users at all.</p>
+      </opt> option is disabled, this option has no effect.</p>
     </option>
 
     <option>
--- ./src/pulse/client.conf.in.orig	2016-01-06 15:33:09.800022947 -0600
+++ ./src/pulse/client.conf.in	2016-01-06 15:39:33.502014775 -0600
@@ -22,8 +22,8 @@
 ; default-server =
 ; default-dbus-server =
 
-; autospawn = yes
-; allow-autospawn-for-root = no
+autospawn = yes
+allow-autospawn-for-root = yes
 ; daemon-binary = @PA_BINARY@
 ; extra-arguments = --log-target=syslog
 
