diff -up pulseaudio-6.0/src/daemon/start-pulseaudio-x11.in.autostart pulseaudio-6.0/src/daemon/start-pulseaudio-x11.in
--- pulseaudio-6.0/src/daemon/start-pulseaudio-x11.in.autostart	2015-02-12 08:10:35.000000000 -0600
+++ pulseaudio-6.0/src/daemon/start-pulseaudio-x11.in	2015-06-22 11:24:13.561614127 -0500
@@ -17,6 +17,9 @@
 
 set -e
 
+# probe to test if autospawn works, else resort to starting manually
+@PACTL_BINARY@ info > /dev/null 2>&1 || @PA_BINARY@ --start "$@"
+
 if [ x"$DISPLAY" != x ] ; then
 
     @PACTL_BINARY@ load-module module-x11-publish "display=$DISPLAY" > /dev/null
