--- ./CMakeLists.txt.orig	2024-01-11 15:29:12.000000000 -0600
+++ ./CMakeLists.txt	2024-03-19 22:59:23.005420760 -0500
@@ -16,28 +16,6 @@
 find_package(Qt5 5.2.0 CONFIG REQUIRED Core Gui Widgets)
 find_package(KF5 5.11.0 REQUIRED Kross)
 
-# ========================= Find Python 2 =========================
-
-# TODO: CMake's FindPythonLibs.cmake needs fixing, because without the
-# following hack find_package(PythonLibs 3) would find the previously found
-# version 2.7. Due to this hack one cannot predefine PYTHON_LIBRARY and/or
-# PYTHON_INCLUDE_DIR by editing CMakeCache.txt.
-unset(PYTHON_INCLUDE_DIR CACHE)
-unset(PYTHON_LIBRARY CACHE)
-unset(PYTHON_LIBRARY_DEBUG CACHE)
-# Unset variables set by find_package(PythonInterp) which is called by KI18n
-unset(PYTHON_VERSION_MAJOR)
-unset(PYTHON_VERSION_MINOR)
-find_package(PythonLibs 2 EXACT)
-
-set(PYTHON2_LIBS_FOUND ${PYTHONLIBS_FOUND})
-set(PYTHON2_LIBRARIES ${PYTHON_LIBRARIES})
-set(PYTHON2_INCLUDE_DIRS ${PYTHON_INCLUDE_DIRS})
-set(PYTHON2_LIBS_VERSION_STRING ${PYTHONLIBS_VERSION_STRING})
-if(PYTHON2_LIBS_FOUND)
-    message(STATUS "Found Python 2: ${PYTHON2_LIBRARIES} (found version \"${PYTHON2_LIBS_VERSION_STRING}\")")
-endif()
-
 # ========================= Find FALCON =========================
 
 ### Falcon support disabled because it does not work with most modern falcon-0.9.x
@@ -61,10 +39,6 @@
 
 # ========================= Traverse subdirectories for actual targets =========================
 
-if(PYTHON2_LIBS_FOUND)
-    ecm_optional_add_subdirectory(python)
-endif(PYTHON2_LIBS_FOUND)
-
 ### Falcon support disabled because it does not work with most modern falcon-0.9.x
 # if(FALCON_FOUND)
 #     ecm_optional_add_subdirectory(falcon)
