diff -Nur parted-2.1.orig/configure parted-2.1/configure
--- parted-2.1.orig/configure	2009-12-20 05:51:57.000000000 -0600
+++ parted-2.1/configure	2009-12-22 21:51:44.828799616 -0600
@@ -13217,7 +13217,7 @@
      # native cc issues annoying warnings and then ignores it,
      # which would cause us to incorrectly conclude that it worked.
      for gl_flags in \
-        '-Wl,--as-needed' \
+        '-Wl,--no-as-needed' \
         '-Wl,-z,ignore' \
         '-z ignore'
      do
