diff -Nur parted-2.3.orig//configure parted-2.3/configure
--- parted-2.3.orig//configure	2010-05-28 13:29:12.000000000 -0500
+++ parted-2.3/configure	2010-07-12 03:23:18.314505370 -0500
@@ -14199,7 +14199,7 @@
      # Don't try bare '--as-needed'; nothing likes it and the HP-UX 11.11
      # native cc issues annoying warnings and then ignores it,
      # which would cause us to incorrectly conclude that it worked.
-     for gl_flags in  '-Wl,--as-needed' \
+     for gl_flags in  '-Wl,--no-as-needed' \
   '-Wl,-z,ignore' \
   '-z ignore'
 
