--- ./login-utils/chfn.c.orig	2020-05-20 07:27:43.465013472 -0500
+++ ./login-utils/chfn.c	2020-06-27 14:11:54.733895362 -0500
@@ -236,12 +236,13 @@
 	if (!def_val)
 		def_val = "";
 	while (true) {
-		printf("%s [%s]: ", question, def_val);
+		printf("%s [%s]:", question, def_val);
 		__fpurge(stdin);
 #ifdef HAVE_LIBREADLINE
 		rl_bind_key('\t', rl_insert);
-		if ((buf = readline(NULL)) == NULL)
+		if ((buf = readline(" ")) == NULL)
 #else
+		printf(" ");
 		if (getline(&buf, &dummy, stdin) < 0)
 #endif
 			errx(EXIT_FAILURE, _("Aborted."));
