--- ./Makefile.orig	2024-05-11 16:51:27.919158388 -0500
+++ ./Makefile	2024-05-11 16:53:10.235160676 -0500
@@ -336,7 +336,7 @@
 # real effect on the reliability of the program, but may concern some
 # people who don't understand them.
 #CFLAGS=-DSYSV3 -O2 -Wall -pipe
-CFLAGS=-DSYSV3 -O2 -pipe
+CFLAGS=-DSYSV3 -O2 -pipe -Wno-error=implicit-function-declaration -Wno-error=implicit-int
 LIB=-lm -lncurses
 
 # All of the source files
