--- ./util.c.orig	2005-07-07 03:16:23.000000000 -0500
+++ ./util.c	2007-03-24 18:38:03.000000000 -0500
@@ -21,6 +21,8 @@
 
 #include "make.h"
 
+#define HAVE_STRERROR 1
+
 #if !defined(MAKE_NATIVE) && !defined(HAVE_STRERROR)
 extern int errno, sys_nerr;
 extern char *sys_errlist[];
@@ -38,6 +40,8 @@
 }
 #endif
 
+#define HAVE_STRDUP 1
+
 #if !defined(MAKE_NATIVE) && !defined(HAVE_STRDUP)
 #include <string.h>
 
