--- ./mtools.conf.orig	2022-09-24 03:43:03.000000000 -0500
+++ ./mtools.conf	2025-02-21 13:17:49.307305213 -0600
@@ -16,11 +16,11 @@
 
 # Example mtools.conf files.  Uncomment the lines which correspond to
 # your architecture and comment out the "SAMPLE FILE" line below
-SAMPLE FILE
+#SAMPLE FILE
 
 # # Linux floppy drives
-# drive a: file="/dev/fd0" exclusive
-# drive b: file="/dev/fd1" exclusive
+drive a: file="/dev/fd0" exclusive
+drive b: file="/dev/fd1" exclusive
 
 # # First SCSI hard disk partition
 # drive c: file="/dev/sda1"
