diff -Nur dhcpcd-5.1.0.orig/dhcpcd-hooks/50-ntp.conf dhcpcd-5.1.0/dhcpcd-hooks/50-ntp.conf
--- dhcpcd-5.1.0.orig/dhcpcd-hooks/50-ntp.conf	2009-09-02 15:21:11.000000000 -0500
+++ dhcpcd-5.1.0/dhcpcd-hooks/50-ntp.conf	2009-09-12 00:36:04.447859980 -0500
@@ -8,14 +8,8 @@
 #   NTP_CONF=/usr/pkg/etc/ntpd.conf
 # to use openntpd from pkgsrc instead of the system provided ntp.
 
-# Detect OpenRC or BSD rc
-# Distributions may want to just have their command here instead of this
-if type rc-service >/dev/null 2>&1 && rc-service --exists ntpd; then
-	ntpd_restart_cmd="rc-service ntpd -- -Ds restart"
-elif [ -x /etc/rc.d/ntpd ]; then
+if [ -x /etc/rc.d/ntpd ]; then
 	ntpd_restart_cmd="/etc/rc.d/ntpd status >/dev/null 2>&1 && /etc/rc.d/ntpd restart"
-elif [ -x /usr/local/etc/rc.d/ntpd ]; then
-	ntpd_restart_cmd="/usr/local/etc/rc.d/ntpd status >/dev/null 2>&1 && /usr/local/etc/rc.d/ntpd restart"
 fi
 
 ntp_conf_dir="$state_dir/ntp.conf"
