--- dhcpcd-2.0.8.orig/src/logger.c	2008-11-07 00:10:23.000000000 +0000
+++ dhcpcd-2.0.8/src/logger.c	2008-11-07 00:13:46.000000000 +0000
@@ -46,21 +46,25 @@
 {
   va_list p;
   va_list p2;
+  va_list p3;
   FILE *f = stderr;
 
   va_start(p, fmt);
   va_copy(p2, p);
+  va_copy(p3, p);
   /* The default to send debug to stderr is just WAY too much console noise, IMHO */
   if(!Daemonized && strcmp(syslog_level_msg[level],"Info")) {
         f = stdout;
         /* fprintf(f, "%s, ", syslog_level_msg[level]); */
 	vfprintf(f, fmt, p);
 	fputc('\n', f);
-	vsyslog(level, fmt, p);
+	vsyslog(level, fmt, p2);
   }
   if ( level <= LogLevel )
-	vsyslog(level, fmt, p2);
+	vsyslog(level, fmt, p3);
   va_end(p);
+  va_end(p2);
+  va_end(p3);
 }
 
 // vim: set ts=4 :
