diff -Naur obexftp-0.23.orig/swig/ruby/ruby_wrap.c obexftp-0.23/swig/ruby/ruby_wrap.c
--- obexftp-0.23.orig/swig/ruby/ruby_wrap.c	2009-02-17 12:30:40.000000000 -0600
+++ obexftp-0.23/swig/ruby/ruby_wrap.c	2010-04-19 12:55:10.445810800 -0500
@@ -2580,7 +2580,11 @@
     
     /* VALUE str = StringValue(argv[0]); // perhaps better? */
     arg2 = STR2CSTR(argv[0]);
+#ifdef RSTRING_LEN
+    arg3 = (int) RSTRING_LEN(argv[0]);
+#else
     arg3 = (int) RSTRING(argv[0])->len;
+#endif
     
     
     
