From: Josselin Mouette <joss@debian.org>
Date: Sat, 23 Nov 2013 20:09:38 +0100
Subject: Add missing MgOpen fonts to 40- and 65-nonlatin.conf

---
 conf.d/40-nonlatin.conf | 8 ++++++++
 conf.d/65-nonlatin.conf | 2 ++
 2 files changed, 10 insertions(+)

diff --git a/conf.d/40-nonlatin.conf b/conf.d/40-nonlatin.conf
index f8d96ce..c99534b 100644
--- a/conf.d/40-nonlatin.conf
+++ b/conf.d/40-nonlatin.conf
@@ -157,10 +157,18 @@
 		<family>Baekmuk Dotum</family>
 		<default><family>sans-serif</family></default>
 	</alias>
+	<alias>
+		<family>MgOpen Moderna</family>
+		<default><family>sans-serif</family></default>
+	</alias>
 	<alias>
 		<family>MgOpen Modata</family>
 		<default><family>sans-serif</family></default>
 	</alias>
+	<alias>
+		<family>MgOpen Cosmetica</family>
+		<default><family>sans-serif</family></default>
+	</alias>
 	<alias>
 		<family>Sazanami Gothic</family>
 		<default><family>sans-serif</family></default>
diff --git a/conf.d/65-nonlatin.conf b/conf.d/65-nonlatin.conf
index 71049f1..ef6c297 100644
--- a/conf.d/65-nonlatin.conf
+++ b/conf.d/65-nonlatin.conf
@@ -110,7 +110,9 @@
 			<family>PingFang HK</family> <!-- han (zh-hk) - macOS -->
 			<family>Hiragino Sans CNS</family> <!-- han (zh-cn) - macOS -->
 			<family>Hiragino Sans GB</family> <!-- han (zh-tw) - macOS -->
+			<family>MgOpen Moderna</family>
 			<family>MgOpen Modata</family>
+			<family>MgOpen Cosmetica</family>
 			<family>VL Gothic</family>
 			<family>IPAMonaGothic</family>
 			<family>IPAGothic</family>
