--- ./CMakeLists.txt.orig	2015-04-07 08:20:28.000000000 -0500
+++ ./CMakeLists.txt	2015-10-24 18:12:52.316604417 -0500
@@ -49,6 +49,7 @@
                     ${CMAKE_CURRENT_SOURCE_DIR}/libs
                     ${CMAKE_CURRENT_SOURCE_DIR}/libs/editor/
                     ${NETWORKMANAGER_INCLUDE_DIRS}
+                    ${NM-UTIL_INCLUDE_DIRS}
                     ${NETWORKMANAGERQT_INCLUDE_DIRS}
                     ${KDE4_INCLUDES})
 
