--- ./src/Profile.cpp.orig	2014-10-31 23:17:02.000000000 -0500
+++ ./src/Profile.cpp	2018-04-01 13:08:59.412975556 -0500
@@ -150,7 +150,7 @@
     setProperty(Command, qgetenv("SHELL"));
     setProperty(Arguments, QStringList() << qgetenv("SHELL"));
     setProperty(Icon, "utilities-terminal");
-    setProperty(Environment, QStringList() << "TERM=xterm");
+    setProperty(Environment, QStringList() << "TERM=konsole");
     setProperty(LocalTabTitleFormat, "%d : %n");
     setProperty(RemoteTabTitleFormat, "(%u) %H");
     setProperty(ShowTerminalSizeHint, true);
--- ./src/Pty.cpp.orig	2014-10-31 23:17:02.000000000 -0500
+++ ./src/Pty.cpp	2018-04-01 13:08:52.282975632 -0500
@@ -218,7 +218,7 @@
 
     // extra safeguard to make sure $TERM is always set
     if (!isTermEnvAdded) {
-        setEnv("TERM", "xterm");
+        setEnv("TERM", "konsole");
     }
 }
 
