diff --git a/psshlib/askpass_server.py b/psshlib/askpass_server.py
index 4293164..4680749 100644
--- a/psshlib/askpass_server.py
+++ b/psshlib/askpass_server.py
@@ -69,7 +69,7 @@ class PasswordServer(object):
         buffer = self.buffermap[fd]
         conn = self.socketmap[fd]
         try:
-            bytes_written = conn.send(buffer)
+            bytes_written = conn.send(buffer.encode('utf-8'))
         except socket.error:
             _, e, _ = sys.exc_info()
             number = e.args[0]
