--- isapnptools-1.26.orig/doc/man2html
+++ isapnptools-1.26/doc/man2html
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #
 # Brute force script to convert isapnptools man pages to html.
 #
--- isapnptools-1.26.orig/debian/changelog
+++ isapnptools-1.26/debian/changelog
@@ -0,0 +1,263 @@
+isapnptools (1.26-5) unstable; urgency=low
+
+  * src/isapnp_main.l: Don't declare variable serial_identifier as static
+    (thanks Andreas Jochens).  Closes: #286477.
+
+ -- Steve M. Robbins <smr@debian.org>  Thu, 23 Dec 2004 00:04:41 -0500
+
+isapnptools (1.26-4) unstable; urgency=low
+
+  * Add amd64 to list of supported arches.  Closes: #257455.
+
+ -- Steve M. Robbins <smr@debian.org>  Wed,  7 Jul 2004 21:21:29 -0400
+
+isapnptools (1.26-3) unstable; urgency=low
+
+  * Applied portability fixes from David Weinehall.  Closes: #254921.
+    - debian/isapnptools.init: Replace "test EXPR1 -o EXPR2" by 
+      "test EXPR1 || test EXPR2".
+    - doc/man2html: Change hash-bang line from /bin/bash to /bin/sh.
+
+ -- Steve M. Robbins <smr@debian.org>  Thu, 24 Jun 2004 20:25:08 -0400
+
+isapnptools (1.26-2) unstable; urgency=low
+
+  * debian/control:
+    Standards-Version updated to 3.5.6.0; no changes required.
+    Build-depend on flex-old rather than flex.
+    Build-depend on linux-kernel-headers, for <linux/pci.h>.
+    Set -dev package section to libdevel.
+  
+  * debian/rules:
+    Suppress accidental execution of auto-* tools.
+    Honour DEB_BUILD_OPTIONS policy.
+  
+  * src/callbacks.c: Include stdlib.h, for declaration of exit().
+  
+  * src/resource.c: Define __KERNEL__ and include <linux/mod_devicetable.h>
+    to switch on definitions of __u32 et al.  Closes: #223166.
+
+ -- Steve M. Robbins <smr@debian.org>  Sat,  6 Dec 2003 22:32:25 -0500
+
+isapnptools (1.26-1) unstable; urgency=low
+
+  * New upstream release.
+  * Revert to explicit list of architectures; does not
+    build on "mipsel".  Closes: #97942.
+  * Do not install /etc/isapnp.conf or /etc/isapnp.gone;
+    supply them as examples, instead.
+  * Init script will exit if the kernel has an isa-pnp.o module.
+  * Install doc/README.lib in -dev package; don't install README
+    files for DOS, OS2, or borland.
+
+ -- Steve M. Robbins <smr@debian.org>  Fri, 18 May 2001 20:10:02 -0400
+
+isapnptools (1.25-1) unstable; urgency=low
+
+  * New upstream release.
+  * Change to architecture "any" (Closes: #94091).
+
+ -- Steve M. Robbins <smr@debian.org>  Mon, 16 Apr 2001 19:19:03 -0400
+
+isapnptools (1.24-3) unstable; urgency=low
+
+  * Fix getopt_long() call to avoid coredump (Closes: #93097).
+
+ -- Steve M. Robbins <smr@debian.org>  Sat,  7 Apr 2001 12:40:52 -0400
+
+isapnptools (1.24-2) unstable; urgency=low
+
+  * In version 1.24-1, all the include files were moved to libisapnp-dev, 
+    so it must conflict with older isapnptools (closes: #90373).
+  * Set section/priority to base/extra, in agreement with the archive
+    override file.
+
+ -- Steve M. Robbins <smr@debian.org>  Mon, 19 Mar 2001 23:12:31 -0500
+
+isapnptools (1.24-1) unstable; urgency=low
+
+  * New Maintainer (closes: #89719).
+  * New upstream version.
+    - scanner is case insensitive (closes: #74455, #75362, #83251).
+  * Repackaged using debhelper.
+    - correctly generates /usr/doc symlink (closes: #52641)
+  * Past NMU fixes that remain fixed:
+    - ensure isapnp is in /sbin (closes: #74893, #74642, #74349, #74487)
+    - dev package installs lib and headers (closes: #64469)
+    - README.lib was missing (closes: #62343, #53259)
+    - no longer uses awk (closes: #29633, #29862)
+    - updated version long ago (closes: #27784)
+    - upstream changelog correctly named (closes: #9664)
+  * Patch previously incorporated without closing bug:
+    - exit init script if test -e /proc/isapnp (closes: #53057)
+  * Closing lots of obsolete bugs (closes: #46674).
+
+ -- Steve M. Robbins <smr@debian.org>  Sun, 18 Mar 2001 20:11:10 -0500
+
+isapnptools (1.23-0.4) unstable; urgency=low
+
+  * Non-Maintainer upload.
+  * Apply patch-1.23a from
+    http://www.roestock.demon.co.uk/isapnptools/patch-1.23a.gz
+  * Created acconfig.h (build fix).
+
+ -- Ivo Timmermans <ivo@debian.org>  Fri, 23 Feb 2001 19:45:39 +0100
+
+isapnptools (1.23-0.3) unstable; urgency=low
+
+  * Non-Maintainer Upload.
+  * Fixed location of isapnp binary - should have tested this with 2.2
+    kernel, not just 2.4..  (Closes: #74642, #74349, #74363, #74642)
+
+ -- Joseph Carter <knghtbrd@debian.org>  Sun, 15 Oct 2000 13:20:19 -0500
+
+isapnptools (1.23-0.2) unstable; urgency=low
+
+  * Non-Maintainer Upload
+  * Fixed drain bamage in isapnp/release.h regarding config.h
+
+ -- Joseph Carter <knghtbrd@debian.org>  Wed,  4 Oct 2000 14:36:03 -0500
+
+isapnptools (1.23-0.1) unstable; urgency=low
+
+  * Non-Maintainer Upload.
+  * New upstream version
+  * Modified debian/* for autosomething
+
+ -- Joseph Carter <knghtbrd@debian.org>  Mon, 25 Sep 2000 01:04:14 -0500
+
+isapnptools (1.21-2.2) unstable; urgency=low
+
+  * NMU again.
+  * Put libisapnp-dev in devel, not libs.
+
+ -- Chris Lawrence <lawrencc@debian.org>  Mon, 24 Jul 2000 16:38:05 -0500
+
+isapnptools (1.21-2.1) unstable; urgency=low
+
+  * NMU.
+  * Create a separate libisapnp-dev package.  (Closes: #64469)
+  * Update Standards-Version in control file.
+  * Properly include section and priority fields.
+  * README.lib md5sums problem fixed.  (Closes: #53259, #62343)
+
+ -- Chris Lawrence <lawrencc@debian.org>  Sat,  1 Jul 2000 14:13:45 -0500
+
+isapnptools (1.21-2) unstable; urgency=low
+
+  * applied patch from Cesar Eduardo Barros to correct the behaviour
+  for IRQ 2. Closes: #50202.
+  * changed the test to not start for 2.3 kernels. Closes: #52818.
+
+ -- Frederic Lepied <Lepied@debian.org>  Thu, 16 Dec 1999 05:56:59 +0100
+
+isapnptools (1.21-1) unstable; urgency=low
+
+  * new upstream version.
+
+ -- Frederic Lepied <Lepied@debian.org>  Sat, 11 Dec 1999 15:52:40 +0100
+
+isapnptools (1.19-1) unstable; urgency=low
+
+  * new upstream version.
+
+ -- Frederic Lepied <Lepied@debian.org>  Thu, 21 Oct 1999 05:51:00 +0200
+
+isapnptools (1.18-1) unstable; urgency=low
+
+  * new upstream version.
+  * Don't start isapnp on startup for 2.3 kernels. Closes: Bug#43634.
+
+ -- Frederic Lepied <Lepied@debian.org>  Sat,  4 Sep 1999 07:06:42 +0200
+
+isapnptools (1.16-4) frozen unstable; urgency=low
+
+  * downgrade pciutils dependency from Recommend to Suggest. Closes Bug#29004.
+
+ -- Frederic Lepied <Lepied@debian.org>  Mon,  9 Nov 1998 06:01:16 +0100
+
+isapnptools (1.16-3) frozen unstable; urgency=low
+
+  * oops upload to frozen too.
+
+ -- Frederic Lepied <Lepied@debian.org>  Wed,  4 Nov 1998 04:06:53 +0100
+
+isapnptools (1.16-2) unstable; urgency=low
+
+  * added missing dependency fields. Closes Bug#28136
+  * (resource.c): use type instead of which to test for
+  the lspci command. Closes Bug#28728.
+
+ -- Frederic Lepied <Lepied@debian.org>  Tue, 27 Oct 1998 09:32:57 +0100
+
+isapnptools (1.16-1) unstable; urgency=low
+
+  * new upstream version.
+
+ -- Frederic Lepied <Lepied@debian.org>  Fri, 16 Oct 1998 08:53:06 +0200
+
+isapnptools (1.13-3.1) unstable; urgency=high
+
+  * Non-maintainer upload
+  * Fix #23770 (startup link in rcS.d)
+  * Add depend on dpkg >= 1.4.0.21 because of update-rc.d
+
+ -- Gergely Madarasz <gorgo@caesar.elte.hu>  Wed, 24 Jun 1998 00:11:59 +0200
+
+isapnptools (1.13-3) unstable; urgency=low
+
+  * made architecture i386 and alpha. Fix Bug#19704.
+
+ -- Frederic Lepied <Lepied@debian.org>  Sun, 15 Mar 1998 20:53:28 +0100
+
+isapnptools (1.13-2) unstable; urgency=low
+
+  * Fix Bug#19334 (reports from lintian).
+
+ -- Frederic Lepied <Lepied@debian.org>  Sun, 15 Mar 1998 14:44:16 +0100
+
+isapnptools (1.13-1) unstable; urgency=low, Closes=15423
+
+  * upstream release.
+  * remade architecture 'any' as it works with the alpha architecture now.
+  * install changelog.Debian compressed. Closes Bug#15423.
+
+ -- Frederic Lepied <Lepied@debian.org>  Wed,  7 Jan 1998 06:22:42 +0100
+
+isapnptools (1.11-3) unstable; urgency=low
+
+  * made architecture i386 (bug #12532).
+
+ -- Frederic Lepied <Lepied@debian.org>  Wed,  8 Oct 1997 11:22:26 +0200
+
+isapnptools (1.11-2) unstable; urgency=low
+
+  * test if isapnp.conf is readable before launching isapnp (bug #13688)
+
+ -- Frederic Lepied <Lepied@debian.org>  Wed,  8 Oct 1997 09:40:09 +0200
+
+isapnptools (1.11-1) unstable; urgency=low
+
+  * new upstream version.
+
+ -- Frederic Lepied <Lepied@debian.org>  Sat, 26 Jul 1997 11:14:14 +0200
+
+isapnptools (1.10-2) unstable; urgency=low
+
+  * compiled with libc6.
+
+ -- Frederic Lepied <Lepied@debian.org>  Sat, 12 Jul 1997 19:48:15 +0200
+
+isapnptools (1.10-1) unstable; urgency=low
+
+  * New upstream release.
+
+ -- Frederic Lepied <Lepied@debian.org>  Thu, 12 Jun 1997 08:47:14 +0200
+
+isapnptools (1.9-1) unstable; urgency=low
+
+  * Initial Release.
+  * Need to be integrated with sysvinit.
+
+ -- Frederic Lepied <Lepied@debian.org>  Thu, 13 Feb 1997 10:28:34 +0100
+
--- isapnptools-1.26.orig/debian/control
+++ isapnptools-1.26/debian/control
@@ -0,0 +1,26 @@
+Source: isapnptools
+Section: base
+Priority: extra
+Maintainer: Steve M. Robbins <smr@debian.org>
+Standards-Version: 3.5.8.0
+Build-Depends: debhelper (>> 3.0.0), flex-old, linux-kernel-headers
+
+Package: isapnptools
+Architecture: alpha amd64 arm i386
+Depends: ${shlibs:Depends}, dpkg (>= 1.4.0.21)
+Description: ISA Plug-And-Play configuration utilities.
+ This program is suitable for all systems, whether or not they
+ include a PnP BIOS. In fact, a PnP BIOS adds some complications
+ because it may already activate some cards so that the drivers
+ can find them, and these tools can unconfigure them, or change
+ their settings causing all sorts of nasty effects.
+
+Package: libisapnp-dev
+Architecture: alpha amd64 arm i386
+Conflicts: isapnptools (<< 1.24-1)
+Section: libdevel
+Priority: optional
+Description: ISA Plug-And-Play development libraries.
+ This package provides a header file and a static library for
+ development of software that detects and configures ISA Plug-And-Play
+ devices.
--- isapnptools-1.26.orig/debian/copyright
+++ isapnptools-1.26/debian/copyright
@@ -0,0 +1,17 @@
+This package was debianized by Frederic Lepied <Lepied@debian.org> on
+Thu, 13 Feb 1997 09:10:53 +0100.
+
+It was downloaded from http://www.roestock.demon.co.uk/isapnptools/index.html
+
+Copyright:
+
+		    GNU GENERAL PUBLIC LICENSE
+		       Version 2, June 1991
+
+ Copyright (C) 1989, 1991 Free Software Foundation, Inc.,
+   59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ Everyone is permitted to copy and distribute verbatim copies
+ of this license document, but changing it is not allowed.
+
+On Debian Linux systems, the complete text of the GNU General
+Public License can be found in `/usr/share/common-licenses/GPL'.
--- isapnptools-1.26.orig/debian/isapnptools.README.Debian
+++ isapnptools-1.26/debian/isapnptools.README.Debian
@@ -0,0 +1,38 @@
+Starting with version 1.26-1 (2001-05-18), the sample configuration
+files are no longer installed in /etc.  Look for them in
+/usr/share/doc/isapnptools/examples.  
+
+The example files are not very useful, in fact.  Start with 
+"pnpdump >isapnp.conf" and then read "man isapnp.conf" to get started.
+
+
+The --reset option to pnpdump may be useful to reset some hardware at
+shutdown time.  However, the pnpdump manpage suggests this may cause a
+lockup, so this is not done by default.  Edit /etc/init.d/isapnp if
+you wish to do so.  See bug report #54686.
+
+
+		Kernel ISA PnP Support
+		----------------------
+
+The 2.4 linux kernels have built-in support for ISA PnP configuration,
+obviating the need for isapnptools.  The init script will silently
+exit if the running kernel has isa-pnp support built-in or compiled as
+a module.
+
+For many, the kernel PnP configuration will be sufficient (it seems to
+be better than isapnptools).  If your card does not get properly
+configured, you will have to peruse the file Documentation/isapnp.txt
+from the kernel sources and come up with the required commands
+to configure it.  Put them into a file, e.g. /etc/proc-isapnp.conf,
+and arrange for this to be read during boot.  If isa-pnp is compiled
+as a module, you can use a module post-install rule such as
+the following:
+
+  post-install isa-pnp cat /etc/proc-isapnp.conf > /proc/isapnp
+
+Install this line into a file in /etc/modutils of your choosing
+(e.g. /etc/modutils/local) then run "update-modules".
+
+
+ -- Steve M. Robbins <smr@debian.org>, Sat Dec  6 23:35:46 2003
--- isapnptools-1.26.orig/debian/isapnptools.docs
+++ isapnptools-1.26/debian/isapnptools.docs
@@ -0,0 +1,8 @@
+AUTHORS
+NEWS
+README
+doc/README.ide
+doc/README.initrd
+doc/README.modules
+doc/isapnpfaq.txt
+
--- isapnptools-1.26.orig/debian/isapnptools.examples
+++ isapnptools-1.26/debian/isapnptools.examples
@@ -0,0 +1,2 @@
+etc/isapnp.gone
+etc/isapnp.conf
--- isapnptools-1.26.orig/debian/isapnptools.files
+++ isapnptools-1.26/debian/isapnptools.files
@@ -0,0 +1,4 @@
+sbin
+usr/sbin
+usr/share/man
+
--- isapnptools-1.26.orig/debian/isapnptools.init
+++ isapnptools-1.26/debian/isapnptools.init
@@ -0,0 +1,29 @@
+#! /bin/sh
+
+# /etc/init.d/isapnp: configure Plug and Play boards
+
+test -x /sbin/isapnp || exit 0
+
+# kernel 2.3.x and later can handle Pnp directly
+if test -e /proc/isapnp || \
+   test -e /lib/modules/`/bin/uname -r`/kernel/drivers/pnp/isa-pnp.o
+then
+    exit 0
+fi
+
+case "$1" in
+    start)
+	if [ -r /etc/isapnp.conf ]; then
+	    /sbin/isapnp /etc/isapnp.conf
+	fi
+	;;
+	
+    stop|force-reload|restart)
+	;;
+
+    *)
+	echo "Usage: $0 {start|stop|restart|force-reload}"
+	exit 1
+esac
+
+exit 0
--- isapnptools-1.26.orig/debian/libisapnp-dev.docs
+++ isapnptools-1.26/debian/libisapnp-dev.docs
@@ -0,0 +1,4 @@
+AUTHORS
+NEWS
+README
+doc/README.lib
--- isapnptools-1.26.orig/debian/libisapnp-dev.examples
+++ isapnptools-1.26/debian/libisapnp-dev.examples
@@ -0,0 +1,2 @@
+demos/demo.c
+demos/demo2.c
--- isapnptools-1.26.orig/debian/libisapnp-dev.files
+++ isapnptools-1.26/debian/libisapnp-dev.files
@@ -0,0 +1,2 @@
+usr/include
+usr/lib
--- isapnptools-1.26.orig/debian/postrm
+++ isapnptools-1.26/debian/postrm
@@ -0,0 +1,7 @@
+#!/bin/sh -e
+
+#DEBHELPER#
+
+if [ "$1" = "purge" ] ; then
+	rm -f /etc/isapnp.conf /etc/isapnp.gone
+fi
--- isapnptools-1.26.orig/debian/rules
+++ isapnptools-1.26/debian/rules
@@ -0,0 +1,115 @@
+#!/usr/bin/make -f
+# 
+# Adapted from /usr/share/doc/debhelper/examples/rules.multi2
+
+# Uncomment this to turn on verbose mode. 
+#export DH_VERBOSE=1
+
+# This is the debhelper compatibility version to use.
+export DH_COMPAT=3
+
+# This has to be exported to make some magic below work.
+export DH_OPTIONS
+
+# Suppress accidental execution of the auto-* tools; see
+# http://lists.debian.org/debian-devel/2001/debian-devel-200111/msg01416.html
+no_auto_tools = ACLOCAL="`pwd`/missing aclocal" \
+        AUTOCONF="`pwd`/missing autoconf" \
+        AUTOMAKE="`pwd`/missing automake" \
+        AUTOHEADER="`pwd`/missing autoheader"
+
+# Customization follows Debian policy 11.1.
+#
+# Variable DEB_BUILD_OPTIONS can have "noopt" or "nostrip"
+
+CFLAGS = -Wall -g
+export CFLAGS
+
+ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
+CFLAGS += -O0
+else
+CFLAGS += -O2
+endif
+
+ifeq (,$(findstring nostrip,$(DEB_BUILD_OPTIONS)))
+INSTALL_PROGRAM += -s
+endif
+
+build: build-stamp
+build-stamp:
+	dh_testdir
+
+	rm -f config.cache
+	./configure --prefix=/usr \
+		--mandir=/usr/share/man --infodir=/usr/share/info
+	$(MAKE) $(no_auto_tools)
+
+	touch build-stamp
+
+clean:
+	dh_testdir
+	dh_testroot
+	rm -f build-stamp
+
+	-cd demos && $(MAKE) distclean
+	-$(MAKE) distclean
+
+	dh_clean
+
+install: DH_OPTIONS=
+install: build
+	dh_testdir
+	dh_testroot
+	dh_clean -k
+	dh_installdirs
+
+	$(MAKE) DESTDIR=`pwd`/debian/tmp install
+	# Move isapnp to /sbin
+	mkdir debian/tmp/sbin
+	mv debian/tmp/usr/sbin/isapnp debian/tmp/sbin
+
+	dh_movefiles
+
+# This single target is used to build all the packages, all at once, or
+# one at a time. So keep in mind: any options passed to commands here will
+# affect _all_ packages. Anything you want to only affect one package
+# should be put in another target, such as the install target.
+binary-common:
+	dh_testdir
+	dh_testroot
+#	dh_installdebconf
+	dh_installdocs
+	dh_installexamples
+	# override init script name, for historical reasons
+	dh_installinit --no-restart-on-upgrade \
+		--init-script=isapnp \
+		-- start 15 S .
+#	dh_installman
+	dh_installinfo
+	dh_installchangelogs ChangeLog
+#	dh_link
+	dh_strip
+	dh_compress
+	dh_fixperms
+#	dh_makeshlibs
+	dh_installdeb
+	dh_shlibdeps
+	dh_gencontrol
+	dh_md5sums
+	dh_builddeb
+
+# Build architecture independant packages using the common target.
+binary-indep: build install
+# (Uncomment this next line if you have such packages.)
+#	 $(MAKE) -f debian/rules DH_OPTIONS=-i binary-common
+
+# Build architecture dependant packages using the common target.
+binary-arch: build install
+	$(MAKE) -f debian/rules DH_OPTIONS=-a binary-common
+
+# Any other binary targets build just one binary package at a time.
+binary-%: build install
+	make -f debian/rules binary-common DH_OPTIONS=-p$*
+
+binary: binary-indep binary-arch
+.PHONY: build clean binary-indep binary-arch binary install
--- isapnptools-1.26.orig/src/callbacks.c
+++ isapnptools-1.26/src/callbacks.c
@@ -51,6 +51,7 @@
 #include <stdio.h>
 #include <errno.h>
 #include <string.h>
+#include <stdlib.h>
 
 /****************************************************************************/
 
--- isapnptools-1.26.orig/src/isapnp_main.l
+++ isapnptools-1.26/src/isapnp_main.l
@@ -363,7 +363,7 @@
 	return 0;
 }
 
-static unsigned char serial_identifier[NUM_CARDS+1][IDENT_LEN];
+unsigned char serial_identifier[NUM_CARDS+1][IDENT_LEN];
 static char *boardid[NUM_CARDS+1];
 static unsigned long serno[NUM_CARDS+1];
 
--- isapnptools-1.26.orig/src/resource.c
+++ isapnptools-1.26/src/resource.c
@@ -24,6 +24,9 @@
 #include <string.h>
 
 #ifdef ENABLE_PCI
+#define __KERNEL__
+#include <linux/mod_devicetable.h>
+#undef __KERNEL__
 #include <linux/pci.h>
 #endif
 
