diff -ru calligra-2.9.11.orig/gemini/desktopviewproxy.cpp calligra-2.9.11/gemini/desktopviewproxy.cpp
--- calligra-2.9.11.orig/gemini/desktopviewproxy.cpp	2016-02-02 20:53:13.000000000 +0100
+++ calligra-2.9.11/gemini/desktopviewproxy.cpp	2017-08-27 17:54:48.900376022 +0200
@@ -31,7 +31,7 @@
 #include <krecentfilesaction.h>
 #include <kactioncollection.h>
 
-#include <boost/config/posix_features.hpp>
+#include <boost/config/detail/posix_features.hpp>
 #include <KConfigGroup>
 
 #include <KoMainWindow.h>
diff -ru calligra-2.9.11.orig/krita/gemini/desktopviewproxy.cpp calligra-2.9.11/krita/gemini/desktopviewproxy.cpp
--- calligra-2.9.11.orig/krita/gemini/desktopviewproxy.cpp	2016-02-02 20:53:14.000000000 +0100
+++ calligra-2.9.11/krita/gemini/desktopviewproxy.cpp	2017-08-27 17:53:54.964372389 +0200
@@ -31,7 +31,7 @@
 #include <krecentfilesaction.h>
 #include <kactioncollection.h>
 
-#include <boost/config/posix_features.hpp>
+#include <boost/config/detail/posix_features.hpp>
 
 #include <KisMainWindow.h>
 #include <KisImportExportManager.h>
