--- ./kexi/migration/mysql/mysqlmigrate.cpp.orig	2016-02-02 13:53:13.000000000 -0600
+++ ./kexi/migration/mysql/mysqlmigrate.cpp	2017-08-30 14:21:01.436709371 -0500
@@ -42,7 +42,7 @@
 #ifdef Q_WS_WIN
 # undef _WIN32_WINNT // avoid redef.
 #endif
-#include <mysql_version.h>
+#include <mariadb_version.h>
 #include <mysql.h>
 #define BOOL bool
 
--- ./kexi/kexidb/drivers/mysql/mysqldriver.cpp.orig	2016-02-02 13:53:13.000000000 -0600
+++ ./kexi/kexidb/drivers/mysql/mysqldriver.cpp	2017-08-30 14:20:24.858710150 -0500
@@ -31,7 +31,7 @@
 #include <QVariant>
 #include <QFile>
 
-#include <mysql_version.h>
+#include <mariadb_version.h>
 #include <mysql.h>
 #define BOOL bool
 
--- ./kexi/kexidb/drivers/mysql/mysqlconnection_p.h.orig	2016-02-02 13:53:13.000000000 -0600
+++ ./kexi/kexidb/drivers/mysql/mysqlconnection_p.h	2017-08-30 14:20:24.861710150 -0500
@@ -30,7 +30,7 @@
 # endif
 # include <my_global.h>
 #endif
-#include <mysql_version.h>
+#include <mariadb_version.h>
 #include <mysql.h>
 
 typedef struct st_mysql MYSQL;
