--- /usr/sbin/grubconfig	2006-03-24 09:28:41.000000000 +0100
+++ _developpement/debug/grubconfig	2007-05-12 16:51:57.000000000 +0200
@@ -53,6 +53,10 @@
 rm -rf $TMP $BOOT_TMP
 mkdir -p $TMP $BOOT_TMP
 
+# We need to ensure the output is in english in order to match the devices
+# with fdisk. 
+export LC_ALL=C
+
 PATH=$PATH:/bin:/usr/bin:/sbin:/usr/sbin
 
 # Menu to check if we want to use VESA framebuffer support:
@@ -475,7 +479,8 @@
  chmod 644 $BOOT/grub/$grub_config
  message=yes
  installcolor;
- umount $BOOT_TMP 2>$TMP/null && rmdir $BOOT_TMP
+ umount $BOOT_TMP 2> $TMP/null
+ rmdir $BOOT_TMP 2> $TMP/null
  rm -rf $TMP
 }
 
@@ -1004,8 +1009,6 @@
         KILL=NO
      fi
    simplegrub
-   umount $BOOT_TMP 2>$TMP/null
-   rm -rf $TMP $BOOT_TMP
    chroot_umount
    exit
  fi
