--- ./etc/a2ps_cfg.in.orig	2007-12-28 19:38:58.000000000 -0600
+++ ./etc/a2ps_cfg.in	2009-03-30 20:11:06.000000000 -0500
@@ -68,7 +68,7 @@
 # 2)	Path to the a2ps resource				#
 #################################################################
 # Default path where a2ps dropped its library files.
-LibraryPath: @libpath@
+LibraryPath: @libpath@:/usr/share/fonts/Type1
 
 # It may be useful to extend it so that a2ps can see some
 # TeX or X11 resources: it likes AFM files and PF[AB] files.
