--- ./configure.orig	Thu Jan 18 21:14:51 2001
+++ ./configure	Thu Jan 18 21:15:16 2001
@@ -8449,7 +8449,8 @@
 # Find out if ghostscript is installed
 ac_psfont_path=
 for ac_dir in /usr/local/ghostscript/fonts \
-  /usr/local/share/ghostscript/fonts;
+  /usr/local/share/ghostscript/fonts \
+  /usr/share/ghostscript/fonts ;
 do
   if test "cd $ac_dir 2>/dev/null && echo *.afm"; then
     ac_psfont_path="$ac_psfont_path:$ac_dir";
