--- ./src/Makefile.in.orig	2006-02-15 21:09:50.000000000 -0600
+++ ./src/Makefile.in	2006-06-25 23:10:19.000000000 -0500
@@ -52,10 +52,10 @@
 #---------------------------------------------------------------------------
 # 1. Linux GPM Mouse support
 #    Uncomment next FOUR lines for GPM mouse support
-#MOUSEFLAGS = -DUSE_GPM_MOUSE
-#MOUSELIB = -lgpm
-#GPMMOUSEO = gpmmouse.o
-#OBJGPMMOUSEO = $(OBJDIR)/$(GPMMOUSEO)
+MOUSEFLAGS = -DUSE_GPM_MOUSE
+MOUSELIB = -lgpm
+GPMMOUSEO = gpmmouse.o
+OBJGPMMOUSEO = $(OBJDIR)/$(GPMMOUSEO)
 
  
 # 2. XFree86 XRENDERFONT (Anti-aliased font) support for XJED
