--- ./fdi/11-x11-synaptics.fdi.orig	2009-10-11 18:11:21.000000000 -0500
+++ ./fdi/11-x11-synaptics.fdi	2010-04-24 12:59:37.000000000 -0500
@@ -3,6 +3,8 @@
   <device>
     <match key="info.capabilities" contains="input.touchpad">
         <merge key="input.x11_driver" type="string">synaptics</merge>
+        <!-- Enable tap click by default: -->
+        <merge key="input.x11_options.TapButton1" type="string">1</merge>
         <!-- Arbitrary options can be passed to the driver using
              the input.x11_options property since xorg-server-1.5. -->
         <!-- EXAMPLES:
