--- ./Makefile.orig	2006-06-15 19:21:52.000000000 -0500
+++ ./Makefile	2006-06-25 00:28:24.000000000 -0500
@@ -39,7 +39,7 @@
 
 CC = $(CROSS_COMPILE)gcc
 CXFLAGS = -ggdb
-CWFLAGS = -Wall -Werror -Wstrict-prototypes
+CWFLAGS = -Wall -Werror -Wstrict-prototypes -Os
 
 ifdef DEBIAN
 CPPFLAGS= -DDEBIAN
@@ -55,7 +55,7 @@
 CFLAGS = $(CWFLAGS) $(CXFLAGS) -DSendmail=\""$(MAILCMD)"\" $(CONFFILEFLAGS)
 
 # If you want a static binary, you might uncomment these
-# LDFLAGS = -static
+LDFLAGS = -static
 # STRIP = -s
 
 INSTALL = /usr/bin/install
