--- ./mkspecs/linux-g++/qmake.conf.orig	Fri Apr 26 00:48:00 2002
+++ ./mkspecs/linux-g++/qmake.conf	Tue May  7 12:05:04 2002
@@ -18,7 +18,7 @@
 QMAKE_CFLAGS_DEPS	= -M
 QMAKE_CFLAGS_WARN_ON	= -Wall -W
 QMAKE_CFLAGS_WARN_OFF	= -w
-QMAKE_CFLAGS_RELEASE	= -O2
+QMAKE_CFLAGS_RELEASE	= -O2 -march=i486 -mtune=i686
 QMAKE_CFLAGS_DEBUG	= -g
 QMAKE_CFLAGS_SHLIB	= -fPIC
 QMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
