--- ./66-wqy-zenhei-sharp-no13px.conf.orig	2009-02-16 12:45:58.000000000 -0600
+++ ./66-wqy-zenhei-sharp-no13px.conf	2018-03-07 17:22:30.603206663 -0600
@@ -6,8 +6,6 @@
   <match target="font">
         <test qual="any" name="family">
                 <string>WenQuanYi Zen Hei</string>
-                <string>文泉驿正黑</string>
-                <string>文泉驛正黑</string>
         </test>
         <test compare="more_eq" name="pixelsize"><double>13.5</double></test>
         <test compare="less" name="pixelsize"><double>14.5</double></test>
--- ./66-wqy-zenhei-sharp.conf.orig	2009-02-12 23:27:50.000000000 -0600
+++ ./66-wqy-zenhei-sharp.conf	2018-03-07 17:22:43.643207743 -0600
@@ -5,8 +5,6 @@
    <match target="font">
         <test qual="any" name="family">
                 <string>WenQuanYi Zen Hei</string>
-                <string>文泉驿正黑</string>
-                <string>文泉驛正黑</string>
         </test>
         <test compare="more_eq" name="pixelsize"><double>12</double></test>
         <test compare="less_eq" name="pixelsize"><double>16</double></test>
@@ -19,8 +17,6 @@
   <match target="font">
         <test qual="any" name="family">
                 <string>WenQuanYi Zen Hei</string>
-                <string>文泉驿正黑</string>
-                <string>文泉驛正黑</string>
         </test>
         <test compare="more_eq" name="pixelsize"><double>13.5</double></test>
         <test compare="less" name="pixelsize"><double>14.5</double></test>
