--- ./agent/main.cpp.orig	2010-07-05 10:08:40.000000000 -0500
+++ ./agent/main.cpp	2011-03-30 18:52:38.636002143 -0500
@@ -21,11 +21,14 @@
 #include <KCmdLineArgs>
 #include <KAboutData>
 #include <KLocale>
+#include <cstdlib>
 
 #include "policykitkde.h"
 
 int main(int argc, char *argv[])
 {
+    setenv("KDE_DEBUG", "true", 1);
+
     KAboutData aboutData("policykit1-kde", "polkit-kde-authentication-agent-1", ki18n("PolicyKit1-KDE"), "0.96.1",
                          ki18n("PolicyKit1-KDE"), KAboutData::License_GPL,
                          ki18n("(c) 2009 Red Hat, Inc."));
