From 576d713ec955faafdc9d7b783f3f4c27b7bc1793 Mon Sep 17 00:00:00 2001
From: Christos Zoulas <christos@zoulas.com>
Date: Tue, 6 Apr 2021 15:43:11 +0000
Subject: [PATCH] Add more ash paths from Cormac Stephenson

---
 magic/Magdir/commands | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/magic/Magdir/commands b/magic/Magdir/commands
index 10f8d2b20..5f6be0cce 100644
--- a/magic/Magdir/commands
+++ b/magic/Magdir/commands
@@ -1,6 +1,6 @@
 
 #------------------------------------------------------------------------------
-# $File: commands,v 1.63 2020/06/06 15:36:30 christos Exp $
+# $File: commands,v 1.64 2021/04/06 15:43:11 christos Exp $
 # commands:  file(1) magic for various shells and interpreters
 #
 #0	string/w	:			shell archive or script for antique kernel text
@@ -38,6 +38,10 @@
 0	search/1	#!/usr/bin/env\ zsh	Paul Falstad's zsh script text executable
 !:mime	text/x-shellscript
 
+0	string/wt	#!\ /bin/ash		Neil Brown's ash script text executable
+!:mime	text/x-shellscript
+0	string/wt	#!\ /usr/bin/ash	Neil Brown's ash script text executable
+!:mime	text/x-shellscript
 0	string/wt	#!\ /usr/local/bin/ash	Neil Brown's ash script text executable
 !:mime	text/x-shellscript
 0	string/wt	#!\ /usr/local/bin/ae	Neil Brown's ae script text executable
