From 500f3d402a94c6950946515d76ebd72872a6d0e2 Mon Sep 17 00:00:00 2001
From: Jim Pryor <profjim@jimpryor.net>
Date: Sun, 1 May 2011 18:56:17 -0400
Subject: [PATCH 2/9] README: /var/spool/cron/crontabs/root

Signed-off-by: Jim Pryor <profjim@jimpryor.net>
---
 README | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/README b/README
index de02958..e33b1aa 100644
--- a/README
+++ b/README
@@ -115,7 +115,7 @@ are normally in /var/spool/cron/cronstamps. These directories normally have perm
 
 Here is the superuser's crontab, created using `sudo crontab -e`:
 
-	-rw-------  0 root   root      513 Jan  6 18:58 /var/spool/cron/root
+	-rw-------  0 root   root      513 Jan  6 18:58 /var/spool/cron/crontabs/root
 
 TESTING
 -------
-- 
2.13.2

