From 2e3717c48ead5a962e2c29bc818672fff36f73f6 Mon Sep 17 00:00:00 2001
From: Jim Pryor <profjim@jimpryor.net>
Date: Sun, 1 May 2011 19:21:21 -0400
Subject: [PATCH 3/9] CHANGELOG tweak

Signed-off-by: Jim Pryor <profjim@jimpryor.net>
---
 CHANGELOG | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/CHANGELOG b/CHANGELOG
index a1ea4a7..0710c3b 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -19,12 +19,12 @@ git
 
   * Documentation and error message updates.
 
-v4.5  30-Apr-2011
+v4.5  1-May-2011
   * Some cron jobs were running multiple times. Now we make sure not to
     ArmJobs that are already running; and not to resynchronize while jobs are
-    running; and to poll the DST setting. (Fixes Arch FS#18681; thanks to Paul
-    Gideon Dann for identifying the second issue; and Tilman Sauerbeck for
-    identifying the third.)
+    running; and to poll the DST setting. (Fixes Arch FS#18681; thanks to Vincent
+    Cappe and Paul Gideon Dann for identifying the second issue; and Tilman
+    Sauerbeck for identifying the third.)
 
   * @monthly was wrongly being parsed the same as @yearly (fixes Arch
     FS#19123). Thanks to Peter Johnson, Paul Gideon Dann, and Tilman Sauerbeck.
-- 
2.13.2

