--- ./scripts/mysql_install_db.sh.orig	2019-08-01 13:25:00.406477885 -0500
+++ ./scripts/mysql_install_db.sh	2019-08-01 13:27:53.316490799 -0500
@@ -480,25 +480,26 @@
 
 if test -n "$user"
 then
-  chown $user "$pamtooldir/auth_pam_tool_dir" && \
-  chmod 0700 "$pamtooldir/auth_pam_tool_dir"
-  if test $? -ne 0
-  then
-      echo "Cannot change ownership of the '$pamtooldir/auth_pam_tool_dir' directory"
-      echo " to the '$user' user. Check that you have the necessary permissions and try again."
-      exit 1
-  fi
-  if test -z "$srcdir"
-  then
-    chown 0 "$pamtooldir/auth_pam_tool_dir/auth_pam_tool" && \
-    chmod 04755 "$pamtooldir/auth_pam_tool_dir/auth_pam_tool"
-    if test $? -ne 0
-    then
-        echo "Couldn't set an owner to '$pamtooldir/auth_pam_tool_dir/auth_pam_tool'."
-        echo " It must be root, the PAM authentication plugin doesn't work otherwise.."
-        echo
-    fi
-  fi
+  # Commented out since Slackware does not currently provide PAM:
+  #chown $user "$pamtooldir/auth_pam_tool_dir" && \
+  #chmod 0700 "$pamtooldir/auth_pam_tool_dir"
+  #if test $? -ne 0
+  #then
+  #    echo "Cannot change ownership of the '$pamtooldir/auth_pam_tool_dir' directory"
+  #    echo " to the '$user' user. Check that you have the necessary permissions and try again."
+  #    exit 1
+  #fi
+  #if test -z "$srcdir"
+  #then
+  #  chown 0 "$pamtooldir/auth_pam_tool_dir/auth_pam_tool" && \
+  #  chmod 04755 "$pamtooldir/auth_pam_tool_dir/auth_pam_tool"
+  #  if test $? -ne 0
+  #  then
+  #      echo "Couldn't set an owner to '$pamtooldir/auth_pam_tool_dir/auth_pam_tool'."
+  #      echo " It must be root, the PAM authentication plugin doesn't work otherwise.."
+  #      echo
+  #  fi
+  #fi
   args="$args --user=$user"
 fi
 
