diff -Nur libwmf-0.2.8.4.orig/configure.ac libwmf-0.2.8.4/configure.ac
--- libwmf-0.2.8.4.orig/configure.ac	2005-07-27 15:48:39.000000000 -0500
+++ libwmf-0.2.8.4/configure.ac	2019-12-28 22:35:15.470031981 -0600
@@ -745,7 +745,7 @@
 if test $LIBWMF_BUILDSTYLE != lite; then
 	PKG_CHECK_MODULES(GDK_PIXBUF,gdk-pixbuf-2.0 >= 2.1.2,[
 		GTK_VERSION=`$PKG_CONFIG --variable=gtk_binary_version gtk+-2.0`
-		GDK_PIXBUF_DIR="gtk-2.0/$GTK_VERSION/loaders"
+		GDK_PIXBUF_DIR="gdk-pixbuf-2.0/$GTK_VERSION/loaders"
 		wmf_gdk_pixbuf=yes
 	],[	wmf_gdk_pixbuf=no
 	])
