--- ./lynx.cfg.orig	Tue Feb 26 23:18:35 2002
+++ ./lynx.cfg	Tue Feb 26 23:24:09 2002
@@ -2841,8 +2841,18 @@
 #COLOR:4:magenta:white
 #COLOR:5:blue:white
 #COLOR:6:red:white
-COLOR:6:brightred:black
+#COLOR:6:brightred:black
 #COLOR:7:magenta:cyan
+COLOR:0:black:white
+COLOR:1:blue:white
+COLOR:2:yellow:blue
+COLOR:3:green:white
+COLOR:4:magenta:white
+COLOR:5:blue:white
+# This makes links a *lot* more visible.  The default settings are like
+# "let's torture the color-blind" :^)
+COLOR:6:brightred:black
+COLOR:7:magenta:cyan
 
 
 .h2 ASSUMED_COLOR
