--- ./lynx.cfg.orig	2021-08-01 13:50:51.000000000 -0500
+++ ./lynx.cfg	2021-09-25 13:27:05.424126198 -0500
@@ -496,7 +496,7 @@
 # ASSUME_CHARSET can also be changed via the 'o'ptions menu but will
 # not be saved as permanent value in user's .lynxrc file to avoid more chaos.
 #
-#ASSUME_CHARSET:iso-8859-1
+ASSUME_CHARSET:utf-8
 
 .h2 ASSUMED_DOC_CHARSET_CHOICE
 .h2 DISPLAY_CHARSET_CHOICE
@@ -3008,8 +3008,18 @@
 #COLOR:4:magenta:white
 #COLOR:5:blue:white
 #COLOR:6:red:white
-COLOR:6:brightred:black
+#COLOR:6:brightred:black
 #COLOR:7:magenta:cyan
+COLOR:0:black:white
+COLOR:1:blue:white
+COLOR:2:yellow:blue
+COLOR:3:green:white
+COLOR:4:magenta:white
+COLOR:5:blue:white
+# This makes links a *lot* more visible.  The default settings are like
+# "let's torture the color-blind" :^)
+COLOR:6:brightred:black
+COLOR:7:magenta:cyan
 
 .h2 COLOR_STYLE
 # Also known as "lss" (lynx style-sheet), the color-style file assigns color
