--- ./add-to-xpdfrc.orig	2004-07-27 17:28:07.000000000 -0500
+++ ./add-to-xpdfrc	2008-03-02 01:45:16.000000000 -0600
@@ -1,8 +1,8 @@
 #----- begin Chinese Traditional support package (2004-jul-27)
-cidToUnicode	Adobe-CNS1	/usr/local/share/xpdf/chinese-traditional/Adobe-CNS1.cidToUnicode
-unicodeMap	Big5		/usr/local/share/xpdf/chinese-traditional/Big5.unicodeMap
-unicodeMap	Big5ascii	/usr/local/share/xpdf/chinese-traditional/Big5ascii.unicodeMap
-cMapDir		Adobe-CNS1	/usr/local/share/xpdf/chinese-traditional/CMap
-toUnicodeDir			/usr/local/share/xpdf/chinese-traditional/CMap
-#displayCIDFontTT	Adobe-CNS1	/usr/..../bkai00mp.ttf
+cidToUnicode	Adobe-CNS1	/usr/share/xpdf/chinese-traditional/Adobe-CNS1.cidToUnicode
+unicodeMap	Big5		/usr/share/xpdf/chinese-traditional/Big5.unicodeMap
+unicodeMap	Big5ascii	/usr/share/xpdf/chinese-traditional/Big5ascii.unicodeMap
+cMapDir		Adobe-CNS1	/usr/share/xpdf/chinese-traditional/CMap
+toUnicodeDir			/usr/share/xpdf/chinese-traditional/CMap
+displayCIDFontTT	Adobe-CNS1	/usr/share/fonts/TTF/wqy-zenhei.ttf
 #----- end Chinese Traditional support package
