--- ./add-to-xpdfrc.orig	2011-09-02 17:24:44.000000000 -0500
+++ ./add-to-xpdfrc	2013-03-28 15:45:42.020412765 -0500
@@ -1,9 +1,9 @@
 #----- begin Chinese Simplified support package (2011-sep-02)
-cidToUnicode	Adobe-GB1	/usr/local/share/xpdf/chinese-simplified/Adobe-GB1.cidToUnicode
-unicodeMap	ISO-2022-CN	/usr/local/share/xpdf/chinese-simplified/ISO-2022-CN.unicodeMap
-unicodeMap	EUC-CN		/usr/local/share/xpdf/chinese-simplified/EUC-CN.unicodeMap
-unicodeMap	GBK		/usr/local/share/xpdf/chinese-simplified/GBK.unicodeMap
-cMapDir		Adobe-GB1	/usr/local/share/xpdf/chinese-simplified/CMap
-toUnicodeDir			/usr/local/share/xpdf/chinese-simplified/CMap
-#fontFileCC	Adobe-GB1	/usr/..../gkai00mp.ttf
+cidToUnicode	Adobe-GB1	/usr/share/xpdf/chinese-simplified/Adobe-GB1.cidToUnicode
+unicodeMap	ISO-2022-CN	/usr/share/xpdf/chinese-simplified/ISO-2022-CN.unicodeMap
+unicodeMap	EUC-CN		/usr/share/xpdf/chinese-simplified/EUC-CN.unicodeMap
+unicodeMap	GBK		/usr/share/xpdf/chinese-simplified/GBK.unicodeMap
+cMapDir		Adobe-GB1	/usr/share/xpdf/chinese-simplified/CMap
+toUnicodeDir			/usr/share/xpdf/chinese-simplified/CMap
+fontFileCC	Adobe-GB1	/usr/share/fonts/TTF/wqy-zenhei.ttc
 #----- end Chinese Simplified support package
