--- ./add-to-xpdfrc.orig	2011-09-02 17:24:44.000000000 -0500
+++ ./add-to-xpdfrc	2013-03-28 15:46:03.285412085 -0500
@@ -1,8 +1,8 @@
 #----- begin Chinese Traditional support package (2011-sep-02)
-cidToUnicode	Adobe-CNS1	/usr/local/share/xpdf/chinese-traditional/Adobe-CNS1.cidToUnicode
-unicodeMap	Big5		/usr/local/share/xpdf/chinese-traditional/Big5.unicodeMap
-unicodeMap	Big5ascii	/usr/local/share/xpdf/chinese-traditional/Big5ascii.unicodeMap
-cMapDir		Adobe-CNS1	/usr/local/share/xpdf/chinese-traditional/CMap
-toUnicodeDir			/usr/local/share/xpdf/chinese-traditional/CMap
-#fontFileCC	Adobe-CNS1	/usr/..../bkai00mp.ttf
+cidToUnicode	Adobe-CNS1	/usr/share/xpdf/chinese-traditional/Adobe-CNS1.cidToUnicode
+unicodeMap	Big5		/usr/share/xpdf/chinese-traditional/Big5.unicodeMap
+unicodeMap	Big5ascii	/usr/share/xpdf/chinese-traditional/Big5ascii.unicodeMap
+cMapDir		Adobe-CNS1	/usr/share/xpdf/chinese-traditional/CMap
+toUnicodeDir			/usr/share/xpdf/chinese-traditional/CMap
+fontFileCC	Adobe-CNS1	/usr/share/fonts/TTF/wqy-zenhei.ttc
 #----- end Chinese Traditional support package
