--- ./add-to-xpdfrc.orig	2008-03-02 01:28:33.000000000 -0600
+++ ./add-to-xpdfrc	2008-03-02 01:28:45.000000000 -0600
@@ -1,4 +1,4 @@
 #----- begin Hebrew support package (2003-feb-16)
-unicodeMap	ISO-8859-8	/usr/local/share/xpdf/hebrew/ISO-8859-8.unicodeMap
-unicodeMap	Windows-1255	/usr/local/share/xpdf/hebrew/Windows-1255.unicodeMap
+unicodeMap	ISO-8859-8	/usr/share/xpdf/hebrew/ISO-8859-8.unicodeMap
+unicodeMap	Windows-1255	/usr/share/xpdf/hebrew/Windows-1255.unicodeMap
 #----- end Hebrew support package
