--- ./add-to-xpdfrc.orig	2019-10-03 13:51:29.775084557 -0500
+++ ./add-to-xpdfrc	2019-10-03 13:59:15.404046027 -0500
@@ -1,8 +1,8 @@
 #----- begin Chinese Traditional support package (2011-sep-02)
-cidToUnicode	Adobe-CNS1	/usr/local/share/xpdf/chinese-traditional/Adobe-CNS1.cidToUnicode
-unicodeMap	Big5		/usr/local/share/xpdf/chinese-traditional/Big5.unicodeMap
-unicodeMap	Big5ascii	/usr/local/share/xpdf/chinese-traditional/Big5ascii.unicodeMap
-cMapDir		Adobe-CNS1	/usr/local/share/xpdf/chinese-traditional/CMap
-toUnicodeDir			/usr/local/share/xpdf/chinese-traditional/CMap
-#fontFileCC	Adobe-CNS1	/usr/..../NotoSansCJKtc-Regular.otf"
+cidToUnicode	Adobe-CNS1	/usr/share/xpdf/chinese-traditional/Adobe-CNS1.cidToUnicode
+unicodeMap	Big5		/usr/share/xpdf/chinese-traditional/Big5.unicodeMap
+unicodeMap	Big5ascii	/usr/share/xpdf/chinese-traditional/Big5ascii.unicodeMap
+cMapDir		Adobe-CNS1	/usr/share/xpdf/chinese-traditional/CMap
+toUnicodeDir			/usr/share/xpdf/chinese-traditional/CMap
+fontFileCC	Adobe-CNS1	/usr/share/fonts/TTF/wqy-zenhei.ttc
 #----- end Chinese Traditional support package
