--- ./add-to-xpdfrc.orig	2021-01-29 12:36:23.966962831 -0600
+++ ./add-to-xpdfrc	2021-01-29 12:38:24.237956178 -0600
@@ -1,10 +1,10 @@
 #----- begin Korean support package (2011-sep-02)
-cidToUnicode	Adobe-Korea1	/usr/local/share/xpdf/korean/Adobe-Korea1.cidToUnicode
-cidToUnicode	Adobe-KR	/usr/local/share/xpdf/korean/Adobe-KR.cidToUnicode
-unicodeMap	ISO-2022-KR	/usr/local/share/xpdf/korean/ISO-2022-KR.unicodeMap
-cMapDir		Adobe-Korea1	/usr/local/share/xpdf/korean/CMap
-cMapDir		Adobe-KR	/usr/local/share/xpdf/korean/CMap
-toUnicodeDir			/usr/local/share/xpdf/korean/CMap
+cidToUnicode	Adobe-Korea1	/usr/share/xpdf/korean/Adobe-Korea1.cidToUnicode
+cidToUnicode	Adobe-KR	/usr/share/xpdf/korean/Adobe-KR.cidToUnicode
+unicodeMap	ISO-2022-KR	/usr/share/xpdf/korean/ISO-2022-KR.unicodeMap
+cMapDir		Adobe-Korea1	/usr/share/xpdf/korean/CMap
+cMapDir		Adobe-KR	/usr/share/xpdf/korean/CMap
+toUnicodeDir			/usr/share/xpdf/korean/CMap
 #fontFileCC	Adobe-Korea1	/usr/..../NotoSansCJKkr-Regular.otf
 #fontFileCC	Adobe-KR	/usr/..../NotoSansCJKkr-Regular.otf
 #----- end Korean support package
