--- ./add-to-xpdfrc.orig	2005-07-07 13:05:05.000000000 -0500
+++ ./add-to-xpdfrc	2008-03-02 01:33:52.000000000 -0600
@@ -1,8 +1,8 @@
 #----- begin Korean support package (2005-jul-07)
-cidToUnicode	Adobe-Korea1	/usr/local/share/xpdf/korean/Adobe-Korea1.cidToUnicode
-unicodeMap	ISO-2022-KR	/usr/local/share/xpdf/korean/ISO-2022-KR.unicodeMap
-cMapDir		Adobe-Korea1	/usr/local/share/xpdf/korean/CMap
-toUnicodeDir			/usr/local/share/xpdf/korean/CMap
+cidToUnicode	Adobe-Korea1	/usr/share/xpdf/korean/Adobe-Korea1.cidToUnicode
+unicodeMap	ISO-2022-KR	/usr/share/xpdf/korean/ISO-2022-KR.unicodeMap
+cMapDir		Adobe-Korea1	/usr/share/xpdf/korean/CMap
+toUnicodeDir			/usr/share/xpdf/korean/CMap
 #displayCIDFontTT	Adobe-Korea1	/usr/..../batang.ttf"
 #displayCIDFontTT	Unidocs-Korea1	/usr/..../batang.ttf"
 #----- end Korean support package
