--- ./add-to-xpdfrc.orig	2004-07-27 17:28:08.000000000 -0500
+++ ./add-to-xpdfrc	2008-03-02 01:46:16.000000000 -0600
@@ -1,9 +1,9 @@
 #----- begin Chinese Simplified support package (2004-jul-27)
-cidToUnicode	Adobe-GB1	/usr/local/share/xpdf/chinese-simplified/Adobe-GB1.cidToUnicode
-unicodeMap	ISO-2022-CN	/usr/local/share/xpdf/chinese-simplified/ISO-2022-CN.unicodeMap
-unicodeMap	EUC-CN		/usr/local/share/xpdf/chinese-simplified/EUC-CN.unicodeMap
-unicodeMap	GBK		/usr/local/share/xpdf/chinese-simplified/GBK.unicodeMap
-cMapDir		Adobe-GB1	/usr/local/share/xpdf/chinese-simplified/CMap
-toUnicodeDir			/usr/local/share/xpdf/chinese-simplified/CMap
-#displayCIDFontTT	Adobe-GB1	/usr/..../gkai00mp.ttf
+cidToUnicode	Adobe-GB1	/usr/share/xpdf/chinese-simplified/Adobe-GB1.cidToUnicode
+unicodeMap	ISO-2022-CN	/usr/share/xpdf/chinese-simplified/ISO-2022-CN.unicodeMap
+unicodeMap	EUC-CN		/usr/share/xpdf/chinese-simplified/EUC-CN.unicodeMap
+unicodeMap	GBK		/usr/share/xpdf/chinese-simplified/GBK.unicodeMap
+cMapDir		Adobe-GB1	/usr/share/xpdf/chinese-simplified/CMap
+toUnicodeDir			/usr/share/xpdf/chinese-simplified/CMap
+displayCIDFontTT	Adobe-GB1	/usr/share/fonts/TTF/wqy-zenhei.ttf
 #----- end Chinese Simplified support package
