--- ./add-to-xpdfrc.orig	2011-08-15 16:24:29.000000000 -0500
+++ ./add-to-xpdfrc	2013-03-28 15:43:21.115417270 -0500
@@ -1,4 +1,4 @@
 #----- begin Cyrillic support package (2011-aug-15)
-nameToUnicode			/usr/local/share/xpdf/cyrillic/Bulgarian.nameToUnicode
-unicodeMap	KOI8-R		/usr/local/share/xpdf/cyrillic/KOI8-R.unicodeMap
+nameToUnicode			/usr/share/xpdf/cyrillic/Bulgarian.nameToUnicode
+unicodeMap	KOI8-R		/usr/share/xpdf/cyrillic/KOI8-R.unicodeMap
 #----- end Cyrillic support package
