--- ./add-to-xpdfrc.orig	2002-11-03 17:49:15.000000000 -0600
+++ ./add-to-xpdfrc	2008-03-02 01:36:36.000000000 -0600
@@ -1,4 +1,4 @@
 #----- begin Thai support package (2002-jan-16)
-nameToUnicode			/usr/local/share/xpdf/thai/Thai.nameToUnicode
-unicodeMap	TIS-620		/usr/local/share/xpdf/thai/TIS-620.unicodeMap
+nameToUnicode			/usr/share/xpdf/thai/Thai.nameToUnicode
+unicodeMap	TIS-620		/usr/share/xpdf/thai/TIS-620.unicodeMap
 #----- end Thai support package
