--- ./add-to-xpdfrc.orig	2011-08-15 16:24:29.000000000 -0500
+++ ./add-to-xpdfrc	2013-03-28 15:43:21.116417270 -0500
@@ -1,4 +1,4 @@
 #----- begin Greek support package (2011-aug-15)
-nameToUnicode			/usr/local/share/xpdf/greek/Greek.nameToUnicode
-unicodeMap	ISO-8859-7	/usr/local/share/xpdf/greek/ISO-8859-7.unicodeMap
+nameToUnicode			/usr/share/xpdf/greek/Greek.nameToUnicode
+unicodeMap	ISO-8859-7	/usr/share/xpdf/greek/ISO-8859-7.unicodeMap
 #----- end Greek support package
