--- ./data/meson.build.orig	2020-11-20 15:02:11.854337200 -0600
+++ ./data/meson.build	2022-02-16 13:40:12.980091649 -0600
@@ -2,7 +2,6 @@
 desktop_conf.set('VERSION', glade_version)
 
 i18n.merge_file(
-  'desktop',
   type: 'desktop',
   input: configure_file(
     input: 'org.gnome.Glade.desktop.in.in',
@@ -18,7 +17,6 @@
 appdata = glade_name + '.appdata.xml'
 
 i18n.merge_file(
-  'appdata',
   input: 'org.gnome.Glade.appdata.xml.in',
   output: '@BASENAME@',
   po_dir: po_dir,
