diff -Nru /tmp/6rrvEn4EMT/eject-2.1.4/eject.c /tmp/Jon9ZzjSLN/eject-2.1.4/eject.c
--- /tmp/6rrvEn4EMT/eject-2.1.4/eject.c	2007-03-12 14:15:26.000000000 +0100
+++ /tmp/Jon9ZzjSLN/eject-2.1.4/eject.c	2007-03-12 14:15:28.000000000 +0100
@@ -552,7 +552,7 @@
 	gettimeofday(&time_start, NULL);
 	
 	/* Send the CDROMEJECT command to the device. */
-	if (ioctl(fd, CDROMEJECT, 0) < 0) {
+	if (ioctl(fd, CDROMEJECT, 0) < 0 && errno != EIO) {
 		perror("ioctl");
 		exit(1);
 	}
