diff -Naur zsh-4.3.10.old/Completion/Linux/Command/_pkgtool zsh-4.3.10/Completion/Linux/Command/_pkgtool
--- zsh-4.3.10.old/Completion/Linux/Command/_pkgtool	2005-06-24 10:15:27.000000000 -0400
+++ zsh-4.3.10/Completion/Linux/Command/_pkgtool	2009-07-07 20:39:01.000000000 -0400
@@ -12,7 +12,7 @@
       "-ask[always ask on package installation regardless of package's priority]" \
       '-priority[provide a priority for the entire package list to use]:priority list:(ADD REC OPT SKP)' \
       '-tagfile[specify a different file to use for package priorities]:tagfile:_files -/' \
-      '*:package file:_files -g "*.tgz(-.)"'
+      '*:package file:_files -g "*.t[gblx]z(-.)"'
   ;;
 
   removepkg)
@@ -29,11 +29,11 @@
       '--install-new[install new packages in addition to upgrading existing ones]' \
       '--reinstall[upgrade even when the same version is already installed]' \
       '--verbose[show all the gory details of the upgrade]' \
-      '*:package file:_files -g "*.tgz(-.)"'
+      '*:package file:_files -g "*.t[gblx]z(-.)"'
   ;;
 
   explodepkg)
-    _wanted file expl 'package file' _files -g '*.tgz(-.)'
+    _wanted file expl 'package file' _files -g '*.t[gblx]z(-.)'
   ;;
 
   makepkg)
