From b55be8ded0c7e1119eb062a5831b246b618c89e8 Mon Sep 17 00:00:00 2001
From: Mike Massonnet <mmassonnet@gmail.com>
Date: Thu, 8 Apr 2010 22:38:50 +0200
Subject: [PATCH] use right comment character (! and not #) in Xft.xrdb file (bug 6314)

 scripts/Xft.xrdb |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/scripts/Xft.xrdb b/scripts/Xft.xrdb
index af45361..cd26513 100644
--- a/scripts/Xft.xrdb
+++ b/scripts/Xft.xrdb
@@ -1,7 +1,7 @@
-# Those are fallback settings, use the ui plugin to change it
-# or add your overrides to ~/.Xresources
-# Xft.hintstyle: hintnone/hintslight/hintmedium/hintfull
-# Xft hinting: 1/0
+! Those are fallback settings, use the ui plugin to change it
+! or add your overrides to ~/.Xresources
+! Xft.hintstyle: hintnone/hintslight/hintmedium/hintfull
+! Xft hinting: 1/0
 
 Xft.hinting: 1
 Xft.hintstyle: hintmedium
