commit 80819d447df5661ee6c9e9cd22f501cee612da8b
Author: Lionel Le Folgoc <mrpouit@gmail.com>
Date:   Mon Jul 5 19:57:53 2010 +0200

    Fix find and replace (bug #5831).

diff --git a/src/search.c b/src/search.c
index 07e29c7..4fb1c5b 100644
--- a/src/search.c
+++ b/src/search.c
@@ -210,8 +210,11 @@ gint run_dialog_find(StructData *sd)
 		G_CALLBACK(toggle_sensitivity), NULL);
 	 g_signal_connect(G_OBJECT(entry_find), "delete-text",
 		G_CALLBACK(toggle_sensitivity), NULL);
-	 if (sd->search.string_find) 
+	 if (sd->search.string_find) {
 		 gtk_entry_set_text(GTK_ENTRY(entry_find), sd->search.string_find);
+		 gtk_dialog_set_response_sensitive(GTK_DIALOG(dialog),
+			GTK_RESPONSE_OK, TRUE);
+	 }
 
 	gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_OK);
 	gtk_entry_set_activates_default(GTK_ENTRY(entry_find), TRUE);
@@ -275,8 +278,11 @@ gint run_dialog_replace(StructData *sd)
 		G_CALLBACK(toggle_sensitivity), NULL);
 	 g_signal_connect(G_OBJECT(entry_find), "delete-text",
 		G_CALLBACK(toggle_sensitivity), NULL);
-	 if (sd->search.string_find) 
+	 if (sd->search.string_find) {
 		 gtk_entry_set_text(GTK_ENTRY(entry_find), sd->search.string_find);
+		 gtk_dialog_set_response_sensitive(GTK_DIALOG(dialog),
+			GTK_RESPONSE_OK, TRUE);
+	}
 	label_replace = gtk_label_new_with_mnemonic(_("Re_place with: "));
 	 gtk_misc_set_alignment(GTK_MISC(label_replace), 0, 0.5);
 	 gtk_table_attach_defaults(GTK_TABLE(table), label_replace, 0, 1, 1, 2);
