diff -Nur Terminal-0.4.0.orig/Makefile.in Terminal-0.4.0/Makefile.in
--- Terminal-0.4.0.orig/Makefile.in	2009-07-20 13:04:04.000000000 -0500
+++ Terminal-0.4.0/Makefile.in	2009-07-20 13:50:20.529864039 -0500
@@ -986,7 +986,7 @@
 
 TerminalHelp: TerminalHelp.in Makefile
 	rm -f TerminalHelp.gen TerminalHelp
-	sed -e "s,\@datadir\@,$(datadir),g" 				\
+	sed -e "s,\@docdir\@,$(docdir),g" 				\
 		< $(srcdir)/TerminalHelp.in				\
 		> TerminalHelp.gen
 	mv TerminalHelp.gen TerminalHelp
diff -Nur Terminal-0.4.0.orig/TerminalHelp.in Terminal-0.4.0/TerminalHelp.in
--- Terminal-0.4.0.orig/TerminalHelp.in	2009-07-20 13:03:50.000000000 -0500
+++ Terminal-0.4.0/TerminalHelp.in	2009-07-20 13:50:20.529864039 -0500
@@ -19,7 +19,7 @@
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 #
 
-HELPDIR="@datadir@/doc/Terminal/"
+HELPDIR="@docdir@/"
 
 if test -n "$LC_ALL"; then
   LC=$LC_ALL
diff -Nur Terminal-0.4.0.orig/doc/C/Makefile.in Terminal-0.4.0/doc/C/Makefile.in
--- Terminal-0.4.0.orig/doc/C/Makefile.in	2009-07-20 13:04:02.000000000 -0500
+++ Terminal-0.4.0/doc/C/Makefile.in	2009-07-20 13:50:20.533867320 -0500
@@ -260,7 +260,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Terminal/C
+TARGET_DIR = $(docdir)/C
 STYLESHEET = ../terminal.xsl
 DOCUMENT = Terminal.xml
 
diff -Nur Terminal-0.4.0.orig/doc/C/images/Makefile.in Terminal-0.4.0/doc/C/images/Makefile.in
--- Terminal-0.4.0.orig/doc/C/images/Makefile.in	2009-07-20 13:04:02.000000000 -0500
+++ Terminal-0.4.0/doc/C/images/Makefile.in	2009-07-20 13:50:20.533867320 -0500
@@ -240,7 +240,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-imagesdir = $(datadir)/doc/Terminal/C/images
+imagesdir = $(docdir)/C/images
 images_DATA = \
 	terminal-compose-shortcut.png					\
 	terminal-edit-preferences.png					\
diff -Nur Terminal-0.4.0.orig/doc/Makefile.in Terminal-0.4.0/doc/Makefile.in
--- Terminal-0.4.0.orig/doc/Makefile.in	2009-07-20 13:04:02.000000000 -0500
+++ Terminal-0.4.0/doc/Makefile.in	2009-07-20 13:51:00.229192942 -0500
@@ -286,7 +286,7 @@
 	fr								\
 	ja
 
-cssdir = $(datadir)/doc/Terminal
+cssdir = $(docdir)/
 css_DATA = \
 	terminal.css
 
diff -Nur Terminal-0.4.0.orig/doc/da/Makefile.in Terminal-0.4.0/doc/da/Makefile.in
--- Terminal-0.4.0.orig/doc/da/Makefile.in	2009-07-20 13:04:02.000000000 -0500
+++ Terminal-0.4.0/doc/da/Makefile.in	2009-07-20 13:50:20.533867320 -0500
@@ -260,7 +260,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Terminal/da
+TARGET_DIR = $(docdir)/da
 STYLESHEET = ../terminal.xsl
 DOCUMENT = Terminal.xml
 
diff -Nur Terminal-0.4.0.orig/doc/da/images/Makefile.in Terminal-0.4.0/doc/da/images/Makefile.in
--- Terminal-0.4.0.orig/doc/da/images/Makefile.in	2009-07-20 13:04:02.000000000 -0500
+++ Terminal-0.4.0/doc/da/images/Makefile.in	2009-07-20 13:50:20.533867320 -0500
@@ -240,7 +240,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-imagesdir = $(datadir)/doc/Terminal/da/images
+imagesdir = $(docdir)/da/images
 images_DATA = \
 	terminal-compose-shortcut.png					\
 	terminal-edit-preferences.png					\
diff -Nur Terminal-0.4.0.orig/doc/fr/Makefile.in Terminal-0.4.0/doc/fr/Makefile.in
--- Terminal-0.4.0.orig/doc/fr/Makefile.in	2009-07-20 13:04:02.000000000 -0500
+++ Terminal-0.4.0/doc/fr/Makefile.in	2009-07-20 13:50:20.533867320 -0500
@@ -260,7 +260,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Terminal/fr
+TARGET_DIR = $(docdir)/fr
 STYLESHEET = ../terminal.xsl
 DOCUMENT = Terminal.xml
 
diff -Nur Terminal-0.4.0.orig/doc/fr/images/Makefile.in Terminal-0.4.0/doc/fr/images/Makefile.in
--- Terminal-0.4.0.orig/doc/fr/images/Makefile.in	2009-07-20 13:04:03.000000000 -0500
+++ Terminal-0.4.0/doc/fr/images/Makefile.in	2009-07-20 13:50:20.533867320 -0500
@@ -240,7 +240,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-imagesdir = $(datadir)/doc/Terminal/fr/images
+imagesdir = $(docdir)/fr/images
 images_DATA = \
 	terminal-compose-shortcut.png				\
 	terminal-edit-preferences.png					\
diff -Nur Terminal-0.4.0.orig/doc/ja/Makefile.in Terminal-0.4.0/doc/ja/Makefile.in
--- Terminal-0.4.0.orig/doc/ja/Makefile.in	2009-07-20 13:04:03.000000000 -0500
+++ Terminal-0.4.0/doc/ja/Makefile.in	2009-07-20 13:50:20.537889876 -0500
@@ -260,7 +260,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Terminal/ja
+TARGET_DIR = $(docdir)/ja
 STYLESHEET = ../terminal.xsl
 DOCUMENT = Terminal.xml
 
diff -Nur Terminal-0.4.0.orig/doc/ja/images/Makefile.in Terminal-0.4.0/doc/ja/images/Makefile.in
--- Terminal-0.4.0.orig/doc/ja/images/Makefile.in	2009-07-20 13:04:03.000000000 -0500
+++ Terminal-0.4.0/doc/ja/images/Makefile.in	2009-07-20 13:50:20.537889876 -0500
@@ -240,7 +240,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-imagesdir = $(datadir)/doc/Terminal/ja/images
+imagesdir = $(docdir)/ja/images
 images_DATA = \
 	terminal-compose-shortcut.png					\
 	terminal-edit-preferences.png					\
