--- thunar-vfs/thunar-vfs-io-local.c.old        2008-06-13 12:43:52.000000000 +0200
+++ thunar-vfs/thunar-vfs-io-local.c    2008-06-13 14:41:46.000000000 +0200
@@ -301,7 +301,7 @@
     }

   /* check whether we have a hidden file here */
-  if ((s - (const guchar *) name) > 1 && (*name == '.' || *(s - 1) == '~'))
+  if ((s - (const guchar *) name) > 1 && (*name == '.' ))
     info->flags = THUNAR_VFS_FILE_FLAGS_HIDDEN;
   else
     info->flags = THUNAR_VFS_FILE_FLAGS_NONE;
@@ -1029,7 +1029,7 @@
       info->display_name = g_strdup (name);

       /* check if this is a hidden file now */
-      if (strlen (name) > 1 && (name[0] == '.' || name[strlen (name) - 1] == '~'))
+      if (strlen (name) > 1 && (name[0] == '.'))
         info->flags |= THUNAR_VFS_FILE_FLAGS_HIDDEN;
       else
         info->flags &= ~THUNAR_VFS_FILE_FLAGS_HIDDEN;
