diff -Nur Thunar-1.0.1.orig/Makefile.in Thunar-1.0.1/Makefile.in
--- Thunar-1.0.1.orig/Makefile.in	2009-04-18 09:41:09.000000000 -0500
+++ Thunar-1.0.1/Makefile.in	2009-07-20 19:36:55.895440657 -0500
@@ -932,7 +932,7 @@
 
 ThunarHelp: ThunarHelp.in Makefile
 	rm -f ThunarHelp.gen ThunarHelp
-	sed -e "s,\@datadir\@,$(datadir),g" 				\
+	sed -e "s,\@htmldir\@,$(htmldir),g" 				\
 		< $(srcdir)/ThunarHelp.in				\
 		> ThunarHelp.gen
 	mv ThunarHelp.gen ThunarHelp
diff -Nur Thunar-1.0.1.orig/ThunarHelp.in Thunar-1.0.1/ThunarHelp.in
--- Thunar-1.0.1.orig/ThunarHelp.in	2008-10-22 01:43:23.000000000 -0500
+++ Thunar-1.0.1/ThunarHelp.in	2009-07-20 19:36:38.235721255 -0500
@@ -19,7 +19,7 @@
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 #
 
-HELPDIR="@datadir@/doc/Thunar/html/"
+HELPDIR="@htmldir@/"
 
 if test -n "$LC_ALL"; then
   LC=$LC_ALL
diff -Nur Thunar-1.0.1.orig/docs/Makefile.in Thunar-1.0.1/docs/Makefile.in
--- Thunar-1.0.1.orig/docs/Makefile.in	2009-04-18 09:40:14.000000000 -0500
+++ Thunar-1.0.1/docs/Makefile.in	2009-07-20 19:36:38.235721255 -0500
@@ -292,7 +292,7 @@
 builddir = @builddir@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(datadir)/doc/Thunar
+docdir = @docdir@
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
diff -Nur Thunar-1.0.1.orig/docs/manual/C/Makefile.in Thunar-1.0.1/docs/manual/C/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/C/Makefile.in	2009-04-18 09:40:19.000000000 -0500
+++ Thunar-1.0.1/docs/manual/C/Makefile.in	2009-07-20 19:36:38.235721255 -0500
@@ -319,7 +319,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/C
+TARGET_DIR = $(htmldir)/C
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.1.orig/docs/manual/C/images/Makefile.in Thunar-1.0.1/docs/manual/C/images/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/C/images/Makefile.in	2009-04-18 09:40:21.000000000 -0500
+++ Thunar-1.0.1/docs/manual/C/images/Makefile.in	2009-07-20 19:36:38.235721255 -0500
@@ -313,7 +313,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/C/images
+imagesdir = $(htmldir)/C/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.1.orig/docs/manual/Makefile.in Thunar-1.0.1/docs/manual/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/Makefile.in	2009-04-18 09:40:22.000000000 -0500
+++ Thunar-1.0.1/docs/manual/Makefile.in	2009-07-20 19:36:38.235721255 -0500
@@ -338,7 +338,7 @@
 	tr								\
 	zh_TW
 
-cssdir = $(datadir)/doc/Thunar/html
+cssdir = $(htmldir)
 css_DATA = \
 	thunar.css
 
diff -Nur Thunar-1.0.1.orig/docs/manual/da/Makefile.in Thunar-1.0.1/docs/manual/da/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/da/Makefile.in	2009-04-18 09:40:22.000000000 -0500
+++ Thunar-1.0.1/docs/manual/da/Makefile.in	2009-07-20 19:36:38.239694094 -0500
@@ -317,7 +317,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/da
+TARGET_DIR = $(htmldir)/da
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.1.orig/docs/manual/da/images/Makefile.in Thunar-1.0.1/docs/manual/da/images/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/da/images/Makefile.in	2009-04-18 09:40:24.000000000 -0500
+++ Thunar-1.0.1/docs/manual/da/images/Makefile.in	2009-07-20 19:36:38.239694094 -0500
@@ -311,7 +311,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/da/images
+imagesdir = $(htmldir)/da/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png					\
diff -Nur Thunar-1.0.1.orig/docs/manual/es/Makefile.in Thunar-1.0.1/docs/manual/es/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/es/Makefile.in	2009-04-18 09:40:24.000000000 -0500
+++ Thunar-1.0.1/docs/manual/es/Makefile.in	2009-07-20 19:36:38.239694094 -0500
@@ -319,7 +319,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/es
+TARGET_DIR = $(htmldir)/es
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.1.orig/docs/manual/es/images/Makefile.in Thunar-1.0.1/docs/manual/es/images/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/es/images/Makefile.in	2009-04-18 09:40:25.000000000 -0500
+++ Thunar-1.0.1/docs/manual/es/images/Makefile.in	2009-07-20 19:36:38.239694094 -0500
@@ -313,7 +313,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/es/images
+imagesdir = $(htmldir)/es/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.1.orig/docs/manual/eu/Makefile.in Thunar-1.0.1/docs/manual/eu/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/eu/Makefile.in	2009-04-18 09:40:25.000000000 -0500
+++ Thunar-1.0.1/docs/manual/eu/Makefile.in	2009-07-20 19:36:38.239694094 -0500
@@ -319,7 +319,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/eu
+TARGET_DIR = $(htmldir)/eu
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.1.orig/docs/manual/eu/images/Makefile.in Thunar-1.0.1/docs/manual/eu/images/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/eu/images/Makefile.in	2009-04-18 09:40:26.000000000 -0500
+++ Thunar-1.0.1/docs/manual/eu/images/Makefile.in	2009-07-20 19:36:38.239694094 -0500
@@ -313,7 +313,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/eu/images
+imagesdir = $(htmldir)/eu/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.1.orig/docs/manual/fr/Makefile.in Thunar-1.0.1/docs/manual/fr/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/fr/Makefile.in	2009-04-18 09:40:26.000000000 -0500
+++ Thunar-1.0.1/docs/manual/fr/Makefile.in	2009-07-20 19:36:38.243692845 -0500
@@ -319,7 +319,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/fr
+TARGET_DIR = $(htmldir)/fr
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.1.orig/docs/manual/fr/images/Makefile.in Thunar-1.0.1/docs/manual/fr/images/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/fr/images/Makefile.in	2009-04-18 09:40:27.000000000 -0500
+++ Thunar-1.0.1/docs/manual/fr/images/Makefile.in	2009-07-20 19:36:38.243692845 -0500
@@ -313,7 +313,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/fr/images
+imagesdir = $(htmldir)/fr/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.1.orig/docs/manual/gl/Makefile.in Thunar-1.0.1/docs/manual/gl/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/gl/Makefile.in	2009-04-18 09:40:27.000000000 -0500
+++ Thunar-1.0.1/docs/manual/gl/Makefile.in	2009-07-20 19:36:38.243692845 -0500
@@ -319,7 +319,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/gl
+TARGET_DIR = $(htmldir)/gl
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.1.orig/docs/manual/gl/images/Makefile.in Thunar-1.0.1/docs/manual/gl/images/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/gl/images/Makefile.in	2009-04-18 09:40:29.000000000 -0500
+++ Thunar-1.0.1/docs/manual/gl/images/Makefile.in	2009-07-20 19:36:38.243692845 -0500
@@ -313,7 +313,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/C/images
+imagesdir = $(htmldir)/C/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.1.orig/docs/manual/it/Makefile.in Thunar-1.0.1/docs/manual/it/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/it/Makefile.in	2009-04-18 09:40:30.000000000 -0500
+++ Thunar-1.0.1/docs/manual/it/Makefile.in	2009-07-20 19:36:38.243692845 -0500
@@ -317,7 +317,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/it
+TARGET_DIR = $(htmldir)/it
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.1.orig/docs/manual/it/images/Makefile.in Thunar-1.0.1/docs/manual/it/images/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/it/images/Makefile.in	2009-04-18 09:40:31.000000000 -0500
+++ Thunar-1.0.1/docs/manual/it/images/Makefile.in	2009-07-20 19:36:38.243692845 -0500
@@ -311,7 +311,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/it/images
+imagesdir = $(htmldir)/it/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.1.orig/docs/manual/ja/Makefile.in Thunar-1.0.1/docs/manual/ja/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/ja/Makefile.in	2009-04-18 09:40:32.000000000 -0500
+++ Thunar-1.0.1/docs/manual/ja/Makefile.in	2009-07-20 19:36:38.247717296 -0500
@@ -319,7 +319,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/ja
+TARGET_DIR = $(htmldir)/ja
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.1.orig/docs/manual/ja/images/Makefile.in Thunar-1.0.1/docs/manual/ja/images/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/ja/images/Makefile.in	2009-04-18 09:40:33.000000000 -0500
+++ Thunar-1.0.1/docs/manual/ja/images/Makefile.in	2009-07-20 19:36:38.247717296 -0500
@@ -313,7 +313,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/ja/images
+imagesdir = $(htmldir)/ja/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.1.orig/docs/manual/nl/Makefile.in Thunar-1.0.1/docs/manual/nl/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/nl/Makefile.in	2009-04-18 09:40:34.000000000 -0500
+++ Thunar-1.0.1/docs/manual/nl/Makefile.in	2009-07-20 19:36:38.247717296 -0500
@@ -319,7 +319,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/nl
+TARGET_DIR = $(htmldir)/nl
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.1.orig/docs/manual/nl/images/Makefile.in Thunar-1.0.1/docs/manual/nl/images/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/nl/images/Makefile.in	2009-04-18 09:40:34.000000000 -0500
+++ Thunar-1.0.1/docs/manual/nl/images/Makefile.in	2009-07-20 19:36:38.247717296 -0500
@@ -313,7 +313,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/nl/images
+imagesdir = $(htmldir)/nl/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.1.orig/docs/manual/pl/Makefile.in Thunar-1.0.1/docs/manual/pl/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/pl/Makefile.in	2009-04-18 09:40:34.000000000 -0500
+++ Thunar-1.0.1/docs/manual/pl/Makefile.in	2009-07-20 19:36:38.247717296 -0500
@@ -319,7 +319,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/pl
+TARGET_DIR = $(htmldir)/pl
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.1.orig/docs/manual/pl/images/Makefile.in Thunar-1.0.1/docs/manual/pl/images/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/pl/images/Makefile.in	2009-04-18 09:40:34.000000000 -0500
+++ Thunar-1.0.1/docs/manual/pl/images/Makefile.in	2009-07-20 19:36:38.247717296 -0500
@@ -313,7 +313,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/pl/images
+imagesdir = $(htmldir)/pl/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.1.orig/docs/manual/ru/Makefile.in Thunar-1.0.1/docs/manual/ru/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/ru/Makefile.in	2009-04-18 09:40:35.000000000 -0500
+++ Thunar-1.0.1/docs/manual/ru/Makefile.in	2009-07-20 19:36:38.251717303 -0500
@@ -319,7 +319,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/ru
+TARGET_DIR = $(htmldir)/ru
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.1.orig/docs/manual/ru/images/Makefile.in Thunar-1.0.1/docs/manual/ru/images/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/ru/images/Makefile.in	2009-04-18 09:40:35.000000000 -0500
+++ Thunar-1.0.1/docs/manual/ru/images/Makefile.in	2009-07-20 19:36:38.251717303 -0500
@@ -313,7 +313,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/ru/images
+imagesdir = $(htmldir)/ru/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.1.orig/docs/manual/tr/Makefile.in Thunar-1.0.1/docs/manual/tr/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/tr/Makefile.in	2009-04-18 09:40:35.000000000 -0500
+++ Thunar-1.0.1/docs/manual/tr/Makefile.in	2009-07-20 19:36:38.251717303 -0500
@@ -317,7 +317,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/tr
+TARGET_DIR = $(htmldir)/tr
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.1.orig/docs/manual/tr/images/Makefile.in Thunar-1.0.1/docs/manual/tr/images/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/tr/images/Makefile.in	2009-04-18 09:40:36.000000000 -0500
+++ Thunar-1.0.1/docs/manual/tr/images/Makefile.in	2009-07-20 19:36:38.251717303 -0500
@@ -311,7 +311,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/tr/images
+imagesdir = $(htmldir)/tr/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.1.orig/docs/manual/zh_TW/Makefile.in Thunar-1.0.1/docs/manual/zh_TW/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/zh_TW/Makefile.in	2009-04-18 09:40:36.000000000 -0500
+++ Thunar-1.0.1/docs/manual/zh_TW/Makefile.in	2009-07-20 19:36:38.251717303 -0500
@@ -319,7 +319,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/zh_TW
+TARGET_DIR = $(htmldir)/zh_TW
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.1.orig/docs/manual/zh_TW/images/Makefile.in Thunar-1.0.1/docs/manual/zh_TW/images/Makefile.in
--- Thunar-1.0.1.orig/docs/manual/zh_TW/images/Makefile.in	2009-04-18 09:40:36.000000000 -0500
+++ Thunar-1.0.1/docs/manual/zh_TW/images/Makefile.in	2009-07-20 19:36:38.251717303 -0500
@@ -313,7 +313,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/zh_TW/images
+imagesdir = $(htmldir)/zh_TW/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
