--- ./thunar-vfs/thunar-vfs-thumb.c.orig	2009-01-12 14:38:49.000000000 -0600
+++ ./thunar-vfs/thunar-vfs-thumb.c	2010-02-12 13:24:33.000000000 -0600
@@ -1169,7 +1169,7 @@
     goto done0;
 
   /* verify the png signature */
-  if (G_LIKELY (png_check_sig ((png_bytep) signature, sizeof (signature))))
+  if (G_LIKELY (png_sig_cmp ((png_bytep) signature, 0, sizeof (signature)) == 0))
     rewind (fp);
   else
     goto done0;
