diff -Nur Thunar-1.0.2.orig//Makefile.in Thunar-1.0.2/Makefile.in
--- Thunar-1.0.2.orig//Makefile.in	2010-05-21 12:11:02.000000000 -0500
+++ Thunar-1.0.2/Makefile.in	2010-05-23 17:44:59.068552125 -0500
@@ -1040,7 +1040,7 @@
 
 ThunarHelp: ThunarHelp.in Makefile
 	rm -f ThunarHelp.gen ThunarHelp
-	sed -e "s,\@datadir\@,$(datadir),g" 				\
+	sed -e "s,\@htmldir\@,$(htmldir),g" 				\
 		< $(srcdir)/ThunarHelp.in				\
 		> ThunarHelp.gen
 	mv ThunarHelp.gen ThunarHelp
diff -Nur Thunar-1.0.2.orig//ThunarHelp.in Thunar-1.0.2/ThunarHelp.in
--- Thunar-1.0.2.orig//ThunarHelp.in	2010-05-21 12:10:52.000000000 -0500
+++ Thunar-1.0.2/ThunarHelp.in	2010-05-23 17:45:26.207677481 -0500
@@ -19,7 +19,7 @@
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 #
 
-HELPDIR="@datadir@/doc/Thunar/html/"
+HELPDIR="@htmldir@/"
 
 if test -n "$LC_ALL"; then
   LC=$LC_ALL
diff -Nur Thunar-1.0.2.orig//docs/Makefile.in Thunar-1.0.2/docs/Makefile.in
--- Thunar-1.0.2.orig//docs/Makefile.in	2010-05-21 12:10:57.000000000 -0500
+++ Thunar-1.0.2/docs/Makefile.in	2010-05-23 17:46:08.776567272 -0500
@@ -322,7 +322,7 @@
 builddir = @builddir@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(datadir)/doc/Thunar
+docdir = @docdir@
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
diff -Nur Thunar-1.0.2.orig//docs/manual/C/Makefile.in Thunar-1.0.2/docs/manual/C/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/C/Makefile.in	2010-05-21 12:10:57.000000000 -0500
+++ Thunar-1.0.2/docs/manual/C/Makefile.in	2010-05-23 17:46:52.539542089 -0500
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/C
+TARGET_DIR = $(htmldir)/C
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.2.orig//docs/manual/C/images/Makefile.in Thunar-1.0.2/docs/manual/C/images/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/C/images/Makefile.in	2010-05-21 12:10:57.000000000 -0500
+++ Thunar-1.0.2/docs/manual/C/images/Makefile.in	2010-05-23 17:47:25.359542363 -0500
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/C/images
+imagesdir = $(htmldir)/C/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.2.orig//docs/manual/Makefile.in Thunar-1.0.2/docs/manual/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/Makefile.in	2010-05-21 12:10:57.000000000 -0500
+++ Thunar-1.0.2/docs/manual/Makefile.in	2010-05-23 17:48:04.650570173 -0500
@@ -370,7 +370,7 @@
 	tr								\
 	zh_TW
 
-cssdir = $(datadir)/doc/Thunar/html
+cssdir = $(htmldir)
 css_DATA = \
 	thunar.css
 
diff -Nur Thunar-1.0.2.orig//docs/manual/am/Makefile.in Thunar-1.0.2/docs/manual/am/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/am/Makefile.in	2010-05-21 12:10:58.000000000 -0500
+++ Thunar-1.0.2/docs/manual/am/Makefile.in	2010-05-23 17:48:35.304542671 -0500
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/am
+TARGET_DIR = $(htmldir)/am
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.2.orig//docs/manual/am/images/Makefile.in Thunar-1.0.2/docs/manual/am/images/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/am/images/Makefile.in	2010-05-21 12:10:58.000000000 -0500
+++ Thunar-1.0.2/docs/manual/am/images/Makefile.in	2010-05-23 17:51:21.111542305 -0500
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/am/images
+imagesdir = $(htmldir)/am/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.2.orig//docs/manual/da/Makefile.in Thunar-1.0.2/docs/manual/da/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/da/Makefile.in	2010-05-21 12:10:58.000000000 -0500
+++ Thunar-1.0.2/docs/manual/da/Makefile.in	2010-05-23 17:48:52.108550193 -0500
@@ -334,7 +334,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/da
+TARGET_DIR = $(htmldir)/da
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.2.orig//docs/manual/da/images/Makefile.in Thunar-1.0.2/docs/manual/da/images/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/da/images/Makefile.in	2010-05-21 12:10:58.000000000 -0500
+++ Thunar-1.0.2/docs/manual/da/images/Makefile.in	2010-05-23 17:51:27.753542103 -0500
@@ -314,7 +314,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/da/images
+imagesdir = $(htmldir)/da/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png					\
diff -Nur Thunar-1.0.2.orig//docs/manual/es/Makefile.in Thunar-1.0.2/docs/manual/es/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/es/Makefile.in	2010-05-21 12:10:58.000000000 -0500
+++ Thunar-1.0.2/docs/manual/es/Makefile.in	2010-05-23 17:48:58.300552519 -0500
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/es
+TARGET_DIR = $(htmldir)/es
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.2.orig//docs/manual/es/images/Makefile.in Thunar-1.0.2/docs/manual/es/images/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/es/images/Makefile.in	2010-05-21 12:10:58.000000000 -0500
+++ Thunar-1.0.2/docs/manual/es/images/Makefile.in	2010-05-23 17:51:34.373542961 -0500
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/es/images
+imagesdir = $(htmldir)/es/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.2.orig//docs/manual/eu/Makefile.in Thunar-1.0.2/docs/manual/eu/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/eu/Makefile.in	2010-05-21 12:10:58.000000000 -0500
+++ Thunar-1.0.2/docs/manual/eu/Makefile.in	2010-05-23 17:49:05.977677434 -0500
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/eu
+TARGET_DIR = $(htmldir)/eu
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.2.orig//docs/manual/eu/images/Makefile.in Thunar-1.0.2/docs/manual/eu/images/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/eu/images/Makefile.in	2010-05-21 12:10:58.000000000 -0500
+++ Thunar-1.0.2/docs/manual/eu/images/Makefile.in	2010-05-23 17:51:42.643542360 -0500
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/eu/images
+imagesdir = $(htmldir)/eu/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.2.orig//docs/manual/fr/Makefile.in Thunar-1.0.2/docs/manual/fr/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/fr/Makefile.in	2010-05-21 12:10:58.000000000 -0500
+++ Thunar-1.0.2/docs/manual/fr/Makefile.in	2010-05-23 17:49:12.495542046 -0500
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/fr
+TARGET_DIR = $(htmldir)/fr
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.2.orig//docs/manual/fr/images/Makefile.in Thunar-1.0.2/docs/manual/fr/images/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/fr/images/Makefile.in	2010-05-21 12:10:58.000000000 -0500
+++ Thunar-1.0.2/docs/manual/fr/images/Makefile.in	2010-05-23 17:51:51.116702819 -0500
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/fr/images
+imagesdir = $(htmldir)/fr/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.2.orig//docs/manual/gl/Makefile.in Thunar-1.0.2/docs/manual/gl/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/gl/Makefile.in	2010-05-21 12:10:58.000000000 -0500
+++ Thunar-1.0.2/docs/manual/gl/Makefile.in	2010-05-23 17:49:19.882542284 -0500
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/gl
+TARGET_DIR = $(htmldir)/gl
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.2.orig//docs/manual/gl/images/Makefile.in Thunar-1.0.2/docs/manual/gl/images/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/gl/images/Makefile.in	2010-05-21 12:10:58.000000000 -0500
+++ Thunar-1.0.2/docs/manual/gl/images/Makefile.in	2010-05-23 17:51:57.596588720 -0500
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/gl/images
+imagesdir = $(htmldir)/gl/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.2.orig//docs/manual/it/Makefile.in Thunar-1.0.2/docs/manual/it/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/it/Makefile.in	2010-05-21 12:10:58.000000000 -0500
+++ Thunar-1.0.2/docs/manual/it/Makefile.in	2010-05-23 17:49:27.597676807 -0500
@@ -334,7 +334,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/it
+TARGET_DIR = $(htmldir)/it
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.2.orig//docs/manual/it/images/Makefile.in Thunar-1.0.2/docs/manual/it/images/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/it/images/Makefile.in	2010-05-21 12:10:58.000000000 -0500
+++ Thunar-1.0.2/docs/manual/it/images/Makefile.in	2010-05-23 17:52:05.152568701 -0500
@@ -314,7 +314,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/it/images
+imagesdir = $(htmldir)/it/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.2.orig//docs/manual/ja/Makefile.in Thunar-1.0.2/docs/manual/ja/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/ja/Makefile.in	2010-05-21 12:10:59.000000000 -0500
+++ Thunar-1.0.2/docs/manual/ja/Makefile.in	2010-05-23 17:49:35.728552141 -0500
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/ja
+TARGET_DIR = $(htmldir)/ja
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.2.orig//docs/manual/ja/images/Makefile.in Thunar-1.0.2/docs/manual/ja/images/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/ja/images/Makefile.in	2010-05-21 12:10:59.000000000 -0500
+++ Thunar-1.0.2/docs/manual/ja/images/Makefile.in	2010-05-23 17:52:13.534542157 -0500
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/ja/images
+imagesdir = $(htmldir)/ja/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.2.orig//docs/manual/nl/Makefile.in Thunar-1.0.2/docs/manual/nl/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/nl/Makefile.in	2010-05-21 12:10:59.000000000 -0500
+++ Thunar-1.0.2/docs/manual/nl/Makefile.in	2010-05-23 17:49:42.981677782 -0500
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/nl
+TARGET_DIR = $(htmldir)/nl
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.2.orig//docs/manual/nl/images/Makefile.in Thunar-1.0.2/docs/manual/nl/images/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/nl/images/Makefile.in	2010-05-21 12:10:59.000000000 -0500
+++ Thunar-1.0.2/docs/manual/nl/images/Makefile.in	2010-05-23 17:52:19.353553300 -0500
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/nl/images
+imagesdir = $(htmldir)/nl/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.2.orig//docs/manual/pl/Makefile.in Thunar-1.0.2/docs/manual/pl/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/pl/Makefile.in	2010-05-21 12:10:59.000000000 -0500
+++ Thunar-1.0.2/docs/manual/pl/Makefile.in	2010-05-23 17:49:51.863542378 -0500
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/pl
+TARGET_DIR = $(htmldir)/pl
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.2.orig//docs/manual/pl/images/Makefile.in Thunar-1.0.2/docs/manual/pl/images/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/pl/images/Makefile.in	2010-05-21 12:10:59.000000000 -0500
+++ Thunar-1.0.2/docs/manual/pl/images/Makefile.in	2010-05-23 17:52:27.285677594 -0500
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/pl/images
+imagesdir = $(htmldir)/pl/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.2.orig//docs/manual/ru/Makefile.in Thunar-1.0.2/docs/manual/ru/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/ru/Makefile.in	2010-05-21 12:10:59.000000000 -0500
+++ Thunar-1.0.2/docs/manual/ru/Makefile.in	2010-05-23 17:49:59.951552903 -0500
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/ru
+TARGET_DIR = $(htmldir)/ru
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.2.orig//docs/manual/ru/images/Makefile.in Thunar-1.0.2/docs/manual/ru/images/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/ru/images/Makefile.in	2010-05-21 12:10:59.000000000 -0500
+++ Thunar-1.0.2/docs/manual/ru/images/Makefile.in	2010-05-23 17:52:33.673677284 -0500
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/ru/images
+imagesdir = $(htmldir)/ru/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.2.orig//docs/manual/tr/Makefile.in Thunar-1.0.2/docs/manual/tr/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/tr/Makefile.in	2010-05-21 12:10:59.000000000 -0500
+++ Thunar-1.0.2/docs/manual/tr/Makefile.in	2010-05-23 17:50:08.415553080 -0500
@@ -334,7 +334,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/tr
+TARGET_DIR = $(htmldir)/tr
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.2.orig//docs/manual/tr/images/Makefile.in Thunar-1.0.2/docs/manual/tr/images/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/tr/images/Makefile.in	2010-05-21 12:10:59.000000000 -0500
+++ Thunar-1.0.2/docs/manual/tr/images/Makefile.in	2010-05-23 17:52:40.320542409 -0500
@@ -314,7 +314,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/tr/images
+imagesdir = $(htmldir)/tr/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
diff -Nur Thunar-1.0.2.orig//docs/manual/zh_TW/Makefile.in Thunar-1.0.2/docs/manual/zh_TW/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/zh_TW/Makefile.in	2010-05-21 12:10:59.000000000 -0500
+++ Thunar-1.0.2/docs/manual/zh_TW/Makefile.in	2010-05-23 17:50:20.989569041 -0500
@@ -336,7 +336,7 @@
 SUBDIRS = \
 	images
 
-TARGET_DIR = $(datadir)/doc/Thunar/html/zh_TW
+TARGET_DIR = $(htmldir)/zh_TW
 STYLESHEET = ../thunar.xsl
 DOCUMENT = Thunar.xml
 
diff -Nur Thunar-1.0.2.orig//docs/manual/zh_TW/images/Makefile.in Thunar-1.0.2/docs/manual/zh_TW/images/Makefile.in
--- Thunar-1.0.2.orig//docs/manual/zh_TW/images/Makefile.in	2010-05-21 12:10:59.000000000 -0500
+++ Thunar-1.0.2/docs/manual/zh_TW/images/Makefile.in	2010-05-23 17:52:47.172552372 -0500
@@ -316,7 +316,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xfconf_query_found = @xfconf_query_found@
-imagesdir = $(datadir)/doc/Thunar/html/zh_TW/images
+imagesdir = $(htmldir)/zh_TW/images
 images_DATA = \
 	bulk-rename.png							\
 	file-manager-window.png						\
