diff -Nur xfce-utils-4.6.2.orig//scripts/xinitrc.in.in xfce-utils-4.6.2/scripts/xinitrc.in.in
--- xfce-utils-4.6.2.orig//scripts/xinitrc.in.in	2010-05-21 11:56:26.000000000 -0500
+++ xfce-utils-4.6.2/scripts/xinitrc.in.in	2011-03-09 22:56:18.687297854 -0600
@@ -97,6 +97,7 @@
 cat /dev/null $XRESOURCES | xrdb -nocpp -merge -
 
 # load local modmap
+test -r /etc/X11/xinit/.Xmodmap && xmodmap /etc/X11/xinit/.Xmodmap
 test -r $HOME/.Xmodmap && xmodmap $HOME/.Xmodmap
 
 # Launch xscreensaver (if available), but only as non-root user
