--- ./scripts/xinitrc.in.in.orig	2009-04-09 18:21:13.000000000 -0500
+++ ./scripts/xinitrc.in.in	2010-06-18 11:33:37.000000000 -0500
@@ -131,7 +131,11 @@
 	x|xno*)
 		;;
 	*)
-		$xfcesm
+                if [ -z "$DESKTOP_SESSION" -a -x /usr/bin/ck-launch-session ]; then
+                        ck-launch-session $xfcesm
+                else
+                        $xfcesm
+                fi
 
 		if test $kill_sshagent -eq 1; then
 			eval `$sshagent -k`
