diff -Nur xfdesktop-4.6.1.orig/modules/menu/directory-data/xfce-education.directory.in xfdesktop-4.6.1/modules/menu/directory-data/xfce-education.directory.in
--- xfdesktop-4.6.1.orig/modules/menu/directory-data/xfce-education.directory.in	2009-01-12 16:03:19.000000000 -0600
+++ xfdesktop-4.6.1/modules/menu/directory-data/xfce-education.directory.in	2009-07-31 01:04:29.029467762 -0500
@@ -2,6 +2,6 @@
 Version=1.0
 Encoding=UTF-8
 Type=Directory
-Icon=
+Icon=applications-education
 _Name=Education
 _Comment=Educational software
