--- ./Imakefile.orig	2025-03-25 16:16:57.515767082 -0500
+++ ./Imakefile	2025-03-25 16:20:02.775979601 -0500
@@ -22,7 +22,7 @@
 		SeSet.o SeSig.o SeString.o SeSubs.o SeSubsX.o SeSupp.o \
 		SeTerm.o SeTrans.o SeWin.o MultiList.o
 
-HELPFILE = -DHELPFILE=\"$(LIBDIR)$(PATHSEP)seyon.help\"
+HELPFILE = -DHELPFILE=\"/usr/share/X11/seyon.help\"
 
 /* These are apparently missing from versions of X prior to R5 */
 
