--- ./Makefile.std.orig	2010-02-19 13:24:58.000000000 -0600
+++ ./Makefile.std	2010-02-19 13:27:20.000000000 -0600
@@ -179,13 +179,13 @@
 JP2K    = -DDOJP2K
 ###
 #JP2KDIR = ../../jasper
-JP2KDIR = /usr/local/lib
+JP2KDIR = /usr
 ###
 #JP2KINC = -I$(JP2KDIR)
-JP2KINC = -I/usr/local/include
+JP2KINC = -I/usr/include
 ###
-#JP2KLIB = -L$(JP2KDIR) -ljasper
-JP2KLIB = $(JP2KDIR)/libjasper.a
+JP2KLIB = -L$(JP2KDIR) -ljasper
+#JP2KLIB = $(JP2KDIR)/libjasper.a
 
 
 ###
