--- ./Makefile.std.orig	2022-02-24 14:03:06.099774095 -0600
+++ ./Makefile.std	2022-02-24 14:04:14.407770317 -0600
@@ -176,15 +176,15 @@
 ### if, for whatever reason, you're unable to get the JasPer JPEG-2000 library
 ### to compile on your machine, *COMMENT OUT* the following lines
 ###
-JP2K    = -DDOJP2K
+#JP2K    = -DDOJP2K
 ###
 #JP2KDIR = ../../jasper
-JP2KDIR = /usr
+#JP2KDIR = /usr
 ###
 #JP2KINC = -I$(JP2KDIR)
-JP2KINC = -I/usr/include
+#JP2KINC = -I/usr/include
 ###
-JP2KLIB = -L$(JP2KDIR) -ljasper
+#JP2KLIB = -L$(JP2KDIR) -ljasper
 #JP2KLIB = $(JP2KDIR)/libjasper.a
 
 
