--- ./routed/trace.c.orig	Sun Feb 25 18:14:10 2001
+++ ./routed/trace.c	Sun Feb 25 18:14:24 2001
@@ -50,6 +50,7 @@
 #include <syslog.h>
 #include <errno.h>
 #include "pathnames.h"
+#include <time.h>
 
 #define	NRECORDS	50		/* size of circular trace buffer */
 
--- ./routed/main.c.orig	Sun Feb 25 18:10:48 2001
+++ ./routed/main.c	Sun Feb 25 18:11:14 2001
@@ -43,6 +43,7 @@
   "$Id: main.c,v 1.17 2000/12/23 18:07:39 jr Exp $";
 
 #include "../version.h"
+#include <time.h>
 
 /*
  * Routing Table Management Daemon
