--- ./plugins/infowidget/geoipmanager.cpp.orig	2021-06-05 16:25:43.000000000 -0500
+++ ./plugins/infowidget/geoipmanager.cpp	2021-06-10 15:57:49.201091341 -0500
@@ -45,7 +45,7 @@
 #ifdef USE_SYSTEM_GEOIP
     geo_ip = GeoIP_open_type(GEOIP_COUNTRY_EDITION, GEOIP_STANDARD);
 #else
-    geoip_data_file = QStandardPaths::locate(QStandardPaths::AppDataLocation, QStringLiteral("geoip.dat"));
+    geoip_data_file = QStringLiteral("/usr/share/apps/ktorrent/GeoIP/geoip.dat");
     if (geoip_data_file.isEmpty())
         geoip_data_file = QStandardPaths::locate(QStandardPaths::AppDataLocation, QStringLiteral("GeoIP.dat"));
 
@@ -56,11 +56,6 @@
         if (geo_ip) {
             QFileInfo fi(geoip_data_file);
             QDateTime now = QDateTime::currentDateTime();
-            if (fi.lastModified().daysTo(now) >= 30) {
-                // if the last time the geoip file was modified, was more then
-                // 30 days ago, redownload it
-                downloadDataBase();
-            }
         }
     }
 #endif
@@ -104,14 +99,6 @@
 
 void GeoIPManager::downloadDataBase()
 {
-    Out(SYS_INW | LOG_IMPORTANT) << "The download GeoIP database loading is defunct, skipping download" << endl;
-    return;
-#ifndef USE_SYSTEM_GEOIP
-    Out(SYS_INW | LOG_NOTICE) << "Downloading GeoIP database: " << geoip_url << endl;
-    download_destination = kt::DataDir(CreateIfNotExists) + geoip_url.fileName();
-    KIO::CopyJob *job = KIO::copy(geoip_url, QUrl::fromLocalFile(download_destination), KIO::Overwrite | KIO::HideProgressInfo);
-    connect(job, &KIO::CopyJob::result, this, &GeoIPManager::databaseDownloadFinished);
-#endif
 }
 
 void GeoIPManager::databaseDownloadFinished(KJob *job)
