--- sddm-0.18.1/data/scripts/Xsession.orig	2018-09-26 21:03:47.000000000 +0200
+++ sddm-0.18.1/data/scripts/Xsession	2020-01-21 11:55:24.997845799 +0100
@@ -91,10 +91,6 @@
 fi
 [ -f $HOME/.Xresources ] && xrdb -merge $HOME/.Xresources
 
-if [ -f "$USERXSESSION" ]; then
-  . "$USERXSESSION"
-fi
-
 if [ -z "$*" ]; then
     exec xmessage -center -buttons OK:0 -default OK "Sorry, $DESKTOP_SESSION is no valid session."
 else
