diff -uar sddm-qt5-7c8cc3a_20140827git.orig/src/helper/backend/PasswdBackend.cpp sddm-qt5-7c8cc3a_20140827git/src/helper/backend/PasswdBackend.cpp
--- sddm-qt5-7c8cc3a_20140827git.orig/src/helper/backend/PasswdBackend.cpp	2014-08-27 16:42:23.000000000 +0200
+++ sddm-qt5-7c8cc3a_20140827git/src/helper/backend/PasswdBackend.cpp	2014-09-15 12:24:06.717173942 +0200
@@ -20,8 +20,8 @@
 
 #include "PasswdBackend.h"
 
-#include "auth/AuthMessages.h"
-#include "helper/HelperApp.h"
+#include "Auth.h"
+#include "../HelperApp.h"
 
 #include <QtCore/QDebug>
 
diff -uar sddm-qt5-7c8cc3a_20140827git.orig/src/helper/backend/PasswdBackend.h sddm-qt5-7c8cc3a_20140827git/src/helper/backend/PasswdBackend.h
--- sddm-qt5-7c8cc3a_20140827git.orig/src/helper/backend/PasswdBackend.h	2014-08-27 16:42:23.000000000 +0200
+++ sddm-qt5-7c8cc3a_20140827git/src/helper/backend/PasswdBackend.h	2014-09-15 12:17:43.135172278 +0200
@@ -21,6 +21,7 @@
 #if !defined(PASSWDBACKEND_H) && !defined(USE_PAM)
 #define PASSWDBACKEND_H
 
+#include "AuthMessages.h"
 #include "../Backend.h"
 
 namespace SDDM {
