--- ./include/config.h.orig	2008-08-06 20:59:51.000000000 -0500
+++ ./include/config.h	2008-08-06 21:00:37.000000000 -0500
@@ -557,10 +557,10 @@
 #define STDC_HEADERS 1
 
 /* System pinerc */
-#define SYSTEM_PINERC "/usr/lib/pine.conf"
+#define SYSTEM_PINERC "/etc/pine.conf"
 
 /* System fixed pinerc */
-#define SYSTEM_PINERC_FIXED "/usr/lib/pine.conf.fixed"
+#define SYSTEM_PINERC_FIXED "/etc/pine.conf.fixed"
 
 /* Pine-Centric Host Specifier */
 #define SYSTYPE "LNX"
