--- ./util/startfluxbox.in.orig	2011-10-29 08:46:15.000000000 -0500
+++ ./util/startfluxbox.in	2012-08-08 23:35:10.841041336 -0500
@@ -50,6 +50,11 @@
 # wmsmixer -w &
 # idesk &
 
+# Start DBUS session bus:
+if [ -z "\$DBUS_SESSION_BUS_ADDRESS" ]; then
+   eval \$(dbus-launch --sh-syntax --exit-with-session)
+fi
+
 # And last but not least we start fluxbox.
 # Because it is the last app you have to run it with ''exec'' before it.
 
