--- ./util/fluxbox-remote.cc.orig	2015-02-08 04:44:45.377187009 -0600
+++ ./util/fluxbox-remote.cc	2021-08-05 23:16:44.301526535 -0500
@@ -73,7 +73,7 @@
     if (strcmp(cmd, "result") == 0) {
         XTextProperty text_prop;
         if (XGetTextProperty(disp, root, &text_prop, atom_result) != 0
-            && text_prop.value > 0
+            && text_prop.value != 0
             && text_prop.nitems > 0) {
 
             printf("%s", text_prop.value);
