--- ./data/keys.orig	2015-02-08 04:44:45.333187008 -0600
+++ ./data/keys	2017-06-09 13:25:14.558517846 -0500
@@ -65,8 +65,8 @@
 
 # volume settings, using common keycodes
 # if these don't work, use xev to find out your real keycodes
-176 :Exec amixer sset Master,0 1+
-174 :Exec amixer sset Master,0 1-
+176 :Exec amixer sset Master,0 5%+
+174 :Exec amixer sset Master,0 5%-
 160 :Exec amixer sset Master,0 toggle
 
 # current window commands
