--- ./loudmouth/Makefile.in.orig	2012-12-01 12:53:52.000000000 -0600
+++ ./loudmouth/Makefile.in	2013-03-20 21:25:34.368522959 -0500
@@ -148,7 +148,7 @@
 CCLD = $(CC)
 LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
 	--mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) \
-	$(LDFLAGS) -o $@
+	$(LDFLAGS) -o $@ -lgobject-2.0
 SOURCES = $(libloudmouth_1_la_SOURCES) $(test_dns_SOURCES)
 DIST_SOURCES = $(am__libloudmouth_1_la_SOURCES_DIST) \
 	$(test_dns_SOURCES)
