--- ./scripts/libast_prereq.sh.orig	2019-05-05 13:34:36.069904205 -0500
+++ ./scripts/libast_prereq.sh	2019-05-05 13:37:19.117888974 -0500
@@ -20,12 +20,12 @@
 
 if [ -z "$CC" ];
 then
-    CC=cc
+    CC=clang
 fi
 
 cd "$MESON_BUILD_ROOT"
 
-if cc --version | grep -q "GCC"
+if $CC --version | grep -q "GCC"
 then
     gcc_major_version=$(cc -dumpversion | cut -d. -f1)
     if [ "$gcc_major_version" -ge 9 ]
