diff -urB strigi-multimedia/CMakeLists.txt new/CMakeLists.txt
--- strigi-multimedia/CMakeLists.txt	2012-04-08 11:47:03.923386124 +0200
+++ new/CMakeLists.txt	2012-04-08 12:29:05.807989600 +0200
@@ -1,3 +1,5 @@
+Project(strigi-multimedia)
+
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/)
 find_package(Qt4 REQUIRED)
 find_package(KDE4 REQUIRED)
@@ -5,12 +7,10 @@
 include(MacroLibrary)
 
 include_directories(${KDE4_INCLUDES} ${QT_INCLUDES})
+add_definitions (${QT_DEFINITIONS} ${KDE4_DEFINITIONS} )
 
 include(CheckIncludeFileCXX)
 
-message (STATUS "port strigi-analyzer !!!")
-if(KFILE_PLUGINS_PORTED) 
-
 macro_optional_find_package(Theora)
 macro_log_feature(THEORA_FOUND "Theora" "A video codec intended for use within the Ogg's project's Ogg multimedia streaming system" "http://www.theora.org" FALSE "" "Required to build the Theora Strigi Analyzer.")
 
@@ -35,5 +35,3 @@
 if (OGGVORBIS_FOUND)
 	add_subdirectory(ogg)
 endif(OGGVORBIS_FOUND)
-
-endif(KFILE_PLUGINS_PORTED)
