--- ./docs/reference/meson.build.orig	2020-09-10 05:42:41.475683000 -0500
+++ ./docs/reference/meson.build	2020-09-15 13:26:21.552796804 -0500
@@ -52,7 +52,7 @@
   endif
   # Check we have the minimum gtk-doc version required. Older versions won't
   # generate correct documentation.
-  dependency('gtk-doc', version : '>=1.32.1',
+  dependency('gtk-doc', version : '>=1.32',
     fallback : ['gtk-doc', 'dummy_dep'],
     default_options : ['tests=false'])
 endif
