--- ./gio/gnetworkmonitornetlink.c.orig	2025-03-06 07:09:13.000000000 -0600
+++ ./gio/gnetworkmonitornetlink.c	2025-03-06 18:22:01.818783403 -0600
@@ -38,14 +38,8 @@
 
 /* must come at the end to pick system includes from
  * gnetworkingprivate.h */
-#ifdef HAVE_LINUX_NETLINK_H
 #include <linux/netlink.h>
 #include <linux/rtnetlink.h>
-#endif
-#ifdef HAVE_NETLINK_NETLINK_H
-#include <netlink/netlink.h>
-#include <netlink/netlink_route.h>
-#endif
 
 static GInitableIface *initable_parent_iface;
 static void g_network_monitor_netlink_iface_init (GNetworkMonitorInterface *iface);
