diff --git a/slacktrack.SlackBuild b/slacktrack.SlackBuild
index be5361c..d2de70d 100755
--- a/slacktrack.SlackBuild
+++ b/slacktrack.SlackBuild
@@ -22,6 +22,8 @@ case "$( uname -m )" in
            PKGEXT=txz ;;
   arm*)    export ARCH=arm
            PKGEXT=txz ;;
+  loongarch64)    export ARCH=loong64
+           PKGEXT=txz ;;
   # Unless $ARCH is already set, use uname -m for all other archs:
      *)    export ARCH=$( uname -m )
            PKGEXT=txz ;;
