diff -Nur pavucontrol-3.0.orig/src/pavucontrol.glade pavucontrol-3.0/src/pavucontrol.glade
--- pavucontrol-3.0.orig/src/pavucontrol.glade	2015-03-24 09:39:00.000000000 -0500
+++ pavucontrol-3.0/src/pavucontrol.glade	2016-01-04 22:40:15.553206280 -0600
@@ -312,7 +312,7 @@
                               <object class="GtkImage" id="image18">
                                 <property name="visible">True</property>
                                 <property name="can_focus">False</property>
-                                <property name="icon_name">stock_lock</property>
+                                <property name="icon_name">security-medium</property>
                                 <property name="icon-size">1</property>
                               </object>
                             </child>
@@ -1474,7 +1474,7 @@
                               <object class="GtkImage" id="image18">
                                 <property name="visible">True</property>
                                 <property name="can_focus">False</property>
-                                <property name="icon_name">stock_lock</property>
+                                <property name="icon_name">security-medium</property>
                                 <property name="icon-size">1</property>
                               </object>
                             </child>
