diff -urNp a/net-snmp-create-v3-user.in b/net-snmp-create-v3-user.in
--- a/net-snmp-create-v3-user.in	2018-07-18 11:11:53.227015237 +0200
+++ b/net-snmp-create-v3-user.in	2018-07-18 11:12:13.375010176 +0200
@@ -137,7 +137,7 @@ fi
 echo $line >> $outfile
 prefix="@prefix@"
 datarootdir="@datarootdir@"
-outfile="@datadir@/snmp/snmpd.conf"
+outfile="/etc/snmp/snmpd.conf"
 line="$token $user"
 echo "adding the following line to $outfile:"
 echo "  " $line
