diff -urNp a/man/netsnmp_config_api.3.def b/man/netsnmp_config_api.3.def
--- a/man/netsnmp_config_api.3.def	2018-07-18 11:18:06.196792766 +0200
+++ b/man/netsnmp_config_api.3.def	2018-07-18 11:20:04.631679886 +0200
@@ -295,7 +295,7 @@ for one particular machine.
 .PP
 The default list of directories to search is \fC SYSCONFDIR/snmp\fP,
 followed by \fC DATADIR/snmp\fP,
-followed by \fC LIBDIR/snmp\fP,
+followed by \fC /usr/lib(64)/snmp\fP,
 followed by \fC $HOME/.snmp\fP.
 This list can be changed by setting the environmental variable
 .I SNMPCONFPATH
@@ -367,7 +367,7 @@ A colon separated list of directories to
 files in.
 Default:
 .br
-SYSCONFDIR/snmp:\:DATADIR/snmp:\:LIBDIR/snmp:\:$HOME/.snmp
+SYSCONFDIR/snmp:\:DATADIR/snmp:\:/usr/lib(64)/snmp:\:$HOME/.snmp
 .SH "SEE ALSO"
 netsnmp_mib_api(3), snmp_api(3)
 .\" Local Variables:
diff -urNp a/man/snmp_config.5.def b/man/snmp_config.5.def
--- a/man/snmp_config.5.def	2018-07-18 11:18:06.194792767 +0200
+++ b/man/snmp_config.5.def	2018-07-18 11:20:56.423626117 +0200
@@ -10,7 +10,7 @@ First off, there are numerous places tha
 found and read from.  By default, the applications look for
 configuration files in the following 4 directories, in order:
 SYSCONFDIR/snmp,
-DATADIR/snmp, LIBDIR/snmp, and $HOME/.snmp.  In each of these
+DATADIR/snmp, /usr/lib(64)/snmp, and $HOME/.snmp.  In each of these
 directories, it looks for files snmp.conf, snmpd.conf and/or
 snmptrapd.conf, as well as snmp.local.conf, snmpd.local.conf
 and/or snmptrapd.local.conf. *.local.conf are always
diff -urNp a/man/snmpd.conf.5.def b/man/snmpd.conf.5.def
--- a/man/snmpd.conf.5.def	2018-07-18 11:18:06.196792766 +0200
+++ b/man/snmpd.conf.5.def	2018-07-18 11:21:44.263574388 +0200
@@ -1559,7 +1559,7 @@ filename), and call the initialisation r
 .RS
 .IP "Note:"
 If the specified PATH is not a fully qualified filename, it will
-be interpreted relative to LIBDIR/snmp/dlmod, and \fC.so\fR
+be interpreted relative to /usr/lib(64)/snmp/dlmod, and \fC.so\fR
 will be appended to the filename.
 .RE
 .PP
