diff --git a/src/aainfo.c b/src/aainfo.c
index d3f6d50..e4eb423 100644
--- a/src/aainfo.c
+++ b/src/aainfo.c
@@ -1,4 +1,4 @@
-
+#include <stdlib.h>
 #include "aalib.h"
 #include "aaint.h"
 int main(int argc, char **argv)
diff --git a/src/aalinuxkbd.c b/src/aalinuxkbd.c
index 610e026..6de433c 100644
--- a/src/aalinuxkbd.c
+++ b/src/aalinuxkbd.c
@@ -25,6 +25,7 @@
 #ifdef GPM_MOUSEDRIVER
 #include <gpm.h>
 #endif
+#include <stdlib.h>
 #include "aalib.h"
 #include "aaint.h"
 static struct termios oldios;
