diff --git a/src/aainfo.c b/src/aainfo.c
index d3f6d50..fc76bed 100644
--- a/src/aainfo.c
+++ b/src/aainfo.c
@@ -1,6 +1,7 @@
 
 #include "aalib.h"
 #include "aaint.h"
+#include <stdlib.h>
 int main(int argc, char **argv)
 {
     aa_context *c;
diff --git a/src/aalinuxkbd.c b/src/aalinuxkbd.c
index 610e026..3e7756a 100644
--- a/src/aalinuxkbd.c
+++ b/src/aalinuxkbd.c
@@ -27,6 +27,7 @@
 #endif
 #include "aalib.h"
 #include "aaint.h"
+#include <stdlib.h>
 static struct termios oldios;
 
 /* random keys not convered anywhere else below */
