diff -Nur xfce4-screensaver-0.1.11.orig/data/xfce4-screensaver-preferences.desktop.in xfce4-screensaver-0.1.11/data/xfce4-screensaver-preferences.desktop.in
--- xfce4-screensaver-0.1.11.orig/data/xfce4-screensaver-preferences.desktop.in	2020-11-07 06:53:45.000000000 -0600
+++ xfce4-screensaver-0.1.11/data/xfce4-screensaver-preferences.desktop.in	2020-12-15 20:16:14.521788731 -0600
@@ -1,6 +1,6 @@
 [Desktop Entry]
-_Name=Screensaver
-_Comment=Set your screensaver preferences
+_Name=xfce4-screensaver
+_Comment=Set your xfce4-screensaver preferences
 Icon=org.xfce.ScreenSaver
 Exec=xfce4-screensaver-preferences
 Terminal=false
