--- ./CMakeLists.txt.orig	2013-08-03 23:22:20.000000000 -0500
+++ ./CMakeLists.txt	2018-01-30 14:37:52.197974764 -0600
@@ -35,6 +35,7 @@
 set(KU_FIRSTGID 500 CACHE STRING "First GID of normal users")
 set(KU_USERPRIVATEGROUP true CACHE STRING "User private groups by default")
 
+include(CheckIncludeFiles)
 check_include_files(shadow.h HAVE_SHADOW_H)
 check_include_files(crypt.h HAVE_CRYPT_H)
 check_include_files(sys/stat.h HAVE_SYS_STAT_H)
