--- ./checklist.c.orig	2008-06-20 15:58:21.000000000 -0500
+++ ./checklist.c	2009-04-22 19:00:18.000000000 -0500
@@ -31,7 +31,7 @@
 
 static int list_width, check_x, item_x, checkflag;
 
-#define MIN_HIGH  (1 + (5 * MARGIN))
+#define MIN_HIGH 4
 
 #define LLEN(n) ((n) * CHECKBOX_TAGS)
 #define ItemData(i)    &items[LLEN(i)]
--- ./menubox.c.orig	2008-06-19 19:54:16.000000000 -0500
+++ ./menubox.c	2009-04-22 19:00:07.000000000 -0500
@@ -35,7 +35,7 @@
     Editing
 } Mode;
 
-#define MIN_HIGH  (1 + (5 * MARGIN))
+#define MIN_HIGH 4
 
 #define INPUT_ROWS     3	/* rows per inputmenu entry */
 
