--- ./apps/konsole/src/ViewContainer.cpp.orig	2011-04-04 15:05:38.000000000 -0500
+++ ./apps/konsole/src/ViewContainer.cpp	2011-04-04 15:08:58.000000000 -0500
@@ -448,7 +448,7 @@
     _closeTabButton->setIcon(KIcon("tab-close"));
     _closeTabButton->adjustSize();
     _closeTabButton->setHidden(true);
-    _closeTabButton->setEnabled(false);
+    _closeTabButton->setEnabled(true);
 
     connect( _tabBar , SIGNAL(currentChanged(int)) , this , SLOT(currentTabChanged(int)) );
     connect( _tabBar , SIGNAL(tabDoubleClicked(int)) , this , SLOT(tabDoubleClicked(int)) );
