--- trunk/KDE/kdebase/apps/konqueror/sidebar/module_manager.h	2009/11/13 18:46:51	1048689
+++ trunk/KDE/kdebase/apps/konqueror/sidebar/module_manager.h	2010/01/07 20:09:11	1071281
@@ -18,6 +18,9 @@
     Boston, MA 02110-1301, USA.
 */
 
+#ifndef MODULE_MANAGER_H
+#define MODULE_MANAGER_H
+
 #include <QStringList>
 #include <kservice.h>
 class KConfigGroup;
@@ -74,3 +77,5 @@
     KConfigGroup *m_config; // owned by SidebarWidget
     QString m_localPath; // local path
 };
+
+#endif

