--- options.h.orig	2013-06-21 21:50:34.859528230 -0500
+++ options.h	2013-06-21 21:52:59.208516285 -0500
@@ -243,7 +243,7 @@
 /* The command to invoke for xauth when using X11 forwarding.
  * "-q" for quiet */
 #ifndef XAUTH_COMMAND
-#define XAUTH_COMMAND "/usr/bin/X11/xauth -q"
+#define XAUTH_COMMAND "/usr/bin/xauth -q"
 #endif
 
 /* if you want to enable running an sftp server (such as the one included with
