diff -uarN dropbear-0.50.orig/options.h dropbear-0.50/options.h
--- dropbear-0.50.orig/options.h	2007-08-08 18:39:38.000000000 +0200
+++ dropbear-0.50/options.h	2008-03-10 14:43:29.261618300 +0100
@@ -209,7 +209,7 @@

 /* This is used by the scp binary when used as a client binary. If you're
  * not using the Dropbear client, you'll need to change it */
-#define _PATH_SSH_PROGRAM "/usr/bin/dbclient"
+#define _PATH_SSH_PROGRAM "/bin/dbclient"

 /* Whether to log commands executed by a client. This only logs the
  * (single) command sent to the server, not what a user did in a
