diff -Nur dropbear-2012.55.orig/options.h dropbear-2012.55/options.h
--- dropbear-2012.55.orig/options.h	2012-02-23 07:47:06.000000000 -0600
+++ dropbear-2012.55/options.h	2012-04-07 15:09:15.676322495 -0500
@@ -180,7 +180,7 @@
  * Public key logins are allowed for blank-password accounts regardless of this
  * setting.  PAM is not affected by this setting, it uses the normal pam.d
  * settings ('nullok' option) */
-/* #define ALLOW_BLANK_PASSWORD */
+#define ALLOW_BLANK_PASSWORD
 
 #define ENABLE_CLI_PASSWORD_AUTH
 #define ENABLE_CLI_PUBKEY_AUTH
