From 0d775547f0e7cf7529538704814469e1db5496f2 Mon Sep 17 00:00:00 2001
From: Detlef Graef <detlef.graef@yahoo.de>
Date: Sun, 5 Jul 2020 08:54:35 +0200
Subject: [PATCH] Remove empty line from file posting.xml

---
 pan/data-impl/profiles.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pan/data-impl/profiles.cc b/pan/data-impl/profiles.cc
index e999fda..4053cae 100644
--- a/pan/data-impl/profiles.cc
+++ b/pan/data-impl/profiles.cc
@@ -240,7 +240,7 @@ ProfilesImpl :: serialize (std::ostream& out) const
     }
     out << indent(--depth) << "</profile>\n";
   }
-  out << indent(--depth) << "</profiles>\n\n";
+  out << indent(--depth) << "</profiles>\n";
   out << indent(--depth) << "</posting>\n";
 }
 
-- 
GitLab

