--- ./configure.ac.orig	2021-09-16 11:17:07.000000000 -0500
+++ ./configure.ac	2021-09-20 12:51:27.786023975 -0500
@@ -159,7 +159,7 @@
                     AC_DEFINE(HAVE_GTK,[1],[GTK+ 3 support])]
                     )
   if test "x$want_gtkspell" = "xyes" ; then
-    PKG_CHECK_MODULES([GTKSPELL], [gtkspell3-3.0 >= $GTKSPELL3_REQUIRED enchant >= $ENCHANT_REQUIRED],
+    PKG_CHECK_MODULES([GTKSPELL], [gtkspell3-3.0 >= $GTKSPELL3_REQUIRED enchant-2 >= $ENCHANT_REQUIRED],
                       [
                       gtkspell_msg=yes
                       AC_DEFINE(HAVE_GTKSPELL,[1],[GtkSpell support for spellchecking])
@@ -172,7 +172,7 @@
   fi
 else
    if test "x$want_gtkspell" = "xyes" ; then
-   PKG_CHECK_MODULES([GTKSPELL], [gtkspell-2.0 >= $GTKSPELL_REQUIRED enchant >= $ENCHANT_REQUIRED],
+   PKG_CHECK_MODULES([GTKSPELL], [gtkspell-2.0 >= $GTKSPELL_REQUIRED enchant-2 >= $ENCHANT_REQUIRED],
                       [
                       gtkspell_msg=yes
                       AC_DEFINE(HAVE_GTKSPELL,[1],[GtkSpell support for spellchecking])
--- ./pan/gui/group-prefs-dialog.cc.orig	2021-09-20 12:51:27.786023975 -0500
+++ ./pan/gui/group-prefs-dialog.cc	2021-09-20 12:52:45.976025724 -0500
@@ -22,7 +22,7 @@
 #include <glib.h>
 #include "gtk-compat.h"
 #ifdef HAVE_GTKSPELL
-#include <enchant/enchant.h>
+#include <enchant-2/enchant.h>
 #endif
 #include <pan/general/debug.h>
 #include <pan/general/macros.h>
