diff -u -r --new-file pan-0.146.orig/configure.ac pan-0.146/configure.ac
--- pan-0.146.orig/configure.ac	2019-05-04 11:10:27.000000000 -0500
+++ pan-0.146/configure.ac	2021-06-29 14:19:49.612442354 -0500
@@ -149,7 +149,7 @@
                     AC_DEFINE(HAVE_GTK,[1],[GTK+ 3 support])]
                     )
   if test "x$want_gtkspell" = "xyes" ; then
-    PKG_CHECK_MODULES([GTKSPELL], [gtkspell3-3.0 >= $GTKSPELL3_REQUIRED enchant >= $ENCHANT_REQUIRED],
+    PKG_CHECK_MODULES([GTKSPELL], [gtkspell3-3.0 >= $GTKSPELL3_REQUIRED enchant-2 >= $ENCHANT_REQUIRED],
                       [
                       gtkspell_msg=yes
                       AC_DEFINE(HAVE_GTKSPELL,[1],[GtkSpell support for spellchecking])
@@ -162,7 +162,7 @@
   fi
 else
    if test "x$want_gtkspell" = "xyes" ; then
-   PKG_CHECK_MODULES([GTKSPELL], [gtkspell-2.0 >= $GTKSPELL_REQUIRED enchant >= $ENCHANT_REQUIRED],
+   PKG_CHECK_MODULES([GTKSPELL], [gtkspell-2.0 >= $GTKSPELL_REQUIRED enchant-2 >= $ENCHANT_REQUIRED],
                       [
                       gtkspell_msg=yes
                       AC_DEFINE(HAVE_GTKSPELL,[1],[GtkSpell support for spellchecking])
diff -u -r --new-file pan-0.146.orig/pan/gui/group-prefs-dialog.cc pan-0.146/pan/gui/group-prefs-dialog.cc
--- pan-0.146.orig/pan/gui/group-prefs-dialog.cc	2017-12-29 11:15:47.000000000 -0600
+++ pan-0.146/pan/gui/group-prefs-dialog.cc	2021-06-29 14:19:49.628442353 -0500
@@ -23,7 +23,7 @@
   #include <glib.h>
   #include "gtk-compat.h"
 #ifdef HAVE_GTKSPELL
-  #include <enchant/enchant.h>
+  #include <enchant-2/enchant.h>
 #endif
 }
 #include <pan/general/debug.h>
