--- ./pan/usenet-utils/numbers.cc.orig	2008-07-04 10:01:53.000000000 -0500
+++ ./pan/usenet-utils/numbers.cc	2010-02-03 18:06:16.000000000 -0600
@@ -19,6 +19,7 @@
 
 #include <config.h>
 #include <cctype>
+#include <cstdio>
 #include <pan/general/string-view.h>
 #include "numbers.h"
 #include <algorithm>
--- ./pan/tasks/nntp.cc.orig	2008-07-05 14:16:06.000000000 -0500
+++ ./pan/tasks/nntp.cc	2010-02-03 18:07:19.000000000 -0600
@@ -20,6 +20,7 @@
 #include <config.h>
 #include <cassert>
 #include <cstdarg>
+#include <cstdio>
 #include <cstdlib> // abort, atoi, strtoul
 extern "C" {
   #include <glib.h>
--- ./pan/tasks/nntp-pool.cc.orig	2008-07-05 01:39:27.000000000 -0500
+++ ./pan/tasks/nntp-pool.cc	2010-02-03 18:07:55.000000000 -0600
@@ -19,6 +19,7 @@
 
 #include <config.h>
 #include <ctime>
+#include <cstdio>
 #include <glib/gi18n.h>
 #include <pan/general/debug.h>
 #include <pan/general/log.h>
--- ./pan/general/progress.cc.orig	2007-08-01 12:00:01.000000000 -0500
+++ ./pan/general/progress.cc	2010-02-03 18:05:38.000000000 -0600
@@ -19,6 +19,7 @@
 
 #include <config.h>
 #include <cstdarg>
+#include <cstdio>
 #include "progress.h"
 #include "string-view.h"
 
--- ./pan/general/log.cc.orig	2007-08-01 12:00:01.000000000 -0500
+++ ./pan/general/log.cc	2010-02-03 18:05:19.000000000 -0600
@@ -20,6 +20,7 @@
 #include <config.h>
 #include <iostream>
 #include <cstdarg>
+#include <cstdio>
 #include "log.h"
 
 using namespace pan;
