diff -u -r --new-file cups-1.3.11.orig/desktop/cups.desktop cups-1.3.11/desktop/cups.desktop
--- cups-1.3.11.orig/desktop/cups.desktop	2008-09-05 19:30:39.000000000 -0500
+++ cups-1.3.11/desktop/cups.desktop	2009-08-21 20:18:23.604682681 -0500
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Categories=System;Printing;HardwareSettings;X-Red-Hat-Base;
-Exec=htmlview http://localhost:631/
+Exec=firefox http://localhost:631/
 Icon=cups
 StartupNotify=false
 Terminal=false
