--- kdelibs-4.4.85/cmake/modules/FindDocBookXML.cmake.orig	2010-05-27 19:25:00.000000000 +0200
+++ kdelibs-4.4.85/cmake/modules/FindDocBookXML.cmake	2010-06-07 16:18:35.000000000 +0200
@@ -12,7 +12,7 @@
 # Redistribution and use is allowed according to the terms of the BSD license.
 # For details see the accompanying COPYING-CMAKE-SCRIPTS file.
 
-set (DOCBOOKXML_CURRENTDTD_VERSION "4.2"
+set (DOCBOOKXML_CURRENTDTD_VERSION "4.5"
      CACHE INTERNAL "Required version of XML DTDs")
 
 set (DTD_PATH_LIST
