--- kdelibs-4.8.0/solid/solid/managerbase.cpp.orig	2011-07-27 20:34:39.000000000 +0200
+++ kdelibs-4.8.0/solid/solid/managerbase.cpp	2012-01-26 09:17:49.409993419 +0100
@@ -90,7 +90,11 @@
 #        endif
 
 #        if defined (HUPNP_FOUND)
-            m_backends << new Solid::Backends::UPnP::UPnPDeviceManager(0);
+            bool solidUpnpEnabled
+                = QString::fromLocal8Bit(qgetenv("SOLID_UPNP")).toInt()==1;
+            if (solidUpnpEnabled) {
+                m_backends << new Solid::Backends::UPnP::UPnPDeviceManager(0);
+            }
 #        endif
     }
 }
