diff -ur vnc_unixsrc.orig/tightvncserver.conf vnc_unixsrc/tightvncserver.conf
--- vnc_unixsrc.orig/tightvncserver.conf	2009-02-12 05:27:18.000000000 +0100
+++ vnc_unixsrc/tightvncserver.conf	2009-06-02 23:55:22.000000000 +0200
@@ -19,15 +19,15 @@
 # $geometry = "1024x768";
 # $depth = 24;
 # $desktopName = "X";
-# $vncClasses = "/usr/local/vnc/classes";
+# $vncClasses = "/usr/share/tightvnc/classes";
 # $vncUserDir = "$ENV{HOME}/.vnc";
-# $fontPath = "unix/:7100";
+# $fontPath = "/usr/share/fonts/misc/,/usr/share/fonts/Type1/,/usr/share/fonts/75dpi/";
 # $authType = "-rfbauth $vncUserDir/passwd";
-# $colorPath = "/usr/lib/X11/rgb";
+# $colorPath = "/usr/share/X11/rgb";
 
 ## Here is another example of setting the font path:
 #
-# $fontPath = "/usr/lib/X11/fonts/misc/"
+# $fontPath = "unix/:7100";
 # $fontPath = "$fontPath,/usr/lib/X11/fonts/75dpi/";
 
 ## You might wish to make your vnc directory under /tmp, to make sure
diff -ur vnc_unixsrc.orig/vncinstall vnc_unixsrc/vncinstall
--- vnc_unixsrc.orig/vncinstall	2002-03-19 12:44:02.000000000 +0100
+++ vnc_unixsrc/vncinstall	2009-06-02 23:51:54.000000000 +0200
@@ -47,7 +47,7 @@
   else
     echo "Copying $f -> $bin_dst/`basename $f`"
     cp -pf $f $bin_dst
-    chmod 0555 $bin_dst/`basename $f`
+    chmod 0755 $bin_dst/`basename $f`
   fi
 
   # Installing man pages
diff -ur vnc_unixsrc.orig/vncserver vnc_unixsrc/vncserver
--- vnc_unixsrc.orig/vncserver	2009-02-12 05:27:18.000000000 +0100
+++ vnc_unixsrc/vncserver	2009-06-02 23:51:39.000000000 +0200
@@ -32,10 +32,11 @@
 $geometry = "1024x768";
 $depth = 24;
 $desktopName = "X";
-$vncClasses = "/usr/local/vnc/classes";
+$vncClasses = "/usr/share/tightvnc/classes";
 $vncUserDir = "$ENV{HOME}/.vnc";
-$fontPath = "unix/:7100";
+$fontPath = "/usr/share/fonts/misc/,/usr/share/fonts/Type1/,/usr/share/fonts/75dpi/";
 $authType = "-rfbauth $vncUserDir/passwd";
+$colorPath = "/usr/share/X11/rgb";
 
 # Read configuration from the system-wide and user files if present.
 
