diff -Nur geeqie-1.4.orig/Makefile.am geeqie-1.4/Makefile.am
--- geeqie-1.4.orig/Makefile.am	2017-12-31 06:31:21.000000000 -0600
+++ geeqie-1.4/Makefile.am	2018-01-02 02:18:15.881141642 -0600
@@ -9,11 +9,7 @@
 
 readmedir = @readmedir@
 
-if HAVE_MARKDOWN
-readme_DATA = README.md COPYING ChangeLog TODO README.lirc AUTHORS README.html ChangeLog.html
-else
-readme_DATA = README.md COPYING ChangeLog TODO README.lirc AUTHORS ChangeLog.html
-endif
+readme_DATA = README.md COPYING ChangeLog TODO README.lirc AUTHORS
 
 desktopdir = $(datadir)/applications
 desktop_in_files = geeqie.desktop.in
@@ -37,9 +33,3 @@
 
 DISTCLEANFILES = config.report
 
-.PHONY: ChangeLog
-ChangeLog.html:
-	./gen_changelog.sh
-
-README.html: README.md
-	./gen_readme.sh
