diff -Nur groff-1.20.orig/Makefile.in groff-1.20/Makefile.in
--- groff-1.20.orig/Makefile.in	2009-01-05 14:45:13.000000000 -0600
+++ groff-1.20/Makefile.in	2009-01-13 22:14:58.237116441 -0600
@@ -146,7 +146,7 @@
 infodir=@infodir@
 
 # `docdir' says where to install documentation files.
-docdir=$(datadir)/doc/groff/$(version)$(revision)
+docdir=@docdir@
 
 # `exampledir' says where to install example files.
 exampledir=$(docdir)/examples
