--- ./MAKEDEV.orig	2002-06-09 12:31:22.000000000 -0700
+++ ./MAKEDEV	2002-06-09 12:31:51.000000000 -0700
@@ -914,7 +914,7 @@
 		base=`index ab $unit`
 		base=`math $base \* 64`
 		makedev hd$unit b $major $base $disk
-		for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20
+                for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32
 		do
 			makedev hd$unit$part b $major `math $base + $part` $disk
 		done
@@ -925,7 +925,7 @@
 		base=`index cd $unit`
 		base=`math $base \* 64`
 		makedev hd$unit b $major $base $disk
-		for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20
+                for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32
 		do
 			makedev hd$unit$part b $major $(( $base + $part )) $disk
 		done
@@ -936,7 +936,7 @@
 		base=`index ef $unit`
 		base=`math $base \* 64`
 		makedev hd$unit b $major $base $disk
-		for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20
+                for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32
 		do
 			makedev hd$unit$part b $major $(( $base + $part )) $disk
 		done
@@ -947,7 +947,7 @@
 		base=`index gh $unit`
 		base=`math $base \* 64`
 		makedev hd$unit b $major $base $disk
-		for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20
+                for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32
 		do
 			makedev hd$unit$part b $major $(( $base + $part )) $disk
 		done
@@ -958,7 +958,7 @@
 		base=`index ij $unit`
 		base=`math $base \* 64`
 		makedev hd$unit b $major $base $disk
-		for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20
+                for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32
 		do
 			makedev hd$unit$part b $major $(( $base + $part )) $disk
 		done
@@ -969,7 +969,7 @@
 		base=`index kl $unit`
 		base=`math $base \* 64`
 		makedev hd$unit b $major $base $disk
-		for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20
+                for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32
 		do
 			makedev hd$unit$part b $major $(( $base + $part )) $disk
 		done
@@ -980,7 +980,7 @@
 		base=`index mn $unit`
 		base=`math $base \* 64`
 		makedev hd$unit b $major $base $disk
-		for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20
+                for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32
 		do
 			makedev hd$unit$part b $major $(( $base + $part )) $disk
 		done
@@ -991,7 +991,7 @@
                 base=`index op $unit`
                 base=`math $base \* 64`
                 makedev hd$unit b $major $base $disk
-                for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20
+                for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32
                 do
                         makedev hd$unit$part b $major $(( $base + $part )) $disk
                 done
@@ -1002,7 +1002,7 @@
                 base=`index qr $unit`
                 base=`math $base \* 64`
                 makedev hd$unit b $major $base $disk
-                for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20
+                for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32
                 do
                         makedev hd$unit$part b $major $(( $base + $part )) $disk
                 done
@@ -1013,7 +1013,7 @@
                 base=`index st $unit`
                 base=`math $base \* 64`
                 makedev hd$unit b $major $base $disk
-                for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20
+                for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32
                 do
                         makedev hd$unit$part b $major $(( $base + $part )) $disk
                 done
