--- ./MAKEDEV.orig	Sun Feb 24 15:46:03 2002
+++ ./MAKEDEV	Sun Feb 24 16:12:18 2002
@@ -147,7 +147,7 @@
 			Character|Block|'')
 				;;
 			*)
-				eval "major_$device=$major"
+				# eval "major_$device=$major"
 				devices="$devices $device"
 				;;
 		esac
@@ -964,7 +964,7 @@
 		done
 		;;
 	hd[k-l])
-		major=`Major ide4 57` || continue
+		major=`Major ide5 57` || continue
 		unit=`suffix $arg hd`
 		base=`index kl $unit`
 		base=`math $base \* 64`
@@ -974,6 +974,50 @@
 			makedev hd$unit$part b $major $(( $base + $part )) $disk
 		done
 		;;
+	hd[m-n])
+		major=`Major ide6 88` || continue
+		unit=`suffix $arg hd`
+		base=`index mn $unit`
+		base=`math $base \* 64`
+		makedev hd$unit b $major $base $disk
+		for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20
+		do
+			makedev hd$unit$part b $major $(( $base + $part )) $disk
+		done
+		;;
+        hd[o-p])
+                major=`Major ide7 89` || continue
+                unit=`suffix $arg hd`
+                base=`index op $unit`
+                base=`math $base \* 64`
+                makedev hd$unit b $major $base $disk
+                for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20
+                do
+                        makedev hd$unit$part b $major $(( $base + $part )) $disk
+                done
+                ;;
+        hd[q-r])
+                major=`Major ide8 90` || continue
+                unit=`suffix $arg hd`
+                base=`index qr $unit`
+                base=`math $base \* 64`
+                makedev hd$unit b $major $base $disk
+                for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20
+                do
+                        makedev hd$unit$part b $major $(( $base + $part )) $disk
+                done
+                ;;
+        hd[s-t])
+                major=`Major ide9 91` || continue
+                unit=`suffix $arg hd`
+                base=`index st $unit`
+                base=`math $base \* 64`
+                makedev hd$unit b $major $base $disk
+                for part in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20
+                do
+                        makedev hd$unit$part b $major $(( $base + $part )) $disk
+                done
+                ;;
 	ht0)
 		major=`Major ht0 37` || continue
 		# Only one IDE tape drive is currently supported; ht0.
