--- ./MAKEDEV.orig	2004-05-20 19:27:42.000000000 -0700
+++ ./MAKEDEV	2004-05-20 19:28:47.000000000 -0700
@@ -17,14 +17,14 @@
     tty="  root tty    0666"
    cons="  root tty    0622"
     vcs="  root root   0600"
-dialout="  root dialout 0660"
+dialout="  root uucp   0660"
   mouse="  root root   0660"
 printer="  root lp     0660"
  floppy="  root floppy 0660"
    disk="  root disk   0660"
    scsi="  root root   0600"
   cdrom="  root cdrom  0660"
-   tape="  root tape   0660"
+   tape="  root disk   0660"
   audio="  root audio  0660"
   video="  root video  0660"
   ibcs2="  root root   0666"
