--- ./conf.d/60-latin.conf.orig	2020-11-27 19:56:42.000000000 -0600
+++ ./conf.d/60-latin.conf	2020-11-29 14:42:35.363035464 -0600
@@ -5,7 +5,10 @@
 	<alias>
 		<family>serif</family>
 		<prefer>
+			<family>Liberation Serif</family>
 			<family>DejaVu Serif</family>
+			<family>Noto Serif</family>
+			<family>Bitstream Vera Serif</family>
 			<family>Times New Roman</family>
 			<family>Thorndale AMT</family>
 			<family>Luxi Serif</family>
@@ -17,7 +20,10 @@
 	<alias>
 		<family>sans-serif</family>
 		<prefer>
+			<family>Liberation Sans</family>
 			<family>DejaVu Sans</family>
+			<family>Noto Sans</family>
+			<family>Bitstream Vera Sans</family>
 			<family>Verdana</family>
 			<family>Arial</family>
 			<family>Albany AMT</family>
@@ -33,7 +39,10 @@
 	<alias>
 		<family>monospace</family>
 		<prefer>
+			<family>Liberation Mono</family>
 			<family>DejaVu Sans Mono</family>
+			<family>Noto Sans Mono</family>
+			<family>Bitstream Vera Sans Mono</family>
 			<family>Inconsolata</family>
 			<family>Andale Mono</family>
 			<family>Courier New</family>
