--- ./conf.d/60-latin.conf.orig	2024-11-27 21:58:06.170032507 -0600
+++ ./conf.d/60-latin.conf	2024-11-27 22:03:54.204040292 -0600
@@ -5,6 +5,7 @@
 	<alias>
 		<family>serif</family>
 		<prefer>
+			<family>Liberation Serif</family>
 			<family>Noto Serif</family>
 			<family>DejaVu Serif</family>
 			<family>Times New Roman</family>
@@ -18,6 +19,7 @@
 	<alias>
 		<family>sans-serif</family>
 		<prefer>
+			<family>Liberation Sans</family>
 			<family>Noto Sans</family>
 			<family>DejaVu Sans</family>
 			<family>Verdana</family>
@@ -36,6 +38,7 @@
 		<family>monospace</family>
 		<prefer>
 			<family>Hack</family>
+			<family>Liberation Mono</family>
 			<family>DejaVu Sans Mono</family>
 			<family>Noto Mono</family>
 			<family>Noto Sans Mono</family>
