--- ./fonts.conf.in.orig	2012-07-23 21:01:32.000000000 -0500
+++ ./fonts.conf.in	2013-02-28 22:32:44.758585955 -0600
@@ -22,13 +22,29 @@
 -->
 
 <!-- Font directory list -->
-
-	<dir>@FC_DEFAULT_FONTS@</dir>
-	@FC_FONTPATH@
+	<dir>/usr/share/fonts/OTF</dir>
+	<dir>/usr/share/fonts/TTF</dir>
+	<dir>/usr/share/fonts/Type1</dir>
+	<dir>/usr/share/fonts/Speedo</dir>
+	<dir>/usr/share/fonts/cyrillic</dir>
+	<dir>/usr/share/fonts/misc</dir>
 	<dir prefix="xdg">fonts</dir>
 	<!-- the following element will be removed in the future -->
 	<dir>~/.fonts</dir>
 
+<!-- Commented out font directories:
+     These are not indexed by default, as fontconfig then seems
+     to prefer bitmapped fonts in some cases...
+	<dir>/usr/share/fonts/100dpi</dir>
+	<dir>/usr/share/fonts/75dpi</dir>
+     The top-level font directory should also not be indexed, as that
+     will include the directories of bitmapped fonts:
+	<dir>@FC_DEFAULT_FONTS@</dir>
+     The empty line below is produced by the FC_FONTPATH macro.
+     Moving it here so it doesn't become a problem in the future:
+	@FC_FONTPATH@
+-->
+
 <!--
   Accept deprecated 'mono' alias, replacing it with 'monospace'
 -->
