diff -Nur fontconfig-2.8.0.orig/fonts.conf.in fontconfig-2.8.0/fonts.conf.in
--- fontconfig-2.8.0.orig/fonts.conf.in	2009-06-28 12:48:47.000000000 -0500
+++ fontconfig-2.8.0/fonts.conf.in	2009-12-01 21:48:48.134935536 -0600
@@ -23,10 +23,21 @@
 
 <!-- Font directory list -->
 
-	<dir>@FC_DEFAULT_FONTS@</dir>
-	@FC_FONTPATH@
+       <dir>/usr/share/fonts/OTF</dir>
+       <dir>/usr/share/fonts/TTF</dir>
+       <dir>/usr/share/fonts/Type1</dir>
+       <dir>/usr/share/fonts/Speedo</dir>
+       <dir>/usr/share/fonts/cyrillic</dir>
+       <dir>/usr/share/fonts/misc</dir>
 	<dir>~/.fonts</dir>
 
+<!--    Commented out font directories:
+These are not indexed by default, as fontconfig then seems
+to prefer bitmapped fonts in some cases...
+       <dir>/usr/share/fonts/100dpi</dir>
+       <dir>/usr/share/fonts/75dpi</dir>
+-->
+
 <!--
   Accept deprecated 'mono' alias, replacing it with 'monospace'
 -->
