--- ./src/meson.build.orig	2025-04-11 01:47:01.000000000 -0500
+++ ./src/meson.build	2025-04-12 13:35:59.361129910 -0500
@@ -80,7 +80,7 @@
   description: 'Font configuration and customization library',
   filebase: 'fontconfig',
   name: 'Fontconfig',
-  requires_private: ['freetype2 ' + freetype_req],
+  requires: ['freetype2 ' + freetype_req],
   version: fc_version,
   variables: [
     'sysconfdir=@0@'.format(join_paths(prefix, get_option('sysconfdir'))),
