diff -ru fontconfig-2.15.0/fonts.conf.in fontconfig-2.15.0.new/fonts.conf.in
--- fontconfig-2.15.0/fonts.conf.in	2023-12-22 14:43:03.000000000 +0100
+++ fontconfig-2.15.0.new/fonts.conf.in	2024-11-22 23:08:01.630099792 +0100
@@ -24,12 +24,28 @@
 
 <!-- Font directory list -->
 
-@FC_DEFAULT_FONTS@
-	@FC_FONTPATH@
+	<dir>/usr/share/fonts/OTF</dir>
+	<dir>/usr/share/fonts/TTF</dir>
+	<dir>/usr/share/fonts/Type1</dir>
+	<dir>/usr/share/fonts/Speedo</dir>
+	<dir>/usr/share/fonts/cyrillic</dir>
+	<dir>/usr/share/fonts/misc</dir>
 	<dir prefix="xdg">fonts</dir>
 	<!-- the following element will be removed in the future -->
 	<dir>~/.fonts</dir>
 
+<!-- Commented out font directories:
+     These are not indexed by default, as fontconfig then seems
+     to prefer bitmapped fonts in some cases...
+	<dir>/usr/share/fonts/100dpi</dir>
+	<dir>/usr/share/fonts/75dpi</dir>
+     The top-level font directory should also not be indexed, as that
+     will include the directories of bitmapped fonts:
+	<dir>/usr/share/fonts</dir>
+     The empty line below is produced by the FC_FONTPATH macro.
+     Moving it here so it doesn't become a problem in the future:
+-->
+
 <!--
   Accept deprecated 'mono' alias, replacing it with 'monospace'
 -->
