diff -Nur fontconfig-2.8.0.orig/conf.d/45-latin.conf fontconfig-2.8.0/conf.d/45-latin.conf
--- fontconfig-2.8.0.orig/conf.d/45-latin.conf	2009-11-16 11:24:52.000000000 -0600
+++ fontconfig-2.8.0/conf.d/45-latin.conf	2009-12-01 21:45:42.639936724 -0600
@@ -10,9 +10,9 @@
   Serif faces
  -->
 	<alias>
-		<family>Bitstream Vera Serif</family>
 		<family>DejaVu Serif</family>
 		<family>Liberation Serif</family>
+		<family>Bitstream Vera Serif</family>
 		<family>Times New Roman</family>
 		<family>Times</family>
 		<family>Nimbus Roman No9 L</family>
@@ -25,9 +25,9 @@
   Sans-serif faces
  -->
 	<alias>
-		<family>Bitstream Vera Sans</family>
 		<family>DejaVu Sans</family>
 		<family>Liberation Sans</family>
+		<family>Bitstream Vera Sans</family>
 		<family>Arial</family>
 		<family>Helvetica</family>
 		<family>Verdana</family>
@@ -41,9 +41,9 @@
   Monospace faces
  -->
  	<alias>
-		<family>Bitstream Vera Sans Mono</family>
 		<family>DejaVu Sans Mono</family>
 		<family>Liberation Mono</family>
+		<family>Bitstream Vera Sans Mono</family>
 		<family>Inconsolata</family>
 		<family>Courier New</family>
 		<family>Courier</family>
diff -Nur fontconfig-2.8.0.orig/conf.d/60-latin.conf fontconfig-2.8.0/conf.d/60-latin.conf
--- fontconfig-2.8.0.orig/conf.d/60-latin.conf	2009-11-16 11:24:52.000000000 -0600
+++ fontconfig-2.8.0/conf.d/60-latin.conf	2009-12-01 21:46:39.925935327 -0600
@@ -4,8 +4,8 @@
 	<alias>
 		<family>serif</family>
 		<prefer>
-			<family>Bitstream Vera Serif</family>
 			<family>DejaVu Serif</family>
+			<family>Bitstream Vera Serif</family>
 			<family>Times New Roman</family>
 			<family>Thorndale AMT</family>
 			<family>Luxi Serif</family>
@@ -16,8 +16,8 @@
 	<alias>
 		<family>sans-serif</family>
 		<prefer>
-			<family>Bitstream Vera Sans</family>
 			<family>DejaVu Sans</family>
+			<family>Bitstream Vera Sans</family>
 			<family>Verdana</family>
 			<family>Arial</family>
 			<family>Albany AMT</family>
@@ -32,8 +32,8 @@
 	<alias>
 		<family>monospace</family>
 		<prefer>
-			<family>Bitstream Vera Sans Mono</family>
 			<family>DejaVu Sans Mono</family>
+			<family>Bitstream Vera Sans Mono</family>
 			<family>Inconsolata</family>
 			<family>Andale Mono</family>
 			<family>Courier New</family>
