--- ./pylib/gyp/common.py.orig	2021-10-10 15:47:45.000000000 -0500
+++ ./pylib/gyp/common.py	2021-10-10 15:49:24.942106994 -0500
@@ -494,7 +494,7 @@
 
 
 # Based on http://code.activestate.com/recipes/576694/.
-class OrderedSet(collections.MutableSet):
+class OrderedSet(collections.abc.MutableSet):
   def __init__(self, iterable=None):
     self.end = end = []
     end += [None, end, end]         # sentinel node for doubly linked list
