commit 7b0a3fd85beb67c990da3fb1b690444a27a6ee42
Author: Tom Hughes <tom@compton.nu>
Date:   Fri Aug 23 09:33:14 2019 +0100

    Fix python 3.8 warnings

diff --git a/pylib/gyp/input.py b/pylib/gyp/input.py
index 42c279cf..013231dc 100644
--- a/pylib/gyp/input.py
+++ b/pylib/gyp/input.py
@@ -1181,7 +1181,7 @@ def LoadVariablesFromVariablesDict(variables, the_dict, the_dict_key):
       if variable_name in variables:
         # If the variable is already set, don't set it.
         continue
-      if the_dict_key is 'variables' and variable_name in the_dict:
+      if the_dict_key == 'variables' and variable_name in the_dict:
         # If the variable is set without a % in the_dict, and the_dict is a
         # variables dict (making |variables| a varaibles sub-dict of a
         # variables dict), use the_dict's definition.
