--- ./dump/inomap.c.orig	2019-11-26 13:32:54.382758726 -0600
+++ ./dump/inomap.c	2019-11-26 13:34:21.686751502 -0600
@@ -79,8 +79,8 @@
 			    bool_t,
 			    bool_t *);
 static void cb_context_free( void );
-static int cb_count_inogrp( void *, int, xfs_inogrp_t *);
-static int cb_add_inogrp( void *, int, xfs_inogrp_t * );
+static int cb_count_inogrp( void *, int, struct xfs_inogrp *);
+static int cb_add_inogrp( void *, int, struct xfs_inogrp * );
 static int cb_add( void *, jdm_fshandle_t *, int, struct xfs_bstat * );
 static bool_t cb_inoinresumerange( xfs_ino_t );
 static bool_t cb_inoresumed( xfs_ino_t );
@@ -488,7 +488,7 @@
 }
 
 static int
-cb_count_inogrp( void *arg1, int fsfd, xfs_inogrp_t *inogrp )
+cb_count_inogrp( void *arg1, int fsfd, struct xfs_inogrp *inogrp )
 {
 	int *count = (int *)arg1;
 	(*count)++;
@@ -1116,7 +1116,7 @@
  * order. adds a new segment to the inomap and ino-to-gen map.
  */
 static int
-cb_add_inogrp( void *arg1, int fsfd, xfs_inogrp_t *inogrp )
+cb_add_inogrp( void *arg1, int fsfd, struct xfs_inogrp *inogrp )
 {
 	hnk_t *hunk;
 	seg_t *segp;
--- ./common/util.h.orig	2019-11-26 13:32:54.397758725 -0600
+++ ./common/util.h	2019-11-26 13:34:21.685751502 -0600
@@ -112,7 +112,7 @@
 extern int inogrp_iter( int fsfd,
 			     int ( * fp )( void *arg1,
 				     		int fsfd,
-						xfs_inogrp_t *inogrp ),
+						struct xfs_inogrp *inogrp ),
 			     void * arg1,
 			     int *statp );
 
--- ./common/util.c.orig	2019-11-26 13:32:54.415758723 -0600
+++ ./common/util.c	2019-11-26 13:34:21.685751502 -0600
@@ -274,23 +274,23 @@
 inogrp_iter( int fsfd,
 	     int ( * fp )( void *arg1,
 				int fsfd,
-				xfs_inogrp_t *inogrp ),
+				struct xfs_inogrp *inogrp ),
 	     void * arg1,
 	     int *statp )
 {
 	xfs_ino_t lastino;
 	int inogrpcnt;
-	xfs_inogrp_t *igrp;
+	struct xfs_inogrp *igrp;
         xfs_fsop_bulkreq_t bulkreq;
 
 	/* stat set with return from callback func */
 	*statp = 0;
 
-	igrp = malloc(INOGRPLEN * sizeof(xfs_inogrp_t));
+	igrp = malloc(INOGRPLEN * sizeof(struct xfs_inogrp));
 	if (!igrp) {
 		mlog(MLOG_NORMAL | MLOG_ERROR,
 		     _("malloc of stream context failed (%d bytes): %s\n"),
-		     INOGRPLEN * sizeof(xfs_inogrp_t),
+		     INOGRPLEN * sizeof(struct xfs_inogrp),
 		     strerror(errno));
 		return -1;
 	}
@@ -302,7 +302,7 @@
 	bulkreq.ubuffer = igrp;
 	bulkreq.ocount = &inogrpcnt;
 	while (!ioctl(fsfd, XFS_IOC_FSINUMBERS, &bulkreq)) {
-		xfs_inogrp_t *p, *endp;
+		struct xfs_inogrp *p, *endp;
 
 		if ( inogrpcnt == 0 ) {
 			free(igrp);
