--- ./dump/content.c.orig	2019-11-26 13:37:51.425734146 -0600
+++ ./dump/content.c	2019-11-26 13:38:52.482729094 -0600
@@ -1390,7 +1390,7 @@
 		stat64_t rootstat;
 		xfs_ino_t lastino = 0;
 		int ocount = 0;
-		xfs_fsop_bulkreq_t bulkreq;
+		struct xfs_fsop_bulkreq bulkreq;
 
 		/* Get the inode of the mount point */
 		rval = fstat64( sc_fsfd, &rootstat );
@@ -2812,7 +2812,7 @@
 {
 	xfs_ino_t lastino;
 	size_t bulkstatcallcnt;
-        xfs_fsop_bulkreq_t bulkreq;
+        struct xfs_fsop_bulkreq bulkreq;
 
 	inomap_reset_context(inomap_contextp);
 
--- ./common/util.c.orig	2019-11-26 13:37:51.523734138 -0600
+++ ./common/util.c	2019-11-26 13:38:52.481729094 -0600
@@ -132,7 +132,7 @@
 	xfs_ino_t lastino;
 	int saved_errno;
 	int bulkstatcnt;
-        xfs_fsop_bulkreq_t bulkreq;
+        struct xfs_fsop_bulkreq bulkreq;
 
 	/* stat set with return from callback func
 	 */
@@ -256,7 +256,7 @@
 	     xfs_ino_t ino,
 	     struct xfs_bstat *statp )
 {
-        xfs_fsop_bulkreq_t bulkreq;
+        struct xfs_fsop_bulkreq bulkreq;
 	int count = 0;
 
 	assert( ino > 0 );
@@ -281,7 +281,7 @@
 	xfs_ino_t lastino;
 	int inogrpcnt;
 	struct xfs_inogrp *igrp;
-        xfs_fsop_bulkreq_t bulkreq;
+        struct xfs_fsop_bulkreq bulkreq;
 
 	/* stat set with return from callback func */
 	*statp = 0;
