diff -u -r scim-anthy-1.2.4.orig/src/scim_anthy_kana.cpp scim-anthy-1.2.4/src/scim_anthy_kana.cpp
--- scim-anthy-1.2.4.orig/src/scim_anthy_kana.cpp	2006-07-13 02:41:51.000000000 -0500
+++ scim-anthy-1.2.4/src/scim_anthy_kana.cpp	2009-06-09 20:40:41.000000000 -0500
@@ -23,6 +23,8 @@
 #include "scim_anthy_default_tables.h"
 #include "scim_anthy_utils.h"
 
+#include <cstring>
+
 using namespace scim_anthy;
 
 static bool
diff -u -r scim-anthy-1.2.4.orig/src/scim_anthy_preedit.cpp scim-anthy-1.2.4/src/scim_anthy_preedit.cpp
--- scim-anthy-1.2.4.orig/src/scim_anthy_preedit.cpp	2006-07-23 20:56:00.000000000 -0500
+++ scim-anthy-1.2.4/src/scim_anthy_preedit.cpp	2009-06-09 20:41:03.000000000 -0500
@@ -26,6 +26,8 @@
 #include "scim_anthy_preedit.h"
 #include "scim_anthy_utils.h"
 
+#include <cstring>
+
 using namespace scim_anthy;
 
 static ConvRule *get_period_rule              (TypingMethod method,
diff -u -r scim-anthy-1.2.4.orig/src/scim_anthy_prefs.cpp scim-anthy-1.2.4/src/scim_anthy_prefs.cpp
--- scim-anthy-1.2.4.orig/src/scim_anthy_prefs.cpp	2006-07-23 20:56:00.000000000 -0500
+++ scim-anthy-1.2.4/src/scim_anthy_prefs.cpp	2009-06-09 20:41:40.000000000 -0500
@@ -23,6 +23,8 @@
 #include "scim_anthy_prefs.h"
 #include "scim_anthy_intl.h"
 
+#include <cstring>
+
 namespace scim_anthy {
 
 BoolConfigData config_bool_common [] =
diff -u -r scim-anthy-1.2.4.orig/src/scim_anthy_setup.cpp scim-anthy-1.2.4/src/scim_anthy_setup.cpp
--- scim-anthy-1.2.4.orig/src/scim_anthy_setup.cpp	2007-05-21 03:22:20.000000000 -0500
+++ scim-anthy-1.2.4/src/scim_anthy_setup.cpp	2009-06-09 20:42:01.000000000 -0500
@@ -30,6 +30,8 @@
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
 
+#include <cstring>
+
 #define Uses_SCIM_CONFIG_BASE
 #define Uses_SCIM_EVENT
 #define SCIM_ANTHY_USE_GTK
diff -u -r scim-anthy-1.2.4.orig/src/scim_anthy_utils.cpp scim-anthy-1.2.4/src/scim_anthy_utils.cpp
--- scim-anthy-1.2.4.orig/src/scim_anthy_utils.cpp	2007-03-28 02:42:10.000000000 -0500
+++ scim-anthy-1.2.4/src/scim_anthy_utils.cpp	2009-06-09 20:41:22.000000000 -0500
@@ -24,6 +24,8 @@
 #include "scim_anthy_utils.h"
 #include "scim_anthy_default_tables.h"
 
+#include <cstring>
+
 namespace scim_anthy {
 
 bool
