diff --git a/date.c b/date.c
index 11c5e5f..c23d7b5 100644
--- a/date.c
+++ b/date.c
@@ -42,7 +42,7 @@ static void		display(const char *, time_t);
 static void		dogmt(void);
 static void		errensure(void);
 static void		timeout(FILE *, const char *, const struct tm *);
-static ATTRIBUTE_NORETURN void usage(void);
+static void usage(void);
 
 int
 main(const int argc, char *argv[])
diff --git a/zdump.c b/zdump.c
index 7acb3e2..71524bf 100644
--- a/zdump.c
+++ b/zdump.c
@@ -125,7 +125,7 @@ is_alpha(char a)
 	}
 }
 
-static ATTRIBUTE_NORETURN void
+static void
 size_overflow(void)
 {
   fprintf(stderr, _("%s: size overflow\n"), progname);
diff --git a/zic.c b/zic.c
index 892414a..6f64bc9 100644
--- a/zic.c
+++ b/zic.c
@@ -459,14 +459,14 @@ static char		roll[TZ_MAX_LEAPS];
 ** Memory allocation.
 */
 
-static ATTRIBUTE_NORETURN void
+static void
 memory_exhausted(const char *msg)
 {
 	fprintf(stderr, _("%s: Memory exhausted: %s\n"), progname, msg);
 	exit(EXIT_FAILURE);
 }
 
-static ATTRIBUTE_NORETURN void
+static void
 size_overflow(void)
 {
   memory_exhausted(_("size overflow"));
@@ -666,7 +666,7 @@ close_file(FILE *stream, char const *dir, char const *name,
   }
 }
 
-static ATTRIBUTE_NORETURN void
+static void
 usage(FILE *stream, int status)
 {
   fprintf(stream,
@@ -3725,7 +3725,7 @@ getfields(char *cp, char **array, int arrayelts)
 	return nsubs;
 }
 
-static ATTRIBUTE_NORETURN void
+static void
 time_overflow(void)
 {
   error(_("time overflow"));
