diff -ur a/bash_completion b/bash_completion
--- a/bash_completion	2006-03-01 16:20:18.000000000 +0000
+++ b/bash_completion	2009-07-15 22:46:27.000000000 +0100
@@ -2852,8 +2852,8 @@
 		return 0
 		;;
 	+([^IZzjy])f)
-		ext='t@(ar?(.@(Z|gz|bz?(2)))|gz|bz?(2))'
-		regex='t\(ar\(\.\(Z\|gz\|bz2\?\)\)\?\|gz\|bz2\?\)'
+		ext='t@(ar?(.@(Z|xz|gz|bz?(2)))|xz|gz|bz?(2))'
+		regex='t\(ar\(\.\(Z\|xz\|gz\|bz2\?\)\)\?\|xz\|gz\|bz2\?\)'
 		;;
 	*[Zz]*f)
 		ext='t?(ar.)@(gz|Z)'
@@ -5519,7 +5519,7 @@
 	COMPREPLY=( $( (cd /var/log/packages; compgen -f -- "$cur") ) )
 } &&
 complete -F _removepkg $filenames removepkg &&
-	complete $dirnames -f -X '!*.tgz' installpkg upgradepkg explodepkg
+	complete $dirnames -f -X '!*.t?z' installpkg upgradepkg explodepkg
 
 # look(1) completion
 #
