diff -Nur bash-completion-1.1.orig//contrib/pkgtools bash-completion-1.1//contrib/pkgtools
--- bash-completion-1.1.orig//contrib/pkgtools	2009-10-09 07:49:31.000000000 -0500
+++ bash-completion-1.1//contrib/pkgtools	2010-03-06 21:56:50.343715164 -0600
@@ -11,7 +11,7 @@
     COMPREPLY=( $( (cd /var/log/packages; compgen -f -- "$cur") ) )
 } &&
 complete -F _removepkg $filenames removepkg &&
-    complete $dirnames -f -X '!*.tgz' installpkg upgradepkg explodepkg
+    complete $dirnames -f -X '!*.t[gx]z' installpkg upgradepkg explodepkg
 
 # Local variables:
 # mode: shell-script
