--- ./src/daemon/XorgDisplayServer.cpp.orig	2020-11-03 03:49:10.000000000 -0600
+++ ./src/daemon/XorgDisplayServer.cpp	2021-08-29 11:36:20.356811948 -0500
@@ -65,7 +65,7 @@
         // create a random hexadecimal number
         const char *digits = "0123456789abcdef";
         for (int i = 0; i < 32; ++i)
-            m_cookie[i] = digits[dis(gen)];
+            m_cookie[i] = QLatin1Char(digits[dis(gen)]);
     }
 
     XorgDisplayServer::~XorgDisplayServer() {
