--- ./src/greeter/SessionModel.cpp.orig	2020-11-03 01:41:06.000000000 -0600
+++ ./src/greeter/SessionModel.cpp	2021-04-26 12:54:48.558769977 -0500
@@ -41,8 +41,8 @@
     SessionModel::SessionModel(QObject *parent) : QAbstractListModel(parent), d(new SessionModelPrivate()) {
         // initial population
         beginResetModel();
-        populate(Session::WaylandSession, mainConfig.Wayland.SessionDir.get());
         populate(Session::X11Session, mainConfig.X11.SessionDir.get());
+        populate(Session::WaylandSession, mainConfig.Wayland.SessionDir.get());
         endResetModel();
 
         // refresh everytime a file is changed, added or removed
