--- ./src/bootstrap/dist.rs.orig	2019-11-07 18:12:33.835944390 -0600
+++ ./src/bootstrap/dist.rs	2019-11-07 22:05:17.991788876 -0600
@@ -676,7 +676,6 @@
         }
 
         builder.ensure(compile::Std { compiler, target });
-        builder.ensure(compile::Rustc { compiler, target });
 
         let image = tmpdir(builder).join(format!("{}-{}-image", name, target));
         let _ = fs::remove_dir_all(&image);
@@ -691,12 +690,12 @@
                 builder.copy(&path, &dst.join(path.file_name().unwrap()));
             }
         }
-        let stamp = dbg!(compile::librustc_stamp(builder, compiler_to_use, target));
-        for (path, host) in builder.read_stamp_file(&stamp) {
-            if !host {
-                builder.copy(&path, &dst.join(path.file_name().unwrap()));
-            }
-        }
+        // let stamp = dbg!(compile::librustc_stamp(builder, compiler_to_use, target));
+        // for (path, host) in builder.read_stamp_file(&stamp) {
+        //    if !host {
+        //        builder.copy(&path, &dst.join(path.file_name().unwrap()));
+        //    }
+        // }
 
         let mut cmd = rust_installer(builder);
         cmd.arg("generate")
