From 841558d3bdc9c9510b841c438838196378741021 Mon Sep 17 00:00:00 2001
From: Nikita Popov <nikita.ppv@gmail.com>
Date: Tue, 4 Feb 2020 19:23:45 +0100
Subject: [PATCH] Remove trailing newline from llvm-config output

---
 src/bootstrap/test.rs | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/bootstrap/test.rs b/src/bootstrap/test.rs
index aa1d1b7c42413..b52fbe4666eb3 100644
--- a/src/bootstrap/test.rs
+++ b/src/bootstrap/test.rs
@@ -1141,6 +1141,8 @@ impl Step for Compiletest {
             let llvm_config = builder.ensure(native::Llvm { target: builder.config.build });
             if !builder.config.dry_run {
                 let llvm_version = output(Command::new(&llvm_config).arg("--version"));
+                // Remove trailing newline from llvm-config output.
+                let llvm_version = llvm_version.trim_end();
                 cmd.arg("--llvm-version").arg(llvm_version);
             }
             if !builder.is_rust_llvm(target) {

