--- ./src/efibootmgr.c.orig	2018-06-10 15:12:10.000000000 -0500
+++ ./src/efibootmgr.c	2018-12-31 13:23:27.284013090 -0600
@@ -1536,9 +1536,6 @@
 					       "invalid numeric value %s\n",
 					       optarg);
 			}
-                        /* XXX efivar-36 accidentally doesn't have a public
-                         * header for this */
-			extern int efi_set_verbose(int verbosity, FILE *errlog);
 			efi_set_verbose(opts.verbose - 2, stderr);
 			break;
 		case 'V':
