From f862d860103511ae953cc2dd1283e1e95bf00376 Mon Sep 17 00:00:00 2001
From: Sly_tom_cat <slytomcat@users.noreply.github.com>
Date: Thu, 24 Oct 2019 22:23:49 +0300
Subject: [PATCH] Remove wrong check of opts.reconnect

---
 src/efibootmgr.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/efibootmgr.c b/src/efibootmgr.c
index 59ba871..f9d01bb 100644
--- a/src/efibootmgr.c
+++ b/src/efibootmgr.c
@@ -1738,9 +1738,6 @@ main(int argc, char **argv)
 			mode = driver;
 	}
 
-	if (opts.reconnect && !opts.driver)
-		errorx(30, "--reconnect is supported only for driver entries.");
-
 	if (!efi_variables_supported())
 		errorx(2, "EFI variables are not supported on this system.");
