--- ./Source/JavaScriptCore/runtime/JSObject.cpp.orig	2014-09-24 06:42:05.000000000 -0500
+++ ./Source/JavaScriptCore/runtime/JSObject.cpp	2015-11-18 22:41:05.673463626 -0600
@@ -1922,6 +1922,10 @@
     }
 }
 
+template void JSObject::putByIndexBeyondVectorLengthWithoutAttributes<Int32Shape>(ExecState* exec, unsigned i, JSValue value);
+template void JSObject::putByIndexBeyondVectorLengthWithoutAttributes<DoubleShape>(ExecState* exec, unsigned i, JSValue value);
+template void JSObject::putByIndexBeyondVectorLengthWithoutAttributes<ContiguousShape>(ExecState* exec, unsigned i, JSValue value);
+
 void JSObject::putByIndexBeyondVectorLengthWithArrayStorage(ExecState* exec, unsigned i, JSValue value, bool shouldThrow, ArrayStorage* storage)
 {
     JSGlobalData& globalData = exec->globalData();
