--- ./src/gui/image/qgifhandler.cpp.orig	2014-04-10 13:37:12.000000000 -0500
+++ ./src/gui/image/qgifhandler.cpp	2014-04-25 13:42:48.468692269 -0500
@@ -358,6 +358,13 @@
                     bits = image->bits();
                     memset(bits, 0, image->byteCount());
                 }
+                
+		// Check if the previous attempt to create the image failed. If it
+		// did then the image is broken and we should give up.
+		if (image->isNull()) {
+		    state = Error;
+		    return -1;
+		}
 
                 disposePrevious(image);
                 disposed = false;
