--- ./calgebra/CMakeLists.txt.orig	2021-08-04 16:49:56.000000000 -0500
+++ ./calgebra/CMakeLists.txt	2021-08-12 22:13:27.013314656 -0500
@@ -13,7 +13,7 @@
 endif()
 
 add_executable(calgebra main.cpp)
-target_link_libraries(calgebra Qt::Core ${READLINE_LIBRARY} KF5::Analitza)
+target_link_libraries(calgebra Qt::Core ${READLINE_LIBRARY} ${CURSES_LIBRARIES} KF5::Analitza)
 
 install(TARGETS calgebra ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
 
