--- kalgebra-14.12.1.orig/calgebra/CMakeLists.txt	2014-11-13 02:59:40.000000000 +0100
+++ kalgebra-14.12.1/calgebra/CMakeLists.txt	2015-01-14 14:35:54.702166188 +0100
@@ -13,7 +13,7 @@
 endif()
 
 add_executable(calgebra main.cpp)
-target_link_libraries(calgebra Qt5::Core ${READLINE_LIBRARY} KF5::Analitza)
+target_link_libraries(calgebra Qt5::Core ${READLINE_LIBRARY} ${CURSES_LIBRARIES} KF5::Analitza)
 
 install(TARGETS calgebra ${INSTALL_TARGETS_DEFAULT_ARGS})
 
