--- ./fig2dev/Imakefile.orig	2002-12-19 09:33:44.000000000 -0800
+++ ./fig2dev/Imakefile	2003-03-03 19:58:09.000000000 -0800
@@ -20,7 +20,7 @@
 XCOMM You should point XFIGLIBDIR to the same directory you did when you compiled
 XCOMM and installed xfig.
 
-XFIGLIBDIR =	/usr/local/lib/X11/xfig
+XFIGLIBDIR =	/usr/lib/X11/xfig
 
 XCOMM ******
 XCOMM If your system has the strerror() function (doesn't have sys_errlist) then
@@ -73,7 +73,7 @@
 XCOMM Change XPMINC if necessary to point to the include file for xpm (xpm.h)
 XCOMM
 
-XCOMM #define USEXPM
+#define USEXPM
 
 #ifdef USEXPM
 XPMLIBDIR = $(USRLIBDIR)
@@ -126,7 +126,7 @@
 
 #ifdef I18N
 I18N_DEFS = -DI18N 
-FIG2DEV_LIBDIR = /usr/local/lib/fig2dev
+FIG2DEV_LIBDIR = /usr/lib/X11/fig2dev
 I18N_DEV_DEFS = $(I18N_DEFS) -DFIG2DEV_LIBDIR=\\\"$(FIG2DEV_LIBDIR)\\\"
 
 install::
