--- ./fig2dev/Imakefile.orig	2010-06-02 14:18:16.000000000 -0500
+++ ./fig2dev/Imakefile	2013-09-18 21:09:04.657010139 -0500
@@ -34,7 +34,7 @@
 XCOMM You should point XFIGLIBDIR to the same directory you did when you compiled
 XCOMM and installed xfig.
 
-XFIGLIBDIR =	/usr/local/lib/X11/xfig
+XFIGLIBDIR =	/usr/lib/X11/xfig
 
 XCOMM ******
 XCOMM If your system doesn't have the strerror() function (has sys_errlist) then
@@ -141,7 +141,7 @@
 
 #ifdef I18N
 I18N_DEFS = -DI18N 
-FIG2DEV_LIBDIR = /usr/local/lib/fig2dev
+FIG2DEV_LIBDIR = /usr/lib/X11/fig2dev
 I18N_DEV_DEFS = $(I18N_DEFS) -DFIG2DEV_LIBDIR=$(FIG2DEV_LIBDIR) -DFIG2DEV_LIBDIR_STR=\\\"$(FIG2DEV_LIBDIR)\\\"
 
 #endif /* I18N */
