--- ./fig2dev/dev/readpng.c.orig	2002-12-18 16:40:35.000000000 -0600
+++ ./fig2dev/dev/readpng.c	2010-02-19 19:25:54.000000000 -0600
@@ -127,7 +127,7 @@
 
 	if (png_get_PLTE(png_ptr, info_ptr, &palette, &num_palette)) {
 	    png_get_hIST(png_ptr, info_ptr, &histogram);
-	    png_set_dither(png_ptr, palette, num_palette, 256, histogram, 0);
+	    /* png_set_dither(png_ptr, palette, num_palette, 256, histogram, 0); */
 	}
     }
     if (color_type == PNG_COLOR_TYPE_GRAY || color_type == PNG_COLOR_TYPE_GRAY_ALPHA) {
