--- klinkstatus/src/tidy/tidyx.h.orig	2009-09-03 09:55:38.886267704 -0500
+++ klinkstatus/src/tidy/tidyx.h	2009-09-03 09:57:29.032269520 -0500
@@ -48,8 +48,8 @@
   Created 2002-07-11 by Charles Reitzel
 */
 
-#include <tidy.h>
-#include <buffio.h>
+#include "/tmp/package-kdewebdev/usr/include/tidy/tidy.h"
+#include "/tmp/package-kdewebdev/usr/include/tidy/buffio.h"
 
 #include <config-tidy.h>
     
