--- cmake/modules/FindLibTidy.cmake.old	2008-07-25 21:47:38.156473750 +0200
+++ cmake/modules/FindLibTidy.cmake	2008-07-25 21:47:50.709258250 +0200
@@ -17,7 +17,7 @@
   set(LibTidy_FIND_QUIETLY TRUE)
 endif (LIBTIDY_INCLUDE_DIR)
 
-FIND_PATH(LIBTIDY_INCLUDE_DIR tidy.h)
+FIND_PATH(LIBTIDY_INCLUDE_DIR tidy/tidy.h tidy.h)
 
 
 FIND_LIBRARY(LIBTIDY_LIBRARIES NAMES tidy)
--- klinkstatus/src/tidy/tidyx.h.old	2008-07-25 21:48:23.151285750 +0200
+++ klinkstatus/src/tidy/tidyx.h	2008-07-25 21:48:36.464117750 +0200
@@ -48,8 +48,8 @@
   Created 2002-07-11 by Charles Reitzel
 */
 
-#include <tidy.h>
-#include <buffio.h>
+#include <tidy/tidy.h>
+#include <tidy/buffio.h>
 
 #include <config-tidy.h>
     
