From 6bdd92202f55d7c530dcbeb2a243604dac546cf1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ville=20Skytt=C3=A4?= <ville.skytta@iki.fi>
Date: Tue, 15 Mar 2016 19:49:55 +0200
Subject: [PATCH 1/5] tar: Don't write to /tmp/jetel

---
 completions/tar | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/completions/tar b/completions/tar
index ef18ad6..6e4c74c 100644
--- a/completions/tar
+++ b/completions/tar
@@ -411,7 +411,7 @@ __tar_try_list_archive()
     shift
 
     read tarball <<<"$(printf -- '%s\n' "$@" \
-        | command sed -n "/^.\{1,\}$regex\$/p" | tee /tmp/jetel)"
+        | command sed -n "/^.\{1,\}$regex\$/p")"
     if [[ -n "$tarball" ]]; then
         local IFS=$'\n'
         COMPREPLY=($(compgen -o filenames -W "$(
-- 
2.7.2

