From 807f903f86ad647425c83f7ad6fc499014c6734e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ville=20Skytt=C3=A4?= <ville.skytta@iki.fi>
Date: Tue, 15 Mar 2016 20:19:10 +0200
Subject: [PATCH 2/5] test suite: Add tar xvf filename test case

---
 test/lib/completions/tar.exp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/test/lib/completions/tar.exp b/test/lib/completions/tar.exp
index 70e38d0..7cdef94 100644
--- a/test/lib/completions/tar.exp
+++ b/test/lib/completions/tar.exp
@@ -59,6 +59,10 @@ set test "old option: create and members"
 assert_complete_dir "dir/ dir2/ archive.tar.xz escape.tar" "tar cTfvv NOT_EXISTS DONT_CREATE.tar " $::srcdir/fixtures/tar
 sync_after_int
 
+set test "old option: extract and archive"
+assert_complete_dir "dir/ dir2/ archive.tar.xz escape.tar" "tar xvf " $::srcdir/fixtures/tar
+sync_after_int
+
 if { "$tar_version" == "gnu" } {
     set test "check short options"
     assert_complete_any "tar -c"
-- 
2.7.2

