From 7ab05bf9d97ce5a6ced7acdfc21235e06542c3f2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ville=20Skytt=C3=A4?= <ville.skytta@iki.fi>
Date: Wed, 16 Mar 2016 12:39:06 +0200
Subject: [PATCH 4/5] tar: Remove unused variable

---
 completions/tar | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/completions/tar b/completions/tar
index 0a5f588..9c44dc7 100644
--- a/completions/tar
+++ b/completions/tar
@@ -696,7 +696,7 @@ _posix_tar()
 
 _tar()
 {
-    local cmd=${COMP_WORDS[0]} output line
+    local cmd=${COMP_WORDS[0]} line
     read line <<<"$($cmd --version)"
     case "$line" in
     *GNU*)
-- 
2.7.2

