--- ./completions/Makefile.in.orig	2016-03-03 06:12:35.000000000 -0600
+++ ./completions/Makefile.in	2016-05-31 14:01:42.323415707 -0500
@@ -1236,7 +1236,7 @@
 		rm -f $(targetdir)/$$file && \
 			$(LN_S) mcrypt $(targetdir)/$$file ; \
 	done
-	for file in mplayer2 mencoder gmplayer kplayer mpv ; do \
+	for file in mplayer2 mencoder gmplayer kplayer ; do \
 		rm -f $(targetdir)/$$file && \
 			$(LN_S) mplayer $(targetdir)/$$file ; \
 	done
--- ./completions/Makefile.am.orig	2016-03-02 13:42:58.000000000 -0600
+++ ./completions/Makefile.am	2016-05-31 14:01:33.554416246 -0500
@@ -802,7 +802,7 @@
 		rm -f $(targetdir)/$$file && \
 			$(LN_S) mcrypt $(targetdir)/$$file ; \
 	done
-	for file in mplayer2 mencoder gmplayer kplayer mpv ; do \
+	for file in mplayer2 mencoder gmplayer kplayer ; do \
 		rm -f $(targetdir)/$$file && \
 			$(LN_S) mplayer $(targetdir)/$$file ; \
 	done
