--- ./Makefile.orig	2020-12-23 14:15:36.000000000 -0600
+++ ./Makefile	2020-12-28 12:15:35.339015286 -0600
@@ -85,10 +85,10 @@
 all: $(PSF) $(PCF)
 
 DESTDIR =
-prefix  = /usr/local
-psfdir  = $(prefix)/share/consolefonts
-x11dir  = $(prefix)/share/fonts/terminus
-otbdir  = $(prefix)/share/fonts/terminus
+prefix  = /usr
+psfdir  = $(prefix)/share/kbd/consolefonts
+x11dir  = $(prefix)/share/fonts/misc
+otbdir  = $(prefix)/share/fonts/misc
 
 install: $(PSF) $(PCF)
 	mkdir -p $(DESTDIR)$(psfdir)
