--- ./Makefile.orig	2011-07-12 11:13:12.000000000 -0500
+++ ./Makefile	2013-02-19 14:37:14.604914231 -0600
@@ -81,9 +81,9 @@
 n12: $(PCF_N12)
 
 DESTDIR	=
-prefix	= /usr/local
-psfdir	= $(prefix)/share/consolefonts
-x11dir	= $(prefix)/share/fonts/terminus
+prefix	= /usr
+psfdir	= $(prefix)/share/kbd/consolefonts
+x11dir	= $(prefix)/share/fonts/misc
 
 install: $(PSF) $(PCF)
 	mkdir -p $(DESTDIR)$(psfdir)
