--- ./netconfig.orig	2017-11-26 13:34:37.652627697 -0600
+++ ./netconfig	2018-02-15 20:01:26.844174383 -0600
@@ -362,10 +362,6 @@
 
 # Write the hostname with domain to /etc/HOSTNAME:
 echo $HOSTNM.$DOMAIN > etc/HOSTNAME
-# Also make sure the hostname is written to /etc/NetworkManager/NetworkManager.conf:
-if [ -w etc/NetworkManager/NetworkManager.conf ]; then
-  sed -i "s/^hostname=.*$/hostname=$HOSTNM/g" etc/NetworkManager/NetworkManager.conf
-fi
 
 dialog --title "CONFIGURATION TYPE FOR '$HOSTNM.$DOMAIN'" \
 --default-item DHCP \
