--- ./rc.inet1.orig	2017-11-26 22:48:14.650274588 -0600
+++ ./rc.inet1	2018-02-15 20:42:45.731997047 -0600
@@ -215,8 +215,8 @@
       else # bring up interface using a static IP address
         if [ -n "${IPADDR[$i]}" ]; then # skip unconfigured interfaces
           # Set up the network card:
-          echo "/etc/rc.d/rc.inet1:  /sbin/ip address add ${IPADDR[$i]}/${NETMASK[$i]} dev ${1}" | $LOGGER
-          /sbin/ip address add ${IPADDR[$i]}/${NETMASK[$i]} dev ${1}
+          echo "/etc/rc.d/rc.inet1:  /sbin/ip address add ${IPADDR[$i]}/${NETMASK[$i]} broadcast + dev ${1}" | $LOGGER
+          /sbin/ip address add ${IPADDR[$i]}/${NETMASK[$i]} broadcast + dev ${1}
           if /sbin/ip link show dev ${1} | grep -wq "state DOWN" ; then
             /sbin/ip link set dev ${1} up  # Bring up interface
           fi
@@ -268,6 +268,8 @@
     if [ -n "${BRNICS[$i]}" ]; then
       br_close $i
     fi
+    # Flush the address from the interface:
+    ip address flush dev ${1}
   fi
 }
 
