--- ./netconfig.orig	2018-02-21 21:21:31.000000000 -0600
+++ ./netconfig	2018-04-12 14:07:10.137023035 -0500
@@ -11,7 +11,7 @@
 #
 
 # If we're doing loopback, we don't want 127.0.0.1 and 255.255.255.0 in
-# rc.inet1.conf... it's better to leave the values null.  However, since
+# rc.inet1.conf... it's better to leave the values null. However, since
 # we use the IP in other places, we need to make a copy for here.
 RCIPCOPY=$IPADDR
 RCMASKCOPY=$NETMASK
@@ -68,7 +68,7 @@
 # Default gateway IP address:
 GATEWAY="$GATEWAY"
 
-# Change this to "yes" for debugging output to stdout.  Unfortunately,
+# Change this to "yes" for debugging output to stdout. Unfortunately,
 # /sbin/hotplug seems to disable stdout so you'll only see debugging output
 # when rc.inet1 is called directly.
 DEBUG_ETH_UP="no"
@@ -103,8 +103,8 @@
 #VIRTIFUSER[1]="someuser"
 #VIRTIFGROUP[1]="somegroup"
 
-## Example config information for wlan0.  Uncomment the lines you need and fill
-## in your data.  (You may not need all of these for your wireless network)
+## Example config information for wlan0. Uncomment the lines you need and fill
+## in your data. (You may not need all of these for your wireless network)
 #IFNAME[4]="wlan0"
 #IPADDR[4]=""
 #NETMASK[4]=""
@@ -173,7 +173,7 @@
 /bin/cat <<EOF >$ETCNETWORKS
 #
 # networks	This file describes a number of netname-to-address
-#		mappings for the TCP/IP subsystem.  It is mostly
+#		mappings for the TCP/IP subsystem. It is mostly
 #		used at boot time, when no name servers are running.
 #
 
@@ -192,14 +192,14 @@
 /bin/cat << EOF > $HOSTS
 #
 # hosts		This file describes a number of hostname-to-address
-#		mappings for the TCP/IP subsystem.  It is mostly
+#		mappings for the TCP/IP subsystem. It is mostly
 #		used at boot time, when no name servers are running.
 #		On small systems, this file can be used instead of a
 #		"named" name server.  Just add the names, addresses
 #		and any aliases to this file...
 #
 # By the way, Arnt Gulbrandsen <agulbra@nvg.unit.no> says that 127.0.0.1
-# should NEVER be named with the name of the machine.  It causes problems
+# should NEVER be named with the name of the machine. It causes problems
 # for some (stupid) programs, irc and reputedly talk. :^)
 #
 
@@ -362,16 +362,21 @@
 
 # Write the hostname with domain to /etc/HOSTNAME:
 echo $HOSTNM.$DOMAIN > etc/HOSTNAME
+## OBSOLETE with recent NetworkManager:
+## Also make sure the hostname is written to /etc/NetworkManager/NetworkManager.conf:
+#if [ -w etc/NetworkManager/NetworkManager.conf ]; then
+#  sed -i "s/^hostname=.*$/hostname=$HOSTNM/g" etc/NetworkManager/NetworkManager.conf
+#fi
 
 dialog --title "CONFIGURATION TYPE FOR '$HOSTNM.$DOMAIN'" \
---default-item DHCP \
+--default-item NetworkManager \
 --menu \
 "Now we need to know how your machine connects to the network.\n\
 If you have an internal network card and an assigned IP address, gateway, \
-and DNS, use the 'static IP' choice to enter these values.  If your IP \
+and DNS, use the 'static IP' choice to enter these values. If your IP \
 address is assigned by a DHCP server (commonly used by cable modem and DSL \
-services), select 'DHCP'.  If you do not have a network card, select \
-the 'loopback' choice.  You may also select 'NetworkManager' if you would \
+services), select 'DHCP'. If you do not have a network card, select \
+the 'loopback' choice. You may also select 'NetworkManager' if you would \
 like to have the NetworkManager daemon automatically handle your wired and \
 wireless network interfaces (this is simple and usually works). \
 Which type of network setup would you like?"  20 70 4 \
@@ -389,10 +394,10 @@
 if [ "$REPLY" = "DHCP" ]; then
   USE_DHCP="yes"
   dialog --title "SET DHCP HOSTNAME"  --inputbox "Some network providers require \
-that the DHCP hostname be set in order to connect.  If so, they'll have assigned \
+that the DHCP hostname be set in order to connect. If so, they'll have assigned \
 a hostname to your machine, which may look something like CC-NUMBER-A (this \
-depends on your ISP).  If you were assigned a DHCP hostname, please enter it \
-below.  If you do not have a DHCP hostname, just hit ENTER or Cancel." 13 62 \
+depends on your ISP). If you were assigned a DHCP hostname, please enter it \
+below. If you do not have a DHCP hostname, just hit ENTER or Cancel." 13 62 \
 2> $TMP/SeTDHCPHOST
   NEW_DHCPHOST="`cat $TMP/SeTDHCPHOST`"
   rm -f $TMP/SeTDHCPHOST
@@ -442,7 +447,7 @@
    NETMASK=`cat $TMP/SeTnetmask`
   fi
   cat << EOF > $TMP/tempmsg
-Enter your netmask.  This will generally look something
+Enter your netmask. This will generally look something
 like this: 255.255.255.0
 Enter netmask (aaa.bbb.ccc.ddd):
 EOF
@@ -556,8 +561,8 @@
   dialog --title "PROBE FOR NETWORK CARD?" --menu "If you like, we \
 can look to see what kind of network card you have in your machine, and \
 if we find one create an /etc/rc.d/rc.netdevice script to load the module \
-for it at boot time.  There's a slight bit of danger that the probing \
-can cause problems, but it almost always works.  If you'd rather configure \
+for it at boot time. There's a slight bit of danger that the probing \
+can cause problems, but it almost always works. If you'd rather configure \
 your system manually, you can skip the probing process and edit \
 /etc/rc.d/rc.modules or /etc/modules.conf later to have it load the right module." \
 16 68 2 \
@@ -601,7 +606,7 @@
      fi
      if [ ! -r /cardfound ]; then
        dialog --title "NO CARD DETECTED" --msgbox "Sorry, but no network \
-card could be probed for on your system.  Some cards (like non-PCI NE2000s) \
+card could be probed for on your system. Some cards (like non-PCI NE2000s) \
 must be supplied with the I/O address to use and can't be probed for safely. \
 You'll have to try to configure the card later by editing \
 /etc/rc.d/rc.modules or recompiling your kernel." 9 70
@@ -621,11 +626,11 @@
 elif [ "$LOOPBACK" = "yes" -a "$NETWORKMANAGER" = "yes" ]; then
   dialog --title "NETWORK SETUP COMPLETE" --yesno "Your networking \
 system is now configured to use NetworkManager for
-wired and wireless network management.  To set up wireless networks
+wired and wireless network management. To set up wireless networks
 and view status, add the Network Management control panel widget to
 your KDE desktop.
 
-Is this correct?  Press 'Yes' to confirm, or 'No' to reconfigure." 0 0
+Is this correct? Press 'Yes' to confirm, or 'No' to reconfigure." 0 0
   RETVAL=$?
 elif [ "$USE_DHCP" = "" ]; then
   while [ 0 ]; do
@@ -634,8 +639,8 @@
 --extra-label Edit \
 --cancel-label Restart \
 --inputmenu \
-"These are the settings you have entered.  To accept them and complete \
-the networking setup, press enter.  If you need to make any changes, you \
+"These are the settings you have entered. To accept them and complete \
+the networking setup, press enter. If you need to make any changes, you \
 can do that now (or reconfigure later using 'netconfig')." \
 22 60 12 \
 "Hostname:" "$HOSTNM" \
@@ -692,11 +697,11 @@
     chmod 755 etc/rc.d/rc.networkmanager
   fi
   if [ "$1" = "" ]; then
-    dialog --msgbox "Settings accepted.  Basic network configuration is complete." 6 40
+    dialog --msgbox "Settings accepted. Basic network configuration is complete." 6 40
   fi
 else
   if [ "$1" = "" ]; then
-    dialog --msgbox "Settings discarded.  Run the 'netconfig' command again if you need to reconfigure your network settings." 6 60
+    dialog --msgbox "Settings discarded. Run the 'netconfig' command again if you need to reconfigure your network settings." 6 60
   fi
 fi
 
