From 8f43a741a36e59b8a9560ff2c4f3e5da2b8dd72b Mon Sep 17 00:00:00 2001
From: Darren 'Tadgy' Austin <darren@afterdark.org.uk>
Date: Wed, 21 Jun 2023 14:50:20 +0100
Subject: Expand the help text for DHCP_IPADDR in rc.inet1.conf.

---
 rc.inet1.conf | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/rc.inet1.conf b/rc.inet1.conf
index e952498..54d0d3e 100644
--- a/rc.inet1.conf
+++ b/rc.inet1.conf
@@ -189,7 +189,8 @@ GATEWAY6=""
 #DHCP_KEEPGW[4]="yes"           # If you don't want the DHCP server to change
                                 # your default gateway
 #DHCP_IPADDR[4]=""              # Request a specific IP address from the DHCP
-                                # server
+                                # server. There is no guarantee the server
+                                # will honour the request for this IP address
 #DHCP_DEBUG[4]="yes"            # Make dhcpcd show verbose diagnostics
 #DHCP_NOIPV4LL[4]="yes"         # Do not assign an ipv4ll address when a DHCP
                                 # server is not found (ipv4ll link-local
-- 
cgit v1.2.3

