--- ./src/exiv2plugin.cpp.orig	2022-08-31 16:22:21.000000000 -0500
+++ ./src/exiv2plugin.cpp	2023-10-27 13:53:53.697705529 -0500
@@ -395,7 +395,7 @@
     std::string strFilename(asc.constData(), asc.length());
 
     try {
-        Image::AutoPtr image = Exiv2::ImageFactory::open(strFilename);
+        Image::UniquePtr image = Exiv2::ImageFactory::open(strFilename);
         if (image.get() != nullptr && image->good()) {
             image->readMetadata();
 
