diff --git a/neofetch b/neofetch
index 48b96d2..2ccb030 100755
--- a/neofetch
+++ b/neofetch
@@ -2208,8 +2208,8 @@ get_cpu() {
 
                 *)
                     cpu="$(awk -F '\\s*: | @' \
-                            '/model name|Hardware|Processor|^cpu model|chip type|^cpu type/ {
-                            cpu=$2; if ($1 == "Hardware") exit } END { print cpu }' "$cpu_file")"
+                            '/[mM]odel [nN]ame|Hardware|Processor|^cpu model|chip type|^cpu type/ {
+                            cpu=$2; if ($1 == "Hardware" || $1 == "Model Name") exit } END { print cpu }' "$cpu_file")"
                 ;;
             esac
 
