diff --git a/neofetch b/neofetch
index 48b96d2..844c16b 100755
--- a/neofetch
+++ b/neofetch
@@ -2206,6 +2206,10 @@ get_cpu() {
                     [[ -z "$cpu" ]] && cpu="$(awk -F':' '/family/ {printf $2; exit}' "$cpu_file")"
                 ;;
 
+                "loongarch"*)
+                    cpu="$(awk -F': ' '/Model/ {print $2; exit}' "$cpu_file")"
+                ;;
+
                 *)
                     cpu="$(awk -F '\\s*: | @' \
                             '/model name|Hardware|Processor|^cpu model|chip type|^cpu type/ {
