diff -Nur LVM2.2.02.84.orig//configure LVM2.2.02.84/configure
--- LVM2.2.02.84.orig//configure	2011-02-04 16:17:54.000000000 -0600
+++ LVM2.2.02.84/configure	2011-02-09 20:23:11.818797797 -0600
@@ -8768,7 +8768,7 @@
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-ludev  $LIBS"
+LIBS="-ludev -lrt $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -8799,7 +8799,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_udev_udev_queue_get_udev_is_active" >&5
 $as_echo "$ac_cv_lib_udev_udev_queue_get_udev_is_active" >&6; }
 if test "x$ac_cv_lib_udev_udev_queue_get_udev_is_active" = x""yes; then :
-  UDEV_PC="libudev"; UDEV_LIBS="-ludev"
+  UDEV_PC="libudev"; UDEV_LIBS="-ludev -lrt"
 else
   as_fn_error $? "bailing out... libudev library is required" "$LINENO" 5
 fi
