diff -Nur LVM2.2.02.114.orig/daemons/dmeventd/dmeventd.c LVM2.2.02.114/daemons/dmeventd/dmeventd.c
--- LVM2.2.02.114.orig/daemons/dmeventd/dmeventd.c	2014-11-28 17:07:42.000000000 -0600
+++ LVM2.2.02.114/daemons/dmeventd/dmeventd.c	2014-12-08 22:46:58.702066654 -0600
@@ -2176,6 +2176,9 @@
 	if (_restart)
 		restart();
 
+	/* Create rundir */
+	mkdir(DEFAULT_DM_RUN_DIR, 0700);
+
 #ifdef __linux__
 	_systemd_activation = _systemd_handover(&fifos);
 #endif
