--- ./client/scripts/linux.orig	2017-07-25 08:39:54.000000000 -0500
+++ ./client/scripts/linux	2017-08-03 15:31:41.067873392 -0500
@@ -22,6 +22,10 @@
 # 4. TIMEOUT not tested. ping has a flag I don't know, and I'm suspicious
 # of the $1 in its args.
 
+# Set a reasonable default path for Linux:
+PATH=/usr/local/sbin:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin
+export PATH
+
 # 5. Script refresh in 2017. The aliasing code was too convoluted and needs
 # to go away. Migrated DHCPv4 script to ip command from iproute2 suite.
 # This is based on Debian script with some tweaks. ifconfig is no longer
