--- xterm.terminfo.orig	2017-12-25 15:08:55.000000000 -0600
+++ xterm.terminfo	2018-04-14 12:28:57.405002630 -0500
@@ -2437,8 +2437,21 @@
 # is widely used for a variety of incompatible terminal emulations including
 # color_xterm and rxvt.
 xterm|X11 terminal emulator,
-	use=xterm-new,
+#	use=xterm-new,
 #	use=xterm-r6,
+# For compatibility purposes, use the same contents as the xterm-new entry,
+# but without "use=ansi+rep,"
+        npc,
+        indn=\E[%p1%dS,
+        kb2=\EOE,
+        kcbt=\E[Z,
+        kent=\EOM,
+        rin=\E[%p1%dT,
+#        use=ansi+rep,
+        use=ecma+strikeout,
+        use=xterm+pcfkeys,
+        use=xterm+tmux,
+        use=xterm-basic,
 
 # This fragment is for people who cannot agree on what the backspace key
 # should send.
