diff -Nur pm-utils-1.4.1.orig//pm/power.d/harddrive pm-utils-1.4.1/pm/power.d/harddrive
--- pm-utils-1.4.1.orig//pm/power.d/harddrive	2010-07-04 09:50:13.000000000 -0500
+++ pm-utils-1.4.1/pm/power.d/harddrive	2010-12-27 14:23:12.729597017 -0600
@@ -11,7 +11,7 @@
 # Default values on battery
 DRIVE_SPINDOWN_VALUE_BAT="${DRIVE_SPINDOWN_VALUE_BAT:-6}"
 DRIVE_WRITE_CACHE_BAT="${DRIVE_WRITE_CACHE_BAT:-0}" 
-DRIVE_POWER_MGMT_BAT="${DRIVE_POWER_MGMT_BAT:-1}"
+DRIVE_POWER_MGMT_BAT="${DRIVE_POWER_MGMT_BAT:-128}"
 DRIVE_ACOUSTIC_MGMT_BAT="${DRIVE_ACOUSTIC_MGMT_BAT:-254}"
 
 # Default devices to operate on
