--- ./VendorP.h.orig	2007-02-14 02:45:33.000000000 -0600
+++ ./VendorP.h	2007-08-08 14:22:17.000000000 -0500
@@ -76,6 +76,8 @@
     XtPointer       extension;          /* pointer to extension record      */
 } VendorShellClassPart;
 
+struct VendorShellClassRec;
+/*
 typedef struct _VendorShellClassRec {
   	CoreClassPart      core_class;
 	CompositeClassPart composite_class;
@@ -85,6 +87,7 @@
 } VendorShellClassRec;
 
 externalref VendorShellClassRec vendorShellClassRec;
+*/
 
 /* New fields for the vendor shell widget. */
 
