--- ./add-to-xpdfrc.orig	2003-07-08 00:43:40.000000000 -0500
+++ ./add-to-xpdfrc	2008-03-02 01:27:17.000000000 -0600
@@ -1,4 +1,4 @@
 #----- begin Cyrillic support package (2003-jun-28)
-nameToUnicode			/usr/local/share/xpdf/cyrillic/Bulgarian.nameToUnicode
-unicodeMap	KOI8-R		/usr/local/share/xpdf/cyrillic/KOI8-R.unicodeMap
+nameToUnicode			/usr/share/xpdf/cyrillic/Bulgarian.nameToUnicode
+unicodeMap	KOI8-R		/usr/share/xpdf/cyrillic/KOI8-R.unicodeMap
 #----- end Cyrillic support package
