--- ./doc/sample-xpdfrc.orig	2005-08-17 00:34:30.000000000 -0500
+++ ./doc/sample-xpdfrc	2007-02-17 21:26:12.000000000 -0600
@@ -29,20 +29,20 @@
 # installed in a "standard" location, xpdf will find them
 # automatically.)
 
-#displayFontT1 Times-Roman		/usr/local/share/ghostscript/fonts/n021003l.pfb
-#displayFontT1 Times-Italic		/usr/local/share/ghostscript/fonts/n021023l.pfb
-#displayFontT1 Times-Bold		/usr/local/share/ghostscript/fonts/n021004l.pfb
-#displayFontT1 Times-BoldItalic		/usr/local/share/ghostscript/fonts/n021024l.pfb
-#displayFontT1 Helvetica		/usr/local/share/ghostscript/fonts/n019003l.pfb
-#displayFontT1 Helvetica-Oblique	/usr/local/share/ghostscript/fonts/n019023l.pfb
-#displayFontT1 Helvetica-Bold		/usr/local/share/ghostscript/fonts/n019004l.pfb
-#displayFontT1 Helvetica-BoldOblique	/usr/local/share/ghostscript/fonts/n019024l.pfb
-#displayFontT1 Courier			/usr/local/share/ghostscript/fonts/n022003l.pfb
-#displayFontT1 Courier-Oblique		/usr/local/share/ghostscript/fonts/n022023l.pfb
-#displayFontT1 Courier-Bold		/usr/local/share/ghostscript/fonts/n022004l.pfb
-#displayFontT1 Courier-BoldOblique	/usr/local/share/ghostscript/fonts/n022024l.pfb
-#displayFontT1 Symbol			/usr/local/share/ghostscript/fonts/s050000l.pfb
-#displayFontT1 ZapfDingbats		/usr/local/share/ghostscript/fonts/d050000l.pfb
+displayFontT1 Times-Roman		/usr/share/ghostscript/fonts/n021003l.pfb
+displayFontT1 Times-Italic		/usr/share/ghostscript/fonts/n021023l.pfb
+displayFontT1 Times-Bold		/usr/share/ghostscript/fonts/n021004l.pfb
+displayFontT1 Times-BoldItalic		/usr/share/ghostscript/fonts/n021024l.pfb
+displayFontT1 Helvetica			/usr/share/ghostscript/fonts/n019003l.pfb
+displayFontT1 Helvetica-Oblique		/usr/share/ghostscript/fonts/n019023l.pfb
+displayFontT1 Helvetica-Bold		/usr/share/ghostscript/fonts/n019004l.pfb
+displayFontT1 Helvetica-BoldOblique	/usr/share/ghostscript/fonts/n019024l.pfb
+displayFontT1 Courier			/usr/share/ghostscript/fonts/n022003l.pfb
+displayFontT1 Courier-Oblique		/usr/share/ghostscript/fonts/n022023l.pfb
+displayFontT1 Courier-Bold		/usr/share/ghostscript/fonts/n022004l.pfb
+displayFontT1 Courier-BoldOblique	/usr/share/ghostscript/fonts/n022024l.pfb
+displayFontT1 Symbol			/usr/share/ghostscript/fonts/s050000l.pfb
+displayFontT1 ZapfDingbats		/usr/share/ghostscript/fonts/d050000l.pfb
 
 # If you need to display PDF files that refer to non-embedded fonts,
 # you should add one or more fontDir options to point to the
@@ -88,4 +88,4 @@
 # Set the command used to run a web browser when a URL hyperlink is
 # clicked.
 
-#urlCommand	"netscape -remote 'openURL(%s)'"
+#urlCommand	"firefox -remote 'openURL(%s)'"
