--- ./add-to-xpdfrc.orig	2003-07-08 00:43:40.000000000 -0500
+++ ./add-to-xpdfrc	2008-03-02 01:28:00.000000000 -0600
@@ -1,4 +1,4 @@
 #----- begin Greek support package (2003-jun-28)
-nameToUnicode			/usr/local/share/xpdf/greek/Greek.nameToUnicode
-unicodeMap	ISO-8859-7	/usr/local/share/xpdf/greek/ISO-8859-7.unicodeMap
+nameToUnicode			/usr/share/xpdf/greek/Greek.nameToUnicode
+unicodeMap	ISO-8859-7	/usr/share/xpdf/greek/ISO-8859-7.unicodeMap
 #----- end Greek support package
