diff -urN ca-certificates-20150426.orig/sbin/update-ca-certificates ca-certificates-20150426/sbin/update-ca-certificates
--- ca-certificates-20150426.orig/sbin/update-ca-certificates	2015-04-01 13:47:25.000000000 +0100
+++ ca-certificates-20150426/sbin/update-ca-certificates	2015-09-14 22:51:39.856021883 +0100
@@ -193,7 +193,7 @@
   echo "Running hooks in $HOOKSDIR..."
   VERBOSE_ARG=
   [ "$verbose" = 0 ] || VERBOSE_ARG="--verbose"
-  eval run-parts "$VERBOSE_ARG" --test -- "$HOOKSDIR" | while read hook
+  eval run-parts "$HOOKSDIR" | while read hook
   do
     ( cat "$ADDED"
       cat "$REMOVED" ) | "$hook" || echo "E: $hook exited with code $?."
