--- ./cacert.org/Makefile.orig
+++ ./cacert.org/Makefile
@@ -7,6 +7,7 @@
 clean:
 
 install:
+	mkdir -p $(CERTSDIR)
 	for p in *.crt; do \
 	 install -m 644 $$p $(CERTSDIR)/cacert.org_$$p ; \
 	done
--- ./mozilla/Makefile.orig
+++ ./mozilla/Makefile
@@ -9,6 +9,7 @@
 	-rm -f *.crt
 
 install:
+	mkdir -p $(CERTSDIR)
 	for p in *.crt; do \
 	 install -m 644 $$p $(CERTSDIR)/$$p ; \
 	done
--- ./sbin/Makefile.orig
+++ ./sbin/Makefile
@@ -8,4 +8,5 @@
 clean:
 
 install:
+	mkdir -p $(DESTDIR)/usr/sbin
 	install -m755 update-ca-certificates $(DESTDIR)/usr/sbin/
--- ./Makefile.orig
+++ ./Makefile
@@ -17,7 +17,7 @@
 
 install:
 	for dir in $(SUBDIRS); do \
-	  mkdir $(DESTDIR)/$(CERTSDIR)/$$dir; \
+	  mkdir -p $(DESTDIR)/$(CERTSDIR)/$$dir; \
 	  $(MAKE) -C $$dir install CERTSDIR=$(DESTDIR)/$(CERTSDIR)/$$dir; \
 	done
 	for dir in sbin; do \
--- ./spi-inc.org/Makefile.orig
+++ ./spi-inc.org/Makefile
@@ -7,6 +7,7 @@
 clean:
 
 install:
+	mkdir -p $PKG/$(CERTSDIR)
 	for p in *.crt; do \
 	 install -m 644 $$p $(CERTSDIR)/$$p ; \
 	done
