diff -Nur ca-certificates-20111211.orig/Makefile ca-certificates-20111211/Makefile
--- ca-certificates-20111211.orig/Makefile	2011-12-17 23:10:21.709842750 -0600
+++ ca-certificates-20111211/Makefile	2011-12-17 23:10:10.096969815 -0600
@@ -3,7 +3,7 @@
 #
 
 CERTSDIR = /usr/share/ca-certificates
-SUBDIRS = mozilla cacert.org debconf.org spi-inc.org
+SUBDIRS = mozilla cacert.org spi-inc.org
 
 all:
 	for dir in $(SUBDIRS); do \
