--- xv-3.10/Makefile.std.orig	2008-11-06 22:16:03.000000000 +0000
+++ xv-3.10/Makefile.std	2008-11-06 22:24:11.000000000 +0000
@@ -46,7 +46,7 @@
 MANDIR = $(PREFIX)/man/man1
 MANSUF = 1
 DOCDIR = $(PREFIX)/doc/xv-3.10a
-LIBDIR = $(PREFIX)/lib/xv
+LIBDIR = $(PREFIX)/lib64/xv
 SYSCONFDIR = $(PREFIX)/etc
 DESTDIR =
 
@@ -200,7 +200,7 @@
 
 
 ### for LINUX, uncomment the following line
-#MCHN = -DLINUX -L/usr/X11R6/lib
+MCHN = -DLINUX -L/usr/X11R6/lib64
 
 
 # For SCO 1.1 (UNIX 3.2v2) machines, uncomment the following:
