--- ./Makefile.std.orig	2010-02-19 13:12:10.000000000 -0600
+++ ./Makefile.std	2010-02-19 13:14:50.000000000 -0600
@@ -56,7 +56,7 @@
 ### NOTE: Users of old K&R compilers (i.e., any version not supporting C89
 ### string concatenation, such as "fub" "ar" => "fubar") should update
 ### xvtext.c:1831 (or thereabouts) if either PREFIX or DOCDIR changes:
-PREFIX = /usr/local
+PREFIX = /usr
 BINDIR = $(PREFIX)/bin
 MANDIR = $(PREFIX)/share/man/man1
 MANSUF = 1
