--- ./gcc/Makefile.in.orig	2023-04-26 12:25:33.525797045 -0500
+++ ./gcc/Makefile.in	2023-04-26 13:40:12.282459243 -0500
@@ -3202,11 +3202,11 @@
 	  cp $(srcdir)/gsyslimits.h $${include_dir}/syslimits.h; \
 	done
 # Install the README
-	if [ x$(STMP_FIXINC) != x ]; then \
-	  rm -f include-fixed/README; \
-	  cp $(srcdir)/../fixincludes/README-fixinc include-fixed/README; \
-	  chmod a+r include-fixed/README; \
-	fi;
+#	if [ x$(STMP_FIXINC) != x ]; then \
+#	  rm -f include-fixed/README; \
+#	  cp $(srcdir)/../fixincludes/README-fixinc include-fixed/README; \
+#	  chmod a+r include-fixed/README; \
+#	fi;
 	$(STAMP) $@
 
 .PHONY: install-gcc-tooldir
@@ -3287,10 +3287,7 @@
 	    (TARGET_MACHINE='$(target)'; srcdir=`cd $(srcdir); ${PWD_COMMAND}`; \
 	      SHELL='$(SHELL)'; MACRO_LIST=`${PWD_COMMAND}`/macro_list ; \
 	      gcc_dir=`${PWD_COMMAND}` ; \
-	      export TARGET_MACHINE srcdir SHELL MACRO_LIST && \
-	      cd $(build_objdir)/fixincludes && \
-	      $(SHELL) ./fixinc.sh "$${gcc_dir}/$${fix_dir}" \
-	        $(BUILD_SYSTEM_HEADER_DIR) $(OTHER_FIXINCLUDES_DIRS) ); \
+	      export TARGET_MACHINE srcdir SHELL MACRO_LIST ); \
 	  done; \
 	fi
 	$(STAMP) stmp-fixinc
