--- ./client/ffmpeg/libavcodec/bitstream.h.orig	2024-05-10 17:20:22.784264430 -0500
+++ ./client/ffmpeg/libavcodec/bitstream.h	2024-05-10 17:20:14.597264247 -0500
@@ -53,7 +53,7 @@
 
 extern const uint8_t ff_reverse[256];
 
-#if defined(ARCH_X86)
+#if defined(ARCH_X8666)
 // avoid +32 for shift optimization (gcc should do that ...)
 static inline  int32_t NEG_SSR32( int32_t a, int8_t s){
     asm ("sarl %1, %0\n\t"
