--- ./Magick++/lib/Magick++/Color.h.orig	2018-08-28 07:31:12.000000000 -0500
+++ ./Magick++/lib/Magick++/Color.h	2018-09-20 21:57:14.555014712 -0500
@@ -122,11 +122,12 @@
     {
       return (static_cast<double>(quantum_)/QuantumRange);
     }
-#endif
+#else
     static double scaleQuantumToDouble(const double quantum_)
     {
       return (quantum_/QuantumRange);
     }
+#endif
 
   protected:
 
