--- ./meson.build.orig	2021-08-17 03:11:30.000000000 -0500
+++ ./meson.build	2022-02-16 14:18:28.805143006 -0600
@@ -253,7 +253,6 @@
 
 foreach desktop: desktop_files
   i18n.merge_file(
-    desktop + '-desktop',
     input: desktop + '.desktop.in',
     output: desktop + '.desktop',
     install: true,
@@ -274,7 +273,6 @@
 appdata = 'nm-connection-editor.appdata.xml'
 
 i18n.merge_file(
-  'desktop',
   input: appdata + '.in',
   output: appdata,
   install: true,
