--- ./configure.in.orig	2010-01-11 14:57:36.000000000 -0600
+++ ./configure.in	2010-02-12 00:48:04.000000000 -0600
@@ -908,12 +908,12 @@
 
 dnl Test for libpng
   if test x$with_libpng != xno && test -z "$LIBPNG"; then
-    AC_MSG_CHECKING(for libpng12)
-    if $PKG_CONFIG --exists libpng12 ; then
+    AC_MSG_CHECKING(for libpng14)
+    if $PKG_CONFIG --exists libpng14 ; then
         AC_MSG_RESULT(yes)
         PNG='png'
-	PNG_DEP_CFLAGS_PACKAGES=libpng12
-	LIBPNG=`$PKG_CONFIG --libs libpng12`
+	PNG_DEP_CFLAGS_PACKAGES=libpng14
+	LIBPNG=`$PKG_CONFIG --libs libpng14`
     else
       AC_MSG_RESULT(no)
       AC_CHECK_LIB(png, png_read_info,
