diff -Nur gtk+-2.18.4.orig/gdk-pixbuf/Makefile.in gtk+-2.18.4/gdk-pixbuf/Makefile.in
--- gtk+-2.18.4.orig/gdk-pixbuf/Makefile.in	2009-11-30 22:08:44.000000000 -0600
+++ gtk+-2.18.4/gdk-pixbuf/Makefile.in	2009-12-03 21:57:10.541765261 -0600
@@ -801,7 +801,7 @@
 LIBJASPER = @LIBJASPER@
 LIBJPEG = @LIBJPEG@
 LIBOBJS = @LIBOBJS@
-LIBPNG = @LIBPNG@
+LIBPNG = @LIBPNG@ -lz
 LIBS = @LIBS@
 LIBTIFF = @LIBTIFF@
 LIBTOOL = @LIBTOOL@
