diff -Nur gtk+-2.14.6.orig/gdk-pixbuf/Makefile.in gtk+-2.14.6/gdk-pixbuf/Makefile.in
--- gtk+-2.14.6.orig/gdk-pixbuf/Makefile.in	2008-12-15 12:42:35.000000000 -0600
+++ gtk+-2.14.6/gdk-pixbuf/Makefile.in	2008-12-16 21:36:45.393854353 -0600
@@ -245,7 +245,7 @@
 LIBJASPER = @LIBJASPER@
 LIBJPEG = @LIBJPEG@
 LIBOBJS = @LIBOBJS@
-LIBPNG = @LIBPNG@
+LIBPNG = @LIBPNG@ -lz
 LIBS = @LIBS@
 LIBTIFF = @LIBTIFF@
 LIBTOOL = @LIBTOOL@
