diff -Nur gtk+-2.24.0.orig/gtk/Makefile.am gtk+-2.24.0/gtk/Makefile.am
--- gtk+-2.24.0.orig/gtk/Makefile.am	2011-01-06 22:41:26.000000000 -0600
+++ gtk+-2.24.0/gtk/Makefile.am	2011-02-11 08:52:32.734533831 -0600
@@ -38,6 +38,7 @@
 	-DGTK_DATADIR=\"$(datadir)\"			\
 	-DGTK_DATA_PREFIX=\"$(prefix)\"			\
 	-DGTK_SYSCONFDIR=\"$(sysconfdir)\"		\
+	-DHOST=\"$(host_triplet)\"			\
 	-DGTK_VERSION=\"$(GTK_VERSION)\"		\
 	-DGTK_BINARY_VERSION=\"$(GTK_BINARY_VERSION)\"	\
 	-DGTK_HOST=\"$(host)\"				\
diff -Nur gtk+-2.24.0.orig/gtk/gtkrc.c gtk+-2.24.0/gtk/gtkrc.c
--- gtk+-2.24.0.orig/gtk/gtkrc.c	2011-01-26 13:46:37.000000000 -0600
+++ gtk+-2.24.0/gtk/gtkrc.c	2011-02-11 08:54:35.145348489 -0600
@@ -450,7 +450,7 @@
       if (im_module_file)
 	result = g_strdup (im_module_file);
       else
-	result = g_build_filename (GTK_SYSCONFDIR, "gtk-2.0", "gtk.immodules", NULL);
+	result = g_build_filename (GTK_SYSCONFDIR, "gtk-2.0", HOST, "gtk.immodules", NULL);
     }
 
   return result;
