--- ./rpc.rusersd/rusers_proc.c.orig	Sun Feb 25 18:16:07 2001
+++ ./rpc.rusersd/rusers_proc.c	Sun Feb 25 18:16:18 2001
@@ -32,6 +32,7 @@
 #include <signal.h>
 #include <sys/types.h>
 #include <sys/time.h>
+#include <time.h>
 #include <utmp.h>
 #include <stdio.h>
 #include <syslog.h>
--- ./rusers/rusers.c.orig	Sun Feb 25 18:16:48 2001
+++ ./rusers/rusers.c	Sun Feb 25 18:17:07 2001
@@ -44,6 +44,7 @@
 #include <utmp.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <time.h>
 
 #include "../version.h"
 
