From 3098be64beb2ddd761735ac2554c6e0764d37f6e Mon Sep 17 00:00:00 2001
From: Xiaotian Wu <wuxiaotian@loongson.cn>
Date: Fri, 14 Oct 2022 12:50:14 +0800
Subject: [PATCH 08/16] fix

Signed-off-by: Shi Pujin <shipujin.t@gmail.com>
---
 src/arch-syscall-validate | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/arch-syscall-validate b/src/arch-syscall-validate
index 618a70c..9fc3b58 100755
--- a/src/arch-syscall-validate
+++ b/src/arch-syscall-validate
@@ -317,10 +317,10 @@ function dump_sys_loongarch64() {
 	local sed_filter=""
 
 	sed_filter+='s/__NR3264_fadvise64/223/;'
+	sed_filter+='s/__NR3264_fstatat//;'
 	sed_filter+='s/__NR3264_fcntl/25/;'
 	sed_filter+='s/__NR3264_fstatfs/44/;'
-	sed_filter+='s/__NR3264_fstatat/79/;'
-	sed_filter+='s/__NR3264_fstat/80/;'
+	sed_filter+='s/__NR3264_fstat//;'
 	sed_filter+='s/__NR3264_ftruncate/46/;'
 	sed_filter+='s/__NR3264_lseek/62/;'
 	sed_filter+='s/__NR3264_lstat/1039/;'
@@ -332,6 +332,7 @@ function dump_sys_loongarch64() {
 
 	gcc -E -dM -I$1/include/uapi \
 		-D__BITS_PER_LONG=64 \
+		-D__ARCH_WANT_NEW_STAT \
 		-D__ARCH_WANT_SYS_CLONE \
 		-D__ARCH_WANT_SYS_CLONE3 \
 		$1/arch/loongarch/include/uapi/asm/unistd.h | \
-- 
2.41.0

