From ae6bfdbb9d018ba3c96c12c692aff5bc307a0ecf Mon Sep 17 00:00:00 2001
From: Xiaotian Wu <wuxiaotian@loongson.cn>
Date: Mon, 5 Feb 2024 22:30:01 +0800
Subject: [PATCH 09/16] remove -D__ARCH_WANT_NEW_STAT

Signed-off-by: Shi Pujin <shipujin.t@gmail.com>
---
 src/arch-syscall-validate | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/arch-syscall-validate b/src/arch-syscall-validate
index 9fc3b58..ef4ab47 100755
--- a/src/arch-syscall-validate
+++ b/src/arch-syscall-validate
@@ -332,7 +332,6 @@ function dump_sys_loongarch64() {
 
 	gcc -E -dM -I$1/include/uapi \
 		-D__BITS_PER_LONG=64 \
-		-D__ARCH_WANT_NEW_STAT \
 		-D__ARCH_WANT_SYS_CLONE \
 		-D__ARCH_WANT_SYS_CLONE3 \
 		$1/arch/loongarch/include/uapi/asm/unistd.h | \
-- 
2.41.0

