From 09c847233ae7798782bdd6c3700a10efcc33153d Mon Sep 17 00:00:00 2001
From: Xiaotian Wu <wuxiaotian@loongson.cn>
Date: Fri, 13 Sep 2024 15:14:20 +0800
Subject: [PATCH 16/16] test case

Signed-off-by: Shi Pujin <shipujin.t@gmail.com>
---
 tests/38-basic-pfc_coverage.pfc | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/38-basic-pfc_coverage.pfc b/tests/38-basic-pfc_coverage.pfc
index 96198dd..f287f1d 100644
--- a/tests/38-basic-pfc_coverage.pfc
+++ b/tests/38-basic-pfc_coverage.pfc
@@ -305,15 +305,15 @@ if ($arch == 3221225730)
   # filter for syscall "open" (4294957130) [priority: 65535]
   if ($syscall == 4294957130)
     action KILL;
-  # filter for syscall "fstat" (4294957050) [priority: 65535]
-  if ($syscall == 4294957050)
-    action KILL_PROCESS;
   # filter for syscall "exit_group" (94) [priority: 65535]
   if ($syscall == 94)
     action LOG;
   # filter for syscall "exit" (93) [priority: 65535]
   if ($syscall == 93)
     action TRACE(1);
+  # filter for syscall "fstat" (80) [priority: 65535]
+  if ($syscall == 80)
+    action KILL_PROCESS;
   # filter for syscall "close" (57) [priority: 65535]
   if ($syscall == 57)
     action ERRNO(1);
-- 
2.41.0

