--- ./Imakefile.orig	2013-09-18 17:32:24.193425727 -0500
+++ ./Imakefile	2013-09-18 17:33:38.835423340 -0500
@@ -428,7 +428,6 @@
 	echo Copying pdf and html files to $(DESTDIR)$(XFIGDOCDIR) ; \
 	if [ -d $(DESTDIR)$(XFIGDOCDIR)/html ]; then set +x; \
 	   else (set -x; $(MKDIRHIER) $(DESTDIR)$(XFIGDOCDIR)/html ); fi ; \
-	$(INSTALL) -m 644 -c xfig_ref_en.pdf $(DESTDIR)$(XFIGDOCDIR) ; \
 	$(INSTALL) -m 644 -c xfig-howto.pdf $(DESTDIR)$(XFIGDOCDIR) ; \
 	if [ -d html ]; then \
 	( cd html ; tar cf - *.* images ) | ( cd $(DESTDIR)$(XFIGDOCDIR)/html ; tar xf - ) ; fi ; \
