--- ./f_readpng.c.orig	2002-09-19 16:07:59.000000000 -0500
+++ ./f_readpng.c	2010-02-19 19:31:42.000000000 -0600
@@ -133,7 +133,7 @@
 
 	if (png_get_PLTE(png_ptr, info_ptr, &palette, &num_palette)) {
 	    png_get_hIST(png_ptr, info_ptr, &histogram);
-	    png_set_dither(png_ptr, palette, num_palette, 256, histogram, 0);
+	    /* png_set_dither(png_ptr, palette, num_palette, 256, histogram, 0); */
 	}
     }
     if (color_type == PNG_COLOR_TYPE_GRAY || color_type == PNG_COLOR_TYPE_GRAY_ALPHA) {
