--- ./Fig.ad.orig	2012-02-16 14:21:34.000000000 -0600
+++ ./Fig.ad	2013-09-18 17:39:13.523412641 -0500
@@ -34,7 +34,7 @@
 ! Spell check program - put your favorite spelling check program here.
 !		It must write the misspelled words to standard output.
 
-Fig.spellcheckcommand:	spell %f
+Fig.spellcheckcommand:	aspell -a %f
 
 ! delay before spinner automatically counts when pressed (milliseconds)
 Fig.spinner_delay:		1000
