--- ./w_cmdpanel.c.orig	2013-06-14 17:39:29.000000000 -0500
+++ ./w_cmdpanel.c	2013-09-18 17:32:05.187426334 -0500
@@ -215,7 +215,9 @@
 menu_def help_menu_items[] = {
 	{"Xfig Reference (HTML)...",	0, launch_refman, False},
 #ifdef FIXED_JAPANESE_PDF
-	{"Xfig Reference (PDF, English)...",	0, launch_refpdf_en, False},
+        /* Why both a PDF and an HTML that are the same?  Remove redundant option */
+        /* and don't distribute the English PDF with the Slackware package. */
+        /* {"Xfig Reference (PDF)...",  0, launch_refpdf_en}, */
 	/* Tom Sato said that the Japanese version of the pdf looked ugly so we'll not distribute it now */
 	{"Xfig Reference (PDF, Japanese)...",	0, launch_refpdf_jp, False},
 #else
