--- ./Imakefile.orig	2012-08-09 12:14:15.000000000 -0500
+++ ./Imakefile	2013-09-18 17:26:47.819436479 -0500
@@ -45,7 +45,7 @@
 XCOMM different tree than the "correct" tree that your X system expects. The usual
 XCOMM purpose of DESTDIR is to test an install process by installing in a benign area.
 
-XCOMM XAPPLOADDIR = /home/user/xfig
+XAPPLOADDIR = /etc/X11/app-defaults
 
 XCOMM Comment out the following definition for XAW3D if you don't to want to use
 XCOMM the 3d Athena Widget Set
@@ -174,7 +174,7 @@
 
 XCOMM XFIGDOCDIR tells where the html and pdf documentation should go
 XCOMM XFIGDOCDIR = $(DOCDIR)/xfig
-XFIGDOCDIR = /usr/local/xfig/doc
+XFIGDOCDIR = /usr/doc/xfig-3.2.5c
 
 XCOMM MANDIR tells where the standard man pages should go (no need to change it
 XCOMM if you want the man pages installed in the standard place on your system
