--- ./w_cmdpanel.c.orig	2006-08-14 18:53:22.000000000 -0500
+++ ./w_cmdpanel.c	2006-08-14 18:55:02.000000000 -0500
@@ -176,7 +176,9 @@
 	/* Tom Sato said that the Japanese version of the pdf looked ugly so we'll not distribute it now */
 	{"Xfig Reference (PDF, Japanese)...",	0, launch_refpdf_jp}
 #else
-	{"Xfig Reference (PDF)...",	0, launch_refpdf_en},
+	/* Why both a PDF and an HTML that are the same?  Remove redundant option */
+	/* and don't distribute the English PDF with the Slackware package. */
+	/* {"Xfig Reference (PDF)...",	0, launch_refpdf_en}, */
 #endif /* FIXED_JAPANESE_PDF */
 	{"Xfig Man Pages (HTML)...",	5, launch_man},
 	{"How-To Guide (PDF)...",	0, launch_howto},
