--- ./Fig.ad.orig	2007-02-19 03:01:42.000000000 -0600
+++ ./Fig.ad	2007-02-19 03:05:49.000000000 -0600
@@ -22,16 +22,24 @@
 ! 		This is for viewing the xfig html reference.
 ! For netscape, this command will open the help pages in a running netscape,
 !     or start a new netscape if one isn't already running
-Fig.browser:	/usr/bin/firefox %f
+! Fig.browser: netscape -remote 'openFile(%f)' || netscape %f
+! This example might work the same for SeaMonkey:
+! Fig.browser:  seamonkey -remote 'openFile(%f)' || seamonkey %f
+! By default, let's try Firefox:
+Fig.browser:  firefox %f
 
 ! pdfviewer - put your favorite pdf viewer here.  
 !		This is for viewing the xfig how-to guide and man pages
-Fig.pdfviewer:			acroread %f
+! Fig.pdfviewer:               acroread %f
+! Fig.pdfviewer:               kpdf %f
+Fig.pdfviewer:                 xpdf %f
 
 ! Spell check program - put your favorite spelling check program here.
 !		It must write the misspelled words to standard output.
 
-Fig.spellcheckcommand:	spell %f
+! Fig.spellcheckcommand:       spell %f
+! Fig.spellcheckcommand:       aspell -a %f
+Fig.spellcheckcommand:         aspell -a %f
 
 ! delay before spinner automatically counts when pressed (milliseconds)
 Fig.spinner_delay:		1000
