--- ./Imakefile.orig	2012-08-09 12:14:15.000000000 -0500
+++ ./Imakefile	2013-09-18 17:20:15.875449009 -0500
@@ -428,7 +428,6 @@
 	echo Copying pdf and html files to $(DESTDIR)$(XFIGDOCDIR) ; \
 	if [ -d $(DESTDIR)$(XFIGDOCDIR)/html ]; then set +x; \
 	   else (set -x; $(MKDIRHIER) $(DESTDIR)$(XFIGDOCDIR)/html ); fi ; \
-	$(INSTALL) -m 644 -c xfig_man.html $(DESTDIR)$(XFIGDOCDIR) ; \
 	$(INSTALL) -m 644 -c xfig_ref_en.pdf $(DESTDIR)$(XFIGDOCDIR) ; \
 	$(INSTALL) -m 644 -c xfig-howto.pdf $(DESTDIR)$(XFIGDOCDIR) ; \
 	if [ -d html ]; then \
