--- Fig-color.ad
+++ Fig-color.ad	2008-04-13 20:50:28.000000000 +0200
@@ -76,6 +76,12 @@ Fig*MenuButton*background: 		gray83
 ! in the color editor popup
 Fig*mixedEdit.background: 		gray83
 
+Fig*mode_panel.background:		gray67
+Fig*mode_panel*topShadow::		white
+Fig*mode_panel*bottomShadow:		gray17
+Fig*mode_panel*topShadowPixel:		white
+Fig*mode_panel*bottomShadowPixel:	gray17
+
 Fig*cancel.background: 			gray88
 Fig*dismiss.background: 		gray88
 Fig*commands*background: 		gray88
@@ -85,12 +91,35 @@ Fig*horizontal.background: 		gray88
 
 Fig*topruler.background: 		gray95
 Fig*sideruler.background: 		gray95
+Fig*topruler.foreground: 		black
+Fig*sideruler.foreground: 		black
 
 ! file panel and scrollbar
 
 Fig*FigList*background: 		gray95
 Fig*List*background: 			gray95
-Fig*Scrollbar.background: 		gray95
+Fig*Scrollbar.background: 		gray77
+Fig*Scrollbar.foreground: 		gray37
+Fig*ScrollbarBackground:		gray67
+Fig*ScrollbarForeground:		gray37
+
+Fig*horizontal.shadowWidth:		2
+Fig*horizontal.topShadowPixel:		gray95
+Fig*horizontal.bottomShadowPixel:	gray37
+Fig*horizontal.topShadow:		gray95
+Fig*horizontal.bottomShadow:		gray37
+
+Fig*Scrollbar.shadowWidth:		2
+Fig*Scrollbar.topShadow:		gray95
+Fig*Scrollbar.bottomShadow:		gray37
+Fig*Scrollbar.topShadowPixel:		gray95
+Fig*Scrollbar.bottomShadowPixel:	gray37
+
+Fig*stdColor.shadowWidth:		2
+Fig*stdColor.topShadow:			gray95
+Fig*stdColor.bottomShadow:		gray37
+Fig*stdColor.topShadowPixel:		gray95
+Fig*stdColor.bottomShadowPixel:		gray37
 
 Fig*Label.background: 			gray80
 Fig*ind_box.background:			gray80
--- Fig.ad
+++ Fig.ad	2008-04-13 20:50:28.000000000 +0200
@@ -10,6 +10,8 @@ Fig.version:			3.2.4
 
 Fig*AllowShellResize: false
 
+Fig.inches: false
+
 ! Image editor - can edit imported image
 Fig.image_editor:		xv
 
@@ -20,7 +22,7 @@ Fig.ghostscript:		gs
 ! 		This is for viewing the xfig html reference.
 ! For netscape, this command will open the help pages in a running netscape,
 !     or start a new netscape if one isn't already running
-Fig.browser:	netscape -remote 'openFile(%f)' || netscape %f
+Fig.browser:	/usr/bin/firefox %f
 
 ! pdfviewer - put your favorite pdf viewer here.  
 !		This is for viewing the xfig how-to guide and man pages
@@ -99,60 +101,62 @@ Fig.internalborderwidth:	1
 ! I include both for compatibility.  Be sure to change both values when
 ! modifying them.
 
-	Fig*horizontal.shadowWidth: 0
-	Fig*horizontal.topShadowPixel: black
-	Fig*horizontal.bottomShadowPixel: white
-	Fig*horizontal.topShadow: black
-	Fig*horizontal.bottomShadow: white
+Fig*horizontal.shadowWidth:		0
+Fig*horizontal.topShadowPixel:		black
+Fig*horizontal.bottomShadowPixel:	white
+Fig*horizontal.topShadow:		black
+Fig*horizontal.bottomShadow:		white
 
 ! the following is done so the widget set doesn't try to make shades of
 ! the scrollbar shadows (even though the width is 0 it takes more colors)
-	Fig*Scrollbar.shadowWidth: 0
-	Fig*Scrollbar.topShadow: black
-	Fig*Scrollbar.bottomShadow: white
-	Fig*Scrollbar.topShadowPixel: black
-	Fig*Scrollbar.bottomShadowPixel: white
+
+Fig*Scrollbar.shadowWidth:		0
+Fig*Scrollbar.topShadow:		black
+Fig*Scrollbar.bottomShadow:		white
+Fig*Scrollbar.topShadowPixel:		black
+Fig*Scrollbar.bottomShadowPixel:	white
 
 ! this is for the standard color buttons in the color panel
-	Fig*stdColor.shadowWidth: 2
-	Fig*stdColor.topShadow: black
-	Fig*stdColor.bottomShadow: white
-	Fig*stdColor.topShadowPixel: black
-	Fig*stdColor.bottomShadowPixel: white
+
+Fig*stdColor.shadowWidth:		2
+Fig*stdColor.topShadow:			black
+Fig*stdColor.bottomShadow:		white
+Fig*stdColor.topShadowPixel:		black
+Fig*stdColor.bottomShadowPixel:		white
 
 ! turn off any shadows for the user color cells
-	Fig*colorMemory.shadowWidth: 0
+Fig*colorMemory.shadowWidth:		0
 
 ! You may need or want to adjust the highlight thickness for the mode
 ! and indicator panels depending on the results.  If it is too thick
 ! it will erase part of the button graphics.
-	Fig*mode_panel*highlightThickness: 1
-	Fig*ind_panel.ind_box.button_form.button.highlightThickness: 1
+Fig*mode_panel*highlightThickness:	1
+Fig*ind_panel.ind_box.button_form.button.highlightThickness: 1
 ! Same for the popup panels from the indicator panel
-	Fig*set_indicator_panel*Command.highlightThickness: 1
+Fig*set_indicator_panel*Command.highlightThickness:	1
 
 ! If you use the 3D Athena widget set and the small icons for the mode panel
 ! (because you have a small screen; see Imakefile) you may want to reduce
 ! the shadow width to 1 or 0.  Uncomment the following in that case:
-!	Fig*mode_panel*shadowWidth: 1
+Fig*mode_panel*shadowWidth:		1
 
 ! The following is for the update marker in the upper-right corner of the
 ! indicator buttons.  You may not want any highlight thickness because
 ! the boxes are so small.
-	Fig*ind_panel.ind_box.button_form.update.highlightThickness: 0
+Fig*ind_panel.ind_box.button_form.update.highlightThickness:	1
 
 ! This sets the shadows for the spinner arrows
-	Fig*spinup.shadowWidth:   2
-	Fig*spindown.shadowWidth: 2
+Fig*spinup.shadowWidth:			0
+Fig*spindown.shadowWidth:		0
 
 ! This gets rid of that funny black line that sometimes borders the canvas
 
-	Fig*canvas.shadowWidth:  0
+Fig*canvas.shadowWidth:			0
 
 !if on monochrome server, or no Fig-color app-defaults file
 !  This background resource shows between the buttons like a border
-	Fig*mode_panel.background:	black
-	Fig*mode_panel.foreground:	black
+Fig*mode_panel.background:		black
+Fig*mode_panel.foreground:		black
 
 ! Specify a private Compose key database file.  If this resource contains a "/"
 ! then the wired-in name of the xfig global directory won't be prefixed to it.
--- Imakefile
+++ Imakefile	2008-04-13 20:52:35.000000000 +0200
@@ -49,7 +49,7 @@ XCOMM Redefine the following if your PNG
 XCOMM are in different places
 
 PNGLIBDIR = $(USRLIBDIR)
-PNGINC = -I/usr/local/include
+PNGINC = -I/usr/include
 
 XCOMM If don't want JPEG support, comment out the #define USEJPEG line
 XCOMM Uncomment the #define for USEJPEG if you want to be able to import
@@ -68,8 +68,8 @@ XCOMM You must have version 5b or newer 
 
 #ifdef USEJPEG
 #ifdef USEINSTALLEDJPEG
-JPEGLIBDIR = /usr/local/lib
-JPEGINC = -I/usr/include/X11
+JPEGLIBDIR = /usr/lib
+JPEGINC = -I/usr/include
 #else
 JPEGLIBDIR = ../jpeg
 JPEGINC = -I$(JPEGLIBDIR)
@@ -87,8 +87,8 @@ XCOMM   in /contrib/libraries.
 XCOMM Change XPMLIBDIR if necessary to point to the xpm library (libXpm)
 XCOMM Change XPMINC if necessary to point to the include file for xpm (xpm.h)
 
-XCOMM #define USEXPM
-XCOMM #define USEXPM_ICON
+#define USEXPM
+#define USEXPM_ICON
 
 #ifdef USEXPM
 XPMLIBDIR = $(USRLIBDIR)
@@ -105,17 +105,20 @@ XCOMM the 3d Athena Widget Set (highly r
 XCOMM Then be sure to change the XAW3DINC to point to the directory where your
 XCOMM 3D Athena widget headers are located
 
-XCOMM #define XAW3D
+#define XAW3D
 
 #ifdef XAW3D
 XAW3DINC = -I/usr/include/X11/Xaw3d
 DUSEXAW3D = -DXAW3D
 XAWLIB = -lXaw3d
+#else
+XAWSRC = SmeBSB.c SimpleMenu.c
+XAWOBJ = SmeBSB.o SimpleMenu.o
 #endif
 
 XCOMM Uncomment the following if you have a wheel mouse.  See docs for description.
 
-XCOMM #define WHEELMOUSE
+#define WHEELMOUSE
 
 #ifdef WHEELMOUSE
 DUSEWHEELMOUSE = -DWHEELMOUSE
@@ -154,7 +157,7 @@ XCOMM uncomment the following line if yo
 XCOMM inline functions. With the "INLINE" keyword, you should notice that
 XCOMM the display will be a bit faster in complex figures
 
-XCOMM USEINLINE = -DUSE_INLINE
+USEINLINE = -DUSE_INLINE
 
 XCOMM use (and change) the following if you want the multi-key data base file
 XCOMM somewhere other than the standard X11 library directory
@@ -166,7 +169,7 @@ XFIGLIBDIR = $(LIBDIR)/xfig
 
 XCOMM XFIGDOCDIR tells where the html and pdf documentation should go
 XCOMM XFIGDOCDIR = $(DOCDIR)/xfig
-XFIGDOCDIR = /usr/share/doc/xfig
+XFIGDOCDIR = /usr/share/doc/packages/xfig
 
 XCOMM MANDIR tells where the standard man pages should go (no need to change it
 XCOMM if you want the man pages installed in the standard place on your system
@@ -200,7 +203,7 @@ XCOMM   instead.
 XCOMM   Add `-DCACHE_SIZE_LIMIT=xxxx' where xxxx is the cache size in kilobytes.
 XCOMM   A cache size of zero turns caching off.
 
-CACHE = -DCACHE_BITMAPS -DCACHE_SIZE_LIMIT=300
+CACHE = -DCACHE_BITMAPS -DCACHE_SIZE_LIMIT=512 -DMAXNUMPTS=50000 -DBSDLPR
 
 XCOMM For SYSV systems with BSD-style printer command which use lpr instead of
 XCOMM lp (SGI is one such machine), add -DBSDLPR to the DEFINES variable
@@ -214,6 +217,7 @@ XCOMM number of vertices.
 XCOMM If you want a compiler other than "cc", define it here
 
 XCOMM CC = /opt/SUNWspro/bin/cc
+CC = gcc
 
 XCOMM *****************************************************
 XCOMM *****************************************************
@@ -234,7 +238,7 @@ DEPLIBJPEG = $(JPEGLIBDIR)/libjpeg.a
 #endif /* USEJPEG */
 
 #ifdef I18N
-I18N_DEFS = -DI18N -DSETLOCALE
+I18N_DEFS = -DI18N
 I18N_SRC = w_i18n.c
 I18N_OBJ = w_i18n.o
 #endif
@@ -304,7 +308,7 @@ XFIGSRC =	d_arc.c d_arcbox.c d_box.c d_e
 		w_export.c w_file.c w_fontbits.c w_fontpanel.c w_grid.c w_icons.c \
 		w_indpanel.c w_library.c w_modepanel.c w_mousefun.c w_msgpanel.c \
 		w_print.c w_rottext.c w_rulers.c w_setup.c w_style.c w_util.c w_zoom.c \
-		$(I18N_SRC) SmeCascade.c SmeBSB.c SimpleMenu.c
+		$(I18N_SRC) SmeCascade.c $(XAWSRC)
 
 XFIGOBJ =	d_arc.o d_arcbox.o d_box.o d_ellipse.o d_picobj.o \
 		d_subspline.o d_line.o d_regpoly.o d_spline.o d_text.o \
@@ -326,7 +330,7 @@ XFIGOBJ =	d_arc.o d_arcbox.o d_box.o d_e
 		w_export.o w_file.o w_fontbits.o w_fontpanel.o w_grid.o w_icons.o \
 		w_indpanel.o w_library.o w_modepanel.o w_mousefun.o w_msgpanel.o \
 		w_print.o w_rottext.o w_rulers.o w_setup.o w_style.o w_util.o w_zoom.o \
-		$(I18N_OBJ) SmeCascade.o SmeBSB.o SimpleMenu.o
+		$(I18N_OBJ) SmeCascade.o $(XAWOBJ)
 
 XCOMM Other dependencies should be handled by "make depend"
 
--- e_addpt.c
+++ e_addpt.c	2008-04-13 20:50:28.000000000 +0200
@@ -52,6 +52,7 @@ point_adding_selected()
     force_nopositioning();
     force_anglegeom();
     constrained = MOVE_ARB;
+    reset_action_on();
 }
 
 static void
--- e_align.c
+++ e_align.c	2008-04-13 20:50:28.000000000 +0200
@@ -63,6 +63,7 @@ align_selected()
     canvas_middlebut_proc = init_align_canvas;
     canvas_rightbut_proc = null_proc;
     set_cursor(pick15_cursor);
+    reset_action_on();
 }
 
 /* align objects to the whole canvas */
--- e_arrow.c
+++ e_arrow.c	2008-04-13 20:50:28.000000000 +0200
@@ -47,6 +47,7 @@ arrow_head_selected()
     canvas_middlebut_proc = point_search_middle;
     canvas_rightbut_proc = null_proc;
     set_cursor(pick9_cursor);
+    reset_action_on();
 }
 
 static void
--- e_break.c
+++ e_break.c	2008-04-13 20:50:28.000000000 +0200
@@ -43,6 +43,7 @@ break_selected()
     canvas_middlebut_proc = object_search_middle;
     canvas_rightbut_proc = null_proc;
     set_cursor(pick15_cursor);
+    reset_action_on();
 }
 
 static void
@@ -82,8 +83,10 @@ init_break(p, type, x, y, px, py, loc_ta
     list_delete_compound(&objects.compounds, cur_c);
     tail(&objects, &object_tails);
     append_objects(&objects, cur_c, &object_tails);
+#if 0
     /* add the depths from this compound */
     add_compound_depth(cur_c);
+#endif
     toggle_markers_in_compound(cur_c);
     set_tags(cur_c, loc_tag);
     set_action(F_BREAK);
--- e_convert.c
+++ e_convert.c	2008-04-13 20:50:28.000000000 +0200
@@ -47,6 +47,7 @@ convert_selected()
     canvas_middlebut_proc = null_proc;
     canvas_rightbut_proc = point_search_right;
     set_cursor(pick15_cursor);
+    reset_action_on();
 }
 
 static void
--- e_deletept.c
+++ e_deletept.c	2008-04-13 20:50:28.000000000 +0200
@@ -43,6 +43,7 @@ delete_point_selected()
     canvas_middlebut_proc = null_proc;
     canvas_rightbut_proc = null_proc;
     set_cursor(pick9_cursor);
+    reset_action_on();
 }
 
 static void
--- e_flip.c
+++ e_flip.c	2008-04-13 20:50:28.000000000 +0200
@@ -61,6 +61,7 @@ flip_ud_selected()
     setcenter = 0;
     setanchor = 0;
     flip_selected();
+    reset_action_on();
 }
 
 void
@@ -76,6 +77,7 @@ flip_lr_selected()
     setcenter = 0;
     setanchor = 0;
     flip_selected();
+    reset_action_on();
 }
 
 static void
--- e_movept.c
+++ e_movept.c	2008-04-13 20:50:28.000000000 +0200
@@ -80,6 +80,7 @@ move_point_selected()
     canvas_rightbut_proc = null_proc;
     set_cursor(pick9_cursor);
     force_anglegeom();
+    reset_action_on();
 }
 
 static void
--- e_rotate.c
+++ e_rotate.c	2008-04-13 20:50:28.000000000 +0200
@@ -94,6 +94,7 @@ rotate_selected()
     canvas_middlebut_proc = object_search_middle;
     canvas_rightbut_proc = set_unset_center;
     set_cursor(pick15_cursor);
+    reset_action_on();
 }
 
 static void
--- e_update.c
+++ e_update.c	2008-04-13 20:50:28.000000000 +0200
@@ -64,6 +64,7 @@ update_selected()
     set_cursor(pick9_cursor);
     /* manage on the update buttons */
     manage_update_buts();
+    reset_action_on();
 }
 
 static int
--- fig.h
+++ fig.h	2008-04-13 20:50:28.000000000 +0200
@@ -26,6 +26,7 @@ extern	char	*my_strdup();
 #if defined(ultrix) || defined(__bsdi__) || defined(Mips) || defined(apollo)
 #include <sys/types.h>	/* for stat structure */
 #endif
+#include <sys/param.h>
 #include <sys/stat.h>
 
 #if defined(__convex__) && defined(__STDC__)
--- mode.h
+++ mode.h	2008-04-13 20:50:28.000000000 +0200
@@ -19,7 +19,7 @@
 #define MODE_H
 
 #define		F_NULL			0
-#define	    FIRST_DRAW_MODE	    F_CIRCLE_BY_RAD
+#define		FIRST_DRAW_MODE		F_CIRCLE_BY_RAD
 #define		F_CIRCLE_BY_RAD		1
 #define		F_CIRCLE_BY_DIA		2
 #define		F_ELLIPSE_BY_RAD	3
@@ -38,7 +38,7 @@
 #define		F_PICOBJ		16
 #define		F_PLACE_LIB_OBJ		17
 
-#define	    FIRST_EDIT_MODE	    F_GLUE
+#define		FIRST_EDIT_MODE		F_GLUE
 #define		F_GLUE			30
 #define		F_BREAK			31
 #define		F_SCALE			32
--- w_canvas.c
+++ w_canvas.c	2008-04-13 20:50:28.000000000 +0200
@@ -536,6 +536,7 @@ reset_click_counter(widget, closure, eve
 
 /* clear the canvas - this can't be called to clear a pixmap, only a window */
 
+void
 clear_canvas()
 {
     /* clear the splash graphic if it is still on the screen */
--- w_canvas.h
+++ w_canvas.h	2008-04-13 20:50:28.000000000 +0200
@@ -33,6 +33,7 @@ extern void	toggle_show_balloons();
 extern void	toggle_show_lengths();
 extern void	toggle_show_vertexnums();
 extern void	toggle_show_borders();
+extern void	clear_canvas();
 
 extern void	canvas_selected();
 extern void	paste_primary_selection();
--- w_menuentry.c
+++ w_menuentry.c	2008-04-13 20:50:28.000000000 +0200
@@ -170,7 +170,8 @@ Region region;
 	gc = entry->sme_bsb.norm_gray_gc;
     }
     
-    if (entry->sme_bsb.label != NULL) {
+/*    if (entry->sme_bsb.label != NULL) { */
+      if (entry->sme_bsb.label == XtName((Widget)(entry))) {
 	int x_loc = entry->sme_bsb.left_margin;
 	int len = strlen(entry->sme_bsb.label);
 	char * label = entry->sme_bsb.label;
--- w_srchrepl.c
+++ w_srchrepl.c	2008-04-13 20:50:28.000000000 +0200
@@ -56,7 +56,6 @@ There is currently no way to undo replac
 #include "w_util.h"
 #include "u_create.h"
 #include <stdarg.h>
-
 #define MAX_MISSPELLED_WORDS	200
 #define	SEARCH_WIDTH		496	/* width of search message and results */
 
--- w_util.c
+++ w_util.c	2008-04-13 20:50:28.000000000 +0200
@@ -1102,7 +1102,7 @@ check_action_on()
 	    finish_text_input(0,0,0);/* finish up any text input */
 	else {
 	    if (cur_mode == F_PLACE_LIB_OBJ)
-		cancel_place_lib_obj();	      
+		cancel_place_lib_obj(0, 0, 0);	      
 	    else {
 		put_msg("Finish (or cancel) the current operation before changing modes");
 		beep();
--- Doc/xfig.html
+++ Doc/xfig.html	2008-04-13 20:50:28.000000000 +0200
@@ -848,6 +848,20 @@ between all buttons and panels (default 
 </DL>
 <P>
 
+<B>-international</B>
+
+<DL COMPACT>
+<DT><DD>
+Switch  on  international  support (mainly Japanese
+and Korean), users of  ISO  Latin  1  (ISO-8859-1)
+probably should not use this, therefore the international
+support is switched off by default for locales using ISO-8859-1*
+codesets. For all other locales this option is already used
+by default.
+</DL>
+
+<P>
+
 
 <B>-jpeg</B>[<B>_quality</B>]
 
@@ -2208,6 +2222,7 @@ inches              boolean true        
                                             -metric (false)
 installowncmap      boolean false           -installowncmap
 internalborderwidth integer 1               -internalBW
+international       boolean false           -international
 jpeg_quality        integer 75              -jpeg_quality
 justify             boolean false           -left (false),
                                             -right (true)
--- Doc/xfig.man
+++ Doc/xfig.man	2008-04-13 20:50:28.000000000 +0200
@@ -531,6 +531,15 @@ Use lines of width
 between all buttons and panels (default = 1).
 .\"-------
 .At
+.BR \-international
+.Ap
+Switch on international support (mainly Japanese and Korean),
+users of ISO Latin 1 (ISO-8859-1) probably should not use this,
+therefore the international support is switched off by default
+for locales using ISO-8859-1* codesets. For all other locales this
+option is already used by default. 
+.\"-------
+.At
 .BR \-jpeg [ _quality ]
 .I quality
 .Ap
