--- ./src/w_cmdpanel.c.orig	2016-07-23 16:10:28.000000000 -0500
+++ ./src/w_cmdpanel.c	2018-02-24 13:36:05.267592616 -0600
@@ -215,7 +215,9 @@
 menu_def help_menu_items[] = {
 	{"Xfig Reference (HTML)...",	0, launch_refman, False},
 #ifdef FIXED_JAPANESE_PDF
-	{"Xfig Reference (PDF, English)...",	0, launch_refpdf_en, False},
+        /* Why both a PDF and an HTML that are the same?  Remove redundant option */
+        /* and don't distribute the English PDF with the Slackware package. */
+        /* {"Xfig Reference (PDF)...",  0, launch_refpdf_en}, */
 	/* Tom Sato said that the Japanese version of the pdf looked ugly so we'll not distribute it now */
 	{"Xfig Reference (PDF, Japanese)...",	0, launch_refpdf_jp, False},
 #else
