diff -ru xfig.3.2.3d.orig/main.c xfig.3.2.3d/main.c
--- xfig.3.2.3d.orig/main.c	2001-01-25 22:49:19.000000000 +0100
+++ xfig.3.2.3d/main.c	2003-05-09 11:58:41.000000000 +0200
@@ -259,18 +259,36 @@
        XtOffset(appresPtr, always_use_fontset), XtRBoolean, (caddr_t) & false},
     {"fixedFontSet", "FontSet", XtRFontSet, sizeof(XFontSet),
        XtOffset(appresPtr, fixed_fontset), XtRString,
-       (caddr_t) "-*-times-medium-r-normal--16-*-*-*-*-*-*-*,"
-	 "-*-*-medium-r-normal--16-*-*-*-*-*-*-*,*--16-*" },
+       (caddr_t) "-URW-Nimbus Roman No9 L-medium-r-normal--34-*-*-*-*-*-*-*,"
+         "-kochi-mincho-medium-r-normal--34-*-*-*-*-*-jisx0208.1983-0,"
+         "-baekmukttf-batang-medium-r-normal--34-*-*-*-*-*-ksc5601.1987-0,"
+         "-*-times-medium-r-normal--34-*-*-*-*-*-*-*,"
+	 "-*-*-medium-r-normal--34-*-*-*-*-*-*-*,*--34-*"
+         "-*-times-medium-r-normal--16-*-*-*-*-*-*-*,"
+         "-*-*-medium-r-normal--16-*-*-*-*-*-*-*,*--16-*"
+    },
     {"normalFontSet", "NormalFontSet", XtRFontSet, sizeof(XFontSet),
        XtOffset(appresPtr, normal_fontset), XtRString,
-       (caddr_t) "-*-times-medium-r-normal--16-*-*-*-*-*-*-*,"
-	 "-*-*-medium-r-normal--16-*-*-*-*-*-*-*,"
-	 "-*-*-*-r-*--16-*-*-*-*-*-*-*" },
+       (caddr_t) "-URW-Nimbus Roman No9 L-medium-r-normal--34-*-*-*-*-*-*-*,"
+         "-kochi-mincho-medium-r-normal--34-*-*-*-*-*-jisx0208.1983-0,"
+         "-baekmukttf-batang-medium-r-normal--34-*-*-*-*-*-ksc5601.1987-0,"
+         "-*-times-medium-r-normal--34-*-*-*-*-*-*-*,"
+	 "-*-*-medium-r-normal--34-*-*-*-*-*-*-*,"
+	 "-*-*-*-r-*--34-*-*-*-*-*-*-*"
+         "-*-times-medium-r-normal--16-*-*-*-*-*-*-*,"
+         "-*-*-medium-r-normal--16-*-*-*-*-*-*-*,*--16-*"
+    },
     {"boldFontSet", "BoldFontSet", XtRFontSet, sizeof(XFontSet),
        XtOffset(appresPtr, bold_fontset), XtRString,
-       (caddr_t) "-*-times-bold-r-normal--16-*-*-*-*-*-*-*,"
-	 "-*-*-bold-r-normal--16-*-*-*-*-*-*-*,"
-	 "-*-*-*-r-*--16-*-*-*-*-*-*-*" },
+       (caddr_t) "-URW-Nimbus Roman No9 L-bold-r-normal--34-*-*-*-*-*-*-*,"
+         "-kochi-gothic-medium-r-normal--34-*-*-*-*-*-jisx0208.1983-0,"
+         "-baekmukttf-dotum-medium-r-normal--34-*-*-*-*-*-ksc5601.1987-0,"
+         "-*-times-bold-r-normal--34-*-*-*-*-*-*-*," 
+	 "-*-*-bold-r-normal--34-*-*-*-*-*-*-*,"
+	 "-*-*-*-r-*--34-*-*-*-*-*-*-*"
+         "-*-times-medium-r-normal--16-*-*-*-*-*-*-*,"
+         "-*-*-medium-r-normal--16-*-*-*-*-*-*-*,*--16-*"
+    },
     {"fontSetSize", "FontSetSize", XtRInt, sizeof(int),
        XtOffset(appresPtr, fontset_size), XtRImmediate, (caddr_t)0 },
     {"inputStyle", "InputStyle", XtRString, sizeof(char *),

