--- ./Imakefile.orig	2007-02-19 03:29:55.000000000 -0600
+++ ./Imakefile	2007-02-19 03:32:49.000000000 -0600
@@ -169,7 +169,7 @@
 
 XCOMM XFIGDOCDIR tells where the html and pdf documentation should go
 XCOMM XFIGDOCDIR = $(DOCDIR)/xfig
-XFIGDOCDIR = /usr/share/doc/packages/xfig
+XFIGDOCDIR = /usr/doc/xfig-3.2.4
 
 XCOMM MANDIR tells where the standard man pages should go (no need to change it
 XCOMM if you want the man pages installed in the standard place on your system
