--- ./getopt.c.orig	Wed Aug  4 14:48:46 1999
+++ ./getopt.c	Wed Aug  4 14:48:55 1999
@@ -22,7 +22,7 @@
 char	*optarg;
 
 int
-getopt(argc, argv, opts)
+maze_getopt(argc, argv, opts)
 int	argc;
 char	**argv, *opts;
 {
--- ./maze.c.orig	Wed Aug  4 14:48:59 1999
+++ ./maze.c	Wed Aug  4 14:49:10 1999
@@ -136,7 +136,7 @@
   int flags;
 
   cmd = argv[0];
-  while ((c = getopt(argc, argv, "rSd:g:")) != EOF)
+  while ((c = maze_getopt(argc, argv, "rSd:g:")) != EOF)
     switch(c)	{
       
     case 'S':
