--- ./Makefile.orig	Wed Aug  4 14:53:55 1999
+++ ./Makefile	Wed Aug  4 14:54:07 1999
@@ -3,7 +3,7 @@
 PROGRAM = xcolormap 
 OBJS = xcolormap.o
 
-LIBS = -lm -lXt -lX11
+LIBS = -lm -L /usr/X11R6/lib -lXt -lX11
 
 all:	$(PROGRAM) 
 
--- ./xcolormap.c.orig	Wed Aug  4 14:53:23 1999
+++ ./xcolormap.c	Wed Aug  4 14:53:49 1999
@@ -113,7 +113,7 @@
 
   if (!TheDisplay)
   {
-	XtWarning ("%s: can't open display, exiting...", APPNAME);
+	XtWarning ("%s: can't open display, exiting...");
 	exit (0);
   }
 
