--- ./kimap/loginjob.cpp.orig	2015-06-24 07:43:14.000000000 -0500
+++ ./kimap/loginjob.cpp	2019-11-08 15:44:05.687811741 -0600
@@ -382,7 +382,7 @@
 
       switch ( d->authState ) {
         case LoginJobPrivate::StartTls:
-          d->sessionInternal()->startSsl( KTcpSocket::TlsV1 );
+          d->sessionInternal()->startSsl( KTcpSocket::SecureProtocols );
           break;
 
         case LoginJobPrivate::Capability:
