--- ./nntp.c.orig	2005-04-19 18:06:51.000000000 -0500
+++ ./nntp.c	2006-06-27 17:14:32.000000000 -0500
@@ -913,8 +913,9 @@
 
     if ((hp = gethostbyname(nntp_server)) != NULL)
 	server_real_name = hp->h_name;
-    else
-	nn_exitmsg(1, "NNTPSERVER is invalid");
+    else {
+	nn_exitmsg(1, "NNTP host is invalid -- please edit /etc/nntpserver");
+    }
     use_nntp = (strcmp(host_name, server_real_name) != 0);
 
     if (use_nntp) {
