# Taken from the Slamd64 repository
# Thanks Fred/Frank
--- nn-6.7.3/inst.sh.orig	2006-08-14 00:19:40.000000000 +0100
+++ nn-6.7.3/inst.sh	2006-08-14 00:32:12.000000000 +0100
@@ -60,11 +60,11 @@
 	fi
 fi
 
-set $RECMAIL
-if [ ! -f "$1" ]
-then
-	echo ERROR: Mailer program $RECMAIL not found.
-fi
+#set $RECMAIL
+#if [ ! -f "$1" ]
+#then
+#	echo ERROR: Mailer program $RECMAIL not found.
+#fi
 ) > ErrorCheck
 
 if [ -s ErrorCheck ]
@@ -246,69 +246,69 @@
 
 bin)
 	echo
-	if [ ! -d "$BIN"/. ]
+	if [ ! -d "$DESTDIR/$BIN"/. ]
 	then
-		echo Directory $BIN does not found!
+		echo Directory $DESTDIR/$BIN does not found!
 		exit 1
 	fi
 
-	echo Installing user programs in $BIN
+	echo Installing user programs in $DESTDIR/$BIN
 
-	if [ -f $BIN/nn ]
+	if [ -f $DESTDIR/$BIN/nn ]
 	then
 	     (
-		cd $BIN
+		cd $DESTDIR/$BIN
 		mv nn nn.old
-		rm -f $BIN_PROG $BIN_LINK
+		rm -f $BIN_PROG $DESTDIR/$BIN_LINK
 	     )
 	fi
 
 	for prog in $BIN_PROG
 	do
-		./inst copy 755 $BIN $prog
+		./inst copy 755 $DESTDIR/$BIN $prog
 	done
 
 	for link in $BIN_LINK
 	do
-		ln $BIN/nn $BIN/$link
+		ln -s nn $DESTDIR/$BIN/$link
 		echo $link linked to nn
 	done
 
-	if [ -f $BIN/nnacct ] ; then
-		chmod 4755 $BIN/nnacct
+	if [ -f $DESTDIR/$BIN/nnacct ] ; then
+		chmod 4755 $DESTDIR/$BIN/nnacct
 		echo nnacct is setuid ${OWNER}.
 	fi
 	;;
 
 aux)
 	echo
-	./inst mkdir $LIB 755 || exit 1
+	./inst mkdir $DESTDIR/$LIB 755 || exit 1
 
 	echo Installing auxiliary programs in $LIB
 
 	for prog in $LIB_PROG
 	do
-		./inst copy 755 $LIB $prog
+		./inst copy 755 $DESTDIR/$LIB $prog
 	done
 
-	./mkprefix conf > ${LIB}/conf
+	./mkprefix conf > $DESTDIR/${LIB}/conf
 	grep "^#" config.h |
-	sed -e '/_MAN_/d' -e 's/[ 	]*\/\*.*$//' >> ${LIB}/conf
-	./inst chmod 644 ${LIB}/conf
+	sed -e '/_MAN_/d' -e 's/[ 	]*\/\*.*$//' >> $DESTDIR/${LIB}/conf
+	./inst chmod 644 $DESTDIR/${LIB}/conf
 	;;
 
 help)
-	./inst mkdir $HELP 755 || exit 1
+	./inst mkdir $DESTDIR/$HELP 755 || exit 1
 
 	echo
-	echo Installing help files in $HELP
+	echo Installing help files in $DESTDIR/$HELP
 
 	cd help
 	for h in *
 	do
 		cd ..
-		./cvt-help < help/$h > $HELP/$h
-		./inst chmod 644 $HELP/$h
+		./cvt-help < help/$h > $DESTDIR/$HELP/$h
+		./inst chmod 644 $DESTDIR/$HELP/$h
 		echo $h
 		cd help
 	done
@@ -327,32 +327,35 @@
 	} |
 	while read DIR SECT SRC
 	do
-		if [ -d "$DIR"/. ]
+		if [ -d "$DESTDIR/$DIR"/. ]
 		then
 			for i in man/*$SRC
 			do
 				MAN=`basename ${i} $SRC`
-				NEW=$DIR/${MAN}.$SECT
-				sed -e '/^\.TH /s/6.7/'${PL}'/' $i > $NEW
-				./inst chmod 644 $NEW
-				echo $MAN in $NEW
+				if [ $MAN != 'nnmaster' ]
+				then
+					NEW=$DESTDIR/$DIR/${MAN}.$SECT
+					sed -e '/^\.TH /s/6.7/'${PL}'/' $i > $NEW
+					./inst chmod 644 $NEW
+					echo $MAN in $NEW
+				fi
 			done
 		else
-			echo $DIR not found or not writeable
+			echo $DESTDIR/$DIR not found or not writeable
 		fi
 	done
 	;;
 
 online)
-	./inst mkdir $HELP 755 || exit 1
+	./inst mkdir $DESTDIR/$HELP 755 || exit 1
 
 	MAN=$HELP/Manual
 
 	echo
-	echo "Formatting online manual $MAN"
+	echo "Formatting online manual $DESTDIR/$MAN"
 	echo ".... (continues in background) ...."
 
-	rm -f $MAN
+	rm -f $DESTDIR/$MAN
 
 	(
 	sed 	-e 's/\\f[BPI]//g' \
@@ -362,9 +365,9 @@
 		-e '/^\.if/d' -e '/^\.ta/d' -e '/^\.nr/d' \
 		-e '/^\.in/d' -e 's/^\.[BI] //' \
 		`ls -1 man/*.? man/*.??` |
-	${AWK} -f format.awk - > $MAN
+	${AWK} -f format.awk - > $DESTDIR/$MAN
 
-	./inst chmod 644 $MAN
+	./inst chmod 644 $DESTDIR/$MAN
 	) &
 	;;
 
@@ -502,9 +505,9 @@
 
 done
 
-if [ -f $LOG ]
+if [ -f $DESTDIR/$LOG ]
 then
-	chmod 666 $LOG
+	chmod 666 $DESTDIR/$LOG
 fi
 
 if $PAUSE
