--- ./util/wmgenmenu.h.orig	2012-05-24 16:28:16.000000000 -0500
+++ ./util/wmgenmenu.h	2012-07-29 15:34:37.242208543 -0500
@@ -458,7 +458,10 @@
 char *OpenSUSE[MAX_NR_APPS][2] = {
 	{ N_("YaST 2"), "yast2" },
 	{ N_("YaST"), "yast !" },
-	{ N_("System Settings"), "systemsettings" },
+        /* The KDE systemsettings is mistaken for an OpenSUSE tool, and adds */
+        /* and OpenSUSE menu that starts it.  I can't think of any reason    */
+        /* this would be useful under WindowMaker, so I'm commenting it out. */
+	/* { N_("System Settings"), "systemsettings" }, */
 	{ N_("UMTSMon"), "umtsmon" },
 	{ NULL, NULL }
 };
