diff -Nur WindowMaker-0.95.2.orig/util/wmaker.inst.in WindowMaker-0.95.2/util/wmaker.inst.in
--- WindowMaker-0.95.2.orig/util/wmaker.inst.in	2012-02-14 13:36:01.000000000 -0600
+++ WindowMaker-0.95.2/util/wmaker.inst.in	2012-02-14 18:25:28.213206250 -0600
@@ -33,20 +33,7 @@
 
 make_script() {
 	ISCRIPT="$1"
-
-	cat << EOF >> "$ISCRIPT"
-# Window Maker default X session startup script 
-
-PATH="\$PATH:$BINDIR"
-
-# If you login from xdm, uncomment this to make error messages appear
-# in the console window.
-#
-# tail -f "$HOME"/.xsession-errors > /dev/console &
-
-exec wmaker
-
-EOF
+	cat /etc/X11/xinit/xinitrc.wmaker >> $ISCRIPT
 	chmod +rx "$ISCRIPT"
 }
 
@@ -298,19 +285,7 @@
 
 trap "show_end_message;exit" 2
 
-echo
-echo "Now the .xinitrc, .Xclients or .xsession script must be updated so that"
-echo "it calls wmaker when you start an X session."
-echo "Type the name of the file that must be changed (normally .xinitrc)."
-echo "If the file already exists, it will be backed up with a .old.$DATE "
-echo "extension"
-echo "If you want to edit it by hand, hit <Control>-C now."
-read file
-
-if test "x$file" = "x"; then
-	echo "Using .xinitrc as a default value"
-	file=.xinitrc
-fi
+file=.xinitrc
 
 if [ -f "$USERDIR/$file" ]; then
 	mv "$USERDIR/$file" "$USERDIR/$file.old.$DATE"
