--- ./util/wmaker.inst.in.orig	2002-05-30 13:13:46.000000000 -0700
+++ ./util/wmaker.inst.in	2002-05-30 13:16:37.000000000 -0700
@@ -37,19 +37,7 @@
 make_script() {
 	ISCRIPT=$1
 
-	cat << EOF >> $ISCRIPT
-# Window Maker default X session startup script 
-
-PATH="\$PATH:$BINDIR"
-
-# If you login from xdm, uncomment this to make error messages appear
-# in the console window.
-#
-# tail -f $HOME/.xsession-errors > /dev/console &
-
-exec wmaker
-
-EOF
+        cat /etc/X11/xinit/xinitrc.wmaker >> $ISCRIPT
 	chmod +rx $ISCRIPT
 }
 
@@ -329,20 +317,7 @@
 
 trap "show_end_message;exit" 2
 
-echo
-echo "Now the .xinitrc, .Xclients or .xsession script must be updated so that"
-echo "it calls wmaker when you start an X session."
-echo "Type the name of the file that must be changed (normally .xinitrc)."
-echo "If the file already exists, it will be backed up with a .old.$DATE "
-echo "extension"
-echo "If you want to edit it by hand, hit <Control>-C now."
-read file
-
-if test "x$file" = "x"; then
-	echo "Using .xinitrc as a default value"
-	file=.xinitrc
-fi
-
+file=.xinitrc
 if [ -f $USERDIR/$file ]; then
 	mv $USERDIR/$file $USERDIR/$file.old.$DATE
 fi
