diff -Nur WindowMaker-0.95.7.orig/util/wmmenugen.c WindowMaker-0.95.7/util/wmmenugen.c
--- WindowMaker-0.95.7.orig/util/wmmenugen.c	2015-08-11 13:41:14.000000000 -0500
+++ WindowMaker-0.95.7/util/wmmenugen.c	2015-11-22 18:23:32.773922263 -0600
@@ -106,7 +106,7 @@
 	parse_locale(NULL, &env_lang, &env_ctry, &env_enc, &env_mod);
 	terminal = find_terminal_emulator();
 
-	for (i = 1; i <= argc; i++)
+	for (i = 1; i < argc; i++)
 	{
 		if (strncmp(argv[i], "-parser", 7) == 0 &&
 		    (argv[i][7] == '=' ||
