--- ./modprobe.c.orig	2010-02-06 13:56:19.000000000 -0600
+++ ./modprobe.c	2010-02-06 14:00:49.000000000 -0600
@@ -971,10 +971,8 @@
 	}
 
 	/* deprecated config file */
-	if (parse_config_file("/etc/modprobe.conf", name, dump_only, removing,
-			      options, commands, aliases, blacklist) > 0)
-		warn("Deprecated config file /etc/modprobe.conf, "
-		      "all config files belong into /etc/modprobe.d/.\n");
+	parse_config_file("/etc/modprobe.conf", name, dump_only, removing,
+			      options, commands, aliases, blacklist);
 
 	/* default config */
 	parse_config_scan("/etc/modprobe.d", name, dump_only, removing,
