diff -Nur module-init-tools-3.6.old/doc/modprobe.conf.sgml module-init-tools-3.6.new/doc/modprobe.conf.sgml
--- module-init-tools-3.6.old/doc/modprobe.conf.sgml	2009-02-04 01:45:24.000000000 -0600
+++ module-init-tools-3.6.new/doc/modprobe.conf.sgml	2009-02-08 01:42:05.097637263 -0600
@@ -35,7 +35,9 @@
     <para>Because the <command>modprobe</command> command can add or
       remove extra more than one module, due to module dependencies,
       we need a method of specifying what options are to be used with
-      those modules.  <filename>/etc/modprobe.conf</filename> (or, if that does not exist, all files under the <filename>/etc/modprobe.d</filename> directory) specifies
+      those modules.  All of the files under the <filename>/etc/modprobe.d/</filename>
+      directory are used, or if that directory does not exist, the
+      <filename>/etc/modprobe.conf</filename> file specifies
       those options, as required.  It can also be used to create
       convenient aliases: alternate names for a module.  Finally, it
       can override the normal <command>modprobe</command> behavior
diff -Nur module-init-tools-3.6.old/modprobe.c module-init-tools-3.6.new/modprobe.c
--- module-init-tools-3.6.old/modprobe.c	2009-02-04 01:45:24.000000000 -0600
+++ module-init-tools-3.6.new/modprobe.c	2009-02-08 01:42:05.099636402 -0600
@@ -1366,8 +1366,8 @@
 
 static const char *default_configs[] = 
 {
-	"/etc/modprobe.conf",
 	"/etc/modprobe.d",
+	"/etc/modprobe.conf",
 };
 
 static void read_toplevel_config(const char *filename,
