--- ./help/help_create_po.pl.orig	2024-01-30 14:09:03.000000000 -0600
+++ ./help/help_create_po.pl	2024-01-30 14:10:33.423012066 -0600
@@ -56,6 +56,9 @@
 open(po, "> $ARGV[1]") or die "Cannot open $ARGV[1]";
 
 print po "# DO NOT EDIT - this file is generated from $ARGV[0]\n";
+print po "msgid \"\"\n";
+print po "msgstr \"\"\n";
+print po "\"Content-Type: text/plain; charset=UTF-8\"\n";
 
 foreach my $id (keys %podefs)
 {
