--- ./test/Makefile.in.orig	2016-01-31 08:31:06.000000000 -0600
+++ ./test/Makefile.in	2018-09-18 17:37:03.435015910 -0500
@@ -122,7 +122,7 @@
 	rotate-test$(EXEEXT) alphamap$(EXEEXT) \
 	gradient-crash-test$(EXEEXT) pixel-test$(EXEEXT) \
 	matrix-test$(EXEEXT) composite-traps-test$(EXEEXT) \
-	region-contains-test$(EXEEXT) glyph-test$(EXEEXT) \
+	region-contains-test$(EXEEXT) \
 	solid-test$(EXEEXT) stress-test$(EXEEXT) cover-test$(EXEEXT) \
 	blitters-test$(EXEEXT) affine-test$(EXEEXT) \
 	scaling-test$(EXEEXT) composite$(EXEEXT) \
@@ -210,12 +210,6 @@
 fetch_test_DEPENDENCIES = libutils.la \
 	$(top_builddir)/pixman/libpixman-1.la $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1)
-glyph_test_SOURCES = glyph-test.c
-glyph_test_OBJECTS = glyph-test.$(OBJEXT)
-glyph_test_LDADD = $(LDADD)
-glyph_test_DEPENDENCIES = libutils.la \
-	$(top_builddir)/pixman/libpixman-1.la $(am__DEPENDENCIES_1) \
-	$(am__DEPENDENCIES_1)
 gradient_crash_test_SOURCES = gradient-crash-test.c
 gradient_crash_test_OBJECTS = gradient-crash-test.$(OBJEXT)
 gradient_crash_test_LDADD = $(LDADD)
@@ -392,7 +386,7 @@
 	affine-test.c alpha-loop.c alphamap.c blitters-test.c \
 	check-formats.c combiner-test.c composite.c \
 	composite-traps-test.c cover-test.c fence-image-self-test.c \
-	fetch-test.c glyph-test.c gradient-crash-test.c \
+	fetch-test.c gradient-crash-test.c \
 	infinite-loop.c lowlevel-blt-bench.c matrix-test.c oob-test.c \
 	pdf-op-test.c pixel-test.c prng-test.c radial-invalid.c \
 	radial-perf-test.c region-contains-test.c region-test.c \
@@ -404,7 +398,7 @@
 	affine-test.c alpha-loop.c alphamap.c blitters-test.c \
 	check-formats.c combiner-test.c composite.c \
 	composite-traps-test.c cover-test.c fence-image-self-test.c \
-	fetch-test.c glyph-test.c gradient-crash-test.c \
+	fetch-test.c gradient-crash-test.c \
 	infinite-loop.c lowlevel-blt-bench.c matrix-test.c oob-test.c \
 	pdf-op-test.c pixel-test.c prng-test.c radial-invalid.c \
 	radial-perf-test.c region-contains-test.c region-test.c \
@@ -816,7 +810,6 @@
 	matrix-test		      \
 	composite-traps-test	      \
 	region-contains-test	      \
-	glyph-test		      \
 	solid-test		      \
 	stress-test		      \
 	cover-test		      \
@@ -965,10 +958,6 @@
 	@rm -f fetch-test$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(fetch_test_OBJECTS) $(fetch_test_LDADD) $(LIBS)
 
-glyph-test$(EXEEXT): $(glyph_test_OBJECTS) $(glyph_test_DEPENDENCIES) $(EXTRA_glyph_test_DEPENDENCIES) 
-	@rm -f glyph-test$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(glyph_test_OBJECTS) $(glyph_test_LDADD) $(LIBS)
-
 gradient-crash-test$(EXEEXT): $(gradient_crash_test_OBJECTS) $(gradient_crash_test_DEPENDENCIES) $(EXTRA_gradient_crash_test_DEPENDENCIES) 
 	@rm -f gradient-crash-test$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(gradient_crash_test_OBJECTS) $(gradient_crash_test_LDADD) $(LIBS)
@@ -1080,7 +1069,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/cover-test.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/fence-image-self-test.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/fetch-test.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/glyph-test.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gradient-crash-test.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/infinite-loop.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/lowlevel-blt-bench.Po@am__quote@
@@ -1487,13 +1475,6 @@
 	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
 	--log-file $$b.log --trs-file $$b.trs \
 	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
-	"$$tst" $(AM_TESTS_FD_REDIRECT)
-glyph-test.log: glyph-test$(EXEEXT)
-	@p='glyph-test$(EXEEXT)'; \
-	b='glyph-test'; \
-	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
-	--log-file $$b.log --trs-file $$b.trs \
-	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
 	"$$tst" $(AM_TESTS_FD_REDIRECT)
 solid-test.log: solid-test$(EXEEXT)
 	@p='solid-test$(EXEEXT)'; \
