--- ./src/headers/input-defines.h.orig	2006-12-11 19:56:14.000000000 -0600
+++ ./src/headers/input-defines.h	2006-12-11 19:59:30.000000000 -0600
@@ -59,23 +59,4 @@
 #define PSMOUSE_SYNAPTICS     7
 #endif
 
-#ifndef HAVE_INPUT_ID
-struct input_id {
-   unsigned short bustype;
-   unsigned short vendor;
-   unsigned short product;
-   unsigned short version;
-};
-#endif
-
-#ifndef HAVE_INPUT_ABSINFO
-struct input_absinfo {
-   int value;
-   int minimum;
-   int maximum;
-   int fuzz;
-   int flat;
-};
-#endif
-
 #endif
