diff -ur gpm-1.20.1.orig/src/headers/config.h.in gpm-1.20.1/src/headers/config.h.in
--- gpm-1.20.1.orig/src/headers/config.h.in	2004-09-04 10:24:11.000000000 -0400
+++ gpm-1.20.1/src/headers/config.h.in	2004-09-04 10:26:02.000000000 -0400
@@ -46,6 +46,12 @@
 /* Define if you have the <linux/input.h> header file.  */
 #undef HAVE_LINUX_INPUT_H
 
+/* Definedif you have the input_id structure */
+#undef HAVE_INPUT_ID
+
+/* Definedif you have the input_absinfo structure */
+#undef HAVE_INPUT_ABSINFO
+
 /* Define if you have the <linux/joystick.h> header file.  */
 #undef HAVE_LINUX_JOYSTICK_H
 
