--- gpm-1.20.1/src/console.c.no-console-error	2006-01-18 17:06:15.000000000 +0100
+++ gpm-1.20.1/src/console.c	2006-01-18 17:08:19.000000000 +0100
@@ -249,8 +249,10 @@
    /* Failed, try OLD console */
    else if (stat(GPM_OLD_CONSOLE, &buf) == 0)
       console.device = GPM_OLD_CONSOLE;
-   else
-      gpm_report(GPM_PR_OOPS, "Can't determine console device"); 
+   else { // XXX handle this gracefully?
+      fprintf( stderr, "no console device found" );
+      exit( 1 );
+   }
 
    return console.device;
 }
