--- gpm-1.20.1/src/lib/liblow.c.orig	2004-10-13 16:34:15.181502369 -0400
+++ gpm-1.20.1/src/lib/liblow.c	2004-10-13 16:34:20.491778415 -0400
@@ -262,13 +262,14 @@
             gpm_report(GPM_PR_ERR,"consolename not set");
             goto err;
          }   
+#if 0
          /* do we really need this check ? */
          if(strncmp(tty,consolename,strlen(consolename)-1)
             || !isdigit(tty[strlen(consolename)-1])) {
             gpm_report(GPM_PR_ERR,"strncmp/isdigit/consolename failed");
             goto err;
          }
-          
+#endif
          conn->vc=atoi(&tty[strlen(consolename)-1]);
       }
 
