diff -u -r gpm-1.20.1.orig/src/report.c gpm-1.20.1/src/report.c
--- gpm-1.20.1.orig/src/report.c	2006-08-20 14:34:30.000000000 -0500
+++ gpm-1.20.1/src/report.c	2006-08-20 14:44:54.000000000 -0500
@@ -48,15 +48,15 @@
  *
  * Startup Mode:
  *    debug    :  - (ignore)
- *    info     :  syslog/stdout
- *    warn/err :  syslog/stderr
+ *    info     :  syslog 
+ *    warn/err :  syslog
  *    oops     :  syslog/stderr  [exit]
  *
  * Running Mode: (daemon)
  *    debug    :  - (ignore)
  *    info     :  syslog
- *    warn     :  syslog+system console
- *    err      :  syslog+system console+current console
+ *    warn     :  syslog
+ *    err      :  syslog
  *    oops     :  syslog/stderr  [_exit]
  *
  * Debug Mode  :
@@ -85,9 +85,6 @@
                syslog(LOG_INFO | LOG_USER, GPM_STRING_INFO);
                vsyslog(LOG_INFO | LOG_USER, text, ap);
 #endif               
-               fprintf(stderr,GPM_STRING_INFO);
-               vfprintf(stderr,text,ap);
-               fprintf(stderr,"\n");
                break;
 
             case GPM_STAT_WARN:
@@ -95,9 +92,6 @@
                syslog(LOG_DAEMON | LOG_WARNING, GPM_STRING_WARN);
                vsyslog(LOG_DAEMON | LOG_WARNING, text, ap);
 #endif               
-               fprintf(stderr,GPM_STRING_WARN);
-               vfprintf(stderr,text,ap);
-               fprintf(stderr,"\n");
                break;
 
             case GPM_STAT_ERR:
@@ -105,9 +99,6 @@
                syslog(LOG_DAEMON | LOG_ERR, GPM_STRING_ERR);
                vsyslog(LOG_DAEMON | LOG_ERR, text, ap);
 #endif               
-               fprintf(stderr,GPM_STRING_ERR);
-               vfprintf(stderr,text,ap);
-               fprintf(stderr,"\n");
                break;
 
             case GPM_STAT_OOPS:
@@ -139,12 +130,6 @@
                syslog(LOG_DAEMON | LOG_WARNING, GPM_STRING_WARN);
                vsyslog(LOG_DAEMON | LOG_WARNING, text, ap);
 #endif               
-               if ((f = fopen(GPM_SYS_CONSOLE, "a")) != NULL) {
-                  fprintf(f, GPM_STRING_WARN);
-                  vfprintf(f, text, ap);
-                  fprintf(f, "\n");
-                  fclose(f);
-               }   
                break;
  
             case GPM_STAT_ERR:
@@ -152,19 +137,6 @@
                syslog(LOG_DAEMON | LOG_ERR, GPM_STRING_ERR);
                vsyslog(LOG_DAEMON | LOG_ERR, text, ap);
 #endif               
-               if ((f = fopen(GPM_SYS_CONSOLE, "a")) != NULL) {
-                  fprintf(f, GPM_STRING_ERR);
-                  vfprintf(f, text, ap);
-                  fprintf(f, "\n");
-                  fclose(f);
-               }
-
-               if ((f = fopen(console.device, "a")) != NULL) {
-                  fprintf(f, GPM_STRING_ERR);
-                  vfprintf(f, text, ap);
-                  fprintf(f, "\n");
-                  fclose(f);
-               }
                break;
 
             case GPM_STAT_OOPS:
