--- ./configure.orig	2023-09-28 14:00:16.498715386 -0500
+++ ./configure	2023-09-28 14:00:55.205713245 -0500
@@ -12426,8 +12426,7 @@
 then :
   printf %s "(cached) " >&6
 else $as_nop
-  eval itz_cv_path_site_lisp=`${EMACS} -batch -l ${srcdir}/exec.el -exec "(mapcar 'print load-path)" 2>/dev/null |
-sed -e '/^$/d' | sed -n -e 2p`
+  eval
 case x${itz_cv_path_site_lisp} in
 x*site-lisp*) ;;
 x*) itz_cv_path_site_lisp='${datadir}/emacs/site-lisp' ;;
