--- ./src/Makefile.in.orig	2006-02-07 15:24:17.000000000 -0600
+++ ./src/Makefile.in	2006-02-07 15:26:03.000000000 -0600
@@ -12,7 +12,7 @@
 include $(top_builddir)/Makefile.include
 
 # Main portion: regular build rules
-MICESRC = mice.c twiddler.c synaptics.c @EVDEV_SRCS@
+MICESRC = mice.c twiddler.c synaptics.c evdev.c
 
 GSRC = main.c gpm.c gpn.c special.c startup.c server_tools.c console.c \
        selection.c client.c optparser.c $(MICESRC)
