--- ./src/daemon/open_console.c.orig	2012-10-26 16:21:38.000000000 -0500
+++ ./src/daemon/open_console.c	2018-02-03 14:48:09.879311780 -0600
@@ -22,6 +22,7 @@
 #include <fcntl.h>                  /* open and co.      */
 #include <sys/stat.h>               /* stat()            */
 #include <sys/ioctl.h>              /* ioctl             */
+#include <sys/types.h>
 
 /* Linux specific (to be outsourced in gpm2 */
 #include <linux/serial.h>           /* for serial console check */
