--- configure.orig	2008-02-27 22:35:01.000000000 -0600
+++ configure	2018-01-31 21:07:13.567970856 -0600
@@ -6619,7 +6619,7 @@
 
     TCL_SRC_DIR_NATIVE=`${CYGPATH} ${TCL_SRC_DIR}`
     TCL_TOP_DIR_NATIVE=\"${TCL_SRC_DIR_NATIVE}\"
-    TCL_GENERIC_DIR_NATIVE=\"${TCL_SRC_DIR_NATIVE}/generic\"
+    TCL_GENERIC_DIR_NATIVE=\"${TCL_SRC_DIR_NATIVE}\"
     TCL_UNIX_DIR_NATIVE=\"${TCL_SRC_DIR_NATIVE}/unix\"
     TCL_WIN_DIR_NATIVE=\"${TCL_SRC_DIR_NATIVE}/win\"
     TCL_BMAP_DIR_NATIVE=\"${TCL_SRC_DIR_NATIVE}/bitmaps\"
@@ -6645,7 +6645,7 @@
 	        ;;
 	esac
     else
-	if test ! -f "${TCL_SRC_DIR}/generic/tclInt.h" ; then
+	if test ! -f "${TCL_SRC_DIR}/tclInt.h" ; then
 	    { { echo "$as_me:$LINENO: error: Cannot find private header tclInt.h in ${TCL_SRC_DIR}" >&5
 echo "$as_me: error: Cannot find private header tclInt.h in ${TCL_SRC_DIR}" >&2;}
    { (exit 1); exit 1; }; }
@@ -6673,7 +6673,7 @@
     TK_TOP_DIR_NATIVE=\"${TK_SRC_DIR_NATIVE}\"
     TK_UNIX_DIR_NATIVE=\"${TK_SRC_DIR_NATIVE}/unix\"
     TK_WIN_DIR_NATIVE=\"${TK_SRC_DIR_NATIVE}/win\"
-    TK_GENERIC_DIR_NATIVE=\"${TK_SRC_DIR_NATIVE}/generic\"
+    TK_GENERIC_DIR_NATIVE=\"${TK_SRC_DIR_NATIVE}\"
     TK_XLIB_DIR_NATIVE=\"${TK_SRC_DIR_NATIVE}/xlib\"
     if test "${TEA_PLATFORM}" = "windows"; then
 	TK_PLATFORM_DIR_NATIVE=${TK_WIN_DIR_NATIVE}
@@ -6700,7 +6700,7 @@
 	        ;;
 	esac
     else
-	if test ! -f "${TK_SRC_DIR}/generic/tkInt.h" ; then
+	if test ! -f "${TK_SRC_DIR}/tkInt.h" ; then
 	    { { echo "$as_me:$LINENO: error: Cannot find private header tkInt.h in ${TK_SRC_DIR}" >&5
 echo "$as_me: error: Cannot find private header tkInt.h in ${TK_SRC_DIR}" >&2;}
    { (exit 1); exit 1; }; }
