--- ./libupower-glib/up-wakeups.c.orig	2013-10-18 09:41:24.000000000 -0500
+++ ./libupower-glib/up-wakeups.c	2018-11-20 23:07:29.603065864 -0600
@@ -185,7 +185,7 @@
 	return array;
 }
 
-#pragma GCC diagnostic error "-Wdeprecated-declarations"
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 
 /**
  * up_wakeups_ensure_properties:
--- ./libupower-glib/up-device.c.orig	2013-10-18 09:41:24.000000000 -0500
+++ ./libupower-glib/up-device.c	2018-11-20 23:07:24.435066237 -0600
@@ -748,7 +748,7 @@
 	return array;
 }
 
-#pragma GCC diagnostic error "-Wdeprecated-declarations"
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 
 /*
  * up_device_set_property:
