--- ./configure.ac.orig	2015-12-18 08:00:34.000000000 -0600
+++ ./configure.ac	2015-12-27 18:44:30.768289611 -0600
@@ -167,7 +167,7 @@
         ])
     ])
     AS_IF([test "x$build_libwrap" != "xno"], [
-        WRAP_LIBS="-lwrap"
+        WRAP_LIBS="-lwrap -lnsl"
         AC_DEFINE([HOSTS_ACCESS], 1, [Use hosts.allow and hosts.deny for access checking of rpc.rquotad])
         COMPILE_OPTS="$COMPILE_OPTS HOSTS_ACCESS"
     ])
