Description: Increase the test timeout for slower buildds
Author: Adrian Bunk <bunk@debian.org>
Bug-Debian: https://bugs.debian.org/878284

--- mozjs52-52.3.1.orig/js/src/Makefile.in
+++ mozjs52-52.3.1/js/src/Makefile.in
@@ -104,7 +104,7 @@ check:: check-style check-masm check-js-
 
 check-jstests:
 	$(wildcard $(RUN_TEST_PROGRAM)) $(PYTHON) -u $(srcdir)/tests/jstests.py \
-		--no-progress --format=automation --timeout 300 \
+		--no-progress --format=automation --timeout 600 \
 		$(JSTESTS_EXTRA_ARGS) \
 		$(DIST)/bin/$(JS_SHELL_NAME)$(BIN_SUFFIX)
 

