--- ./Modules/FindDoxygen.cmake.orig	2018-12-13 05:57:40.000000000 -0600
+++ ./Modules/FindDoxygen.cmake	2018-12-31 13:45:34.300895249 -0600
@@ -708,8 +708,11 @@
         if(_line MATCHES "([A-Z][A-Z0-9_]+)( *=)(.*)")
             set(_key "${CMAKE_MATCH_1}")
             set(_eql "${CMAKE_MATCH_2}")
-            string(REPLACE ";" "\\\n" _value "${CMAKE_MATCH_3}")
-            list(APPEND _Doxygen_tpl_params "${_key}${_eql}${_value}")
+            #string(REPLACE ";" "\\\n" _value "${CMAKE_MATCH_3}")
+            #list(APPEND _Doxygen_tpl_params "${_key}${_eql}${_value}")
+            string(REPLACE "\\" "\\\\" _value1 "${CMAKE_MATCH_3}")
+            string(REPLACE ";" "\\\n" _value2 "${_value1}")
+            list(APPEND _Doxygen_tpl_params "${_key}${_eql}${_value2}")
         endif()
     endforeach()
 
