--- ./installdir/htdig.conf.orig	2007-04-23 17:06:21.000000000 -0500
+++ ./installdir/htdig.conf	2007-04-23 17:08:46.000000000 -0500
@@ -20,7 +20,12 @@
 #
 # Define common_dir, which is used in several of the examples.
 #
-common_dir:     /var/www/html/htdig
+common_dir:     /var/www/htdocs/htdig
+
+#
+# Define synonyms dictionary.
+#
+synonym_dictionary: /var/www/htdocs/htdig/synonyms
 
 #
 # Disable the translation of the ISO Latin 1 SGML entities from the
