--- a/CMakeLists.txt	2018-06-07 22:22:27.000000000 +0800
+++ b/CMakeLists.txt	2018-06-20 20:55:56.000000000 +0800
@@ -45,7 +45,7 @@
 option(COPY_MESSAGING_HOST_FILE_HOME "Copy the native messaging hosts json file to user home dir" FALSE)
 add_feature_info(COPY_MESSAGING_HOST_FILE_HOME ${COPY_MESSAGING_HOST_FILE_HOME} "Enable this option to copy the native messaging hosts json file to home dir if you install plasma-browser-integration to custom prefix (non-/usr)")
 
-set(MOZILLA_DIR "${CMAKE_INSTALL_PREFIX}/lib/mozilla" CACHE STRING "Mozilla directory")
+set(MOZILLA_DIR "${CMAKE_INSTALL_PREFIX}/${KDE_INSTALL_LIBDIR}/mozilla" CACHE STRING "Mozilla directory")
 add_feature_info(MOZILLA_DIR On "Mozilla directory is '${MOZILLA_DIR}'")
 
 add_subdirectory(host)
