--- ./cmake/modules/HCT.cmake.orig	2024-02-03 14:07:18.000000000 -0600
+++ ./cmake/modules/HCT.cmake	2024-02-03 14:29:21.535797775 -0600
@@ -92,9 +92,9 @@
   # file, and define the verification command
   if(NOT copy_sources)
     set(output ${temp_output})
-    if (CLANG_FORMAT_EXE) # Only verify sources if clang-format is available.
-      set(verification COMMAND ${CMAKE_COMMAND} -E compare_files ${temp_output} ${full_output})
-    endif()
+    #if (CLANG_FORMAT_EXE) # Only verify sources if clang-format is available.
+    #  set(verification COMMAND ${CMAKE_COMMAND} -E compare_files ${temp_output} ${full_output})
+    #endif()
   endif()
   if(WIN32 AND NOT HLSL_AUTOCRLF)
     set(force_lf "--force-lf")
