diff --git a/external/VulkanMemoryAllocator/include/vk_mem_alloc.h b/external/VulkanMemoryAllocator/include/vk_mem_alloc.h
index 696af1d..fbe52cb 100644
--- a/external/VulkanMemoryAllocator/include/vk_mem_alloc.h
+++ b/external/VulkanMemoryAllocator/include/vk_mem_alloc.h
@@ -3576,6 +3576,7 @@ VMA_CALL_PRE void VMA_CALL_POST vmaDestroyImage(
 #include <cstdlib>
 #include <cstring>
 #include <utility>
+#include <cstdio>
 
 /*******************************************************************************
 CONFIGURATION SECTION
