--- ./framework/util/page_guard_manager.cpp.orig	2021-06-15 14:04:31.000000000 -0500
+++ ./framework/util/page_guard_manager.cpp	2022-02-12 18:53:36.586025716 -0600
@@ -110,7 +110,7 @@
 static struct sigaction s_old_sigaction = {};
 static stack_t          s_old_stack     = {};
 
-static uint8_t s_alt_stack[SIGSTKSZ];
+static uint8_t s_alt_stack[_SC_SIGSTKSZ];
 
 static void PageGuardExceptionHandler(int id, siginfo_t* info, void* data)
 {
