--- ./build/_deps/sdl-src/src/audio/pipewire/SDL_pipewire.c.orig	2025-03-31 13:45:45.410012021 -0500
+++ ./build/_deps/sdl-src/src/audio/pipewire/SDL_pipewire.c	2025-03-31 13:52:02.134020449 -0500
@@ -529,7 +529,7 @@
 
         // Need to parse the parameters to get the sample rate
         for (i = 0; i < info->n_params; ++i) {
-            pw_node_enum_params(node->proxy, 0, info->params[i].id, 0, 0, NULL);
+            pw_node_enum_params((struct pw_node*)node->proxy, 0, info->params[i].id, 0, 0, NULL);
         }
 
         hotplug_core_sync(node);
