--- ./utils/update_build_version.py.orig	2024-01-03 11:45:00.000000000 -0600
+++ ./utils/update_build_version.py	2024-02-03 13:48:45.412743279 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # Copyright 2016 The Shaderc Authors. All rights reserved.
 #
--- ./utils/add_copyright.py.orig	2024-01-03 11:45:00.000000000 -0600
+++ ./utils/add_copyright.py	2024-02-03 13:48:14.417742586 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2015 The Shaderc Authors. All rights reserved.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
--- ./utils/remove-file-by-suffix.py.orig	2024-01-03 11:45:00.000000000 -0600
+++ ./utils/remove-file-by-suffix.py	2024-02-03 13:48:31.279742963 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # Copyright 2015 The Shaderc Authors. All rights reserved.
 #
