--- ./layers/utils/shader_utils.cpp.orig	2025-05-17 13:01:20.000000000 -0500
+++ ./layers/utils/shader_utils.cpp	2025-05-17 13:21:36.902074668 -0500
@@ -176,10 +176,10 @@
         // --allow-localsizeid
         out_options.SetAllowLocalSizeId(true);
     }
-    if (settings.allow_offset_texture_operand) {
-        // --allow-offset-texture-operand
-        out_options.SetAllowOffsetTextureOperand(true);
-    }
+//    if (settings.allow_offset_texture_operand) {
+//        // --allow-offset-texture-operand
+//        out_options.SetAllowOffsetTextureOperand(true);
+//    }
 
     // Faster validation without friendly names.
     out_options.SetFriendlyNames(false);
