--- ./submodules/jsoncpp/src/lib_json/json_reader.cpp.orig	2022-03-06 13:14:43.000000000 -0600
+++ ./submodules/jsoncpp/src/lib_json/json_reader.cpp	2023-02-14 14:46:31.107270967 -0600
@@ -212,7 +212,7 @@
     break;
   case tokenNull:
     {
-    Value v;
+    Value v(0);
     currentValue().swapPayload(v);
     currentValue().setOffsetStart(token.start_ - begin_);
     currentValue().setOffsetLimit(token.end_ - begin_);
@@ -225,7 +225,7 @@
       // "Un-read" the current token and mark the current value as a null
       // token.
       current_--;
-      Value v;
+      Value v(0);
       currentValue().swapPayload(v);
       currentValue().setOffsetStart(current_ - begin_ - 1);
       currentValue().setOffsetLimit(current_ - begin_);
@@ -1134,7 +1134,7 @@
     break;
   case tokenNull:
     {
-    Value v;
+    Value v(0);
     currentValue().swapPayload(v);
     currentValue().setOffsetStart(token.start_ - begin_);
     currentValue().setOffsetLimit(token.end_ - begin_);
@@ -1171,7 +1171,7 @@
       // "Un-read" the current token and mark the current value as a null
       // token.
       current_--;
-      Value v;
+      Value v(0);
       currentValue().swapPayload(v);
       currentValue().setOffsetStart(current_ - begin_ - 1);
       currentValue().setOffsetLimit(current_ - begin_);
