--- ./configure.in.orig	2008-11-12 12:09:44.000000000 -0600
+++ ./configure.in	2008-11-12 12:51:44.000000000 -0600
@@ -214,7 +214,7 @@
 
 #ifndef ACCT_FILE
 #  if defined(__FreeBSD__) || defined(__linux__)
-#    define ACCT_FILE "/var/log/account/pacct"
+#    define ACCT_FILE "/var/log/pacct"
 #  else
 #    if defined(__NetBSD__)
 #      define ACCT_FILE "/var/account/acct"
@@ -234,7 +234,7 @@
 
 #ifndef SAVACCT_FILE
 #  if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__linux__)
-#    define SAVACCT_FILE "/var/log/account/savacct"
+#    define SAVACCT_FILE "/var/log/savacct"
 #  else
 #    if defined(sun) || defined(AMIX)
 #      define SAVACCT_FILE "/var/adm/savacct"
@@ -250,7 +250,7 @@
 
 #ifndef USRACCT_FILE
 #  if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__linux__)
-#    define USRACCT_FILE "/var/log/account/usracct"
+#    define USRACCT_FILE "/var/log/usracct"
 #  else
 #    if defined(sun) || defined(AMIX)
 #      define USRACCT_FILE "/var/adm/usracct"
