--- ./configure.ac.orig	2010-02-12 10:39:09.000000000 -0600
+++ ./configure.ac	2010-07-28 12:35:30.000000000 -0500
@@ -187,7 +187,7 @@
 
 #ifndef ACCT_FILE
 #  if defined(__FreeBSD__) || defined(__linux__)
-#    define ACCT_FILE "/var/account/pacct"
+#    define ACCT_FILE "/var/log/pacct"
 #  else
 #    if defined(__NetBSD__)
 #      define ACCT_FILE "/var/account/acct"
@@ -207,7 +207,7 @@
 
 #ifndef SAVACCT_FILE
 #  if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__linux__)
-#    define SAVACCT_FILE "/var/account/savacct"
+#    define SAVACCT_FILE "/var/log/savacct"
 #  else
 #    if defined(sun) || defined(AMIX)
 #      define SAVACCT_FILE "/var/adm/savacct"
@@ -223,7 +223,7 @@
 
 #ifndef USRACCT_FILE
 #  if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__linux__)
-#    define USRACCT_FILE "/var/account/usracct"
+#    define USRACCT_FILE "/var/log/usracct"
 #  else
 #    if defined(sun) || defined(AMIX)
 #      define USRACCT_FILE "/var/adm/usracct"
