--- ./talkd/announce.c.orig	Sat Jul 22 08:27:39 2000
+++ ./talkd/announce.c	Sun Feb 25 18:46:08 2001
@@ -40,6 +40,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/time.h>
+#include <time.h>
 #include <sys/wait.h>
 #include <sys/socket.h>
 #include <errno.h>
--- ./talkd/table.c.orig	Sun Feb 25 18:46:23 2001
+++ ./talkd/table.c	Sun Feb 25 18:46:34 2001
@@ -47,6 +47,7 @@
  */
 #include <sys/param.h>
 #include <sys/time.h>
+#include <time.h>
 #include <sys/socket.h>
 #include <syslog.h>
 #include <unistd.h>
