--- ./kio/kio/kdirwatch.cpp.orig	2009-06-18 19:29:01.000000000 -0500
+++ ./kio/kio/kdirwatch.cpp	2009-06-18 19:32:41.000000000 -0500
@@ -73,20 +73,20 @@
 #define _S390_BITOPS_H
 #include <sys/inotify.h>
 
-static inline int inotify_init (void)
-{
-  return syscall (__NR_inotify_init);
-}
-
-static inline int inotify_add_watch (int fd, const char *name, __u32 mask)
-{
-  return syscall (__NR_inotify_add_watch, fd, name, mask);
-}
-
-static inline int inotify_rm_watch (int fd, __u32 wd)
-{
-  return syscall (__NR_inotify_rm_watch, fd, wd);
-}
+// static inline int inotify_init (void)
+// {
+//   return syscall (__NR_inotify_init);
+// }
+
+// static inline int inotify_add_watch (int fd, const char *name, __u32 mask)
+// {
+//   return syscall (__NR_inotify_add_watch, fd, name, mask);
+// }
+
+// static inline int inotify_rm_watch (int fd, __u32 wd)
+// {
+//   return syscall (__NR_inotify_rm_watch, fd, wd);
+// }
 
 #ifndef  IN_ONLYDIR
 #define  IN_ONLYDIR 0x01000000 
