diff -Naur kdelibs-3.5.7.orig/kdecore/kpty.cpp kdelibs-3.5.7/kdecore/kpty.cpp
--- kdelibs-3.5.7.orig/kdecore/kpty.cpp	2006-05-22 19:14:21.000000000 +0100
+++ kdelibs-3.5.7/kdecore/kpty.cpp	2007-06-10 14:08:18.000000000 +0100
@@ -401,7 +401,9 @@
 #ifdef HAVE_UTEMPTER
     KProcess_Utmp utmp;
     utmp.cmdFd = d->masterFd;
-    utmp << "/usr/sbin/utempter" << "-a" << d->ttyName << "";
+    utmp << "/usr/lib/utempter/utempter" << "add";
+    if (remotehost)
+      utmp << remotehost;
     utmp.start(KProcess::Block);
     Q_UNUSED(user);
     Q_UNUSED(remotehost);
@@ -444,7 +446,7 @@
 #ifdef HAVE_UTEMPTER
     KProcess_Utmp utmp;
     utmp.cmdFd = d->masterFd;
-    utmp << "/usr/sbin/utempter" << "-d" << d->ttyName;
+    utmp << "/usr/lib/utempter/utempter" << "del";
     utmp.start(KProcess::Block);
 #elif defined(USE_LOGIN)
     const char *str_ptr = d->ttyName.data();
