--- ./Lib/distutils/sysconfig.py.orig	2020-10-05 10:07:58.000000000 -0500
+++ ./Lib/distutils/sysconfig.py	2020-10-24 17:31:20.860207925 -0500
@@ -151,7 +151,7 @@
             libdir = sys.platlibdir
         else:
             # Pure Python
-            libdir = "lib"
+            libdir = "lib64"
         libpython = os.path.join(prefix, libdir,
                                  "python" + get_python_version())
         if standard_lib:
--- ./Lib/distutils/command/install.py.orig	2020-10-05 10:07:58.000000000 -0500
+++ ./Lib/distutils/command/install.py	2020-10-24 17:29:58.230206077 -0500
@@ -29,14 +29,14 @@
 
 INSTALL_SCHEMES = {
     'unix_prefix': {
-        'purelib': '$base/lib/python$py_version_short/site-packages',
+        'purelib': '$base/lib64/python$py_version_short/site-packages',
         'platlib': '$platbase/$platlibdir/python$py_version_short/site-packages',
         'headers': '$base/include/python$py_version_short$abiflags/$dist_name',
         'scripts': '$base/bin',
         'data'   : '$base',
         },
     'unix_home': {
-        'purelib': '$base/lib/python',
+        'purelib': '$base/lib64/python',
         'platlib': '$base/$platlibdir/python',
         'headers': '$base/include/python/$dist_name',
         'scripts': '$base/bin',
