--- ./CMakeLists.txt.orig	2024-02-23 16:50:39.000000000 -0000
+++ ./CMakeLists.txt	2024-03-14 20:49:46.015432993 -0000
@@ -271,11 +271,7 @@
     endif()
   endif()
 endif()
-if(BUILD_WITH_QT6)
-  find_package(QT NAMES Qt6 COMPONENTS Core REQUIRED)
-else()
-  find_package(QT NAMES Qt5 Qt6 COMPONENTS Core REQUIRED)
-endif()
+find_package(QT NAMES Qt5 COMPONENTS Core REQUIRED)
 if(QT_VERSION_MAJOR EQUAL 6 AND QT_VERSION_MINOR LESS 2)
   set(WITH_MULTIMEDIA OFF)
 endif()
