--- ./pycups.egg-info/requires.txt.orig	2024-04-17 09:28:43.000000000 -0500
+++ ./pycups.egg-info/requires.txt	2024-04-17 15:29:21.533926348 -0500
@@ -1,2 +1 @@
-gcc
-python3-devel
+
--- ./setup.py.orig	2024-04-17 09:27:05.000000000 -0500
+++ ./setup.py	2024-04-17 15:29:36.532925519 -0500
@@ -61,10 +61,6 @@
            "Programming Language :: Python :: 3",
            ],
        license="GPLv2+",
-       install_requires=[
-           'gcc',
-           'python3-devel',
-       ],
        ext_modules=[Extension("cups",
                               ["cupsmodule.c", "cupsconnection.c",
                                "cupsppd.c", "cupsipp.c"],
--- ./PKG-INFO.orig	2024-04-17 09:28:43.393506800 -0500
+++ ./PKG-INFO	2024-04-17 15:29:05.376927242 -0500
@@ -16,8 +16,6 @@
 Classifier: Programming Language :: Python
 Classifier: Programming Language :: Python :: 3
 License-File: COPYING
-Requires-Dist: gcc
-Requires-Dist: python3-devel
 
 This is a set of Python bindings for the libcups library from the
 CUPS project.
