--- ./Makefile.orig	1999-11-06 23:55:19.000000000 -0600
+++ ./Makefile	2025-04-02 13:41:10.287638816 -0500
@@ -11,7 +11,7 @@
 
 # See the README for a description of these options.
 #OPTIONS= -DNO_32K_CASCADE -DTESTMODE=G640x480x32K -DBUG_WORKAROUND -DONLY_1_8_MODES
-OPTIONS= -DBUG_WORKAROUND
+OPTIONS= -DBUG_WORKAROUND -Wno-error=incompatible-pointer-types -Wno-error=implicit-int -Wno-error=implicit-function-declaration -Wno-error=return-mismatch -Wno-error=implicit-int -Wno-error=int-conversion
 
 # This should be the location of the JPEG v6 header files.
 JPEG_HEADERS=	-I /usr/local/include
