diff -c -r seejpeg-1.10.orig/Makefile seejpeg-1.10/Makefile
*** seejpeg-1.10.orig/Makefile	Sat Nov  6 21:55:19 1999
--- seejpeg-1.10/Makefile	Sat Apr 29 22:51:54 2000
***************
*** 6,25 ****
  # $Id: Makefile,v 1.8 1999/11/07 05:55:19 evan Exp $	
  
  # The installation directories.
! BIN=	/usr/local/bin
! MAN=	/usr/local/man/man1
  
  # See the README for a description of these options.
  #OPTIONS= -DNO_32K_CASCADE -DTESTMODE=G640x480x32K -DBUG_WORKAROUND -DONLY_1_8_MODES
  OPTIONS= -DBUG_WORKAROUND
  
  # This should be the location of the JPEG v6 header files.
! JPEG_HEADERS=	-I /usr/local/include
! #JPEG_HEADERS=	-I /usr/X11R6/include
  
  # This should be the location of the JPEG v6 libraries.
! JPEG_LIBS=	-L /usr/local/lib
! #JPEG_LIBS=	-L /usr/X11R6/lib
  
  # These may be changed, if desired.
  
--- 6,25 ----
  # $Id: Makefile,v 1.8 1999/11/07 05:55:19 evan Exp $	
  
  # The installation directories.
! BIN=	/usr/bin
! MAN=	/usr/man/man1
  
  # See the README for a description of these options.
  #OPTIONS= -DNO_32K_CASCADE -DTESTMODE=G640x480x32K -DBUG_WORKAROUND -DONLY_1_8_MODES
  OPTIONS= -DBUG_WORKAROUND
  
  # This should be the location of the JPEG v6 header files.
! #JPEG_HEADERS=	-I /usr/local/include
! JPEG_HEADERS=	-I /usr/X11R6/include
  
  # This should be the location of the JPEG v6 libraries.
! #JPEG_LIBS=	-L /usr/local/lib
! JPEG_LIBS=	-L /usr/X11R6/lib
  
  # These may be changed, if desired.
  
***************
*** 29,35 ****
  # Editing of the rest of this Makefile shouldn't be necessary.
  
  CFLAGS=		$(JPEG_HEADERS) -I jpeglib $(CCOPTS) $(OPTIONS)
! LIBS=		-lvga $(JPEG_LIBS) -ljpeg -lm
  LDOPTS=		-s
  
  SEEJPEG=	seejpeg.o jpeg.o image.o display.o cmap.o \
--- 29,35 ----
  # Editing of the rest of this Makefile shouldn't be necessary.
  
  CFLAGS=		$(JPEG_HEADERS) -I jpeglib $(CCOPTS) $(OPTIONS)
! LIBS=		-lvga jpeglib/libjpeg.a -lm
  LDOPTS=		-s
  
  SEEJPEG=	seejpeg.o jpeg.o image.o display.o cmap.o \
