--- ./unix/Makefile.orig	2009-01-18 16:41:18.000000000 -0600
+++ ./unix/Makefile	2024-05-11 12:50:37.503835132 -0500
@@ -545,7 +545,7 @@
 #	make $(MAKEF) unzips CF="${CF} `cat flags`"
 
 generic_gcc:
-	$(MAKE) $(MAKEF) generic CC=gcc IZ_BZIP2="$(IZ_BZIP2)"
+	$(MAKE) $(MAKEF) generic CC="gcc -std=gnu89" IZ_BZIP2="$(IZ_BZIP2)"
 
 # extensions to perform SVR4 package-creation after compilation
 generic_pkg:	generic svr4package
