--- ./src/config.h.orig	2009-02-05 06:14:24.000000000 -0600
+++ ./src/config.h	2009-02-19 17:51:07.000000000 -0600
@@ -45,7 +45,7 @@
 #   elif defined(__sun__) || defined (__sun)
 #      define LEASEFILE "/var/cache/dnsmasq.leases"
 #   else
-#      define LEASEFILE "/var/lib/misc/dnsmasq.leases"
+#      define LEASEFILE "/var/state/dnsmasq/dnsmasq.leases"
 #   endif
 #endif
 
--- ./dnsmasq.conf.example.orig	2009-01-03 14:28:51.000000000 -0600
+++ ./dnsmasq.conf.example	2009-02-19 17:50:29.000000000 -0600
@@ -381,7 +381,7 @@
 # The DHCP server needs somewhere on disk to keep its lease database.
 # This defaults to a sane location, but if you want to change it, use
 # the line below.
-#dhcp-leasefile=/var/lib/misc/dnsmasq.leases
+#dhcp-leasefile=/var/state/dnsmasq/dnsmasq.leases
 
 # Set the DHCP server to authoritative mode. In this mode it will barge in
 # and take over the lease for any client which broadcasts on the network,
