--- ./modules/fdlibm/src/math_private.h.orig	2023-07-31 15:30:17.157093845 -0500
+++ ./modules/fdlibm/src/math_private.h	2023-07-31 15:31:19.745095245 -0500
@@ -30,8 +30,8 @@
  * Adapted from https://github.com/freebsd/freebsd-src/search?q=__double_t
  */
 
-typedef double      __double_t;
-typedef __double_t  double_t;
+typedef long double __double_t;
+typedef long double double_t;
 typedef float       __float_t;
 
 /*
