--- ./mozilla/tools/profiler/core/platform.h.orig	2020-02-17 17:39:44.000000000 -0600
+++ ./mozilla/tools/profiler/core/platform.h	2020-02-29 22:43:51.381996340 -0600
@@ -43,32 +43,6 @@
 #include <vector>
 #include "StackTop.h"
 
-// We need a definition of gettid(), but glibc doesn't provide a
-// wrapper for it.
-#if defined(__GLIBC__)
-#include <unistd.h>
-#include <sys/syscall.h>
-static inline pid_t gettid()
-{
-  return (pid_t) syscall(SYS_gettid);
-}
-#elif defined(GP_OS_darwin)
-#include <unistd.h>
-#include <sys/syscall.h>
-static inline pid_t gettid()
-{
-  return (pid_t) syscall(SYS_thread_selfid);
-}
-#elif defined(GP_OS_android)
-#include <unistd.h>
-#elif defined(GP_OS_windows)
-#include <windows.h>
-#include <process.h>
-#ifndef getpid
-#define getpid _getpid
-#endif
-#endif
-
 extern mozilla::LazyLogModule gProfilerLog;
 
 // These are for MOZ_LOG="prof:3" or higher. It's the default logging level for
