--- ./xpfe/bootstrap/mozilla.in.orig	2006-08-20 17:44:28.000000000 -0500
+++ ./xpfe/bootstrap/mozilla.in	2007-02-19 21:25:32.000000000 -0600
@@ -60,6 +60,13 @@
   MOZ_PIS_SESSION_PID="$$"
   MOZ_PIS_USER_DIR="${MOZ_USER_DIR}"
   export MOZ_PIS_API MOZ_PIS_MOZBINDIR MOZ_PIS_SESSION_PID MOZ_PIS_USER_DIR
+  # Include /usr/lib/mozilla/plugins in the plugin path:
+  if [ "$MOZ_PLUGIN_PATH" ] ; then
+    MOZ_PLUGIN_PATH=$MOZ_PLUGIN_PATH:${moz_libdir}/plugins:/usr/lib/mozilla/plugins
+  else
+    MOZ_PLUGIN_PATH=${moz_libdir}/plugins:/usr/lib/mozilla/plugins
+  fi
+  export MOZ_PLUGIN_PATH
   
   case "${1}" in
     "start")
