--- ./mozilla/build/moz.configure/rust.configure.orig	2019-05-26 08:32:14.000000000 -0500
+++ ./mozilla/build/moz.configure/rust.configure	2019-09-08 22:44:00.615109300 -0500
@@ -105,7 +105,7 @@
             # OpenBSD
             ('x86_64', 'OpenBSD'): 'x86_64-unknown-openbsd',
             # Linux
-            ('x86', 'Linux'): 'i586-unknown-linux-gnu',
+            ('x86', 'Linux'): 'i686-unknown-linux-gnu',
             # Linux
             ('x86_64', 'Linux'): 'x86_64-unknown-linux-gnu',
             # OS X and iOS
