--- ./mozilla/modules/libpref/init/all.js.orig	2017-07-07 00:37:54.000000000 -0500
+++ ./mozilla/modules/libpref/init/all.js	2017-07-21 11:21:29.152341810 -0500
@@ -227,6 +227,9 @@
 // If true, all popups won't hide automatically on blur
 pref("ui.popup.disable_autohide", false);
 
+// Default to GTK+ v2 type scrolling:
+pref("ui.scrollToClick", 0);
+
 pref("browser.display.use_document_fonts",  1);  // 0 = never, 1 = quick, 2 = always
 // 0 = default: always, except in high contrast mode
 // 1 = always
