From 9aadd2bfd66333318461c97cc7744ccdb84c24b5 Mon Sep 17 00:00:00 2001
From: Max Verevkin <me@maxverevkin.tk>
Date: Thu, 5 Nov 2020 14:16:32 +0200
Subject: [PATCH 4/4] Pavilion 13 x360 keys

Signed-off-by: Anthony G. Basile <blueness@gentoo.org>
---
 hwdb/60-keyboard.hwdb | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/hwdb/60-keyboard.hwdb b/hwdb/60-keyboard.hwdb
index 15f0a970b..b8ffac2b8 100644
--- a/hwdb/60-keyboard.hwdb
+++ b/hwdb/60-keyboard.hwdb
@@ -508,6 +508,14 @@ evdev:atkbd:dmi:bvn*:bvr*:bd*:svnHewlett-Packard*:pnHP*Pavilion*dv7*Notebook*PC:
  KEYBOARD_KEY_c6=break
  KEYBOARD_KEY_94=reserved
 
+
+# Pavilion 13 x360 (Tablet mode and SYSRQ key)
+evdev:atkbd:dmi:bvn*:bvr*:bd*:svnHewlett-Packard*:pn*[pP][aA][vV][iI][lL][iI][oO][nN]*13*x360*:pvr*
+ KEYBOARD_KEY_d9=unknown
+ KEYBOARD_KEY_d7=!f22                                   # touchpad on
+ KEYBOARD_KEY_d2=sysrq                                  # Fn+Print = SYSRQ
+
+
 # Pavilion and Spectre x360 13 (Prevents random airplane mode activation)
 evdev:atkbd:dmi:bvn*:bvr*:bd*:svnHewlett-Packard*:pn*[sS][pP][eE][cC][tT][rR][eE]*x360*13*:pvr*
 evdev:atkbd:dmi:bvn*:bvr*:bd*:svnHP*:pn*[sS][pP][eE][cC][tT][rR][eE]*x360Convertible*:pvr*
-- 
2.29.2

