diff --git a/src/udev/udev-event.c b/src/udev/udev-event.c
index d0befba29..790498c97 100644
--- a/src/udev/udev-event.c
+++ b/src/udev/udev-event.c
@@ -855,6 +855,11 @@ void udev_event_execute_rules(struct udev_event *event,
         if (udev_device_get_subsystem(dev) == NULL)
                 return;
 
+        if (streq(udev_device_get_action(dev), "bind") || streq(udev_device_get_action(dev), "unbind")) {
+            // Ignore bind/unbind events
+            return;
+        }
+
         if (streq(udev_device_get_action(dev), "remove")) {
                 udev_device_read_db(dev);
                 udev_device_tag_index(dev, NULL, false);

