--- ./rules/50-udev-default.rules.orig	2023-05-16 10:08:56.000000000 -0500
+++ ./rules/50-udev-default.rules	2023-05-16 13:15:25.690216651 -0500
@@ -31,12 +31,12 @@
 
 SUBSYSTEM=="video4linux", GROUP="video"
 SUBSYSTEM=="graphics", GROUP="video"
-SUBSYSTEM=="drm", KERNEL!="renderD*", GROUP="video"
+SUBSYSTEM=="drm", GROUP="video"
 SUBSYSTEM=="dvb", GROUP="video"
 SUBSYSTEM=="media", GROUP="video"
 SUBSYSTEM=="cec", GROUP="video"
 
-SUBSYSTEM=="drm", KERNEL=="renderD*", GROUP="video", MODE="0666"
+SUBSYSTEM=="drm", GROUP="video", MODE="0666"
 SUBSYSTEM=="kfd", GROUP="video", MODE="0666"
 
 SUBSYSTEM=="misc", KERNEL=="sgx_enclave", GROUP="sgx", MODE="0660"
