--- ./rules/50-udev-default.rules.orig	2023-05-16 10:08:56.000000000 -0500
+++ ./rules/50-udev-default.rules	2023-05-17 12:45:42.392847645 -0500
@@ -39,8 +39,9 @@
 SUBSYSTEM=="drm", KERNEL=="renderD*", GROUP="video", MODE="0666"
 SUBSYSTEM=="kfd", GROUP="video", MODE="0666"
 
-SUBSYSTEM=="misc", KERNEL=="sgx_enclave", GROUP="sgx", MODE="0660"
-SUBSYSTEM=="misc", KERNEL=="sgx_vepc", GROUP="sgx", MODE="0660"
+# This is obsolete. No sense starting to support it now.
+#SUBSYSTEM=="misc", KERNEL=="sgx_enclave", GROUP="sgx", MODE="0660"
+#SUBSYSTEM=="misc", KERNEL=="sgx_vepc", GROUP="sgx", MODE="0660"
 
 # When using static_node= with non-default permissions, also update
 # tmpfiles.d/static-nodes-permissions.conf.in to keep permissions synchronized.
