--- ./nping/Makefile.in.orig	2025-02-10 14:32:48.000000000 -0600
+++ ./nping/Makefile.in	2025-05-08 13:59:14.982724762 -0500
@@ -49,7 +49,7 @@
 STATIC =
 LDFLAGS = @LDFLAGS@ $(DBGFLAGS) $(STATIC)
 OPENSSL_LIBS = @OPENSSL_LIBS@
-LIBS =  $(NSOCKDIR)/src/libnsock.a ../libnetutil/libnetutil.a $(NBASEDIR)/libnbase.a $(OPENSSL_LIBS) @LIBPCAP_LIBS@ @LIBDNET_LIBS@ @LIBS@
+LIBS =  $(NSOCKDIR)/src/libnsock.a ../libnetutil/libnetutil.a $(NBASEDIR)/libnbase.a $(OPENSSL_LIBS) -lpcap @LIBDNET_LIBS@ @LIBS@
 # LIBS =  -lefence @LIBS@
 # LIBS =  -lrmalloc @LIBS@
 INSTALL = @INSTALL@
