Description: Fix compilation by adding missing #include(s)
Author: Mathieu Malaterre <mathieu@debian>
Bug-Debian: https://bugs.debian.org/799162
Forwarded: no

--- dmapi-2.2.10.orig/libdm/dm_handle2path.c
+++ dmapi-2.2.10/libdm/dm_handle2path.c
@@ -22,7 +22,12 @@
 
 #include <mntent.h>
 #include <dirent.h>
+#include <string.h>
 #ifdef linux
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
 #include "getdents.h"
 #endif
 
