--- ./include/buildmacros.orig	2007-01-22 23:56:41.000000000 -0600
+++ ./include/buildmacros	2008-04-01 00:04:57.000000000 -0500
@@ -26,7 +26,7 @@
 	  $(LFILES:.l=.o) \
 	  $(YFILES:%.y=%.tab.o)
 
-INSTALL	= $(TOPDIR)/install-sh -o $(PKG_USER) -g $(PKG_GROUP)
+INSTALL	= $(TOPDIR)/install-sh
 
 SHELL = /bin/sh
 IMAGES_DIR = $(TOPDIR)/all-images
--- ./include/builddefs.in.orig	2007-05-28 02:00:24.000000000 -0500
+++ ./include/builddefs.in	2008-04-01 00:04:57.000000000 -0500
@@ -23,14 +23,14 @@
 PKG_VERSION	= @pkg_version@
 PKG_PLATFORM	= @pkg_platform@
 PKG_DISTRIBUTION= @pkg_distribution@
-PKG_BIN_DIR	= @bindir@
-PKG_SBIN_DIR	= @sbindir@
-PKG_LIB_DIR	= @libdir@@libdirsuffix@
-PKG_DEVLIB_DIR	= @libexecdir@@libdirsuffix@
-PKG_INC_DIR	= @includedir@/xfs
-PKG_MAN_DIR	= @mandir@
-PKG_DOC_DIR	= @datadir@/doc/@pkg_name@
-PKG_LOCALE_DIR	= @datadir@/locale
+PKG_BIN_DIR	= $(DESTDIR)@bindir@
+PKG_SBIN_DIR	= $(DESTDIR)@sbindir@
+PKG_LIB_DIR	= $(DESTDIR)@libdir@@libdirsuffix@
+PKG_DEVLIB_DIR	= $(DESTDIR)@libexecdir@@libdirsuffix@
+PKG_INC_DIR	= $(DESTDIR)@includedir@/xfs
+PKG_MAN_DIR	= $(DESTDIR)@mandir@
+PKG_DOC_DIR	= $(DESTDIR)@datadir@/doc/@pkg_name@
+PKG_LOCALE_DIR	= $(DESTDIR)@datadir@/locale
 
 CC		= @cc@
 AWK		= @awk@
