diff --git a/libdm/getdents.c b/libdm/getdents.c
index 1e8e44e..ec6107f 100644
--- a/libdm/getdents.c
+++ b/libdm/getdents.c
@@ -225,7 +225,7 @@ __getdents_wrap (int fd, char *buf, size_t nbytes)
     dp = (DIRENT_TYPE *) buf;
     skdp = kdp = alloca (red_nbytes);
 
-    retval = syscall(SYS_getdents, fd, kdp, red_nbytes);
+    retval = syscall(SYS_getdents64, fd, kdp, red_nbytes);
 
     if (retval == -1)
       return -1;
