--- ./src/update-desktop-database.c.orig	2010-03-09 20:33:41.000000000 -0600
+++ ./src/update-desktop-database.c	2010-05-11 09:00:08.000000000 -0500
@@ -135,15 +135,9 @@
         case MU_VALID:
           break;
         case MU_DISCOURAGED:
-          udd_print (_("Warning in file \"%s\": usage of MIME type \"%s\" is "
-                       "discouraged (%s)\n"),
-                     desktop_file, mime_types[i], valid_error);
           g_free (valid_error);
           break;
         case MU_INVALID:
-          udd_print (_("Error in file \"%s\": \"%s\" is an invalid MIME type "
-                       "(%s)\n"),
-                     desktop_file, mime_types[i], valid_error);
           g_free (valid_error);
           /* not a break: we continue to the next mime type */
           continue;
