--- ./src/libical/icalrecur.c.orig	2020-03-07 08:42:42.000000000 -0600
+++ ./src/libical/icalrecur.c	2020-10-30 03:56:12.509914568 -0500
@@ -142,6 +142,14 @@
 #include <stddef.h>     /* For offsetof() macro */
 #include <stdlib.h>
 
+#ifndef FALSE
+#define FALSE (0)
+#endif
+
+#ifndef TRUE
+#define TRUE (!FALSE)
+#endif
+
 #if defined(HAVE_LIBICU)
 #include <unicode/ucal.h>
 #include <unicode/ustring.h>
