--- ./Makefile.orig	2024-05-11 14:25:05.399961923 -0500
+++ ./Makefile	2024-05-11 14:25:19.301962234 -0500
@@ -86,7 +86,8 @@
  #-Wimplicit -Wshadow -Wid-clash-6 #-Wuninitialized
 
 # The place to put your favourite extra cc flag
-CFLAGS0 = -O -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 #$(GCC_WARNINGS)
+CFLAGS0 = -O -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Wno-error=implicit-int -Wno-error=implicit-function-declaration #$(GCC_WARNINGS)
+
 LDFLAGS0= -s
 # Read my libs :-)
 LIBS=
