diff -Nur dnsmasq-2.52.orig/dnsmasq.conf.example dnsmasq-2.52/dnsmasq.conf.example
--- dnsmasq-2.52.orig/dnsmasq.conf.example	2009-11-25 06:55:16.000000000 -0600
+++ dnsmasq-2.52/dnsmasq.conf.example	2010-01-22 15:32:41.137807274 -0600
@@ -415,7 +415,7 @@
 # The DHCP server needs somewhere on disk to keep its lease database.
 # This defaults to a sane location, but if you want to change it, use
 # the line below.
-#dhcp-leasefile=/var/lib/misc/dnsmasq.leases
+#dhcp-leasefile=/var/state/dnsmasq/dnsmasq.leases
 
 # Set the DHCP server to authoritative mode. In this mode it will barge in
 # and take over the lease for any client which broadcasts on the network,
diff -Nur dnsmasq-2.52.orig/src/config.h dnsmasq-2.52/src/config.h
--- dnsmasq-2.52.orig/src/config.h	2010-01-21 08:22:06.000000000 -0600
+++ dnsmasq-2.52/src/config.h	2010-01-22 15:32:41.136833058 -0600
@@ -47,7 +47,7 @@
 #   elif defined(__sun__) || defined (__sun)
 #      define LEASEFILE "/var/cache/dnsmasq.leases"
 #   else
-#      define LEASEFILE "/var/lib/misc/dnsmasq.leases"
+#      define LEASEFILE "/var/state/dnsmasq/dnsmasq.leases"
 #   endif
 #endif
 
