diff -u -r kompare-20.08.3.orig/komparepart/kompare_part.cpp kompare-20.08.3/komparepart/kompare_part.cpp
--- kompare-20.08.3.orig/komparepart/kompare_part.cpp	2020-10-11 01:17:46.000000000 -0500
+++ kompare-20.08.3/komparepart/kompare_part.cpp	2020-11-21 19:06:25.761427059 -0600
@@ -610,7 +610,7 @@
 void KomparePart::slotFilePrint()
 {
     QPrinter printer;
-    printer.setOrientation(QPrinter::Landscape);
+    printer.setPageOrientation(QPageLayout::Landscape);
     QPrintDialog* dlg = new QPrintDialog(&printer, nullptr);
 
     if (dlg->exec() == QDialog::Accepted)
@@ -625,7 +625,7 @@
 void KomparePart::slotFilePrintPreview()
 {
     QPrinter printer;
-    printer.setOrientation(QPrinter::Landscape);
+    printer.setPageOrientation(QPageLayout::Landscape);
     QPrintPreviewDialog dlg(&printer);
 
     connect(&dlg, &QPrintPreviewDialog::paintRequested, this, &KomparePart::slotPaintRequested);
