From ca2b7fdf3f684a55280174ecf20935c7f2e7c897 Mon Sep 17 00:00:00 2001
From: Robby Workman <rw@rlworkman.net>
Date: Tue, 17 Mar 2015 22:10:56 +0100
Subject: [PATCH] g_free() on a GtkWidget is wrong on so many levels.. (bug
 #11713)

---
 panel-plugin/systemload.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/panel-plugin/systemload.c b/panel-plugin/systemload.c
index 628947e..4df29e5 100644
--- a/panel-plugin/systemload.c
+++ b/panel-plugin/systemload.c
@@ -394,7 +394,6 @@ monitor_free(XfcePanelPlugin *plugin, t_global_monitor *global)
         g_source_remove(global->timeout_id);
 
     g_free(global->command.command_text);
-    g_free(global->menu_item);
 
     for(count = 0; count < 3; count++)
     {
-- 
2.3.3

