--- ./lib/tevent/wscript.orig	2016-03-08 05:07:45.000000000 -0600
+++ ./lib/tevent/wscript	2016-04-11 12:45:14.299532166 -0500
@@ -92,11 +92,8 @@
     if bld.CONFIG_SET('HAVE_SOLARIS_PORTS'):
         SRC += ' tevent_port.c'
 
-    if bld.env.standalone_tevent:
-        bld.env.PKGCONFIGDIR = '${LIBDIR}/pkgconfig'
-        private_library = False
-    else:
-        private_library = True
+    bld.env.PKGCONFIGDIR = '${LIBDIR}/pkgconfig'
+    private_library = False
 
     if not bld.CONFIG_SET('USING_SYSTEM_TEVENT'):
         bld.SAMBA_LIBRARY('tevent',
--- ./lib/tdb/wscript.orig	2016-01-26 05:45:46.000000000 -0600
+++ ./lib/tdb/wscript	2016-04-11 12:45:14.301532166 -0500
@@ -117,11 +117,8 @@
 
     COMMON_SRC = bld.SUBDIR('common', COMMON_FILES)
 
-    if bld.env.standalone_tdb:
-        bld.env.PKGCONFIGDIR = '${LIBDIR}/pkgconfig'
-        private_library = False
-    else:
-        private_library = True
+    bld.env.PKGCONFIGDIR = '${LIBDIR}/pkgconfig'
+    private_library = False
 
     if not bld.CONFIG_SET('USING_SYSTEM_TDB'):
 
--- ./lib/talloc/wscript.orig	2016-04-11 12:45:14.295532166 -0500
+++ ./lib/talloc/wscript	2016-04-11 12:45:26.761531768 -0500
@@ -77,41 +77,39 @@
 def build(bld):
     bld.RECURSE('lib/replace')
 
-    if bld.env.standalone_talloc:
-        bld.env.PKGCONFIGDIR = '${LIBDIR}/pkgconfig'
-        bld.env.TALLOC_VERSION = VERSION
-        private_library = False
-
-        # should we also install the symlink to libtalloc1.so here?
-        bld.SAMBA_LIBRARY('talloc-compat1-%s' % (VERSION),
-                          'compat/talloc_compat1.c',
-                          public_deps='talloc',
-                          soname='libtalloc.so.1',
-                          pc_files=[],
-                          public_headers=[],
-                          enabled=bld.env.TALLOC_COMPAT1)
-
-        testsuite_deps = 'talloc'
-        if bld.CONFIG_SET('HAVE_PTHREAD'):
-            testsuite_deps += ' pthread'
-
-        bld.SAMBA_BINARY('talloc_testsuite',
-                         'testsuite_main.c testsuite.c',
-                         testsuite_deps,
-                         install=False)
-
-        bld.SAMBA_BINARY('talloc_test_magic_differs_helper',
-                         'test_magic_differs_helper.c',
-                         'talloc', install=False)
+    bld.env.PKGCONFIGDIR = '${LIBDIR}/pkgconfig'
+    bld.env.TALLOC_VERSION = VERSION
+    private_library = False
+
+    # should we also install the symlink to libtalloc1.so here?
+    bld.SAMBA_LIBRARY('talloc-compat1-%s' % (VERSION),
+                      'compat/talloc_compat1.c',
+                      public_deps='talloc',
+                      soname='libtalloc.so.1',
+                      pc_files=[],
+                      public_headers=[],
+                      enabled=bld.env.TALLOC_COMPAT1)
+
+    testsuite_deps = 'talloc'
+    if bld.CONFIG_SET('HAVE_PTHREAD'):
+        testsuite_deps += ' pthread'
+
+    bld.SAMBA_BINARY('talloc_testsuite',
+                     'testsuite_main.c testsuite.c',
+                     testsuite_deps,
+                     install=False)
+
+    bld.SAMBA_BINARY('talloc_test_magic_differs_helper',
+                     'test_magic_differs_helper.c',
+                     'talloc', install=False)
 
-    else:
-        private_library = True
 
     if not bld.CONFIG_SET('USING_SYSTEM_TALLOC'):
 
         bld.SAMBA_LIBRARY('talloc',
                           'talloc.c',
                           deps='replace',
+                          includes='.',
                           abi_directory='ABI',
                           abi_match='talloc* _talloc*',
                           hide_symbols=True,
