diff -u -r emacspeak-32.0.orig/servers/linux-espeak/Makefile emacspeak-32.0/servers/linux-espeak/Makefile
--- emacspeak-32.0.orig/servers/linux-espeak/Makefile	2009-08-26 12:19:01.000000000 -0500
+++ emacspeak-32.0/servers/linux-espeak/Makefile	2010-05-13 17:17:12.000000000 -0500
@@ -4,7 +4,7 @@
 PREFIX = /usr
 LIBPARENTDIR = ${PREFIX}/share/emacs/site-lisp
 LIBDIR =$(LIBPARENTDIR)/emacspeak/servers/linux-espeak
-TCL_VERSION = 8.4
+TCL_VERSION = 8.5
 TCL_INCLUDE= /usr/include/tcl$(TCL_VERSION)
 CXXFLAGS+= -g    -O2 -fPIC  -DPIC  -pedantic -ansi -Wall -Wno-long-long -I$(TCL_INCLUDE)
 
diff -u -r emacspeak-32.0.orig/servers/linux-outloud/Makefile emacspeak-32.0/servers/linux-outloud/Makefile
--- emacspeak-32.0.orig/servers/linux-outloud/Makefile	2009-08-28 16:07:35.000000000 -0500
+++ emacspeak-32.0/servers/linux-outloud/Makefile	2010-05-13 17:17:18.000000000 -0500
@@ -5,7 +5,7 @@
 LIBPARENTDIR = ${PREFIX}/share/emacs/site-lisp
 LIBDIR =$(LIBPARENTDIR)/emacspeak/servers/linux-outloud
 
-TCL_VERSION=8.4
+TCL_VERSION=8.5
 TCL_INCLUDE=/usr/include/tcl$(TCL_VERSION)
 CFLAGS+= -g    -O2 -fPIC  -DPIC  -pedantic -ansi -Wall  -Wno-long-long -I$(TCL_INCLUDE)
 CXXFLAGS+= $(CFLAGS)
diff -u -r emacspeak-32.0.orig/servers/software-dtk/Makefile emacspeak-32.0/servers/software-dtk/Makefile
--- emacspeak-32.0.orig/servers/software-dtk/Makefile	2009-08-28 16:07:35.000000000 -0500
+++ emacspeak-32.0/servers/software-dtk/Makefile	2010-05-13 17:17:46.000000000 -0500
@@ -2,7 +2,7 @@
 #
 LABEL=1.1
 DIST=software-dtk-$(LABEL)
-TCL_VERSION=8.4
+TCL_VERSION=8.5
 CFLAGS+=-O3 -fpic -pedantic -ansi -Wall -I/usr/include/tcl$(TCL_VERSION)
 CPPFLAGS+=-DSTDC_HEADERS=1 -Dclock_t=long -DRETSIGTYPE=void 
 INSTALL=install
