--- ./servers/software-dtk/Makefile.orig	2009-08-28 16:07:35.000000000 -0500
+++ ./servers/software-dtk/Makefile	2011-03-12 14:43:44.000000000 -0600
@@ -2,7 +2,7 @@
 #
 LABEL=1.1
 DIST=software-dtk-$(LABEL)
-TCL_VERSION=8.4
+TCL_VERSION=8.6
 CFLAGS+=-O3 -fpic -pedantic -ansi -Wall -I/usr/include/tcl$(TCL_VERSION)
 CPPFLAGS+=-DSTDC_HEADERS=1 -Dclock_t=long -DRETSIGTYPE=void 
 INSTALL=install
--- ./servers/linux-espeak/Makefile.orig	2009-08-26 12:19:01.000000000 -0500
+++ ./servers/linux-espeak/Makefile	2011-03-12 14:43:44.000000000 -0600
@@ -4,7 +4,7 @@
 PREFIX = /usr
 LIBPARENTDIR = ${PREFIX}/share/emacs/site-lisp
 LIBDIR =$(LIBPARENTDIR)/emacspeak/servers/linux-espeak
-TCL_VERSION = 8.4
+TCL_VERSION = 8.6
 TCL_INCLUDE= /usr/include/tcl$(TCL_VERSION)
 CXXFLAGS+= -g    -O2 -fPIC  -DPIC  -pedantic -ansi -Wall -Wno-long-long -I$(TCL_INCLUDE)
 
--- ./servers/linux-outloud/Makefile.orig	2010-11-23 12:57:29.000000000 -0600
+++ ./servers/linux-outloud/Makefile	2011-03-12 14:43:44.000000000 -0600
@@ -5,7 +5,7 @@
 LIBPARENTDIR = ${PREFIX}/share/emacs/site-lisp
 LIBDIR =$(LIBPARENTDIR)/emacspeak/servers/linux-outloud
 
-TCL_VERSION=8.4
+TCL_VERSION=8.6
 TCL_INCLUDE=/usr/include/tcl$(TCL_VERSION)
 #CFLAGS+= -g    -O2 -fPIC  -DPIC  -pedantic -ansi -Wall  -Wno-long-long -I$(TCL_INCLUDE)
 CFLAGS+= -m32 -g    -O2 -fPIC  -DPIC  -pedantic -ansi -Wall  -Wno-long-long -I$(TCL_INCLUDE)
