From aacd44860129372e5d4ff1494664317f18cbb615 Mon Sep 17 00:00:00 2001
From: Denys Vlasenko <vda.linux@googlemail.com>
Date: Sun, 17 Jun 2012 18:21:30 +0000
Subject: wget: fix wget-supports--P testsuite failure. Closes 4940

Signed-off-by: Denys Vlasenko <vda.linux@googlemail.com>
---
diff --git a/networking/wget.c b/networking/wget.c
index 5d58450..6d8f8a5 100644
--- a/networking/wget.c
+++ b/networking/wget.c
@@ -621,13 +621,11 @@ static void download_one_url(const char *url)
 		if (G.fname_out[0] == '/' || !G.fname_out[0])
 			G.fname_out = (char*)"index.html";
 		/* -P DIR is considered only if there was no -O FILE */
+		if (G.dir_prefix)
+			G.fname_out = fname_out_alloc = concat_path_file(G.dir_prefix, G.fname_out);
 		else {
-			if (G.dir_prefix)
-				G.fname_out = fname_out_alloc = concat_path_file(G.dir_prefix, G.fname_out);
-			else {
-				/* redirects may free target.path later, need to make a copy */
-				G.fname_out = fname_out_alloc = xstrdup(G.fname_out);
-			}
+			/* redirects may free target.path later, need to make a copy */
+			G.fname_out = fname_out_alloc = xstrdup(G.fname_out);
 		}
 	}
 #if ENABLE_FEATURE_WGET_STATUSBAR
--
cgit v0.9.0.1-2-gef13
