--- ./miscutils/eject.c.orig	2009-12-12 15:13:28.000000000 -0600
+++ ./miscutils/eject.c	2009-12-31 14:04:20.000000000 -0600
@@ -13,6 +13,10 @@
  * Most of the dirty work blatantly ripped off from cat.c =)
  */
 
+/* kludge for 2.6.32.2 /usr/include/scsi/scsi.h */
+#include <stdint.h>
+typedef uint8_t u8;
+
 #include <sys/mount.h>
 #include "libbb.h"
 /* Must be after libbb.h: they need size_t */
