--- ./CMakeLists.txt.orig	2024-02-10 21:59:22.000000000 -0600
+++ ./CMakeLists.txt	2025-03-15 16:56:50.180564786 -0500
@@ -72,17 +72,6 @@
 find_package(BISON REQUIRED)
 set_package_properties(BISON PROPERTIES PURPOSE "MapCSS parser for indoor map rendering.")
 
-if (NOT WIN32)
-    # try upstream cmake config first
-    set(protobuf_MODULE_COMPATIBLE ON)
-    find_package(Protobuf CONFIG)
-    if(NOT Protobuf_FOUND)
-    # fall back to cmake's protobuf module
-       find_package(Protobuf)
-    endif()
-    set_package_properties(Protobuf PROPERTIES TYPE OPTIONAL PURPOSE "Parsing of OSM PBF files.")
-endif()
-
 ecm_set_disabled_deprecation_versions(QT 5.15.2  KF 5.102.0)
 
 add_subdirectory(src)
