--- src/conf.sh.orig	Sat Aug 12 20:25:11 1995
+++ src/conf.sh	Sat Aug 12 20:25:45 1995
@@ -77,10 +77,10 @@
 rm -f a.d &&
 date >a.d &&
 chmod 0 a.d &&
-{ test -w a.d || cp /dev/null a.d 2>/dev/null; } && {
-	echo >&3 "$n$0: This command should not be run with superuser permissions."
-	exit 1
-}
+#{ test -w a.d || cp /dev/null a.d 2>/dev/null; } && {
+#	echo >&3 "$n$0: This command should not be run with superuser permissions."
+#	exit 1
+#}
 echo >&3 OK
 rm -f a.d || exit
 
