--- ./include/misc.h.orig	2017-07-24 20:34:11.000000000 -0500
+++ ./include/misc.h	2018-09-18 19:00:40.783069613 -0500
@@ -73,7 +73,7 @@
 int get_dma_info(dma_info_t *dma_info);
 void clean_after_dma_check(int fd, dma_info_t *dma_info);
 
-int valid_offset(int fd, loff_t offset);
+int valid_offset(int fd, off_t offset);
 unsigned long count_blocks(const char *filename, int blocksize);
 
 void print_how_far(FILE * fp, unsigned long *passed, unsigned long total,
--- ./include/reiserfs_lib.h.orig	2016-06-07 09:22:56.000000000 -0500
+++ ./include/reiserfs_lib.h	2018-09-18 19:00:40.786069613 -0500
@@ -297,8 +297,8 @@
 char *key_of_what(const struct reiserfs_key *key);
 int type_unknown(const struct reiserfs_key *key);
 void set_type(int format, struct reiserfs_key *key, int type);
-void set_offset(int format, struct reiserfs_key *key, loff_t offset);
-void set_type_and_offset(int format, struct reiserfs_key *key, loff_t offset,
+void set_offset(int format, struct reiserfs_key *key, off_t offset);
+void set_type_and_offset(int format, struct reiserfs_key *key, off_t offset,
 			 int type);
 
 typedef int (*check_unfm_func_t) (reiserfs_filsys_t , __u32);
