--- ./configure.ac.orig	2020-06-08 05:17:27.000000000 -0500
+++ ./configure.ac	2020-06-13 14:11:04.857950668 -0500
@@ -712,6 +712,7 @@
 	po/Makefile.in \
 	Make.xml.rules \
 	modules/Makefile \
+	modules/pam_chroot/Makefile modules/pam_console/Makefile modules/pam_postgresok/Makefile \
 	modules/pam_access/Makefile modules/pam_cracklib/Makefile \
         modules/pam_debug/Makefile modules/pam_deny/Makefile \
 	modules/pam_echo/Makefile modules/pam_env/Makefile \
--- ./modules/Makefile.am.orig	2020-06-08 05:17:27.000000000 -0500
+++ ./modules/Makefile.am	2020-06-13 14:12:28.614946035 -0500
@@ -53,6 +53,8 @@
 SUBDIRS := \
 	pam_access \
 	$(MAYBE_PAM_CRACKLIB) \
+	pam_chroot \
+	pam_console \
 	pam_debug \
 	pam_deny \
 	pam_echo \
@@ -76,6 +78,7 @@
 	$(MAYBE_PAM_NAMESPACE) \
 	pam_nologin \
 	pam_permit \
+	pam_postgresok \
 	pam_pwhistory \
 	$(MAYBE_PAM_RHOSTS) \
 	pam_rootok \
