diff -up Linux-PAM-1.6.0/configure.ac.redhat-modules Linux-PAM-1.6.0/configure.ac
--- Linux-PAM-1.6.0/configure.ac.redhat-modules	2024-01-23 13:16:34.854753145 +0100
+++ Linux-PAM-1.6.0/configure.ac	2024-01-23 13:17:52.855859922 +0100
@@ -774,6 +774,8 @@ AC_CONFIG_FILES([Makefile libpam/Makefil
 	po/Makefile.in \
 	Make.xml.rules \
 	modules/Makefile \
+	modules/pam_chroot/Makefile \
+	modules/pam_postgresok/Makefile \
 	modules/pam_access/Makefile \
 	modules/pam_canonicalize_user/Makefile \
 	modules/pam_debug/Makefile modules/pam_deny/Makefile \
diff -up Linux-PAM-1.6.0/modules/Makefile.am.redhat-modules Linux-PAM-1.6.0/modules/Makefile.am
--- Linux-PAM-1.6.0/modules/Makefile.am.redhat-modules	2024-01-17 11:29:36.000000000 +0100
+++ Linux-PAM-1.6.0/modules/Makefile.am	2024-01-23 13:16:34.855753147 +0100
@@ -48,6 +48,8 @@ SUBDIRS := \
 	pam_debug \
 	pam_deny \
 	pam_echo \
+	pam_chroot \
+	pam_postgresok \
 	pam_env \
 	pam_exec \
 	pam_faildelay \
