--- ./xmms/main.c.orig	2006-07-16 08:40:04.000000000 -0500
+++ ./xmms/main.c	2013-05-02 21:25:33.648185044 -0500
@@ -502,7 +502,7 @@
 	if (cfg.outputplugin == NULL)
 	{
 #ifdef HAVE_OSS
-		cfg.outputplugin = g_strdup_printf("%s/%s/libOSS.so", PLUGIN_DIR, plugin_dir_list[0]);
+		cfg.outputplugin = g_strdup_printf("%s/%s/libALSA.so", PLUGIN_DIR, plugin_dir_list[0]);
 #elif defined(sun)
 		cfg.outputplugin = g_strdup_printf("%s/%s/libSolaris.so", PLUGIN_DIR, plugin_dir_list[0]);
 #else
