diff -ur libcap-ng-0.7.8.orig/bindings/python3/Makefile.am libcap-ng-0.7.8/bindings/python3/Makefile.am
--- libcap-ng-0.7.8.orig/bindings/python3/Makefile.am	2016-07-24 17:47:34.000000000 +0200
+++ libcap-ng-0.7.8/bindings/python3/Makefile.am	2017-06-24 16:32:53.957501204 +0200
@@ -26,6 +26,8 @@
 LIBS = ${top_builddir}/src/libcap-ng.la
 SWIG_FLAGS = -python -py3 -modern
 SWIG_INCLUDES = ${AM_CPPFLAGS}
+# REMOVE SOMEDAY - without overriding this macro, it uses python2
+am__py_compile = PYTHON=$(PYTHON3) $(SHELL) $(py_compile)
 py3exec_PYTHON = capng.py
 py3exec_LTLIBRARIES = _capng.la
 py3exec_SOLIBRARIES = _capng.so
