--- ./src/common/util.c.orig	2009-08-16 04:40:16.000000000 -0500
+++ ./src/common/util.c	2013-05-13 22:18:22.364370023 -0500
@@ -39,7 +39,7 @@
 #include <errno.h>
 #include "xchat.h"
 #include "xchatc.h"
-#include <glib/gmarkup.h>
+#include <glib.h>
 #include <ctype.h>
 #include "util.h"
 #include "../../config.h"
--- ./src/common/text.c.orig	2010-05-29 21:14:41.000000000 -0500
+++ ./src/common/text.c	2013-05-13 22:17:58.028371777 -0500
@@ -28,7 +28,7 @@
 #include <sys/mman.h>
 
 #include "xchat.h"
-#include <glib/ghash.h>
+#include <glib.h>
 #include "cfgfiles.h"
 #include "chanopt.h"
 #include "plugin.h"
--- ./src/common/xchat.h.orig	2009-08-16 04:40:16.000000000 -0500
+++ ./src/common/xchat.h	2013-05-13 22:16:41.332377305 -0500
@@ -1,10 +1,6 @@
 #include "../../config.h"
 
-#include <glib/gslist.h>
-#include <glib/glist.h>
-#include <glib/gutils.h>
-#include <glib/giochannel.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <time.h>			/* need time_t */
 
 #ifndef XCHAT_H
--- ./src/common/servlist.c.orig	2010-05-16 02:24:26.000000000 -0500
+++ ./src/common/servlist.c	2013-05-13 22:17:29.900373805 -0500
@@ -24,7 +24,7 @@
 #include <unistd.h>
 
 #include "xchat.h"
-#include <glib/ghash.h>
+#include <glib.h>
 
 #include "cfgfiles.h"
 #include "fe.h"
