diff -up doxygen-1.9.1/src/docparser.cpp.me doxygen-1.9.1/src/docparser.cpp
--- doxygen-1.9.1/src/docparser.cpp.me	2021-02-10 10:52:09.481086282 +0100
+++ doxygen-1.9.1/src/docparser.cpp	2021-02-10 10:55:32.418781686 +0100
@@ -951,7 +951,7 @@ static void handlePendingStyleCommands(D
       children.append(new DocStyleChange(parent,g_nodeStack.count(),sc->style(),sc->tagName(),FALSE));
       g_initialStyleStack.push(sc);
       g_styleStack.pop();
-      sc = g_styleStack.top();
+      sc = !g_styleStack.isEmpty() ? g_styleStack.top() : 0;
     }
   }
 }
