diff -up doxygen-1.9.1/src/context.cpp.me doxygen-1.9.1/src/context.cpp
--- doxygen-1.9.1/src/context.cpp.me	2021-02-10 10:56:44.575251899 +0100
+++ doxygen-1.9.1/src/context.cpp	2021-02-10 10:57:54.522809720 +0100
@@ -4062,11 +4062,11 @@ class MemberContext::Private : public De
         s_inst.addProperty("nameWithContextFor",  &Private::nameWithContextFor);
         init=TRUE;
       }
-      if (md && !md->cookie()) { md->setCookie(new MemberContext::Private::Cachable(md)); }
+      if (!md->cookie()) { md->setCookie(new MemberContext::Private::Cachable(md)); }
 
       Cachable &cache = getCache();
       cache.propertyAttrs.reset(TemplateList::alloc());
-      if (md && md->isProperty())
+      if (md->isProperty())
       {
         if (md->isGettable())           cache.propertyAttrs->append("get");
         if (md->isPrivateGettable())    cache.propertyAttrs->append("private get");
@@ -4076,7 +4076,7 @@ class MemberContext::Private : public De
         if (md->isProtectedSettable())  cache.propertyAttrs->append("protected set");
       }
       cache.eventAttrs.reset(TemplateList::alloc());
-      if (md && md->isEvent())
+      if (md->isEvent())
       {
         if (md->isAddable())   cache.eventAttrs->append("add");
         if (md->isRemovable()) cache.eventAttrs->append("remove");
diff -up doxygen-1.9.1/src/docparser.cpp.me doxygen-1.9.1/src/docparser.cpp
--- doxygen-1.9.1/src/docparser.cpp.me	2021-02-10 10:58:11.491500540 +0100
+++ doxygen-1.9.1/src/docparser.cpp	2021-02-10 11:00:35.905512597 +0100
@@ -1524,7 +1524,7 @@ reparsetoken:
           {
             QCString scope;
             doctokenizerYYsetStateSetScope();
-            doctokenizerYYlex();
+            (void)doctokenizerYYlex();
             scope = g_token->name;
             g_context = scope;
             //printf("Found scope='%s'\n",scope.data());
diff -up doxygen-1.9.1/src/dotgroupcollaboration.cpp.me doxygen-1.9.1/src/dotgroupcollaboration.cpp
--- doxygen-1.9.1/src/dotgroupcollaboration.cpp.me	2021-02-10 11:01:10.588530954 +0100
+++ doxygen-1.9.1/src/dotgroupcollaboration.cpp	2021-02-10 11:02:54.216221350 +0100
@@ -309,6 +309,7 @@ void DotGroupCollaboration::Edge::write(
     {
       if (first) first=FALSE; else t << "\\n";
       t << DotNode::convertLabel(link->label);
+      count++;
     }
     if (count==maxLabels) t << "\\n...";
     t << "\"";
