diff -Naur tigervnc-1.8.0.orig/common/rfb/CConnection.cxx tigervnc-1.8.0/common/rfb/CConnection.cxx
--- tigervnc-1.8.0.orig/common/rfb/CConnection.cxx	2017-05-16 15:53:28.000000000 +0200
+++ tigervnc-1.8.0/common/rfb/CConnection.cxx	2018-05-15 09:44:02.021640000 +0200
@@ -138,6 +138,13 @@
   vlog.info("Server supports RFB protocol version %d.%d",
             server.majorVersion, server.minorVersion);
 
+  // Mark's Hack
+  if (server.minorVersion==16) {
+    server.setVersion(3,3);
+    vlog.info("Hack #1 for 3.16");
+    server.minorVersion=3;
+  }
+
   // The only official RFB protocol versions are currently 3.3, 3.7 and 3.8
   if (server.beforeVersion(3,3)) {
     vlog.error("Server gave unsupported RFB protocol version %d.%d",
