--- ./vncserver.orig	2021-11-22 13:18:49.000000000 -0600
+++ ./vncserver	2021-11-22 13:23:05.128071832 -0600
@@ -437,7 +437,7 @@
 	die "$prog: couldn't find \"$cmd\" on your PATH.\n";
     }
 
-    foreach $cmd ("/etc/X11/xinit/Xsession", "/etc/X11/Xsession") {
+    foreach $cmd ("/etc/X11/xinit/Xsession", "/etc/X11/Xsession", "$ENV{HOME}/.vnc/xstartup", "$ENV{HOME}/.xinitrc", "/etc/X11/xinit/xinitrc") {
         if (-x "$cmd") {
             $Xsession = $cmd;
             last;
