--- ./vncserver.orig	2019-12-20 01:02:02.000000000 -0600
+++ ./vncserver	2021-11-15 15:52:29.648106674 -0600
@@ -205,7 +205,7 @@
 $default_opts{geometry} = $geometry if ($geometry);
 $default_opts{depth} = $depth if ($depth);
 $default_opts{pixelformat} = $pixelformat if ($pixelformat);
-$default_opts{rfbwait} = 30000;
+#$default_opts{rfbwait} = 30000;
 $default_opts{rfbauth} = "$vncUserDir/passwd";
 $default_opts{rfbport} = $vncPort;
 $default_opts{fp} = $fontPath if ($fontPath);
