From 743b892d410d134de1b34022dfb5c2777017a54a Mon Sep 17 00:00:00 2001
From: Shi Pujin <shipujin.t@gmail.com>
Date: Thu, 14 Nov 2024 04:50:44 +0800
Subject: [PATCH] qtbase: add LoongArch support


diff --git a/qtbase/mkspecs/devices/common/linux_loongarch_device_post.conf b/qtbase/mkspecs/devices/common/linux_loongarch_device_post.conf
new file mode 100644
index 0000000000..85d5799da6
--- /dev/null
+++ b/qtbase/mkspecs/devices/common/linux_loongarch_device_post.conf
@@ -0,0 +1,2 @@
+
+include(linux_device_post.conf)
diff --git a/qtbase/mkspecs/devices/loongarch64-generic-g++/qmake.conf b/qtbase/mkspecs/devices/loongarch64-generic-g++/qmake.conf
new file mode 100644
index 0000000000..3fba1b4532
--- /dev/null
+++ b/qtbase/mkspecs/devices/loongarch64-generic-g++/qmake.conf
@@ -0,0 +1,10 @@
+#
+# Generic qmake configuration for building with g++ on loongarch64 devices.
+#
+# A minimal configure line could look something like this:
+# ./configure -device loongarch64-generic-g++ -device-option CROSS_COMPILE=loongarch64-slackware-linux-
+
+
+include(../common/linux_device_pre.conf)
+include(../common/linux_loongarch_device_post.conf)
+load(qt_config)
diff --git a/qtbase/mkspecs/devices/loongarch64-generic-g++/qplatformdefs.h b/qtbase/mkspecs/devices/loongarch64-generic-g++/qplatformdefs.h
new file mode 100644
index 0000000000..6a35ed45d5
--- /dev/null
+++ b/qtbase/mkspecs/devices/loongarch64-generic-g++/qplatformdefs.h
@@ -0,0 +1,40 @@
+/****************************************************************************
+**
+** Copyright (C) 2016 The Qt Company Ltd.
+** Contact: https://www.qt.io/licensing/
+**
+** This file is part of the qmake spec of the Qt Toolkit.
+**
+** $QT_BEGIN_LICENSE:LGPL$
+** Commercial License Usage
+** Licensees holding valid commercial Qt licenses may use this file in
+** accordance with the commercial license agreement provided with the
+** Software or, alternatively, in accordance with the terms contained in
+** a written agreement between you and The Qt Company. For licensing terms
+** and conditions see https://www.qt.io/terms-conditions. For further
+** information use the contact form at https://www.qt.io/contact-us.
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 3 as published by the Free Software
+** Foundation and appearing in the file LICENSE.LGPL3 included in the
+** packaging of this file. Please review the following information to
+** ensure the GNU Lesser General Public License version 3 requirements
+** will be met: https://www.gnu.org/licenses/lgpl-3.0.html.
+**
+** GNU General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU
+** General Public License version 2.0 or (at your option) the GNU General
+** Public license version 3 or any later version approved by the KDE Free
+** Qt Foundation. The licenses are as published by the Free Software
+** Foundation and appearing in the file LICENSE.GPL2 and LICENSE.GPL3
+** included in the packaging of this file. Please review the following
+** information to ensure the GNU General Public License requirements will
+** be met: https://www.gnu.org/licenses/gpl-2.0.html and
+** https://www.gnu.org/licenses/gpl-3.0.html.
+**
+** $QT_END_LICENSE$
+**
+****************************************************************************/
+
+#include "../../linux-g++/qplatformdefs.h"
diff --git a/qtbase/mkspecs/linux-loongarch64-gnu-g++/qmake.conf b/qtbase/mkspecs/linux-loongarch64-gnu-g++/qmake.conf
new file mode 100644
index 0000000000..092081c99f
--- /dev/null
+++ b/qtbase/mkspecs/linux-loongarch64-gnu-g++/qmake.conf
@@ -0,0 +1,24 @@
+#
+# qmake configuration for building with loongarch64-slackware-linux-g++
+#
+
+MAKEFILE_GENERATOR      = UNIX
+CONFIG                 += incremental
+QMAKE_INCREMENTAL_STYLE = sublib
+
+include(../common/linux.conf)
+include(../common/gcc-base-unix.conf)
+include(../common/g++-unix.conf)
+
+# modifications to g++.conf
+QMAKE_CC                = loongarch64-slackware-linux-gcc
+QMAKE_CXX               = loongarch64-slackware-linux-g++
+QMAKE_LINK              = loongarch64-slackware-linux-g++
+QMAKE_LINK_SHLIB        = loongarch64-slackware-linux-g++
+
+# modifications to linux.conf
+QMAKE_AR                = loongarch64-slackware-linux-ar cqs
+QMAKE_OBJCOPY           = loongarch64-slackware-linux-objcopy
+QMAKE_NM                = loongarch64-slackware-linux-nm -P
+QMAKE_STRIP             = loongarch64-slackware-linux-strip
+load(qt_config)
diff --git a/qtbase/mkspecs/linux-loongarch64-gnu-g++/qplatformdefs.h b/qtbase/mkspecs/linux-loongarch64-gnu-g++/qplatformdefs.h
new file mode 100644
index 0000000000..2697a75e44
--- /dev/null
+++ b/qtbase/mkspecs/linux-loongarch64-gnu-g++/qplatformdefs.h
@@ -0,0 +1,40 @@
+/****************************************************************************
+**
+** Copyright (C) 2017 The Qt Company Ltd.
+** Contact: https://www.qt.io/licensing/
+**
+** This file is part of the qmake spec of the Qt Toolkit.
+**
+** $QT_BEGIN_LICENSE:LGPL$
+** Commercial License Usage
+** Licensees holding valid commercial Qt licenses may use this file in
+** accordance with the commercial license agreement provided with the
+** Software or, alternatively, in accordance with the terms contained in
+** a written agreement between you and The Qt Company. For licensing terms
+** and conditions see https://www.qt.io/terms-conditions. For further
+** information use the contact form at https://www.qt.io/contact-us.
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 3 as published by the Free Software
+** Foundation and appearing in the file LICENSE.LGPL3 included in the
+** packaging of this file. Please review the following information to
+** ensure the GNU Lesser General Public License version 3 requirements
+** will be met: https://www.gnu.org/licenses/lgpl-3.0.html.
+**
+** GNU General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU
+** General Public License version 2.0 or (at your option) the GNU General
+** Public license version 3 or any later version approved by the KDE Free
+** Qt Foundation. The licenses are as published by the Free Software
+** Foundation and appearing in the file LICENSE.GPL2 and LICENSE.GPL3
+** included in the packaging of this file. Please review the following
+** information to ensure the GNU General Public License requirements will
+** be met: https://www.gnu.org/licenses/gpl-2.0.html and
+** https://www.gnu.org/licenses/gpl-3.0.html.
+**
+** $QT_END_LICENSE$
+**
+****************************************************************************/
+
+#include "../linux-g++/qplatformdefs.h"
diff --git a/qtbase/src/3rdparty/double-conversion/include/double-conversion/utils.h b/qtbase/src/3rdparty/double-conversion/include/double-conversion/utils.h
index 70e697ca00..93da575807 100644
--- a/qtbase/src/3rdparty/double-conversion/include/double-conversion/utils.h
+++ b/qtbase/src/3rdparty/double-conversion/include/double-conversion/utils.h
@@ -93,6 +93,7 @@ int main(int argc, char** argv) {
 #if defined(_M_X64) || defined(__x86_64__) || \
     defined(__ARMEL__) || defined(__avr32__) || defined(_M_ARM) || defined(_M_ARM64) || \
     defined(__hppa__) || defined(__ia64__) || \
+    defined(__loongarch__) || \
     defined(__mips__) || \
     defined(__powerpc__) || defined(__ppc__) || defined(__ppc64__) || \
     defined(_POWER) || defined(_ARCH_PPC) || defined(_ARCH_PPC64) || \
diff --git a/qtbase/src/corelib/global/archdetect.cpp b/qtbase/src/corelib/global/archdetect.cpp
index 1d00b7f5a5..74f3c8fc86 100644
--- a/qtbase/src/corelib/global/archdetect.cpp
+++ b/qtbase/src/corelib/global/archdetect.cpp
@@ -59,6 +59,8 @@
 #  define ARCH_PROCESSOR "x86_64"
 #elif defined(Q_PROCESSOR_IA64)
 #  define ARCH_PROCESSOR "ia64"
+#elif defined(Q_PROCESSOR_LOONGARCH_64)
+#  define ARCH_PROCESSOR "loongarch64"
 #elif defined(Q_PROCESSOR_MIPS_64)
 #  define ARCH_PROCESSOR "mips64"
 #elif defined(Q_PROCESSOR_MIPS)
diff --git a/qtbase/src/corelib/global/qprocessordetection.h b/qtbase/src/corelib/global/qprocessordetection.h
index 8d65720850..ccdfdc58fd 100644
--- a/qtbase/src/corelib/global/qprocessordetection.h
+++ b/qtbase/src/corelib/global/qprocessordetection.h
@@ -223,6 +223,17 @@
 #  define Q_PROCESSOR_WORDSIZE   8
 // Q_BYTE_ORDER not defined, use endianness auto-detection
 
+/*
+    LoongArch family, known variants: 64-bit
+
+    LoongArch is little-endian.
+*/
+#elif defined(__loongarch__)
+#  define Q_PROCESSOR_LOONGARCH
+#  define Q_PROCESSOR_LOONGARCH_64
+#  define Q_BYTE_ORDER Q_LITTLE_ENDIAN
+#  define Q_PROCESSOR_WORDSIZE   8
+
 /*
     MIPS family, known revisions: I, II, III, IV, 32, 64
 
-- 
2.41.0

