--- ./qtdeclarative/src/qml/jsruntime/qv4regexp_p.h.orig	2020-10-27 03:02:12.000000000 -0500
+++ ./qtdeclarative/src/qml/jsruntime/qv4regexp_p.h	2021-08-05 13:38:38.366445122 -0500
@@ -57,7 +57,7 @@
 #include <wtf/FastAllocBase.h>
 #include <wtf/BumpPointerAllocator.h>
 
-#include <limits.h>
+#include <limits>
 
 #include <yarr/Yarr.h>
 #include <yarr/YarrInterpreter.h>
--- ./qtdeclarative/src/qmldebug/qqmlprofilerevent_p.h.orig	2020-10-27 03:02:12.000000000 -0500
+++ ./qtdeclarative/src/qmldebug/qqmlprofilerevent_p.h	2021-08-05 13:39:39.671441731 -0500
@@ -48,6 +48,7 @@
 #include <QtCore/qmetatype.h>
 
 #include <initializer_list>
+#include <limits>
 #include <type_traits>
 
 //
