--- ./qtbase/src/concurrent/qtconcurrentthreadengine.h.orig	2020-10-27 03:02:11.000000000 -0500
+++ ./qtbase/src/concurrent/qtconcurrentthreadengine.h	2021-08-05 14:20:37.774305766 -0500
@@ -247,8 +247,9 @@
 class ThreadEngineStarter<void> : public ThreadEngineStarterBase<void>
 {
 public:
-    ThreadEngineStarter<void>(ThreadEngine<void> *_threadEngine)
-    :ThreadEngineStarterBase<void>(_threadEngine) {}
+    ThreadEngineStarter(ThreadEngine<void> *_threadEngine)
+        : ThreadEngineStarterBase<void>(_threadEngine) {}
+
 
     void startBlocking()
     {
