--- ./knetworkconf/backends/platform.pl.cmake.orig	2010-07-06 01:54:42.000000000 -0500
+++ ./knetworkconf/backends/platform.pl.cmake	2011-03-13 16:43:44.000000000 -0500
@@ -93,6 +93,10 @@
   "slackware-9.1.0" => "Slackware 9.1.0",
   "slackware-10.0.0" => "Slackware 10.0.0",
   "slackware-10.1.0" => "Slackware 10.1.0",
+  "slackware-12.2.0" => "Slackware 12.2.0",
+  "slackware-13.0.0.0.0" => "Slackware 13.0.0.0.0",
+  "slackware-13.1.0" => "Slackware 13.1.0",
+  "slackware-13.37.0" => "Slackware 13.37.0",
   "freebsd-4"       => "FreeBSD 4",
   "freebsd-5"       => "FreeBSD 5",
   "freebsd-6"       => "FreeBSD 6",
--- ./knetworkconf/backends/network-conf.cmake.orig	2010-07-06 01:54:42.000000000 -0500
+++ ./knetworkconf/backends/network-conf.cmake	2011-03-13 16:43:13.000000000 -0500
@@ -77,7 +77,9 @@
               "mandriva-2006.1", "mandriva-2007.0", "mandriva-2007.1", "mandriva-2008.0", "mandriva-2008.1",
 	      "mandriva-2009.0", "mandriva-2009.1", "mandriva-2010.0",
               "vine-3.0", "vine-3.1",
-              "slackware-9.1.0", "slackware-10.0.0", "slackware-10.1.0",
+              "slackware-9.1.0", "slackware-10.0.0", "slackware-10.1.0", "slackware-12.2.0", "slackware-13.0.0.0.0",
+              "slackware-13.1.0",
+              "slackware-13.37.0",
 	      "freebsd-5", "freebsd-6", "freebsd-7", "freebsd-8", "freebsd-9",
               "gentoo", "vlos-1.2", "ark");
 
--- ./knetworkconf/backends/service.pl.cmake.orig	2010-07-06 01:54:42.000000000 -0500
+++ ./knetworkconf/backends/service.pl.cmake	2011-03-13 16:42:51.000000000 -0500
@@ -281,6 +281,10 @@
      "slackware-9.1.0" => "slackware-9.1.0",
      "slackware-10.0.0" => "slackware-9.1.0",
      "slackware-10.1.0" => "slackware-9.1.0",
+     "slackware-12.2.0" => "slackware-9.1.0",
+     "slackware-13.0.0.0.0" => "slackware-9.1.0",
+     "slackware-13.1.0" => "slackware-9.1.0",
+     "slackware-13.37.0" => "slackware-9.1.0",
 
      "gentoo"         => "gentoo",
      "vlos-1.2"       => "gentoo",
@@ -317,7 +321,11 @@
                       "nonetwork" => "RECOVER"
                      },
      "freebsd-5"  => {"rc" => "GRAPHICAL" },
-     "slackware-9.1.0" => {"4" => "GRAPHICAL" }
+     "slackware-9.1.0" => {"0" => "HALT",
+                           "3" => "NONE",
+                           "4" => "GRAPHICAL",
+                           "6" => "REBOOT"
+                          }
     );
 
   $distro = $dist_map{$gst_dist};
--- ./knetworkconf/backends/network.pl.cmake.orig	2010-07-06 01:54:42.000000000 -0500
+++ ./knetworkconf/backends/network.pl.cmake	2011-03-13 16:42:24.000000000 -0500
@@ -528,6 +528,10 @@
        "slackware-9.1.0" => "redhat-6.2",
        "slackware-10.0.0" => "redhat-6.2",
        "slackware-10.1.0" => "redhat-6.2",
+       "slackware-12.2.0" => "redhat-6.2",
+       "slackware-13.0.0.0.0" => "redhat-6.2",
+       "slackware-13.1.0" => "redhat-6.2",
+       "slackware-13.37.0" => "redhat-6.2",
        "gentoo"       => "debian-2.2",
        "vlos-1.2"     => "debian-2.2",
        "freebsd-5"    => "debian-2.2",
@@ -1833,6 +1837,10 @@
 	  "slackware-9.1.0" => \&gst_network_deb22_get_file,
 	  "slackware-10.0.0" => \&gst_network_deb22_get_file,
 	  "slackware-10.1.0" => \&gst_network_deb22_get_file,
+	  "slackware-12.2.0" => \&gst_network_deb22_get_file,
+	  "slackware-13.0.0.0.0" => \&gst_network_deb22_get_file,
+	  "slackware-13.1.0" => \&gst_network_deb22_get_file,
+	  "slackware-13.37.0" => \&gst_network_deb22_get_file,
 	  "gentoo"       => \&gst_network_deb22_get_file,
 	  "freebsd-5"    => \&gst_network_deb22_get_file,
 	  "freebsd-6"    => \&gst_network_deb22_get_file,
@@ -3958,6 +3966,10 @@
 	  "slackware-9.1.0" => "",
 	  "slackware-10.0.0" => "",
 	  "slackware-10.1.0" => "",
+	  "slackware-12.2.0" => "",
+	  "slackware-13.0.0.0.0" => "",
+	  "slackware-13.1.0" => "",
+	  "slackware-13.37.0" => "",
 	  "gentoo"       => "",
 	  "vlos-1.2"     => "",
 	  "freebsd-5"    => "",
@@ -4109,6 +4121,10 @@
 	  "slackware-9.1.0" => "slackware-9.1.0",
 	  "slackware-10.0.0" => "slackware-9.1.0",
 	  "slackware-10.1.0" => "slackware-9.1.0",
+	  "slackware-12.2.0" => "slackware-9.1.0",
+	  "slackware-13.0.0.0.0" => "slackware-9.1.0",
+	  "slackware-13.1.0" => "slackware-9.1.0",
+	  "slackware-13.37.0" => "slackware-9.1.0",
 	  "gentoo"       => "gentoo",
 	  "vlos-1.2"     => "vlos-1.2",
 	  "freebsd-5"    => "freebsd-5",
@@ -4395,6 +4411,7 @@
        [ "hostname",      \&gst_parse_fq_hostname, HOSTNAME ],
        [ "nameserver",    \&gst_parse_split_all_unique_hash_comment, RESOLV_CONF,  "nameserver", "[ \t]+" ],
        [ "searchdomain",  \&gst_parse_split_first_array_unique, RESOLV_CONF,  "search", "[ \t]+", "[ \t]+" ],
+       [ "domain",        \&gst_parse_fq_domain,   HOSTNAME ],
        [ "domain",        \&gst_parse_split_first_str,   RESOLV_CONF,  "domain", "[ \t]+" ],
        [ "order",         \&gst_parse_split_first_array, HOST_CONF,    "order", "[ \t]+", ",[ \t]*" ],
        [ "hostmatch",     \&gst_parse_split_first_bool,  HOST_CONF,    "multi", "[ \t]+" ],
@@ -4611,6 +4628,10 @@
 	  "slackware-9.1.0" => "slackware-9.1.0",
 	  "slackware-10.0.0" => "slackware-9.1.0",
 	  "slackware-10.1.0" => "slackware-9.1.0",
+	  "slackware-12.2.0" => "slackware-9.1.0",
+	  "slackware-13.0.0.0.0" => "slackware-9.1.0",
+	  "slackware-13.1.0" => "slackware-9.1.0",
+	  "slackware-13.37.0" => "slackware-9.1.0",
 	  "gentoo"       => "gentoo",
 	  "vlos-1.2"     => "vlos-1.2",
 	  "freebsd-5"    => "freebsd-5",
@@ -5508,6 +5529,10 @@
 	  "slackware-9.1.0" => "slackware-9.1.0",
 	  "slackware-10.0.0" => "slackware-9.1.0",
 	  "slackware-10.1.0" => "slackware-9.1.0",
+	  "slackware-12.2.0" => "slackware-9.1.0",
+	  "slackware-13.0.0.0.0" => "slackware-9.1.0",
+	  "slackware-13.1.0" => "slackware-9.1.0",
+	  "slackware-13.37.0" => "slackware-9.1.0",
 	  "gentoo"       => "gentoo",
 	  "vlos-1.2"     => "vlos-1.2",
 	  "freebsd-5"    => "freebsd-5",
@@ -5781,6 +5806,7 @@
        [ "hostname",      \&gst_replace_fq_hostname, HOSTNAME, "%hostname%", "%domain%" ],
        [ "hostname",      \&gst_network_run_hostname ],
        [ "gateway",       \&gst_replace_rcinet1conf_global, RC_INET_CONF, GATEWAY ],
+       [ "domain",        \&gst_parse_fq_domain,   HOSTNAME ],
        [ "domain",        \&gst_replace_join_first_str,   RESOLV_CONF, "domain", "[ \t]+" ],
        [ "nameserver",    \&gst_replace_join_all,         RESOLV_CONF, "nameserver", "[ \t]+" ],
        [ "searchdomain",  \&gst_replace_join_first_array, RESOLV_CONF, "search", "[ \t]+", "[ \t]+" ],
@@ -5960,6 +5986,10 @@
 	  "slackware-9.1.0" => "slackware-9.1.0",
 	  "slackware-10.0.0" => "slackware-9.1.0",
 	  "slackware-10.1.0" => "slackware-9.1.0",
+	  "slackware-12.2.0" => "slackware-9.1.0",
+	  "slackware-13.0.0.0.0" => "slackware-9.1.0",
+	  "slackware-13.1.0" => "slackware-9.1.0",
+	  "slackware-13.37.0" => "slackware-9.1.0",
 	  "gentoo"       => "gentoo",
 	  "vlos-1.2"     => "vlos-1.2",
 	  "freebsd-5"    => "freebsd-5",
