From 181d1a670a06bb3fecbd7ad990f222d6e2285dc9 Mon Sep 17 00:00:00 2001
From: Mike Sharov <msharov@users.sourceforge.net>
Date: Mon, 12 Apr 2021 14:56:31 -0400
Subject: [PATCH] Explicitly use ncursesw package. Fixes #55.

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index 9831da9..8e0ab5b 100755
--- a/configure
+++ b/configure
@@ -34,7 +34,7 @@ seds=[s/^#undef \(USE_UNSUPPORTED_AND_BROKEN_CODE\)/#define \1/]
 progs="CC=gcc CC=clang CC=cc INSTALL=install MSGFMT=msgfmt"
 
 # Libs found using pkg-config
-pkgs="libcurl libxml-2.0 libcrypto ncurses"
+pkgs="libcurl libxml-2.0 libcrypto ncursesw"
 # Default pkg flags to substitute when pkg-config is not found
 pkg_libs="-lcurl -lxml2 -lcrypto -lncursesw"
 pkg_cflags="-I\/usr\/include\/libxml2"
