From 1b88d708109272e6d65d57840fdb1dfadf7d9cec Mon Sep 17 00:00:00 2001
From: Mike Sharov <msharov@users.sourceforge.net>
Date: Mon, 12 Apr 2021 14:54:53 -0400
Subject: [PATCH] Fix configure --with-debug

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index 608f772..9831da9 100755
--- a/configure
+++ b/configure
@@ -15,7 +15,7 @@ components="
 {
 name=[with-debug]
 desc=[	Compile for debugging]
-seds=[s/^#\(DEBUG\)/\1/]
+seds=[s/^#\(debug\)/\1/]
 }{
 name=[with-native]
 desc=[	Use -march=native]
