--- ./Config.mk.orig	2018-05-07 16:40:21.596003821 -0500
+++ ./Config.mk	2018-05-07 16:45:44.398003832 -0500
@@ -9,10 +9,10 @@
 INSTALLSCR	:= ${INSTALL} -D -p -m 755
 INSTALLDATA	:= ${INSTALL} -D -p -m 644
 
-PREFIX		:= /usr
+PREFIX		:= /tmp/package-snownews/usr
 BINDIR		:= ${PREFIX}/bin
 LOCALEPATH	:= ${PREFIX}/share/locale
-MANPATH		:= /usr/man
+MANPATH		:= ${PREFIX}/man
 
 CFLAGS		:= -Wall -Wextra -Os -std=c11 -D_GNU_SOURCE -D_DEFAULT_SOURCE -I/usr/include/libxml2
 LDFLAGS		:= -lcrypto -lxml2 -lncurses -lz -lm
