--- ./Config.mk.orig	2018-11-07 18:54:52.571283011 -0600
+++ ./Config.mk	2018-11-07 18:56:20.898274760 -0600
@@ -17,10 +17,10 @@
 
 ################ Destination #########################################
 
-PREFIX		:= /usr
+PREFIX		:= /tmp/package-snownews/usr
 BINDIR		:= ${PREFIX}/bin
 LOCALEPATH	:= ${PREFIX}/share/locale
-MANPATH		:= /usr/man
+MANPATH		:= ${PREFIX}/man
 BUILDDIR	:= /tmp/root/make/${NAME}
 PKGDIR		:= 
 O		:= .o/
