--- ./src/engine/zoom.c.orig	2013-11-01 02:54:21.000000000 -0500
+++ ./src/engine/zoom.c	2015-04-05 00:38:10.332028884 -0500
@@ -933,6 +933,7 @@
 }
 
 #ifndef USE_i386ASM
+REGISTERS(0)
 static /*INLINE */ void
 moveoldpoints(void /*@unused@ */ *data1,
 	      struct taskinfo /*@unused@ */ *task,
--- ./src/engine/zoomd.c.orig	2013-11-01 02:54:21.000000000 -0500
+++ ./src/engine/zoomd.c	2015-04-05 00:38:55.436032139 -0500
@@ -280,7 +280,8 @@
 #endif
 #undef bpp1
 
-static INLINE void fillline(int line)
+REGISTERS(0)
+static void fillline(int line)
 {
     register unsigned char *RESTRICT vbuff = cimage.currlines[line];
     const struct filltable *RESTRICT table = (struct filltable *) tmpdata;
