--- ./kioslave/info/kde-info2html.orig	2014-10-31 00:50:20.000000000 +0100
+++ ./kioslave/info/kde-info2html	2018-11-01 18:30:29.718723253 +0100
@@ -172,7 +172,7 @@
 		next if $looking && !/\* Menu/;
 		$looking = 0;
 		my @item = &ParseMenuItem($_,'dir');
-		if (!defined(@item)) { next }
+        if (!@item) { next }
     		my ($MenuLinkTag, $MenuLinkFile, $MenuLinkRef, $MenuLinkText) = @item;
 		if ($MenuLinkRef eq $FileName) {
 			&Redirect($MenuLinkFile, $MenuLinkTag);
@@ -430,7 +430,7 @@
 sub MenuItem2HTML {
     my ($Line, $BaseInfoFile) = @_;
     my @parse_results = &ParseMenuItem($Line, $BaseInfoFile);
-    if (!defined (@parse_results)) { return $Line; }
+    if (!@parse_results) { return $Line; }
     my ($MenuLinkTag, $MenuLinkFile, $MenuLinkRef, $MenuLinkText) = @parse_results;
     #-- produce a HTML line
     return "<tr class=\"infomenutr\"><td class=\"infomenutd\" style=\"width:30%\"><ul><li><a href=\"info:/$MenuLinkFile/$MenuLinkTag\">$MenuLinkRef</a></ul></td><td class=\"infomenutd\">$MenuLinkText";
