--- ./solid-networkstatus/kded/CMakeLists.txt.orig	2014-10-30 18:50:20.000000000 -0500
+++ ./solid-networkstatus/kded/CMakeLists.txt	2015-10-24 00:10:47.043649546 -0500
@@ -21,6 +21,7 @@
         MESSAGE(STATUS "Adding in-process NetworkManager service-wart to kded module")
         set(kded_networkstatus_PART_SRCS ${kded_networkstatus_PART_SRCS} networkmanagerstatus.cpp)
         include_directories(${NETWORKMANAGER_INCLUDE_DIRS})
+        include_directories(${NM-UTIL_INCLUDE_DIRS})
         add_definitions(-DNM_BACKEND_ENABLED)
     endif(NM_0_7)
 endif(NETWORKMANAGER_FOUND)
