--- ./Makefile.orig	2004-10-11 15:16:15.000000000 -0700
+++ ./Makefile	2005-07-25 00:18:46.000000000 -0700
@@ -17,7 +17,7 @@
 ## Uncomment this to build tools using static version of the library
 ## Mostly useful for embedded platforms without ldd, or to create
 ## a local version (non-root).
-# BUILD_STATIC = y
+BUILD_STATIC = y
 
 ## Uncomment this to build without using libm (less efficient)
 ## This is mostly useful for embedded platforms without maths.
