diff -Nur ca-certificates-20090814.orig/Makefile ca-certificates-20090814/Makefile
--- ca-certificates-20090814.orig/Makefile	2009-07-08 16:18:57.000000000 -0500
+++ ca-certificates-20090814/Makefile	2009-12-17 00:48:14.499550845 -0600
@@ -17,7 +17,7 @@
 
 install:
 	for dir in $(SUBDIRS); do \
-	  mkdir $(DESTDIR)/$(CERTSDIR)/$$dir; \
+	  mkdir -p $(DESTDIR)/$(CERTSDIR)/$$dir; \
 	  $(MAKE) -C $$dir install CERTSDIR=$(DESTDIR)/$(CERTSDIR)/$$dir; \
 	done
 	for dir in sbin; do \
diff -Nur ca-certificates-20090814.orig/brasil.gov.br/Makefile ca-certificates-20090814/brasil.gov.br/Makefile
--- ca-certificates-20090814.orig/brasil.gov.br/Makefile	2007-02-02 00:23:19.000000000 -0600
+++ ca-certificates-20090814/brasil.gov.br/Makefile	2009-12-17 00:48:14.499550845 -0600
@@ -8,5 +8,6 @@
 
 install:
 	for p in *.crt; do \
+	 mkdir -p $(CERTSDIR) ; \
 	 install -m 644 $$p $(CERTSDIR)/$$p ; \
 	done
diff -Nur ca-certificates-20090814.orig/cacert.org/Makefile ca-certificates-20090814/cacert.org/Makefile
--- ca-certificates-20090814.orig/cacert.org/Makefile	2009-07-08 16:19:37.000000000 -0500
+++ ca-certificates-20090814/cacert.org/Makefile	2009-12-17 00:48:14.499550845 -0600
@@ -9,5 +9,6 @@
 
 install:
 	cat root.crt class3.crt > cacert.org.crt
+	mkdir -p $(CERTSDIR)
 	install -m 644 cacert.org.crt $(CERTSDIR)/cacert.org.crt
 
diff -Nur ca-certificates-20090814.orig/gouv.fr/Makefile ca-certificates-20090814/gouv.fr/Makefile
--- ca-certificates-20090814.orig/gouv.fr/Makefile	2008-06-23 13:57:45.000000000 -0500
+++ ca-certificates-20090814/gouv.fr/Makefile	2009-12-17 00:48:14.500550426 -0600
@@ -8,6 +8,7 @@
 
 install:
 	for p in *.crt; do \
+	 mkdir -p $(CERTSDIR) ; \
 	 install -m 644 $$p $(CERTSDIR)/$$p ; \
 	done
 
diff -Nur ca-certificates-20090814.orig/mozilla/Makefile ca-certificates-20090814/mozilla/Makefile
--- ca-certificates-20090814.orig/mozilla/Makefile	2009-06-24 05:56:54.000000000 -0500
+++ ca-certificates-20090814/mozilla/Makefile	2009-12-17 00:48:14.500550426 -0600
@@ -10,5 +10,6 @@
 
 install:
 	for p in *.crt; do \
+	 mkdir -p $(CERTSDIR) ; \
 	 install -m 644 $$p $(CERTSDIR)/$$p ; \
 	done
diff -Nur ca-certificates-20090814.orig/sbin/Makefile ca-certificates-20090814/sbin/Makefile
--- ca-certificates-20090814.orig/sbin/Makefile	2007-02-02 00:23:19.000000000 -0600
+++ ca-certificates-20090814/sbin/Makefile	2009-12-17 00:49:05.976571309 -0600
@@ -8,4 +8,5 @@
 clean:
 
 install:
+	mkdir -p $(DESTDIR)/usr/sbin
 	install -m755 update-ca-certificates $(DESTDIR)/usr/sbin/
diff -Nur ca-certificates-20090814.orig/signet.pl/Makefile ca-certificates-20090814/signet.pl/Makefile
--- ca-certificates-20090814.orig/signet.pl/Makefile	2007-02-02 00:23:19.000000000 -0600
+++ ca-certificates-20090814/signet.pl/Makefile	2009-12-17 00:48:14.501550915 -0600
@@ -8,5 +8,6 @@
 
 install:
 	for p in *.crt; do \
+	 mkdir -p $(CERTSDIR) ; \
 	 install -m 644 $$p $(CERTSDIR)/$$p ; \
 	done
diff -Nur ca-certificates-20090814.orig/spi-inc.org/Makefile ca-certificates-20090814/spi-inc.org/Makefile
--- ca-certificates-20090814.orig/spi-inc.org/Makefile	2007-02-02 00:23:19.000000000 -0600
+++ ca-certificates-20090814/spi-inc.org/Makefile	2009-12-17 00:48:14.501550915 -0600
@@ -8,6 +8,7 @@
 
 install:
 	for p in *.crt; do \
+	 mkdir -p $(CERTSDIR) ; \
 	 install -m 644 $$p $(CERTSDIR)/$$p ; \
 	done
 
diff -Nur ca-certificates-20090814.orig/telesec.de/Makefile ca-certificates-20090814/telesec.de/Makefile
--- ca-certificates-20090814.orig/telesec.de/Makefile	2008-03-19 09:03:09.000000000 -0500
+++ ca-certificates-20090814/telesec.de/Makefile	2009-12-17 00:48:14.566586954 -0600
@@ -8,6 +8,7 @@
 
 install:
 	for p in *.crt; do \
+	 mkdir -p $(CERTSDIR) ; \
 	 install -m 644 $$p $(CERTSDIR)/$$p ; \
 	done
 
