diff -Nur ca-certificates-20090814.orig/Makefile ca-certificates-20090814/Makefile
--- ca-certificates-20090814.orig/Makefile	2009-07-08 16:18:57.000000000 -0500
+++ ca-certificates-20090814/Makefile	2009-12-17 00:44:30.540551055 -0600
@@ -3,7 +3,7 @@
 #
 
 CERTSDIR = /usr/share/ca-certificates
-SUBDIRS = spi-inc.org debconf.org mozilla cacert.org brasil.gov.br signet.pl telesec.de gouv.fr
+SUBDIRS = spi-inc.org mozilla cacert.org brasil.gov.br signet.pl telesec.de gouv.fr
 
 all:
 	for dir in $(SUBDIRS); do \
