diff -Nur ca-certificates-20090814.orig/sbin/update-ca-certificates ca-certificates-20090814/sbin/update-ca-certificates
--- ca-certificates-20090814.orig/sbin/update-ca-certificates	2009-07-08 16:23:12.000000000 -0500
+++ ca-certificates-20090814/sbin/update-ca-certificates	2009-12-17 00:42:41.760550636 -0600
@@ -148,9 +148,7 @@
 
 HOOKSDIR=/etc/ca-certificates/update.d
 echo -n "Running hooks in $HOOKSDIR...."
-VERBOSE_ARG=
-[ "$verbose" = 0 ] || VERBOSE_ARG=--verbose
-eval run-parts $VERBOSE_ARG --test -- $HOOKSDIR | while read hook
+eval run-parts $HOOKSDIR | while read hook
 do
   ( cat $ADDED
     cat $REMOVED ) | $hook || echo E: $hook exited with code $?.
