diff -u -r newspost-2.1.1.orig/base/newspost.c newspost-2.1.1/base/newspost.c
--- newspost-2.1.1.orig/base/newspost.c	2003-04-23 10:33:23.000000000 -0500
+++ newspost-2.1.1/base/newspost.c	2018-02-02 16:14:51.133929233 -0600
@@ -342,7 +342,7 @@
 	file = fopen(filename, "r");
 	if (file != NULL) {
 		while (!feof(file)) {
-			line = getline(line, file);
+			line = newspost_getline(line, file);
 			if(line == NULL){
 				text_buffer = buff_add(text_buffer, "\r\n");
 				continue;
diff -u -r newspost-2.1.1.orig/base/utils.c newspost-2.1.1/base/utils.c
--- newspost-2.1.1.orig/base/utils.c	2003-04-23 10:33:23.000000000 -0500
+++ newspost-2.1.1/base/utils.c	2018-02-02 16:14:36.751929386 -0600
@@ -43,7 +43,7 @@
 	return NULL;
 }
 
-Buff * getline(Buff *buff, FILE *file){
+Buff * newspost_getline(Buff *buff, FILE *file){
 	char c = fgetc(file);
 	buff = buff_free(buff);
 	while(TRUE){
diff -u -r newspost-2.1.1.orig/base/utils.h newspost-2.1.1/base/utils.h
--- newspost-2.1.1.orig/base/utils.h	2003-02-08 09:09:41.000000000 -0600
+++ newspost-2.1.1/base/utils.h	2018-02-02 16:15:07.069929063 -0600
@@ -26,7 +26,7 @@
 file_entry * file_entry_alloc();
 file_entry * file_entry_free(file_entry *fe);
 
-Buff * getline(Buff *buff, FILE *file);
+Buff * newspost_getline(Buff *buff, FILE *file);
 Buff *buff_add(Buff *buff, char *data, ... );
 Buff * buff_free(Buff *buff);
 Buff *buff_create(Buff *buff, char *data, ... );
diff -u -r newspost-2.1.1.orig/ui/options.c newspost-2.1.1/ui/options.c
--- newspost-2.1.1.orig/ui/options.c	2003-04-23 10:33:33.000000000 -0500
+++ newspost-2.1.1/ui/options.c	2018-02-02 16:16:53.144927936 -0600
@@ -332,7 +332,7 @@
 		file = fopen(filename->data, "r");
 		if (file != NULL) {
 			while (!feof(file)) {
-				line = getline(line, file);
+				line = newspost_getline(line, file);
 				linenum++;
 				if(line == NULL) continue;
 
@@ -429,7 +429,7 @@
 				linenum = 0;
 				while (linenum < 8) {
 					linenum++;
-					line = getline(line, file);
+					line = newspost_getline(line, file);
 					if(line == NULL) continue;
 
 					switch (linenum) {
