diff -Nur xscreensaver-5.45.orig/driver/screensaver-properties.desktop.in xscreensaver-5.45/driver/screensaver-properties.desktop.in
--- xscreensaver-5.45.orig/driver/screensaver-properties.desktop.in	2009-10-06 21:26:41.000000000 -0500
+++ xscreensaver-5.45/driver/screensaver-properties.desktop.in	2020-12-15 20:12:13.223955366 -0600
@@ -2,7 +2,7 @@
 Exec=xscreensaver-demo
 Icon=xscreensaver
 Terminal=false
-_Name=Screensaver
-_Comment=Change screensaver properties
+_Name=XScreenSaver
+_Comment=Change XScreenSaver properties
 Type=Application
 Categories=Settings;DesktopSettings;Security;X-XFCE;
