--- ./driver/prefs.c.orig	2016-05-23 22:08:27.000000000 -0500
+++ ./driver/prefs.c	2016-05-30 15:29:36.446721679 -0500
@@ -1734,6 +1734,17 @@
      shipping the last version with the old license and then never
      upgrading it again -- which would be the worst possible outcome for
      everyone involved, most especially the users.
+
+     ---
+
+     NOTE:  This feature is disabled by Slackware...  we do not ship multi-year
+     old versions, nor do we think it is a good idea to include nag screens
+     that activate when an expiration date is reached.  In this case, the
+     nag screen was activated after only one year, which practically insures
+     that this will be seen before the next stable Slackware release.
+     If there's a problem that needs fixing, we'll issue a fix.
+     No nag screens please.  Thanks for possibly reconsidering this!
+
   */
 
   time_t now = time ((time_t *) 0);				/*       */
@@ -1766,5 +1777,7 @@
   months = ((((tm->tm_year + 1900) * 12) + tm->tm_mon) -	/*   h   */
             (y * 12 + m));					/*   p   */
 							  	/*   .   */
-  return (months >= 17);					/*       */
+  /* Disable date checking for Slackware:  */                   /*  \ö/  */
+  /* return (months >= 17); */					/*       */
+  return ( 0 );
 }
