--- ./driver/dialog.c.orig	2021-10-04 17:18:28.000000000 -0500
+++ ./driver/dialog.c	2023-03-05 14:03:13.158084489 -0600
@@ -1333,6 +1333,15 @@
   lines[i].align = CENTER;
   i++;
 
+/*
+  NOTE:  This feature is disabled by Slackware...  we do not ship multi-year
+  old versions, nor do we think it is a good idea to include nag screens
+  that activate when an expiration date is reached.  In this case, the
+  nag screen was activated after only one year, which practically insures
+  that this will be seen before the next stable Slackware release.
+  If there's a problem that needs fixing, we'll issue a fix.
+  No nag screens please.  Thanks for possibly reconsidering this!
+
   if (time ((time_t *) 0) - XSCREENSAVER_RELEASED > 60*60*24*30*17)
     {
       lines[i].text  = _("Update available!\nThis version is very old.\n");
@@ -1343,7 +1352,10 @@
       lines[i].align = CENTER;
       i++;
     }
-  else if (strstr (ws->version, "a") ||
+  else
+*/
+
+  if (strstr (ws->version, "a") ||
       strstr (ws->version, "b"))
     {
       lines[i].text  = _("PRE-RELEASE VERSION");
