--- ./driver/XScreenSaver.ad.in.orig	2022-12-11 14:32:55.000000000 -0600
+++ ./driver/XScreenSaver.ad.in	2022-12-12 13:21:20.761072115 -0600
@@ -411,6 +411,7 @@
 				spotlight --root			    \n\
 				squiral --root				    \n\
 				wander --root				    \n\
+				electricsheep --root 1			    \n\
 				xflame --root				    \n\
 				xmatrix --root				    \n\
 @GL_KLUDGE@ GL: 				gflux --root				    \n\
--- ./driver/XScreenSaver_ad.h.orig	2022-12-11 15:08:39.000000000 -0600
+++ ./driver/XScreenSaver_ad.h	2022-12-12 13:21:24.404072196 -0600
@@ -266,6 +266,7 @@
 				spotlight --root			    \\n\
 				squiral --root				    \\n\
 				wander --root				    \\n\
+				electricsheep --root 1			    \\n\
 				xflame --root				    \\n\
 				xmatrix --root				    \\n\
   GL: 				gflux --root				    \\n\
