--- ./Bindings/Python/brlapi.pyx.orig	2013-04-01 15:37:26.000000000 -0500
+++ ./Bindings/Python/brlapi.pyx	2013-09-09 18:01:39.196869555 -0500
@@ -497,7 +497,7 @@
 		if (type(dots) == unicode):
 			dots = dots.encode('latin1')
 		if (len(dots) < dispSize):
-			dots = dots + b"".center(dispSize - len(dots), '\0')
+			dots = dots + "".center(dispSize - len(dots), '\0')
 		c_dots = dots
 		c_udots = <unsigned char *>c_dots
 		c_brlapi.Py_BEGIN_ALLOW_THREADS
