diff -Naur irssi-1.2.2/src/fe-text/term-terminfo.c irssi-1.2.2.patched/src/fe-text/term-terminfo.c
--- irssi-1.2.2/src/fe-text/term-terminfo.c	2019-08-29 09:48:46.000000000 -0400
+++ irssi-1.2.2.patched/src/fe-text/term-terminfo.c	2020-04-26 04:15:02.007493067 -0400
@@ -673,6 +673,7 @@
 static int input_utf8(const unsigned char *buffer, int size, unichar *result)
 {
 	unichar c = g_utf8_get_char_validated((char *)buffer, size);
+	if(*buffer == '\0') c = (unichar)-1;
 
 	switch (c) {
 	case (unichar)-1:
