diff -ur pykde4-4.14.3.old/sip/kio/kautomount.sip pykde4-4.14.3/sip/kio/kautomount.sip
--- pykde4-4.14.3.old/sip/kio/kautomount.sip	2014-08-14 22:17:11.000000000 +0200
+++ pykde4-4.14.3/sip/kio/kautomount.sip	2017-02-25 19:17:16.511589230 +0100
@@ -20,7 +20,7 @@
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 
-class KAutoMount : QObject
+class KAutoMount : QObject /Abstract/
 {
 %TypeHeaderCode
 #include <kautomount.h>
@@ -42,7 +42,7 @@
 // KAutoMount
 
 
-class KAutoUnmount : QObject
+class KAutoUnmount : QObject /Abstract/
 {
 %TypeHeaderCode
 #include <kautomount.h>
