--- ./php.ini-development.orig	2017-07-05 17:25:00.000000000 -0500
+++ ./php.ini-development	2017-07-07 14:25:29.270476234 -0500
@@ -879,6 +879,50 @@
 ; If you only provide the name of the extension, PHP will look for it in its
 ; default extension directory.
 ;
+
+; These are the extensions that are available for use on this PHP build.
+; If you do NOT need any of these extensions, you should comment them out.
+; That will probably increase both your security and performance.
+
+extension=bcmath.so
+extension=bz2.so
+extension=calendar.so
+extension=ctype.so
+extension=curl.so
+extension=dba.so
+extension=dom.so
+extension=enchant.so
+extension=exif.so
+extension=ftp.so
+extension=gd.so
+extension=gettext.so
+extension=gmp.so
+extension=iconv.so
+extension=intl.so
+extension=ldap.so
+extension=mbstring.so
+extension=mysql.so
+extension=mysqli.so
+extension=odbc.so
+extension=openssl.so
+zend_extension=opcache.so
+extension=pdo.so
+extension=pdo_mysql.so
+extension=pdo_sqlite.so
+extension=pdo_odbc.so
+extension=pspell.so
+extension=shmop.so
+extension=snmp.so
+extension=soap.so
+extension=sqlite3.so
+extension=tokenizer.so
+extension=wddx.so
+extension=xmlreader.so
+extension=xmlrpc.so
+extension=xsl.so
+extension=zip.so
+extension=zlib.so
+
 ; Windows Extensions
 ; Note that ODBC support is built in, so no dll is needed for it.
 ; Note that many DLL files are located in the extensions/ (PHP 4) ext/ (PHP 5)
@@ -1442,7 +1486,7 @@
 ; where MODE is the octal representation of the mode. Note that this
 ; does not overwrite the process's umask.
 ; http://php.net/session.save-path
-;session.save_path = "/tmp"
+session.save_path = "/var/lib/php"
 
 ; Whether to use strict session mode.
 ; Strict session mode does not accept uninitialized session ID and regenerate
