From 706116c651fd9abc9284516b47cea6cd536c7bcd Mon Sep 17 00:00:00 2001
From: Paul Eggert <eggert@cs.ucla.edu>
Date: Fri, 21 Feb 2025 23:10:06 -0700
Subject: [PATCH] maint: mention bug#76452 in NEWS

---
 NEWS | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/NEWS b/NEWS
index 5c23daf..73d98d6 100644
--- a/NEWS
+++ b/NEWS
@@ -2,6 +2,12 @@ GNU diffutils NEWS                                    -*- outline -*-
 
 * Noteworthy changes in release ?.? (????-??-??) [?]
 
+** Bug fixes
+
+  diff -r no longer merely summarizes when comparing an empty regular
+  file to a nonempty regular file.
+  [bug#76452 introduced in 3.11]
+
 
 * Noteworthy changes in release 3.11 (2025-02-02) [stable]
 
-- 
2.34.1

