diff -Nur diffutils-2.9.orig/src/sdiff.c diffutils-2.9.new/src/sdiff.c
--- diffutils-2.9.orig/src/sdiff.c	2010-02-11 03:39:17.000000000 -0600
+++ diffutils-2.9.new/src/sdiff.c	2010-02-14 02:28:19.077953690 -0600
@@ -715,7 +715,7 @@
       ck_fclose (out);
 
       {
-	int wstatus;
+	int wstatus = 0;
 	int werrno = 0;
 
 #if ! (HAVE_WORKING_FORK || HAVE_WORKING_VFORK)
@@ -724,11 +724,6 @@
 	  werrno = errno;
 #else
 	ck_fclose (diffout);
-	while (waitpid (diffpid, &wstatus, 0) < 0)
-	  if (errno == EINTR)
-	    checksigs ();
-	  else
-	    perror_fatal ("waitpid");
 	diffpid = 0;
 #endif
 
