diff -ur xchat-2.8.6.orig/src/fe-gtk/about.c xchat-2.8.6/src/fe-gtk/about.c
--- xchat-2.8.6.orig/src/fe-gtk/about.c	2008-04-01 10:58:38.000000000 +0200
+++ xchat-2.8.6/src/fe-gtk/about.c	2008-10-12 14:12:59.329706289 +0200
@@ -16,7 +16,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA
  */
 
-#define GTK_DISABLE_DEPRECATED
+/* #define GTK_DISABLE_DEPRECATED */
 
 #include <stdio.h>
 #include <stdlib.h>
diff -ur xchat-2.8.6.orig/src/fe-gtk/ascii.c xchat-2.8.6/src/fe-gtk/ascii.c
--- xchat-2.8.6.orig/src/fe-gtk/ascii.c	2008-02-05 11:02:50.000000000 +0100
+++ xchat-2.8.6/src/fe-gtk/ascii.c	2008-10-12 14:12:59.329706289 +0200
@@ -16,7 +16,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA
  */
 
-#define GTK_DISABLE_DEPRECATED
+/* #define GTK_DISABLE_DEPRECATED */
 
 #include <stdio.h>
 #include <stdlib.h>
diff -ur xchat-2.8.6.orig/src/fe-gtk/banlist.c xchat-2.8.6/src/fe-gtk/banlist.c
--- xchat-2.8.6.orig/src/fe-gtk/banlist.c	2008-02-05 11:02:50.000000000 +0100
+++ xchat-2.8.6/src/fe-gtk/banlist.c	2008-10-12 14:12:59.329706289 +0200
@@ -16,7 +16,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA
  */
 
-#define GTK_DISABLE_DEPRECATED
+/* #define GTK_DISABLE_DEPRECATED */
 
 #include <stdio.h>
 #include <stdlib.h>
diff -ur xchat-2.8.6.orig/src/fe-gtk/fe-gtk.c xchat-2.8.6/src/fe-gtk/fe-gtk.c
--- xchat-2.8.6.orig/src/fe-gtk/fe-gtk.c	2008-02-24 09:39:44.000000000 +0100
+++ xchat-2.8.6/src/fe-gtk/fe-gtk.c	2008-10-12 14:12:59.339702951 +0200
@@ -16,7 +16,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA
  */
 
-#define GTK_DISABLE_DEPRECATED
+/* #define GTK_DISABLE_DEPRECATED */
 
 #include <stdio.h>
 #include <string.h>
diff -ur xchat-2.8.6.orig/src/fe-gtk/ignoregui.c xchat-2.8.6/src/fe-gtk/ignoregui.c
--- xchat-2.8.6.orig/src/fe-gtk/ignoregui.c	2008-02-05 11:02:52.000000000 +0100
+++ xchat-2.8.6/src/fe-gtk/ignoregui.c	2008-10-12 14:12:59.343034799 +0200
@@ -16,7 +16,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA
  */
 
-#define GTK_DISABLE_DEPRECATED
+/* #define GTK_DISABLE_DEPRECATED */
 
 #include <stdio.h>
 #include <stdlib.h>
diff -ur xchat-2.8.6.orig/src/fe-gtk/maingui.c xchat-2.8.6/src/fe-gtk/maingui.c
--- xchat-2.8.6.orig/src/fe-gtk/maingui.c	2008-04-01 10:53:41.000000000 +0200
+++ xchat-2.8.6/src/fe-gtk/maingui.c	2008-10-12 14:12:59.349699891 +0200
@@ -16,7 +16,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA
  */
 
-#define GTK_DISABLE_DEPRECATED
+/* #define GTK_DISABLE_DEPRECATED */
 
 #include <stdlib.h>
 #include <string.h>
diff -ur xchat-2.8.6.orig/src/fe-gtk/menu.c xchat-2.8.6/src/fe-gtk/menu.c
--- xchat-2.8.6.orig/src/fe-gtk/menu.c	2008-06-08 09:59:37.000000000 +0200
+++ xchat-2.8.6/src/fe-gtk/menu.c	2008-10-12 14:12:59.349699891 +0200
@@ -16,7 +16,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA
  */
 
-#define GTK_DISABLE_DEPRECATED
+/* #define GTK_DISABLE_DEPRECATED */
 
 #include <stdio.h>
 #include <stdlib.h>
diff -ur xchat-2.8.6.orig/src/fe-gtk/notifygui.c xchat-2.8.6/src/fe-gtk/notifygui.c
--- xchat-2.8.6.orig/src/fe-gtk/notifygui.c	2008-02-05 11:03:34.000000000 +0100
+++ xchat-2.8.6/src/fe-gtk/notifygui.c	2008-10-12 14:12:59.363033500 +0200
@@ -16,7 +16,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA
  */
 
-#define GTK_DISABLE_DEPRECATED
+/* #define GTK_DISABLE_DEPRECATED */
 
 #include <stdio.h>
 #include <stdlib.h>
diff -ur xchat-2.8.6.orig/src/fe-gtk/palette.c xchat-2.8.6/src/fe-gtk/palette.c
--- xchat-2.8.6.orig/src/fe-gtk/palette.c	2008-02-05 11:02:52.000000000 +0100
+++ xchat-2.8.6/src/fe-gtk/palette.c	2008-10-12 14:12:59.363033500 +0200
@@ -16,7 +16,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA
  */
 
-#define GTK_DISABLE_DEPRECATED
+/* #define GTK_DISABLE_DEPRECATED */
 
 #include <stdio.h>
 #include <stdlib.h>
diff -ur xchat-2.8.6.orig/src/fe-gtk/pixmaps.c xchat-2.8.6/src/fe-gtk/pixmaps.c
--- xchat-2.8.6.orig/src/fe-gtk/pixmaps.c	2008-02-05 11:02:52.000000000 +0100
+++ xchat-2.8.6/src/fe-gtk/pixmaps.c	2008-10-12 14:12:59.363033500 +0200
@@ -16,7 +16,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA
  */
 
-#define GTK_DISABLE_DEPRECATED
+/* #define GTK_DISABLE_DEPRECATED */
 
 #include <stdio.h>
 #include <string.h>
diff -ur xchat-2.8.6.orig/src/fe-gtk/plugingui.c xchat-2.8.6/src/fe-gtk/plugingui.c
--- xchat-2.8.6.orig/src/fe-gtk/plugingui.c	2008-02-05 11:02:52.000000000 +0100
+++ xchat-2.8.6/src/fe-gtk/plugingui.c	2008-10-12 14:12:59.363033500 +0200
@@ -16,7 +16,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA
  */
 
-#define GTK_DISABLE_DEPRECATED
+/* #define GTK_DISABLE_DEPRECATED */
 
 #include <string.h>
 #include <stdio.h>
diff -ur xchat-2.8.6.orig/src/fe-gtk/rawlog.c xchat-2.8.6/src/fe-gtk/rawlog.c
--- xchat-2.8.6.orig/src/fe-gtk/rawlog.c	2008-02-24 05:49:37.000000000 +0100
+++ xchat-2.8.6/src/fe-gtk/rawlog.c	2008-10-12 14:12:59.366369608 +0200
@@ -16,7 +16,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA
  */
 
-#define GTK_DISABLE_DEPRECATED
+/* #define GTK_DISABLE_DEPRECATED */
 
 #include <stdio.h>
 #include <string.h>
diff -ur xchat-2.8.6.orig/src/fe-gtk/search.c xchat-2.8.6/src/fe-gtk/search.c
--- xchat-2.8.6.orig/src/fe-gtk/search.c	2008-02-05 11:02:53.000000000 +0100
+++ xchat-2.8.6/src/fe-gtk/search.c	2008-10-12 14:12:59.369703062 +0200
@@ -16,7 +16,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA
  */
 
-#define GTK_DISABLE_DEPRECATED
+/* #define GTK_DISABLE_DEPRECATED */
 
 #include <stdio.h>
 #include <string.h>
diff -ur xchat-2.8.6.orig/src/fe-gtk/urlgrab.c xchat-2.8.6/src/fe-gtk/urlgrab.c
--- xchat-2.8.6.orig/src/fe-gtk/urlgrab.c	2008-02-05 11:02:53.000000000 +0100
+++ xchat-2.8.6/src/fe-gtk/urlgrab.c	2008-10-12 14:12:59.376368714 +0200
@@ -16,7 +16,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA
  */
 
-#define GTK_DISABLE_DEPRECATED
+/* #define GTK_DISABLE_DEPRECATED */
 
 #include <stdio.h>
 #include <string.h>
diff -ur xchat-2.8.6.orig/src/fe-gtk/userlistgui.c xchat-2.8.6/src/fe-gtk/userlistgui.c
--- xchat-2.8.6.orig/src/fe-gtk/userlistgui.c	2008-02-05 11:02:53.000000000 +0100
+++ xchat-2.8.6/src/fe-gtk/userlistgui.c	2008-10-12 14:12:59.379703216 +0200
@@ -16,7 +16,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA
  */
 
-#define GTK_DISABLE_DEPRECATED
+/* #define GTK_DISABLE_DEPRECATED */
 
 #include <stdio.h>
 #include <string.h>
diff -ur xchat-2.8.6.orig/src/fe-gtk/xtext.h xchat-2.8.6/src/fe-gtk/xtext.h
--- xchat-2.8.6.orig/src/fe-gtk/xtext.h	2008-02-24 05:48:02.000000000 +0100
+++ xchat-2.8.6/src/fe-gtk/xtext.h	2008-10-12 14:10:49.203049619 +0200
@@ -1,6 +1,7 @@
 #ifndef __XTEXT_H__
 #define __XTEXT_H__
 
+#include <gtk/gtktypeutils.h>
 #include <gtk/gtkadjustment.h>
 #ifdef USE_XFT
 #include <X11/Xft/Xft.h>
