--- ./conf/postfix-script.orig	2016-01-31 15:05:46.000000000 -0600
+++ ./conf/postfix-script	2017-11-05 15:09:40.838097647 -0600
@@ -275,7 +275,7 @@
 	find $todo ! -user root \
 	    -exec $WARN not owned by root: {} \;
 
-	find $todo \( -perm -020 -o -perm -002 \) \
+	find $todo -type f \( -perm -020 -o -perm -002 \) \
 	    -exec $WARN group or other writable: {} \;
 
 	# Check Postfix mail_owner-owned directory tree owner/permissions.
