--- ./conf/postfix-script.orig	2019-02-28 13:02:24.783759137 -0600
+++ ./conf/postfix-script	2019-02-28 13:05:25.516761071 -0600
@@ -139,7 +139,10 @@
 		# Foreground this so it can be stopped. All inodes are cached.
 		$daemon_directory/postfix-script check-warn
 	fi
-	$INFO starting the Postfix mail system || exit 1
+	$INFO starting the Postfix mail system > /dev/null 2> /dev/null || exit 1
+	# Because this is called during Slackware init, let's make it look
+	# similar to everything else printed on the screen:
+	echo "Starting the Postfix mail system:  $command_directory/postfix start"
 	case $1 in
 	start)
 	    # NOTE: wait in foreground process to get the initialization status.
