From 24d77f5b06c704506b4ae9d2eabdb96e56e12dae Mon Sep 17 00:00:00 2001
From: Liquider <liquiderz@gmail.com>
Date: Sun, 16 Feb 2014 14:04:20 +0100
Subject: [PATCH 1/3] Set notifications as override redirect windows (bug
 #10686).

---
 xfce4-notifyd/xfce-notify-window.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xfce4-notifyd/xfce-notify-window.c b/xfce4-notifyd/xfce-notify-window.c
index f8ffdf6..29287fa 100644
--- a/xfce4-notifyd/xfce-notify-window.c
+++ b/xfce4-notifyd/xfce-notify-window.c
@@ -318,6 +318,7 @@ xfce_notify_window_realize(GtkWidget *widget)
 
     gdk_window_set_type_hint(widget->window,
                              GDK_WINDOW_TYPE_HINT_NOTIFICATION);
+    gdk_window_set_override_redirect(gtk_widget_get_window(widget), TRUE);
     xfce_notify_window_start_expiration(window);
 }
 
-- 
2.3.0

