From 93123ab4b7b73e8295102e709d5eeea398774650 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=A9r=C3=B4me=20Guelfucci?= <jeromeg@xfce.org>
Date: Sun, 16 Feb 2014 14:09:36 +0100
Subject: [PATCH 2/3] Use accessor function to get the GdkWindow.

---
 xfce4-notifyd/xfce-notify-window.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xfce4-notifyd/xfce-notify-window.c b/xfce4-notifyd/xfce-notify-window.c
index 29287fa..e2fc40a 100644
--- a/xfce4-notifyd/xfce-notify-window.c
+++ b/xfce4-notifyd/xfce-notify-window.c
@@ -316,7 +316,7 @@ xfce_notify_window_realize(GtkWidget *widget)
 
     GTK_WIDGET_CLASS(xfce_notify_window_parent_class)->realize(widget);
 
-    gdk_window_set_type_hint(widget->window,
+    gdk_window_set_type_hint(gtk_widget_get_window(widget),
                              GDK_WINDOW_TYPE_HINT_NOTIFICATION);
     gdk_window_set_override_redirect(gtk_widget_get_window(widget), TRUE);
     xfce_notify_window_start_expiration(window);
-- 
2.3.0

