--- ./libusb/os/linux_usbfs.c.orig	2010-04-22 16:14:46.000000000 -0500
+++ ./libusb/os/linux_usbfs.c	2011-03-24 18:40:38.000000000 -0500
@@ -1055,7 +1055,8 @@
 	hpriv->fd = open(filename, O_RDWR);
 	if (hpriv->fd < 0) {
 		if (errno == EACCES) {
-			fprintf(stderr, "libusb couldn't open USB device %s: "
+                        usbi_err(HANDLE_CTX(handle),
+			        "libusb couldn't open USB device %s: "
 				"Permission denied.\n"
 				"libusb requires write access to USB device nodes.\n",
 				filename);
