--- ./hw/xfree86/common/xf86pciBus.c.orig	2010-07-01 12:17:35.000000000 -0500
+++ ./hw/xfree86/common/xf86pciBus.c	2010-11-18 16:56:57.633003654 -0600
@@ -1118,7 +1118,7 @@
 	    break;
 	case 0x102b:		    driverList[0] = "mga";	break;
 	case 0x10c8:		    driverList[0] = "neomagic"; break;
-	case 0x10de: case 0x12d2:   driverList[0] = "nv";	break;
+	case 0x10de: case 0x12d2:   driverList[0] = "nouveau" ; driverList[1] = "nv";	break;
 	case 0x1106:		    driverList[0] = "openchrome"; break;
         case 0x1b36:		    driverList[0] = "qxl"; break;
 	case 0x1163:		    driverList[0] = "rendition"; break;
