--- ./libksieve/kmanagesieve/session.cpp.orig	2015-06-25 12:47:38.000000000 -0500
+++ ./libksieve/kmanagesieve/session.cpp	2019-11-08 15:49:06.212790078 -0600
@@ -368,7 +368,7 @@
         m_sslCheck->setInterval(60*1000);
         connect(m_sslCheck, SIGNAL(timeout()), this, SLOT(slotSslTimeout()));
     }
-    m_socket->setAdvertisedSslVersion( KTcpSocket::TlsV1 );
+    m_socket->setAdvertisedSslVersion( KTcpSocket::SecureProtocols );
     m_socket->ignoreSslErrors();
     connect(m_socket, SIGNAL(encrypted()), SLOT(slotEncryptedDone()));
     m_sslCheck->start();
