From c6befda3aea297610d76e0a0c5109e28ff18baab Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Mon, 12 Jun 2023 08:51:16 +0800
Subject: [PATCH] add loongarch64 support


diff --git a/ipc/chromium/src/build/build_config.h b/ipc/chromium/src/build/build_config.h
index de26701b2d..76f74de83f 100644
--- a/ipc/chromium/src/build/build_config.h
+++ b/ipc/chromium/src/build/build_config.h
@@ -125,6 +125,9 @@
 #elif defined(__riscv) && __riscv_xlen == 64
 #  define ARCH_CPU_RISCV64 1
 #  define ARCH_CPU_64_BITS 1
+#elif defined(__loongarch_lp64)
+#  define ARCH_CPU_LOONGARCH64 1
+#  define ARCH_CPU_64_BITS 1
 #else
 #  error Please add support for your architecture in build/build_config.h
 #endif
diff --git a/third_party/libwebrtc/build/build_config.h b/third_party/libwebrtc/build/build_config.h
index c39ae9da50..28191de026 100644
--- a/third_party/libwebrtc/build/build_config.h
+++ b/third_party/libwebrtc/build/build_config.h
@@ -210,6 +210,10 @@
 #define ARCH_CPU_SPARC 1
 #define ARCH_CPU_32_BITS 1
 #define ARCH_CPU_BIG_ENDIAN 1
+#elif defined(__loongarch_lp64)
+#define ARCH_CPU_LOONGARCH64 1
+#define ARCH_CPU_64_BITS 1
+#define ARCH_CPU_LITTLE_ENDIAN 1
 #else
 #error Please add support for your architecture in build/build_config.h
 #endif
diff --git a/toolkit/moz.configure b/toolkit/moz.configure
index e51e97678c..aee3b36bba 100644
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -2192,7 +2192,7 @@ with only_when(compile_environment | artifact_builds):
                 use_nasm = False
         elif target.cpu == "x86_64":
             flags = ["-D__x86_64__", "-DPIC", "-DELF", "-Pconfig_unix64.asm"]
-        elif target.cpu in ("x86", "arm", "aarch64"):
+        elif target.cpu in ("x86", "arm", "aarch64", "loongarch64"):
             flac_only = True
         else:
             enable = False
-- 
2.39.1

