From ce0eb01dddae974e7754a36665dbb8256a975b34 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Mon, 12 Jun 2023 09:17:07 +0800
Subject: [PATCH 2/2] fix for gcc13


diff --git a/gfx/2d/Rect.h b/gfx/2d/Rect.h
index 4875ad6714..f52437bbdc 100644
--- a/gfx/2d/Rect.h
+++ b/gfx/2d/Rect.h
@@ -15,6 +15,7 @@
 #include "mozilla/Maybe.h"
 
 #include <cmath>
+#include <cstdint>
 
 namespace mozilla {
 
diff --git a/toolkit/components/telemetry/pingsender/pingsender.cpp b/toolkit/components/telemetry/pingsender/pingsender.cpp
index 01fdb63c29..3e8bab5fc9 100644
--- a/toolkit/components/telemetry/pingsender/pingsender.cpp
+++ b/toolkit/components/telemetry/pingsender/pingsender.cpp
@@ -9,6 +9,7 @@
 #include <fstream>
 #include <iomanip>
 #include <string>
+#include <cstdint>
 #include <vector>
 
 #include <zlib.h>
-- 
2.39.1

