--- ./modules/libpref/init/all.js.orig	2017-04-10 21:13:21.000000000 -0500
+++ ./modules/libpref/init/all.js	2017-05-21 15:29:42.803757223 -0500
@@ -235,6 +235,9 @@
 // If true, all popups won't hide automatically on blur
 pref("ui.popup.disable_autohide", false);
 
+// Default to GTK+ v2 type scrolling:
+pref("ui.scrollToClick", 0);
+
 pref("browser.display.use_document_fonts",  1);  // 0 = never, 1 = quick, 2 = always
 // 0 = default: always, except in high contrast mode
 // 1 = always
