--- ./modules/libpref/init/all.js.orig	2019-07-08 13:28:45.106153934 -0500
+++ ./modules/libpref/init/all.js	2019-07-08 13:41:02.035201116 -0500
@@ -258,6 +258,9 @@
 // If true, all popups won't hide automatically on blur
 pref("ui.popup.disable_autohide", false);
 
+// Default to GTK+ v2 type scrolling:
+pref("ui.scrollToClick", 0);
+
 #ifdef XP_MACOSX
 pref("ui.touchbar.layout", "Back,Forward,Reload,OpenLocation,NewTab,Share");
 #endif
