--- ./config/makefiles/rust.mk.orig	2020-07-08 06:51:15.000000000 -0500
+++ ./config/makefiles/rust.mk	2020-07-24 12:23:31.769077708 -0500
@@ -61,7 +61,11 @@
 # Enable link-time optimization for release builds, but not when linking
 # gkrust_gtest.
 ifeq (,$(findstring gkrust_gtest,$(RUST_LIBRARY_FILE)))
+# Pass -Clto for older versions of rust, and CARGO_PROFILE_RELEASE_LTO=true
+# for newer ones that support it. Combining the latter with -Clto works, so
+# set both everywhere.
 cargo_rustc_flags += -Clto
+export CARGO_PROFILE_RELEASE_LTO=true
 endif
 endif
 endif
