diff -up ConsoleKit-0.3.0/doc/dbus/Makefile.in.skipvalidation ConsoleKit-0.3.0/doc/dbus/Makefile.in
--- ConsoleKit-0.3.0/doc/dbus/Makefile.in.skipvalidation	2009-02-27 20:17:36.784329106 -0500
+++ ConsoleKit-0.3.0/doc/dbus/Makefile.in	2009-02-27 20:18:15.701316733 -0500
@@ -427,7 +427,7 @@ uninstall-am: uninstall-htmldocDATA
 
 
 @DOCBOOK_DOCS_ENABLED_TRUE@ConsoleKit.html : $(SPEC_XML_FILES)
-@DOCBOOK_DOCS_ENABLED_TRUE@	$(XMLTO) xhtml-nochunks -m $(top_srcdir)/doc/dbus/config.xsl ConsoleKit.xml
+@DOCBOOK_DOCS_ENABLED_TRUE@	$(XMLTO) --skip-validation xhtml-nochunks -m $(top_srcdir)/doc/dbus/config.xsl ConsoleKit.xml
 
 org.freedesktop.ConsoleKit.Manager.ref.xml : $(top_srcdir)/src/org.freedesktop.ConsoleKit.Manager.xml spec-to-docbook.xsl
 	$(XSLTPROC) $(top_srcdir)/doc/dbus/spec-to-docbook.xsl $< | tail -n +2 > $@
