--- ./src/evdev.c.orig	2020-08-12 22:50:03.612642300 -0500
+++ ./src/evdev.c	2020-08-20 17:26:17.663305924 -0500
@@ -1027,7 +1027,7 @@
 
 	/* if we have a current libinput_dispatch() snapshot, compare our
 	 * event time with the one from the snapshot. If we have more than
-	 * 10ms delay, complain about it. This catches delays in processing
+	 * 75ms delay, complain about it. This catches delays in processing
 	 * where there is no steady event flow and thus SYN_DROPPED may not
 	 * get hit by the kernel despite us being too slow.
 	 */
@@ -1035,7 +1035,7 @@
 		return;
 
 	tdelta = us2ms(libinput->dispatch_time - input_event_time(ev));
-	if (tdelta > 10) {
+	if (tdelta > 75) {
 		evdev_log_bug_client_ratelimit(device,
 					       &device->delay_warning_limit,
 					       "event processing lagging behind by %dms, your system is too slow\n",
