--- ./src/evdev.c.orig	2022-02-19 06:32:09.000000000 -0600
+++ ./src/evdev.c	2022-03-04 00:16:00.850979376 -0600
@@ -1105,7 +1105,7 @@
 
 	/* if we have a current libinput_dispatch() snapshot, compare our
 	 * event time with the one from the snapshot. If we have more than
-	 * 10ms delay, complain about it. This catches delays in processing
+	 * 75ms delay, complain about it. This catches delays in processing
 	 * where there is no steady event flow and thus SYN_DROPPED may not
 	 * get hit by the kernel despite us being too slow.
 	 */
@@ -1114,7 +1114,7 @@
 		return;
 
 	tdelta = us2ms(libinput->dispatch_time - eventtime);
-	if (tdelta > 20) {
+	if (tdelta > 75) {
 		evdev_log_bug_client_ratelimit(device,
 					       &device->delay_warning_limit,
 					       "event processing lagging behind by %dms, your system is too slow\n",
