--- ./src/evdev.c.orig	2021-04-29 23:25:41.531076400 -0500
+++ ./src/evdev.c	2021-04-30 12:39:02.636755868 -0500
@@ -1028,7 +1028,7 @@
 
 	/* if we have a current libinput_dispatch() snapshot, compare our
 	 * event time with the one from the snapshot. If we have more than
-	 * 10ms delay, complain about it. This catches delays in processing
+	 * 75ms delay, complain about it. This catches delays in processing
 	 * where there is no steady event flow and thus SYN_DROPPED may not
 	 * get hit by the kernel despite us being too slow.
 	 */
@@ -1037,7 +1037,7 @@
 		return;
 
 	tdelta = us2ms(libinput->dispatch_time - eventtime);
-	if (tdelta > 10) {
+	if (tdelta > 75) {
 		evdev_log_bug_client_ratelimit(device,
 					       &device->delay_warning_limit,
 					       "event processing lagging behind by %dms, your system is too slow\n",
