Author: Cyril Brulebois <kibi@debian.org>
Description: Fixes FTBFS on kfreebsd (Closes: #557174).
Last-Update: 2015-09-06

--- squashfs-tools-4.3.orig/squashfs-tools/mksquashfs.c
+++ squashfs-tools-4.3/squashfs-tools/mksquashfs.c
@@ -51,7 +51,7 @@
 #include <limits.h>
 #include <ctype.h>
 
-#ifndef linux
+#if !defined(linux) && !defined(__GLIBC__)
 #define __BYTE_ORDER BYTE_ORDER
 #define __BIG_ENDIAN BIG_ENDIAN
 #define __LITTLE_ENDIAN LITTLE_ENDIAN
@@ -4105,7 +4105,7 @@ void initialise_threads(int readq, int f
 		BAD_ERROR("Failed to set signal mask in intialise_threads\n");
 
 	if(processors == -1) {
-#ifndef linux
+#if !defined(linux) && !defined(__GLIBC__)
 		int mib[2];
 		size_t len = sizeof(processors);
 
--- squashfs-tools-4.3.orig/squashfs-tools/read_fs.c
+++ squashfs-tools-4.3/squashfs-tools/read_fs.c
@@ -35,7 +35,7 @@
 #include <limits.h>
 #include <dirent.h>
 
-#ifndef linux
+#if !defined(linux) && !defined(__GLIBC__)
 #define __BYTE_ORDER BYTE_ORDER
 #define __BIG_ENDIAN BIG_ENDIAN
 #define __LITTLE_ENDIAN LITTLE_ENDIAN
--- squashfs-tools-4.3.orig/squashfs-tools/read_xattrs.c
+++ squashfs-tools-4.3/squashfs-tools/read_xattrs.c
@@ -31,7 +31,7 @@
 #include <stdio.h>
 #include <string.h>
 
-#ifndef linux
+#if !defined(linux) && !defined(__GLIBC__)
 #define __BYTE_ORDER BYTE_ORDER
 #define __BIG_ENDIAN BIG_ENDIAN
 #define __LITTLE_ENDIAN LITTLE_ENDIAN
--- squashfs-tools-4.3.orig/squashfs-tools/swap.c
+++ squashfs-tools-4.3/squashfs-tools/swap.c
@@ -19,7 +19,7 @@
  * swap.c
  */
 
-#ifndef linux
+#if !defined(linux) && !defined(__GLIBC__)
 #define __BYTE_ORDER BYTE_ORDER
 #define __BIG_ENDIAN BIG_ENDIAN
 #define __LITTLE_ENDIAN LITTLE_ENDIAN
--- squashfs-tools-4.3.orig/squashfs-tools/unsquashfs.c
+++ squashfs-tools-4.3/squashfs-tools/unsquashfs.c
@@ -2190,7 +2190,7 @@ void initialise_threads(int fragment_buf
 			"\n");
 
 	if(processors == -1) {
-#ifndef linux
+#if !defined(linux) && !defined(__GLIBC__)
 		int mib[2];
 		size_t len = sizeof(processors);
 
--- squashfs-tools-4.3.orig/squashfs-tools/unsquashfs.h
+++ squashfs-tools-4.3/squashfs-tools/unsquashfs.h
@@ -47,7 +47,7 @@
 #include <sys/ioctl.h>
 #include <sys/time.h>
 
-#ifndef linux
+#if !defined(linux) && !defined(__GLIBC__)
 #define __BYTE_ORDER BYTE_ORDER
 #define __BIG_ENDIAN BIG_ENDIAN
 #define __LITTLE_ENDIAN LITTLE_ENDIAN
--- squashfs-tools-4.3.orig/squashfs-tools/xz_wrapper.h
+++ squashfs-tools-4.3/squashfs-tools/xz_wrapper.h
@@ -24,7 +24,7 @@
  *
  */
 
-#ifndef linux
+#if !defined(linux) && !defined(__GLIBC__)
 #define __BYTE_ORDER BYTE_ORDER
 #define __BIG_ENDIAN BIG_ENDIAN
 #define __LITTLE_ENDIAN LITTLE_ENDIAN
--- squashfs-tools-4.3.orig/squashfs-tools/gzip_wrapper.h
+++ squashfs-tools-4.3/squashfs-tools/gzip_wrapper.h
@@ -24,7 +24,7 @@
  *
  */
 
-#ifndef linux
+#if !defined(linux) && !defined(__GLIBC__)
 #define __BYTE_ORDER BYTE_ORDER
 #define __BIG_ENDIAN BIG_ENDIAN
 #define __LITTLE_ENDIAN LITTLE_ENDIAN
--- squashfs-tools-4.3.orig/squashfs-tools/lz4_wrapper.h
+++ squashfs-tools-4.3/squashfs-tools/lz4_wrapper.h
@@ -24,7 +24,7 @@
  *
  */
 
-#ifndef linux
+#if !defined(linux) && !defined(__GLIBC__)
 #define __BYTE_ORDER BYTE_ORDER
 #define __BIG_ENDIAN BIG_ENDIAN
 #define __LITTLE_ENDIAN LITTLE_ENDIAN
--- squashfs-tools-4.3.orig/squashfs-tools/lzo_wrapper.h
+++ squashfs-tools-4.3/squashfs-tools/lzo_wrapper.h
@@ -24,7 +24,7 @@
  *
  */
 
-#ifndef linux
+#if !defined(linux) && !defined(__GLIBC__)
 #define __BYTE_ORDER BYTE_ORDER
 #define __BIG_ENDIAN BIG_ENDIAN
 #define __LITTLE_ENDIAN LITTLE_ENDIAN
