--- ./data/configs/blueman.rules.orig	2020-04-28 03:20:19.000000000 -0500
+++ ./data/configs/blueman.rules	2020-05-09 13:52:27.000000000 -0500
@@ -1,10 +1,10 @@
-/* Allow users in wheel group to use blueman feature requiring root without authentication */
+/* Allow users in netdev group to use blueman feature requiring root without authentication */
 polkit.addRule(function(action, subject) {
     if ((action.id == "org.blueman.network.setup" ||
          action.id == "org.blueman.dhcp.client" ||
          action.id == "org.blueman.rfkill.setstate" ||
          action.id == "org.blueman.pppd.pppconnect") &&
-        subject.isInGroup("wheel")) {
+        subject.isInGroup("netdev")) {
         return polkit.Result.YES;
     }
 });
