--- ./firefox.orig	2008-05-29 15:21:18.000000000 -0500
+++ ./firefox	2008-06-17 12:19:26.000000000 -0500
@@ -54,6 +54,14 @@
 
 moz_libdir=/usr/local/lib/firefox-3.0
 
+# Include /usr/lib/mozilla/plugins in the plugin path:
+if [ "$MOZ_PLUGIN_PATH" ] ; then
+  MOZ_PLUGIN_PATH=$MOZ_PLUGIN_PATH:${moz_libdir}/plugins:/usr/lib/mozilla/plugins
+else
+  MOZ_PLUGIN_PATH=${moz_libdir}/plugins:/usr/lib/mozilla/plugins
+fi
+export MOZ_PLUGIN_PATH
+
 # Use run-mozilla.sh in the current dir if it exists
 # If not, then start resolving symlinks until we find run-mozilla.sh
 found=0
