diff -Nur gnome-icon-theme-symbolic-3.8.0.orig/configure gnome-icon-theme-symbolic-3.8.0/configure
--- gnome-icon-theme-symbolic-3.8.0.orig/configure	2013-03-26 03:44:16.000000000 -0500
+++ gnome-icon-theme-symbolic-3.8.0/configure	2013-03-29 20:15:33.496885348 -0500
@@ -2685,7 +2685,7 @@
 
 
 
-ac_config_files="$ac_config_files gnome-icon-theme-symbolic.pc Makefile src/Makefile"
+ac_config_files="$ac_config_files Makefile src/Makefile"
 
 
 cat >confcache <<\_ACEOF
