diff -Nur gnome-icon-theme-2.20.0.orig/Makefile.in gnome-icon-theme-2.20.0/Makefile.in
--- gnome-icon-theme-2.20.0.orig/Makefile.in	2007-09-17 15:07:53.000000000 -0500
+++ gnome-icon-theme-2.20.0/Makefile.in	2008-02-09 19:56:18.425736840 -0600
@@ -218,7 +218,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 SUBDIRS = 8x8 16x16 22x22 24x24 32x32 48x48 scalable po
-pkgconfigdir = $(datadir)/pkgconfig
+pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = gnome-icon-theme.pc
 THEME_IN_FILES = index.theme.in
 theme_DATA = index.theme
