--- ./CMakeLists.txt.orig	2014-10-09 02:07:16.000000000 -0500
+++ ./CMakeLists.txt	2018-01-29 22:34:52.281573013 -0600
@@ -9,6 +9,7 @@
 # X11_xf86vmode discovery is done by FindX11
 macro_log_feature(X11_xf86vmode_FOUND "X11 Video Mode Extension (xf86vm)" "Required to build kgamma." "http://www.x.org" TRUE "" "")
 
+include(CheckIncludeFiles)
 check_include_files(strstream HAVE_STRSTREAM_H)
 configure_file(config-kgamma.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-kgamma.h)
 
