diff -up lesstif-0.95.0/scripts/motif-config.in.multilib lesstif-0.95.0/scripts/motif-config.in
--- lesstif-0.95.0/scripts/motif-config.in.multilib	2005-01-14 20:36:00.000000000 +0100
+++ lesstif-0.95.0/scripts/motif-config.in	2007-10-21 13:52:43.000000000 +0200
@@ -2,14 +2,16 @@
 #
 # $Header: /stuff/slackware/non-public/alien/slackware64/source/l/lesstif/RCS/lesstif-0.95.0-x86_64.patch,v 1.1 2008/10/17 20:29:26 eha Exp eha $
 #
-L="@prefix@"
 XC="@X_CFLAGS@"
-XL="@X_LIBS@"
-XR="@XRENDER_LIBS@"
+X_PRE_LIBS="@X_PRE_LIBS@"
+X_LIBS="@X_LIBS@"
+X_EXTRA_LIBS="@X_EXTRA_LIBS@"
 V="@PACKAGE_NAME@ @PACKAGE_VERSION@"
 
-prefix=$L
-exec_prefix=$L
+prefix="@prefix@"
+exec_prefix="@exec_prefix@"
+libdir="@libdir@"
+includedir="@includedir@"
 exec_prefix_set=no
 
 usage()
@@ -80,10 +82,10 @@ if test "$echo_exec_prefix" = "yes"; the
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo "-I$L/include $XC"
+      echo "-I$includedir $XC"
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo "-L$L/lib -L/usr/X11R6/lib -lXm -lXt $my_glib_libs $XR -lXext -lX11"
+      echo "-L$libdir ${X_LIBS} -lXm -lXt ${X_PRE_LIBS} -lX11 ${X_EXTRA_LIBS}"
 fi      
 

