# Fix for installing Dt on x86_64

diff -u --recursive lesstif-0.95.0-orig/clients/Motif-2.1/mwm/Makefile.in lesstif-0.95.0/clients/Motif-2.1/mwm/Makefile.in
--- lesstif-0.95.0-orig/clients/Motif-2.1/mwm/Makefile.in	2006-06-10 11:35:34.000000000 +0200
+++ lesstif-0.95.0/clients/Motif-2.1/mwm/Makefile.in	2006-08-23 00:27:05.000000000 +0200
@@ -301,7 +301,7 @@
 		$(X_LIBS) $(XPLIB) -lXt $(X_PRE_LIBS) -lXext -lX11 $(X_EXTRA_LIBS)
 
 appdir = $(libdir)/X11/app-defaults
-mwmddir = $(libdir)/X11/mwm
+mwmddir = $(sysconfdir)/mwm
 mwmd_DATA = system.mwmrc alt.map README
 AM_YFLAGS = -d
 all: all-am
diff -u --recursive lesstif-0.95.0-orig/lib/Dt/Makefile.in lesstif-0.95.0/lib/Dt/Makefile.in
--- lesstif-0.95.0-orig/lib/Dt/Makefile.in	2006-06-10 11:35:38.000000000 +0200
+++ lesstif-0.95.0/lib/Dt/Makefile.in	2006-08-23 00:28:39.000000000 +0200
@@ -248,7 +248,8 @@
 infodir = @infodir@
 install_sh = @install_sh@
 lib_version_suffix = @lib_version_suffix@
-libdir = $(exec_prefix)/lib
+#libdir = $(exec_prefix)/lib
+libdir = @libdir@
 libexecdir = @libexecdir@
 localstatedir = @localstatedir@
 man2html_cmd = @man2html_cmd@
