--- scim-bridge-0.4.16/client-qt/scim-bridge-client-imcontext-qt.cpp.bak	2008-11-02 14:44:46.000000000 +0800
+++ scim-bridge-0.4.16/client-qt/scim-bridge-client-imcontext-qt.cpp	2010-09-11 22:46:10.396979441 +0800
@@ -220,7 +220,12 @@
 {
     scim_bridge_pdebugln (4, "ScimBridgeClientIMContextImpl::setFocusWidget ()");
     QInputContext::setFocusWidget (widget);
-    focus_in ();
+       if (widget == NULL) {
+       focus_out ();
+       }
+       else {
+       focus_in ();
+       }
     update ();
 }
 
@@ -442,9 +447,11 @@
         }
     }
 
-    set_preedit_shown (false);
-    update_preedit ();
-
+       if (preedit_shown) {
+       set_preedit_shown (false);
+       update_preedit ();
+       }
+       
     focused_imcontext = NULL;
 }
 
