--- scim-bridge-0.4.16/agent/scim-bridge-agent-application.cpp.bak	2009-01-18 01:04:20.000000000 +0800
+++ scim-bridge-0.4.16/agent/scim-bridge-agent-application.cpp	2010-09-11 22:38:27.196979199 +0800
@@ -65,7 +65,7 @@
     unsigned int tmp_uint;
 
     int option = 0;
-    while (option != EOF) {
+    while (option != -1) {
         option = getopt_long (argc, argv, short_options, long_options, NULL);
         switch (option) {
             case 'v':
