--- scim-bridge-0.4.16/agent/scim-bridge-agent-imcontext.h.bak	2008-11-02 14:44:49.000000000 +0800
+++ scim-bridge-0.4.16/agent/scim-bridge-agent-imcontext.h	2010-09-11 23:12:18.085979961 +0800
@@ -130,6 +130,21 @@
         static void set_enabled_by_default (bool enabled);
 
         /**
+         * Get the hotkey help messages.
+         *
+         * @return help messages about hotkeys.
+         */
+        static scim::String get_help_hotkeys ();
+
+        /**
+         * Set the hotkey help messages.
+         *
+         * @param hotkey_str Help message about hotkeys.
+         *
+         */
+        static void set_help_hotkeys (const scim::String &hotkey_str);
+
+        /**
          * Destructor.
          */
         virtual ~ScimBridgeAgentIMContext () {}
