--- scim-bridge-0.4.16/agent/scim-bridge-agent-imcontext.cpp.bak	2009-02-03 22:36:14.000000000 +0800
+++ scim-bridge-0.4.16/agent/scim-bridge-agent-imcontext.cpp	2010-09-11 23:25:00.636978519 +0800
@@ -60,6 +60,8 @@
 
 static bool on_the_spot_enabled = true;
 
+static String help_hotkeys = "";
+
 /* Class definition */
 class ScimBridgeAgentIMContextImpl: public ScimBridgeAgentIMContext
 {
@@ -274,6 +276,12 @@
 }
 
 
+void ScimBridgeAgentIMContext::set_help_hotkeys (const String &hotkey_str)
+{
+    help_hotkeys = hotkey_str;
+}
+
+
 ScimBridgeAgentIMContextImpl::ScimBridgeAgentIMContextImpl (ScimBridgeAgentClientListener *new_client_listener):
 client_listener (new_client_listener), imengine (NULL), enabled (false), focused(false) ,preedit_mode (PREEDIT_ANY),
 preedit_shown (false), preedit_cursor_position (0)
@@ -654,6 +662,8 @@
 
 void ScimBridgeAgentIMContextImpl::focus_out ()
 {
+    if (!imengine_shared && imengine == NULL) alloc_imengine ();
+
     ScimBridgeAgentIMContext *focused_imcontext = static_cast<ScimBridgeAgentIMContext*> (get_imengine ()->get_frontend_data ());
 
 	if ( !focused )
@@ -923,9 +933,11 @@
 
 void ScimBridgeAgentIMContextImpl::panel_request_help ()
 {
-    String help =  String ("Smart Common Input Method platform ") +
-        String (SCIM_VERSION) +
-        String ("\n(C) 2002-2005 James Su <suzhe@tsinghua.org.cn>\n\n");
+    String help =  String ("SCIM Bridge") + 
+        String (VERSION) +
+        String ("\n(C) 2006-2008 Ryo Dairiki <ryo-dairiki@users.sourceforge.net>\n") +
+        help_hotkeys +
+        String ("\n\n");
 
         IMEngineFactoryPointer factory = scim_backend->get_factory (get_imengine ()->get_factory_uuid ());
         if (factory.null ()) factory = fallback_imengine_factory;
