--- ./startx.cpp.orig	2018-03-11 16:07:52.205154560 -0500
+++ ./startx.cpp	2018-03-11 16:09:46.094163992 -0500
@@ -201,6 +201,7 @@
     server=$defaultserver
 
 #ifdef __linux__
+  if [ -x /usr/lib/systemd/systemd -o -x /lib/systemd/systemd ]; then
     XCOMM When starting the defaultserver start X on the current tty to avoid
     XCOMM the startx session being seen as inactive:
     XCOMM "https://bugzilla.redhat.com/show_bug.cgi?id=806491"
@@ -209,6 +210,7 @@
         tty_num=$(echo "$tty" | grep -oE '[0-9]+$')
         vtarg="vt$tty_num -keeptty"
     fi
+  fi
 #endif
 
     XCOMM For compatibility reasons, only use xserverrc if there were no server command line arguments
