--- ./startx.in.orig	2025-03-09 15:44:26.000000000 -0500
+++ ./startx.in	2025-03-10 13:27:15.681102705 -0500
@@ -284,6 +284,10 @@
 
     # create a file with auth information for the server. ':0' is a dummy.
     xserverauthfile="$HOME/.serverauth.$$"
+    if [ ! -e $xserverauthfile ]; then
+      touch $xserverauthfile
+      chmod 600 $xserverauthfile
+    fi
     trap "rm -f '$xserverauthfile'" HUP INT QUIT ILL TRAP BUS TERM
     xauth -q -f "$xserverauthfile" << EOF
 add :$dummy . $mcookie
