diff -Nur wicd-1.7.3.orig/setup.py wicd-1.7.3/setup.py
--- wicd-1.7.3.orig/setup.py	2014-12-21 13:40:46.000000000 -0600
+++ wicd-1.7.3/setup.py	2014-12-23 13:03:13.181102639 -0600
@@ -617,9 +617,6 @@
             import wpath
 
         if not wpath.no_install_i18n:
-            if os.path.exists('translations'):
-                shutil.rmtree('translations/')
-            os.makedirs('translations')
 
             oldlang = os.environ['LANG']
             os.environ['LANG'] = 'C'
@@ -652,10 +649,6 @@
                 except (OSError, ValueError):
                     print 'ARGH'
 
-                if compile_po:
-                    os.makedirs('translations/' + lang + '/LC_MESSAGES/')
-                    os.system('pybabel compile -D wicd -i %s -l %s -d translations/' % (pofile, lang))
-
             os.environ['LANG'] = oldlang
 
 class uninstall(Command):
