--- ./wicd/misc.py.orig	2012-04-30 14:20:09.000000000 -0500
+++ ./wicd/misc.py	2012-05-09 14:08:24.611298308 -0500
@@ -430,7 +430,7 @@
     """ Sanitize property names to be used in config-files. """
     allowed = string.ascii_letters + '_' + string.digits
     table = string.maketrans(allowed, ' ' * len(allowed))
-    return s.translate(None, table)
+    return s.encode('ascii', 'replace').translate(None, table)
 
 def sanitize_escaped(s):
     """ Sanitize double-escaped unicode strings. """
