--- ./src/alsa/alsa.c.orig	2011-02-22 11:29:46.000000000 -0600
+++ ./src/alsa/alsa.c	2011-03-11 15:02:53.340999637 -0600
@@ -207,10 +207,20 @@
             static gboolean warned = FALSE;
             if (! warned)
             {
-                fprintf (stderr, "\n** WARNING **\nAudacious has detected that "
+                /* fprintf (stderr, "\n** WARNING **\nAudacious has detected that "
                  "your ALSA device has a broken timer.  A workaround\nis being "
                  "used to prevent CPU overload.  Please report this problem to "
-                 "your\nLinux distributor or to the ALSA developers.\n\n");
+                 "your\nLinux distributor or to the ALSA developers.\n\n"); */
+                  /* ^^^ COMMENTED OUT ^^^ */
+
+                /* I'm glad there is a workaround for this, and the reports
+                   are that it works, so spamming the console about it seems
+                   a little bit extreme.  It's not an issue to be solved here
+                   by Slackware, and I would be surprised if the ALSA
+                   developers did not know by now, given this warning.
+                   FWIW, this seems a strange method of communicating with the
+                   ALSA developers.  Did they report the bug to them? */
+
                 warned = TRUE;
             }
 
