--- ./etc/psf/etc2ps.sh.orig	2004-10-26 20:28:38.000000000 -0700
+++ ./etc/psf/etc2ps.sh	2004-10-28 11:43:19.000000000 -0700
@@ -9,14 +9,14 @@
 # tag in the case.
 #
 
-DVIPSPATH=/usr/local/tex/bin
-DVIPS=/usr/local/tex/bin/dvips
+DVIPSPATH=/usr/share/texmf/bin
+DVIPS=/usr/share/texmf/bin/dvips
 DVIPSARGS="-f -q"
 
 TROFF2PS=/usr/local/psroff/troff2/troff2ps
 TROFF2PSARGS="-Z -O-.10"
 
-PATH=/usr/bin:$DVIPSPATH; export PATH
+PATH=$PATH:$DVIPSPATH; export PATH
 
 case $1 in
 
