From c57c0a39b6745203ebec36f37318ea3b0b4820f9 Mon Sep 17 00:00:00 2001
From: Igor Murzov <igor@gplsoft.org>
Date: Sun, 18 Apr 2010 22:06:56 +0400
Subject: [PATCH 3/6] allow every user to use rpm2tgz

---
 rpm2tgz |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/rpm2tgz b/rpm2tgz
index c41f456..4818b74 100755
--- a/rpm2tgz
+++ b/rpm2tgz
@@ -216,10 +216,10 @@ for i in $* ; do
       if [ "$META_NAME" = "true" ]; then
         get_meta_data $i
         cd $TMPDIR
-        makepkg -l y -c n $CWD/$PRGNAM-$VERSION-$ARCH-${BUILD}.tgz
+        /sbin/makepkg -l y -c n $CWD/$PRGNAM-$VERSION-$ARCH-${BUILD}.tgz
       else
         cd $TMPDIR
-        makepkg -l y -c n $CWD/$(basename $i .rpm).tgz
+        /sbin/makepkg -l y -c n $CWD/$(basename $i .rpm).tgz
       fi
     )
   else
-- 
1.7.0.2

