From 88fbabed7dc236642021f8263f1b25bf67c6806e Mon Sep 17 00:00:00 2001
From: Tz-Huan Huang <tzhuan@gmail.com>
Date: Fri, 24 Oct 2014 13:12:32 +0800
Subject: [PATCH] Disable subdir-objects

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 3d89eb1..f3901ba 100644
--- a/configure.ac
+++ b/configure.ac
@@ -78,7 +78,7 @@ AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE", [The gettext domain])
 ###########################################################
 
 # Init automake stuff
-AM_INIT_AUTOMAKE([subdir-objects])
+AM_INIT_AUTOMAKE([])
 AM_CONFIG_HEADER([config.h])
 
 # Init libtool
