diff -r -u akonadi-1.13.0.orig/server/src/storage/mysql-global.conf akonadi-1.13.0/server/src/storage/mysql-global.conf
--- akonadi-1.13.0.orig/server/src/storage/mysql-global.conf	2014-08-10 12:38:58.000000000 +0200
+++ akonadi-1.13.0/server/src/storage/mysql-global.conf	2017-08-31 19:43:35.000000000 +0200
@@ -37,13 +37,13 @@
 # use InnoDB for transactions and better crash recovery
 default_storage_engine=innodb
 
-# memory pool InnoDB uses to store data dictionary information and other internal data structures (default:1M)
-# Deprecated in MySQL >= 5.6.3
-innodb_additional_mem_pool_size=1M
+# memory pool InnoDB uses to store data dictionary information and other internal data structures (default:8M)
+# Deprecated in MySQL >= 5.6.3, removed in 5.7 (works in MariaDB)
+# innodb_additional_mem_pool_size=8M
 
 # memory buffer InnoDB uses to cache data and indexes of its tables (default:128M)
 # Larger values means less I/O
-innodb_buffer_pool_size=80M
+innodb_buffer_pool_size=128M
 
 # Create a .ibd file for each table (default:0)
 innodb_file_per_table=1
