--- tetex-src-3.0/texk/web2c/fmtutil.in.fmtutil	2005-02-06 09:27:46.000000000 +0900
+++ tetex-src-3.0/texk/web2c/fmtutil.in	2005-02-07 15:22:31.828492229 +0900
@@ -55,7 +55,9 @@
 
 # Change "amstex.ini -> bamstex.ini" and "- -> language.dat"
 # if you want babel support in amstex:
-#! amstex	pdfetex		-		-translate-file=cp227.tcx *amstex.ini
+amstex	pdfetex		-		-translate-file=cp227.tcx *amstex.ini
+bplain pdfetex		language.dat	-translate-file=cp227.tcx bplain.ini
+bamstex pdfetex		language.dat	-translate-file=cp227.tcx bamstex.ini
 
 # Change "pdfamstex.ini -> pdfbamstex.ini" and "- -> language.dat"
 # if you want babel support in pdfamstex:
