--- ./texk/kpathsea/texmf.in-teTeX.orig	2005-02-05 18:04:35.000000000 -0600
+++ ./texk/kpathsea/texmf.in-teTeX	2007-06-27 21:44:36.000000000 -0500
@@ -119,7 +119,7 @@
 % Where generated fonts may be written.  This tree is used when the sources
 % were found in a system tree and either that tree wasn't writable, or the
 % varfonts feature was enabled in MT_FEATURES in mktex.cnf.
-VARTEXFONTS = @vartexfonts@
+VARTEXFONTS = $TEXMFVAR
 
 % Where to look for ls-R files.  There need not be an ls-R in the
 % directories in this path, but if there is one, Kpathsea will use it.
