--- tetex-src-3.0/texk/tetex/texconfig.tmpcleanup        2005-11-04 11:19:47.000000000 +0100
+++ tetex-src-3.0/texk/tetex/texconfig  2005-11-04 14:22:18.000000000 +0100
@@ -36,7 +36,8 @@
   TEXSOURCES TFMFONTS TRFONTS TTFONTS VFFONTS WEB2C WEBINPUTS
 "
 tmpdir=`mktemp -d "${TMP-/tmp}/$progname.XXXXXX"` || exit 1
-needsCleanup=false
+trap -- "cleanup;" EXIT 1 2 3 7 13 15
+needsCleanup=true
 lastUpdatedFile=
 
 ###############################################################################
--- tetex-src-3.0/texk/tetex/updmap.tmpcleanup        2005-11-04 11:19:47.000000000 +0100
+++ tetex-src-3.0/texk/tetex/updmap  2005-11-04 14:22:18.000000000 +0100
@@ -1274,7 +1274,7 @@
   texhashEnabled=true
   mkmapEnabled=true
   verbose=true
-  needsCleanup=false
+  needsCleanup=true
 
   cnfFileShort=updmap.cfg
   cnfFile=
@@ -1282,7 +1282,8 @@
   : ${TEXMFMAIN=`kpsewhich -var-value=TEXMFMAIN`}
 
   tmpdir=`mktemp -d ${TMP-/tmp}/$progname.XXXXXX` || \
-	abort "could not create directory"
+	abort "could not create temporary directory"
+  trap -- "cleanup;" EXIT 1 2 3 7 13 15
   tmp1=$tmpdir/a
   tmp2=$tmpdir/b
   tmp3=$tmpdir/c
--- tetex-src-3.0/texk/tetex/fmtutil.tmpcleanup        2005-11-04 11:19:47.000000000 +0100
+++ tetex-src-3.0/texk/tetex/fmtutil  2005-11-04 14:22:18.000000000 +0100
@@ -351,11 +351,13 @@
   destdir=     # global variable: where do we put the format files?
   cnf_file=    # global variable: full name of the config file
   cmd=         # desired action from command line
-  needsCleanup=false
+  needsCleanup=true
   need_find_hyphenfile=false
   cfgparam=
   cfgmaint=
-  tmpdir=`mktemp -d ${TMP-/tmp}/$progname.XXXXXX`
+  tmpdir=`mktemp -d ${TMP-/tmp}/$progname.XXXXXX` || \
+      abort "could not create temporary directory"
+  trap -- "cleanup;" EXIT 1 2 3 7 13 15
   verboseFlag=true
 
   # mktexfmtMode: if called as mktexfmt, set to true. Will echo the generated
