--- ./texk/web2c/fmtutil.cnf.orig	2005-07-17 17:09:33.000000000 -0700
+++ ./texk/web2c/fmtutil.cnf	2005-07-17 17:11:20.000000000 -0700
@@ -40,7 +40,7 @@
 # if you want babel support in tex:
 tex		tex		-		-translate-file=cp227.tcx tex.ini
 latex		pdfetex		language.dat	-translate-file=cp227.tcx *latex.ini
-etex		pdfetex		language.def	-translate-file=cp227.tcx *etex.ini
+etex		etex		language.def	-translate-file=cp227.tcx *etex.ini
 pdftex		pdfetex		-	-translate-file=cp227.tcx *pdftex.ini
 pdflatex	pdfetex		language.dat	-translate-file=cp227.tcx *pdflatex.ini
 pdfetex		pdfetex		language.def	-translate-file=cp227.tcx *pdfetex.ini
