--- ./texk/xdvik/gui/xm_prefsP.h.orig	2004-04-21 16:12:09.000000000 -0500
+++ ./texk/xdvik/gui/xm_prefsP.h	2006-05-16 16:39:38.000000000 -0500
@@ -96,8 +96,10 @@
 #define Xdvi_APPLY_STR				"Apply"
 
 #define Xdvi_PREFS_BROWSER_DEFAULTS \
+	"firefox -remote 'openURL(%s,new-window)'\n" \
+	"seamonkey -remote 'openURL(%s,new-window)'\n" \
 	"mozilla -remote 'openURL(%s,new-window)'\n" \
-	"netscape -raise -remote 'openURL(%s,new-window)'\n" \
+	"xterm -e links %s\n" \
 	"xterm -e lynx %s\n" \
 	Xdvi_ADD_COMMAND_STR
 #define Xdvi_PREFS_EDITOR_DEFAULTS \
--- ./texk/xdvik/browser.c.orig	2004-11-11 19:25:26.000000000 -0600
+++ ./texk/xdvik/browser.c	2006-05-16 16:39:52.000000000 -0500
@@ -39,7 +39,10 @@
 #include "statusline.h"
 
 static const char *const default_browser_cmd =
-	"netscape -raise -remote \"openURL(%s,new-window)\""
+	":firefox -remote \"openURL(%s,new-window)\""
+	":seamonkey -remote \"openURL(%s,new-window)\""
+	":mozilla -remote \"openURL(%s,new-window)\""
+	":xterm -e links %s"
 	":xterm -e lynx %s"
 	":xterm -e wget %s";
 
--- ./texk/xdvik/xdvi.1.in.orig	2005-01-20 10:51:45.000000000 -0600
+++ ./texk/xdvik/xdvi.1.in	2006-05-16 16:40:25.000000000 -0500
@@ -365,7 +365,7 @@
 .B WWWBROWSER
 (in that order) are used to determine the browser command. If these are
 not set either, the following default value is used:
-.SB netscape -raise -remote "openURL(%s,new-window)":xterm -e lynx %s:xterm -e wget %s
+.SB firefox -remote -remote "openURL(%s,new-window)":seamonkey -remote "openURL(%s,new-window)":mozilla -remote "openURL(%s,new-window)":xterm -e links %s:xterm -e lynx %s:xterm -e wget %s
 .\"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
 .TP
 .BI \-bw " width"
