diff -rup a/configure b/configure
--- a/configure	2023-02-08 21:27:48.733647760 -0500
+++ b/configure	2023-02-08 21:28:31.201222024 -0500
@@ -57429,6 +57429,7 @@ else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
+	 #include <stdio.h>
 	 int main()
 	 {
 	    unsigned long long x=0,x1;
diff -rup a/m4/needtrio.m4 b/m4/needtrio.m4
--- a/m4/needtrio.m4	2016-02-20 08:57:52.000000000 -0500
+++ b/m4/needtrio.m4	2023-02-08 21:28:13.642571126 -0500
@@ -9,6 +9,7 @@ AC_DEFUN([LFTP_NEED_TRIO],[
       else
 
       AC_RUN_IFELSE([AC_LANG_SOURCE([[
+         #include <stdio.h>
 	 int main()
 	 {
 	    unsigned long long x=0,x1;
