--- ./mkinitrd.orig	2021-04-27 13:53:00.500036921 -0500
+++ ./mkinitrd	2021-07-16 13:04:05.246136283 -0500
@@ -237,6 +237,13 @@
   # Then copy all remaining libs our initrd files link against:
   COUNT=1
   PRFX=$(mktemp --suffix -ldd)
+
+  # Make sure a temporary directory was actually created:
+  if [ -z "${PRFX}" ]; then
+    echo "ERROR: Could not create a temp directory"
+    exit 1
+  fi
+
   TMPFILE=${PRFX}${COUNT}
 
   find $SOURCE_TREE -type f -exec ldd {} 2>/dev/null \; | unify_libs > $TMPFILE
