--- ./mkinitrd.orig	2021-04-27 13:53:00.500036921 -0500
+++ ./mkinitrd	2022-01-26 14:31:33.134016601 -0600
@@ -755,7 +755,7 @@
         fi
 
         if ! grep -Eq " $(basename $SRCMOD .ko)(\.| |$)" $SOURCE_TREE/load_kernel_modules 2>/dev/null ; then
-          LINE="$(echo "modprobe -v $(basename ${SRCMOD%%.gz} .ko)" )"
+          LINE="$(echo "modprobe -v $(basename ${SRCMOD%%.ko*})" )"
 
           # Test to see if arguments should be passed
           # Over-ride the previously defined LINE variable if so
