--- ./mkinitrd.orig	2022-07-09 12:56:10.493030210 -0500
+++ ./mkinitrd	2022-07-09 12:57:38.848032187 -0500
@@ -702,9 +702,9 @@
   cp -a /sbin/$i $SOURCE_TREE/sbin ;
 done
 
-# Make sure modules.builtin and modules.order are there (for kmod):
+# Make sure modules.builtin, modules.builtin.modinfo, and modules.order are there (for kmod):
 for kver in $(echo ${KERNEL_VERSION} | tr ':' '\n') ; do
-  cp /lib/modules/$kver/modules.{builtin,order} \
+  cp /lib/modules/$kver/modules.{builtin,builtin.modinfo,order} \
      $SOURCE_TREE/lib/modules/$kver
 done
 
