--- ./mkinitrd_command_generator.sh.orig	2024-11-08 16:01:47.271041636 -0600
+++ ./mkinitrd_command_generator.sh	2024-12-10 12:50:57.114973388 -0600
@@ -442,11 +442,19 @@
     SRCHLIST=$(echo $SRCHLIST | cut -d' ' -f2-) # cut ELEM from the list
     if [ "$ELEM" = "-o" ]; then
       IMGFILE=$(echo $SRCHLIST | cut -d' ' -f1)
-      break
+    elif [ "$ELEM" = "-h" ]; then
+      RESUMEDEV=$(echo $SRCHLIST | cut -d' ' -f1)
+    elif [ "$ELEM" = "-l" ]; then
+      KEYMAP=$(echo $SRCHLIST | cut -d' ' -f1)
+    elif [ "$ELEM" = "-K" ]; then
+      LUKSKEY="$(echo $SRCHLIST | cut -d' ' -f1)"
     fi
   done
 fi
 IMGFILE=${IMGFILE:-"/boot/initrd.gz"}
+RESUMEDEV=${RESUMEDEV:-""}
+KEYMAP="${KEYMAP:-"us"}"
+LUKSKEY="${LUKSKEY:-""}"
 
 # Get information about the root device / root filesystem:
 ROOTDEV=${ROOTDEV:-$(get_root_device)}
@@ -859,6 +867,7 @@
 	KEYMAP="$KEYMAP"
 	MODULE_LIST="$(echo $MLIST | cut -f2 -d\ )"
 	LUKSDEV="$REALDEV"
+	LUKSKEY="${LUKSKEY}"
 	ROOTDEV="$ROOTDEV"
 	ROOTFS="$ROOTFS"
 	RESUMEDEV="$RESUMEDEV"
