diff -ruN libpng-1.2.12/scripts/makefile.linux libpng-1.2.12-ldhack/scripts/makefile.linux
--- libpng-1.2.12/scripts/makefile.linux	2006-06-27 22:22:40.000000000 +0200
+++ libpng-1.2.12-ldhack/scripts/makefile.linux	2006-08-25 14:27:08.000000000 +0200
@@ -110,12 +110,12 @@
 	$(LN_SF) $(LIBSOVER) $(LIBSOMAJ)
 
 $(LIBSOVER): $(OBJSDLL)
-	$(CC) -shared -Wl,-soname,$(LIBSOMAJ) -o $(LIBSOVER) $(OBJSDLL)
+	$(CC) -shared -Wl,-soname,$(LIBSOMAJ) -o $(LIBSOVER) $(OBJSDLL) -lm -lz
 
 $(OLDSOVER): $(OBJSDLL)
 	$(CC) -shared -Wl,-soname,$(OLDSOMAJ) \
 	-o $(OLDSOVER) \
-	$(OBJSDLL)
+	$(OBJSDLL) -lm -lz
 
 pngtest: pngtest.o $(LIBSO)
 	$(CC) -o pngtest $(CFLAGS) pngtest.o $(LDFLAGS)
