diff --git a/kde/window-decorator-kde4/decorator.cpp b/kde/window-decorator-kde4/decorator.cpp
index 96b818d..0bf4718 100644
--- a/kde/window-decorator-kde4/decorator.cpp
+++ b/kde/window-decorator-kde4/decorator.cpp
@@ -649,9 +649,10 @@ KWD::Decorator::x11EventFilter (XEvent *xevent)
 	    }
 	    else if (action == Atoms::toolkitActionRunDialogAtom)
 	    {
-		QDBusInterface krunner ("org.kde.krunner", "/Interface",
-					"org.kde.krunner.Interface");
-		krunner.call ("display", "");
+		QDBusInterface krunner ("org.kde.krunner", "/App",
+					"org.kde.krunner.App");
+		if (krunner.isValid ())
+		    krunner.call ("display");
 	    }
 	    else if (action == Atoms::toolkitActionForceQuitDialogAtom)
 	    {

