diff -Nur libgphoto2-2.4.11.orig/packaging/generic/print-camera-list.c libgphoto2-2.4.11/packaging/generic/print-camera-list.c
--- libgphoto2-2.4.11.orig/packaging/generic/print-camera-list.c	2011-04-06 16:42:09.000000000 -0500
+++ libgphoto2-2.4.11/packaging/generic/print-camera-list.c	2011-12-10 20:31:37.621092267 -0600
@@ -409,7 +409,7 @@
 		"ACTION!=\"add\", GOTO=\"libgphoto2_rules_end\"\n"
 		"SUBSYSTEM!=\"usb\", GOTO=\"libgphoto2_usb_end\"\n"
 		"ENV{DEVTYPE}!=\"usb_device\", GOTO=\"libgphoto2_usb_end\"\n\n"
-		"ENV{ID_USB_INTERFACES}==\"\", IMPORT{program}=\"usb_id --export %%p\"\n"
+		"ENV{ID_USB_INTERFACES}==\"\", IMPORT{builtin}=\"usb_id\"\n"
 		/* ignore mass storage class having devices in mark-up */
 		"ENV{ID_USB_INTERFACES}==\"*:08*:*\", GOTO=\"libgphoto2_usb_end\"\n"
 		/* shortcut the most common camera driver, ptp class, so we avoid parsing 1000
