--- ./socket.c.orig	2013-04-23 15:00:45.000000000 -0500
+++ ./socket.c	2018-09-26 11:56:14.957489314 -0500
@@ -1007,6 +1007,8 @@
         	SSL_use_RSAPrivateKey_file(_ssl_context[sock], mykey, SSL_FILETYPE_PEM);
 	}
 
+	SSL_set_tlsext_host_name(_ssl_context[sock],servercname);
+
 	if (SSL_set_fd(_ssl_context[sock], sock) == 0 
 	    || SSL_connect(_ssl_context[sock]) < 1) {
 		ERR_print_errors_fp(stderr);
