--- ./src/lib/pdf/pdfdocument.cpp.orig	2024-02-10 05:38:52.000000000 -0600
+++ ./src/lib/pdf/pdfdocument.cpp	2024-05-21 17:08:28.607184297 -0500
@@ -329,13 +329,9 @@
         return {};
     }
 
-    if (s->hasUnicodeMarker() || s->hasUnicodeMarkerLE()) {
-        return QString::fromUtf16(reinterpret_cast<const char16_t*>(s->toStr().c_str()), s->toStr().size() / 2);
-    } else {
         int len = 0;
         std::unique_ptr<const char[]> utf16Data(pdfDocEncodingToUTF16(s->toStr(), &len));
         return QString::fromUtf16(reinterpret_cast<const char16_t*>(utf16Data.get()), len / 2);
-    }
 
     return QString::fromUtf8(s->c_str());
 }
