--- ./configure.orig	2009-06-16 07:25:50.000000000 -0500
+++ ./configure	2010-05-03 00:12:23.000000000 -0500
@@ -27890,7 +27890,7 @@
 
   if test "x$png_REQUIRES" = x; then
     # libpng13 is GnuWin32's libpng-1.2.8 :-(
-    for l in libpng12 libpng13 libpng10 ; do
+    for l in libpng14 libpng12 libpng13 libpng10 ; do
       if $PKG_CONFIG --exists $l ; then
         png_REQUIRES=$l
         use_png=yes
