--- kdebindings-4.3.0/ruby/krossruby/rubyconfig.h.org	2009-01-16 08:04:29.000000000 -0700
+++ kdebindings-4.3.0/ruby/krossruby/rubyconfig.h	2009-08-08 23:14:59.917148261 -0600
@@ -21,12 +21,12 @@
 #define KROSS_RUBY_CONFIG_H
 
 #include <ruby.h>
-#include <env.h>
-#include <rubysig.h>
-#include <node.h>
-#include <version.h>
+//#include <env.h>
+#include <ruby/backward/rubysig.h>
+#include <ruby/node.h>
+#include <ruby/version.h>
 
-#include <st.h>
+#include <ruby/st.h>
 //#include <typeinfo>
 
 #include <kross/core/krossconfig.h>
