--- ./src/py_client.py.orig	2012-03-23 16:24:40.000000000 -0500
+++ ./src/py_client.py	2018-03-11 14:24:54.540483549 -0500
@@ -585,7 +585,8 @@
           'union'   : py_union,
           'request' : py_request,
           'event'   : py_event,
-          'error'   : py_error
+          'error'   : py_error,
+          'eventstruct' : lambda *a, **k: None
           }
 
 # Boilerplate below this point
