--- ./PyKOpeningHours/CMakeLists.txt.orig	2021-11-02 10:24:56.000000000 -0500
+++ ./PyKOpeningHours/CMakeLists.txt	2021-11-24 14:36:44.808100571 -0600
@@ -24,7 +24,7 @@
 target_link_libraries(PyKOpeningHours PUBLIC ${Boost_LIBRARIES} ${PYTHON_LIBRARIES} KOpeningHours)
 target_include_directories(PyKOpeningHours PRIVATE ${PYTHON_INCLUDE_DIRS})
 
-set(_install_dir lib/python${PYTHONLIBS_VERSION_MAJOR}.${PYTHONLIBS_VERSION_MINOR}/site-packages)
+set(_install_dir lib64/python${PYTHONLIBS_VERSION_MAJOR}.${PYTHONLIBS_VERSION_MINOR}/site-packages)
 
 install(TARGETS PyKOpeningHours DESTINATION ${_install_dir}/PyKOpeningHours)
 install(FILES PyKOpeningHours.pyi __init__.py DESTINATION ${_install_dir}/PyKOpeningHours)
