--- ./src/init.c.orig	2018-11-24 10:38:12.000000000 -0600
+++ ./src/init.c	2019-01-08 23:56:31.143232075 -0600
@@ -9,12 +9,8 @@
  * Version:     init.c 2.90 18-Jun-2018 jsmith@resonatingmedia.com
  */
 
-/*
-Version information is not placed in the top-level Makefile by default
-*/
-#ifndef VERSION
-#define VERSION "2.91"
-#endif
+#define VERSION "2.93"
+
 /*
  *		This file is part of the sysvinit suite,
  *		Copyright (C) 1991-2004 Miquel van Smoorenburg.
