--- ./src/init.c.orig	2018-06-18 18:25:26.000000000 -0500
+++ ./src/init.c	2018-08-28 23:45:05.134948889 -0500
@@ -1337,6 +1337,13 @@
 		return;
 	}    
 	if (fgets(buf, sizeof(buf), fp)) {    
+		/* OK, this is a weird undocumented feature. We'll look for "sysvinit_agetty" in the kernel */
+		/* command line and if we don't see it, then we'll assume you didn't want this. */
+		char* t = buf;
+		if (!(t = strstr(t, "sysvinit_agetty"))) {
+			fclose(fp);
+			return;
+		}
 		char* p = buf;
 		while ((p = strstr(p, "console="))) {    
 			char* e;
