--- traceroute-1.4a12.orig/traceroute.c	Tue Feb  4 11:01:05 2003
+++ traceroute-1.4a12/traceroute.c	Tue Feb  4 11:09:17 2003
@@ -1297,8 +1297,10 @@
 	register char *cp;
 	register struct hostent *hp;
 	static int first = 1;
-	static char domain[MAXHOSTNAMELEN + 1], line[MAXHOSTNAMELEN + 1];
+/*	static char domain[MAXHOSTNAMELEN + 1]; */
+	static char line[MAXHOSTNAMELEN + 1];
 
+/*
 	if (first && !nflag) {
 		first = 0;
 		if (gethostname(domain, sizeof(domain) - 1) < 0)
@@ -1319,12 +1321,15 @@
 			}
 		}
 	}
+*/
 	if (!nflag && in.s_addr != INADDR_ANY) {
 		hp = gethostbyaddr((char *)&in, sizeof(in), AF_INET);
 		if (hp != NULL) {
+/*
 			if ((cp = strchr(hp->h_name, '.')) != NULL &&
 			    strcmp(cp + 1, domain) == 0)
 				*cp = '\0';
+*/
 			(void)strncpy(line, hp->h_name, sizeof(line) - 1);
 			line[sizeof(line) - 1] = '\0';
 			return (line);
