Subject: [MAKEFLOPPIES] Fix usage printing to fit the manpage
Author: Matteo Cypriani <mcy@lm7.fr>
Bug-Debian: http://bugs.debian.org/275877
Last-Update: 2010-05-12
--- a/src/MAKEFLOPPIES
+++ b/src/MAKEFLOPPIES
@@ -169,10 +169,8 @@
 	echo 'Options:
 	-l	Local (make files in local directory, not /dev)
 	-n	Dry run (just report what would be done, do not do anything)
-	-t	Name devices for drive type
-	-d	Name devices for drive type
-	-m	Name devices for media type
-	-u	Use the same letter (u) for all 3 1/2 devices
+	-t | -d	Use the old naming convention for 3 1/2 devices (e.g. fd0H720 instead of fd0u720)
+	-m | -u	Base the name for the created devices on the type of the media (e.g. fd0h720 instead of fd0u720)
 	-g	Allow access only for group floppy
 	-v	Verbose
 ' >&2
