Subject: [MAKEFLOPPIES] Exit if devfs is detected
Author: Taral <taral@taral.net>
Bug-Debian: http://bugs.debian.org/96845
Last-Update: 2010-05-12
--- a/src/MAKEFLOPPIES
+++ b/src/MAKEFLOPPIES
@@ -10,6 +10,13 @@
 
 set -e
 set -u
+
+# check for devfs
+if [ -e /dev/.devfsd ]; then
+    echo "devfs found, skipping MAKEFLOPPIES..." >&2
+    exit 0
+fi
+
 MAJOR=2
 TMPDEVICE=/dev/tmpfloppy$$
 if [ ! -b /dev/fd0 ] ; then
