Subject: [MAKEFLOPPIES] Fix the chown calls to use ':' instead of '.'
Author: Jochen Voss <voss@debian.org>
Bug-Debian: http://bugs.debian.org/205367
Last-Update: 2010-05-12
--- a/src/MAKEFLOPPIES
+++ b/src/MAKEFLOPPIES
@@ -216,7 +216,7 @@
 	echo "Creating \"$FILE\", ID=$DRIVE, Type=$CMOS ($CN)"
 	if [ -z "$DRYRUN" ]; then
 		mknod "$FILE" b "$MAJOR" "$BASENUMBER"
-		chown root.floppy "$FILE"
+		chown root:floppy "$FILE"
 		chmod ${PERMISSION} "$FILE"
 	fi
 	if [ -n "$VERBOSE" -o -n "$DRYRUN" ]; then
@@ -240,7 +240,7 @@
 			fi
 			if [ -z "$DRYRUN" ]; then
 				mknod "${FILE}${NAME}" b "$MAJOR" "$MINOR"
-				chown root.floppy "${FILE}${NAME}"
+				chown root:floppy "${FILE}${NAME}"
 				chmod ${PERMISSION} "${FILE}${NAME}"
 			fi
 			if [ -n "$VERBOSE" -o -n "$DRYRUN" ]; then
