Subject: [superformat] Inactivate setting of variables out of environment
Author: Anthony Fok <foka@gpu.srv.ualberta.ca>
Bug-Debian: http://bugs.debian.org/12166
Last-Update: 2021-01-30
--- a/src/enh_options.c
+++ b/src/enh_options.c
@@ -528,6 +528,7 @@
 	
 
 	/* setting of variables out of environment */
+/* Commented out on 1998-08-28 by AF, Re: Debian Bug#12166
 	for( option_index=0; option_index<length; option_index++){
 		if ( !my_xopts[option_index] ){
 			s=getenv(eo[option_index].longo);
@@ -539,6 +540,7 @@
 			}
 		}
 	}
+*/
 	if ( ret ){
 		print_usage(argv[0], eo, userparams);
 		return '?';
