--- screen-4.4.0/etc/etcscreenrc	2015-09-05 14:28:22.000000000 -0500
+++ screen-4.2.1/etc/etcscreenrc	2014-04-25 14:08:40.000000000 -0500
@@ -68,8 +68,8 @@
 #
 # Do not use xterms alternate window buffer. 
 # This one would not add lines to the scrollback buffer.
-termcap xterm|xterms|xs ti=\E7\E[?47l
-terminfo xterm|xterms|xs ti=\E7\E[?47l
+#termcap xterm|xterms|xs ti=\E7\E[?47l
+#terminfo xterm|xterms|xs ti=\E7\E[?47l
 
 #make hp700 termcap/info better
 termcap  hp700 'Z0=\E[?3h:Z1=\E[?3l:hs:ts=\E[62"p\E[0$~\E[2$~\E[1$}:fs=\E[0}\E[61"p:ds=\E[62"p\E[1$~\E[61"p:ic@'
