--- ./doc/screen.texinfo.orig	2025-05-13 13:22:32.861053013 -0500
+++ ./doc/screen.texinfo	2025-05-13 13:27:35.290059779 -0500
@@ -5940,7 +5940,7 @@
 hardware and operating system, the compiler used to compile, a
 description of the bug behavior, and the conditions that triggered the
 bug. In case of security issue please send an mail to
-@w{@samp{alexander_naumov@opensuse.org}}.
+@w{@samp{alexander_naumov@@opensuse.org}}.
 
 @node Availability,  , Reporting Bugs, Bugs
 @section Availability
