--- ./src/CMakeLists.txt.orig	2024-05-04 06:37:54.000000000 -0500
+++ ./src/CMakeLists.txt	2025-04-10 13:32:54.283487753 -0500
@@ -13,7 +13,7 @@
 # so simply set standard for the while library
 # Current version of FindGperf in ECM does not support a version check yet,
 # so for now always lower to C++14.
-set_target_properties(KF5Codecs PROPERTIES CXX_STANDARD 14)
+set_target_properties(KF5Codecs PROPERTIES CXX_STANDARD 17)
 
 target_sources(KF5Codecs PRIVATE
     kcharsets.cpp
