--- ./scripts/90-xcm-i2c.rules.orig	2016-12-09 15:50:46.000000000 -0600
+++ ./scripts/90-xcm-i2c.rules	2018-11-11 15:25:36.438669935 -0600
@@ -1,8 +1,11 @@
 # load the device node kernel module
-SUBSYSTEM=="i2c-adapter", ACTION=="add", RUN+="/sbin/modprobe i2c-dev"
+#SUBSYSTEM=="i2c-adapter", ACTION=="add", RUN+="/sbin/modprobe i2c-dev"
 
-# openSUSE_11.3 preferes:
-SUBSYSTEM=="i2c", RUN+="/sbin/modprobe -bv i2c-dev"
+# openSUSE_11.3 prefers:
+#SUBSYSTEM=="i2c", RUN+="/sbin/modprobe -bv i2c-dev"
+
+# Here's what Slackware uses:
+SUBSYSTEM=="i2c", RUN+="/sbin/modprobe -b i2c-dev"
 
 # match only DDC devices
 KERNEL=="i2c-[0-9]*", ACTION=="add", PROGRAM=="/usr/bin/xcmddc --i2c /dev/%k --identify", RESULT=="0", GROUP="video", MODE="0660"
