diff -Nur logrotate-3.7.8.orig/Makefile logrotate-3.7.8/Makefile
--- logrotate-3.7.8.orig/Makefile	2008-05-19 05:25:54.000000000 -0500
+++ logrotate-3.7.8/Makefile	2009-05-09 23:25:12.060201375 -0500
@@ -1,7 +1,7 @@
 VERSION = $(shell awk '/Version:/ { print $$2 }' logrotate.spec)
 OS_NAME = $(shell uname -s)
 LFS = $(shell echo `getconf LFS_CFLAGS 2>/dev/null`)
-CFLAGS = -Wall -D_GNU_SOURCE -D$(OS_NAME) -DVERSION=\"$(VERSION)\" $(RPM_OPT_FLAGS) $(LFS)
+CFLAGS = -Wall $(EXTRA_CFLAGS) -D_GNU_SOURCE -D$(OS_NAME) -DVERSION=\"$(VERSION)\" $(RPM_OPT_FLAGS) $(LFS)
 PROG = logrotate
 MAN = logrotate.8
 LOADLIBES = -lpopt
diff -Nur logrotate-3.7.8.orig/config.c logrotate-3.7.8/config.c
--- logrotate-3.7.8.orig/config.c	2008-06-02 05:26:14.000000000 -0500
+++ logrotate-3.7.8/config.c	2009-05-09 23:53:36.748287318 -0500
@@ -33,8 +33,9 @@
 #define isblank(c) 	( (c) == ' ' || (c) == '\t' ) ? 1 : 0
 #endif
 
-static char *defTabooExts[] = { ".rpmsave", ".rpmorig", "~", ",v",
-    ".rpmnew", ".swp", ".cfsaved", ".rhn-cfg-tmp-*"
+static char *defTabooExts[] = { ".new", ".orig", ".old", ".bak",
+    ".rpmsave", ".rpmorig", "~", ",v", ".rpmnew", ".swp", ".cfsaved", 
+    ".rhn-cfg-tmp-*"
 };
 static int defTabooCount = sizeof(defTabooExts) / sizeof(char *);
 
diff -Nur logrotate-3.7.8.orig/config.h logrotate-3.7.8/config.h
--- logrotate-3.7.8.orig/config.h	2005-11-12 13:07:56.000000000 -0600
+++ logrotate-3.7.8/config.h	2009-05-09 23:58:17.040241145 -0500
@@ -13,7 +13,7 @@
 #define DEFAULT_MAIL_COMMAND "/usr/bin/mailx"
 #define COMPRESS_COMMAND "/usr/local/bin/gzip"
 #define UNCOMPRESS_COMMAND "/usr/local/bin/gunzip"
-#define STATEFILE "/var/log/logrotate.status"
+#define STATEFILE "/var/lib/logrotate/status"
 #endif
 
 /*
diff -Nur logrotate-3.7.8.orig/logrotate.8 logrotate-3.7.8/logrotate.8
--- logrotate-3.7.8.orig/logrotate.8	2008-12-06 08:05:40.000000000 -0600
+++ logrotate-3.7.8/logrotate.8	2009-05-09 23:56:45.137687740 -0500
@@ -442,7 +442,7 @@
 for information on the taboo extensions). If a + precedes the list of
 extensions, the current taboo extension list is augmented, otherwise it
 is replaced. At startup, the taboo extension list 
-contains .rpmorig, .rpmsave, ,v, .swp, .rpmnew, ~, .cfsaved and .rhn-cfg-tmp-*.
+contains .new, .orig, .bak, .old, .rpmorig, .rpmsave, ,v, .swp, .rpmnew, ~, .cfsaved and .rhn-cfg-tmp-*.
 
 .TP
 \fBweekly\fR
@@ -459,7 +459,7 @@
 .SH FILES
 .PD 0
 .TP 27
-\fI/var/lib/logrotate.status\fR
+\fI/var/lib/logrotate/status\fR
 Default state file.
 .TP 27
 \fI/etc/logrotate.conf\fR
