--- ./logrotate.8.orig	2018-12-04 09:10:46.000000000 -0600
+++ ./logrotate.8	2018-12-13 18:26:31.756882880 -0600
@@ -608,6 +608,10 @@
 .IR .ucf\-dist ,
 .IR .ucf\-new ,
 .IR .ucf\-old ,
+.IR .new
+.IR .old
+.IR .orig
+.IR .bak
 .IR ~
 
 .TP
--- ./logrotate.8.in.orig	2018-08-09 04:17:37.000000000 -0500
+++ ./logrotate.8.in	2018-12-13 18:27:14.471878890 -0600
@@ -608,6 +608,10 @@
 .IR .ucf\-dist ,
 .IR .ucf\-new ,
 .IR .ucf\-old ,
+.IR .new
+.IR .old
+.IR .orig
+.IR .bak
 .IR ~
 
 .TP
--- ./config.c.orig	2018-10-05 10:01:21.000000000 -0500
+++ ./config.c	2018-12-13 18:25:29.650888682 -0600
@@ -149,6 +149,10 @@
     ".ucf-dist",
     ".ucf-new",
     ".ucf-old",
+    ".new",
+    ".old",
+    ".orig",
+    ".bak",
     "~"
 };
 static const int defTabooCount = sizeof(defTabooExts) / sizeof(char *);
