diff -ur rdist-6.1.5.orig/Makefile rdist-6.1.5/Makefile
--- rdist-6.1.5.orig/Makefile	Mon Nov  9 19:36:06 1998
+++ rdist-6.1.5/Makefile	Tue Apr  3 10:56:28 2001
@@ -17,9 +17,10 @@
 SHELL=/bin/sh
 
 all install install.man: FRC
-	@for t in $(TARGETS); do \
+	@for t in $(TARGETS); do ( \
 		echo Making \"$@\" in \"$$t\"; \
 		(cd $$t; ${MAKE} $@); \
+		) || exit 1; \
 	done
 
 clean:
diff -ur rdist-6.1.5.orig/Makefile.local rdist-6.1.5/Makefile.local
--- rdist-6.1.5.orig/Makefile.local	Mon Nov  9 19:36:31 1998
+++ rdist-6.1.5/Makefile.local	Tue Apr  3 10:56:28 2001
@@ -15,6 +15,20 @@
 # or add your own line.
 #
 #BIN_DIR	= /usr/bin
+BIN_DIR = /usr/bin
+MAN_1_DIR = /usr/man/man1
+MAN_8_DIR = /usr/man/man8
+RDIST_MODE = 755
+RDISTD_MODE = 755
+MAN_MODE = 644
+MAN_OWNER = root
+MAN_GROUP = root
+BIN_GROUP = root
+
+INSTALL = install
+YACC = bison -y
+
+OPT = -Wall -O2 -pipe
 
 #
 # Add any local definitions you want pass to the compiler to DEFS_LOCAL
Only in rdist-6.1.5: Makefile.local~
Only in rdist-6.1.5: Makefile~
diff -ur rdist-6.1.5.orig/config/os-linux.h rdist-6.1.5/config/os-linux.h
--- rdist-6.1.5.orig/config/os-linux.h	Mon Nov  9 19:59:59 1998
+++ rdist-6.1.5/config/os-linux.h	Tue Apr  3 10:56:28 2001
@@ -93,8 +93,8 @@
 /*
  * Our types, usually these are uid_t and gid_t.
  */
-typedef uid_t UID_T;	/* Must be signed */
-typedef gid_t GID_T;	/* Must be signed */
+typedef int UID_T;	/* Must be signed */
+typedef int GID_T;	/* Must be signed */
 
 /*
  * Generic pointer, used by memcpy, malloc, etc.  Usually char or void.
Only in rdist-6.1.5/config: os-linux.h~
diff -ur rdist-6.1.5.orig/doc/Makefile.real rdist-6.1.5/doc/Makefile.real
--- rdist-6.1.5.orig/doc/Makefile.real	Mon Nov  9 21:40:06 1998
+++ rdist-6.1.5/doc/Makefile.real	Tue Apr  3 10:56:28 2001
@@ -16,7 +16,7 @@
 
 doc:
 
-install:
+install: install.man
 
 install.man: rdist.man rdistd.man
 	${INSTALL} ${INSTALL_ARGS} rdist.man ${CLIENT_DEST}
Only in rdist-6.1.5/doc: Makefile.real~
diff -ur rdist-6.1.5.orig/doc/rdist.man rdist-6.1.5/doc/rdist.man
--- rdist-6.1.5.orig/doc/rdist.man	Mon Nov  9 21:38:53 1998
+++ rdist-6.1.5/doc/rdist.man	Tue Apr  3 10:56:28 2001
@@ -171,7 +171,7 @@
 .I rdist
 was compiled with the location of the old rdist
 (usually either
-.I /usr/ucb/oldrdist
+.I /usr/bin/oldrdist
 or
 .I /usr/old/rdist)
 and that program is available at run time.
@@ -511,7 +511,7 @@
 may be a colon seperated list of possible pathnames.
 In this case, the first component of the path to exist is used.
 i.e. 
-.B "/usr/ucb/rsh:/usr/bin/remsh",
+.B "/usr/bin/rsh:/usr/bin/remsh",
 .B /usr/bsd/rsh.
 .TP
 .B "\-t \fItimeout\fR"
Only in rdist-6.1.5/doc: rdist.man~
diff -ur rdist-6.1.5.orig/src/Makefile.real rdist-6.1.5/src/Makefile.real
--- rdist-6.1.5.orig/src/Makefile.real	Mon Nov  9 20:03:46 1998
+++ rdist-6.1.5/src/Makefile.real	Tue Apr  3 10:56:28 2001
@@ -52,9 +52,9 @@
 		$(COMMONOBJS) $(MISSINGOBJS)
 
 install:
-	$(INSTALL) ${IN_ARGS} -o $(BIN_OWNER) -m ${RDIST_MODE} \
+	$(INSTALL) ${IN_ARGS} -s -o $(BIN_OWNER) -m ${RDIST_MODE} \
 		$(CLIENT_BIN) $(BIN_DIR)/$(CLIENT)
-	$(INSTALL) ${IN_ARGS} -o ${BIN_OWNER} -m ${RDISTD_MODE} \
+	$(INSTALL) ${IN_ARGS} -s -o ${BIN_OWNER} -m ${RDISTD_MODE} \
 		$(SERVER_BIN) $(BIN_DIR)/$(SERVER)
 
 install.man:
Only in rdist-6.1.5/src: Makefile.real~
