--- ./less.nro.orig	Mon May 27 17:14:24 2002
+++ ./less.nro	Mon May 27 17:14:46 2002
@@ -947,10 +947,10 @@
 Otherwise,
 .I less 
 looks in a standard place for the system-wide lesskey file:
-On Unix systems, the system-wide lesskey file is /usr/local/etc/sysless.
+On Unix systems, the system-wide lesskey file is /etc/sysless.
 (However, if 
 .I less 
-was built with a different sysconf directory than /usr/local/etc,
+was built with a different sysconf directory than /etc,
 that directory is where the sysless file is found.)
 On MS-DOS and Windows systems, the system-wide lesskey file is c:\\_sysless.
 On OS/2 systems, the system-wide lesskey file is c:\\sysless.ini.
