--- ./src/metamail/Makefile.in.orig	2024-05-10 16:12:29.131173303 -0500
+++ ./src/metamail/Makefile.in	2024-05-10 16:18:58.068182003 -0500
@@ -82,7 +82,7 @@
 
 MAINTAINERCLEANFILES = Makefile.in
 
-CFLAGS = -lncurses
+CFLAGS = -lncurses -Wno-error=return-mismatch -Wno-error=implicit-function-declaration -Wno-error=implicit-int
 
 INCLUDES = -I$(top_builddir) -I$(top_builddir)/metamail                         -I$(top_srcdir) -I$(top_srcdir)/metamail
 
--- ./src/richmail/Makefile.in.orig	2024-05-10 16:12:29.130173303 -0500
+++ ./src/richmail/Makefile.in	2024-05-10 16:20:28.612184029 -0500
@@ -82,7 +82,7 @@
 
 MAINTAINERCLEANFILES = Makefile.in
 
-CFLAGS = -lncurses
+CFLAGS = -lncurses -Wno-error=return-mismatch -Wno-error=implicit-function-declaration -Wno-error=implicit-int
 
 INCLUDES = -I$(top_builddir) -I$(top_builddir)/metamail                         -I$(top_srcdir) -I$(top_srcdir)/metamail
 
