diff -U3 -d -r -N xfce4-pulseaudio-plugin-0.2.4/panel-plugin/pulseaudio-button.c xfce4-pulseaudio-plugin-0.2.4.fixed/panel-plugin/pulseaudio-button.c
--- xfce4-pulseaudio-plugin-0.2.4/panel-plugin/pulseaudio-button.c	2015-06-06 23:57:55.000000000 +0200
+++ xfce4-pulseaudio-plugin-0.2.4.fixed/panel-plugin/pulseaudio-button.c	2016-06-22 21:51:07.258482863 +0200
@@ -137,7 +137,7 @@
 
   /* Setup Gtk style */
   css_provider = gtk_css_provider_new ();
-  gtk_css_provider_load_from_data (css_provider, "#pulseaudio-button { -GtkWidget-focus-padding: 0; -GtkWidget-focus-line-width: 0; -GtkButton-default-border: 0; -GtkButton-inner-border: 0; padding: 1px; border-width: 1px;}", -1, NULL);
+  gtk_css_provider_load_from_data (css_provider, "#pulseaudio-button { -GtkWidget-focus-padding: 0; -GtkWidget-focus-line-width: 0; -GtkButton-default-border: 0; -GtkButton-inner-border: 0; padding: 1px; border-width: 0px; border-radius: 0px; background-color:@theme_bg_color;}", -1, NULL);
   gtk_style_context_add_provider (GTK_STYLE_CONTEXT (gtk_widget_get_style_context (GTK_WIDGET (button))), GTK_STYLE_PROVIDER (css_provider), GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
 
   /* Intercept scroll events */
