--- ./iproc.c.orig	2008-05-20 21:35:18.000000000 -0500
+++ ./iproc.c	2019-08-19 17:28:01.899194246 -0500
@@ -536,14 +536,6 @@
 
 # ifdef SVR4_PTYS
 #  include <stdlib.h>	/* for grantpt and unlockpt, at least in Solaris 2.3 */
-#  if _XOPEN_SOURCE >= 500
-    /* Linux/glibc no longer pretends to support STREAMS (XSR) (2008) */
-#   if _XOPEN_STREAMS != -1
-#    include <stropts.h>
-#   endif
-#  else
-#   include <sys/stropts.h>
-#  endif
   extern char	*ptsname proto((int /*filedes*/));	/* get name of slave */
 # endif
 
