--- ./fp.h.orig	2002-03-20 15:49:38.000000000 -0600
+++ ./fp.h	2018-08-16 12:43:15.763002319 -0500
@@ -17,7 +17,8 @@
 # define scr_putchar(c)	f_putc((c), jstdout)
 extern void		flushscreen proto((void));
 # ifndef SMALL
-#  define MAXTTYBUF	2048
+// TTK here -- upping from 2048 to 256K
+#  define MAXTTYBUF	262144
 # else
 #  define MAXTTYBUF	512
 # endif
--- ./insert.c.orig	2002-03-20 15:49:38.000000000 -0600
+++ ./insert.c	2018-08-16 12:43:15.773002319 -0500
@@ -504,7 +506,7 @@
 		atchar = 0;
 	}
 
-	getline(atline->l_dline, genbuf);
+	j_getline(atline->l_dline, genbuf);
 	atchar += tchar;
 	linecopy(genbuf, atchar, save);
 	atline->l_dline = putline(genbuf);
--- ./jove.h.orig	2002-03-20 15:49:38.000000000 -0600
+++ ./jove.h	2018-08-16 12:43:15.797002319 -0500
@@ -267,7 +267,8 @@
 
 /* term.c: universal termcap-like declarations */
 
-#define MAXCOLS		256	/* maximum number of columns */
+// TTK here -- boosting to 256K
+#define MAXCOLS		262144	/* maximum number of columns */
 
 extern int
 	SG,		/* number of magic cookies left by SO and SE */
--- ./io.h.orig	2002-03-20 15:49:38.000000000 -0600
+++ ./io.h	2018-08-16 12:43:15.790002319 -0500
@@ -40,7 +40,7 @@
 	close_file proto((File *fp)),
 	d_cache_init proto((void)),
 	file_write proto((char *fname, bool app)),
-	getline proto((daddr addr, char *buf)),
+	j_getline proto((daddr addr, char *buf)),
 	lsave proto((void)),
 	putreg proto((File *fp,LinePtr line1,int char1,LinePtr line2,int char2,bool makesure)),
 	read_file proto((char *file, bool is_insert)),
--- ./io.c.orig	2010-04-16 00:41:46.000000000 -0500
+++ ./io.c	2018-08-16 12:43:15.782002319 -0500
@@ -1356,9 +1356,9 @@
 
 void
 #ifdef USE_PROTOTYPES
-getline proto((daddr addr, register char *buf))
+j_getline proto((daddr addr, register char *buf))
 #else
-getline(addr, buf)
+j_getline(addr, buf)
 daddr	addr;
 register char	*buf;
 #endif
