--- ./src/login.c.orig	2023-09-25 10:52:15.000000000 -0500
+++ ./src/login.c	2023-10-24 20:24:47.101964317 -0500
@@ -705,6 +705,15 @@
 
 		/* Make the login prompt look like we want it */
 		if (gethostname (hostn, sizeof (hostn)) == 0) {
+			/* Trim away everything after the first '.': */
+			int i = 0;
+			while (hostn[i] != '\0' && i < sizeof(hostn) - 1) {
+				if (hostn[i] == '.') {
+					hostn[i] = '\0';
+					break;
+				}
+				i++;
+			}
 			snprintf (loginprompt,
 			          sizeof (loginprompt),
 			          _("%s login: "), hostn);
--- ./lib/loginprompt.c.orig	2023-09-25 10:52:15.000000000 -0500
+++ ./lib/loginprompt.c	2023-10-24 20:24:47.101964317 -0500
@@ -71,6 +71,15 @@
 		}
 	}
 	(void) gethostname (buf, sizeof buf);
+        /* Trim away everything after the first '.': */
+        i = 0;
+        while (buf[i] != '\0' && i < sizeof(buf) - 1) {
+                if (buf[i] == '.') {
+                        buf[i] = '\0';
+                        break;
+                }
+                i++;
+        }
 	printf (_("\n%s login: "), buf);
 	(void) fflush (stdout);
 
