--- ./libmisc/loginprompt.c.orig	2014-03-01 12:59:51.000000000 -0600
+++ ./libmisc/loginprompt.c	2017-11-21 18:17:27.492000123 -0600
@@ -99,6 +99,15 @@
 			}
 		}
 		(void) gethostname (buf, sizeof buf);
+		/* Trim away everything after the first '.': */
+		i = 0;
+		while (buf[i] != '\0' && i < sizeof(buf) - 1) {
+			if (buf[i] == '.') {
+				buf[i] = '\0';
+				break;
+			}
+			i++;
+		}
 		printf (prompt, buf);
 		(void) fflush (stdout);
 	}
