--- ./man/ru/Makefile.in.orig	2011-02-15 16:18:21.000000000 -0600
+++ ./man/ru/Makefile.in	2012-06-26 23:21:34.866292111 -0500
@@ -228,7 +228,7 @@
 	grpunconv.8 gshadow.5 lastlog.8 login.1 login.defs.5 logoutd.8 \
 	newgrp.1 newusers.8 nologin.8 passwd.1 passwd.5 pwck.8 \
 	pwconv.8 pwunconv.8 sg.1 shadow.3 shadow.5 su.1 suauth.5 \
-	useradd.8 userdel.8 usermod.8 vigr.8 vipw.8 $(am__append_1)
+	useradd.8 userdel.8 usermod.8 vigr.8 vipw.8
 man_nopam = \
 	limits.5 \
 	login.access.5 \
