--- ./lib/loginprompt.c.orig	2024-02-29 19:50:52.000000000 -0600
+++ ./lib/loginprompt.c	2024-03-09 13:29:58.373925583 -0600
@@ -71,6 +71,15 @@
 		}
 	}
 	(void) gethostname (buf, sizeof buf);
+        /* Trim away everything after the first '.': */
+        i = 0;
+        while (buf[i] != '\0' && i < sizeof(buf) - 1) {
+                if (buf[i] == '.') {
+                        buf[i] = '\0';
+                        break;
+                }
+                i++;
+        }
 	printf (_("\n%s login: "), buf);
 	(void) fflush (stdout);
 
--- ./src/login.c.orig	2024-03-09 13:29:58.373925583 -0600
+++ ./src/login.c	2024-03-09 13:31:39.344919998 -0600
@@ -703,6 +703,15 @@
 
 		/* Make the login prompt look like we want it */
 		if (gethostname (hostn, sizeof (hostn)) == 0) {
+                        /* Trim away everything after the first '.': */
+                        int i = 0;
+                        while (hostn[i] != '\0' && i < sizeof(hostn) - 1) {
+                                if (hostn[i] == '.') {
+                                        hostn[i] = '\0';
+                                        break;
+                                }
+                                i++;
+                        }
 			SNPRINTF(loginprompt, _("%s login: "), hostn);
 		} else {
 			STRTCPY(loginprompt, _("login: "));
