--- ./runtime/vimrc.new.orig	2008-07-02 15:11:53.000000000 -0500
+++ ./runtime/vimrc.new	2009-05-28 13:19:24.000000000 -0500
@@ -87,6 +87,9 @@
 
 endif " has("autocmd")
 
+" Make vim work with the 'crontab -e' command
+set backupskip+=/var/spool/cron/*
+
 " Convenient command to see the difference between the current buffer and the
 " file it was loaded from, thus the changes you made.
 " Only define it when not defined already.
