--- ./src/lex.c.orig	1993-12-07 23:36:49.000000000 -0600
+++ ./src/lex.c	2018-02-01 15:27:54.224877642 -0600
@@ -1044,7 +1044,7 @@
 
 
 
-void getline()
+void p2c_getline()
 {
     char *cp, *cp2;
 
@@ -1071,7 +1071,7 @@
 			infname = stralloc(cp);
 			infname[cp2 - cp] = 0;
 		    }
-		    getline();
+		    p2c_getline();
 		    return;
 		}
 		if (copysource && *inbuf) {
@@ -1088,7 +1088,7 @@
                     fprintf(stderr, "\n");
                 if (inputkind == INP_INCFILE) {
                     pop_input();
-                    getline();
+                    p2c_getline();
                 } else
                     strcpy(inbuf, "\001");
             }
@@ -1179,7 +1179,7 @@
         infname = fname;
         inf_lnum = 0;
     } else
-        inf_lnum--;     /* adjust for extra getline() */
+        inf_lnum--;     /* adjust for extra p2c_getline() */
     *inbuf = 0;
     inbufptr = inbuf;
     gettok();
@@ -2367,7 +2367,7 @@
 		else
 		    commentline(CMT_POST);
 		trailing = 0;
-                getline();
+                p2c_getline();
 		i = 0;
 		for (;;) {
 		    if (*inbufptr == ' ') {
@@ -2419,7 +2419,7 @@
         if (isspace(*inbufptr)) {
             inbufptr++;
         } else if (!*inbufptr) {
-            getline();
+            p2c_getline();
         } else if (*inbufptr == '{') {
             inbufptr++;
             comment(0);
@@ -2513,7 +2513,7 @@
         switch (*inbufptr++) {
 
             case 0:
-                getline();
+                p2c_getline();
                 break;
 
             case ' ':
@@ -2765,7 +2765,7 @@
             case 0:
 	        if (commenting_flag)
 		    saveinputcomment(inbufptr-1);
-                getline();
+                p2c_getline();
 		cp = curtokbuf;
 		for (;;) {
 		    inbufindent = 0;
@@ -2782,7 +2782,7 @@
 		    }
 		    if (!*inbufptr && !commenting_flag) {   /* blank line */
 			*cp++ = '\001';
-			getline();
+			p2c_getline();
 		    } else
 			break;
 		}
@@ -2797,10 +2797,10 @@
 		*cp++ = '\001';
 		*cp++ = '\014';
 		if (!*inbufptr && !commenting_flag) {
-		    getline();
+		    p2c_getline();
 		    while (!*inbufptr) {
 			*cp++ = '\001';
-			getline();
+			p2c_getline();
 		    }
 		}
 		*cp = 0;
--- ./src/parse.c.orig	2018-02-01 15:25:56.206878896 -0600
+++ ./src/parse.c	2018-02-01 15:28:00.997877570 -0600
@@ -5073,7 +5073,7 @@
 	out_include(fname, 1);
     outsection(majorspace);
     pop_input();
-    getline();
+    p2c_getline();
     gettok();
 }
 
--- ./src/TAGS.orig	1993-12-07 23:36:55.000000000 -0600
+++ ./src/TAGS	2018-02-01 15:27:50.927877677 -0600
@@ -176,7 +176,7 @@
 char getchartok(2466,60243
 Static int getflag(2451,60064
 char *getinlinepart(2414,59241
-void getline(1047,22000
+void p2c_getline(1047,22000
 char *getparenstr(2481,60499
 void gettok(2728,65611
 void gettok(2752,66193
