diff --git a/util/grub.d/00_header.in b/util/grub.d/00_header.in
index 46a0e47ac..3c26ce286 100644
--- a/util/grub.d/00_header.in
+++ b/util/grub.d/00_header.in
@@ -175,7 +175,6 @@ EOF
     font="`make_system_path_relative_to_its_root "${font_path}"`"
 fi
 
-if loadfont \$font ; then
 EOF
 	    else
     cat << EOF
@@ -184,24 +183,12 @@ EOF
 	    fi
 	fi
 
-    cat << EOF
-  set gfxmode=${GRUB_GFXMODE}
-  load_video
-  insmod gfxterm
-EOF
-
 # Gettext variables and module
 if [ "x${grub_lang}" != "xC" ] && [ "x${LANG}" != "xPOSIX" ] && [ "x${LANG}" != "x" ]; then
   cat << EOF
-  set locale_dir=\$prefix/locale
-  set lang=${grub_lang}
-  insmod gettext
 EOF
 fi
 
-cat <<EOF
-fi
-EOF
 fi
 
 case x${GRUB_TERMINAL_INPUT} in
