--- ./util/grub-mkimagexx.c.orig	2017-04-24 04:16:00.000000000 -0500
+++ ./util/grub-mkimagexx.c	2018-09-22 13:35:33.089870295 -0500
@@ -832,6 +832,7 @@
 		  break;
 
 		case R_X86_64_PC32:
+		case R_X86_64_PLT32:
 		  {
 		    grub_uint32_t *t32 = (grub_uint32_t *) target;
 		    *t32 = grub_host_to_target64 (grub_target_to_host32 (*t32)
--- ./util/grub-module-verifier.c.orig	2017-04-24 04:16:00.000000000 -0500
+++ ./util/grub-module-verifier.c	2018-09-22 13:36:05.328867627 -0500
@@ -19,6 +19,7 @@
       -1
     }, (int[]){
       R_X86_64_PC32,
+      R_X86_64_PLT32,
       -1
     }
   },
