--- ./util/grub.d/00_header.in.orig	2012-05-03 18:04:39.000000000 -0500
+++ ./util/grub.d/00_header.in	2013-09-17 01:43:37.870841064 -0500
@@ -143,7 +143,7 @@
 EOF
     else
 	for dir in "${pkgdatadir}" "`echo '/@bootdirname@/@grubdirname@' | sed "s,//*,/,g"`" /usr/share/grub ; do
-	    for basename in unicode unifont ascii; do
+	    for basename in unicode unifont dejavusansmono ascii; do
 		path="${dir}/${basename}.pf2"
 		if is_path_readable_by_grub "${path}" > /dev/null ; then
 		    font_path="${path}"
@@ -156,7 +156,7 @@
 	if [ -n "${font_path}" ] ; then
     cat << EOF
 if [ x\$feature_default_font_path = xy ] ; then
-   font=unicode
+   font=dejavusansmono
 else
 EOF
                 # Make the font accessible
@@ -169,7 +169,7 @@
 EOF
 	    else
     cat << EOF
-if loadfont unicode ; then
+if loadfont dejavusansmono ; then
 EOF
 	    fi
 	fi
--- ./util/grub-install.in.orig	2012-06-26 16:31:03.000000000 -0500
+++ ./util/grub-install.in	2013-09-17 01:42:02.678834196 -0500
@@ -552,6 +552,11 @@
     cp "${pkgdatadir}"/themes/starfield/* "${grubdir}"/themes/starfield
 fi
 
+if test -f "${pkgdatadir}"/dejavusansmono.pf2; then
+    mkdir -p "${grubdir}"/fonts
+    cp "${pkgdatadir}"/dejavusansmono.pf2 "${grubdir}"/fonts
+fi
+
 if test -f "${pkgdatadir}"/unicode.pf2; then
     mkdir -p "${grubdir}"/fonts
     cp "${pkgdatadir}"/unicode.pf2 "${grubdir}"/fonts
