diff -Nur ppp-2.4.7.orig/chat/Makefile.linux ppp-2.4.7/chat/Makefile.linux
--- ppp-2.4.7.orig/chat/Makefile.linux	2014-08-09 07:31:39.000000000 -0500
+++ ppp-2.4.7/chat/Makefile.linux	2014-08-09 17:10:27.423374415 -0500
@@ -2,7 +2,7 @@
 
 DESTDIR = $(INSTROOT)@DESTDIR@
 BINDIR = $(DESTDIR)/sbin
-MANDIR = $(DESTDIR)/share/man/man8
+MANDIR = $(DESTDIR)/man/man8
 
 CDEF1=	-DTERMIOS			# Use the termios structure
 CDEF2=	-DSIGTYPE=void			# Standard definition
diff -Nur ppp-2.4.7.orig/linux/Makefile.top ppp-2.4.7/linux/Makefile.top
--- ppp-2.4.7.orig/linux/Makefile.top	2014-08-09 07:31:39.000000000 -0500
+++ ppp-2.4.7/linux/Makefile.top	2014-08-09 17:11:14.156384165 -0500
@@ -3,7 +3,7 @@
 DESTDIR = $(INSTROOT)@DESTDIR@
 BINDIR = $(DESTDIR)/sbin
 INCDIR = $(DESTDIR)/include
-MANDIR = $(DESTDIR)/share/man
+MANDIR = $(DESTDIR)/man
 ETCDIR = $(INSTROOT)@SYSCONF@/ppp
 
 # uid 0 = root
diff -Nur ppp-2.4.7.orig/pppd/Makefile.linux ppp-2.4.7/pppd/Makefile.linux
--- ppp-2.4.7.orig/pppd/Makefile.linux	2014-08-09 07:31:39.000000000 -0500
+++ ppp-2.4.7/pppd/Makefile.linux	2014-08-09 17:07:27.458308345 -0500
@@ -6,7 +6,7 @@
 # Default installation locations
 DESTDIR = $(INSTROOT)@DESTDIR@
 BINDIR = $(DESTDIR)/sbin
-MANDIR = $(DESTDIR)/share/man/man8
+MANDIR = $(DESTDIR)/man/man8
 INCDIR = $(DESTDIR)/include
 
 TARGETS = pppd
@@ -32,7 +32,7 @@
 
 # CC = gcc
 #
-COPTS = -O2 -pipe -Wall -g
+COPTS = -O2 -pipe -Wall
 LIBS =
 
 # Uncomment the next 2 lines to include support for Microsoft's
@@ -68,7 +68,7 @@
 PLUGIN=y
 
 # Enable Microsoft proprietary Callback Control Protocol
-#CBCP=y
+CBCP=y
 
 # Enable EAP SRP-SHA1 authentication (requires libsrp)
 #USE_SRP=y
diff -Nur ppp-2.4.7.orig/pppd/plugins/Makefile.linux ppp-2.4.7/pppd/plugins/Makefile.linux
--- ppp-2.4.7.orig/pppd/plugins/Makefile.linux	2014-08-09 07:31:39.000000000 -0500
+++ ppp-2.4.7/pppd/plugins/Makefile.linux	2014-08-09 17:11:10.734383553 -0500
@@ -6,7 +6,7 @@
 
 DESTDIR = $(INSTROOT)@DESTDIR@
 BINDIR = $(DESTDIR)/sbin
-MANDIR = $(DESTDIR)/share/man/man8
+MANDIR = $(DESTDIR)/man/man8
 LIBDIR = $(DESTDIR)/lib/pppd/$(VERSION)
 
 SUBDIRS := rp-pppoe pppoatm pppol2tp
diff -Nur ppp-2.4.7.orig/pppd/plugins/radius/Makefile.linux ppp-2.4.7/pppd/plugins/radius/Makefile.linux
--- ppp-2.4.7.orig/pppd/plugins/radius/Makefile.linux	2014-08-09 07:31:39.000000000 -0500
+++ ppp-2.4.7/pppd/plugins/radius/Makefile.linux	2014-08-09 17:11:08.470383139 -0500
@@ -4,7 +4,7 @@
 #
 
 DESTDIR = $(INSTROOT)@DESTDIR@
-MANDIR = $(DESTDIR)/share/man/man8
+MANDIR = $(DESTDIR)/man/man8
 LIBDIR = $(DESTDIR)/lib/pppd/$(VERSION)
 
 VERSION = $(shell awk -F '"' '/VERSION/ { print $$2; }' ../../patchlevel.h)
diff -Nur ppp-2.4.7.orig/pppdump/Makefile.linux ppp-2.4.7/pppdump/Makefile.linux
--- ppp-2.4.7.orig/pppdump/Makefile.linux	2014-08-09 07:31:39.000000000 -0500
+++ ppp-2.4.7/pppdump/Makefile.linux	2014-08-09 17:10:35.672376354 -0500
@@ -1,6 +1,6 @@
 DESTDIR = $(INSTROOT)@DESTDIR@
 BINDIR = $(DESTDIR)/sbin
-MANDIR = $(DESTDIR)/share/man/man8
+MANDIR = $(DESTDIR)/man/man8
 
 CFLAGS= -O -I../include/net
 OBJS = pppdump.o bsd-comp.o deflate.o zlib.o
diff -Nur ppp-2.4.7.orig/pppstats/Makefile.linux ppp-2.4.7/pppstats/Makefile.linux
--- ppp-2.4.7.orig/pppstats/Makefile.linux	2014-08-09 07:31:39.000000000 -0500
+++ ppp-2.4.7/pppstats/Makefile.linux	2014-08-09 17:10:42.293377843 -0500
@@ -4,7 +4,7 @@
 #
 DESTDIR = $(INSTROOT)@DESTDIR@
 BINDIR = $(DESTDIR)/sbin
-MANDIR = $(DESTDIR)/share/man/man8
+MANDIR = $(DESTDIR)/man/man8
 
 PPPSTATSRCS = pppstats.c
 PPPSTATOBJS = pppstats.o
