--- ./chat/Makefile.linux.orig	2021-01-04 17:06:37.000000000 -0600
+++ ./chat/Makefile.linux	2021-01-25 12:56:35.570645743 -0600
@@ -4,7 +4,7 @@
 
 DESTDIR = $(INSTROOT)@DESTDIR@
 BINDIR = $(DESTDIR)/sbin
-MANDIR = $(DESTDIR)/share/man/man8
+MANDIR = $(DESTDIR)/man/man8
 
 CDEF1=	-DTERMIOS			# Use the termios structure
 CDEF2=	-DSIGTYPE=void			# Standard definition
--- ./pppdump/Makefile.linux.orig	2021-01-04 17:06:37.000000000 -0600
+++ ./pppdump/Makefile.linux	2021-01-25 12:56:35.570645743 -0600
@@ -4,7 +4,7 @@
 
 DESTDIR = $(INSTROOT)@DESTDIR@
 BINDIR = $(DESTDIR)/sbin
-MANDIR = $(DESTDIR)/share/man/man8
+MANDIR = $(DESTDIR)/man/man8
 
 CFLAGS = $(COPTS) -I../include/net
 OBJS = pppdump.o bsd-comp.o deflate.o zlib.o
--- ./pppd/Makefile.linux.orig	2021-01-04 17:06:37.000000000 -0600
+++ ./pppd/Makefile.linux	2021-01-25 12:56:35.570645743 -0600
@@ -10,7 +10,7 @@
 # Default installation locations
 DESTDIR = $(INSTROOT)@DESTDIR@
 BINDIR = $(DESTDIR)/sbin
-MANDIR = $(DESTDIR)/share/man/man8
+MANDIR = $(DESTDIR)/man/man8
 INCDIR = $(DESTDIR)/include
 
 TARGETS = pppd
@@ -74,7 +74,7 @@
 PLUGIN=y
 
 # Enable Microsoft proprietary Callback Control Protocol
-#CBCP=y
+CBCP=y
 
 # Enable EAP SRP-SHA1 authentication (requires libsrp)
 #USE_SRP=y
--- ./pppd/plugins/Makefile.linux.orig	2021-01-04 17:06:37.000000000 -0600
+++ ./pppd/plugins/Makefile.linux	2021-01-25 12:56:35.570645743 -0600
@@ -4,7 +4,7 @@
 
 DESTDIR = $(INSTROOT)@DESTDIR@
 BINDIR = $(DESTDIR)/sbin
-MANDIR = $(DESTDIR)/share/man/man8
+MANDIR = $(DESTDIR)/man/man8
 LIBDIR = $(DESTDIR)/lib/pppd/$(VERSION)
 
 CFLAGS	= $(COPTS) -I.. -I../../include -fPIC
--- ./pppd/plugins/radius/Makefile.linux.orig	2021-01-04 17:06:37.000000000 -0600
+++ ./pppd/plugins/radius/Makefile.linux	2021-01-25 12:56:35.570645743 -0600
@@ -8,7 +8,7 @@
 COPTS=@CFLAGS@
 
 DESTDIR = $(INSTROOT)@DESTDIR@
-MANDIR = $(DESTDIR)/share/man/man8
+MANDIR = $(DESTDIR)/man/man8
 LIBDIR = $(DESTDIR)/lib/pppd/$(VERSION)
 
 VERSION = $(shell awk -F '"' '/VERSION/ { print $$2; }' ../../patchlevel.h)
--- ./linux/Makefile.top.orig	2021-01-04 17:06:37.000000000 -0600
+++ ./linux/Makefile.top	2021-01-25 12:56:35.570645743 -0600
@@ -3,7 +3,7 @@
 DESTDIR = $(INSTROOT)@DESTDIR@
 BINDIR = $(DESTDIR)/sbin
 INCDIR = $(DESTDIR)/include
-MANDIR = $(DESTDIR)/share/man
+MANDIR = $(DESTDIR)/man
 ETCDIR = $(INSTROOT)@SYSCONF@/ppp
 
 # uid 0 = root
--- ./pppstats/Makefile.linux.orig	2021-01-04 17:06:37.000000000 -0600
+++ ./pppstats/Makefile.linux	2021-01-25 12:56:35.570645743 -0600
@@ -7,7 +7,7 @@
 
 DESTDIR = $(INSTROOT)@DESTDIR@
 BINDIR = $(DESTDIR)/sbin
-MANDIR = $(DESTDIR)/share/man/man8
+MANDIR = $(DESTDIR)/man/man8
 
 PPPSTATSRCS = pppstats.c
 PPPSTATOBJS = pppstats.o
