--- ./pppsetup.orig	Sun May 19 19:01:29 2002
+++ ./pppsetup	Sun May 19 19:06:07 2002
@@ -861,10 +861,12 @@
 echo "multi on" >> /etc/host.conf
 fi
 
-if [ ! -z $DOMAINNAME ]; then        
+if [ ! -z $DOMAINNAME ]; then
+# backup file
+cp -a /etc/resolv.conf /etc/resolv.conf.$$ 2> /dev/null
 echo "search $DOMAINNAME" > /etc/resolv.conf
 else
-rm -f /etc/resolv.conf 2>/dev/null 
+rm -f /etc/resolv.conf 2>/dev/null
 DOMAINNAME="unknown.com"
 fi
 
@@ -875,6 +877,8 @@
 fi
 
 if [ ! -s /etc/hosts ]; then
+# backup file
+cp -a /etc/hosts /etc/hosts.bak.$$ 2> /dev/null
 echo "127.0.0.1    localhost" > /etc/hosts
 if [ "$DOMAINNAME" = "unknown.com" ]; then
 echo "0.0.0.0     `hostname 2>/dev/null`.localnet `hostname 2>/dev/null`" >> /etc/hosts
