--- ppp-off.orig	Fri Oct 20 17:43:50 2000
+++ ppp-off	Fri Oct 20 17:46:32 2000
@@ -21,14 +21,8 @@
 # This just checks if demand dialing is running, if so it
 # says Demand Dialing Stoped.
 
-ps x 2>/dev/null > /tmp/grep.tmp
-
-if grep 2>/dev/null "options.demand" /tmp/grep.tmp >/dev/null
-then
-rm -f /tmp/grep.tmp
-D="echo Demand Dialing Stoped."
-else
-rm -f /tmp/grep.tmp
+if ps x | fgrep "options.demand" 1> /dev/null 2> /dev/null ; then
+        D="echo Demand Dialing Stoped."
 fi
 
 #
