--- ./gdk/wayland/cursor/meson.build.orig	2022-12-09 12:40:50.120829020 -0600
+++ ./gdk/wayland/cursor/meson.build	2022-12-09 12:40:50.120829020 -0600
@@ -0,0 +1,12 @@
+wayland_cursor_sources = files([
+  'wayland-cursor.c',
+  'xcursor.c',
+  'os-compatibility.c'
+])
+
+libwayland_cursor = static_library('wayland+cursor',
+  sources: wayland_cursor_sources,
+  include_directories: [ confinc, ],
+  dependencies: [ glib_dep, wlclientdep, ],
+  c_args: common_cflags,
+)
