--- ./Makefile.in.orig	2020-04-27 09:37:05.000000000 -0500
+++ ./Makefile.in	2020-04-27 13:08:49.593281753 -0500
@@ -1241,7 +1241,6 @@
 dist-hook:
 	mkdir $(distdir)/subprojects
 	cp -p $(srcdir)/subprojects/*.wrap $(distdir)/subprojects
-	$(top_srcdir)/check-version.py $(top_srcdir)/configure.ac $(top_srcdir)/meson.build
 
 distclean-local:
 	if test "$(srcdir)" = "."; then :; else \
--- ./meson.build.orig	2020-04-27 09:35:59.000000000 -0500
+++ ./meson.build	2020-04-27 13:09:29.805285185 -0500
@@ -987,16 +987,6 @@
     install_dir : join_paths(gtk_datadir, 'gtk-3.0', 'valgrind'))
 endif
 
-test(
-  'version-check',
-  find_program('check-version.py'),
-  args: [
-    join_paths(meson.current_source_dir(), 'configure.ac'),
-    join_paths(meson.current_source_dir(), 'meson.build'),
-  ],
-  suite: 'gtk',
-)
-
 summary = [
   '',
   '------',
--- ./Makefile.am.orig	2020-04-23 12:14:53.000000000 -0500
+++ ./Makefile.am	2020-04-27 13:08:28.015279911 -0500
@@ -84,7 +84,6 @@
 dist-hook:
 	mkdir $(distdir)/subprojects
 	cp -p $(srcdir)/subprojects/*.wrap $(distdir)/subprojects
-	$(top_srcdir)/check-version.py $(top_srcdir)/configure.ac $(top_srcdir)/meson.build
 
 distclean-local:
 	if test "$(srcdir)" = "."; then :; else \
