diff -Nur gtk+-3.0.12.orig//gtk/Makefile.am gtk+-3.0.12/gtk/Makefile.am
--- gtk+-3.0.12.orig//gtk/Makefile.am	2011-07-27 20:34:32.000000000 -0500
+++ gtk+-3.0.12/gtk/Makefile.am	2011-08-03 20:35:55.780035282 -0500
@@ -37,6 +37,7 @@
 	-DGTK_LIBDIR=\"$(libdir)\"			\
 	-DGTK_DATADIR=\"$(datadir)\"			\
 	-DGTK_DATA_PREFIX=\"$(prefix)\"			\
+	-DHOST=\"$(host_triplet)\"			\
 	-DGTK_SYSCONFDIR=\"$(sysconfdir)\"		\
 	-DGTK_VERSION=\"$(GTK_VERSION)\"		\
 	-DGTK_BINARY_VERSION=\"$(GTK_BINARY_VERSION)\"	\
