diff -u -r kaddressbook-20.08.3.orig/src/printing/compact/compactstyle.cpp kaddressbook-20.08.3/src/printing/compact/compactstyle.cpp
--- kaddressbook-20.08.3.orig/src/printing/compact/compactstyle.cpp	2020-10-31 12:34:25.000000000 -0500
+++ kaddressbook-20.08.3/src/printing/compact/compactstyle.cpp	2020-11-21 18:57:47.894415475 -0600
@@ -184,7 +184,7 @@
 
     // print
     QPrinter *printer = wizard()->printer();
-    printer->setPageMargins(20, 20, 20, 20, QPrinter::DevicePixel);
+    printer->setPageMargins(QMarginsF(20, 20, 20, 20), QPageLayout::Point);
 
     progress->addMessage(i18n("Setting up document"));
 
diff -u -r kaddressbook-20.08.3.orig/src/printing/grantlee/grantleeprintstyle.cpp kaddressbook-20.08.3/src/printing/grantlee/grantleeprintstyle.cpp
--- kaddressbook-20.08.3.orig/src/printing/grantlee/grantleeprintstyle.cpp	2020-10-31 12:34:25.000000000 -0500
+++ kaddressbook-20.08.3/src/printing/grantlee/grantleeprintstyle.cpp	2020-11-21 18:58:45.776416770 -0600
@@ -52,7 +52,7 @@
 void GrantleePrintStyle::print(const KContacts::Addressee::List &contacts, PrintProgress *progress)
 {
     QPrinter *printer = wizard()->printer();
-    printer->setPageMargins(20, 20, 20, 20, QPrinter::DevicePixel);
+    printer->setPageMargins(QMarginsF(20, 20, 20, 20), QPageLayout::Point);
 
     progress->addMessage(i18n("Setting up document"));
 
diff -u -r kaddressbook-20.08.3.orig/src/printing/mike/mikesstyle.cpp kaddressbook-20.08.3/src/printing/mike/mikesstyle.cpp
--- kaddressbook-20.08.3.orig/src/printing/mike/mikesstyle.cpp	2020-10-31 12:34:25.000000000 -0500
+++ kaddressbook-20.08.3/src/printing/mike/mikesstyle.cpp	2020-11-21 18:59:05.607417213 -0600
@@ -123,7 +123,7 @@
 void MikesStyle::print(const KContacts::Addressee::List &contacts, PrintProgress *progress)
 {
     QPrinter *printer = wizard()->printer();
-    printer->setPageMargins(20, 20, 20, 20, QPrinter::DevicePixel);
+    printer->setPageMargins(QMarginsF(20, 20, 20, 20), QPageLayout::Point);
 
     progress->addMessage(i18n("Setting up document"));
 
diff -u -r kaddressbook-20.08.3.orig/src/printing/ringbinder/ringbinderstyle.cpp kaddressbook-20.08.3/src/printing/ringbinder/ringbinderstyle.cpp
--- kaddressbook-20.08.3.orig/src/printing/ringbinder/ringbinderstyle.cpp	2020-10-31 12:34:25.000000000 -0500
+++ kaddressbook-20.08.3/src/printing/ringbinder/ringbinderstyle.cpp	2020-11-21 18:59:38.963417959 -0600
@@ -175,7 +175,7 @@
     config.sync();
 
     QPrinter *printer = wizard()->printer();
-    printer->setPageMargins(50, 20, 0, 50, QPrinter::DevicePixel);
+    printer->setPageMargins(QMarginsF(50, 20, 0, 50), QPageLayout::Point);
 
     progress->addMessage(i18n("Setting up document"));
 
