--- a/default-config/Makefile.am	2017-01-20 22:30:43.356970266 +0000
+++ b/default-config/Makefile.am	2017-01-20 22:49:56.060484603 +0000
@@ -17,9 +17,9 @@
 
 install-data-hook:
 	cp -r $(srcdir)/images $(inst_location)
-	ln -sf $(inst_location)/FvwmScript-DateTime $(inst_location)/..
-	ln -sf $(inst_location)/FvwmScript-ConfirmQuit $(inst_location)/..
-	ln -sf $(inst_location)/FvwmScript-ConfirmCopyConfig $(inst_location)/..
+	ln -sf default-config/FvwmScript-DateTime $(inst_location)/..
+	ln -sf default-config/FvwmScript-ConfirmQuit $(inst_location)/..
+	ln -sf default-config/FvwmScript-ConfirmCopyConfig $(inst_location)/..
 
 uninstall-hook:
 	rm -fr $(DESTDIR)/$(configdir)
