--- ./fvwm/colormaps.c.orig	Mon Sep 13 14:10:25 1999
+++ ./fvwm/colormaps.c	Mon Sep 13 14:12:03 1999
@@ -90,10 +90,13 @@
   /* Reinstall the colormap that we think should be installed,
    * UNLESS and unrecognized window has the focus - it might be
    * an override-redirect window that has its own colormap. */
-  if((ReInstall)&&(Scr.UnknownWinFocused == None))
-    {
-      XInstallColormap(dpy,last_cmap);
-    }
+  /* Commented out, since although this behavior is an X standard, it breaks
+   * xlockmore and some other nifty programs, and having it this way doesn't
+   * cause any problems that I'm aware of. */
+  /* if((ReInstall)&&(Scr.UnknownWinFocused == None)) */
+  /*  { */
+  /*    XInstallColormap(dpy,last_cmap); */
+  /*  } */
 }
 
 /************************************************************************
