--- ./etc/httpd/httpd.conf.orig	2007-05-17 23:40:15.000000000 -0500
+++ ./etc/httpd/httpd.conf	2007-05-18 15:55:38.000000000 -0500
@@ -88,7 +88,7 @@
 LoadModule proxy_http_module lib/httpd/modules/mod_proxy_http.so
 LoadModule proxy_ajp_module lib/httpd/modules/mod_proxy_ajp.so
 LoadModule proxy_balancer_module lib/httpd/modules/mod_proxy_balancer.so
-LoadModule ssl_module lib/httpd/modules/mod_ssl.so
+#LoadModule ssl_module lib/httpd/modules/mod_ssl.so
 LoadModule mime_module lib/httpd/modules/mod_mime.so
 LoadModule dav_module lib/httpd/modules/mod_dav.so
 LoadModule status_module lib/httpd/modules/mod_status.so
