--- ./os-probes/init/common/10filesystems.orig	2013-07-23 02:38:08.000000000 -0500
+++ ./os-probes/init/common/10filesystems	2013-08-30 15:21:07.382157112 -0500
@@ -2,7 +2,7 @@
 # Make sure filesystems are available.
 set +e	# ignore errors from modprobe
 
-FILESYSTEMS='ext2 ext3 ext4 xfs jfs msdos vfat ntfs minix hfs hfsplus qnx4 ufs btrfs'
+FILESYSTEMS='ext2 ext3 ext4 reiserfs xfs jfs msdos vfat ntfs minix hfs hfsplus qnx4 ufs btrfs'
 # fuse is needed to make grub-mount work.
 FILESYSTEMS="$FILESYSTEMS fuse"
 # The Ubuntu kernel udebs put a number of filesystem modules in
