--- ./buildtools/configure.pl.orig	2024-04-12 12:11:55.000000000 -0500
+++ ./buildtools/configure.pl	2024-05-10 19:05:04.538404953 -0500
@@ -1537,7 +1537,7 @@
         " -pedantic -fno-common " .
         "-Wall -Wno-uninitialized -Wmissing-declarations -Wimplicit " .
         "-Wwrite-strings -Wmissing-prototypes -Wundef " .
-        "-Wno-unknown-pragmas ";
+        "-Wno-unknown-pragmas -Wno-error=incompatible-pointer-types ";
 
     if (wnostrictoverflowWorks($gccCommandName)) {
         # The compiler generates some optimizations based on the assumption
