--- ./converter/other/pnmtotiffcmyk.c.orig	2023-12-29 13:47:21.000000000 -0600
+++ ./converter/other/pnmtotiffcmyk.c	2023-12-29 13:55:43.849898342 -0600
@@ -989,8 +989,9 @@
 
   if ( (err = parseOpts( argc, argv, rt )) ) goto exit ;
 
-  if ( (err = rt->in->open( rt->in, rt )) ) goto exit ;
-  if ( (err = rt->out->open( rt->out, rt )) ) goto exit ;
+  if ( (err = (rt->in->open)( rt->in, rt )) ) goto exit ;
+  if ( (err = (rt->out->open)( rt->out, rt )) ) goto exit ;
+
 
   while ( rt->in->hasMore( rt->in ) ) {
     if ( (err = rt->in->next( rt->in, &r, &g, &b )) ) goto exit ;
