--- xorg-cf-files-1.0.6/Imake.tmpl.orig	2021-04-15 11:06:19.978750979 -0700
+++ xorg-cf-files-1.0.6/Imake.tmpl	2021-04-15 11:07:02.421749295 -0700
@@ -1081,21 +1081,21 @@
 #if HasLargeTmp || SystemV4
 #define ArCmd ArCmdBase cq
 #else
-#define ArCmd ArCmdBase clq
+#define ArCmd ArCmdBase cq
 #endif
 #endif
 #ifndef ArAddCmd
 #if HasLargeTmp || SystemV4
 #define ArAddCmd ArCmdBase ru
 #else
-#define ArAddCmd ArCmdBase rul
+#define ArAddCmd ArCmdBase ru
 #endif
 #endif
 #ifndef ArExtCmd
 #if HasLargeTmp || SystemV4
 #define ArExtCmd ArCmdBase x
 #else
-#define ArExtCmd ArCmdBase xl
+#define ArExtCmd ArCmdBase x
 #endif
 #endif
 #ifndef BootstrapCFlags
