--- ./include/Makefile.in.orig	2014-11-12 05:59:58.000000000 -0600
+++ ./include/Makefile.in	2015-03-18 20:16:10.909912988 -0500
@@ -314,7 +314,7 @@
 AUTOMAKE_OPTIONS = foreign
 DISTCLEANFILES = ffitarget.h
 EXTRA_DIST = ffi.h.in ffi_common.h
-includesdir = $(libdir)/@PACKAGE_NAME@-@PACKAGE_VERSION@/include
+includesdir = $(includedir)
 nodist_includes_HEADERS = ffi.h ffitarget.h
 all: all-am
 
--- ./include/Makefile.am.orig	2014-11-08 06:47:24.000000000 -0600
+++ ./include/Makefile.am	2015-03-18 20:14:30.883905771 -0500
@@ -5,5 +5,5 @@
 DISTCLEANFILES=ffitarget.h
 EXTRA_DIST=ffi.h.in ffi_common.h
 
-includesdir = $(libdir)/@PACKAGE_NAME@-@PACKAGE_VERSION@/include
+includesdir = $(includedir)
 nodist_includes_HEADERS = ffi.h ffitarget.h
--- ./libffi.pc.in.orig	2014-11-08 06:47:24.000000000 -0600
+++ ./libffi.pc.in	2015-03-18 20:14:30.884905771 -0500
@@ -2,10 +2,10 @@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 toolexeclibdir=@toolexeclibdir@
-includedir=${libdir}/@PACKAGE_NAME@-@PACKAGE_VERSION@/include
+includedir=@includedir@
 
 Name: @PACKAGE_NAME@
 Description: Library supporting Foreign Function Interfaces
 Version: @PACKAGE_VERSION@
-Libs: -L${toolexeclibdir} -lffi
-Cflags: -I${includedir}
+Libs: -lffi
+Cflags:
