--- ./modules/Makefile.am.orig	2019-07-16 13:18:28.619322386 -0500
+++ ./modules/Makefile.am	2019-07-16 13:45:49.260371056 -0500
@@ -10,7 +10,7 @@
 	pam_mkhomedir pam_motd pam_namespace pam_nologin \
 	pam_permit pam_pwhistory pam_rhosts pam_rootok pam_securetty \
 	pam_selinux pam_sepermit pam_shells pam_stress \
-	pam_succeed_if pam_time pam_timestamp \
+	pam_succeed_if pam_tally2 pam_time pam_timestamp \
 	pam_tty_audit pam_umask \
 	pam_unix pam_userdb pam_warn pam_wheel pam_xauth
 
