--- ./boost/function/function_template.hpp.orig	2008-10-16 08:21:50.000000000 -0500
+++ ./boost/function/function_template.hpp	2009-08-24 20:45:28.000000000 -0500
@@ -950,10 +950,10 @@
           f.vtable->manager(f.functor, this->functor,
                             boost::detail::function::move_functor_tag);
 		  f.vtable = 0;
-#if !defined(BOOST_NO_EXCEPTIONS)      
         } else {
           clear();
         }
+#if !defined(BOOST_NO_EXCEPTIONS)      
       } catch (...) {
         vtable = 0;
         throw;
