--- ./core/cmake/macros/MacroSSE.cmake.orig	2023-04-12 11:24:31.000000000 -0500
+++ ./core/cmake/macros/MacroSSE.cmake	2023-04-15 14:08:38.586047012 -0500
@@ -90,28 +90,12 @@
         string(REGEX REPLACE "^.*(sse4_1).*$" "\\1" SSE_THERE ${CPUINFO})
         string(COMPARE EQUAL "sse4_1" "${SSE_THERE}" SSE41_TRUE)
 
-        if(SSE41_TRUE)
-
-            set(SSE4_1_FOUND true CACHE BOOL "SSE4.1 available on host")
-
-        else()
-
-            set(SSE4_1_FOUND false CACHE BOOL "SSE4.1 available on host")
-
-        endif()
+        set(SSE4_1_FOUND false CACHE BOOL "SSE4.1 available on host")
 
         string(REGEX REPLACE "^.*(sse4_2).*$" "\\1" SSE_THERE ${CPUINFO})
         string(COMPARE EQUAL "sse4_2" "${SSE_THERE}" SSE42_TRUE)
 
-        if(SSE42_TRUE)
-
-            set(SSE4_2_FOUND true CACHE BOOL "SSE4.2 available on host")
-
-        else()
-
-            set(SSE4_2_FOUND false CACHE BOOL "SSE4.2 available on host")
-
-        endif()
+        set(SSE4_2_FOUND false CACHE BOOL "SSE4.2 available on host")
 
     elseif(CMAKE_SYSTEM_NAME MATCHES "Darwin")
 
