--- libpurple/plugins/ssl/ssl-gnutls.c.old	2008-10-17 18:58:31.000000000 +0200
+++ libpurple/plugins/ssl/ssl-gnutls.c	2008-10-17 18:53:03.000000000 +0200
@@ -693,7 +693,8 @@
 					(Verisign and possibly others have
 					root certificates that predate the
 					current standard) */
-				     GNUTLS_VERIFY_ALLOW_X509_V1_CA_CRT,
+				     GNUTLS_VERIFY_ALLOW_X509_V1_CA_CRT 
+				     | GNUTLS_VERIFY_ALLOW_SIGN_RSA_MD5,
 				     &verify);
 
 	if (ret != 0) {
