--- ./scripts/Makefile.extrawarn.orig	2025-05-02 00:59:33.000000000 -0500
+++ ./scripts/Makefile.extrawarn	2025-05-02 15:00:59.507092623 -0500
@@ -82,8 +82,8 @@
 # Warn if there is an enum types mismatch
 KBUILD_CFLAGS += $(call cc-option,-Wenum-conversion)
 
-# Explicitly clear padding bits during variable initialization
-KBUILD_CFLAGS += $(call cc-option,-fzero-init-padding-bits=all)
+## Explicitly clear padding bits during variable initialization
+#KBUILD_CFLAGS += $(call cc-option,-fzero-init-padding-bits=all)
 
 KBUILD_CFLAGS += -Wextra
 KBUILD_CFLAGS += -Wunused
