diff -Nur file-5.00.orig/src/readelf.h file-5.00/src/readelf.h
--- file-5.00.orig/src/readelf.h	2008-05-28 15:58:16.000000000 -0500
+++ file-5.00/src/readelf.h	2009-02-08 22:17:26.681636542 -0600
@@ -233,7 +233,7 @@
 #define NT_OPENBSD_VERSION	1
 #define NT_DRAGONFLY_VERSION	1
 /* GNU executables (name = "GNU") */
-#define NT_GNU_VERSION		1
+#define NT_GNU_VERSION		0
 
 /* GNU OS tags */
 #define GNU_OS_LINUX	0
