# Remove non-existant struct member:

--- ./modules/berkeley-db/bdb.c.orig	2018-02-04 13:38:36.000000000 -0600
+++ ./modules/berkeley-db/bdb.c	2018-02-04 15:29:06.147941837 -0600
@@ -2216,7 +2216,6 @@
   c_data.compact_timeout = timeout;
   c_data.compact_pages = pages;
   SYSCALL(db->compact,(db,txn,pstart,pstop,&c_data,flags,&end));
-  pushSTACK(uint32_to_I(c_data.compact_empty_buckets));
   pushSTACK(uint32_to_I(c_data.compact_pages_free));
   pushSTACK(uint32_to_I(c_data.compact_pages_examine));
   pushSTACK(uint32_to_I(c_data.compact_levels));
