diff -u -r --new-file clisp-2.50_20191103_c26de7873.orig/src/makemake.in clisp-2.50_20191103_c26de7873/src/makemake.in
--- clisp-2.50_20191103_c26de7873.orig/src/makemake.in	2019-12-30 13:50:09.000000000 -0600
+++ clisp-2.50_20191103_c26de7873/src/makemake.in	2020-04-24 22:40:43.272361884 -0500
@@ -1995,7 +1995,7 @@
 
 GLLIB_A=libgnu.a
 # libgnu.a must come _before_ -lws2_32 -lintl &c
-LIBS="${GLLIB_A} ${LIBS} ${LIBSOCKET}"
+LIBS="-Wl,--whole-archive ${GLLIB_A} -Wl,--no-whole-archive ${LIBS} -ldl ${LIBSOCKET}"
 FILES="${GLLIB_A} "${FILES}
 
 FILES='lisp.a '$FILES
