--- ./doc/example-config/README.in.orig	2017-06-23 06:18:28.000000000 -0500
+++ ./doc/example-config/README.in	2017-11-14 19:04:46.621109623 -0600
@@ -1,2 +1,15 @@
-Configuration files go to this directory. See example configuration files in
-@exampledir@/
+Configuration files go to this directory.
+
+The configuration files that are provided here should work out of the box on
+Slackware once you've installed security certificates to support SSL/TLS.
+
+See the config file beneath this directory:  conf.d/10-ssl.conf
+
+For a script that will install self-signed certificates, see mkcert.sh in
+/usr/doc/dovecot-2.*
+
+It is also recommended to edit conf.d/10-auth.conf and uncomment this line:
+#disable_plaintext_auth = yes
+
+This will prevent usernames and passwords from being sent until a secure
+connection has been established.
