--- ./doc/example-config/conf.d/auth-system.conf.ext.orig	2017-12-22 07:53:36.000000000 -0600
+++ ./doc/example-config/conf.d/auth-system.conf.ext	2018-01-31 17:05:06.840878097 -0600
@@ -7,12 +7,12 @@
 # PAM is typically used with either userdb passwd or userdb static.
 # REMEMBER: You'll need /etc/pam.d/dovecot file created for PAM
 # authentication to actually work. <doc/wiki/PasswordDatabase.PAM.txt>
-passdb {
-  driver = pam
+#passdb {
+  #driver = pam
   # [session=yes] [setcred=yes] [failure_show_msg=yes] [max_requests=<n>]
   # [cache_key=<key>] [<service name>]
   #args = dovecot
-}
+#}
 
 # System users (NSS, /etc/passwd, or similar).
 # In many systems nowadays this uses Name Service Switch, which is
@@ -24,13 +24,17 @@
 #}
 
 # Shadow passwords for system users (NSS, /etc/shadow or similar).
-# Deprecated by PAM nowadays.
+# This is the default on Slackware systems.
 # <doc/wiki/PasswordDatabase.Shadow.txt>
-#passdb {
-  #driver = shadow
+passdb {
+  driver = shadow
   # [blocking=no]
   #args = 
-#}
+}
+# Auth worker to authenticate shadow passwords on Slackware:
+service auth-worker {
+  group = shadow
+}
 
 # PAM-like authentication for OpenBSD.
 # <doc/wiki/PasswordDatabase.BSDAuth.txt>
