--- ./write/write.c.orig	Sun Feb 25 18:22:21 2001
+++ ./write/write.c	Sun Feb 25 18:22:35 2001
@@ -73,9 +73,7 @@
 #include <fcntl.h>
 #include <signal.h>
 #include <utmp.h>
-#if 0
-	#include <time.h>
-#endif
+#include <time.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
--- ./writed/writed.c.orig	Sun Feb 25 18:23:17 2001
+++ ./writed/writed.c	Sun Feb 25 18:24:08 2001
@@ -67,6 +67,7 @@
 #include <stdlib.h>
 #include <errno.h>
 #include <unistd.h>
+#include <sys/socket.h>
 
 #ifdef ultrix  /* missing protos */
 struct sockaddr;
