--- ./builds/freetype.mk.orig	2018-04-22 04:41:36.000000000 -0500
+++ ./builds/freetype.mk	2018-09-13 14:34:39.641002962 -0500
@@ -248,20 +248,24 @@
 	$(FT_COMPILE) $T$(subst /,$(COMPILER_SEP),$@ $<)
 
 
-# ftver component
+# COMMENTED OUT:
+# Otherwise having windres on the system causes a Windows-specific resource
+# file to be compiled, breaking the build.
 #
-#  The VERSIONINFO resource `ftver.rc' contains version and copyright
-#  to be compiled by windres and tagged into DLL usually.
+## ftver component
+##
+##  The VERSIONINFO resource `ftver.rc' contains version and copyright
+##  to be compiled by windres and tagged into DLL usually.
+##
+#ifneq ($(RC),)
+#  FTVER_SRC := $(BASE_DIR)/ftver.rc
+#  FTVER_OBJ := $(OBJ_DIR)/ftver.$O
 #
-ifneq ($(RC),)
-  FTVER_SRC := $(BASE_DIR)/ftver.rc
-  FTVER_OBJ := $(OBJ_DIR)/ftver.$O
-
-  OBJECTS_LIST += $(FTVER_OBJ)
-
-  $(FTVER_OBJ): $(FTVER_SRC)
-	$(RC) -o $@ $<
-endif
+#  OBJECTS_LIST += $(FTVER_OBJ)
+#
+#  $(FTVER_OBJ): $(FTVER_SRC)
+#	$(RC) -o $@ $<
+#endif
 
 
 # All FreeType library objects.
