--- ./src/wc.c.orig	2023-04-10 05:14:08.000000000 -0500
+++ ./src/wc.c	2023-08-26 14:22:10.058092456 -0500
@@ -174,14 +174,7 @@
 
   if (avx_enabled)
     {
-      eax = ebx = ecx = edx = 0;
-      if (! __get_cpuid_count (7, 0, &eax, &ebx, &ecx, &edx))
-        getcpuid_ok = false;
-      else
-        {
-          if (! (ebx & bit_AVX2))
-            avx_enabled = false;  /* Hardware doesn't support it.  */
-        }
+        avx_enabled = false;  /* Never use AVX2. */
     }
 
 
