diff -aurN xf86-video-tseng-1.2.5/src/tseng_driver.c xf86-video-tseng-1.2.5-mod/src/tseng_driver.c
--- xf86-video-tseng-1.2.5/src/tseng_driver.c	2012-07-17 07:21:17.000000000 +0200
+++ xf86-video-tseng-1.2.5-mod/src/tseng_driver.c	2018-07-14 13:48:02.634774719 +0200
@@ -1040,10 +1040,6 @@
     else
 	pTseng->Bytesperpixel = 1;  /* this is fake for < 8bpp, but simplifies other code */
 
-    /* hardware limits */
-    pScrn->maxHValue = Tseng_HMAX;
-    pScrn->maxVValue = Tseng_VMAX;
-
     /*
      * This must happen after pScrn->display has been set because
      * xf86SetWeight references it.
@@ -1129,15 +1125,14 @@
     /*
      * xf86ValidateModes will check that the mode HTotal and VTotal values
      * don't exceed the chipset's limit if pScrn->maxHValue and
-     * pScrn->maxVValue are set.  Since our TsengValidMode() already takes
      * care of this, we don't worry about setting them here.
      */
 
     /* Select valid modes from those available */
     i = xf86ValidateModes(pScrn, pScrn->monitor->Modes,
 	pScrn->display->modes, &pTseng->clockRange,
-	NULL, 32, pScrn->maxHValue, 8*pTseng->Bytesperpixel, /* H limits */
-	0, pScrn->maxVValue,	       /* V limits */
+	NULL, 32, Tseng_HMAX, 8*pTseng->Bytesperpixel, /* H limits */
+	0, Tseng_VMAX,	       /* V limits */
 	pScrn->display->virtualX,
 	pScrn->display->virtualY,
 	pTseng->FbMapSize,
