--- ./libutil++/bfd_spu_support.cpp.orig	2011-07-04 21:25:04.000000000 -0500
+++ ./libutil++/bfd_spu_support.cpp	2012-12-20 18:42:01.400977322 -0600
@@ -9,6 +9,8 @@
  * (C) Copyright IBM Corporation 2007
  */
 
+#include "config.h"
+
 #include "bfd_support.h"
 #include "op_bfd.h"
 #include "config.h"
--- ./libutil++/bfd_support.cpp.orig	2011-07-04 21:25:04.000000000 -0500
+++ ./libutil++/bfd_support.cpp	2012-12-20 18:41:31.160978289 -0600
@@ -8,6 +8,8 @@
  * @author John Levon
  */
 
+#include "config.h"
+
 #include "bfd_support.h"
 
 #include "op_bfd.h"
