diff -u -r --new-file oprofile-0.9.7.orig/libopagent/opagent.c oprofile-0.9.7/libopagent/opagent.c
--- oprofile-0.9.7.orig/libopagent/opagent.c	2011-07-04 21:25:04.000000000 -0500
+++ oprofile-0.9.7/libopagent/opagent.c	2012-06-20 10:59:17.851896225 -0500
@@ -60,6 +60,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <time.h>
+#include "config.h"
 #include <bfd.h>
 
 #include "opagent.h"
diff -u -r --new-file oprofile-0.9.7.orig/libpp/symbol.h oprofile-0.9.7/libpp/symbol.h
--- oprofile-0.9.7.orig/libpp/symbol.h	2011-07-04 21:25:04.000000000 -0500
+++ oprofile-0.9.7/libpp/symbol.h	2012-06-20 10:59:23.579895917 -0500
@@ -18,6 +18,7 @@
 #include "format_flags.h"
 #include "op_types.h"
 
+#include "config.h"
 #include <bfd.h>
 #include <stdint.h>
 
diff -u -r --new-file oprofile-0.9.7.orig/opjitconv/debug_line.c oprofile-0.9.7/opjitconv/debug_line.c
--- oprofile-0.9.7.orig/opjitconv/debug_line.c	2011-07-04 21:25:04.000000000 -0500
+++ oprofile-0.9.7/opjitconv/debug_line.c	2012-06-20 10:59:40.620895008 -0500
@@ -13,6 +13,7 @@
 #include <string.h>
 #include <stddef.h>
 #include <stdio.h>
+#include "config.h"
 #include <bfd.h>
 #include <limits.h>
 
diff -u -r --new-file oprofile-0.9.7.orig/opjitconv/opjitconv.h oprofile-0.9.7/opjitconv/opjitconv.h
--- oprofile-0.9.7.orig/opjitconv/opjitconv.h	2011-07-04 21:25:04.000000000 -0500
+++ oprofile-0.9.7/opjitconv/opjitconv.h	2012-06-20 10:59:35.387895285 -0500
@@ -26,6 +26,7 @@
 #define OP_JIT_CONV_ALREADY_DONE 5
 #define OP_JIT_CONV_TMPDIR_NOT_REMOVED 6
 
+#include "config.h"
 #include <bfd.h>
 #include <stddef.h>
 #include <sys/stat.h>
