diff -up doxygen-1.5.8/configure.qt4 doxygen-1.5.8/configure
--- doxygen-1.5.8/configure.qt4	2008-12-26 20:22:39.000000000 +0100
+++ doxygen-1.5.8/configure	2009-02-06 11:00:44.000000000 +0100
@@ -268,7 +268,7 @@ if test "$f_wizard" = YES; then
   if test -z "$QTDIR"; then
     echo "  QTDIR environment variable not set!"
     echo -n "  Checking for Qt..."
-    for d in /usr/{lib,share,qt}/{qt-4,qt4,qt,qt*,4}; do
+    for d in /usr/{lib64,lib,share,qt}/{qt-4,qt4,qt,qt*,4}; do
       if test -x "$d/bin/qmake"; then
         QTDIR=$d
       fi
