--- ./misc.h.orig	2005-08-15 14:50:05.000000000 -0500
+++ ./misc.h	2007-05-10 16:46:23.000000000 -0500
@@ -24,7 +24,7 @@
 long random(void);
 #endif
 
-#if !defined(__VMS) & !defined(linux) & !defined(__EMX__) & !defined(__FreeBSD__) & !defined(__CYGWIN__)
+#if !defined(__VMS) & !defined(linux) & !defined(__EMX__) & !defined(__FreeBSD__) & !defined(__CYGWIN__) & !defined(__GLIBC__)
 #if defined(BSD4_4) || defined(HPArchitecture) || defined(SGIArchitecture) || defined(_AIX) || defined(_SCO_DS)
 void srandom(unsigned int);
 #else
--- ./Local.config.orig	2005-04-30 16:33:14.000000000 -0500
+++ ./Local.config	2007-05-10 16:47:02.000000000 -0500
@@ -132,11 +132,11 @@
 XCOMM  You must have the XPM library.
 XCOMM  Make sure the path name to the library is correct.
 
-XPM_LIB     = -L/usr/X11R6/lib -lXpm
+XPM_LIB     = -lXpm
 
 XCOMM  Make sure this points to the location of the 'xpm.h' file.
 
-XPM_INCLUDE = -I/usr/X11R6/include/X11
+XPM_INCLUDE = -I/usr/include/X11
 
 
 DEPENDFLAGS = -I./bitmaps
--- ./configure.orig	2005-06-03 07:26:11.000000000 -0500
+++ ./configure	2007-05-10 16:46:23.000000000 -0500
@@ -7,7 +7,7 @@
    echo "(looking for Xaw, Xaw3d, Xaw95, neXtaw)"
    echo ""
 else
-   WIDGETS=`echo $1 | tr [a:z] [A:Z]`
+   WIDGETS=`echo $1`
 fi
 
 if test "$WIDGETS" = "" ; then
@@ -54,7 +54,7 @@
    fi
 fi
 
-if test "$WIDGETS" = "XAW3DG" ; then
+if test "$WIDGETS" = "xaw3dg" ; then
    ln -sf /usr/include/X11/Xaw3d xaw_incdir
    echo "XAWLIB_DEFINES = -DXAW3D -DXAW3DG"   
    echo "XAWLIB_DEFINES = -DXAW3D -DXAW3DG" > Local.xawdefs
