--- ./modules/fdlibm/src/math_private.h.orig	2024-03-19 18:53:52.828031985 -0500
+++ ./modules/fdlibm/src/math_private.h	2024-03-19 18:55:57.193034767 -0500
@@ -30,8 +30,8 @@
  * Adapted from https://github.com/freebsd/freebsd-src/search?q=__double_t
  */
 
-typedef double      __double_t;
-typedef __double_t  double_t;
+typedef long double      __double_t;
+typedef long double  double_t;
 typedef float       __float_t;
 
 /*
