--- ./util/grub-mkconfig_lib.in.orig	2023-12-05 08:46:53.000000000 -0600
+++ ./util/grub-mkconfig_lib.in	2024-01-02 13:44:41.164913702 -0600
@@ -191,6 +191,7 @@
     case "$1" in
       *.dpkg-*) return 1 ;; # debian dpkg
       *.rpmsave|*.rpmnew) return 1 ;;
+      *.new|*.orig) return 1 ;; # slackware incoming or backed up config file
       README*|*/README*)  return 1 ;; # documentation
       *.sig) return 1 ;; # signatures
     esac
