--- ./util/grub-mkconfig.in.orig	2021-03-10 07:42:22.000000000 -0600
+++ ./util/grub-mkconfig.in	2021-09-10 13:33:17.602163090 -0500
@@ -278,6 +278,8 @@
     *~) ;;
     # emacsen autosave files. FIXME: support other editors
     */\#*\#) ;;
+    # Slackware incoming config files should be skipped as well:
+    *.new) ;;
     *)
       if grub_file_is_not_garbage "$i" && test -x "$i" ; then
         echo
