--- ./util/grub.d/10_linux.in.orig	2022-12-07 07:47:11.000000000 -0600
+++ ./util/grub.d/10_linux.in	2024-07-24 14:14:09.362582694 -0500
@@ -30,6 +30,9 @@
 
 if [ "x${GRUB_DISTRIBUTOR}" = "x" ] ; then
   OS=GNU/Linux
+elif echo ${GRUB_DISTRIBUTOR} | grep -q Slackware ; then # opting out, thanks
+  OS="${GRUB_DISTRIBUTOR}"
+  CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr 'A-Z' 'a-z' | cut -d' ' -f1|LC_ALL=C sed 's,[^[:alnum:]_],_,g') ${CLASS}"
 else
   OS="${GRUB_DISTRIBUTOR} GNU/Linux"
   CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr 'A-Z' 'a-z' | cut -d' ' -f1|LC_ALL=C sed 's,[^[:alnum:]_],_,g') ${CLASS}"
