--- ./util/grub.d/10_linux.in.orig	2024-07-25 00:18:12.270876545 -0500
+++ ./util/grub.d/10_linux.in	2024-07-25 00:21:10.292942662 -0500
@@ -209,6 +209,12 @@
 
 is_top_level=true
 for linux in ${reverse_sorted_list}; do
+
+  # Skip symbolic links to kernels if requested:
+  if [ -L "$linux" -a "$GRUB_DISABLE_KERNEL_SYMLINKS" = "true" ]; then
+      continue
+  fi
+
   gettext_printf "Found linux image: %s\n" "$linux" >&2
   basename=`basename $linux`
   dirname=`dirname $linux`
