--- ./util/grub.d/00_header.in.orig	2012-05-03 18:04:39.000000000 -0500
+++ ./util/grub.d/00_header.in	2013-09-17 01:43:37.870841064 -0500
@@ -143,7 +143,7 @@
 EOF
     else
 	for dir in "${pkgdatadir}" "`echo '/@bootdirname@/@grubdirname@' | sed "s,//*,/,g"`" /usr/share/grub ; do
-	    for basename in unicode unifont ascii; do
+	    for basename in unicode unifont dejavusansmono ascii; do
 		path="${dir}/${basename}.pf2"
 		if is_path_readable_by_grub "${path}" > /dev/null ; then
 		    font_path="${path}"
@@ -156,7 +156,7 @@
 	if [ -n "${font_path}" ] ; then
     cat << EOF
 if [ x\$feature_default_font_path = xy ] ; then
-   font=unicode
+   font=dejavusansmono
 else
 EOF
                 # Make the font accessible
@@ -169,7 +169,7 @@
 EOF
 	    else
     cat << EOF
-if loadfont unicode ; then
+if loadfont dejavusansmono ; then
 EOF
 	    fi
 	fi
diff -Naur grub-2.02~beta2.orig/util/grub-install.c grub-2.02~beta2/util/grub-install.c
--- grub-2.02~beta2.orig/util/grub-install.c	2013-12-24 08:40:31.000000000 -0800
+++ grub-2.02~beta2/util/grub-install.c.new	2015-01-27 22:07:56.714000000 -0800
@@ -827,7 +827,7 @@
   grub_util_host_init (&argc, &argv);
   product_version = xstrdup (PACKAGE_VERSION);
   pkgdatadir = grub_util_get_pkgdatadir ();
-  label_font = grub_util_path_concat (2, pkgdatadir, "unicode.pf2");
+  label_font = grub_util_path_concat (2, pkgdatadir, "dejavusansmono.pf2");
 
   argp_parse (&argp, argc, argv, 0, 0, 0);
 
