--- ./mkinitrd.orig	2022-01-26 14:33:29.000000000 -0600
+++ ./mkinitrd	2022-04-26 13:55:17.511004239 -0500
@@ -778,6 +778,9 @@
           echo "$LINE" >> $SOURCE_TREE/load_kernel_modules
         fi
 
+        # Ensure that the destination directory for the module exists:
+        mkdir -p "$(dirname $SOURCE_TREE/$SRCMOD)"
+
         # Try to add the module to the initrd-tree.  This should be done
         # even if it exists there already as we may have changed compilers
         # or otherwise caused the modules in the initrd-tree to need
