--- ./mkinitrd_command_generator.sh.orig	2023-02-28 15:19:08.085991698 -0600
+++ ./mkinitrd_command_generator.sh	2023-02-28 15:22:03.264982008 -0600
@@ -308,11 +308,15 @@
       USING_RAID=1
       break
     fi
-  done
-  # Additional check in case $BASEDEV is a partition of /dev/$MD:
-  for BLK in $(sfdisk -ld /dev/$MD 2> /dev/null | grep "^/dev" | cut -d " " -f 1); do
-    if [ "$BASEDEV" = "$BLK" ]; then
-      USING_RAID=1
+    # Additional check in case $BASEDEV is a partition of /dev/$MD:
+    for BLK in $(sfdisk -ld /dev/$MD 2> /dev/null | grep "^/dev" | cut -d " " -f 1); do
+      if [ "$BASEDEV" = "$BLK" ]; then
+        USING_RAID=1
+        break
+      fi
+    done
+    # Break out of outer loop if needed.
+    if [ "$USING_RAID" = "1" ]; then
       break
     fi
   done
