diff -up ksh-20120801/src/cmd/ksh93/sh/init.c.nomulti ksh-20120801/src/cmd/ksh93/sh/init.c
--- ksh-20120801/src/cmd/ksh93/sh/init.c.nomulti	2013-10-08 20:46:46.202471042 +0200
+++ ksh-20120801/src/cmd/ksh93/sh/init.c	2013-10-08 20:46:57.912331483 +0200
@@ -1446,7 +1446,6 @@ Shell_t *sh_init(register int argc,regis
 #endif /* SHOPT_TIMEOUT */
 	/* initialize jobs table */
 	job_clear();
-	sh_onoption(SH_MULTILINE);
 	if(argc>0)
 	{
 		/* check for restricted shell */
