diff -up ksh-20120801/src/cmd/ksh93/sh/xec.c.nohupfork ksh-20120801/src/cmd/ksh93/sh/xec.c
--- ksh-20120801/src/cmd/ksh93/sh/xec.c.nohupfork	2015-08-27 14:25:38.925378019 +0200
+++ ksh-20120801/src/cmd/ksh93/sh/xec.c	2015-08-27 14:30:42.267058087 +0200
@@ -2067,6 +2067,7 @@ int sh_exec(register const Shnode_t *t,
 				{
 					sh_exec(t->par.partre,flags);
 					shp->st.trapcom[0]=0;
+					sh_offoption(SH_INTERACTIVE);
 					sh_done(shp,0);
 				}
 			}
