diff -up ksh-20120620/src/cmd/ksh93/data/builtins.c.builtins ksh-20120620/src/cmd/ksh93/data/builtins.c
--- ksh-20120620/src/cmd/ksh93/data/builtins.c.builtins	2012-06-19 10:02:12.000000000 +0200
+++ ksh-20120620/src/cmd/ksh93/data/builtins.c	2012-06-22 12:35:05.587717588 +0200
@@ -131,20 +131,28 @@ const struct shtable3 shtab_builtins[] =
 #undef	mktemp		/* undo possible map-libc mktemp => _ast_mktemp */
 #include SHOPT_CMDLIB_HDR
 #else
+#if 1
 	CMDLIST(basename)
 	CMDLIST(chmod)
 	CMDLIST(dirname)
 	CMDLIST(getconf)
 	CMDLIST(head)
+#if 0
+does not work when ACLs are used
 	CMDLIST(mkdir)
+#endif
 	CMDLIST(logname)
+#if 1
+//does not work in chrooted environments, because /dev/fd/? is missing
 	CMDLIST(cat)
+#endif
 	CMDLIST(cmp)
 	CMDLIST(cut)
 	CMDLIST(uname)
 	CMDLIST(wc)
 	CMDLIST(sync)
 #endif
+#endif
 #if SHOPT_REGRESS
 	"__regress__",		NV_BLTIN|BLT_ENV,	bltin(__regress__),
 #endif
