diff -up ksh-20120801/src/cmd/ksh93/bltins/cd_pwd.c.cdfix4 ksh-20120801/src/cmd/ksh93/bltins/cd_pwd.c
--- ksh-20120801/src/cmd/ksh93/bltins/cd_pwd.c.cdfix4	2014-08-26 15:24:57.276953822 +0200
+++ ksh-20120801/src/cmd/ksh93/bltins/cd_pwd.c	2014-08-26 15:25:34.738770361 +0200
@@ -143,9 +143,9 @@ int sh_diropenat(Shell_t *shp, int dir,
 	}
 
 	/* Move fd to a number > 10 and *register* the fd number with the shell */
-	shfd = sh_fcntl(fd, F_dupfd_cloexec, 10);
+	shfd = fcntl(fd, F_dupfd_cloexec, 10);
 	savederrno=errno;
-	sh_close(fd);
+	close(fd);
 	errno=savederrno;
 	return(shfd);
 }
