diff -up ksh-20120801/src/cmd/ksh93/sh/expand.c.fikspand ksh-20120801/src/cmd/ksh93/sh/expand.c
--- ksh-20120801/src/cmd/ksh93/sh/expand.c.fikspand	2010-11-24 05:46:30.000000000 +0100
+++ ksh-20120801/src/cmd/ksh93/sh/expand.c	2014-05-22 12:55:46.252717371 +0200
@@ -278,6 +278,8 @@ int path_generate(Shell_t *shp,struct ar
 	char comma, range=0;
 	int first, last, incr, count = 0;
 	char tmp[32], end[1];
+	if(!sh_isoption(SH_BRACEEXPAND))
+		return path_expand(shp,todo->argval,arghead);
 	todo->argchn.ap = 0;
 again:
 	apin = ap = todo;
