--- ./factor/factor.c.orig	2004-02-08 07:20:03.000000000 -0600
+++ ./factor/factor.c	2019-02-17 15:38:29.289767009 -0600
@@ -96,8 +96,8 @@
  * We are able to sieve 2^32-1 because this byte table yields all primes
  * up to 65537 and 65537^2 > 2^32-1.
  */
-extern const ubig prime[];
-extern const ubig *pr_limit;		/* largest prime in the prime array */
+extern const uint64_t prime[];
+extern const uint64_t *const pr_limit;		/* largest prime in the prime array */
 
 #define	PRIME_CHECKS	5
 
@@ -202,7 +202,7 @@
 void
 pr_fact(BIGNUM *val)
 {
-	const ubig *fact;		/* The factor found. */
+	const uint64_t *fact;		/* The factor found. */
 
 	/* Firewall - catch 0 and 1. */
 	if (BN_is_zero(val))	/* Historical practice; 0 just exits. */
