--- ./configure.orig	2004-02-08 19:37:31.000000000 -0600
+++ ./configure	2019-02-17 16:07:03.818785361 -0600
@@ -521,7 +521,7 @@
 game_ask dm "File to disable games playing" dm_nogamesfile /etc/nogames
 game_ask dm "Log file for dm" dm_logfile "$varlibdir/games.log"
 game_ask fish "File for fish instructions" fish_instrfile "$sharedir/fish.instr"
-game_ask fortune "Directory for fortune files" fortune_dir "$sharedir/fortune"
+game_ask fortune "Directory for fortune files" fortune_dir "$sharedir/fortunes"
 game_ask hack "Directory for hack variable data" hack_dir "$varlibdir/hack"
 
 if building_in hack; then
