--- ./vsf_findlibs.sh.orig	2012-03-27 21:17:41.000000000 -0500
+++ ./vsf_findlibs.sh	2018-05-07 16:10:58.744003755 -0500
@@ -68,10 +68,10 @@
 # Solaris sendfile
 locate_library /usr/lib/libsendfile.so && echo "-lsendfile";
 
-# OpenSSL
-if find_func SSL_library_init ssl.o; then
+# Always link with OpenSSL:
+#if find_func SSL_library_init ssl.o; then
   echo "-lssl -lcrypto";
-fi
+#fi
 
 exit 0;
 
