--- ./uninstall.py.orig	2017-11-23 06:10:34.000000000 -0600
+++ ./uninstall.py	2017-11-27 18:41:11.976513695 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2011-2015 HP Development Company, L.P.
--- ./logcapture.py.orig	2017-11-23 06:10:34.000000000 -0600
+++ ./logcapture.py	2017-11-27 18:38:34.880515365 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
--- ./doctor.py.orig	2017-11-23 06:10:34.000000000 -0600
+++ ./doctor.py	2017-11-27 18:39:14.193514947 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2012-2020 HP Development Company, L.P.
--- ./upgrade.py.orig	2017-11-23 06:10:34.000000000 -0600
+++ ./upgrade.py	2017-11-27 18:36:56.502516410 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2011-2015 HP Development Company, L.P.
