--- ./Makefile.am.orig	2024-05-10 18:14:51.431337550 -0500
+++ ./Makefile.am	2024-05-10 18:15:37.078338571 -0500
@@ -306,13 +306,6 @@
 pcarddir = $(hplipdir)/pcard
 dist_pcard_DATA = pcard/__init__.py pcard/photocard.py
 
-# pcardext
-pcardextdir = $(pyexecdir)
-pcardext_LTLIBRARIES = pcardext.la
-pcardext_la_LDFLAGS = -module -avoid-version
-pcardext_la_SOURCES = pcard/pcardext/pcardext.c pcard/fat.c pcard/pcardext/pcardext.h
-pcardext_la_CFLAGS = -I$(PYTHONINCLUDEDIR)
-
 # prnt
 prntdir = $(hplipdir)/prnt
 dist_prnt_DATA = prnt/cups.py prnt/__init__.py prnt/ldl.py prnt/pcl.py prnt/colorcal.py
