--- ./upgrade.py.orig	2013-03-08 01:15:32.000000000 -0600
+++ ./upgrade.py	2013-03-19 23:41:06.360870795 -0500
@@ -214,6 +214,13 @@
 try:
     change_spinner_state(False)
     core =  CoreInstall(MODE_CHECK)
+
+    # To reenable upgrade in Slackware (although it probably won't work),
+    # delete the following 3 lines:
+    log.info("HPLIP upgrade function is disabled in Slackware.")
+    log.info("Not attempting to download upgrades.")
+    clean_exit(0)
+
     if not utils.check_network_connection():
         log.error("Either Internet is not working or Wget is not installed.")
         clean_exit(0)
