--- ./utils/exportfs/exportfs.c.orig	2020-10-21 08:15:26.000000000 -0500
+++ ./utils/exportfs/exportfs.c	2020-10-26 14:59:29.733988620 -0500
@@ -176,10 +176,10 @@
 		xlog(L_ERROR, "-r and -u are incompatible");
 		return 1;
 	}
-printf("point 1\n");
+/* printf("point 1\n"); */
 	if (!setup_state_path_names(progname, ETAB, ETABTMP, ETABLCK, &etab))
 		return 1;
-printf("point 2\n");
+/* printf("point 2\n"); */
 	if (optind == argc && ! f_all) {
 		if (force_flush) {
 			cache_flush(1);
@@ -193,7 +193,7 @@
 			return 0;
 		}
 	}
-printf("point 3\n");
+/* printf("point 3\n"); */
 
 	/*
 	 * Serialize things as best we can
