--- ./configure.orig	2008-03-20 06:20:01.000000000 -0500
+++ ./configure	2008-03-31 17:19:22.000000000 -0500
@@ -3933,7 +3933,7 @@
                     CPPFLAGS="-I${with_tcpw} ${CPPFLAGS}"
                 fi
             fi
-            LIBWRAP="-lwrap"
+            LIBWRAP="-lwrap -lnsl"
             LIBS="$LIBWRAP $LIBS"
             { echo "$as_me:$LINENO: checking for libwrap" >&5
 echo $ECHO_N "checking for libwrap... $ECHO_C" >&6; }
