--- ./binutils/readelf.c.orig	2024-01-28 18:00:00.000000000 -0600
+++ ./binutils/readelf.c	2024-01-29 13:46:17.699015450 -0600
@@ -13680,12 +13680,13 @@
       unsigned int vis = ELF_ST_VISIBILITY (psym->st_other);
 
       printf (" %-7s", get_symbol_visibility (vis));
-
+#if 0
       /* Check to see if any other bits in the st_other field are set.
 	 FIXME: Displaying this information here disrupts the layout
 	 of the table being generated.  */
       if (psym->st_other ^ vis)
 	printf (" [%s] ", get_symbol_other (filedata, psym->st_other ^ vis));
+#endif
     }
 
   bool is_special;
@@ -13770,6 +13771,16 @@
 		version_string);
     }
 
+#if 1
+    {
+      unsigned int vis = ELF_ST_VISIBILITY (psym->st_other);
+
+      /* Check to see if any other bits in the st_other field are set.  */
+      if (psym->st_other ^ vis)
+       printf (" \t[%s]", get_symbol_other (filedata, psym->st_other ^ vis));
+    }
+#endif
+
   putchar ('\n');
 
   if (ELF_ST_BIND (psym->st_info) == STB_LOCAL
