--- ./cimg.cc.orig	2008-02-22 09:51:22.000000000 -0600
+++ ./cimg.cc	2015-03-23 11:10:46.570421240 -0500
@@ -94,11 +94,11 @@
       ct == PNG_COLOR_TYPE_GRAY_ALPHA)
     png_set_gray_to_rgb(pngp);
 
-  alloc(pngp->width,pngp->height);
+  alloc(png_get_image_width(pngp, infp), png_get_image_height(pngp, infp));
   if (!ok) { fclose(f); return; }
   ok = 0;
 
-  for(i=0;i<pngp->height;i++) {
+  for(i=0;i<png_get_image_height(pngp, infp);i++) {
     png_read_row(pngp, (png_bytep) (&data[i*rowlen]), NULL);
   }
 
