--- gnuchess.SlackBuild	2006-04-04 03:47:10.000000000 +0300
+++ gnuchess.SlackBuild	2007-05-08 17:06:11.000000000 +0300
@@ -4,10 +4,10 @@
 
 VERGNUCHESS=5.07
 VERSJENG=11.2
-VEREBOARD=0.9.5
+VEREBOARD=1.0.3
 VERXBOARD=4.2.7
 ARCH=${ARCH:-i486}
-BUILD=${BUILD:-2}
+BUILD=${BUILD:-3}
 
 if [ "$ARCH" = "i386" ]; then
   SLKCFLAGS="-O2 -march=i386 -mcpu=i686"
@@ -27,6 +27,8 @@
 rm -rf gnuchess-$VERGNUCHESS
 tar xjvf $CWD/gnuchess-$VERGNUCHESS.tar.bz2
 cd gnuchess-$VERGNUCHESS
+# make gnuchess compile with gcc4
+zcat $CWD/gnuchess.gcc4_fix.diff.gz | patch -p1 --verbose --backup --suffix=.orig
 chown -R root:root .
 find . -perm 664 -exec chmod 644 {} \;
 find . -perm 600 -exec chmod 644 {} \;
@@ -93,33 +95,13 @@
 find . -perm 511 -exec chmod 755 {} \;
 find . -perm 711 -exec chmod 755 {} \;
 find . -perm 555 -exec chmod 755 {} \;
-CFLAGS="$SLKCFLAGS" \
 ./configure \
-  --prefix=/usr
+  --prefix=/usr --extra-flags=`echo $SLKCFLAGS | sed -e "y/ /:/"`
 make -j3
-#make install DESTDIR=$PKG
-install -d $PKG/usr/bin
-install -m 0755 eboard $PKG/usr/bin/eboard
-install -m 0755 eboard-config   $PKG/usr/bin/eboard-config
-install -m 0755 eboard-addtheme $PKG/usr/bin/eboard-addtheme
-install -d $PKG/usr/man/man1
-install -d $PKG/usr/man/man6
-install -m 0644 eboard.6 $PKG/usr/man/man6
-install -m 0644 eboard-addtheme.1 $PKG/usr/man/man1
-install -m 0644 eboard-config.1 $PKG/usr/man/man1
-install -d $PKG/usr/share/eboard
-install -m 0644 classic.png $PKG/usr/share/eboard/classic.png
-install -m 0644 ghouls.png $PKG/usr/share/eboard/ghouls.png
-install -m 0644 plastic.png $PKG/usr/share/eboard/plastic.png
-install -m 0644 eboard_themes.conf $PKG/usr/share/eboard/eboard_themes.conf
-install -m 0644 NAG.en.txt  $PKG/usr/share/eboard/NAG.en.txt
-( cd multilang
-  install -d $PKG/usr/share/eboard
-  install -m 0644 eboard.cs.dict eboard.de.dict eboard.es.dict eboard.it.dict eboard.pt_BR.dict $PKG/usr/share/eboard
-)
+make install DESTDIR=$PKG
 mkdir -p $PKG/usr/doc/eboard-$VEREBOARD
 cp -a \
-  AUTHORS COPYING README TODO \
+  AUTHORS COPYING README TODO Documentation/*.txt \
   $PKG/usr/doc/eboard-$VEREBOARD
 
 ### XBoard
@@ -140,6 +122,8 @@
 find . -perm 555 -exec chmod 755 {} \;
 # --infodir is fubar.
 zcat $CWD/xboard.infodir.diff.gz | patch -p1 --verbose --backup --suffix=.orig
+# Taken from Debian to fix CVE-2004-2552
+zcat $CWD/xboard.buffer_overflow_fix.diff.gz | patch -p1 --verbose --backup --suffix=.orig || exit 1
 CFLAGS="$SLKCFLAGS" \
 ./configure \
    --prefix=/usr \
@@ -153,6 +137,7 @@
 install -c ./pxboard $PKG/usr/bin/pxboard
 install -c -m 644 ./xboard.man $PKG/usr/man/man6/xboard.6
 install -c -m 644 ./zic2xpm.man $PKG/usr/man/man6/zic2xpm.6
+mkdir -p $PKG/usr/info
 install -c -m 644 xboard.info $PKG/usr/info/xboard.info
 mkdir -p $PKG/usr/doc/xboard-$VERXBOARD
 cp -a \
@@ -165,8 +150,9 @@
   find . | xargs file | grep "shared object" | grep ELF | cut -f 1 -d : | xargs strip --strip-unneeded 2> /dev/null
 )
 gzip -9 $PKG/usr/man/man?/*.?
+gzip -9 $PKG/usr/info/*.info
 mkdir -p $PKG/usr/share/pixmaps
-cat $CWD/chess.png > $PKG/usr/share/pixmaps/chess.png
+cp $CWD/chess.png $PKG/usr/share/pixmaps/chess.png
 mkdir -p $PKG/usr/share/applications
 cat $CWD/eboard.desktop > $PKG/usr/share/applications/eboard.desktop
 cat $CWD/xboard.desktop > $PKG/usr/share/applications/xboard.desktop
