--- ./src/input.c.orig	2010-02-19 21:34:10.000000000 -0600
+++ ./src/input.c	2010-02-19 21:36:33.000000000 -0600
@@ -127,7 +127,7 @@
 	      (RealGameCnt+1)/2 + 1 );
     }
     pthread_mutex_lock(&input_mutex);
-    getline(prompt);
+    chess_getline(prompt);
     input_status = INPUT_AVAILABLE;
     pthread_cond_signal(&input_cond);
     pthread_mutex_unlock(&input_mutex);
@@ -173,13 +173,13 @@
 {
 #ifdef HAVE_LIBREADLINE
   if (isatty(STDIN_FILENO)) {
-    getline = getline_readline;
+    chess_getline = getline_readline;
     using_history();
   } else {
-    getline = getline_standard;
+    chess_getline = getline_standard;
   }
 #else
-  getline = getline_standard;
+  chess_getline = getline_standard;
 #endif
   /* Start input thread */
   pthread_create(&input_thread, NULL, input_func, NULL);
--- ./src/common.h.orig	2003-06-30 06:28:38.000000000 -0500
+++ ./src/common.h	2010-02-19 21:35:33.000000000 -0600
@@ -745,7 +745,7 @@
  * Input routine, initialized to one of the specific
  * input routines. The given argument is the prompt.
  */
-void (*getline) (char *);
+void (*chess_getline) (char *);
 
 #define MAXSTR 128
 extern char inputstr[MAXSTR];
