--- ./xboard.conf.in.orig	2013-02-20 23:59:30.000000000 -0600
+++ ./xboard.conf.in	2013-03-19 21:39:17.377182426 -0500
@@ -67,12 +67,12 @@
 ;
 ; Engines & adjudicatons in engine-engine games
 ;
--firstChessProgram fairymax
--secondChessProgram fairymax
--firstChessProgramNames {fairymax
+-firstChessProgram gnuchess
+-secondChessProgram gnuchess
+-firstChessProgramNames {"GNU Chess" -fcp gnuchess
+fairymax
 "Fruit 2.1" -fcp fruit -fUCI
 "Crafty" -fcp crafty
-"GNU Chess" -fcp gnuchess
 }
 -niceEngines 0
 -polyglotDir ""
