--- ./xboard.conf.orig	2015-03-20 13:32:54.035406604 -0500
+++ ./xboard.conf	2015-03-20 13:35:24.433417456 -0500
@@ -73,11 +73,11 @@
 ;
 ; Engines & adjudicatons in engine-engine games
 ;
--firstChessProgram fairymax
--firstChessProgramNames {fairymax
+-firstChessProgram gnuchess
+-firstChessProgramNames {"GNU Chess" -fcp gnuchess
+fairymax
 "Fruit 2.1" -fcp fruit -fUCI
 "Crafty" -fcp crafty
-"GNU Chess" -fcp gnuchess
 }
 -niceEngines 0
 -polyglotDir ""
