--- ./Makefile.in.orig	2003-11-28 13:12:57.000000000 -0800
+++ ./Makefile.in	2004-06-26 15:47:07.000000000 -0700
@@ -21,7 +21,7 @@
 # Yuck; the autoconf generated configure script sets an infodir variable
 # and provides a command-line flag to change it, but the default is 
 # /usr/info instead of /usr/share/info.
-infodir = $(prefix)/share/info
+infodir = /usr/info
 
 CC = @CC@
 CFLAGS = @CFLAGS@ @CPPFLAGS@ -DINFODIR=\"$(infodir)\"
