From 476b5a62fcda711ce2d7412a310b8182cca6940e Mon Sep 17 00:00:00 2001
From: Detlef Graef <detlef.graef@yahoo.de>
Date: Fri, 26 Apr 2019 20:59:54 +0200
Subject: [PATCH] Fix for issue #77. Segfault when build with gmime-3

---
 pan/gui/pan.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pan/gui/pan.cc b/pan/gui/pan.cc
index 8d28fdb..fb3a696 100644
--- a/pan/gui/pan.cc
+++ b/pan/gui/pan.cc
@@ -906,6 +906,7 @@ main (int argc, char *argv[])
   g_thread_init (0);
 #endif
 #ifdef HAVE_GMIME_30
+  g_mime_init ();
 #else
   g_mime_init (GMIME_ENABLE_RFC2047_WORKAROUNDS);
 #endif  
