diff -uri icmpinfo-1.11/defs.h icmpinfo-1.11.fixed/defs.h
--- icmpinfo-1.11/defs.h	1994-05-26 15:36:40.000000000 +0200
+++ icmpinfo-1.11.fixed/defs.h	2008-04-07 20:13:09.000000000 +0200
@@ -3,10 +3,11 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h> /* exit */
 #include <errno.h>
 extern int	errno;
 
-#include <sys/time.h>
+#include <time.h>
 #include <sys/param.h>
 #include <sys/socket.h>
 #include <sys/file.h>
diff -uri icmpinfo-1.11/err.c icmpinfo-1.11.fixed/err.c
--- icmpinfo-1.11/err.c	1994-04-22 19:44:48.000000000 +0200
+++ icmpinfo-1.11.fixed/err.c	2008-04-07 20:13:47.000000000 +0200
@@ -1,4 +1,5 @@
 #include <stdio.h>
+#include <stdlib.h> /* exit */
 
 extern char *pname;
 
diff -uri icmpinfo-1.11/pid.c icmpinfo-1.11.fixed/pid.c
--- icmpinfo-1.11/pid.c	2008-04-07 20:16:58.000000000 +0200
+++ icmpinfo-1.11.fixed/pid.c	2008-04-07 20:14:12.000000000 +0200
@@ -1,4 +1,5 @@
 #include <stdio.h>
+#include <stdlib.h> /* exit */
 #include <signal.h>
 
 #define PIDFILE                 "/var/run/icmpinfo.pid"
diff -uri icmpinfo-1.11/print.c icmpinfo-1.11.fixed/print.c
--- icmpinfo-1.11/print.c	2008-04-07 20:16:58.000000000 +0200
+++ icmpinfo-1.11.fixed/print.c	2008-04-07 20:15:23.000000000 +0200
@@ -9,6 +9,7 @@
           it should be ok, now */
 
 #include <string.h>
+#include <stddef.h> /* offsetof */
 #include	"defs.h"
 
 #ifndef ANSI_OFFSETOF
