diff -Nur cgmanager-0.39.orig/Makefile.in cgmanager-0.39/Makefile.in
--- cgmanager-0.39.orig/Makefile.in	2015-09-03 17:59:13.000000000 -0500
+++ cgmanager-0.39/Makefile.in	2015-11-19 19:48:33.407492836 -0600
@@ -503,15 +503,6 @@
 libcgmanager_la_LIBADD = \
 	$(DBUS_LIBS) $(NIH_LIBS) $(NIH_DBUS_LIBS)
 
-pam_LTLIBRARIES = pam_cgm.la
-pam_cgm_la_SOURCES = pam/pam_cgm.c pam/cgmanager.c pam/cgmanager.h
-pam_cgm_la_CFLAGS = $(AM_CFLAGS)
-pam_cgm_la_LIBADD = $(AM_LIBS) $(PAM_LIBS) -L$(top_srcdir) -lcgmanager
-pam_cgm_la_LDFLAGS = $(AM_LDFLAGS) -module -avoid-version -shared
-pam_cgm_la_DEPENDENCIES = libcgmanager.la
-all: $(BUILT_SOURCES) config.h
-	$(MAKE) $(AM_MAKEFLAGS) all-recursive
-
 .SUFFIXES:
 .SUFFIXES: .c .lo .o .obj
 am--refresh: Makefile
diff -Nur cgmanager-0.39.orig/configure cgmanager-0.39/configure
--- cgmanager-0.39.orig/configure	2015-09-03 17:59:14.000000000 -0500
+++ cgmanager-0.39/configure	2015-11-19 19:48:08.425776320 -0600
@@ -13401,53 +13401,6 @@
 
 
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for pam_authenticate in -lpam" >&5
-$as_echo_n "checking for pam_authenticate in -lpam... " >&6; }
-if ${ac_cv_lib_pam_pam_authenticate+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-  ac_check_lib_save_LIBS=$LIBS
-LIBS="-lpam  $LIBS"
-cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-/* Override any GCC internal prototype to avoid an error.
-   Use char because int might match the return type of a GCC
-   builtin and then its argument prototype would still apply.  */
-#ifdef __cplusplus
-extern "C"
-#endif
-char pam_authenticate ();
-int
-main ()
-{
-return pam_authenticate ();
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-  ac_cv_lib_pam_pam_authenticate=yes
-else
-  ac_cv_lib_pam_pam_authenticate=no
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-LIBS=$ac_check_lib_save_LIBS
-fi
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_pam_pam_authenticate" >&5
-$as_echo "$ac_cv_lib_pam_pam_authenticate" >&6; }
-if test "x$ac_cv_lib_pam_pam_authenticate" = xyes; then :
-  PAM_LIBS="-lpam"
-else
-  as_fn_error $? "*** libpam not found." "$LINENO" 5
-
-fi
-
-
-
-
-
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
 # tests run on this system so they can be shared between configure
