--- ./Makefile.orig	2010-04-04 02:36:49.000000000 -0500
+++ ./Makefile	2011-02-25 13:45:41.443999917 -0600
@@ -19,10 +19,10 @@
 # can be found in /usr/share/common-licenses/GPL-3 file.
 
 DESTDIR =
-PREFIX = /usr/local
-SBINDIR = $(PREFIX)/sbin
-DOCDIR = $(PREFIX)/share/doc
-MANDIR = $(PREFIX)/share/man
+PREFIX = /usr
+SBINDIR = /sbin
+DOCDIR = $(PREFIX)/doc
+MANDIR = $(PREFIX)/man
 
 #OPTFLAGS = -O2 -fomit-frame-pointer -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
 OPTFLAGS = -O2 -fomit-frame-pointer $(shell getconf LFS_CFLAGS)
