--- ./Makefile.in.orig	2015-12-28 15:31:15.000000000 -0600
+++ ./Makefile.in	2018-08-16 15:04:37.525329502 -0500
@@ -1639,7 +1639,7 @@
 DISTCLEANFILES = tool-versions.txt
 
 # "tests" is the gnulib unit test dir.
-SUBDIRS = gl tests build-aux lib find xargs locate doc po m4
+SUBDIRS = gl tests build-aux lib find xargs doc po m4
 ACLOCAL_AMFLAGS = -I gl/m4 -I m4
 TESTFILE_SUFFIXES = .exp .xo .xe .xi
 all: config.h
--- ./Makefile.am.orig	2015-12-28 15:10:23.000000000 -0600
+++ ./Makefile.am	2018-08-16 15:04:25.998328424 -0500
@@ -9,7 +9,7 @@
 
 
 # "tests" is the gnulib unit test dir.
-SUBDIRS = gl tests build-aux lib find xargs locate doc po m4
+SUBDIRS = gl tests build-aux lib find xargs doc po m4
 
 ACLOCAL_AMFLAGS = -I gl/m4 -I m4
 
