--- ./configure.ac.orig	2019-06-23 04:00:40.000000000 -0500
+++ ./configure.ac	2019-08-30 13:22:48.133114947 -0500
@@ -94,7 +94,6 @@
 
 dnl for gnulib
 gl_EARLY
-AC_PROG_LN_S
 AC_PROG_INSTALL
 AC_CHECK_TOOLS([AR], [ar])
 AC_CHECK_TOOLS([RANLIB], [ranlib], [:])
@@ -309,8 +308,6 @@
 gl/Makefile
 gl/lib/Makefile
 lib/Makefile
-locate/Makefile
-locate/testsuite/Makefile
 m4/Makefile
 po/Makefile.in
 po/Makefile
--- ./doc/find.texi.orig	2019-08-30 13:21:12.626110869 -0500
+++ ./doc/find.texi	2019-08-30 13:22:56.395115300 -0500
@@ -7,7 +7,6 @@
 @c %**end of header
 
 @include version.texi
-@include dblocation.texi
 
 @iftex
 @finalout
--- ./doc/Makefile.am.orig	2019-06-23 04:00:40.000000000 -0500
+++ ./doc/Makefile.am	2019-08-30 13:21:04.629110527 -0500
@@ -16,11 +16,9 @@
 
 info_TEXINFOS = find.texi find-maint.texi
 find_TEXINFOS = perm.texi parse-datetime.texi regexprops.texi fdl.texi
-BUILT_SOURCES = dblocation.texi
-nodist_find_TEXINFOS = dblocation.texi
 find_maint_TEXINFOS = fdl.texi
 MOSTLYCLEANFILES = find.cps
-CLEANFILES = find.txt find_mono.html findutils.texi_html_node.tar.gz dblocation.texi \
+CLEANFILES = find.txt find_mono.html findutils.texi_html_node.tar.gz \
 	find_mono.html findutils.texi_html_node.tar.gz \
 	find-info.tar.gz find.texi find.texi.tar.gz \
 	find.txt  tmp-doc-install find_mono.html.gz
@@ -72,10 +70,6 @@
 findutils.texi_html_node.tar.gz: find.html
 	tar zcf $@ $<
 
-
-dblocation.texi: ../locate/dblocation.texi
-	$(LN_S) ../locate/dblocation.texi $@
-
 find-info.tar.gz:
 	$(MKDIR_P) tmp-doc-install/info
 	$(MAKE) $(AM_MAKEFLAGS) \
--- ./Makefile.in.orig	2019-08-29 14:15:43.000000000 -0500
+++ ./Makefile.in	2019-08-30 13:18:16.959103368 -0500
@@ -1955,11 +1955,11 @@
 DISTCLEANFILES = tool-versions.txt
 
 # "gnulib-tests" is the gnulib unit test dir.
-SUBDIRS = gl build-aux lib find xargs locate doc po m4 gnulib-tests
+SUBDIRS = gl build-aux lib find xargs doc po m4 gnulib-tests
 ALL_RECURSIVE_TARGETS = check-root
 ACLOCAL_AMFLAGS = -I gl/m4 -I m4
 TESTFILE_SUFFIXES = .exp .xo .xe .xi
-built_programs = find oldfind xargs frcode locate updatedb
+built_programs = find oldfind xargs
 
 # Indirections required so that we'll still be able to know the
 # complete list of our tests even if the user overrides TESTS
--- ./Makefile.am.orig	2019-06-23 04:00:40.000000000 -0500
+++ ./Makefile.am	2019-08-30 13:18:16.961103368 -0500
@@ -35,7 +35,7 @@
 
 
 # "gnulib-tests" is the gnulib unit test dir.
-SUBDIRS = gl build-aux lib find xargs locate doc po m4 gnulib-tests
+SUBDIRS = gl build-aux lib find xargs doc po m4 gnulib-tests
 
 ALL_RECURSIVE_TARGETS =
 
