--- ./tests/CMakeLists.txt.orig	2014-08-03 16:14:49.000000000 -0500
+++ ./tests/CMakeLists.txt	2015-10-24 12:55:55.767426554 -0500
@@ -6,7 +6,6 @@
         include_directories(${QT_INCLUDES} ${KDE4_INCLUDES} ${CMAKE_CURRENT_BINARY_DIR} ${X11_Xrandr_INCLUDE_PATH})
         kde4_add_unit_test(${_testname} ${_testname}.cpp ${_testname}.moc)
         target_link_libraries(${_testname} ${QT_QTTEST_LIBRARY} ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} kscreen)
-        add_test(${_testname} ${CMAKE_CURRENT_BINARY_DIR}/${_testname})
     endforeach(_testname)
 endmacro(KSCREEN_TESTS)
 
