--- ./localedata/locales/en_US.orig	2019-01-31 10:45:36.000000000 -0600
+++ ./localedata/locales/en_US	2019-01-31 22:35:02.033009166 -0600
@@ -118,7 +118,10 @@
 t_fmt_ampm "%I:%M:%S %p"
 %
 % Appropriate date and time representation for date(1)
-date_fmt "%a %d %b %Y %r %Z"
+% [ Slackware editor's note - no thank you, US users
+% of UNIX-like systems expect a 24 hour clock, just
+% like users in the vast majority of other locales. ]
+%date_fmt "%a %d %b %Y %r %Z"
 %
 % Strings for AM/PM
 %
