--- ./localedata/Makefile.orig	2018-02-04 23:06:03.700212576 -0600
+++ ./localedata/Makefile	2018-02-04 23:07:10.102218076 -0600
@@ -223,6 +223,7 @@
 	echo -n '...'; \
 	input=`echo $$locale | sed 's/\([^.]*\)[^@]*\(.*\)/\1\2/'`; \
 	$(LOCALEDEF) $$flags --alias-file=../intl/locale.alias \
+		     --no-archive \
 		     -i locales/$$input -f charmaps/$$charset \
 		     $(addprefix --prefix=,$(install_root)) $$locale \
 	&& echo ' done'; \
