--- a/inetd.c	2013-11-14 10:34:46.604127237 +0100
+++ b/inetd.c	2013-11-14 10:35:56.168907312 +0100
@@ -165,6 +165,7 @@ static char xtraid[] = "$Slackware: inet
 #include <stdlib.h>
 #include <unistd.h>
 #include <string.h>
+#include <time.h>
 /* #include <login_cap.h> */
 #include <rpc/rpc.h>
 #include <rpc/pmap_clnt.h>
@@ -2096,7 +2097,7 @@ daytime_stream(s, sep)		/* Return human-
 	struct servtab *sep;
 {
 	char buffer[256];
-	time_t time(), clock;
+	time_t clock;
 
 	clock = time(NULL);
 
@@ -2111,7 +2112,7 @@ daytime_dg(s, sep)		/* Return human-read
 	struct servtab *sep;
 {
 	char buffer[256];
-	time_t time(), clock;
+	time_t clock;
 	/* struct sockaddr_storage ss; */
 	struct sockaddr sa;
 	int size;
