--- ./kstars/CMakeLists.txt.orig	2023-05-31 03:23:38.000000000 -0500
+++ ./kstars/CMakeLists.txt	2023-06-10 15:22:26.822136184 -0500
@@ -1338,6 +1338,11 @@
 
 endif(INDI_FOUND)
 
+# Force link with GSL:
+find_package(GSL REQUIRED)
+include_directories(${GSL_INCLUDE_DIRS})
+target_link_libraries(KStarsLib ${GSL_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} KF5::Notifications)
+
 if(WCSLIB_FOUND)
     target_link_libraries(KStarsLib ${WCSLIB_LIBRARIES})
     if (BUILD_KSTARS_LITE)
