--- ./bfd/bfd-in2.h.orig	2013-02-27 14:28:03.000000000 -0600
+++ ./bfd/bfd-in2.h	2013-04-02 15:57:44.847202021 -0500
@@ -32,11 +32,6 @@
 #ifndef __BFD_H_SEEN__
 #define __BFD_H_SEEN__
 
-/* PR 14072: Ensure that config.h is included first.  */
-#if !defined PACKAGE && !defined PACKAGE_VERSION
-#error config.h must be included before this header
-#endif
-
 #ifdef __cplusplus
 extern "C" {
 #endif
--- ./bfd/bfd-in.h.orig	2013-02-27 14:28:03.000000000 -0600
+++ ./bfd/bfd-in.h	2013-04-02 15:57:44.846202021 -0500
@@ -25,11 +25,6 @@
 #ifndef __BFD_H_SEEN__
 #define __BFD_H_SEEN__
 
-/* PR 14072: Ensure that config.h is included first.  */
-#if !defined PACKAGE && !defined PACKAGE_VERSION
-#error config.h must be included before this header
-#endif
-
 #ifdef __cplusplus
 extern "C" {
 #endif
