--- ./src/bluetooth.conf.orig	2010-10-04 06:41:01.000000000 -0500
+++ ./src/bluetooth.conf	2010-11-07 19:50:47.229999322 -0600
@@ -11,13 +11,18 @@
     <allow own="org.bluez"/>
     <allow send_destination="org.bluez"/>
     <allow send_interface="org.bluez.Agent"/>
+    <allow send_interface="org.bluez.Adapter"/>
     <allow send_interface="org.bluez.HandsfreeAgent"/>
     <allow send_interface="org.bluez.MediaEndpoint"/>
     <allow send_interface="org.bluez.Watcher"/>
   </policy>
 
-  <policy at_console="true">
+  <policy group="plugdev">
     <allow send_destination="org.bluez"/>
+    <allow send_interface="org.bluez.Agent"/>
+    <allow send_interface="org.bluez.Adapter"/>
+    <allow send_destination="org.bluez"
+           send_interface="org.freedesktop.DBus.Introspectable"/>
   </policy>
 
   <!-- allow users of lp group (printing subsystem) to 
