From b2c3df6015d2ab1b57e2532b05298803b28f2902 Mon Sep 17 00:00:00 2001
From: Peter de Ridder <peter@xfce.org>
Date: Wed, 4 Mar 2015 11:03:28 +0100
Subject: [PATCH] Make sure icon == NULL if no icon is found

---
 dialogs/appearance-settings/main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dialogs/appearance-settings/main.c b/dialogs/appearance-settings/main.c
index 3eeb348..6532182 100644
--- a/dialogs/appearance-settings/main.c
+++ b/dialogs/appearance-settings/main.c
@@ -720,6 +720,7 @@ appearance_settings_load_icon_themes (preview_data *pd)
 
                     for (p = 0; p < 4; p++)
                     {
+                        icon = NULL;
                         if (gtk_icon_theme_has_icon (icon_theme, preview_icons[p]))
                             icon = gtk_icon_theme_load_icon (icon_theme, preview_icons[p], 16, 0, NULL);
                         else if (gtk_icon_theme_has_icon (icon_theme, "image-missing"))
-- 
2.3.1

