--- ./meson.build.orig	2025-03-18 03:38:44.000000000 -0500
+++ ./meson.build	2025-03-19 16:40:29.995944908 -0500
@@ -13,6 +13,7 @@
 cc = meson.get_compiler('c')
 gkr_prefix = get_option('prefix')
 gkr_bindir = gkr_prefix / get_option('bindir')
+gkr_libdir = gkr_prefix / get_option('pamdir')
 gkr_pkglibdir = gkr_prefix / get_option('libdir') / 'gnome-keyring' / 'devel'
 config_h_inc = include_directories('.')
 po_dir = meson.current_source_dir() / 'po'
--- ./pam/meson.build.orig	2025-03-18 03:38:44.000000000 -0500
+++ ./pam/meson.build	2025-03-19 16:42:25.393938525 -0500
@@ -33,7 +33,7 @@
   link_depends: pam_gkr_symbolmap,
   include_directories: config_h_inc,
   install: true,
-  install_dir: get_option('libdir') / 'security',
+  install_dir: get_option('pamdir'),
 )
 
 libpam_gkr_dep = declare_dependency(
--- ./meson_options.txt.orig	2025-03-18 03:38:44.000000000 -0500
+++ ./meson_options.txt	2025-03-19 16:41:30.823941543 -0500
@@ -45,3 +45,9 @@
   value: true,
   description: 'Build manpage',
 )
+
+option('pamdir',
+  type: 'string',
+  value: '',
+  description: 'Pam Library directory (e.g., lib or lib64)',
+)
