--- ./sysstat.in.orig	2006-07-05 10:58:00.000000000 -0500
+++ ./sysstat.in	2006-08-14 22:03:13.000000000 -0500
@@ -22,21 +22,21 @@
 
 RETVAL=0
 # Remove flag indicating that sadc was successfully launched
-rm -f /tmp/sysstat.run
+rm -f /var/run/sysstat.run
 
 # See how we were called.
 case "$1" in
   start)
         echo -n "Calling the system activity data collector (sadc): "
-        SU SA_LIB_DIR/sadc -F -L - && touch /tmp/sysstat.run QUOTE
+        SU SA_LIB_DIR/sadc -F -L - && touch /var/run/sysstat.run QUOTE
 
 # Try to guess if sadc was successfully launched. The difficulty
 # here is that the exit code is lost when the above command is
 # run via "su foo -c ..."
-	if [ ! -f /tmp/sysstat.run ]; then
+	if [ ! -f /var/run/sysstat.run ]; then
 		RETVAL=1
 	else
-		rm -f /tmp/sysstat.run
+		rm -f /var/run/sysstat.run
 	fi
         echo
         ;;
