--- ./Makefile.orig	2006-07-05 10:46:04.000000000 -0500
+++ ./Makefile	2006-08-14 21:40:38.000000000 -0500
@@ -34,7 +34,7 @@
 MAN8_DIR = $(MAN_DIR)/man8
 DOC_DIR = $(PREFIX)/doc/sysstat-$(VERSION)
 NLS_DIR = $(PREFIX)/share/locale
-SYSCONFIG_DIR = /etc/sysconfig
+SYSCONFIG_DIR = /etc/sysstat
 
 # Compiler flags
 CFLAGS = -Wall -Wstrict-prototypes -pipe -O2 -fno-strength-reduce
--- ./sa2.in.orig	2006-06-09 03:17:26.000000000 -0500
+++ ./sa2.in	2006-08-14 21:38:35.000000000 -0500
@@ -7,7 +7,7 @@
 #   make history configurable
 #
 HISTORY=7
-[ -r /etc/sysconfig/sysstat ] && . /etc/sysconfig/sysstat
+[ -r /etc/sysstat/sysstat ] && . /etc/sysstat/sysstat
 [ ${HISTORY} -gt 28 ] && HISTORY=28
 S_TIME_FORMAT=ISO ; export S_TIME_FORMAT
 umask 0022
--- ./README.orig	2006-07-05 10:51:54.000000000 -0500
+++ ./README	2006-08-14 21:39:27.000000000 -0500
@@ -91,8 +91,8 @@
 ${PREFIX}/doc/sysstat-x.y.z/*
 /var/log/sa
 ${INIT_DIR}/sysstat
-/etc/sysconfig/sysstat
-/etc/sysconfig/sysstat.ioconf
+/etc/sysstat/sysstat
+/etc/sysstat/sysstat.ioconf
 /etc/cron.d/sysstat
 /etc/rc.d/rc.sysstat		for [SLACKWARE]
 ${RC_DIR}/rc2.d/S03sysstat
--- ./ioconf.h.orig	2006-07-09 03:19:06.000000000 -0500
+++ ./ioconf.h	2006-08-14 21:36:48.000000000 -0500
@@ -20,7 +20,7 @@
 #define MAX_BLKDEV	255
 #endif
 
-#define IOCONF	"/etc/sysconfig/sysstat.ioconf"
+#define IOCONF	"/etc/sysstat/sysstat.ioconf"
 
 #define K_NODEV	"nodev"
 
