--- ./build/Configure.sh.orig	2006-06-09 03:21:27.000000000 -0500
+++ ./build/Configure.sh	2006-08-14 21:54:37.000000000 -0500
@@ -10,10 +10,10 @@
 
 # Installation directory
 
-PREFIX=`${ASK} 'Installation directory:' "/usr/local" "prefix"`
+PREFIX=`${ASK} 'Installation directory:' "/usr" "prefix"`
 if [ ! -d ${PREFIX} ]; then
-	echo "WARNING: Directory ${PREFIX} not found: Using default (/usr/local)."
-	PREFIX=/usr/local
+	echo "WARNING: Directory ${PREFIX} not found: Using default (/usr)."
+	PREFIX=/usr
 fi
 
 # sadc directory
--- ./build/CONFIG.def.orig	2006-06-09 03:20:56.000000000 -0500
+++ ./build/CONFIG.def	2006-08-14 21:54:19.000000000 -0500
@@ -2,8 +2,8 @@
 # (C) 2000-2006 Sebastien GODARD (sysstat <at> wanadoo.fr)
 
 # Directories
-PREFIX = /usr/local
-SA_LIB_DIR = /usr/local/lib/sa
+PREFIX = /usr
+SA_LIB_DIR = /usr/lib/sa
 SADC_PATH = ${SA_LIB_DIR}/sadc
 SA_DIR = /var/log/sa
 # Do not use ${PREFIX}/share/man by default: not necessarily in man path!
--- ./build/CONFIG.orig	2006-07-09 04:47:17.000000000 -0500
+++ ./build/CONFIG	2006-08-14 21:54:10.000000000 -0500
@@ -2,8 +2,8 @@
 # (C) 2000-2006 Sebastien GODARD (sysstat <at> wanadoo.fr)
 
 # Directories
-PREFIX = /usr/local
-SA_LIB_DIR = /usr/local/lib/sa
+PREFIX = /usr
+SA_LIB_DIR = /usr/lib/sa
 SADC_PATH = ${SA_LIB_DIR}/sadc
 SA_DIR = /var/log/sa
 # Do not use ${PREFIX}/share/man by default: not necessarily in man path!
